// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/markdown/v1/mode_html_to_markdown.schema.js
/**
 * Markdown Node - Version 1 - Zod Schema
 * Discriminator: mode=htmlToMarkdown
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      mode: z.literal('htmlToMarkdown').default('htmlToMarkdown'),
      html: stringOrExpression.optional(),
      destinationKey: stringOrExpression.optional(),
      options: z.object({ bulletMarker: stringOrExpression.optional(), codeFence: stringOrExpression.optional(), emDelimiter: stringOrExpression.optional(), globalEscape: z.unknown().optional(), ignore: stringOrExpression.optional(), keepDataImages: booleanOrExpression.optional(), lineStartEscape: z.unknown().optional(), maxConsecutiveNewlines: numberOrExpression.optional(), useLinkReferenceDefinitions: booleanOrExpression.optional(), strongDelimiter: stringOrExpression.optional(), codeBlockStyle: z.union([z.literal('fence'), z.literal('indented'), expressionSchema]).optional(), textReplace: z.unknown().optional(), blockElements: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
