// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/syncroMsp/v1/resource_ticket/operation_get_all.ts
/**
 * SyncroMSP Node - Version 1
 * Discriminator: resource=ticket, operation=getAll
 */


interface Credentials {
  syncroMspApi: CredentialReference;
}

/** Retrieve many customers */
export type SyncroMspV1TicketGetAllParams = {
  resource: 'ticket';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 25
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Search query, it can be anything related to ticket data like user etc
     */
    query?: string | Expression<string> | PlaceholderValue;
    /** Status
     * @default New
     */
    status?: 'Customer Reply' | 'In Progress' | 'New' | 'Resolved' | 'Scheduled' | 'Waiting for Parts' | 'Waiting on Customer' | Expression<string>;
  };
};

export type SyncroMspV1TicketGetAllNode = {
  type: 'n8n-nodes-base.syncroMsp';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SyncroMspV1TicketGetAllParams>;
};
