// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/actionNetwork/v1/resource_signature/operation_create.schema.js
/**
 * Action Network Node - Version 1 - Zod Schema
 * Discriminator: resource=signature, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('signature'),
      operation: z.literal('create').default('create'),
      petitionId: stringOrExpression.optional(),
      personId: stringOrExpression.optional(),
      simple: booleanOrExpression.optional(),
      additionalFields: z.object({ comments: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
