// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleCalendar/v13/resource_event/operation_delete.ts
/**
 * Google Calendar Node - Version 1.3
 * Discriminator: resource=event, operation=delete
 */


interface Credentials {
  googleCalendarOAuth2Api: CredentialReference;
}

/** Delete an event */
export type GoogleCalendarV13EventDeleteParams = {
  resource: 'event';
  operation: 'delete';
/**
 * Google Calendar to operate on
 * @default {"mode":"list","value":""}
 */
    calendar?: { __rl: true; mode: 'list' | 'id'; value: string; cachedResultName?: string };
/**
 * Event ID
 */
    eventId?: string | Expression<string> | PlaceholderValue;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Whether to send notifications about the creation of the new event
     */
    sendUpdates?: 'all' | 'externalOnly' | 'none' | Expression<string>;
  };
};

export type GoogleCalendarV13EventDeleteOutput = {
  success?: boolean;
};

export type GoogleCalendarV13EventDeleteNode = {
  type: 'n8n-nodes-base.googleCalendar';
  version: 1.3;
  credentials?: Credentials;
  config: NodeConfig<GoogleCalendarV13EventDeleteParams>;
  output?: Items<GoogleCalendarV13EventDeleteOutput>;
};
