// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/currents/v1/resource_signature/operation_generate.ts
/**
 * Currents Node - Version 1
 * Discriminator: resource=signature, operation=generate
 */


interface Credentials {
  currentsApi: CredentialReference;
}

/** Generate unique test signatures */
export type CurrentsV1SignatureGenerateParams = {
  resource: 'signature';
  operation: 'generate';
/**
 * The Currents project
 * @default {"mode":"list","value":""}
 */
    projectId?: { __rl: true; mode: 'list' | 'id'; value: string; cachedResultName?: string };
/**
 * The complete path to the spec file
 */
    specFilePath?: string | Expression<string> | PlaceholderValue;
/**
 * The test title. For nested describe blocks, use " &gt; " as separator (e.g., "Login &gt; should login with valid credentials").
 */
    testTitle?: string | Expression<string> | PlaceholderValue;
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type CurrentsV1SignatureGenerateNode = {
  type: 'n8n-nodes-base.currents';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<CurrentsV1SignatureGenerateParams>;
};
