// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftTeams/v1/resource_task/operation_get_all.ts
/**
 * Microsoft Teams Node - Version 1
 * Discriminator: resource=task, operation=getAll
 */


interface Credentials {
  microsoftTeamsOAuth2Api: CredentialReference;
}

/** Get many channels */
export type MicrosoftTeamsV1TaskGetAllParams = {
  resource: 'task';
  operation: 'getAll';
/**
 * Group Source
 * @default all
 */
    groupSource?: 'all' | 'mine' | Expression<string>;
/**
 * Tasks For
 * @default member
 */
    tasksFor?: 'member' | 'plan' | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    groupId?: string | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 * @displayOptions.show { tasksFor: ["member"] }
 */
    memberId?: string | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 * @displayOptions.show { tasksFor: ["plan"] }
 */
    planId?: string | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
};

export type MicrosoftTeamsV1TaskGetAllNode = {
  type: 'n8n-nodes-base.microsoftTeams';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MicrosoftTeamsV1TaskGetAllParams>;
};
