// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftTeams/v1/resource_task/operation_create.schema.js
/**
 * Microsoft Teams Node - Version 1 - Zod Schema
 * Discriminator: resource=task, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('task'),
      operation: z.literal('create').default('create'),
      groupSource: z.union([z.literal('all'), z.literal('mine'), expressionSchema]).optional(),
      groupId: stringOrExpression.optional(),
      planId: stringOrExpression.optional(),
      bucketId: stringOrExpression.optional(),
      title: stringOrExpression.optional(),
      additionalFields: z.object({ assignedTo: stringOrExpression.optional(), dueDateTime: stringOrExpression.optional(), labels: z.array(z.string()).optional(), percentComplete: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
