// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/philipsHue/v1/resource_light/operation_update.schema.js
/**
 * Philips Hue Node - Version 1 - Zod Schema
 * Discriminator: resource=light, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('light').default('light'),
      operation: z.literal('update').default('update'),
      lightId: stringOrExpression.optional(),
      on: booleanOrExpression.optional(),
      additionalFields: z.object({ alert: z.union([z.literal('none'), z.literal('select'), z.literal('lselect'), expressionSchema]).optional(), bri: numberOrExpression.optional(), bri_inc: numberOrExpression.optional(), ct: numberOrExpression.optional(), ct_inc: numberOrExpression.optional(), xy: stringOrExpression.optional(), xy_inc: stringOrExpression.optional(), effect: z.union([z.literal('none'), z.literal('colorloop'), expressionSchema]).optional(), hue: numberOrExpression.optional(), hue_inc: numberOrExpression.optional(), sat: numberOrExpression.optional(), sat_inc: numberOrExpression.optional(), transitiontime: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
