// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleDrive/v3/resource_drive/operation_create.ts
/**
 * Google Drive Node - Version 3
 * Discriminator: resource=drive, operation=create
 */


interface Credentials {
  googleApi: CredentialReference;
  googleDriveOAuth2Api: CredentialReference;
}

/** Create a shared drive */
export type GoogleDriveV3DriveCreateParams = {
  resource: 'drive';
  operation: 'create';
  authentication?: 'oAuth2' | 'serviceAccount' | Expression<string>;
/**
 * The name of the shared drive to create
 */
    name?: string | Expression<string> | PlaceholderValue;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Capabilities
     * @default {}
     */
    capabilities?: {
    /** Whether the current user can add children to folders in this shared drive
     * @default false
     */
    canAddChildren?: boolean | Expression<boolean>;
    /** Whether the current user can change the copyRequiresWriterPermission restriction of this shared drive
     * @default false
     */
    canChangeCopyRequiresWriterPermissionRestriction?: boolean | Expression<boolean>;
    /** Whether the current user can change the domainUsersOnly restriction of this shared drive
     * @default false
     */
    canChangeDomainUsersOnlyRestriction?: boolean | Expression<boolean>;
    /** Whether the current user can change the background of this shared drive
     * @default false
     */
    canChangeDriveBackground?: boolean | Expression<boolean>;
    /** Whether the current user can change the driveMembersOnly restriction of this shared drive
     * @default false
     */
    canChangeDriveMembersOnlyRestriction?: boolean | Expression<boolean>;
    /** Whether the current user can comment on files in this shared drive
     * @default false
     */
    canComment?: boolean | Expression<boolean>;
    /** Whether the current user can copy files in this shared drive
     * @default false
     */
    canCopy?: boolean | Expression<boolean>;
    /** Whether the current user can delete children from folders in this shared drive
     * @default false
     */
    canDeleteChildren?: boolean | Expression<boolean>;
    /** Whether the current user can delete this shared drive. Attempting to delete the shared drive may still fail if there are untrashed items inside the shared drive.
     * @default false
     */
    canDeleteDrive?: boolean | Expression<boolean>;
    /** Whether the current user can download files in this shared drive
     * @default false
     */
    canDownload?: boolean | Expression<boolean>;
    /** Whether the current user can edit files in this shared drive
     * @default false
     */
    canEdit?: boolean | Expression<boolean>;
    /** Whether the current user can list the children of folders in this shared drive
     * @default false
     */
    canListChildren?: boolean | Expression<boolean>;
    /** Whether the current user can add members to this shared drive or remove them or change their role
     * @default false
     */
    canManageMembers?: boolean | Expression<boolean>;
    /** Whether the current user can read the revisions resource of files in this shared drive
     * @default false
     */
    canReadRevisions?: boolean | Expression<boolean>;
    /** Whether the current user can rename files or folders in this shared drive
     * @default false
     */
    canRename?: boolean | Expression<boolean>;
    /** Whether the current user can rename this shared drive
     * @default false
     */
    canRenameDrive?: boolean | Expression<boolean>;
    /** Whether the current user can share files or folders in this shared drive
     * @default false
     */
    canShare?: boolean | Expression<boolean>;
    /** Whether the current user can trash children from folders in this shared drive
     * @default false
     */
    canTrashChildren?: boolean | Expression<boolean>;
  };
    /** The color of this shared drive as an RGB hex string
     */
    colorRgb?: string | Expression<string>;
    /** Whether the shared drive is hidden from default view
     * @default false
     */
    hidden?: boolean | Expression<boolean>;
    /** Restrictions
     * @default {}
     */
    restrictions?: {
    /** Whether the options to copy, print, or download files inside this shared drive, should be disabled for readers and commenters. When this restriction is set to true, it will override the similarly named field to true for any file inside this shared drive.
     * @default false
     */
    adminManagedRestrictions?: boolean | Expression<boolean>;
    /** Whether the options to copy, print, or download files inside this shared drive, should be disabled for readers and commenters. When this restriction is set to true, it will override the similarly named field to true for any file inside this shared drive.
     * @default false
     */
    copyRequiresWriterPermission?: boolean | Expression<boolean>;
    /** Whether access to this shared drive and items inside this shared drive is restricted to users of the domain to which this shared drive belongs. This restriction may be overridden by other sharing policies controlled outside of this shared drive.
     * @default false
     */
    domainUsersOnly?: boolean | Expression<boolean>;
    /** Whether access to items inside this shared drive is restricted to its members
     * @default false
     */
    driveMembersOnly?: boolean | Expression<boolean>;
  };
  };
};

export type GoogleDriveV3DriveCreateOutput = {
  id?: string;
  kind?: string;
  name?: string;
};

export type GoogleDriveV3DriveCreateNode = {
  type: 'n8n-nodes-base.googleDrive';
  version: 3;
  credentials?: Credentials;
  config: NodeConfig<GoogleDriveV3DriveCreateParams>;
  output?: Items<GoogleDriveV3DriveCreateOutput>;
};
