// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleDrive/v3/resource_drive/operation_update.ts
/**
 * Google Drive Node - Version 3
 * Discriminator: resource=drive, operation=update
 */


interface Credentials {
  googleApi: CredentialReference;
  googleDriveOAuth2Api: CredentialReference;
}

/** Update a shared drive */
export type GoogleDriveV3DriveUpdateParams = {
  resource: 'drive';
  operation: 'update';
  authentication?: 'oAuth2' | 'serviceAccount' | Expression<string>;
/**
 * The shared drive to update
 * @default {"mode":"list","value":""}
 */
    driveId?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
/**
 * Update Fields
 * @default {}
 */
    options?: {
    /** The color of this shared drive as an RGB hex string
     */
    colorRgb?: string | Expression<string>;
    /** The updated name of the shared drive
     */
    name?: string | Expression<string> | PlaceholderValue;
    /** Restrictions
     * @default {}
     */
    restrictions?: {
    /** Whether the options to copy, print, or download files inside this shared drive, should be disabled for readers and commenters. When this restriction is set to true, it will override the similarly named field to true for any file inside this shared drive.
     * @default false
     */
    adminManagedRestrictions?: boolean | Expression<boolean>;
    /** Whether the options to copy, print, or download files inside this shared drive, should be disabled for readers and commenters. When this restriction is set to true, it will override the similarly named field to true for any file inside this shared drive.
     * @default false
     */
    copyRequiresWriterPermission?: boolean | Expression<boolean>;
    /** Whether access to this shared drive and items inside this shared drive is restricted to users of the domain to which this shared drive belongs. This restriction may be overridden by other sharing policies controlled outside of this shared drive.
     * @default false
     */
    domainUsersOnly?: boolean | Expression<boolean>;
    /** Whether access to items inside this shared drive is restricted to its members
     * @default false
     */
    driveMembersOnly?: boolean | Expression<boolean>;
  };
  };
};

export type GoogleDriveV3DriveUpdateOutput = {
  id?: string;
  kind?: string;
  name?: string;
};

export type GoogleDriveV3DriveUpdateNode = {
  type: 'n8n-nodes-base.googleDrive';
  version: 3;
  credentials?: Credentials;
  config: NodeConfig<GoogleDriveV3DriveUpdateParams>;
  output?: Items<GoogleDriveV3DriveUpdateOutput>;
};
