// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/reddit/v1/resource_profile/operation_get.ts
/**
 * Reddit Node - Version 1
 * Discriminator: resource=profile, operation=get
 */


interface Credentials {
  redditOAuth2Api: CredentialReference;
}

export type RedditV1ProfileGetParams = {
  resource: 'profile';
  operation: 'get';
/**
 * Details of my account to retrieve
 * @default identity
 */
    details?: 'blockedUsers' | 'friends' | 'identity' | 'karma' | 'prefs' | 'saved' | 'trophies' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @displayOptions.show { details: ["saved"] }
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { details: ["saved"], returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
};

export type RedditV1ProfileGetOutput = {
  awards_on_streams?: boolean;
  chat_group_rollout?: boolean;
  chat_subreddit?: boolean;
  chat_user_settings?: boolean;
  cookie_consent_banner?: boolean;
  crowd_control_for_post?: boolean;
  do_not_track?: boolean;
  expensive_coins_package?: boolean;
  images_in_comments?: boolean;
  is_email_permission_required?: boolean;
  mod_awards?: boolean;
  mod_service_mute_reads?: boolean;
  mod_service_mute_writes?: boolean;
  modlog_copyright_removal?: boolean;
  modmail_harassment_filter?: boolean;
  mweb_xpromo_interstitial_comments_android?: boolean;
  mweb_xpromo_interstitial_comments_ios?: boolean;
  mweb_xpromo_modal_listing_click_daily_dismissible_android?: boolean;
  mweb_xpromo_modal_listing_click_daily_dismissible_ios?: boolean;
  mweb_xpromo_revamp_v2?: {
    experiment_id?: number;
    owner?: string;
    variant?: string;
  };
  noreferrer_to_noopener?: boolean;
  premium_subscriptions_table?: boolean;
  promoted_trend_blanks?: boolean;
  resized_styles_images?: boolean;
  show_amp_link?: boolean;
  use_pref_account_deployment?: boolean;
};

export type RedditV1ProfileGetNode = {
  type: 'n8n-nodes-base.reddit';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<RedditV1ProfileGetParams>;
  output?: Items<RedditV1ProfileGetOutput>;
};
