// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/reddit/v1/resource_profile/operation_get.schema.js
/**
 * Reddit Node - Version 1 - Zod Schema
 * Discriminator: resource=profile, operation=get
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('profile'),
      operation: z.literal('get'),
      details: z.union([z.literal('blockedUsers'), z.literal('friends'), z.literal('identity'), z.literal('karma'), z.literal('prefs'), z.literal('saved'), z.literal('trophies'), expressionSchema]).optional(),
      returnAll: resolveSchema({ parameters, schema: booleanOrExpression, required: false, displayOptions: {"show":{"details":["saved"]}}, defaults: {"details":"identity"} }),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"details":["saved"],"returnAll":[false]}}, defaults: {"details":"identity","returnAll":false} }),
    }).optional(),
  });
};
