// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/unleashedSoftware/v1/resource_sales_order/operation_get_all.schema.js
/**
 * Unleashed Software Node - Version 1 - Zod Schema
 * Discriminator: resource=salesOrder, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('salesOrder').default('salesOrder'),
      operation: z.literal('getAll').default('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filters: z.object({ customerId: stringOrExpression.optional(), customerCode: stringOrExpression.optional(), endDate: stringOrExpression.optional(), modifiedSince: stringOrExpression.optional(), orderNumber: stringOrExpression.optional(), orderStatus: z.array(z.union([z.literal('Backordered'), z.literal('Completed'), z.literal('Deleted'), z.literal('Parked'), z.literal('Placed')])).optional(), startDate: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
