// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/unleashedSoftware/v1/resource_sales_order/operation_get_all.ts
/**
 * Unleashed Software Node - Version 1
 * Discriminator: resource=salesOrder, operation=getAll
 */


interface Credentials {
  unleashedSoftwareApi: CredentialReference;
}

/** Get many sales orders */
export type UnleashedSoftwareV1SalesOrderGetAllParams = {
  resource: 'salesOrder';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Only returns orders for a specified Customer GUID. The CustomerId can be specified as a list of comma-separated GUIDs.
     */
    customerId?: string | Expression<string> | PlaceholderValue;
    /** Returns orders that start with the specific customer code
     */
    customerCode?: string | Expression<string> | PlaceholderValue;
    /** Returns orders with order date before the specified date. UTC.
     */
    endDate?: string | Expression<string>;
    /** Returns orders created or edited after a specified date, must be UTC format
     */
    modifiedSince?: string | Expression<string>;
    /** Returns a single order with the specified order number. If set, it overrides all other filters.
     */
    orderNumber?: string | Expression<string> | PlaceholderValue;
    /** Returns orders with the specified status. If no orderStatus filter is specified, then we exclude "Deleted" by default.
     * @default []
     */
    orderStatus?: Array<'Backordered' | 'Completed' | 'Deleted' | 'Parked' | 'Placed'>;
    /** Returns orders with order date after the specified date. UTC.
     */
    startDate?: string | Expression<string>;
  };
};

export type UnleashedSoftwareV1SalesOrderGetAllOutput = {
  AllocateProduct?: boolean;
  CreatedBy?: string;
  CreatedOn?: string;
  Currency?: {
    CurrencyCode?: string;
    DefaultBuyRate?: null;
    DefaultSellRate?: null;
    Description?: string;
    Guid?: string;
    LastModifiedOn?: string;
  };
  Customer?: {
    CurrencyId?: number;
    CustomerCode?: string;
    CustomerName?: string;
    Guid?: string;
    LastModifiedOn?: string;
  };
  Guid?: string;
  LastModifiedBy?: string;
  LastModifiedOn?: string;
  OrderDate?: string;
  OrderNumber?: string;
  OrderStatus?: string;
  ReceivedDate?: null;
  SalesAccount?: null;
  SalesOrderLines?: Array<{
    BatchNumbers?: null;
    DueDate?: string;
    Guid?: string;
    LastModifiedOn?: string;
    LineNumber?: number;
    OrderQuantity?: number;
    Product?: {
      Guid?: string;
      ProductDescription?: string;
    };
    SerialNumbers?: null;
    XeroTaxCode?: string;
  }>;
  SalesPerson?: {
    Email?: string;
    FullName?: string;
    Guid?: string;
    LastModifiedOn?: string;
    Obsolete?: boolean;
  };
  SaveAddress?: boolean;
  SendAccountingJournalOnly?: boolean;
  SourceId?: null;
  Tax?: {
    CanApplyToExpenses?: boolean;
    CanApplyToRevenue?: boolean;
    Description?: string;
    Guid?: string;
    LastModifiedOn?: string;
    Obsolete?: boolean;
    TaxCode?: string;
  };
  Warehouse?: {
    FaxNumber?: null;
    Guid?: string;
    IsDefault?: boolean;
    LastModifiedOn?: string;
    Obsolete?: boolean;
    WarehouseCode?: string;
    WarehouseName?: string;
  };
  XeroTaxCode?: string;
};

export type UnleashedSoftwareV1SalesOrderGetAllNode = {
  type: 'n8n-nodes-base.unleashedSoftware';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<UnleashedSoftwareV1SalesOrderGetAllParams>;
  output?: Items<UnleashedSoftwareV1SalesOrderGetAllOutput>;
};
