// n8n/dist/controllers/oauth/oauth1-credential.controller.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OAuth1CredentialController = void 0;
const decorators_1 = require("@n8n/decorators");
const axios_1 = __importDefault(require("axios"));
const n8n_workflow_1 = require("n8n-workflow");
const oauth_service_1 = require("../../oauth/oauth.service");
const backend_common_1 = require("@n8n/backend-common");
let OAuth1CredentialController = class OAuth1CredentialController {
    constructor(oauthService, logger) {
        this.oauthService = oauthService;
        this.logger = logger;
    }
    async getAuthUri(req) {
        const credential = await this.oauthService.getCredential(req);
        const uri = await this.oauthService.generateAOauth1AuthUri(credential, {
            cid: credential.id,
            origin: 'static-credential',
            userId: oauth_service_1.skipAuthOnOAuthCallback ? undefined : req.user.id,
        });
        this.logger.debug('OAuth1 authorization successful for new credential', {
            userId: req.user.id,
            credentialId: credential.id,
        });
        return uri;
    }
    async handleCallback(req, res) {
        try {
            const { oauth_verifier, oauth_token, state: encodedState } = req.query;
            if (!oauth_verifier || !oauth_token || !encodedState) {
                return this.oauthService.renderCallbackError(res, 'Insufficient parameters for OAuth1 callback.', `Received following query parameters: ${JSON.stringify(req.query)}`);
            }
            const [credential, _, oauthCredentials, state] = await this.oauthService.resolveCredential(req);
            const oauthToken = await axios_1.default.post(oauthCredentials.accessTokenUrl, { oauth_token, oauth_verifier }, { headers: { 'content-type': 'application/x-www-form-urlencoded' } });
            const paramParser = new URLSearchParams(oauthToken.data);
            const oauthTokenData = Object.fromEntries(paramParser.entries());
            if (!state.origin || state.origin === 'static-credential') {
                await this.oauthService.encryptAndSaveData(credential, { oauthTokenData }, ['csrfSecret']);
                this.logger.debug('OAuth1 callback successful for new credential', {
                    credentialId: credential.id,
                });
                return res.render('oauth-callback');
            }
            if (state.origin === 'dynamic-credential') {
                if (!state.credentialResolverId || typeof state.credentialResolverId !== 'string') {
                    return this.oauthService.renderCallbackError(res, 'Credential resolver ID is required');
                }
                if (!state.authorizationHeader ||
                    typeof state.authorizationHeader !== 'string' ||
                    !state.authorizationHeader.startsWith('Bearer ')) {
                    return this.oauthService.renderCallbackError(res, 'Authorization header is required');
                }
                await this.oauthService.saveDynamicCredential(credential, oauthTokenData, state.authorizationHeader.split('Bearer ')[1], state.credentialResolverId, state.authMetadata ?? {});
                return res.render('oauth-callback');
            }
        }
        catch (e) {
            const error = (0, n8n_workflow_1.ensureError)(e);
            return this.oauthService.renderCallbackError(res, error.message, 'body' in error ? (0, n8n_workflow_1.jsonStringify)(error.body) : undefined);
        }
    }
};
exports.OAuth1CredentialController = OAuth1CredentialController;
__decorate([
    (0, decorators_1.Get)('/auth'),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", Promise)
], OAuth1CredentialController.prototype, "getAuthUri", null);
__decorate([
    (0, decorators_1.Get)('/callback', { usesTemplates: true, skipAuth: oauth_service_1.skipAuthOnOAuthCallback }),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object, Object]),
    __metadata("design:returntype", Promise)
], OAuth1CredentialController.prototype, "handleCallback", null);
exports.OAuth1CredentialController = OAuth1CredentialController = __decorate([
    (0, decorators_1.RestController)('/oauth1-credential'),
    __metadata("design:paramtypes", [oauth_service_1.OauthService,
        backend_common_1.Logger])
], OAuth1CredentialController);
