// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/github/v1/resource_issue/operation_edit.schema.js
/**
 * GitHub Node - Version 1 - Zod Schema
 * Discriminator: resource=issue, operation=edit
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('issue').default('issue'),
      operation: z.literal('edit'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      owner: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('url'), z.literal('name')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      repository: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('url'), z.literal('name')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      issueNumber: numberOrExpression.optional(),
      editFields: z.object({ assignees: z.unknown().optional(), body: stringOrExpression.optional(), labels: z.unknown().optional(), state: z.union([z.literal('closed'), z.literal('open'), expressionSchema]).optional(), state_reason: z.union([z.literal('completed'), z.literal('not_planned'), z.literal('reopened'), expressionSchema]).optional(), title: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
