// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/freshservice/v1/resource_ticket/operation_get_all.ts
/**
 * Freshservice Node - Version 1
 * Discriminator: resource=ticket, operation=getAll
 */


interface Credentials {
  freshserviceApi: CredentialReference;
}

/** Retrieve many agents */
export type FreshserviceV1TicketGetAllParams = {
  resource: 'ticket';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** ID of the agent to whom the tickets have been assigned. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    agent_id?: string | Expression<string>;
    /** ID of the group to which the tickets have been assigned. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    group_id?: string | Expression<string>;
    /** Impact
     * @default 1
     */
    impact?: 1 | 2 | 3 | Expression<number>;
    /** Priority
     * @default 1
     */
    priority?: 1 | 2 | 3 | 4 | Expression<number>;
    /** Status
     * @default 2
     */
    status?: 2 | 3 | 4 | 5 | Expression<number>;
    /** Date when the ticket was created
     */
    created_at?: string | Expression<string>;
    /** Date when the ticket is due to be resolved
     */
    due_by?: string | Expression<string>;
  };
};

export type FreshserviceV1TicketGetAllNode = {
  type: 'n8n-nodes-base.freshservice';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<FreshserviceV1TicketGetAllParams>;
};
