// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/freshservice/v1/resource_ticket/operation_get_all.schema.js
/**
 * Freshservice Node - Version 1 - Zod Schema
 * Discriminator: resource=ticket, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('ticket'),
      operation: z.literal('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filters: z.object({ agent_id: stringOrExpression.optional(), group_id: stringOrExpression.optional(), impact: z.union([z.literal(1), z.literal(2), z.literal(3), expressionSchema]).optional(), priority: z.union([z.literal(1), z.literal(2), z.literal(3), z.literal(4), expressionSchema]).optional(), status: z.union([z.literal(2), z.literal(3), z.literal(4), z.literal(5), expressionSchema]).optional(), created_at: stringOrExpression.optional(), due_by: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
