// @n8n/n8n-nodes-langchain/dist/nodes/vendors/OpenAi/OpenAi.node.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OpenAi = void 0;
const n8n_workflow_1 = require("n8n-workflow");
const description_1 = require("./helpers/description");
const OpenAiV1_node_1 = require("./v1/OpenAiV1.node");
const OpenAiV2_node_1 = require("./v2/OpenAiV2.node");
class OpenAi extends n8n_workflow_1.VersionedNodeType {
    constructor() {
        const baseDescription = {
            displayName: 'OpenAI',
            name: 'openAi',
            icon: { light: 'file:openAi.svg', dark: 'file:openAi.dark.svg' },
            group: ['transform'],
            defaultVersion: 2.1,
            subtitle: `={{(${description_1.prettifyOperation})($parameter.resource, $parameter.operation)}}`,
            description: 'Message an assistant or GPT, analyze images, generate audio, etc.',
            codex: {
                alias: [
                    'LangChain',
                    'ChatGPT',
                    'Sora',
                    'DallE',
                    'whisper',
                    'audio',
                    'transcribe',
                    'tts',
                    'assistant',
                ],
                categories: ['AI'],
                subcategories: {
                    AI: ['Agents', 'Miscellaneous', 'Root Nodes'],
                },
                resources: {
                    primaryDocumentation: [
                        {
                            url: 'https://docs.n8n.io/integrations/builtin/app-nodes/n8n-nodes-langchain.openai/',
                        },
                    ],
                },
            },
            builderHint: {
                message: 'For text generation, reasoning and tools, use AI Agent with OpenAI Chat Model. This OpenAI node is for specialized operations: image generation (DALL-E), audio (Whisper, TTS), and video generation (Sora).',
                relatedNodes: [
                    {
                        nodeType: '@n8n/n8n-nodes-langchain.agent',
                        relationHint: 'Prefer for most LLM tasks',
                    },
                    {
                        nodeType: '@n8n/n8n-nodes-langchain.lmChatOpenAi',
                        relationHint: 'Prefer for most LLM tasks',
                    },
                ],
            },
        };
        const nodeVersions = {
            1: new OpenAiV1_node_1.OpenAiV1(baseDescription),
            1.1: new OpenAiV1_node_1.OpenAiV1(baseDescription),
            1.2: new OpenAiV1_node_1.OpenAiV1(baseDescription),
            1.3: new OpenAiV1_node_1.OpenAiV1(baseDescription),
            1.4: new OpenAiV1_node_1.OpenAiV1(baseDescription),
            1.5: new OpenAiV1_node_1.OpenAiV1(baseDescription),
            1.6: new OpenAiV1_node_1.OpenAiV1(baseDescription),
            1.7: new OpenAiV1_node_1.OpenAiV1(baseDescription),
            1.8: new OpenAiV1_node_1.OpenAiV1(baseDescription),
            2: new OpenAiV2_node_1.OpenAiV2(baseDescription),
            2.1: new OpenAiV2_node_1.OpenAiV2(baseDescription),
        };
        super(nodeVersions, baseDescription);
    }
}
exports.OpenAi = OpenAi;
