// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mailerLite/v2/resource_subscriber/operation_update.schema.js
/**
 * MailerLite Node - Version 2 - Zod Schema
 * Discriminator: resource=subscriber, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('subscriber').default('subscriber'),
      operation: z.literal('update'),
      subscriberId: stringOrExpression.optional(),
      additionalFields: z.object({ customFieldsUi: z.unknown().optional(), status: z.union([z.literal('active'), z.literal('bounced'), z.literal('junk'), z.literal('unconfirmed'), z.literal('unsubscribed'), expressionSchema]).optional(), subscribed_at: stringOrExpression.optional(), ip_address: stringOrExpression.optional(), opted_in_at: stringOrExpression.optional(), optin_ip: stringOrExpression.optional(), unsubscribed_at: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
