// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftOutlook/v2/resource_message_attachment/operation_get.schema.js
/**
 * Microsoft Outlook Node - Version 2 - Zod Schema
 * Discriminator: resource=messageAttachment, operation=get
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('messageAttachment'),
      operation: z.literal('get'),
      messageId: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      attachmentId: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      options: z.object({ fields: z.array(z.union([z.literal('contentType'), z.literal('isInline'), z.literal('lastModifiedDateTime'), z.literal('name'), z.literal('size')])).optional() }).optional(),
    }).optional(),
  });
};
