// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleFirebaseCloudFirestore/v11/resource_document/operation_get.schema.js
/**
 * Google Cloud Firestore Node - Version 1.1 - Zod Schema
 * Discriminator: resource=document, operation=get
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('document').default('document'),
      operation: z.literal('get').default('get'),
      authentication: z.union([z.literal('googleFirebaseCloudFirestoreOAuth2Api'), z.literal('serviceAccount'), expressionSchema]).optional(),
      projectId: stringOrExpression.optional(),
      database: stringOrExpression.optional(),
      collection: stringOrExpression.optional(),
      documentId: stringOrExpression.optional(),
      simple: booleanOrExpression.optional(),
    }).optional(),
  });
};
