// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleFirebaseCloudFirestore/v11/resource_document/operation_upsert.ts
/**
 * Google Cloud Firestore Node - Version 1.1
 * Discriminator: resource=document, operation=upsert
 */


interface Credentials {
  googleFirebaseCloudFirestoreOAuth2Api: CredentialReference;
  googleApi: CredentialReference;
}

/** Create a new document, or update the current one if it already exists (upsert) */
export type GoogleFirebaseCloudFirestoreV11DocumentUpsertParams = {
  resource: 'document';
  operation: 'upsert';
  authentication?: 'googleFirebaseCloudFirestoreOAuth2Api' | 'serviceAccount' | Expression<string>;
/**
 * As displayed in firebase console URL. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    projectId?: string | Expression<string>;
/**
 * Usually the provided default value will work
 * @default (default)
 */
    database?: string | Expression<string> | PlaceholderValue;
/**
 * Collection name
 */
    collection?: string | Expression<string> | PlaceholderValue;
/**
 * Name of the field in an input item that contains the document ID
 */
    updateKey?: string | Expression<string> | PlaceholderValue;
/**
 * Columns to insert
 */
    columns?: string | Expression<string> | PlaceholderValue;
};

export type GoogleFirebaseCloudFirestoreV11DocumentUpsertOutput = {
  updateTime?: string;
};

export type GoogleFirebaseCloudFirestoreV11DocumentUpsertNode = {
  type: 'n8n-nodes-base.googleFirebaseCloudFirestore';
  version: 1.1;
  credentials?: Credentials;
  config: NodeConfig<GoogleFirebaseCloudFirestoreV11DocumentUpsertParams>;
  output?: Items<GoogleFirebaseCloudFirestoreV11DocumentUpsertOutput>;
};
