// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/goToWebinar/v1/resource_session/operation_get_all.ts
/**
 * GoToWebinar Node - Version 1
 * Discriminator: resource=session, operation=getAll
 */


interface Credentials {
  goToWebinarOAuth2Api: CredentialReference;
}

export type GoToWebinarV1SessionGetAllParams = {
  resource: 'session';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 10
 */
    limit?: number | Expression<number>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Time Range
     * @default {}
     */
    times?: {
        /** Times Properties
     */
    timesProperties?: {
      /** Start of the datetime range for the session
       */
      fromTime?: string | Expression<string>;
      /** End of the datetime range for the session
       */
      toTime?: string | Expression<string>;
    };
  };
    /** Webinar by which to filter the sessions to retrieve. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     * @default {}
     */
    webinarKey?: string | Expression<string>;
  };
};

export type GoToWebinarV1SessionGetAllNode = {
  type: 'n8n-nodes-base.goToWebinar';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GoToWebinarV1SessionGetAllParams>;
};
