// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/goToWebinar/v1/resource_session/operation_get_details.schema.js
/**
 * GoToWebinar Node - Version 1 - Zod Schema
 * Discriminator: resource=session, operation=getDetails
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('session'),
      operation: z.literal('getDetails'),
      webinarKey: stringOrExpression.optional(),
      sessionKey: stringOrExpression.optional(),
      details: z.union([z.literal('performance'), z.literal('polls'), z.literal('questions'), z.literal('surveys'), expressionSchema]).optional(),
    }).optional(),
  });
};
