// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/slack/v2/resource_file/operation_get_all.schema.js
/**
 * Slack Node - Version 2 - Zod Schema
 * Discriminator: resource=file, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('file'),
      operation: z.literal('getAll'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filters: z.object({ channelId: stringOrExpression.optional(), showFilesHidden: booleanOrExpression.optional(), tsFrom: stringOrExpression.optional(), tsTo: stringOrExpression.optional(), types: z.array(z.union([z.literal('all'), z.literal('gdocs'), z.literal('images'), z.literal('pdfs'), z.literal('snippets'), z.literal('spaces'), z.literal('zips')])).optional(), userId: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
