// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/zoom/v1/resource_meeting/operation_get_all.ts
/**
 * Zoom Node - Version 1
 * Discriminator: resource=meeting, operation=getAll
 */


interface Credentials {
  zoomApi: CredentialReference;
  zoomOAuth2Api: CredentialReference;
}

/** Retrieve many meetings */
export type ZoomV1MeetingGetAllParams = {
  resource: 'meeting';
  operation: 'getAll';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 30
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Meeting type
     * @default live
     */
    type?: 'scheduled' | 'live' | 'upcoming' | Expression<string>;
  };
};

export type ZoomV1MeetingGetAllOutput = {
  created_at?: string;
  duration?: number;
  host_id?: string;
  id?: number;
  join_url?: string;
  start_time?: string;
  timezone?: string;
  topic?: string;
  type?: number;
  uuid?: string;
};

export type ZoomV1MeetingGetAllNode = {
  type: 'n8n-nodes-base.zoom';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ZoomV1MeetingGetAllParams>;
  output?: Items<ZoomV1MeetingGetAllOutput>;
};
