// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/theHiveProject/v1/resource_case/operation_add_attachment.ts
/**
 * TheHive 5 Node - Version 1
 * Discriminator: resource=case, operation=addAttachment
 */


interface Credentials {
  theHiveProjectApi: CredentialReference;
}

export type TheHiveProjectV1CaseAddAttachmentParams = {
  resource: 'case';
  operation: 'addAttachment';
/**
 * Case
 * @default {"mode":"list","value":""}
 */
    caseId?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
/**
 * Array of supported attachments to add to the message
 * @default {}
 */
    attachmentsUi?: {
        /** Values
     */
    values?: Array<{
      /** Add the field name from the input node
       * @hint The name of the field with the attachment in the node input
       * @default data
       */
      field?: string | Expression<string> | PlaceholderValue;
    }>;
  };
/**
 * Options
 * @default {}
 */
    options?: {
    /** Whether to rename the file in case a file with the same name already exists
     * @default false
     */
    canRename?: boolean | Expression<boolean>;
  };
};

export type TheHiveProjectV1CaseAddAttachmentNode = {
  type: 'n8n-nodes-base.theHiveProject';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<TheHiveProjectV1CaseAddAttachmentParams>;
};
