// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/copper/v1/resource_opportunity/operation_get_all.ts
/**
 * Copper Node - Version 1
 * Discriminator: resource=opportunity, operation=getAll
 */


interface Credentials {
  copperApi: CredentialReference;
}

export type CopperV1OpportunityGetAllParams = {
  resource: 'opportunity';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 5
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filterFields?: {
    /** Comma-separated IDs of the primary companies to filter by
     */
    company_ids?: string | Expression<string> | PlaceholderValue;
    /** Comma-separated IDs of the customer sources to filter by
     */
    customer_source_ids?: string | Expression<string> | PlaceholderValue;
  };
};

export type CopperV1OpportunityGetAllNode = {
  type: 'n8n-nodes-base.copper';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<CopperV1OpportunityGetAllParams>;
};
