// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/copper/v1/resource_opportunity/operation_get_all.schema.js
/**
 * Copper Node - Version 1 - Zod Schema
 * Discriminator: resource=opportunity, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('opportunity'),
      operation: z.literal('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filterFields: z.object({ company_ids: stringOrExpression.optional(), customer_source_ids: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
