// @microsoft/agents-a365-tooling/dist/esm/McpToolServerConfigurationService.js
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
import fs from 'fs';
import path from 'path';
import axios from 'axios';
import { Utility } from './Utility';
import { StreamableHTTPClientTransport } from '@modelcontextprotocol/sdk/client/streamableHttp.js';
import { Client } from '@modelcontextprotocol/sdk/client/index.js';
/**
 * Service responsible for discovering and normalizing MCP (Model Context Protocol)
 * tool servers and producing configuration objects consumable by the Claude SDK.
 */
export class McpToolServerConfigurationService {
    /**
     * Construct a McpToolServerConfigurationService.
     */
    constructor() {
        this.logger = console;
    }
    /**
     * Return MCP server definitions for the given agent. In development (NODE_ENV=Development) this reads the local ToolingManifest.json; otherwise it queries the remote tooling gateway.
     *
     * @param agenticAppId The agentic app id for which to discover servers.
     * @param authToken Optional bearer token used when querying the remote tooling gateway.
     * @returns A promise resolving to an array of normalized MCP server configuration objects.
     */
    async listToolServers(agenticAppId, authToken) {
        return await (this.isDevScenario() ? this.getMCPServerConfigsFromManifest() : this.getMCPServerConfigsFromToolingGateway(agenticAppId, authToken));
    }
    /**
     * Connect to the MCP server and return tools with names prefixed by the server name.
     * Throws if the server URL is missing or the client fails to list tools.
     */
    async getMcpClientTools(mcpServerName, mcpServerConfig) {
        if (!mcpServerConfig) {
            throw new Error('Invalid MCP Server Configuration');
        }
        if (!mcpServerConfig.url) {
            throw new Error('MCP Server URL cannot be null or empty');
        }
        const transport = new StreamableHTTPClientTransport(new URL(mcpServerConfig.url), {
            requestInit: {
                headers: mcpServerConfig.headers
            }
        });
        const mcpClient = new Client({
            name: mcpServerName + ' Client',
            version: '1.0',
        });
        await mcpClient.connect(transport);
        const toolsObj = await mcpClient.listTools();
        await mcpClient.close();
        return toolsObj.tools;
    }
    /**
     * Query the tooling gateway for MCP servers for the specified agent and normalize each entry's mcpServerUniqueName into a full URL using Utility.BuildMcpServerUrl.
     * Throws an error if the gateway call fails.
     *
     * @param agenticAppId The agentic app id used by the tooling gateway to scope results.
     * @param authToken Optional Bearer token to include in the Authorization header when calling the gateway.
     * @throws Error when the gateway call fails or returns an unexpected payload.
     */
    async getMCPServerConfigsFromToolingGateway(agenticAppId, authToken) {
        // Validate the authentication token
        Utility.ValidateAuthToken(authToken);
        const configEndpoint = Utility.GetToolingGatewayForDigitalWorker(agenticAppId);
        try {
            const response = await axios.get(configEndpoint, {
                headers: {
                    'Authorization': authToken ? `Bearer ${authToken}` : undefined,
                },
                timeout: 10000 // 10 seconds timeout
            });
            return (response.data) || [];
        }
        catch (err) {
            const error = err;
            throw new Error(`Failed to read MCP servers from endpoint: ${error.code || 'UNKNOWN'} ${error.message || 'Unknown error'}`);
        }
    }
    /**
     * Read MCP servers from a local ToolingManifest.json file (development only).
     * Searches process.cwd() and process.argv[1] for the manifest file.
     *
     * Reads MCP server configurations from ToolingManifest.json in the application's content root.
     * The file should be located at: [ProjectRoot]/ToolingManifest.json
     *
     * Example ToolingManifest.json:
     * {
     *   "mcpServers": [
     *     {
     *       "mcpServerName": "mailMCPServerConfig",
     *       "mcpServerUniqueName": "mcp_MailTools"
     *     },
     *     {
     *       "mcpServerName": "sharePointMCPServerConfig",
     *       "mcpServerUniqueName": "mcp_SharePointTools"
     *     }
     *   ]
     * }
     */
    async getMCPServerConfigsFromManifest() {
        let manifestPath = path.join(process.cwd(), 'ToolingManifest.json');
        if (!fs.existsSync(manifestPath)) {
            this.logger.warn(`ToolingManifest.json not found at ${manifestPath}, checking argv[1] location.`);
            manifestPath = path.join(path.dirname(process.argv[1] || ''), 'ToolingManifest.json');
        }
        if (!fs.existsSync(manifestPath)) {
            this.logger.warn(`ToolingManifest.json not found at ${manifestPath}`);
            return [];
        }
        try {
            const jsonContent = fs.readFileSync(manifestPath, 'utf-8');
            const manifestData = JSON.parse(jsonContent);
            const mcpServers = manifestData.mcpServers || [];
            return mcpServers.map((s) => {
                return {
                    mcpServerName: s.mcpServerName,
                    url: Utility.BuildMcpServerUrl(s.mcpServerName)
                };
            });
        }
        catch (err) {
            const error = err;
            this.logger.error(`Error reading or parsing ToolingManifest.json: ${error.message || 'Unknown error'}`);
            return [];
        }
    }
    /**
     * Detect if the process is running in a development scenario based on environment variables.
     *
     * @returns {boolean} True when running in a development environment.
     */
    isDevScenario() {
        const environment = process.env.NODE_ENV || '';
        return environment.toLowerCase() === 'development';
    }
}
