// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftExcel/v1/resource_workbook/operation_get_all.ts
/**
 * Microsoft Excel Node - Version 1
 * Discriminator: resource=workbook, operation=getAll
 */


interface Credentials {
  microsoftExcelOAuth2Api: CredentialReference;
}

/** Workbook is the top level object which contains related workbook objects such as worksheets, tables, ranges, etc */
export type MicrosoftExcelV1WorkbookGetAllParams = {
  resource: 'workbook';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Fields the response will containt. Multiple can be added separated by ,.
     */
    fields?: string | Expression<string> | PlaceholderValue;
  };
};

export type MicrosoftExcelV1WorkbookGetAllNode = {
  type: 'n8n-nodes-base.microsoftExcel';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MicrosoftExcelV1WorkbookGetAllParams>;
};
