// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftExcel/v1/resource_workbook/operation_add_worksheet.ts
/**
 * Microsoft Excel Node - Version 1
 * Discriminator: resource=workbook, operation=addWorksheet
 */


interface Credentials {
  microsoftExcelOAuth2Api: CredentialReference;
}

/** Workbook is the top level object which contains related workbook objects such as worksheets, tables, ranges, etc */
export type MicrosoftExcelV1WorkbookAddWorksheetParams = {
  resource: 'workbook';
  operation: 'addWorksheet';
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    workbook?: string | Expression<string>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** The name of the worksheet to be added. If specified, name should be unqiue. If not specified, Excel determines the name of the new worksheet.
     */
    name?: string | Expression<string> | PlaceholderValue;
  };
};

export type MicrosoftExcelV1WorkbookAddWorksheetNode = {
  type: 'n8n-nodes-base.microsoftExcel';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MicrosoftExcelV1WorkbookAddWorksheetParams>;
};
