// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleCalendar/v12/resource_event/operation_get_all.schema.js
/**
 * Google Calendar Node - Version 1.2 - Zod Schema
 * Discriminator: resource=event, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('event').default('event'),
      operation: z.literal('getAll'),
      calendar: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      options: z.object({ timeMin: stringOrExpression.optional(), timeMax: stringOrExpression.optional(), singleEvents: booleanOrExpression.optional(), fields: stringOrExpression.optional(), iCalUID: stringOrExpression.optional(), maxAttendees: numberOrExpression.optional(), orderBy: z.union([z.literal('startTime'), z.literal('updated'), expressionSchema]).optional(), query: stringOrExpression.optional(), recurringEventHandling: z.union([z.literal('expand'), z.literal('first'), z.literal('next'), expressionSchema]).optional(), showDeleted: booleanOrExpression.optional(), showHiddenInvitations: booleanOrExpression.optional(), timeZone: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(), updatedMin: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
