// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleCloudStorage/v1/resource_bucket/operation_get.ts
/**
 * Google Cloud Storage Node - Version 1
 * Discriminator: resource=bucket, operation=get
 */


interface Credentials {
  googleCloudStorageOAuth2Api: CredentialReference;
}

/** Get metadata for a specific Bucket */
export type GoogleCloudStorageV1BucketGetParams = {
  resource: 'bucket';
  operation: 'get';
/**
 * Bucket Name
 */
    bucketName?: string | Expression<string> | PlaceholderValue;
/**
 * Projection
 * @default noAcl
 */
    projection?: 'full' | 'noAcl';
/**
 * Filters
 * @default {}
 */
    getFilters?: {
    /** Only return data if the metageneration value of the Bucket matches the sent value
     * @default 0
     */
    ifMetagenerationMatch?: number | Expression<number>;
    /** Only return data if the metageneration value of the Bucket does not match the sent value
     * @default 0
     */
    ifMetagenerationNotMatch?: number | Expression<number>;
  };
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type GoogleCloudStorageV1BucketGetOutput = {
  etag?: string;
  generation?: string;
  hierarchicalNamespace?: {
    enabled?: boolean;
  };
  iamConfiguration?: {
    bucketPolicyOnly?: {
      enabled?: boolean;
      lockedTime?: string;
    };
    publicAccessPrevention?: string;
    uniformBucketLevelAccess?: {
      enabled?: boolean;
      lockedTime?: string;
    };
  };
  id?: string;
  kind?: string;
  location?: string;
  locationType?: string;
  metageneration?: string;
  name?: string;
  projectNumber?: string;
  selfLink?: string;
  softDeletePolicy?: {
    effectiveTime?: string;
    retentionDurationSeconds?: string;
  };
  storageClass?: string;
  timeCreated?: string;
  updated?: string;
};

export type GoogleCloudStorageV1BucketGetNode = {
  type: 'n8n-nodes-base.googleCloudStorage';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GoogleCloudStorageV1BucketGetParams>;
  output?: Items<GoogleCloudStorageV1BucketGetOutput>;
};
