// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleCloudStorage/v1/resource_bucket/operation_create.ts
/**
 * Google Cloud Storage Node - Version 1
 * Discriminator: resource=bucket, operation=create
 */


interface Credentials {
  googleCloudStorageOAuth2Api: CredentialReference;
}

/** Create a new Bucket */
export type GoogleCloudStorageV1BucketCreateParams = {
  resource: 'bucket';
  operation: 'create';
/**
 * Project ID
 */
    projectId?: string | Expression<string> | PlaceholderValue;
/**
 * Bucket Name
 */
    bucketName?: string | Expression<string> | PlaceholderValue;
/**
 * Projection
 * @default noAcl
 */
    projection?: 'full' | 'noAcl';
/**
 * Predefined Access Control
 * @default {}
 */
    createAcl?: {
    /** Predefined ACL
     * @default authenticatedRead
     */
    predefinedAcl?: 'authenticatedRead' | 'private' | 'projectPrivate' | 'publicRead' | 'publicReadWrite';
    /** Predefined Default Object ACL
     * @default authenticatedRead
     */
    predefinedDefaultObjectAcl?: 'authenticatedRead' | 'bucketOwnerFullControl' | 'bucketOwnerRead' | 'private' | 'projectPrivate' | 'publicRead';
  };
/**
 * Additional Parameters
 * @default {}
 */
    createBody?: {
    /** Access Control
     * @default []
     */
    acl?: IDataObject | string;
    /** Billing
     * @default {}
     */
    billing?: IDataObject | string;
    /** CORS
     * @default []
     */
    cors?: IDataObject | string;
    /** Custom Placement Config
     * @default {}
     */
    customPlacementConfig?: IDataObject | string;
    /** Data Locations
     * @default []
     */
    dataLocations?: IDataObject | string;
    /** Default Event Based Hold
     * @default true
     */
    defaultEventBasedHold?: boolean;
    /** Default Object ACL
     * @default []
     */
    defaultObjectAcl?: IDataObject | string;
    /** Encryption
     * @default {}
     */
    encryption?: IDataObject | string;
    /** IAM Configuration
     * @default {}
     */
    iamConfiguration?: IDataObject | string;
    /** Labels
     * @default {}
     */
    labels?: IDataObject | string;
    /** Lifecycle
     * @default {}
     */
    lifecycle?: IDataObject | string;
    /** Location
     * @default US
     */
    location?: string;
    /** Logging
     * @default {}
     */
    logging?: IDataObject | string;
    /** Retention Policy
     * @default {}
     */
    retentionPolicy?: IDataObject | string;
    /** Recovery Point Objective
     * @default DEFAULT
     */
    rpo?: string;
    /** Storage Class
     * @default STANDARD
     */
    storageClass?: string;
    /** Versioning
     * @default {}
     */
    versioning?: IDataObject | string;
    /** Website
     * @default {}
     */
    website?: IDataObject | string;
  };
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type GoogleCloudStorageV1BucketCreateOutput = {
  etag?: string;
  generation?: string;
  iamConfiguration?: {
    bucketPolicyOnly?: {
      enabled?: boolean;
    };
    publicAccessPrevention?: string;
    uniformBucketLevelAccess?: {
      enabled?: boolean;
    };
  };
  id?: string;
  kind?: string;
  location?: string;
  locationType?: string;
  metageneration?: string;
  name?: string;
  projectNumber?: string;
  rpo?: string;
  selfLink?: string;
  softDeletePolicy?: {
    effectiveTime?: string;
    retentionDurationSeconds?: string;
  };
  storageClass?: string;
  timeCreated?: string;
  updated?: string;
};

export type GoogleCloudStorageV1BucketCreateNode = {
  type: 'n8n-nodes-base.googleCloudStorage';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GoogleCloudStorageV1BucketCreateParams>;
  output?: Items<GoogleCloudStorageV1BucketCreateOutput>;
};
