// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftOutlook/v1/resource_message/operation_reply.schema.js
/**
 * Microsoft Outlook Node - Version 1 - Zod Schema
 * Discriminator: resource=message, operation=reply
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('message').default('message'),
      operation: z.literal('reply'),
      messageId: stringOrExpression.optional(),
      replyType: z.union([z.literal('reply'), z.literal('replyAll'), expressionSchema]).optional(),
      comment: stringOrExpression.optional(),
      send: booleanOrExpression.optional(),
      additionalFields: resolveSchema({ parameters, schema: z.object({ attachments: z.unknown().optional(), bccRecipients: stringOrExpression.optional(), bodyContent: stringOrExpression.optional(), bodyContentType: z.union([z.literal('html'), z.literal('Text'), expressionSchema]).optional(), ccRecipients: stringOrExpression.optional(), internetMessageHeaders: z.unknown().optional(), from: stringOrExpression.optional(), importance: z.union([z.literal('Low'), z.literal('Normal'), z.literal('High'), expressionSchema]).optional(), isReadReceiptRequested: booleanOrExpression.optional(), toRecipients: stringOrExpression.optional(), replyTo: stringOrExpression.optional(), subject: stringOrExpression.optional() }), required: false, displayOptions: {"show":{"replyType":["reply"]}}, defaults: {"replyType":"reply"} }),
    }).optional(),
  });
};
