// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/hackerNews/v1/resource_all/operation_get_all.schema.js
/**
 * Hacker News Node - Version 1 - Zod Schema
 * Discriminator: resource=all, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('all'),
      operation: z.literal('getAll').default('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      additionalFields: z.object({ keyword: stringOrExpression.optional(), tags: z.array(z.union([z.literal('ask_hn'), z.literal('comment'), z.literal('front_page'), z.literal('poll'), z.literal('show_hn'), z.literal('story')])).optional() }).optional(),
    }).optional(),
  });
};
