// @n8n/n8n-nodes-langchain/dist/node-definitions/nodes/n8n-nodes-langchain/openAi/v12/resource_audio/operation_translate.ts
/**
 * OpenAI Node - Version 1.2
 * Discriminator: resource=audio, operation=translate
 */


interface Credentials {
  openAiApi: CredentialReference;
}

/** Translates audio into text in English */
export type LcOpenAiV12AudioTranslateParams = {
  resource: 'audio';
  operation: 'translate';
/**
 * Name of the binary property which contains the audio file in one of these formats: flac, mp3, mp4, mpeg, mpga, m4a, ogg, wav, or webm
 * @hint The name of the input field containing the binary file data to be processed
 * @default data
 */
    binaryPropertyName?: string | Expression<string> | PlaceholderValue;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Output Randomness (Temperature)
     * @default 0
     */
    temperature?: number | Expression<number>;
  };
};

export interface LcOpenAiV12AudioTranslateSubnodeConfig {
  tools?: ToolInstance[];
  memory?: MemoryInstance;
}

export type LcOpenAiV12AudioTranslateNode = {
  type: '@n8n/n8n-nodes-langchain.openAi';
  version: 1.2;
  credentials?: Credentials;
  config: NodeConfig<LcOpenAiV12AudioTranslateParams> & { subnodes?: LcOpenAiV12AudioTranslateSubnodeConfig };
};
