// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/invoiceNinja/v1/resource_bank_transaction/operation_match_payment.schema.js
/**
 * Invoice Ninja Node - Version 1 - Zod Schema
 * Discriminator: resource=bank_transaction, operation=matchPayment
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('bank_transaction'),
      operation: z.literal('matchPayment'),
      apiVersion: z.union([z.literal('v4'), z.literal('v5'), expressionSchema]).optional(),
      bankTransactionId: stringOrExpression.optional(),
      paymentId: stringOrExpression.optional(),
    }).optional(),
  });
};
