// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/zulip/v1/resource_user/operation_update.schema.js
/**
 * Zulip Node - Version 1 - Zod Schema
 * Discriminator: resource=user, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('user'),
      operation: z.literal('update'),
      userId: stringOrExpression.optional(),
      additionalFields: z.object({ fullName: stringOrExpression.optional(), isAdmin: booleanOrExpression.optional(), isGuest: booleanOrExpression.optional(), profileData: z.unknown().optional(), role: z.union([z.literal(600), z.literal(400), z.literal(200), z.literal(300), z.literal(100), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
