// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/zulip/v1/resource_user/operation_get.ts
/**
 * Zulip Node - Version 1
 * Discriminator: resource=user, operation=get
 */


interface Credentials {
  zulipApi: CredentialReference;
}

/** Get a message */
export type ZulipV1UserGetParams = {
  resource: 'user';
  operation: 'get';
/**
 * The ID of user to get
 */
    userId?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Whether the client supports computing gravatars URLs. If enabled, avatar_url will be included in the response only if there is a Zulip avatar, and will be null for users who are using gravatar as their avatar.
     * @default false
     */
    clientGravatar?: boolean | Expression<boolean>;
    /** Whether the client wants custom profile field data to be included in the response
     * @default false
     */
    includeCustomProfileFields?: boolean | Expression<boolean>;
  };
};

export type ZulipV1UserGetNode = {
  type: 'n8n-nodes-base.zulip';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ZulipV1UserGetParams>;
};
