// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/nasa/v1/resource_astronomy_picture_of_the_day/operation_get.ts
/**
 * NASA Node - Version 1
 * Discriminator: resource=astronomyPictureOfTheDay, operation=get
 */


interface Credentials {
  nasaApi: CredentialReference;
}

/** Get the Astronomy Picture of the Day */
export type NasaV1AstronomyPictureOfTheDayGetParams = {
  resource: 'astronomyPictureOfTheDay';
  operation: 'get';
/**
 * By default just the URL of the image is returned. When set to true the image will be downloaded.
 * @default true
 */
    download?: boolean | Expression<boolean>;
/**
 * Put Output File in Field
 * @hint The name of the output binary field to put the file in
 * @displayOptions.show { download: [true] }
 * @default data
 */
    binaryPropertyName?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Date
     */
    date?: string | Expression<string>;
  };
};

export type NasaV1AstronomyPictureOfTheDayGetOutput = {
  copyright?: string;
  date?: string;
  explanation?: string;
  hdurl?: string;
  media_type?: string;
  service_version?: string;
  title?: string;
  url?: string;
};

export type NasaV1AstronomyPictureOfTheDayGetNode = {
  type: 'n8n-nodes-base.nasa';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<NasaV1AstronomyPictureOfTheDayGetParams>;
  output?: Items<NasaV1AstronomyPictureOfTheDayGetOutput>;
};
