// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/nasa/v1/resource_astronomy_picture_of_the_day/operation_get.schema.js
/**
 * NASA Node - Version 1 - Zod Schema
 * Discriminator: resource=astronomyPictureOfTheDay, operation=get
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('astronomyPictureOfTheDay').default('astronomyPictureOfTheDay'),
      operation: z.literal('get').default('get'),
      download: booleanOrExpression.optional(),
      binaryPropertyName: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"download":[true]}}, defaults: {"download":true} }),
      additionalFields: z.object({ date: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
