// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/reddit/v1/resource_user/operation_get.ts
/**
 * Reddit Node - Version 1
 * Discriminator: resource=user, operation=get
 */


interface Credentials {
  redditOAuth2Api: CredentialReference;
}

export type RedditV1UserGetParams = {
  resource: 'user';
  operation: 'get';
/**
 * Reddit ID of the user to retrieve
 */
    username?: string | Expression<string> | PlaceholderValue;
/**
 * Details of the user to retrieve
 * @default about
 */
    details?: 'about' | 'comments' | 'gilded' | 'overview' | 'submitted' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @displayOptions.show { details: ["overview", "submitted", "comments", "gilded"] }
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { details: ["comments", "gilded", "overview", "submitted"], returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
};

export type RedditV1UserGetNode = {
  type: 'n8n-nodes-base.reddit';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<RedditV1UserGetParams>;
};
