// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/reddit/v1/resource_user/operation_get.schema.js
/**
 * Reddit Node - Version 1 - Zod Schema
 * Discriminator: resource=user, operation=get
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('user'),
      operation: z.literal('get'),
      username: stringOrExpression.optional(),
      details: z.union([z.literal('about'), z.literal('comments'), z.literal('gilded'), z.literal('overview'), z.literal('submitted'), expressionSchema]).optional(),
      returnAll: resolveSchema({ parameters, schema: booleanOrExpression, required: false, displayOptions: {"show":{"details":["overview","submitted","comments","gilded"]}}, defaults: {"details":"about"} }),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"details":["comments","gilded","overview","submitted"],"returnAll":[false]}}, defaults: {"details":"about","returnAll":false} }),
    }).optional(),
  });
};
