// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleSlides/v2/resource_presentation/operation_replace_text.ts
/**
 * Google Slides Node - Version 2
 * Discriminator: resource=presentation, operation=replaceText
 */


interface Credentials {
  googleApi: CredentialReference;
  googleSlidesOAuth2Api: CredentialReference;
}

/** Replace text in a presentation */
export type GoogleSlidesV2PresentationReplaceTextParams = {
  resource: 'presentation';
  operation: 'replaceText';
/**
 * Authentication
 * @default oAuth2
 */
    authentication?: 'oAuth2' | 'serviceAccount' | Expression<string>;
/**
 * ID of the presentation to retrieve. Found in the presentation URL: &lt;code&gt;https://docs.google.com/presentation/d/PRESENTATION_ID/edit&lt;/code&gt;
 */
    presentationId?: string | Expression<string> | PlaceholderValue;
/**
 * Texts To Replace
 * @default {}
 */
    textUi?: {
        /** Text
     */
    textValues?: Array<{
      /** Whether the search should respect case. True : the search is case sensitive. False : the search is case insensitive.
       * @default false
       */
      matchCase?: boolean | Expression<boolean>;
      /** If non-empty, limits the matches to slide elements only on the given slides. Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
       * @default []
       */
      pageObjectIds?: string[];
      /** The text to search for in the slide
       */
      text?: string | Expression<string> | PlaceholderValue;
      /** The text that will replace the matched text
       */
      replaceText?: string | Expression<string> | PlaceholderValue;
    }>;
  };
/**
 * Options
 * @default {}
 */
    options?: {
    /** The revision ID of the presentation required for the write request. If specified and the requiredRevisionId doesn't exactly match the presentation's current revisionId, the request will not be processed and will return a 400 bad request error.
     */
    revisionId?: string | Expression<string> | PlaceholderValue;
  };
};

export type GoogleSlidesV2PresentationReplaceTextOutput = {
  presentationId?: string;
  replies?: Array<{
    replaceAllText?: {
      occurrencesChanged?: number;
    };
  }>;
  writeControl?: {
    requiredRevisionId?: string;
  };
};

export type GoogleSlidesV2PresentationReplaceTextNode = {
  type: 'n8n-nodes-base.googleSlides';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<GoogleSlidesV2PresentationReplaceTextParams>;
  output?: Items<GoogleSlidesV2PresentationReplaceTextOutput>;
};
