// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleContacts/v1/resource_contact/operation_get.ts
/**
 * Google Contacts Node - Version 1
 * Discriminator: resource=contact, operation=get
 */


interface Credentials {
  googleContactsOAuth2Api: CredentialReference;
}

/** Get a contact */
export type GoogleContactsV1ContactGetParams = {
  resource: 'contact';
  operation: 'get';
/**
 * Contact ID
 */
    contactId?: string | Expression<string> | PlaceholderValue;
/**
 * A field mask to restrict which fields on each person are returned. Multiple fields can be specified by separating them with commas.
 * @default []
 */
    fields?: Array<'*' | 'addresses' | 'biographies' | 'birthdays' | 'coverPhotos' | 'emailAddresses' | 'events' | 'genders' | 'imClients' | 'interests' | 'locales' | 'memberships' | 'metadata' | 'names' | 'nicknames' | 'occupations' | 'organizations' | 'phoneNumbers' | 'photos' | 'relations' | 'residences' | 'sipAddresses' | 'skills' | 'urls' | 'userDefined'>;
/**
 * Whether to return the data exactly in the way it got received from the API
 * @default false
 */
    rawData?: boolean | Expression<boolean>;
};

export type GoogleContactsV1ContactGetOutput = {
  contactId?: string;
  etag?: string;
  names?: {
    displayName?: string;
    displayNameLastFirst?: string;
    familyName?: string;
    givenName?: string;
    unstructuredName?: string;
  };
  resourceName?: string;
};

export type GoogleContactsV1ContactGetNode = {
  type: 'n8n-nodes-base.googleContacts';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GoogleContactsV1ContactGetParams>;
  output?: Items<GoogleContactsV1ContactGetOutput>;
};
