// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleContacts/v1/resource_contact/operation_get_all.schema.js
/**
 * Google Contacts Node - Version 1 - Zod Schema
 * Discriminator: resource=contact, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('contact').default('contact'),
      operation: z.literal('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      fields: z.array(z.union([z.literal('*'), z.literal('addresses'), z.literal('biographies'), z.literal('birthdays'), z.literal('coverPhotos'), z.literal('emailAddresses'), z.literal('events'), z.literal('genders'), z.literal('imClients'), z.literal('interests'), z.literal('locales'), z.literal('memberships'), z.literal('metadata'), z.literal('names'), z.literal('nicknames'), z.literal('occupations'), z.literal('organizations'), z.literal('phoneNumbers'), z.literal('photos'), z.literal('relations'), z.literal('residences'), z.literal('sipAddresses'), z.literal('skills'), z.literal('urls'), z.literal('userDefined')])).optional(),
      useQuery: booleanOrExpression.optional(),
      query: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"useQuery":[true]}}, defaults: {"useQuery":false} }),
      rawData: booleanOrExpression.optional(),
      options: resolveSchema({ parameters, schema: z.object({ sortOrder: z.union([z.literal('LAST_MODIFIED_ASCENDING'), z.literal('LAST_MODIFIED_DESCENDING'), z.literal('FIRST_NAME_ASCENDING'), z.literal('LAST_NAME_ASCENDING'), expressionSchema]).optional() }), required: false, displayOptions: {"show":{"useQuery":[false]}}, defaults: {"useQuery":false} }),
    }).optional(),
  });
};
