// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/dhl/v1.schema.js
/**
 * DHL Node - Version 1 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    resource: z.unknown().optional(),
    operation: resolveSchema({ parameters, schema: z.union([z.literal('get')]), required: false, displayOptions: {"show":{"resource":["shipment"]}}, defaults: {"resource":"shipment"} }),
    trackingNumber: stringOrExpression.optional(),
    options: z.object({ recipientPostalCode: stringOrExpression.optional() }).optional(),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
