// @n8n/n8n-nodes-langchain/dist/node-definitions/nodes/n8n-nodes-langchain/openAi/v21/resource_conversation/operation_update.ts
/**
 * OpenAI Node - Version 2.1
 * Discriminator: resource=conversation, operation=update
 */


interface Credentials {
  openAiApi: CredentialReference;
}

/** Update a conversation */
export type LcOpenAiV21ConversationUpdateParams = {
  resource: 'conversation';
  operation: 'update';
/**
 * The ID of the conversation to update
 */
    conversationId?: string | Expression<string> | PlaceholderValue;
/**
 * Set of 16 key-value pairs that can be attached to an object. This can be useful for storing additional information about the object in a structured format, and querying for objects via API or the dashboard. Keys are strings with a maximum length of 64 characters. Values are strings with a maximum length of 512 characters.
 * @default {}
 */
    metadata?: IDataObject | string | Expression<string>;
};

export interface LcOpenAiV21ConversationUpdateSubnodeConfig {
  tools?: ToolInstance[];
  memory?: MemoryInstance;
}

export type LcOpenAiV21ConversationUpdateNode = {
  type: '@n8n/n8n-nodes-langchain.openAi';
  version: 2.1;
  credentials?: Credentials;
  config: NodeConfig<LcOpenAiV21ConversationUpdateParams> & { subnodes?: LcOpenAiV21ConversationUpdateSubnodeConfig };
};
