// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/quickbase/v1/resource_record/operation_create.schema.js
/**
 * Quick Base Node - Version 1 - Zod Schema
 * Discriminator: resource=record, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('record').default('record'),
      operation: z.literal('create'),
      tableId: stringOrExpression.optional(),
      columns: stringOrExpression.optional(),
      simple: booleanOrExpression.optional(),
      options: z.object({ fields: z.array(z.string()).optional(), useFieldIDs: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
