// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/quickbase/v1/resource_record/operation_upsert.ts
/**
 * Quick Base Node - Version 1
 * Discriminator: resource=record, operation=upsert
 */


interface Credentials {
  quickbaseApi: CredentialReference;
}

/** Create a new record, or update the current one if it already exists (upsert) */
export type QuickbaseV1RecordUpsertParams = {
  resource: 'record';
  operation: 'upsert';
/**
 * The table identifier
 */
    tableId?: string | Expression<string> | PlaceholderValue;
/**
 * Comma-separated list of the properties which should used as columns for the new rows
 */
    columns?: string | Expression<string> | PlaceholderValue;
/**
 * Update can use the key field on the table, or any other supported unique field
 */
    updateKey?: string | Expression<string> | PlaceholderValue;
/**
 * &lt;p&gt;You're updating records in a Quick Base table with data from an external file. In order for a merge like this to work, Quick Base needs a way to match records in the source data with corresponding records in the destination table.&lt;/p&gt;&lt;p&gt;You make this possible by choosing the field in the app table that holds unique matching values. This is called a merge field.&lt;/p&gt;. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    mergeFieldId?: string | Expression<string>;
/**
 * Whether to return a simplified version of the response instead of the raw data
 * @default true
 */
    simple?: boolean | Expression<boolean>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Specify an array of field IDs that will return data for any updates or added record. Record ID (FID 3) is always returned if any field ID is requested. Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     * @default []
     */
    fields?: string[];
    /** Whether to use Field IDs instead of Field Names in Columns
     * @default false
     */
    useFieldIDs?: boolean | Expression<boolean>;
  };
};

export type QuickbaseV1RecordUpsertNode = {
  type: 'n8n-nodes-base.quickbase';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<QuickbaseV1RecordUpsertParams>;
};
