// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/zammad/v1/resource_ticket/operation_get_all.ts
/**
 * Zammad Node - Version 1
 * Discriminator: resource=ticket, operation=getAll
 */


interface Credentials {
  zammadBasicAuthApi: CredentialReference;
  zammadTokenAuthApi: CredentialReference;
}

/** Get many groups */
export type ZammadV1TicketGetAllParams = {
  resource: 'ticket';
  operation: 'getAll';
  authentication?: 'basicAuth' | 'tokenAuth' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
};

export type ZammadV1TicketGetAllOutput = {
  article_count?: number;
  create_article_sender_id?: number;
  create_article_type_id?: number;
  created_at?: string;
  created_by_id?: number;
  customer_id?: number;
  escalation_at?: null;
  group_id?: number;
  id?: number;
  note?: null;
  number?: string;
  owner_id?: number;
  pending_time?: null;
  preferences?: {
    channel_id?: number;
  };
  priority_id?: number;
  state_id?: number;
  title?: string;
  updated_at?: string;
  updated_by_id?: number;
};

export type ZammadV1TicketGetAllNode = {
  type: 'n8n-nodes-base.zammad';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ZammadV1TicketGetAllParams>;
  output?: Items<ZammadV1TicketGetAllOutput>;
};
