// @n8n/n8n-nodes-langchain/dist/node-definitions/nodes/n8n-nodes-langchain/vectorStoreZep/v13/mode_insert.schema.js
/**
 * Zep Vector Store Node - Version 1.3 - Zod Schema
 * Discriminator: mode=insert
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, embeddingInstanceSchema, documentLoaderInstanceSchema }) {

  // Static subnode schema
  const subnodesSchema = z.object({
    embedding: z.union([embeddingInstanceSchema, z.array(embeddingInstanceSchema)]),
    documentLoader: z.union([documentLoaderInstanceSchema, z.array(documentLoaderInstanceSchema)]),
  }).strict();

  return z.object({
    parameters: z.object({
      mode: z.literal('insert'),
      ragStarterCallout: z.unknown().optional(),
      collectionName: stringOrExpression.optional(),
      embeddingBatchSize: numberOrExpression.optional(),
      options: z.object({ embeddingDimensions: numberOrExpression.optional(), isAutoEmbedded: booleanOrExpression.optional() }).optional(),
    }).optional(),
    subnodes: subnodesSchema,
  });
};
