// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/set/v32/mode_manual.schema.js
/**
 * Edit Fields (Set) Node - Version 3.2 - Zod Schema
 * Discriminator: mode=manual
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      mode: z.literal('manual').default('manual'),
      duplicateItem: booleanOrExpression.optional(),
      duplicateCount: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"duplicateItem":[true]}}, defaults: {"duplicateItem":false} }),
      fields: z.object({ values: z.array(z.object({ name: stringOrExpression.optional(), type: z.union([z.literal('stringValue'), z.literal('numberValue'), z.literal('booleanValue'), z.literal('arrayValue'), z.literal('objectValue'), expressionSchema]).optional(), stringValue: stringOrExpression.optional(), numberValue: stringOrExpression.optional(), booleanValue: z.union([z.literal('true'), z.literal('false'), expressionSchema]).optional(), arrayValue: stringOrExpression.optional(), objectValue: z.union([iDataObjectSchema, z.string()]).optional() })).optional() }).optional(),
      include: z.union([z.literal('all'), z.literal('none'), z.literal('selected'), z.literal('except'), expressionSchema]).optional(),
      includeFields: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"include":["selected"]}}, defaults: {"include":"all"} }),
      excludeFields: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"include":["except"]}}, defaults: {"include":"all"} }),
      options: z.object({ includeBinary: booleanOrExpression.optional(), stripBinary: booleanOrExpression.optional(), ignoreConversionErrors: booleanOrExpression.optional(), dotNotation: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
