// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/splitOut/v1.schema.js
/**
 * Split Out Node - Version 1 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    fieldToSplitOut: stringOrExpression.optional(),
    include: z.union([z.literal('noOtherFields'), z.literal('allOtherFields'), z.literal('selectedOtherFields'), expressionSchema]).optional(),
    fieldsToInclude: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"include":["selectedOtherFields"]}}, defaults: {"include":"noOtherFields"} }),
    options: z.object({ disableDotNotation: booleanOrExpression.optional(), destinationFieldName: stringOrExpression.optional(), includeBinary: booleanOrExpression.optional() }).optional(),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
