// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftDynamicsCrm/v1/resource_account/operation_get_all.ts
/**
 * Microsoft Dynamics CRM Node - Version 1
 * Discriminator: resource=account, operation=getAll
 */


interface Credentials {
  microsoftDynamicsOAuth2Api: CredentialReference;
}

export type MicrosoftDynamicsCrmV1AccountGetAllParams = {
  resource: 'account';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 5
 */
    limit?: number | Expression<number>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     * @default []
     */
    returnFields?: string[];
    /** Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     * @default []
     */
    expandFields?: string[];
  };
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Query to filter the results. Check &lt;a href="https://docs.microsoft.com/en-us/powerapps/developer/data-platform/webapi/query-data-web-api#filter-results" target="_blank"&gt;filters&lt;/a&gt;.
     */
    query?: string | Expression<string> | PlaceholderValue;
  };
};

export type MicrosoftDynamicsCrmV1AccountGetAllNode = {
  type: 'n8n-nodes-base.microsoftDynamicsCrm';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MicrosoftDynamicsCrmV1AccountGetAllParams>;
};
