// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/homeAssistant/v1/resource_service/operation_call.schema.js
/**
 * Home Assistant Node - Version 1 - Zod Schema
 * Discriminator: resource=service, operation=call
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('service'),
      operation: z.literal('call'),
      domain: stringOrExpression.optional(),
      service: stringOrExpression.optional(),
      serviceAttributes: z.object({ attributes: z.array(z.object({ name: stringOrExpression.optional(), value: stringOrExpression.optional() })).optional() }).optional(),
    }).optional(),
  });
};
