// n8n-nodes-base/dist/nodes/Currents/descriptions/common.descriptions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.filterTagsOption = exports.filterGroupsOption = exports.filterBranchesOption = exports.filterAuthorsOption = exports.projectRLC = void 0;
exports.projectRLC = {
    displayName: 'Project',
    name: 'projectId',
    type: 'resourceLocator',
    default: { mode: 'list', value: '' },
    required: true,
    description: 'The Currents project',
    modes: [
        {
            displayName: 'From List',
            name: 'list',
            type: 'list',
            placeholder: 'Select a project...',
            typeOptions: {
                searchListMethod: 'getProjects',
                searchable: true,
            },
        },
        {
            displayName: 'By ID',
            name: 'id',
            type: 'string',
            placeholder: 'e.g. abc123',
        },
    ],
};
/** Expression: comma-separated string → array for brackets[] query serialization */
const COMMA_TO_ARRAY_VALUE = "={{ $value && String($value).trim() ? String($value).split(',').map(v => v.trim()).filter(Boolean) : undefined }}";
exports.filterAuthorsOption = {
    displayName: 'Git Authors',
    name: 'authors',
    type: 'string',
    default: '',
    routing: {
        send: {
            type: 'query',
            property: 'authors',
            value: COMMA_TO_ARRAY_VALUE,
        },
    },
    description: 'Filter by git authors (comma-separated for multiple)',
};
exports.filterBranchesOption = {
    displayName: 'Branches',
    name: 'branches',
    type: 'string',
    default: '',
    routing: {
        send: {
            type: 'query',
            property: 'branches',
            value: COMMA_TO_ARRAY_VALUE,
        },
    },
    description: 'Filter by branches (comma-separated for multiple)',
};
exports.filterGroupsOption = {
    displayName: 'Groups',
    name: 'groups',
    type: 'string',
    default: '',
    routing: {
        send: {
            type: 'query',
            property: 'groups',
            value: COMMA_TO_ARRAY_VALUE,
        },
    },
    description: 'Filter by groups (comma-separated for multiple)',
};
exports.filterTagsOption = {
    displayName: 'Tags',
    name: 'tags',
    type: 'string',
    default: '',
    routing: {
        send: {
            type: 'query',
            property: 'tags',
            value: COMMA_TO_ARRAY_VALUE,
        },
    },
    description: 'Filter by tags (comma-separated for multiple)',
};
