// n8n-nodes-base/dist/nodes/Currents/descriptions/InstanceDescription.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceFields = exports.instanceOperations = void 0;
exports.instanceOperations = [
    {
        displayName: 'Operation',
        name: 'operation',
        type: 'options',
        noDataExpression: true,
        displayOptions: {
            show: {
                resource: ['instance'],
            },
        },
        options: [
            {
                name: 'Get',
                value: 'get',
                description: 'Get a spec file execution instance with full test results',
                routing: {
                    request: {
                        method: 'GET',
                        url: '=/instances/{{$parameter["instanceId"]}}',
                    },
                    output: {
                        postReceive: [
                            {
                                type: 'rootProperty',
                                properties: {
                                    property: 'data',
                                },
                            },
                        ],
                    },
                },
                action: 'Get an instance',
            },
        ],
        default: 'get',
    },
];
exports.instanceFields = [
    {
        displayName: 'Instance ID',
        name: 'instanceId',
        type: 'string',
        required: true,
        default: '',
        displayOptions: {
            show: {
                resource: ['instance'],
                operation: ['get'],
            },
        },
        description: 'The ID of the spec file execution instance',
    },
];
