// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/github/v11/resource_user/operation_get_user_issues.schema.js
/**
 * GitHub Node - Version 1.1 - Zod Schema
 * Discriminator: resource=user, operation=getUserIssues
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('user'),
      operation: z.literal('getUserIssues'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      getUserIssuesFilters: z.object({ mentioned: stringOrExpression.optional(), labels: stringOrExpression.optional(), since: stringOrExpression.optional(), state: z.union([z.literal('all'), z.literal('closed'), z.literal('open'), expressionSchema]).optional(), sort: z.union([z.literal('created'), z.literal('updated'), z.literal('comments'), expressionSchema]).optional(), direction: z.union([z.literal('asc'), z.literal('desc'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
