// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/github/v11/resource_user/operation_get_user_issues.ts
/**
 * GitHub Node - Version 1.1
 * Discriminator: resource=user, operation=getUserIssues
 */


interface Credentials {
  githubApi: CredentialReference;
  githubOAuth2Api: CredentialReference;
}

/** Returns the issues assigned to the user */
export type GithubV11UserGetUserIssuesParams = {
  resource: 'user';
  operation: 'getUserIssues';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    getUserIssuesFilters?: {
    /** Return only issues in which a specific user was mentioned
     */
    mentioned?: string | Expression<string> | PlaceholderValue;
    /** Return only issues with the given labels. Multiple labels can be separated by comma.
     */
    labels?: string | Expression<string> | PlaceholderValue;
    /** Return only issues updated at or after this time
     */
    since?: string | Expression<string>;
    /** The state to set
     * @default open
     */
    state?: 'all' | 'closed' | 'open' | Expression<string>;
    /** The order the issues should be returned in
     * @default created
     */
    sort?: 'created' | 'updated' | 'comments' | Expression<string>;
    /** The sort order
     * @default desc
     */
    direction?: 'asc' | 'desc' | Expression<string>;
  };
};

export type GithubV11UserGetUserIssuesNode = {
  type: 'n8n-nodes-base.github';
  version: 1.1;
  credentials?: Credentials;
  config: NodeConfig<GithubV11UserGetUserIssuesParams>;
};
