// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/iterable/v1/resource_user_list/operation_remove.schema.js
/**
 * Iterable Node - Version 1 - Zod Schema
 * Discriminator: resource=userList, operation=remove
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('userList'),
      operation: z.literal('remove'),
      listId: stringOrExpression.optional(),
      identifier: z.union([z.literal('email'), z.literal('userId'), expressionSchema]).optional(),
      value: stringOrExpression.optional(),
      additionalFields: z.object({ campaignId: numberOrExpression.optional(), channelUnsubscribe: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
