// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/jinaAi/v1/resource_reader/operation_search.schema.js
/**
 * Jina AI Node - Version 1 - Zod Schema
 * Discriminator: resource=reader, operation=search
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('reader').default('reader'),
      operation: z.literal('search'),
      searchQuery: stringOrExpression.optional(),
      simplify: booleanOrExpression.optional(),
      options: z.object({ outputFormat: z.union([z.literal('html'), z.literal(''), z.literal('markdown'), z.literal('screenshot'), z.literal('text'), expressionSchema]).optional(), siteFilter: stringOrExpression.optional(), pageNumber: numberOrExpression.optional() }).optional(),
      requestOptions: z.object({ batching: z.unknown().optional(), allowUnauthorizedCerts: z.boolean().optional(), proxy: stringOrExpression.optional(), timeout: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
