// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/nasa/v1/resource_donki_interplanetary_shock/operation_get.ts
/**
 * NASA Node - Version 1
 * Discriminator: resource=donkiInterplanetaryShock, operation=get
 */


interface Credentials {
  nasaApi: CredentialReference;
}

/** Get the Astronomy Picture of the Day */
export type NasaV1DonkiInterplanetaryShockGetParams = {
  resource: 'donkiInterplanetaryShock';
  operation: 'get';
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Start Date
     */
    startDate?: string | Expression<string>;
    /** End Date
     */
    endDate?: string | Expression<string>;
    /** The location of the geomagnetic storm
     * @default ALL
     */
    location?: 'ALL' | 'earth' | 'MESSENGER' | 'STEREO A' | 'STEREO B' | Expression<string>;
    /** The catalog of the geomagnetic storm
     * @default ALL
     */
    catalog?: 'ALL' | 'SWRC_CATALOG' | 'WINSLOW_MESSENGER_ICME_CATALOG' | Expression<string>;
  };
};

export type NasaV1DonkiInterplanetaryShockGetNode = {
  type: 'n8n-nodes-base.nasa';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<NasaV1DonkiInterplanetaryShockGetParams>;
};
