// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/compareDatasets/v2.schema.js
/**
 * Compare Datasets Node - Version 2 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    mergeByFields: z.object({ values: z.array(z.object({ field1: stringOrExpression.optional(), field2: stringOrExpression.optional() })).optional() }).optional(),
    resolve: z.union([z.literal('preferInput1'), z.literal('preferInput2'), z.literal('mix'), z.literal('includeBoth'), expressionSchema]).optional(),
    fuzzyCompare: booleanOrExpression.optional(),
    preferWhenMix: resolveSchema({ parameters, schema: z.union([z.literal('input1'), z.literal('input2'), expressionSchema]), required: false, displayOptions: {"show":{"resolve":["mix"]}}, defaults: {"resolve":"preferInput2"} }),
    exceptWhenMix: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"resolve":["mix"]}}, defaults: {"resolve":"preferInput2"} }),
    options: z.object({ skipFields: stringOrExpression.optional(), fuzzyCompare: booleanOrExpression.optional(), disableDotNotation: booleanOrExpression.optional(), multipleMatches: z.union([z.literal('first'), z.literal('all'), expressionSchema]).optional() }).optional(),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
