// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/twake/v1/resource_message/operation_send.schema.js
/**
 * Twake Node - Version 1 - Zod Schema
 * Discriminator: resource=message, operation=send
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('message').default('message'),
      operation: z.literal('send').default('send'),
      channelId: stringOrExpression.optional(),
      content: stringOrExpression.optional(),
      additionalFields: z.object({ senderIcon: stringOrExpression.optional(), senderName: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
