// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/elasticSecurity/v1/resource_case/operation_create.schema.js
/**
 * Elastic Security Node - Version 1 - Zod Schema
 * Discriminator: resource=case, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('case').default('case'),
      operation: z.literal('create').default('create'),
      title: stringOrExpression.optional(),
      connectorId: stringOrExpression.optional(),
      connectorType: z.union([z.literal('.resilient'), z.literal('.jira'), z.literal('.servicenow'), expressionSchema]).optional(),
      issueType: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"connectorType":[".jira"]}}, defaults: {"connectorType":".jira"} }),
      priority: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"connectorType":[".jira"]}}, defaults: {"connectorType":".jira"} }),
      urgency: resolveSchema({ parameters, schema: z.union([z.literal(1), z.literal(2), z.literal(3), expressionSchema]), required: false, displayOptions: {"show":{"connectorType":[".servicenow"]}}, defaults: {"connectorType":".jira"} }),
      severity: resolveSchema({ parameters, schema: z.union([z.literal(1), z.literal(2), z.literal(3), expressionSchema]), required: false, displayOptions: {"show":{"connectorType":[".servicenow"]}}, defaults: {"connectorType":".jira"} }),
      impact: resolveSchema({ parameters, schema: z.union([z.literal(1), z.literal(2), z.literal(3), expressionSchema]), required: false, displayOptions: {"show":{"connectorType":[".servicenow"]}}, defaults: {"connectorType":".jira"} }),
      category: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"connectorType":[".servicenow"]}}, defaults: {"connectorType":".jira"} }),
      issueTypes: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"connectorType":[".resilient"]}}, defaults: {"connectorType":".jira"} }),
      severityCode: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"connectorType":[".resilient"]}}, defaults: {"connectorType":".jira"} }),
      additionalFields: z.object({ description: stringOrExpression.optional(), owner: stringOrExpression.optional(), syncAlerts: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
