// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/elasticSecurity/v1/resource_case/operation_get_all.schema.js
/**
 * Elastic Security Node - Version 1 - Zod Schema
 * Discriminator: resource=case, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('case').default('case'),
      operation: z.literal('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filters: z.object({ status: z.union([z.literal('open'), z.literal('in-progress'), z.literal('closed'), expressionSchema]).optional(), tags: z.array(z.string()).optional() }).optional(),
      sortOptions: z.object({ sortOptionsProperties: z.object({ sortField: z.union([z.literal('createdAt'), z.literal('updatedAt'), expressionSchema]).optional(), sortOrder: z.union([z.literal('asc'), z.literal('desc'), expressionSchema]).optional() }).optional() }).optional(),
    }).optional(),
  });
};
