// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/azureCosmosDb/v1/resource_item/operation_create.ts
/**
 * Azure Cosmos DB Node - Version 1
 * Discriminator: resource=item, operation=create
 */


interface Credentials {
  microsoftAzureCosmosDbSharedKeyApi: CredentialReference;
}

/** Create a container */
export type AzureCosmosDbV1ItemCreateParams = {
  resource: 'item';
  operation: 'create';
/**
 * Select the container you want to use
 * @default {"mode":"list","value":""}
 */
    container?: { __rl: true; mode: 'list' | 'id'; value: string; cachedResultName?: string };
/**
 * The item contents as a JSON object
 * @hint The item requires an ID and partition key value if a custom key is set
 * @displayOptions.hide { container: [""] }
 */
    customProperties?: IDataObject | string | Expression<string>;
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type AzureCosmosDbV1ItemCreateOutput = {
  _attachments?: string;
  _etag?: string;
  _rid?: string;
  _self?: string;
  _ts?: number;
  id?: string;
};

export type AzureCosmosDbV1ItemCreateNode = {
  type: 'n8n-nodes-base.azureCosmosDb';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<AzureCosmosDbV1ItemCreateParams>;
  output?: Items<AzureCosmosDbV1ItemCreateOutput>;
};
