// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftExcel/v21/resource_table/operation_add_table.schema.js
/**
 * Microsoft Excel 365 Node - Version 2.1 - Zod Schema
 * Discriminator: resource=table, operation=addTable
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('table'),
      operation: z.literal('addTable'),
      workbook: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      worksheet: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      selectRange: z.union([z.literal('auto'), z.literal('manual'), expressionSchema]).optional(),
      range: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"selectRange":["manual"]}}, defaults: {"selectRange":"auto"} }),
      hasHeaders: booleanOrExpression.optional(),
    }).optional(),
  });
};
