// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/harvest/v1/resource_time_entry/operation_create_by_duration.ts
/**
 * Harvest Node - Version 1
 * Discriminator: resource=timeEntry, operation=createByDuration
 */


interface Credentials {
  harvestApi: CredentialReference;
  harvestOAuth2Api: CredentialReference;
}

/** Create a time entry via duration */
export type HarvestV1TimeEntryCreateByDurationParams = {
  resource: 'timeEntry';
  operation: 'createByDuration';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    accountId?: string | Expression<string>;
/**
 * The ID of the project to associate with the time entry
 */
    projectId?: string | Expression<string> | PlaceholderValue;
/**
 * The ID of the task to associate with the time entry
 */
    taskId?: string | Expression<string> | PlaceholderValue;
/**
 * The ISO 8601 formatted date the time entry was spent
 */
    spentDate?: string | Expression<string>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** The current amount of time tracked
     * @default 0
     */
    hours?: number | Expression<number>;
    /** These are notes about the time entry
     */
    notes?: string | Expression<string> | PlaceholderValue;
    /** The ID of the user to associate with the time entry. Defaults to the currently authenticated user’s ID.
     */
    user_id?: string | Expression<string> | PlaceholderValue;
  };
};

export type HarvestV1TimeEntryCreateByDurationNode = {
  type: 'n8n-nodes-base.harvest';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<HarvestV1TimeEntryCreateByDurationParams>;
};
