// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/harvest/v1/resource_time_entry/operation_get_all.ts
/**
 * Harvest Node - Version 1
 * Discriminator: resource=timeEntry, operation=getAll
 */


interface Credentials {
  harvestApi: CredentialReference;
  harvestOAuth2Api: CredentialReference;
}

/** Get data of many clients */
export type HarvestV1TimeEntryGetAllParams = {
  resource: 'timeEntry';
  operation: 'getAll';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    accountId?: string | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Only return time entries belonging to the client with the given ID
     */
    client_id?: string | Expression<string> | PlaceholderValue;
    /** Only return time entries with a spent_date on or after the given date
     */
    from?: string | Expression<string>;
    /** Whether to only return time entries that have been invoiced and false to return time entries that have not been invoiced
     * @default true
     */
    is_billed?: boolean | Expression<boolean>;
    /** Whether to only return running time entries and false to return non-running time entries
     * @default true
     */
    is_running?: boolean | Expression<boolean>;
    /** The page number to use in pagination. For instance, if you make a list request and receive 100 records, your subsequent call can include page=2 to retrieve the next page of the list. (Default: 1)
     * @default 1
     */
    page?: number | Expression<number>;
    /** Only return time entries with a spent_date on or before the given date
     */
    to?: string | Expression<string>;
    /** Only return time entries that have been updated since the given date and time
     */
    updated_since?: string | Expression<string>;
    /** Only return time entries belonging to the user with the given ID
     */
    user_id?: string | Expression<string> | PlaceholderValue;
  };
};

export type HarvestV1TimeEntryGetAllOutput = {
  billable?: boolean;
  budgeted?: boolean;
  client?: {
    currency?: string;
    id?: number;
    name?: string;
  };
  created_at?: string;
  id?: number;
  is_billed?: boolean;
  is_closed?: boolean;
  is_locked?: boolean;
  is_running?: boolean;
  project?: {
    id?: number;
    name?: string;
  };
  spent_date?: string;
  task?: {
    id?: number;
    name?: string;
  };
  task_assignment?: {
    billable?: boolean;
    created_at?: string;
    id?: number;
    updated_at?: string;
  };
  updated_at?: string;
  user?: {
    id?: number;
    name?: string;
  };
  user_assignment?: {
    created_at?: string;
    id?: number;
    is_active?: boolean;
    is_project_manager?: boolean;
    updated_at?: string;
    use_default_rates?: boolean;
  };
};

export type HarvestV1TimeEntryGetAllNode = {
  type: 'n8n-nodes-base.harvest';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<HarvestV1TimeEntryGetAllParams>;
  output?: Items<HarvestV1TimeEntryGetAllOutput>;
};
