// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/harvest/v1/resource_time_entry/operation_create_by_start_end.schema.js
/**
 * Harvest Node - Version 1 - Zod Schema
 * Discriminator: resource=timeEntry, operation=createByStartEnd
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('timeEntry'),
      operation: z.literal('createByStartEnd'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      accountId: stringOrExpression.optional(),
      projectId: stringOrExpression.optional(),
      taskId: stringOrExpression.optional(),
      spentDate: stringOrExpression.optional(),
      additionalFields: z.object({ ended_time: stringOrExpression.optional(), notes: stringOrExpression.optional(), started_time: stringOrExpression.optional(), user_id: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
