// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/harvest/v1/resource_time_entry/operation_update.ts
/**
 * Harvest Node - Version 1
 * Discriminator: resource=timeEntry, operation=update
 */


interface Credentials {
  harvestApi: CredentialReference;
  harvestOAuth2Api: CredentialReference;
}

/** Update a client */
export type HarvestV1TimeEntryUpdateParams = {
  resource: 'timeEntry';
  operation: 'update';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    accountId?: string | Expression<string>;
/**
 * The ID of the time entry to update
 */
    id?: string | Expression<string> | PlaceholderValue;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** The time the entry ended
     */
    ended_time?: string | Expression<string> | PlaceholderValue;
    /** The current amount of time tracked
     * @default 0
     */
    hours?: number | Expression<number>;
    /** These are notes about the time entry
     */
    notes?: string | Expression<string> | PlaceholderValue;
    /** The time the entry started. Defaults to the current time. Example: “8:00am”.
     */
    started_time?: string | Expression<string> | PlaceholderValue;
  };
};

export type HarvestV1TimeEntryUpdateNode = {
  type: 'n8n-nodes-base.harvest';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<HarvestV1TimeEntryUpdateParams>;
};
