// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedrive/v1/resource_product/operation_get_all.ts
/**
 * Pipedrive Node - Version 1
 * Discriminator: resource=product, operation=getAll
 */


interface Credentials {
  pipedriveApi: CredentialReference;
  pipedriveOAuth2Api: CredentialReference;
}

/** Get data of many activities */
export type PipedriveV1ProductGetAllParams = {
  resource: 'product';
  operation: 'getAll';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * By default do custom properties get returned only as ID instead of their actual name. Also option fields contain only the ID instead of their actual value. If this option gets set they get automatically resolved.
 * @default false
 */
    resolveProperties?: boolean | Expression<boolean>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
};

export type PipedriveV1ProductGetAllOutput = {
  '1e901332a0666ef17792122457262233cfdb3d15'?: null;
  '73794f1db542a544402ab6cd42aaf337d2d8e51b'?: null;
  active_flag?: boolean;
  add_time?: string;
  billing_frequency?: string;
  billing_frequency_cycles?: null;
  first_char?: string;
  id?: number;
  name?: string;
  owner_id?: {
    active_flag?: boolean;
    email?: string;
    has_pic?: number;
    id?: number;
    name?: string;
    pic_hash?: null;
    value?: number;
  };
  prices?: Array<{
    currency?: string;
    id?: number;
    notes?: string;
    product_id?: number;
  }>;
  product_variations?: Array<{
    id?: number;
    name?: string;
    prices?: Array<{
      comment?: string;
      cost?: number;
      currency?: string;
      id?: number;
      overhead_cost?: number;
      price?: number;
      price_formatted?: string;
      product_id?: number;
      product_variation_id?: number;
    }>;
    product_id?: number;
  }>;
  selectable?: boolean;
  tax?: number;
  update_time?: string;
  visible_to?: string;
};

export type PipedriveV1ProductGetAllNode = {
  type: 'n8n-nodes-base.pipedrive';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PipedriveV1ProductGetAllParams>;
  output?: Items<PipedriveV1ProductGetAllOutput>;
};
