// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftSharePoint/v1/resource_item/operation_get_all.ts
/**
 * Microsoft SharePoint Node - Version 1
 * Discriminator: resource=item, operation=getAll
 */


interface Credentials {
  microsoftSharePointOAuth2Api: CredentialReference;
}

/** Get specific items in a list or list many items */
export type MicrosoftSharePointV1ItemGetAllParams = {
  resource: 'item';
  operation: 'getAll';
/**
 * Select the site to retrieve lists from
 * @default {"mode":"list","value":""}
 */
    site?: { __rl: true; mode: 'list' | 'id'; value: string; cachedResultName?: string };
/**
 * Select the list you want to search for items in
 * @displayOptions.hide { site: [""] }
 * @default {"mode":"list","value":""}
 */
    list?: { __rl: true; mode: 'list' | 'id'; value: string; cachedResultName?: string };
/**
 * The formula will be evaluated for each record. &lt;a href="https://learn.microsoft.com/en-us/graph/filter-query-parameter"&gt;More info&lt;/a&gt;.
 * @hint If empty, all the items will be returned
 */
    filter?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** The fields you want to include in the output
     * @displayOptions.hide { /simplify: [true] }
     * @default []
     */
    fields?: Array<'contentType' | 'createdDateTime' | 'createdBy' | 'fields' | 'id' | 'lastModifiedDateTime' | 'lastModifiedBy' | 'parentReference' | 'webUrl'>;
  };
/**
 * Simplify
 * @default true
 */
    simplify?: boolean | Expression<boolean>;
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type MicrosoftSharePointV1ItemGetAllOutput = {
  '@odata.etag'?: string;
  contentType?: {
    id?: string;
    name?: string;
  };
  createdBy?: {
    user?: {
      displayName?: string;
      email?: string;
      id?: string;
    };
  };
  createdDateTime?: string;
  eTag?: string;
  fields?: {
    _ComplianceFlags?: string;
    _ComplianceTag?: string;
    _ComplianceTagUserId?: string;
    _ComplianceTagWrittenTime?: string;
    _UIVersionString?: string;
    '@odata.etag'?: string;
    AppEditorLookupId?: string;
    Attachments?: boolean;
    AuthorLookupId?: string;
    ContentType?: string;
    Created?: string;
    Edit?: string;
    EditorLookupId?: string;
    FolderChildCount?: string;
    ID?: number;
    ItemChildCount?: string;
    Modified?: string;
    Title?: string;
  };
  id?: string;
  lastModifiedBy?: {
    application?: {
      displayName?: string;
      id?: string;
    };
    user?: {
      displayName?: string;
      email?: string;
      id?: string;
    };
  };
  lastModifiedDateTime?: string;
  parentReference?: {
    id?: string;
    listId?: string;
    siteId?: string;
  };
  webUrl?: string;
};

export type MicrosoftSharePointV1ItemGetAllNode = {
  type: 'n8n-nodes-base.microsoftSharePoint';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MicrosoftSharePointV1ItemGetAllParams>;
  output?: Items<MicrosoftSharePointV1ItemGetAllOutput>;
};
