// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftSharePoint/v1/resource_item/operation_get.ts
/**
 * Microsoft SharePoint Node - Version 1
 * Discriminator: resource=item, operation=get
 */


interface Credentials {
  microsoftSharePointOAuth2Api: CredentialReference;
}

/** Retrieve an item from a list */
export type MicrosoftSharePointV1ItemGetParams = {
  resource: 'item';
  operation: 'get';
/**
 * Select the site to retrieve lists from
 * @default {"mode":"list","value":""}
 */
    site?: { __rl: true; mode: 'list' | 'id'; value: string; cachedResultName?: string };
/**
 * Select the list you want to retrieve an item from
 * @displayOptions.hide { site: [""] }
 * @default {"mode":"list","value":""}
 */
    list?: { __rl: true; mode: 'list' | 'id'; value: string; cachedResultName?: string };
/**
 * Select the item you want to get
 * @displayOptions.hide { site: [""], list: [""] }
 * @default {"mode":"list","value":""}
 */
    item?: { __rl: true; mode: 'list' | 'id'; value: string; cachedResultName?: string };
/**
 * Simplify
 * @default true
 */
    simplify?: boolean | Expression<boolean>;
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type MicrosoftSharePointV1ItemGetOutput = {
  '@odata.context'?: string;
  '@odata.etag'?: string;
  contentType?: {
    id?: string;
    name?: string;
  };
  createdBy?: {
    user?: {
      displayName?: string;
      email?: string;
      id?: string;
    };
  };
  createdDateTime?: string;
  eTag?: string;
  fields?: {
    _ComplianceFlags?: string;
    _ComplianceTag?: string;
    _ComplianceTagUserId?: string;
    _ComplianceTagWrittenTime?: string;
    _UIVersionString?: string;
    '@odata.etag'?: string;
    AuthorLookupId?: string;
    ContentType?: string;
    Created?: string;
    Edit?: string;
    EditorLookupId?: string;
    FolderChildCount?: string;
    ID?: number;
    ItemChildCount?: string;
    Modified?: string;
    Title?: string;
  };
  'fields@odata.navigationLink'?: string;
  id?: string;
  lastModifiedBy?: {
    application?: {
      displayName?: string;
      id?: string;
    };
    user?: {
      displayName?: string;
      email?: string;
      id?: string;
    };
  };
  lastModifiedDateTime?: string;
  parentReference?: {
    id?: string;
    listId?: string;
    siteId?: string;
  };
  webUrl?: string;
};

export type MicrosoftSharePointV1ItemGetNode = {
  type: 'n8n-nodes-base.microsoftSharePoint';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MicrosoftSharePointV1ItemGetParams>;
  output?: Items<MicrosoftSharePointV1ItemGetOutput>;
};
