// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleAds/v1/resource_campaign/operation_get_all.ts
/**
 * Google Ads Node - Version 1
 * Discriminator: resource=campaign, operation=getAll
 */


interface Credentials {
  googleAdsOAuth2Api: CredentialReference;
}

/** Get many campaigns linked to the specified account */
export type GoogleAdsV1CampaignGetAllParams = {
  resource: 'campaign';
  operation: 'getAll';
/**
 * Manager Customer ID
 */
    managerCustomerId?: string | Expression<string> | PlaceholderValue;
/**
 * Client Customer ID
 */
    clientCustomerId?: string | Expression<string> | PlaceholderValue;
/**
 * Additional options for fetching campaigns
 * @default {}
 */
    additionalOptions?: {
    /** Filters statistics by period
     * @default allTime
     */
    dateRange?: 'allTime' | 'TODAY' | 'YESTERDAY' | 'LAST_7_DAYS' | 'LAST_BUSINESS_WEEK' | 'THIS_MONTH' | 'LAST_MONTH' | 'LAST_14_DAYS' | 'LAST_30_DAYS' | Expression<string>;
    /** Filters campaigns by status
     * @default all
     */
    campaignStatus?: 'all' | 'ENABLED' | 'PAUSED' | 'REMOVED' | Expression<string>;
  };
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type GoogleAdsV1CampaignGetAllOutput = {
  advertisingChannelType?: string;
  amountMicros?: string;
  averageCost?: number;
  averageCpm?: number;
  costMicros?: string;
  ctr?: number;
  id?: string;
  impressions?: string;
  interactionRate?: number;
  interactions?: string;
  name?: string;
  period?: string;
  resourceName?: string;
  status?: string;
  videoViews?: string;
};

export type GoogleAdsV1CampaignGetAllNode = {
  type: 'n8n-nodes-base.googleAds';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GoogleAdsV1CampaignGetAllParams>;
  output?: Items<GoogleAdsV1CampaignGetAllOutput>;
};
