// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleAds/v1/resource_campaign/operation_get_all.schema.js
/**
 * Google Ads Node - Version 1 - Zod Schema
 * Discriminator: resource=campaign, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('campaign').default('campaign'),
      operation: z.literal('getAll').default('getAll'),
      managerCustomerId: stringOrExpression.optional(),
      clientCustomerId: stringOrExpression.optional(),
      additionalOptions: z.object({ dateRange: z.union([z.literal('allTime'), z.literal('TODAY'), z.literal('YESTERDAY'), z.literal('LAST_7_DAYS'), z.literal('LAST_BUSINESS_WEEK'), z.literal('THIS_MONTH'), z.literal('LAST_MONTH'), z.literal('LAST_14_DAYS'), z.literal('LAST_30_DAYS'), expressionSchema]).optional(), campaignStatus: z.union([z.literal('all'), z.literal('ENABLED'), z.literal('PAUSED'), z.literal('REMOVED'), expressionSchema]).optional() }).optional(),
      requestOptions: z.object({ batching: z.unknown().optional(), allowUnauthorizedCerts: z.boolean().optional(), proxy: stringOrExpression.optional(), timeout: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
