// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/trello/v1/resource_card/operation_create.schema.js
/**
 * Trello Node - Version 1 - Zod Schema
 * Discriminator: resource=card, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('card').default('card'),
      operation: z.literal('create'),
      listId: stringOrExpression.optional(),
      name: stringOrExpression.optional(),
      description: stringOrExpression.optional(),
      additionalFields: z.object({ due: stringOrExpression.optional(), dueComplete: booleanOrExpression.optional(), pos: stringOrExpression.optional(), idMembers: stringOrExpression.optional(), idLabels: stringOrExpression.optional(), urlSource: stringOrExpression.optional(), idCardSource: stringOrExpression.optional(), keepFromSource: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
