// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/trello/v1/resource_card/operation_get.ts
/**
 * Trello Node - Version 1
 * Discriminator: resource=card, operation=get
 */


interface Credentials {
  trelloApi: CredentialReference;
}

/** Get the data of an attachment */
export type TrelloV1CardGetParams = {
  resource: 'card';
  operation: 'get';
/**
 * The ID of the card
 * @default {"mode":"list","value":""}
 */
    id?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Fields to return. Either "all" or a comma-separated list: badges, checkItemStates, closed, dateLastActivity, desc, descData, due, email, idBoard, idChecklists, idLabels, idList, idMembers, idShort, idAttachmentCover, manualCoverAttachment, labels, name, pos, shortUrl, url.
     * @default all
     */
    fields?: string | Expression<string> | PlaceholderValue;
    /** Whether to return the board object the card is on
     * @default false
     */
    board?: boolean | Expression<boolean>;
    /** Fields to return. Either "all" or a comma-separated list: name, desc, descData, closed, idOrganization, pinned, url, prefs.
     * @default all
     */
    board_fields?: string | Expression<string> | PlaceholderValue;
    /** Whether to include the customFieldItems
     * @default false
     */
    customFieldItems?: boolean | Expression<boolean>;
    /** Whether to return member objects for members on the card
     * @default false
     */
    members?: boolean | Expression<boolean>;
    /** Fields to return. Either "all" or a comma-separated list: avatarHash, fullName, initials, username.
     * @default all
     */
    member_fields?: string | Expression<string> | PlaceholderValue;
    /** Whether to include pluginData on the card with the response
     * @default false
     */
    pluginData?: boolean | Expression<boolean>;
    /** Whether to include sticker models with the response
     * @default false
     */
    stickers?: boolean | Expression<boolean>;
    /** Fields to return. Either "all" or a comma-separated list of sticker fields.
     * @default all
     */
    sticker_fields?: string | Expression<string> | PlaceholderValue;
  };
};

export type TrelloV1CardGetOutput = {
  badges?: {
    attachments?: number;
    attachmentsByType?: {
      trello?: {
        board?: number;
        card?: number;
      };
    };
    checkItems?: number;
    checkItemsChecked?: number;
    checkItemsEarliestDue?: null;
    comments?: number;
    description?: boolean;
    dueComplete?: boolean;
    externalSource?: null;
    fogbugz?: string;
    lastUpdatedByAi?: boolean;
    location?: boolean;
    subscribed?: boolean;
    viewingMemberVoted?: boolean;
    votes?: number;
  };
  cardRole?: null;
  checkItemStates?: Array<{
    idCheckItem?: string;
    state?: string;
  }>;
  closed?: boolean;
  cover?: {
    brightness?: string;
    idUploadedBackground?: null;
    size?: string;
  };
  dateLastActivity?: string;
  desc?: string;
  dueComplete?: boolean;
  email?: null;
  id?: string;
  idBoard?: string;
  idChecklists?: Array<string>;
  idLabels?: Array<string>;
  idList?: string;
  idMembers?: Array<string>;
  idShort?: number;
  isTemplate?: boolean;
  labels?: Array<{
    id?: string;
    idBoard?: string;
    idOrganization?: string;
    name?: string;
    nodeId?: string;
    uses?: number;
  }>;
  manualCoverAttachment?: boolean;
  mirrorSourceId?: null;
  name?: string;
  nodeId?: string;
  pinned?: boolean;
  shortLink?: string;
  shortUrl?: string;
  subscribed?: boolean;
  url?: string;
};

export type TrelloV1CardGetNode = {
  type: 'n8n-nodes-base.trello';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<TrelloV1CardGetParams>;
  output?: Items<TrelloV1CardGetOutput>;
};
