// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleFirebaseCloudFirestore/v1/resource_collection/operation_get_all.ts
/**
 * Google Cloud Firestore Node - Version 1
 * Discriminator: resource=collection, operation=getAll
 */


interface Credentials {
  googleFirebaseCloudFirestoreOAuth2Api: CredentialReference;
  googleApi: CredentialReference;
}

/** Get many documents from a collection */
export type GoogleFirebaseCloudFirestoreV1CollectionGetAllParams = {
  resource: 'collection';
  operation: 'getAll';
  authentication?: 'googleFirebaseCloudFirestoreOAuth2Api' | 'serviceAccount' | Expression<string>;
/**
 * As displayed in firebase console URL. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    projectId?: string | Expression<string>;
/**
 * Usually the provided default value will work
 * @default (default)
 */
    database?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
};

export type GoogleFirebaseCloudFirestoreV1CollectionGetAllOutput = {
  name?: string;
};

export type GoogleFirebaseCloudFirestoreV1CollectionGetAllNode = {
  type: 'n8n-nodes-base.googleFirebaseCloudFirestore';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GoogleFirebaseCloudFirestoreV1CollectionGetAllParams>;
  output?: Items<GoogleFirebaseCloudFirestoreV1CollectionGetAllOutput>;
};
