// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/securityScorecard/v1/resource_company/operation_get_score_plan.ts
/**
 * SecurityScorecard Node - Version 1
 * Discriminator: resource=company, operation=getScorePlan
 */


interface Credentials {
  securityScorecardApi: CredentialReference;
}

/** Get company's score improvement plan */
export type SecurityScorecardV1CompanyGetScorePlanParams = {
  resource: 'company';
  operation: 'getScorePlan';
/**
 * Primary identifier of a company or scorecard, i.e. domain.
 */
    scorecardIdentifier?: string | Expression<string> | PlaceholderValue;
/**
 * Score target
 * @default 0
 */
    score?: number | Expression<number>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
};

export type SecurityScorecardV1CompanyGetScorePlanNode = {
  type: 'n8n-nodes-base.securityScorecard';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SecurityScorecardV1CompanyGetScorePlanParams>;
};
