// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/securityScorecard/v1/resource_company/operation_get_historical_score.ts
/**
 * SecurityScorecard Node - Version 1
 * Discriminator: resource=company, operation=getHistoricalScore
 */


interface Credentials {
  securityScorecardApi: CredentialReference;
}

/** Get company's historical scores */
export type SecurityScorecardV1CompanyGetHistoricalScoreParams = {
  resource: 'company';
  operation: 'getHistoricalScore';
/**
 * Primary identifier of a company or scorecard, i.e. domain.
 */
    scorecardIdentifier?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Whether to return a simplified version of the response instead of the raw data
 * @default true
 */
    simple?: boolean | Expression<boolean>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** History start date
     */
    date_from?: string | Expression<string>;
    /** History end date
     */
    date_to?: string | Expression<string>;
    /** Date granularity
     * @default daily
     */
    timing?: 'daily' | 'weekly' | 'monthly' | Expression<string>;
  };
};

export type SecurityScorecardV1CompanyGetHistoricalScoreNode = {
  type: 'n8n-nodes-base.securityScorecard';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SecurityScorecardV1CompanyGetHistoricalScoreParams>;
};
