// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/securityScorecard/v1/resource_company/operation_get_historical_score.schema.js
/**
 * SecurityScorecard Node - Version 1 - Zod Schema
 * Discriminator: resource=company, operation=getHistoricalScore
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('company').default('company'),
      operation: z.literal('getHistoricalScore'),
      scorecardIdentifier: stringOrExpression.optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      simple: booleanOrExpression.optional(),
      options: z.object({ date_from: stringOrExpression.optional(), date_to: stringOrExpression.optional(), timing: z.union([z.literal('daily'), z.literal('weekly'), z.literal('monthly'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
