// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/salesforce/v1/resource_flow/operation_get_all.ts
/**
 * Salesforce Node - Version 1
 * Discriminator: resource=flow, operation=getAll
 */


interface Credentials {
  salesforceOAuth2Api: CredentialReference;
  salesforceJwtApi: CredentialReference;
}

/** Represents an autolaunched flow */
export type SalesforceV1FlowGetAllParams = {
  resource: 'flow';
  operation: 'getAll';
/**
 * OAuth Authorization Flow
 * @default oAuth2
 */
    authentication?: 'oAuth2' | 'jwt' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
};

export type SalesforceV1FlowGetAllOutput = {
  label?: string;
  name?: string;
  type?: string;
  url?: string;
};

export type SalesforceV1FlowGetAllNode = {
  type: 'n8n-nodes-base.salesforce';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SalesforceV1FlowGetAllParams>;
  output?: Items<SalesforceV1FlowGetAllOutput>;
};
