// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/freshservice/v1/resource_release/operation_create.schema.js
/**
 * Freshservice Node - Version 1 - Zod Schema
 * Discriminator: resource=release, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('release'),
      operation: z.literal('create').default('create'),
      subject: stringOrExpression.optional(),
      releaseType: z.union([z.literal(1), z.literal(2), z.literal(3), z.literal(4), expressionSchema]).optional(),
      priority: z.union([z.literal(1), z.literal(2), z.literal(3), z.literal(4), expressionSchema]).optional(),
      status: z.union([z.literal(1), z.literal(2), z.literal(3), z.literal(4), z.literal(5), expressionSchema]).optional(),
      plannedStartDate: stringOrExpression.optional(),
      plannedEndDate: stringOrExpression.optional(),
      additionalFields: z.object({ department_id: stringOrExpression.optional(), description: stringOrExpression.optional(), group_id: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
