// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/freshservice/v1/resource_release/operation_create.ts
/**
 * Freshservice Node - Version 1
 * Discriminator: resource=release, operation=create
 */


interface Credentials {
  freshserviceApi: CredentialReference;
}

/** Create an agent */
export type FreshserviceV1ReleaseCreateParams = {
  resource: 'release';
  operation: 'create';
/**
 * Subject
 */
    subject?: string | Expression<string> | PlaceholderValue;
/**
 * Release Type
 * @default 1
 */
    releaseType?: 1 | 2 | 3 | 4 | Expression<number>;
/**
 * Priority
 * @default 1
 */
    priority?: 1 | 2 | 3 | 4 | Expression<number>;
/**
 * Status
 * @default 1
 */
    status?: 1 | 2 | 3 | 4 | 5 | Expression<number>;
/**
 * Planned Start Date
 */
    plannedStartDate?: string | Expression<string>;
/**
 * Planned End Date
 */
    plannedEndDate?: string | Expression<string>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** ID of the department initiating the release. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    department_id?: string | Expression<string>;
    /** HTML supported
     */
    description?: string | Expression<string> | PlaceholderValue;
    /** ID of the agent group to which the release is assigned. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    group_id?: string | Expression<string>;
  };
};

export type FreshserviceV1ReleaseCreateNode = {
  type: 'n8n-nodes-base.freshservice';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<FreshserviceV1ReleaseCreateParams>;
};
