// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/notion/v2/resource_block/operation_append.ts
/**
 * Notion Node - Version 2
 * Discriminator: resource=block, operation=append
 */


interface Credentials {
  notionApi: CredentialReference;
}

/** Append a block */
export type NotionV2BlockAppendParams = {
  resource: 'block';
  operation: 'append';
/**
 * The Notion Block to append blocks to
 * @default {"mode":"url","value":""}
 */
    blockId?: { __rl: true; mode: 'url' | 'id'; value: string; cachedResultName?: string };
/**
 * Blocks
 * @default {}
 */
    blockUi?: {
        /** Block
     */
    blockValues?: Array<{
      /** Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
       * @default paragraph
       */
      type?: string | Expression<string>;
      /** Rich Text
       * @displayOptions.show { type: ["paragraph"] }
       * @default false
       */
      richText?: boolean | Expression<boolean>;
      /** Text
       * @displayOptions.show { type: ["paragraph"], richText: [false] }
       */
      textContent?: string | Expression<string> | PlaceholderValue;
      /** Rich text in the block
       * @displayOptions.show { type: ["paragraph"], richText: [true] }
       * @default {}
       */
      text?: {
        /** Text
     */
    text?: Array<{
      /** Type
       * @default text
       */
      textType?: 'equation' | 'mention' | 'text' | Expression<string>;
      /** Text content. This field contains the actual content of your text and is probably the field you'll use most often.
       * @displayOptions.show { textType: ["text"] }
       */
      text?: string | Expression<string> | PlaceholderValue;
      /** Is Link
       * @displayOptions.show { textType: ["text"] }
       * @default false
       */
      isLink?: boolean | Expression<boolean>;
      /** The URL that this link points to
       * @displayOptions.show { textType: ["text"], isLink: [true] }
       */
      textLink?: string | Expression<string> | PlaceholderValue;
      /** An inline mention of a user, page, database, or date. In the app these are created by typing @ followed by the name of a user, page, database, or a date.
       * @displayOptions.show { textType: ["mention"] }
       */
      mentionType?: 'database' | 'date' | 'page' | 'user' | Expression<string>;
      /** The ID of the user being mentioned. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
       * @displayOptions.show { mentionType: ["user"] }
       */
      user?: string | Expression<string>;
      /** The ID of the page being mentioned
       * @displayOptions.show { mentionType: ["page"] }
       */
      page?: string | Expression<string> | PlaceholderValue;
      /** The Notion Database being mentioned
       * @displayOptions.show { mentionType: ["database"] }
       * @default {"mode":"list","value":""}
       */
      database?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
      /** Whether or not you want to define a date range
       * @displayOptions.show { mentionType: ["date"] }
       * @default false
       */
      range?: boolean | Expression<boolean>;
      /** An ISO 8601 format date, with optional time
       * @displayOptions.show { mentionType: ["date"], range: [false] }
       */
      date?: string | Expression<string>;
      /** An ISO 8601 format date, with optional time
       * @displayOptions.show { mentionType: ["date"], range: [true] }
       */
      dateStart?: string | Expression<string>;
      /** An ISO 8601 formatted date, with optional time. Represents the end of a date range.
       * @displayOptions.show { range: [true], mentionType: ["date"] }
       */
      dateEnd?: string | Expression<string>;
      /** Expression
       * @displayOptions.show { textType: ["equation"] }
       */
      expression?: string | Expression<string> | PlaceholderValue;
      /** All annotations that apply to this rich text
       * @default {}
       */
      annotationUi?: {
    /** Whether the text is bolded
     * @default false
     */
    bold?: boolean | Expression<boolean>;
    /** Whether the text is italicized
     * @default false
     */
    italic?: boolean | Expression<boolean>;
    /** Whether the text is struck through
     * @default false
     */
    strikethrough?: boolean | Expression<boolean>;
    /** Whether the text is underlined
     * @default false
     */
    underline?: boolean | Expression<boolean>;
    /** Whether the text is code style
     * @default false
     */
    code?: boolean | Expression<boolean>;
    /** Color of the text
     */
    color?: 'default' | 'gray' | 'brown' | 'orange' | 'yellow' | 'green' | 'blue' | 'purple' | 'pink' | 'red' | 'gray_background' | 'brown_background' | 'orange_background' | 'yellow_background' | 'green_background' | 'blue_background' | 'purple_background' | 'pink_background' | 'red_background' | Expression<string>;
  };
    }>;
  };
      /** Rich Text
       * @displayOptions.show { type: ["heading_1"] }
       * @default false
       */
      richText?: boolean | Expression<boolean>;
      /** Text
       * @displayOptions.show { type: ["heading_1"], richText: [false] }
       */
      textContent?: string | Expression<string> | PlaceholderValue;
      /** Rich text in the block
       * @displayOptions.show { type: ["heading_1"], richText: [true] }
       * @default {}
       */
      text?: {
        /** Text
     */
    text?: Array<{
      /** Type
       * @default text
       */
      textType?: 'equation' | 'mention' | 'text' | Expression<string>;
      /** Text content. This field contains the actual content of your text and is probably the field you'll use most often.
       * @displayOptions.show { textType: ["text"] }
       */
      text?: string | Expression<string> | PlaceholderValue;
      /** Is Link
       * @displayOptions.show { textType: ["text"] }
       * @default false
       */
      isLink?: boolean | Expression<boolean>;
      /** The URL that this link points to
       * @displayOptions.show { textType: ["text"], isLink: [true] }
       */
      textLink?: string | Expression<string> | PlaceholderValue;
      /** An inline mention of a user, page, database, or date. In the app these are created by typing @ followed by the name of a user, page, database, or a date.
       * @displayOptions.show { textType: ["mention"] }
       */
      mentionType?: 'database' | 'date' | 'page' | 'user' | Expression<string>;
      /** The ID of the user being mentioned. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
       * @displayOptions.show { mentionType: ["user"] }
       */
      user?: string | Expression<string>;
      /** The ID of the page being mentioned
       * @displayOptions.show { mentionType: ["page"] }
       */
      page?: string | Expression<string> | PlaceholderValue;
      /** The Notion Database being mentioned
       * @displayOptions.show { mentionType: ["database"] }
       * @default {"mode":"list","value":""}
       */
      database?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
      /** Whether or not you want to define a date range
       * @displayOptions.show { mentionType: ["date"] }
       * @default false
       */
      range?: boolean | Expression<boolean>;
      /** An ISO 8601 format date, with optional time
       * @displayOptions.show { mentionType: ["date"], range: [false] }
       */
      date?: string | Expression<string>;
      /** An ISO 8601 format date, with optional time
       * @displayOptions.show { mentionType: ["date"], range: [true] }
       */
      dateStart?: string | Expression<string>;
      /** An ISO 8601 formatted date, with optional time. Represents the end of a date range.
       * @displayOptions.show { range: [true], mentionType: ["date"] }
       */
      dateEnd?: string | Expression<string>;
      /** Expression
       * @displayOptions.show { textType: ["equation"] }
       */
      expression?: string | Expression<string> | PlaceholderValue;
      /** All annotations that apply to this rich text
       * @default {}
       */
      annotationUi?: {
    /** Whether the text is bolded
     * @default false
     */
    bold?: boolean | Expression<boolean>;
    /** Whether the text is italicized
     * @default false
     */
    italic?: boolean | Expression<boolean>;
    /** Whether the text is struck through
     * @default false
     */
    strikethrough?: boolean | Expression<boolean>;
    /** Whether the text is underlined
     * @default false
     */
    underline?: boolean | Expression<boolean>;
    /** Whether the text is code style
     * @default false
     */
    code?: boolean | Expression<boolean>;
    /** Color of the text
     */
    color?: 'default' | 'gray' | 'brown' | 'orange' | 'yellow' | 'green' | 'blue' | 'purple' | 'pink' | 'red' | 'gray_background' | 'brown_background' | 'orange_background' | 'yellow_background' | 'green_background' | 'blue_background' | 'purple_background' | 'pink_background' | 'red_background' | Expression<string>;
  };
    }>;
  };
      /** Rich Text
       * @displayOptions.show { type: ["heading_2"] }
       * @default false
       */
      richText?: boolean | Expression<boolean>;
      /** Text
       * @displayOptions.show { type: ["heading_2"], richText: [false] }
       */
      textContent?: string | Expression<string> | PlaceholderValue;
      /** Rich text in the block
       * @displayOptions.show { type: ["heading_2"], richText: [true] }
       * @default {}
       */
      text?: {
        /** Text
     */
    text?: Array<{
      /** Type
       * @default text
       */
      textType?: 'equation' | 'mention' | 'text' | Expression<string>;
      /** Text content. This field contains the actual content of your text and is probably the field you'll use most often.
       * @displayOptions.show { textType: ["text"] }
       */
      text?: string | Expression<string> | PlaceholderValue;
      /** Is Link
       * @displayOptions.show { textType: ["text"] }
       * @default false
       */
      isLink?: boolean | Expression<boolean>;
      /** The URL that this link points to
       * @displayOptions.show { textType: ["text"], isLink: [true] }
       */
      textLink?: string | Expression<string> | PlaceholderValue;
      /** An inline mention of a user, page, database, or date. In the app these are created by typing @ followed by the name of a user, page, database, or a date.
       * @displayOptions.show { textType: ["mention"] }
       */
      mentionType?: 'database' | 'date' | 'page' | 'user' | Expression<string>;
      /** The ID of the user being mentioned. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
       * @displayOptions.show { mentionType: ["user"] }
       */
      user?: string | Expression<string>;
      /** The ID of the page being mentioned
       * @displayOptions.show { mentionType: ["page"] }
       */
      page?: string | Expression<string> | PlaceholderValue;
      /** The Notion Database being mentioned
       * @displayOptions.show { mentionType: ["database"] }
       * @default {"mode":"list","value":""}
       */
      database?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
      /** Whether or not you want to define a date range
       * @displayOptions.show { mentionType: ["date"] }
       * @default false
       */
      range?: boolean | Expression<boolean>;
      /** An ISO 8601 format date, with optional time
       * @displayOptions.show { mentionType: ["date"], range: [false] }
       */
      date?: string | Expression<string>;
      /** An ISO 8601 format date, with optional time
       * @displayOptions.show { mentionType: ["date"], range: [true] }
       */
      dateStart?: string | Expression<string>;
      /** An ISO 8601 formatted date, with optional time. Represents the end of a date range.
       * @displayOptions.show { range: [true], mentionType: ["date"] }
       */
      dateEnd?: string | Expression<string>;
      /** Expression
       * @displayOptions.show { textType: ["equation"] }
       */
      expression?: string | Expression<string> | PlaceholderValue;
      /** All annotations that apply to this rich text
       * @default {}
       */
      annotationUi?: {
    /** Whether the text is bolded
     * @default false
     */
    bold?: boolean | Expression<boolean>;
    /** Whether the text is italicized
     * @default false
     */
    italic?: boolean | Expression<boolean>;
    /** Whether the text is struck through
     * @default false
     */
    strikethrough?: boolean | Expression<boolean>;
    /** Whether the text is underlined
     * @default false
     */
    underline?: boolean | Expression<boolean>;
    /** Whether the text is code style
     * @default false
     */
    code?: boolean | Expression<boolean>;
    /** Color of the text
     */
    color?: 'default' | 'gray' | 'brown' | 'orange' | 'yellow' | 'green' | 'blue' | 'purple' | 'pink' | 'red' | 'gray_background' | 'brown_background' | 'orange_background' | 'yellow_background' | 'green_background' | 'blue_background' | 'purple_background' | 'pink_background' | 'red_background' | Expression<string>;
  };
    }>;
  };
      /** Rich Text
       * @displayOptions.show { type: ["heading_3"] }
       * @default false
       */
      richText?: boolean | Expression<boolean>;
      /** Text
       * @displayOptions.show { type: ["heading_3"], richText: [false] }
       */
      textContent?: string | Expression<string> | PlaceholderValue;
      /** Rich text in the block
       * @displayOptions.show { type: ["heading_3"], richText: [true] }
       * @default {}
       */
      text?: {
        /** Text
     */
    text?: Array<{
      /** Type
       * @default text
       */
      textType?: 'equation' | 'mention' | 'text' | Expression<string>;
      /** Text content. This field contains the actual content of your text and is probably the field you'll use most often.
       * @displayOptions.show { textType: ["text"] }
       */
      text?: string | Expression<string> | PlaceholderValue;
      /** Is Link
       * @displayOptions.show { textType: ["text"] }
       * @default false
       */
      isLink?: boolean | Expression<boolean>;
      /** The URL that this link points to
       * @displayOptions.show { textType: ["text"], isLink: [true] }
       */
      textLink?: string | Expression<string> | PlaceholderValue;
      /** An inline mention of a user, page, database, or date. In the app these are created by typing @ followed by the name of a user, page, database, or a date.
       * @displayOptions.show { textType: ["mention"] }
       */
      mentionType?: 'database' | 'date' | 'page' | 'user' | Expression<string>;
      /** The ID of the user being mentioned. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
       * @displayOptions.show { mentionType: ["user"] }
       */
      user?: string | Expression<string>;
      /** The ID of the page being mentioned
       * @displayOptions.show { mentionType: ["page"] }
       */
      page?: string | Expression<string> | PlaceholderValue;
      /** The Notion Database being mentioned
       * @displayOptions.show { mentionType: ["database"] }
       * @default {"mode":"list","value":""}
       */
      database?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
      /** Whether or not you want to define a date range
       * @displayOptions.show { mentionType: ["date"] }
       * @default false
       */
      range?: boolean | Expression<boolean>;
      /** An ISO 8601 format date, with optional time
       * @displayOptions.show { mentionType: ["date"], range: [false] }
       */
      date?: string | Expression<string>;
      /** An ISO 8601 format date, with optional time
       * @displayOptions.show { mentionType: ["date"], range: [true] }
       */
      dateStart?: string | Expression<string>;
      /** An ISO 8601 formatted date, with optional time. Represents the end of a date range.
       * @displayOptions.show { range: [true], mentionType: ["date"] }
       */
      dateEnd?: string | Expression<string>;
      /** Expression
       * @displayOptions.show { textType: ["equation"] }
       */
      expression?: string | Expression<string> | PlaceholderValue;
      /** All annotations that apply to this rich text
       * @default {}
       */
      annotationUi?: {
    /** Whether the text is bolded
     * @default false
     */
    bold?: boolean | Expression<boolean>;
    /** Whether the text is italicized
     * @default false
     */
    italic?: boolean | Expression<boolean>;
    /** Whether the text is struck through
     * @default false
     */
    strikethrough?: boolean | Expression<boolean>;
    /** Whether the text is underlined
     * @default false
     */
    underline?: boolean | Expression<boolean>;
    /** Whether the text is code style
     * @default false
     */
    code?: boolean | Expression<boolean>;
    /** Color of the text
     */
    color?: 'default' | 'gray' | 'brown' | 'orange' | 'yellow' | 'green' | 'blue' | 'purple' | 'pink' | 'red' | 'gray_background' | 'brown_background' | 'orange_background' | 'yellow_background' | 'green_background' | 'blue_background' | 'purple_background' | 'pink_background' | 'red_background' | Expression<string>;
  };
    }>;
  };
      /** Rich Text
       * @displayOptions.show { type: ["toggle"] }
       * @default false
       */
      richText?: boolean | Expression<boolean>;
      /** Text
       * @displayOptions.show { type: ["toggle"], richText: [false] }
       */
      textContent?: string | Expression<string> | PlaceholderValue;
      /** Rich text in the block
       * @displayOptions.show { type: ["toggle"], richText: [true] }
       * @default {}
       */
      text?: {
        /** Text
     */
    text?: Array<{
      /** Type
       * @default text
       */
      textType?: 'equation' | 'mention' | 'text' | Expression<string>;
      /** Text content. This field contains the actual content of your text and is probably the field you'll use most often.
       * @displayOptions.show { textType: ["text"] }
       */
      text?: string | Expression<string> | PlaceholderValue;
      /** Is Link
       * @displayOptions.show { textType: ["text"] }
       * @default false
       */
      isLink?: boolean | Expression<boolean>;
      /** The URL that this link points to
       * @displayOptions.show { textType: ["text"], isLink: [true] }
       */
      textLink?: string | Expression<string> | PlaceholderValue;
      /** An inline mention of a user, page, database, or date. In the app these are created by typing @ followed by the name of a user, page, database, or a date.
       * @displayOptions.show { textType: ["mention"] }
       */
      mentionType?: 'database' | 'date' | 'page' | 'user' | Expression<string>;
      /** The ID of the user being mentioned. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
       * @displayOptions.show { mentionType: ["user"] }
       */
      user?: string | Expression<string>;
      /** The ID of the page being mentioned
       * @displayOptions.show { mentionType: ["page"] }
       */
      page?: string | Expression<string> | PlaceholderValue;
      /** The Notion Database being mentioned
       * @displayOptions.show { mentionType: ["database"] }
       * @default {"mode":"list","value":""}
       */
      database?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
      /** Whether or not you want to define a date range
       * @displayOptions.show { mentionType: ["date"] }
       * @default false
       */
      range?: boolean | Expression<boolean>;
      /** An ISO 8601 format date, with optional time
       * @displayOptions.show { mentionType: ["date"], range: [false] }
       */
      date?: string | Expression<string>;
      /** An ISO 8601 format date, with optional time
       * @displayOptions.show { mentionType: ["date"], range: [true] }
       */
      dateStart?: string | Expression<string>;
      /** An ISO 8601 formatted date, with optional time. Represents the end of a date range.
       * @displayOptions.show { range: [true], mentionType: ["date"] }
       */
      dateEnd?: string | Expression<string>;
      /** Expression
       * @displayOptions.show { textType: ["equation"] }
       */
      expression?: string | Expression<string> | PlaceholderValue;
      /** All annotations that apply to this rich text
       * @default {}
       */
      annotationUi?: {
    /** Whether the text is bolded
     * @default false
     */
    bold?: boolean | Expression<boolean>;
    /** Whether the text is italicized
     * @default false
     */
    italic?: boolean | Expression<boolean>;
    /** Whether the text is struck through
     * @default false
     */
    strikethrough?: boolean | Expression<boolean>;
    /** Whether the text is underlined
     * @default false
     */
    underline?: boolean | Expression<boolean>;
    /** Whether the text is code style
     * @default false
     */
    code?: boolean | Expression<boolean>;
    /** Color of the text
     */
    color?: 'default' | 'gray' | 'brown' | 'orange' | 'yellow' | 'green' | 'blue' | 'purple' | 'pink' | 'red' | 'gray_background' | 'brown_background' | 'orange_background' | 'yellow_background' | 'green_background' | 'blue_background' | 'purple_background' | 'pink_background' | 'red_background' | Expression<string>;
  };
    }>;
  };
      /** Whether the to_do is checked or not
       * @displayOptions.show { type: ["to_do"] }
       * @default false
       */
      checked?: boolean | Expression<boolean>;
      /** Rich Text
       * @displayOptions.show { type: ["to_do"] }
       * @default false
       */
      richText?: boolean | Expression<boolean>;
      /** Text
       * @displayOptions.show { type: ["to_do"], richText: [false] }
       */
      textContent?: string | Expression<string> | PlaceholderValue;
      /** Rich text in the block
       * @displayOptions.show { type: ["to_do"], richText: [true] }
       * @default {}
       */
      text?: {
        /** Text
     */
    text?: Array<{
      /** Type
       * @default text
       */
      textType?: 'equation' | 'mention' | 'text' | Expression<string>;
      /** Text content. This field contains the actual content of your text and is probably the field you'll use most often.
       * @displayOptions.show { textType: ["text"] }
       */
      text?: string | Expression<string> | PlaceholderValue;
      /** Is Link
       * @displayOptions.show { textType: ["text"] }
       * @default false
       */
      isLink?: boolean | Expression<boolean>;
      /** The URL that this link points to
       * @displayOptions.show { textType: ["text"], isLink: [true] }
       */
      textLink?: string | Expression<string> | PlaceholderValue;
      /** An inline mention of a user, page, database, or date. In the app these are created by typing @ followed by the name of a user, page, database, or a date.
       * @displayOptions.show { textType: ["mention"] }
       */
      mentionType?: 'database' | 'date' | 'page' | 'user' | Expression<string>;
      /** The ID of the user being mentioned. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
       * @displayOptions.show { mentionType: ["user"] }
       */
      user?: string | Expression<string>;
      /** The ID of the page being mentioned
       * @displayOptions.show { mentionType: ["page"] }
       */
      page?: string | Expression<string> | PlaceholderValue;
      /** The Notion Database being mentioned
       * @displayOptions.show { mentionType: ["database"] }
       * @default {"mode":"list","value":""}
       */
      database?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
      /** Whether or not you want to define a date range
       * @displayOptions.show { mentionType: ["date"] }
       * @default false
       */
      range?: boolean | Expression<boolean>;
      /** An ISO 8601 format date, with optional time
       * @displayOptions.show { mentionType: ["date"], range: [false] }
       */
      date?: string | Expression<string>;
      /** An ISO 8601 format date, with optional time
       * @displayOptions.show { mentionType: ["date"], range: [true] }
       */
      dateStart?: string | Expression<string>;
      /** An ISO 8601 formatted date, with optional time. Represents the end of a date range.
       * @displayOptions.show { range: [true], mentionType: ["date"] }
       */
      dateEnd?: string | Expression<string>;
      /** Expression
       * @displayOptions.show { textType: ["equation"] }
       */
      expression?: string | Expression<string> | PlaceholderValue;
      /** All annotations that apply to this rich text
       * @default {}
       */
      annotationUi?: {
    /** Whether the text is bolded
     * @default false
     */
    bold?: boolean | Expression<boolean>;
    /** Whether the text is italicized
     * @default false
     */
    italic?: boolean | Expression<boolean>;
    /** Whether the text is struck through
     * @default false
     */
    strikethrough?: boolean | Expression<boolean>;
    /** Whether the text is underlined
     * @default false
     */
    underline?: boolean | Expression<boolean>;
    /** Whether the text is code style
     * @default false
     */
    code?: boolean | Expression<boolean>;
    /** Color of the text
     */
    color?: 'default' | 'gray' | 'brown' | 'orange' | 'yellow' | 'green' | 'blue' | 'purple' | 'pink' | 'red' | 'gray_background' | 'brown_background' | 'orange_background' | 'yellow_background' | 'green_background' | 'blue_background' | 'purple_background' | 'pink_background' | 'red_background' | Expression<string>;
  };
    }>;
  };
      /** Plain text of page title
       * @displayOptions.show { type: ["child_page"] }
       */
      title?: string | Expression<string> | PlaceholderValue;
      /** Rich Text
       * @displayOptions.show { type: ["bulleted_list_item"] }
       * @default false
       */
      richText?: boolean | Expression<boolean>;
      /** Text
       * @displayOptions.show { type: ["bulleted_list_item"], richText: [false] }
       */
      textContent?: string | Expression<string> | PlaceholderValue;
      /** Rich text in the block
       * @displayOptions.show { type: ["bulleted_list_item"], richText: [true] }
       * @default {}
       */
      text?: {
        /** Text
     */
    text?: Array<{
      /** Type
       * @default text
       */
      textType?: 'equation' | 'mention' | 'text' | Expression<string>;
      /** Text content. This field contains the actual content of your text and is probably the field you'll use most often.
       * @displayOptions.show { textType: ["text"] }
       */
      text?: string | Expression<string> | PlaceholderValue;
      /** Is Link
       * @displayOptions.show { textType: ["text"] }
       * @default false
       */
      isLink?: boolean | Expression<boolean>;
      /** The URL that this link points to
       * @displayOptions.show { textType: ["text"], isLink: [true] }
       */
      textLink?: string | Expression<string> | PlaceholderValue;
      /** An inline mention of a user, page, database, or date. In the app these are created by typing @ followed by the name of a user, page, database, or a date.
       * @displayOptions.show { textType: ["mention"] }
       */
      mentionType?: 'database' | 'date' | 'page' | 'user' | Expression<string>;
      /** The ID of the user being mentioned. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
       * @displayOptions.show { mentionType: ["user"] }
       */
      user?: string | Expression<string>;
      /** The ID of the page being mentioned
       * @displayOptions.show { mentionType: ["page"] }
       */
      page?: string | Expression<string> | PlaceholderValue;
      /** The Notion Database being mentioned
       * @displayOptions.show { mentionType: ["database"] }
       * @default {"mode":"list","value":""}
       */
      database?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
      /** Whether or not you want to define a date range
       * @displayOptions.show { mentionType: ["date"] }
       * @default false
       */
      range?: boolean | Expression<boolean>;
      /** An ISO 8601 format date, with optional time
       * @displayOptions.show { mentionType: ["date"], range: [false] }
       */
      date?: string | Expression<string>;
      /** An ISO 8601 format date, with optional time
       * @displayOptions.show { mentionType: ["date"], range: [true] }
       */
      dateStart?: string | Expression<string>;
      /** An ISO 8601 formatted date, with optional time. Represents the end of a date range.
       * @displayOptions.show { range: [true], mentionType: ["date"] }
       */
      dateEnd?: string | Expression<string>;
      /** Expression
       * @displayOptions.show { textType: ["equation"] }
       */
      expression?: string | Expression<string> | PlaceholderValue;
      /** All annotations that apply to this rich text
       * @default {}
       */
      annotationUi?: {
    /** Whether the text is bolded
     * @default false
     */
    bold?: boolean | Expression<boolean>;
    /** Whether the text is italicized
     * @default false
     */
    italic?: boolean | Expression<boolean>;
    /** Whether the text is struck through
     * @default false
     */
    strikethrough?: boolean | Expression<boolean>;
    /** Whether the text is underlined
     * @default false
     */
    underline?: boolean | Expression<boolean>;
    /** Whether the text is code style
     * @default false
     */
    code?: boolean | Expression<boolean>;
    /** Color of the text
     */
    color?: 'default' | 'gray' | 'brown' | 'orange' | 'yellow' | 'green' | 'blue' | 'purple' | 'pink' | 'red' | 'gray_background' | 'brown_background' | 'orange_background' | 'yellow_background' | 'green_background' | 'blue_background' | 'purple_background' | 'pink_background' | 'red_background' | Expression<string>;
  };
    }>;
  };
      /** Rich Text
       * @displayOptions.show { type: ["numbered_list_item"] }
       * @default false
       */
      richText?: boolean | Expression<boolean>;
      /** Text
       * @displayOptions.show { type: ["numbered_list_item"], richText: [false] }
       */
      textContent?: string | Expression<string> | PlaceholderValue;
      /** Rich text in the block
       * @displayOptions.show { type: ["numbered_list_item"], richText: [true] }
       * @default {}
       */
      text?: {
        /** Text
     */
    text?: Array<{
      /** Type
       * @default text
       */
      textType?: 'equation' | 'mention' | 'text' | Expression<string>;
      /** Text content. This field contains the actual content of your text and is probably the field you'll use most often.
       * @displayOptions.show { textType: ["text"] }
       */
      text?: string | Expression<string> | PlaceholderValue;
      /** Is Link
       * @displayOptions.show { textType: ["text"] }
       * @default false
       */
      isLink?: boolean | Expression<boolean>;
      /** The URL that this link points to
       * @displayOptions.show { textType: ["text"], isLink: [true] }
       */
      textLink?: string | Expression<string> | PlaceholderValue;
      /** An inline mention of a user, page, database, or date. In the app these are created by typing @ followed by the name of a user, page, database, or a date.
       * @displayOptions.show { textType: ["mention"] }
       */
      mentionType?: 'database' | 'date' | 'page' | 'user' | Expression<string>;
      /** The ID of the user being mentioned. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
       * @displayOptions.show { mentionType: ["user"] }
       */
      user?: string | Expression<string>;
      /** The ID of the page being mentioned
       * @displayOptions.show { mentionType: ["page"] }
       */
      page?: string | Expression<string> | PlaceholderValue;
      /** The Notion Database being mentioned
       * @displayOptions.show { mentionType: ["database"] }
       * @default {"mode":"list","value":""}
       */
      database?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
      /** Whether or not you want to define a date range
       * @displayOptions.show { mentionType: ["date"] }
       * @default false
       */
      range?: boolean | Expression<boolean>;
      /** An ISO 8601 format date, with optional time
       * @displayOptions.show { mentionType: ["date"], range: [false] }
       */
      date?: string | Expression<string>;
      /** An ISO 8601 format date, with optional time
       * @displayOptions.show { mentionType: ["date"], range: [true] }
       */
      dateStart?: string | Expression<string>;
      /** An ISO 8601 formatted date, with optional time. Represents the end of a date range.
       * @displayOptions.show { range: [true], mentionType: ["date"] }
       */
      dateEnd?: string | Expression<string>;
      /** Expression
       * @displayOptions.show { textType: ["equation"] }
       */
      expression?: string | Expression<string> | PlaceholderValue;
      /** All annotations that apply to this rich text
       * @default {}
       */
      annotationUi?: {
    /** Whether the text is bolded
     * @default false
     */
    bold?: boolean | Expression<boolean>;
    /** Whether the text is italicized
     * @default false
     */
    italic?: boolean | Expression<boolean>;
    /** Whether the text is struck through
     * @default false
     */
    strikethrough?: boolean | Expression<boolean>;
    /** Whether the text is underlined
     * @default false
     */
    underline?: boolean | Expression<boolean>;
    /** Whether the text is code style
     * @default false
     */
    code?: boolean | Expression<boolean>;
    /** Color of the text
     */
    color?: 'default' | 'gray' | 'brown' | 'orange' | 'yellow' | 'green' | 'blue' | 'purple' | 'pink' | 'red' | 'gray_background' | 'brown_background' | 'orange_background' | 'yellow_background' | 'green_background' | 'blue_background' | 'purple_background' | 'pink_background' | 'red_background' | Expression<string>;
  };
    }>;
  };
      /** Image file reference
       * @displayOptions.show { type: ["image"] }
       */
      url?: string | Expression<string> | PlaceholderValue;
    }>;
  };
};

export type NotionV2BlockAppendOutput = {
  has_more?: boolean;
  next_cursor?: null;
  object?: string;
  request_id?: string;
  results?: Array<{
    archived?: boolean;
    created_by?: {
      id?: string;
      object?: string;
    };
    created_time?: string;
    has_children?: boolean;
    id?: string;
    in_trash?: boolean;
    last_edited_by?: {
      id?: string;
      object?: string;
    };
    last_edited_time?: string;
    object?: string;
    paragraph?: {
      color?: string;
      text?: Array<{
        annotations?: {
          bold?: boolean;
          code?: boolean;
          color?: string;
          italic?: boolean;
          strikethrough?: boolean;
          underline?: boolean;
        };
        plain_text?: string;
        text?: {
          content?: string;
        };
        type?: string;
      }>;
    };
    parent?: {
      page_id?: string;
      type?: string;
    };
    type?: string;
  }>;
};

export type NotionV2BlockAppendNode = {
  type: 'n8n-nodes-base.notion';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<NotionV2BlockAppendParams>;
  output?: Items<NotionV2BlockAppendOutput>;
};
