// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleDrive/v2/resource_file/operation_update.ts
/**
 * Google Drive Node - Version 2
 * Discriminator: resource=file, operation=update
 */


interface Credentials {
  googleApi: CredentialReference;
  googleDriveOAuth2Api: CredentialReference;
}

/** Update a file */
export type GoogleDriveV2FileUpdateParams = {
  resource: 'file';
  operation: 'update';
/**
 * Authentication
 * @default oAuth2
 */
    authentication?: 'oAuth2' | 'serviceAccount' | Expression<string>;
/**
 * The ID of the file
 * @default {"mode":"list","value":""}
 */
    fileId?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** The name of the file
     */
    fileName?: string | Expression<string> | PlaceholderValue;
    /** Whether to set the 'keepForever' field in the new head revision. This is only applicable to files with binary content in Google Drive. Only 200 revisions for the file can be kept forever. If the limit is reached, try deleting pinned revisions.
     * @default false
     */
    keepRevisionForever?: boolean | Expression<boolean>;
    /** Whether to move a file to the trash. Only the owner may trash a file.
     * @default false
     */
    trashed?: boolean | Expression<boolean>;
    /** A language hint for OCR processing during image import (ISO 639-1 code)
     */
    ocrLanguage?: string | Expression<string> | PlaceholderValue;
    /** The ID of the parent to set
     */
    parentId?: string | Expression<string> | PlaceholderValue;
    /** Whether to use the uploaded content as indexable text
     * @default false
     */
    useContentAsIndexableText?: boolean | Expression<boolean>;
  };
/**
 * Options
 * @default {}
 */
    options?: {
    /** The fields to return
     * @default []
     */
    fields?: Array<'*' | 'explicitlyTrashed' | 'exportLinks' | 'hasThumbnail' | 'iconLink' | 'id' | 'kind' | 'mimeType' | 'name' | 'permissions' | 'shared' | 'spaces' | 'starred' | 'thumbnailLink' | 'trashed' | 'version' | 'webViewLink'>;
  };
};

export type GoogleDriveV2FileUpdateNode = {
  type: 'n8n-nodes-base.googleDrive';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<GoogleDriveV2FileUpdateParams>;
};
