// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/sendInBlue/v1/resource_email/operation_send.ts
/**
 * Brevo Node - Version 1
 * Discriminator: resource=email, operation=send
 */


interface Credentials {
  sendInBlueApi: CredentialReference;
}

export type SendInBlueV1EmailSendParams = {
  resource: 'email';
  operation: 'send';
/**
 * Send HTML
 * @default false
 */
    sendHTML?: boolean | Expression<boolean>;
/**
 * Subject of the email
 */
    subject?: string | Expression<string> | PlaceholderValue;
/**
 * Text content of the message
 * @displayOptions.show { sendHTML: [false] }
 */
    textContent?: string | Expression<string> | PlaceholderValue;
/**
 * HTML content of the message
 * @displayOptions.show { sendHTML: [true] }
 */
    htmlContent?: string | Expression<string> | PlaceholderValue;
/**
 * Sender
 */
    sender?: string | Expression<string> | PlaceholderValue;
/**
 * Receipients
 */
    receipients?: string | Expression<string> | PlaceholderValue;
/**
 * Additional fields to add
 * @default {}
 */
    additionalFields?: {
    /** Attachments
     * @default {}
     */
    emailAttachments?: {
        /** Attachment Data
     */
    attachment?: {
      /** The name of the incoming field containing the binary file data to be processed
       */
      binaryPropertyName?: string | Expression<string> | PlaceholderValue;
    };
  };
    /** Receipients BCC
     * @default {}
     */
    receipientsBCC?: {
        /** Receipient
     */
    receipientBcc?: {
      /** Receipient
       */
      bcc?: string | Expression<string> | PlaceholderValue;
    };
  };
    /** Receipients CC
     * @default {}
     */
    receipientsCC?: {
        /** Receipient
     */
    receipientCc?: {
      /** Receipient
       */
      cc?: string | Expression<string> | PlaceholderValue;
    };
  };
    /** Add tags to your emails to find them more easily
     * @default {}
     */
    emailTags?: {
        /** Tags
     */
    tags?: {
      /** Tag
       */
      tag?: string | Expression<string> | PlaceholderValue;
    };
  };
  };
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type SendInBlueV1EmailSendNode = {
  type: 'n8n-nodes-base.sendInBlue';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SendInBlueV1EmailSendParams>;
};
