// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/sendInBlue/v1/resource_email/operation_send_template.ts
/**
 * Brevo Node - Version 1
 * Discriminator: resource=email, operation=sendTemplate
 */


interface Credentials {
  sendInBlueApi: CredentialReference;
}

export type SendInBlueV1EmailSendTemplateParams = {
  resource: 'email';
  operation: 'sendTemplate';
/**
 * Template ID
 */
    templateId?: string | Expression<string>;
/**
 * Receipients
 */
    receipients?: string | Expression<string> | PlaceholderValue;
/**
 * Additional fields to add
 * @default {}
 */
    additionalFields?: {
    /** Attachments
     * @default {}
     */
    emailAttachments?: {
        /** Attachment Data
     */
    attachment?: {
      /** The name of the incoming field containing the binary file data to be processed
       */
      binaryPropertyName?: string | Expression<string> | PlaceholderValue;
    };
  };
    /** Add tags to your emails to find them more easily
     * @default {}
     */
    emailTags?: {
        /** Tags
     */
    tags?: {
      /** Tag
       */
      tag?: string | Expression<string> | PlaceholderValue;
    };
  };
    /** Pass a set of attributes to customize the template
     * @default {}
     */
    templateParameters?: {
        /** Parameters
     */
    parameterValues?: {
      /** Comma-separated key=value pairs
       */
      parameters?: string | Expression<string> | PlaceholderValue;
    };
  };
  };
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type SendInBlueV1EmailSendTemplateNode = {
  type: 'n8n-nodes-base.sendInBlue';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SendInBlueV1EmailSendTemplateParams>;
};
