// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleFirebaseCloudFirestore/v1/resource_document/operation_get_all.ts
/**
 * Google Cloud Firestore Node - Version 1
 * Discriminator: resource=document, operation=getAll
 */


interface Credentials {
  googleFirebaseCloudFirestoreOAuth2Api: CredentialReference;
  googleApi: CredentialReference;
}

/** Get many documents from a collection */
export type GoogleFirebaseCloudFirestoreV1DocumentGetAllParams = {
  resource: 'document';
  operation: 'getAll';
  authentication?: 'googleFirebaseCloudFirestoreOAuth2Api' | 'serviceAccount' | Expression<string>;
/**
 * As displayed in firebase console URL. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    projectId?: string | Expression<string>;
/**
 * Usually the provided default value will work
 * @default (default)
 */
    database?: string | Expression<string> | PlaceholderValue;
/**
 * Collection name
 */
    collection?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Whether to return a simplified version of the response instead of the raw data
 * @default true
 */
    simple?: boolean | Expression<boolean>;
};

export type GoogleFirebaseCloudFirestoreV1DocumentGetAllOutput = {
  _createTime?: string;
  _id?: string;
  _name?: string;
  _updateTime?: string;
  createTime?: string;
  id?: string;
  updateTime?: string;
};

export type GoogleFirebaseCloudFirestoreV1DocumentGetAllNode = {
  type: 'n8n-nodes-base.googleFirebaseCloudFirestore';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GoogleFirebaseCloudFirestoreV1DocumentGetAllParams>;
  output?: Items<GoogleFirebaseCloudFirestoreV1DocumentGetAllOutput>;
};
