// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftGraphSecurity/v1/resource_secure_score_control_profile/operation_update.ts
/**
 * Microsoft Graph Security Node - Version 1
 * Discriminator: resource=secureScoreControlProfile, operation=update
 */


interface Credentials {
  microsoftGraphSecurityOAuth2Api: CredentialReference;
}

export type MicrosoftGraphSecurityV1SecureScoreControlProfileUpdateParams = {
  resource: 'secureScoreControlProfile';
  operation: 'update';
/**
 * ID of the secure score control profile to update
 */
    secureScoreControlProfileId?: string | Expression<string> | PlaceholderValue;
/**
 * Name of the provider of the security product or service
 */
    provider?: string | Expression<string> | PlaceholderValue;
/**
 * Name of the vendor of the security product or service
 */
    vendor?: string | Expression<string> | PlaceholderValue;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** Analyst driven setting on the control
     * @default Default
     */
    state?: 'Default' | 'Ignored' | 'ThirdParty' | Expression<string>;
  };
};

export type MicrosoftGraphSecurityV1SecureScoreControlProfileUpdateNode = {
  type: 'n8n-nodes-base.microsoftGraphSecurity';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MicrosoftGraphSecurityV1SecureScoreControlProfileUpdateParams>;
};
