// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/dropbox/v1/resource_search/operation_query.schema.js
/**
 * Dropbox Node - Version 1 - Zod Schema
 * Discriminator: resource=search, operation=query
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('search'),
      operation: z.literal('query'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      query: stringOrExpression.optional(),
      fileStatus: z.union([z.literal('active'), z.literal('deleted'), expressionSchema]).optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      simple: booleanOrExpression.optional(),
      filters: z.object({ file_categories: z.array(z.union([z.literal('audio'), z.literal('document'), z.literal('paper'), z.literal('folder'), z.literal('image'), z.literal('other'), z.literal('pdf'), z.literal('presentation'), z.literal('spreadsheet'), z.literal('video')])).optional(), file_extensions: stringOrExpression.optional(), path: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
