// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleAnalytics/v1/resource_user_activity/operation_search.schema.js
/**
 * Google Analytics Node - Version 1 - Zod Schema
 * Discriminator: resource=userActivity, operation=search
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('userActivity'),
      operation: z.literal('search'),
      viewId: stringOrExpression.optional(),
      userId: stringOrExpression.optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      additionalFields: z.object({ activityTypes: z.array(z.union([z.literal('ECOMMERCE'), z.literal('EVENT'), z.literal('GOAL'), z.literal('PAGEVIEW'), z.literal('SCREENVIEW')])).optional() }).optional(),
    }).optional(),
  });
};
