// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedrive/v1/resource_deal_product/operation_add.schema.js
/**
 * Pipedrive Node - Version 1 - Zod Schema
 * Discriminator: resource=dealProduct, operation=add
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('dealProduct'),
      operation: z.literal('add'),
      authentication: z.union([z.literal('apiToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      dealId: stringOrExpression.optional(),
      productId: stringOrExpression.optional(),
      item_price: numberOrExpression.optional(),
      quantity: numberOrExpression.optional(),
      additionalFields: z.object({ comments: stringOrExpression.optional(), discount_percentage: numberOrExpression.optional(), product_variation_id: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
