// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedrive/v1/resource_deal_product/operation_remove.ts
/**
 * Pipedrive Node - Version 1
 * Discriminator: resource=dealProduct, operation=remove
 */


interface Credentials {
  pipedriveApi: CredentialReference;
  pipedriveOAuth2Api: CredentialReference;
}

/** Remove a product from a deal */
export type PipedriveV1DealProductRemoveParams = {
  resource: 'dealProduct';
  operation: 'remove';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * The ID of the deal whose product to remove. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    dealId?: string | Expression<string>;
/**
 * ID of the deal-product (the ID of the product attached to the deal). Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    productAttachmentId?: string | Expression<string>;
};

export type PipedriveV1DealProductRemoveNode = {
  type: 'n8n-nodes-base.pipedrive';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PipedriveV1DealProductRemoveParams>;
};
