// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedrive/v1/resource_deal_product/operation_get_all.ts
/**
 * Pipedrive Node - Version 1
 * Discriminator: resource=dealProduct, operation=getAll
 */


interface Credentials {
  pipedriveApi: CredentialReference;
  pipedriveOAuth2Api: CredentialReference;
}

/** Get data of many activities */
export type PipedriveV1DealProductGetAllParams = {
  resource: 'dealProduct';
  operation: 'getAll';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * The ID of the deal whose products to retrieve. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    dealId?: string | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
};

export type PipedriveV1DealProductGetAllOutput = {
  active_flag?: boolean;
  add_time?: string;
  billing_frequency?: string;
  currency?: string;
  deal_id?: number;
  discount?: number;
  discount_type?: string;
  enabled_flag?: boolean;
  id?: number;
  last_edit?: string;
  name?: string;
  order_nr?: number;
  product?: null;
  product_id?: number;
  quantity?: number;
  quantity_formatted?: string;
  sum_formatted?: string;
  tax_method?: string;
};

export type PipedriveV1DealProductGetAllNode = {
  type: 'n8n-nodes-base.pipedrive';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PipedriveV1DealProductGetAllParams>;
  output?: Items<PipedriveV1DealProductGetAllOutput>;
};
