// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/shopify/v1/resource_order/operation_create.ts
/**
 * Shopify Node - Version 1
 * Discriminator: resource=order, operation=create
 */


interface Credentials {
  shopifyApi: CredentialReference;
  shopifyAccessTokenApi: CredentialReference;
  shopifyOAuth2Api: CredentialReference;
}

/** Create an order */
export type ShopifyV1OrderCreateParams = {
  resource: 'order';
  operation: 'create';
  authentication?: 'accessToken' | 'oAuth2' | 'apiKey' | Expression<string>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Billing Address
     * @default {}
     */
    billingAddressUi?: {
        /** Billing Address
     */
    billingAddressValues?: {
      /** First Name
       */
      firstName?: string | Expression<string> | PlaceholderValue;
      /** Last Name
       */
      lastName?: string | Expression<string> | PlaceholderValue;
      /** Company
       */
      company?: string | Expression<string> | PlaceholderValue;
      /** Country
       */
      country?: string | Expression<string> | PlaceholderValue;
      /** Address Line 1
       */
      address1?: string | Expression<string> | PlaceholderValue;
      /** Address Line 2
       */
      address2?: string | Expression<string> | PlaceholderValue;
      /** City
       */
      city?: string | Expression<string> | PlaceholderValue;
      /** Province
       */
      province?: string | Expression<string> | PlaceholderValue;
      /** Zip Code
       */
      zip?: string | Expression<string> | PlaceholderValue;
      /** Phone
       */
      phone?: string | Expression<string> | PlaceholderValue;
    };
  };
    /** Discount Codes
     * @default {}
     */
    discountCodesUi?: {
        /** Discount Code
     */
    discountCodesValues?: Array<{
      /** The amount that's deducted from the order total
       */
      amount?: string | Expression<string> | PlaceholderValue;
      /** When the associated discount application is of type code
       */
      code?: string | Expression<string> | PlaceholderValue;
      /** When the associated discount application is of type code
       * @default fixedAmount
       */
      type?: 'fixedAmount' | 'percentage' | 'shipping' | Expression<string>;
    }>;
  };
    /** The customer's email address
     */
    email?: string | Expression<string> | PlaceholderValue;
    /** The order's status in terms of fulfilled line items
     */
    fulfillmentStatus?: 'fulfilled' | 'null' | 'partial' | 'restocked' | Expression<string>;
    /** The behaviour to use when updating inventory
     * @default bypass
     */
    inventoryBehaviour?: 'bypass' | 'decrementIgnoringPolicy' | 'decrementObeyingPolicy' | Expression<string>;
    /** The ID of the physical location where the order was processed. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    locationId?: string | Expression<string>;
    /** An optional note that a shop owner can attach to the order
     */
    note?: string | Expression<string> | PlaceholderValue;
    /** Whether to send a shipping confirmation to the customer
     * @default false
     */
    sendFulfillmentReceipt?: boolean | Expression<boolean>;
    /** Whether to send an order confirmation to the customer
     * @default false
     */
    sendReceipt?: boolean | Expression<boolean>;
    /** Shipping Address
     * @default {}
     */
    shippingAddressUi?: {
        /** Shipping Address
     */
    shippingAddressValues?: {
      /** First Name
       */
      firstName?: string | Expression<string> | PlaceholderValue;
      /** Last Name
       */
      lastName?: string | Expression<string> | PlaceholderValue;
      /** Company
       */
      company?: string | Expression<string> | PlaceholderValue;
      /** Country
       */
      country?: string | Expression<string> | PlaceholderValue;
      /** Address Line 1
       */
      address1?: string | Expression<string> | PlaceholderValue;
      /** Address Line 2
       */
      address2?: string | Expression<string> | PlaceholderValue;
      /** City
       */
      city?: string | Expression<string> | PlaceholderValue;
      /** Province
       */
      province?: string | Expression<string> | PlaceholderValue;
      /** Zip Code
       */
      zip?: string | Expression<string> | PlaceholderValue;
      /** Phone
       */
      phone?: string | Expression<string> | PlaceholderValue;
    };
  };
    /** Where the order originated. Can be set only during order creation, and is not writeable afterwards.
     */
    sourceName?: string | Expression<string> | PlaceholderValue;
    /** Tags attached to the order, formatted as a string of comma-separated values
     */
    tags?: string | Expression<string> | PlaceholderValue;
    /** Whether this is a test order
     * @default false
     */
    test?: boolean | Expression<boolean>;
  };
/**
 * Line Items
 * @default {}
 */
    limeItemsUi?: {
        /** Line Item
     */
    lineItemValues?: Array<{
      /** The ID of the product that the line item belongs to. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
       */
      productId?: string | Expression<string>;
      /** The ID of the product variant
       */
      variantId?: string | Expression<string> | PlaceholderValue;
      /** The title of the product
       */
      title?: string | Expression<string> | PlaceholderValue;
      /** The weight of the item in grams
       */
      grams?: string | Expression<string> | PlaceholderValue;
      /** The number of items that were purchased
       * @default 1
       */
      quantity?: number | Expression<number>;
      /** Price
       */
      price?: string | Expression<string> | PlaceholderValue;
    }>;
  };
};

export type ShopifyV1OrderCreateNode = {
  type: 'n8n-nodes-base.shopify';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ShopifyV1OrderCreateParams>;
};
