// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/shopify/v1/resource_order/operation_get_all.ts
/**
 * Shopify Node - Version 1
 * Discriminator: resource=order, operation=getAll
 */


interface Credentials {
  shopifyApi: CredentialReference;
  shopifyAccessTokenApi: CredentialReference;
  shopifyOAuth2Api: CredentialReference;
}

/** Get many orders */
export type ShopifyV1OrderGetAllParams = {
  resource: 'order';
  operation: 'getAll';
  authentication?: 'accessToken' | 'oAuth2' | 'apiKey' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Show orders attributed to a certain app, specified by the app ID. Set as current to show orders for the app currently consuming the API.
     */
    attributionAppId?: string | Expression<string> | PlaceholderValue;
    /** Show orders created at or after date
     */
    createdAtMin?: string | Expression<string>;
    /** Show orders created at or before date
     */
    createdAtMax?: string | Expression<string>;
    /** Filter orders by their financial status
     * @default any
     */
    financialStatus?: 'any' | 'authorized' | 'paid' | 'partiallyPaid' | 'partiallyRefunded' | 'pending' | 'refunded' | 'unpaid' | 'voided' | Expression<string>;
    /** Filter orders by their fulfillment status
     * @default any
     */
    fulfillmentStatus?: 'any' | 'partial' | 'shipped' | 'unfulfilled' | 'unshipped' | Expression<string>;
    /** Fields the orders will return, formatted as a string of comma-separated values. By default all the fields are returned.
     */
    fields?: string | Expression<string> | PlaceholderValue;
    /** Retrieve only orders specified by a comma-separated list of order IDs
     */
    ids?: string | Expression<string> | PlaceholderValue;
    /** Show orders imported at or before date
     */
    processedAtMax?: string | Expression<string>;
    /** Show orders imported at or after date
     */
    processedAtMin?: string | Expression<string>;
    /** Filter orders by their status
     * @default open
     */
    status?: 'any' | 'Cancelled' | 'closed' | 'open' | Expression<string>;
    /** Show orders after the specified ID
     */
    sinceId?: string | Expression<string> | PlaceholderValue;
    /** Show orders last updated at or after date
     */
    updatedAtMax?: string | Expression<string>;
    /** Show orders last updated at or before date
     */
    updatedAtMin?: string | Expression<string>;
  };
};

export type ShopifyV1OrderGetAllOutput = {
  created_at?: string;
  id?: number;
  line_items?: Array<{
    admin_graphql_api_id?: string;
    current_quantity?: number;
    discount_allocations?: Array<{
      amount?: string;
      amount_set?: {
        presentment_money?: {
          amount?: string;
          currency_code?: string;
        };
        shop_money?: {
          amount?: string;
          currency_code?: string;
        };
      };
      discount_application_index?: number;
    }>;
    fulfillable_quantity?: number;
    fulfillment_service?: string;
    gift_card?: boolean;
    grams?: number;
    id?: number;
    name?: string;
    price?: string;
    price_set?: {
      presentment_money?: {
        amount?: string;
        currency_code?: string;
      };
      shop_money?: {
        amount?: string;
        currency_code?: string;
      };
    };
    product_exists?: boolean;
    properties?: Array<{
      name?: string;
    }>;
    quantity?: number;
    requires_shipping?: boolean;
    tax_lines?: Array<{
      channel_liable?: boolean;
      price?: string;
      price_set?: {
        presentment_money?: {
          amount?: string;
          currency_code?: string;
        };
        shop_money?: {
          amount?: string;
          currency_code?: string;
        };
      };
      title?: string;
    }>;
    taxable?: boolean;
    title?: string;
    total_discount?: string;
    total_discount_set?: {
      presentment_money?: {
        amount?: string;
        currency_code?: string;
      };
      shop_money?: {
        amount?: string;
        currency_code?: string;
      };
    };
    vendor?: string;
  }>;
};

export type ShopifyV1OrderGetAllNode = {
  type: 'n8n-nodes-base.shopify';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ShopifyV1OrderGetAllParams>;
  output?: Items<ShopifyV1OrderGetAllOutput>;
};
