// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/shopify/v1/resource_order/operation_get_all.schema.js
/**
 * Shopify Node - Version 1 - Zod Schema
 * Discriminator: resource=order, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('order').default('order'),
      operation: z.literal('getAll'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), z.literal('apiKey'), expressionSchema]).optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      options: z.object({ attributionAppId: stringOrExpression.optional(), createdAtMin: stringOrExpression.optional(), createdAtMax: stringOrExpression.optional(), financialStatus: z.union([z.literal('any'), z.literal('authorized'), z.literal('paid'), z.literal('partiallyPaid'), z.literal('partiallyRefunded'), z.literal('pending'), z.literal('refunded'), z.literal('unpaid'), z.literal('voided'), expressionSchema]).optional(), fulfillmentStatus: z.union([z.literal('any'), z.literal('partial'), z.literal('shipped'), z.literal('unfulfilled'), z.literal('unshipped'), expressionSchema]).optional(), fields: stringOrExpression.optional(), ids: stringOrExpression.optional(), processedAtMax: stringOrExpression.optional(), processedAtMin: stringOrExpression.optional(), status: z.union([z.literal('any'), z.literal('Cancelled'), z.literal('closed'), z.literal('open'), expressionSchema]).optional(), sinceId: stringOrExpression.optional(), updatedAtMax: stringOrExpression.optional(), updatedAtMin: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
