// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/shopify/v1/resource_order/operation_create.schema.js
/**
 * Shopify Node - Version 1 - Zod Schema
 * Discriminator: resource=order, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('order').default('order'),
      operation: z.literal('create').default('create'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), z.literal('apiKey'), expressionSchema]).optional(),
      additionalFields: z.object({ billingAddressUi: z.unknown().optional(), discountCodesUi: z.unknown().optional(), email: stringOrExpression.optional(), fulfillmentStatus: z.union([z.literal('fulfilled'), z.literal('null'), z.literal('partial'), z.literal('restocked'), expressionSchema]).optional(), inventoryBehaviour: z.union([z.literal('bypass'), z.literal('decrementIgnoringPolicy'), z.literal('decrementObeyingPolicy'), expressionSchema]).optional(), locationId: stringOrExpression.optional(), note: stringOrExpression.optional(), sendFulfillmentReceipt: booleanOrExpression.optional(), sendReceipt: booleanOrExpression.optional(), shippingAddressUi: z.unknown().optional(), sourceName: stringOrExpression.optional(), tags: stringOrExpression.optional(), test: booleanOrExpression.optional() }).optional(),
      limeItemsUi: z.object({ lineItemValues: z.array(z.object({ productId: stringOrExpression.optional(), variantId: stringOrExpression.optional(), title: stringOrExpression.optional(), grams: stringOrExpression.optional(), quantity: numberOrExpression.optional(), price: stringOrExpression.optional() })).optional() }).optional(),
    }).optional(),
  });
};
