// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/invoiceNinja/v1/resource_invoice/operation_get_all.ts
/**
 * Invoice Ninja Node - Version 1
 * Discriminator: resource=invoice, operation=getAll
 */


interface Credentials {
  invoiceNinjaApi: CredentialReference;
}

/** Get data of many clients */
export type InvoiceNinjaV1InvoiceGetAllParams = {
  resource: 'invoice';
  operation: 'getAll';
/**
 * API Version
 * @default v4
 */
    apiVersion?: 'v4' | 'v5' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Invoice Number
     */
    invoiceNumber?: string | Expression<string> | PlaceholderValue;
    /** Include
     * @default client
     */
    include?: 'client' | Expression<string>;
    /** Status
     * @default active
     */
    status?: 'active' | 'archived' | 'deleted' | Expression<string>;
    /** Created At
     */
    createdAt?: string | Expression<string>;
    /** Updated At
     */
    updatedAt?: string | Expression<string>;
    /** Is Deleted
     * @default false
     */
    isDeleted?: boolean | Expression<boolean>;
    /** Client Status
     * @default all
     */
    clientStatus?: 'all' | 'paid' | 'unpaid' | 'overdue' | Expression<string>;
  };
};

export type InvoiceNinjaV1InvoiceGetAllNode = {
  type: 'n8n-nodes-base.invoiceNinja';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<InvoiceNinjaV1InvoiceGetAllParams>;
};
