// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/wise/v1/resource_quote/operation_create.ts
/**
 * Wise Node - Version 1
 * Discriminator: resource=quote, operation=create
 */


interface Credentials {
  wiseApi: CredentialReference;
}

export type WiseV1QuoteCreateParams = {
  resource: 'quote';
  operation: 'create';
/**
 * ID of the user profile to create the quote under. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 * @default []
 */
    profileId?: string | Expression<string>;
/**
 * ID of the account that will receive the funds. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 * @default []
 */
    targetAccountId?: string | Expression<string>;
/**
 * Whether the amount is to be sent or received
 * @default source
 */
    amountType?: 'source' | 'target' | Expression<string>;
/**
 * Amount of funds for the quote to create
 * @default 1
 */
    amount?: number | Expression<number>;
/**
 * Code of the currency to send for the quote to create
 */
    sourceCurrency?: string | Expression<string> | PlaceholderValue;
/**
 * Code of the currency to receive for the quote to create
 */
    targetCurrency?: string | Expression<string> | PlaceholderValue;
};

export type WiseV1QuoteCreateNode = {
  type: 'n8n-nodes-base.wise';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<WiseV1QuoteCreateParams>;
};
