// @microsoft/agents-a365-runtime/dist/cjs/environment-utils.js
"use strict";
// ------------------------------------------------------------------------------
// Copyright (c) Microsoft Corporation. All rights reserved.
// ------------------------------------------------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.DEVELOPMENT_ENVIRONMENT_NAME = exports.PRODUCTION_ENVIRONMENT_NAME = exports.PROD_OBSERVABILITY_CLUSTER_CATEGORY = exports.PROD_MCP_PLATFORM_AUTHENTICATION_SCOPE = exports.PROD_OBSERVABILITY_SCOPE = void 0;
exports.getObservabilityAuthenticationScope = getObservabilityAuthenticationScope;
exports.getClusterCategory = getClusterCategory;
exports.isDevelopmentEnvironment = isDevelopmentEnvironment;
exports.getMcpPlatformAuthenticationScope = getMcpPlatformAuthenticationScope;
/**
 * Utility logic for environment-related operations.
 */
exports.PROD_OBSERVABILITY_SCOPE = 'https://api.powerplatform.com/.default';
exports.PROD_MCP_PLATFORM_AUTHENTICATION_SCOPE = 'ea9ffc3e-8a23-4a7d-836d-234d7c7565c1/.default';
exports.PROD_OBSERVABILITY_CLUSTER_CATEGORY = 'prod';
// Default environment names
exports.PRODUCTION_ENVIRONMENT_NAME = 'production';
exports.DEVELOPMENT_ENVIRONMENT_NAME = 'Development';
/**
 * Returns the scope for authenticating to the observability service
 *
 * @returns The authentication scope for the current environment.
 */
function getObservabilityAuthenticationScope() {
    // Always return production scope
    return [exports.PROD_OBSERVABILITY_SCOPE];
}
/**
 * Gets the cluster category from environment variables.
 *
 * @returns The cluster category from CLUSTER_CATEGORY env var, defaults to 'prod'.
 */
function getClusterCategory() {
    const clusterCategory = process.env.CLUSTER_CATEGORY;
    if (!clusterCategory) {
        return 'prod';
    }
    return clusterCategory.toLowerCase();
}
/**
 * Returns true if the current environment is a development environment.
 *
 * @returns True if the current environment is development, false otherwise.
 */
function isDevelopmentEnvironment() {
    const clusterCategory = getClusterCategory();
    return ['local', 'dev'].includes(clusterCategory);
}
/**
 * Gets the MCP platform authentication scope from environment variables.
 *
 * @returns The MCP platform authentication scope from MCP_PLATFORM_AUTHENTICATION_SCOPE env var, defaults to production scope.
 */
function getMcpPlatformAuthenticationScope() {
    return process.env.MCP_PLATFORM_AUTHENTICATION_SCOPE || exports.PROD_MCP_PLATFORM_AUTHENTICATION_SCOPE;
}
