// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/respondToWebhook/v15.schema.js
/**
 * Respond to Webhook Node - Version 1.5 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    enableResponseOutput: booleanOrExpression.optional(),
    respondWith: z.union([z.literal('allIncomingItems'), z.literal('binary'), z.literal('firstIncomingItem'), z.literal('json'), z.literal('jwt'), z.literal('noData'), z.literal('redirect'), z.literal('text')]).optional(),
    redirectURL: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"respondWith":["redirect"]}}, defaults: {"respondWith":"firstIncomingItem"} }),
    responseBody: resolveSchema({ parameters, schema: z.union([iDataObjectSchema, z.string()]), required: false, displayOptions: {"show":{"respondWith":["json","text"]}}, defaults: {"respondWith":"firstIncomingItem"} }),
    payload: resolveSchema({ parameters, schema: z.union([iDataObjectSchema, z.string()]), required: false, displayOptions: {"show":{"respondWith":["jwt"]}}, defaults: {"respondWith":"firstIncomingItem"} }),
    responseDataSource: resolveSchema({ parameters, schema: z.union([z.literal('automatically'), z.literal('set'), expressionSchema]), required: false, displayOptions: {"show":{"respondWith":["binary"]}}, defaults: {"respondWith":"firstIncomingItem"} }),
    inputFieldName: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"respondWith":["binary"],"responseDataSource":["set"]}}, defaults: {"respondWith":"firstIncomingItem","responseDataSource":"automatically"} }),
    options: z.object({ responseCode: numberOrExpression.optional(), responseHeaders: z.unknown().optional(), responseKey: stringOrExpression.optional(), enableStreaming: booleanOrExpression.optional() }).optional(),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
