// @n8n/n8n-nodes-langchain/dist/nodes/vector_store/ChatHubVectorStoreQdrant/ChatHubVectorStoreQdrant.node.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChatHubVectorStoreQdrant = void 0;
const qdrant_1 = require("@langchain/qdrant");
const n8n_workflow_1 = require("n8n-workflow");
const Qdrant_utils_1 = require("../VectorStoreQdrant/Qdrant.utils");
const userScoped_1 = require("../shared/userScoped");
const ai_utilities_1 = require("@n8n/ai-utilities");
const INDEXED_PAYLOAD_FIELDS = [
    'metadata.userId',
    'metadata.agentId',
    'metadata.fileKnowledgeId',
];
async function ensurePayloadIndexes(client, collectionName) {
    for (const fieldName of INDEXED_PAYLOAD_FIELDS) {
        await client.createPayloadIndex(collectionName, {
            field_name: fieldName,
            field_schema: 'keyword',
        });
    }
}
async function deleteDocuments(payload) {
    const { filter = {} } = (typeof payload === 'string' ? (0, n8n_workflow_1.jsonParse)(payload) : (payload ?? {}));
    const credentials = await this.getCredentials('chatHubVectorStoreQdrantApi');
    const userId = (0, userScoped_1.ensureUserId)(this);
    const client = (0, Qdrant_utils_1.createQdrantClient)(credentials);
    await ensurePayloadIndexes(client, credentials.collectionName);
    const must = [
        { key: 'metadata.userId', match: { value: userId } },
        ...Object.entries(filter).map(([key, value]) => {
            if (Array.isArray(value)) {
                return { key: `metadata.${key}`, match: { any: value } };
            }
            return { key: `metadata.${key}`, match: { value } };
        }),
    ];
    this.logger.debug(`Deleting Qdrant vector store documents... Filter: ${JSON.stringify(must)}`);
    await client.delete(credentials.collectionName, { filter: { must } });
    return null;
}
class ChatHubVectorStoreQdrant extends (0, ai_utilities_1.createVectorStoreNode)({
    meta: {
        displayName: 'ChatHub Qdrant Vector Store',
        name: 'chatHubVectorStoreQdrant',
        description: 'Internal-use vector store for ChatHub',
        icon: 'file:../VectorStoreQdrant/qdrant.svg',
        docsUrl: 'https://docs.n8n.io',
        credentials: [
            {
                name: 'chatHubVectorStoreQdrantApi',
                required: true,
            },
        ],
        operationModes: ['load', 'insert', 'retrieve', 'retrieve-as-tool'],
    },
    hidden: true,
    methods: { actionHandler: { deleteDocuments } },
    sharedFields: [],
    insertFields: [],
    loadFields: [],
    retrieveFields: [],
    async getVectorStoreClient(context, filter, embeddings) {
        const credentials = await context.getCredentials('chatHubVectorStoreQdrantApi');
        const userId = (0, userScoped_1.ensureUserId)(context);
        const client = (0, Qdrant_utils_1.createQdrantClient)(credentials);
        const config = { client, collectionName: credentials.collectionName };
        const store = await qdrant_1.QdrantVectorStore.fromExistingCollection(embeddings, config);
        await ensurePayloadIndexes(client, credentials.collectionName);
        const originalSearch = store.similaritySearch.bind(store);
        const originalSearchWithScore = store.similaritySearchWithScore.bind(store);
        const originalSearchVectorWithScore = store.similaritySearchVectorWithScore.bind(store);
        function createFinalFilter(anotherFilter) {
            const originalMust = anotherFilter
                ? Array.isArray(anotherFilter.must)
                    ? anotherFilter.must
                    : [anotherFilter.must ?? {}]
                : [];
            const must = filter ? (Array.isArray(filter.must) ? filter.must : [filter.must ?? {}]) : [];
            const final = {
                ...anotherFilter,
                ...filter,
                must: [...originalMust, ...must, { key: 'metadata.userId', match: { value: userId } }],
            };
            context.logger.debug(`Querying Qdrant vector store... Filter: ${JSON.stringify(final)}`);
            return final;
        }
        store.similaritySearch = async (query, k, f, callbacks) => await originalSearch(query, k, createFinalFilter(f), callbacks);
        store.similaritySearchWithScore = async (query, k, f, callbacks) => await originalSearchWithScore(query, k, createFinalFilter(f), callbacks);
        store.similaritySearchVectorWithScore = async (query, k, f) => await originalSearchVectorWithScore(query, k, createFinalFilter(f));
        return store;
    },
    async populateVectorStore(context, embeddings, documents) {
        const credentials = await context.getCredentials('chatHubVectorStoreQdrantApi');
        const userId = (0, userScoped_1.ensureUserId)(context);
        const client = (0, Qdrant_utils_1.createQdrantClient)(credentials);
        const config = { client, collectionName: credentials.collectionName };
        await qdrant_1.QdrantVectorStore.fromDocuments(documents.map((d) => ({ ...d, metadata: { ...d.metadata, userId } })), embeddings, config);
        await ensurePayloadIndexes(client, credentials.collectionName);
    },
}) {
}
exports.ChatHubVectorStoreQdrant = ChatHubVectorStoreQdrant;
