// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/youTube/v1/resource_playlist/operation_create.schema.js
/**
 * YouTube Node - Version 1 - Zod Schema
 * Discriminator: resource=playlist, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('playlist'),
      operation: z.literal('create'),
      title: stringOrExpression.optional(),
      options: z.object({ description: stringOrExpression.optional(), privacyStatus: z.union([z.literal('private'), z.literal('public'), z.literal('unlisted'), expressionSchema]).optional(), tags: stringOrExpression.optional(), defaultLanguage: stringOrExpression.optional(), onBehalfOfContentOwnerChannel: stringOrExpression.optional(), onBehalfOfContentOwner: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
