// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/youTube/v1/resource_playlist/operation_update.ts
/**
 * YouTube Node - Version 1
 * Discriminator: resource=playlist, operation=update
 */


interface Credentials {
  youTubeOAuth2Api: CredentialReference;
}

/** Update a channel */
export type YouTubeV1PlaylistUpdateParams = {
  resource: 'playlist';
  operation: 'update';
/**
 * The playlist's title
 */
    playlistId?: string | Expression<string> | PlaceholderValue;
/**
 * The playlist's title
 */
    title?: string | Expression<string> | PlaceholderValue;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** The language of the text in the playlist resource's title and description properties. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    defaultLanguage?: string | Expression<string>;
    /** The playlist's description
     */
    description?: string | Expression<string> | PlaceholderValue;
    /** The onBehalfOfContentOwner parameter indicates that the request's authorization credentials identify a YouTube CMS user who is acting on behalf of the content owner specified in the parameter value
     */
    onBehalfOfContentOwner?: string | Expression<string> | PlaceholderValue;
    /** The playlist's privacy status
     */
    privacyStatus?: 'private' | 'public' | 'unlisted' | Expression<string>;
    /** Keyword tags associated with the playlist. Mulplie can be defined separated by comma.
     */
    tags?: string | Expression<string> | PlaceholderValue;
  };
};

export type YouTubeV1PlaylistUpdateNode = {
  type: 'n8n-nodes-base.youTube';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<YouTubeV1PlaylistUpdateParams>;
};
