// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/youTube/v1/resource_playlist/operation_get_all.schema.js
/**
 * YouTube Node - Version 1 - Zod Schema
 * Discriminator: resource=playlist, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('playlist'),
      operation: z.literal('getAll').default('getAll'),
      part: z.array(z.union([z.literal('*'), z.literal('contentDetails'), z.literal('id'), z.literal('localizations'), z.literal('player'), z.literal('snippet'), z.literal('status')])).optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filters: z.object({ channelId: stringOrExpression.optional(), id: stringOrExpression.optional() }).optional(),
      options: z.object({ onBehalfOfContentOwnerChannel: stringOrExpression.optional(), onBehalfOfContentOwner: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
