// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/youTube/v1/resource_playlist/operation_create.ts
/**
 * YouTube Node - Version 1
 * Discriminator: resource=playlist, operation=create
 */


interface Credentials {
  youTubeOAuth2Api: CredentialReference;
}

/** Create a playlist */
export type YouTubeV1PlaylistCreateParams = {
  resource: 'playlist';
  operation: 'create';
/**
 * The playlist's title
 */
    title?: string | Expression<string> | PlaceholderValue;
/**
 * Options
 * @default {}
 */
    options?: {
    /** The playlist's description
     */
    description?: string | Expression<string> | PlaceholderValue;
    /** The playlist's privacy status
     */
    privacyStatus?: 'private' | 'public' | 'unlisted' | Expression<string>;
    /** Keyword tags associated with the playlist. Mulplie can be defined separated by comma.
     */
    tags?: string | Expression<string> | PlaceholderValue;
    /** The language of the text in the playlist resource's title and description properties. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    defaultLanguage?: string | Expression<string>;
    /** The onBehalfOfContentOwnerChannel parameter specifies the YouTube channel ID of the channel to which a video is being added. This parameter is required when a request specifies a value for the onBehalfOfContentOwner parameter, and it can only be used in conjunction with that parameter.
     */
    onBehalfOfContentOwnerChannel?: string | Expression<string> | PlaceholderValue;
    /** The onBehalfOfContentOwner parameter indicates that the request's authorization credentials identify a YouTube CMS user who is acting on behalf of the content owner specified in the parameter value
     */
    onBehalfOfContentOwner?: string | Expression<string> | PlaceholderValue;
  };
};

export type YouTubeV1PlaylistCreateNode = {
  type: 'n8n-nodes-base.youTube';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<YouTubeV1PlaylistCreateParams>;
};
