// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/clockify/v1/resource_project/operation_update.schema.js
/**
 * Clockify Node - Version 1 - Zod Schema
 * Discriminator: resource=project, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('project').default('project'),
      operation: z.literal('update'),
      workspaceId: stringOrExpression.optional(),
      projectId: stringOrExpression.optional(),
      updateFields: z.object({ billable: booleanOrExpression.optional(), color: stringOrExpression.optional(), clientId: stringOrExpression.optional(), estimateUi: z.unknown().optional(), isPublic: booleanOrExpression.optional(), name: stringOrExpression.optional(), note: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
