// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/lemlist/v2/resource_enrich/operation_get.ts
/**
 * Lemlist Node - Version 2
 * Discriminator: resource=enrich, operation=get
 */


interface Credentials {
  lemlistApi: CredentialReference;
}

export type LemlistV2EnrichGetParams = {
  resource: 'enrich';
  operation: 'get';
/**
 * ID of the enrichment to retrieve
 */
    enrichId?: string | Expression<string> | PlaceholderValue;
};

export type LemlistV2EnrichGetOutput = {
  data?: {
    email?: {
      email?: string;
      notFound?: boolean;
      status?: string;
    };
    linkedin?: {
      companyDescription?: string;
      companyDomain?: string;
      companyEmployeesOnLinkedin?: number;
      companyFoundedOn?: number;
      companyHeadQuarter?: string;
      companyId?: number;
      companyIndustry?: string;
      companyLinkedinUrl?: string;
      companyLogo?: string;
      companyName?: string;
      companySize?: string;
      companySpecialities?: string;
      companyTagline?: string;
      companyType?: string;
      companyWebsite?: string;
      firstName?: string;
      industry?: string;
      languages?: string;
      lastName?: string;
      linkedinClassicId?: string;
      linkedinMemberId?: number;
      linkedinUrl?: string;
      locationName?: string;
      occupation?: string;
      picture?: string;
      positionGroups?: Array<{
        company?: {
          description?: string;
          domain?: string;
          employeesOnLinkedin?: number;
          foundedOn?: number;
          headQuarter?: string;
          id?: number;
          industry?: string;
          linkedinUrl?: string;
          linkedinUrlSalesNav?: string;
          logo?: string;
          name?: string;
          size?: string;
          specialities?: string;
          tagline?: string;
          type?: string;
          website?: string;
        };
        date?: {
          end?: {
            month?: number;
            year?: number;
          };
          start?: {
            month?: number;
            year?: number;
          };
        };
        profilePositions?: Array<{
          date?: {
            end?: {
              month?: number;
              year?: number;
            };
            start?: {
              month?: number;
              year?: number;
            };
          };
          title?: string;
        }>;
      }>;
      project?: string;
      skills?: string;
      summary?: string;
      tagline?: string;
    };
  };
  enrichmentId?: string;
  enrichmentStatus?: string;
  input?: {
    companyDomain?: string;
    companyName?: string;
    email?: string;
    firstName?: string;
    lastName?: string;
    linkedinUrl?: string;
  };
};

export type LemlistV2EnrichGetNode = {
  type: 'n8n-nodes-base.lemlist';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<LemlistV2EnrichGetParams>;
  output?: Items<LemlistV2EnrichGetOutput>;
};
