// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/zammad/v1/resource_organization/operation_create.ts
/**
 * Zammad Node - Version 1
 * Discriminator: resource=organization, operation=create
 */


interface Credentials {
  zammadBasicAuthApi: CredentialReference;
  zammadTokenAuthApi: CredentialReference;
}

/** Create a group */
export type ZammadV1OrganizationCreateParams = {
  resource: 'organization';
  operation: 'create';
  authentication?: 'basicAuth' | 'tokenAuth' | Expression<string>;
/**
 * Organization Name
 */
    name?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Whether the organization is shared with other instances
     * @default false
     */
    shared?: boolean | Expression<boolean>;
    /** The domain associated with the organization
     */
    domain?: string | Expression<string> | PlaceholderValue;
    /** Whether to assign users based on their email domain
     * @default true
     */
    domain_assignment?: boolean | Expression<boolean>;
    /** Whether the organization is active
     * @default true
     */
    active?: boolean | Expression<boolean>;
    /** Whether the organization is marked as VIP
     * @default false
     */
    vip?: boolean | Expression<boolean>;
    /** A note about the organization
     */
    note?: string | Expression<string> | PlaceholderValue;
    /** Custom Fields
     * @default {}
     */
    customFieldsUi?: {
        /** Custom Field
     */
    customFieldPairs?: Array<{
      /** Name of the custom field to set. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
       */
      name?: string | Expression<string>;
      /** Value to set on the custom field
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
  };
};

export type ZammadV1OrganizationCreateNode = {
  type: 'n8n-nodes-base.zammad';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ZammadV1OrganizationCreateParams>;
};
