// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/zammad/v1/resource_organization/operation_get_all.ts
/**
 * Zammad Node - Version 1
 * Discriminator: resource=organization, operation=getAll
 */


interface Credentials {
  zammadBasicAuthApi: CredentialReference;
  zammadTokenAuthApi: CredentialReference;
}

/** Get many groups */
export type ZammadV1OrganizationGetAllParams = {
  resource: 'organization';
  operation: 'getAll';
  authentication?: 'basicAuth' | 'tokenAuth' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
};

export type ZammadV1OrganizationGetAllOutput = {
  active?: boolean;
  cod_torre?: string;
  contrato_sla?: string;
  created_at?: string;
  created_by_id?: number;
  dom_2nd?: string;
  domain?: string;
  domain_assignment?: boolean;
  id?: number;
  id_tele?: string;
  member_ids?: Array<number>;
  name?: string;
  r7regiao?: string;
  sctntid?: string;
  secondary_member_ids?: Array<number>;
  shared?: boolean;
  stellar_case_min_score?: number;
  stellar_ia?: boolean;
  stellarcyber_cases?: boolean;
  updated_at?: string;
  updated_by_id?: number;
  vip?: boolean;
};

export type ZammadV1OrganizationGetAllNode = {
  type: 'n8n-nodes-base.zammad';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ZammadV1OrganizationGetAllParams>;
  output?: Items<ZammadV1OrganizationGetAllOutput>;
};
