// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/zohoCrm/v1/resource_vendor/operation_upsert.ts
/**
 * Zoho CRM Node - Version 1
 * Discriminator: resource=vendor, operation=upsert
 */


interface Credentials {
  zohoOAuth2Api: CredentialReference;
}

/** Create a new record, or update the current one if it already exists (upsert) */
export type ZohoCrmV1VendorUpsertParams = {
  resource: 'vendor';
  operation: 'upsert';
/**
 * Name of the vendor. If a record with this vendor name exists it will be updated, otherwise a new one will be created.
 */
    vendorName?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Address
     * @default {}
     */
    Address?: {
        /** Address Fields
     */
    address_fields?: {
      /** Street
       */
      Street?: string | Expression<string> | PlaceholderValue;
      /** City
       */
      City?: string | Expression<string> | PlaceholderValue;
      /** State
       */
      State?: string | Expression<string> | PlaceholderValue;
      /** Country
       */
      Country?: string | Expression<string> | PlaceholderValue;
      /** Zip Code
       */
      Zip_Code?: string | Expression<string> | PlaceholderValue;
    };
  };
    /** Category
     */
    Category?: string | Expression<string> | PlaceholderValue;
    /** Currency
     * @default USD
     */
    Currency?: 'USD' | 'EUR' | 'AED' | 'AFN' | 'ALL' | 'ARS' | 'AUD' | 'AZN' | 'BBD' | 'BDT' | 'BGN' | 'BMD' | 'BND' | 'BOB' | 'BRL' | 'BSD' | 'BWP' | 'BZD' | 'CAD' | 'CHF' | 'CLP' | 'CNY' | 'COP' | 'CRC' | 'CZK' | 'DKK' | 'DOP' | 'DZD' | 'EGP' | 'FJD' | 'GBP' | 'GTQ' | 'HKD' | 'HNL' | 'HRK' | 'HUF' | 'IDR' | 'ILS' | 'INR' | 'JMD' | 'JPY' | 'KES' | 'KRW' | 'KZT' | 'LAK' | 'LBP' | 'LKR' | 'LRD' | 'MAD' | 'MMK' | 'MOP' | 'MRO' | 'MUR' | 'MVR' | 'MXN' | 'MYR' | 'NIO' | 'NOK' | 'NPR' | 'NZD' | 'PEN' | 'PGK' | 'PHP' | 'PKR' | 'PLN' | 'QAR' | 'RON' | 'RUB' | 'SAR' | 'SBD' | 'SCR' | 'SEK' | 'SGD' | 'SYP' | 'THB' | 'TOP' | 'TRY' | 'TTD' | 'TWD' | 'UAH' | 'VND' | 'VUV' | 'WST' | 'XCD' | 'XOF' | 'YER' | 'ZAR' | Expression<string>;
    /** Filter by custom fields
     * @default {}
     */
    customFields?: {
        /** Custom Field
     */
    customFields?: Array<{
      /** Custom field to set a value to
       */
      fieldId?: string | Expression<string>;
      /** Value to set on custom field
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
    /** Description
     */
    Description?: string | Expression<string> | PlaceholderValue;
    /** Email
     */
    Email?: string | Expression<string> | PlaceholderValue;
    /** Phone
     */
    Phone?: string | Expression<string> | PlaceholderValue;
    /** Website
     */
    Website?: string | Expression<string> | PlaceholderValue;
  };
};

export type ZohoCrmV1VendorUpsertNode = {
  type: 'n8n-nodes-base.zohoCrm';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ZohoCrmV1VendorUpsertParams>;
};
