// @microsoft/agents-a365-tooling-extensions-langchain/dist/esm/McpToolRegistrationService.js
"use strict";
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
Object.defineProperty(exports, "__esModule", { value: true });
exports.McpToolRegistrationService = void 0;
// Microsoft Agent 365 SDK
const agents_a365_tooling_1 = require("@microsoft/agents-a365-tooling");
const agents_a365_runtime_1 = require("@microsoft/agents-a365-runtime");
// LangChain SDKs
const langchain_1 = require("langchain");
const mcp_adapters_1 = require("@langchain/mcp-adapters");
/**
 * Discover MCP servers and list tools formatted for the LangChain Orchestrator.
 * Uses listToolServers to fetch server configs and getTools to enumerate tools.
 */
class McpToolRegistrationService {
    constructor() {
        this.configService = new agents_a365_tooling_1.McpToolServerConfigurationService();
    }
    /**
     * Registers MCP tool servers and updates agent options with discovered tools and server configs.
     * Call this to enable dynamic LangChain tool access based on the current MCP environment.
     * @param agent The LangChain Agent instance to which MCP servers will be added.
     * @param authorization Authorization object for token exchange.
     * @param authHandlerName The name of the auth handler to use for token exchange.
     * @param turnContext The TurnContext of the current request.
     * @param authToken Optional bearer token for MCP server access.
     * @returns The updated Agent instance with registered MCP servers.
     */
    async addToolServersToAgent(agent, authorization, authHandlerName, turnContext, authToken) {
        if (!agent) {
            throw new Error('Langchain Agent is Required');
        }
        if (!authToken) {
            authToken = await agents_a365_runtime_1.AgenticAuthenticationService.GetAgenticUserToken(authorization, authHandlerName, turnContext);
        }
        // Validate the authentication token
        agents_a365_tooling_1.Utility.ValidateAuthToken(authToken);
        const agenticAppId = agents_a365_runtime_1.Utility.ResolveAgentIdentity(turnContext, authToken);
        const servers = await this.configService.listToolServers(agenticAppId, authToken);
        const mcpServers = {};
        for (const server of servers) {
            // Compose headers if values are available
            const headers = agents_a365_tooling_1.Utility.GetToolRequestHeaders(authToken, turnContext);
            // Create Connection instance for LangChain agents
            mcpServers[server.mcpServerName] = {
                type: 'http',
                url: server.url,
                headers: headers
            };
        }
        const mcpClientConfig = {};
        mcpClientConfig.mcpServers = Object.assign(mcpClientConfig.mcpServers ?? {}, mcpServers);
        const multiServerMcpClient = new mcp_adapters_1.MultiServerMCPClient(mcpClientConfig);
        const mcpTools = await multiServerMcpClient.getTools();
        // Merge existing agent tools with MCP tools
        const existingTools = agent.options.tools ?? [];
        const allTools = [...existingTools, ...mcpTools];
        // Create the agent with existing options and combined tools
        return (0, langchain_1.createAgent)({
            ...agent.options,
            tools: allTools,
        });
    }
}
exports.McpToolRegistrationService = McpToolRegistrationService;
