// @n8n/ai-workflow-builder/dist/types/categorization.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TechniqueDescription = exports.WorkflowTechnique = void 0;
exports.WorkflowTechnique = {
    SCHEDULING: 'scheduling',
    CHATBOT: 'chatbot',
    FORM_INPUT: 'form_input',
    SCRAPING_AND_RESEARCH: 'scraping_and_research',
    MONITORING: 'monitoring',
    ENRICHMENT: 'enrichment',
    TRIAGE: 'triage',
    CONTENT_GENERATION: 'content_generation',
    DOCUMENT_PROCESSING: 'document_processing',
    DATA_EXTRACTION: 'data_extraction',
    DATA_ANALYSIS: 'data_analysis',
    DATA_TRANSFORMATION: 'data_transformation',
    DATA_PERSISTENCE: 'data_persistence',
    NOTIFICATION: 'notification',
    KNOWLEDGE_BASE: 'knowledge_base',
    HUMAN_IN_THE_LOOP: 'human_in_the_loop',
};
exports.TechniqueDescription = {
    [exports.WorkflowTechnique.SCHEDULING]: 'Running an action at a specific time or interval',
    [exports.WorkflowTechnique.CHATBOT]: 'Receiving chat messages and replying (built-in chat, Telegram, Slack, MS Teams, etc.)',
    [exports.WorkflowTechnique.FORM_INPUT]: 'Gathering data from users via forms',
    [exports.WorkflowTechnique.SCRAPING_AND_RESEARCH]: 'Methodically collecting information from websites or APIs to compile structured data',
    [exports.WorkflowTechnique.MONITORING]: 'Repeatedly checking service/website status and taking action when conditions are met',
    [exports.WorkflowTechnique.ENRICHMENT]: 'Adding extra details to existing data by merging information from other sources',
    [exports.WorkflowTechnique.TRIAGE]: 'Classifying data for routing or prioritization',
    [exports.WorkflowTechnique.CONTENT_GENERATION]: 'Creating text, images, audio, video, etc.',
    [exports.WorkflowTechnique.DOCUMENT_PROCESSING]: 'Taking action on content within files (PDFs, Word docs, images)',
    [exports.WorkflowTechnique.DATA_EXTRACTION]: 'Pulling specific information from structured or unstructured inputs',
    [exports.WorkflowTechnique.DATA_ANALYSIS]: 'Examining data to find patterns, trends, anomalies, or insights',
    [exports.WorkflowTechnique.DATA_TRANSFORMATION]: 'Cleaning, formatting, or restructuring data (including summarization)',
    [exports.WorkflowTechnique.DATA_PERSISTENCE]: 'Storing, updating, or retrieving records from persistent storage (Google Sheets, Airtable, built-in Data Tables)',
    [exports.WorkflowTechnique.NOTIFICATION]: 'Sending alerts or updates via email, chat, SMS when events occur',
    [exports.WorkflowTechnique.KNOWLEDGE_BASE]: 'Building or using a centralized information collection (usually vector database for LLM use)',
    [exports.WorkflowTechnique.HUMAN_IN_THE_LOOP]: 'Pausing for human decision/input before resuming',
};
