// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/graphql/v1.schema.js
/**
 * GraphQL Node - Version 1 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    authentication: z.union([z.literal('basicAuth'), z.literal('customAuth'), z.literal('digestAuth'), z.literal('headerAuth'), z.literal('none'), z.literal('oAuth1'), z.literal('oAuth2'), z.literal('queryAuth'), expressionSchema]).optional(),
    requestMethod: z.union([z.literal('GET'), z.literal('POST'), expressionSchema]).optional(),
    endpoint: stringOrExpression.optional(),
    allowUnauthorizedCerts: booleanOrExpression.optional(),
    requestFormat: resolveSchema({ parameters, schema: z.union([z.literal('graphql'), z.literal('json'), expressionSchema]), required: false, displayOptions: {"show":{"requestMethod":["POST"]}}, defaults: {"requestMethod":"POST"} }),
    query: stringOrExpression.optional(),
    variables: resolveSchema({ parameters, schema: z.union([iDataObjectSchema, z.string()]), required: false, displayOptions: {"show":{"requestFormat":["json"],"requestMethod":["POST"]}}, defaults: {"requestFormat":"graphql","requestMethod":"POST"} }),
    operationName: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"requestFormat":["json"],"requestMethod":["POST"]}}, defaults: {"requestFormat":"graphql","requestMethod":"POST"} }),
    responseFormat: z.union([z.literal('json'), z.literal('string'), expressionSchema]).optional(),
    dataPropertyName: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"responseFormat":["string"]}}, defaults: {"responseFormat":"json"} }),
    headerParametersUi: z.object({ parameter: z.array(z.object({ name: stringOrExpression.optional(), value: stringOrExpression.optional() })).optional() }).optional(),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
