// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/onfleet/v1/resource_team/operation_create.ts
/**
 * Onfleet Node - Version 1
 * Discriminator: resource=team, operation=create
 */


interface Credentials {
  onfleetApi: CredentialReference;
}

/** Create a new Onfleet admin */
export type OnfleetV1TeamCreateParams = {
  resource: 'team';
  operation: 'create';
/**
 * A unique name for the team
 */
    name?: string | Expression<string> | PlaceholderValue;
/**
 * A list of workers. Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 * @default []
 */
    workers?: string[];
/**
 * A list of managing administrators. Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 * @default []
 */
    managers?: string[];
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** The team's hub. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    hub?: string | Expression<string>;
    /** Whether or not to allow drivers to self-assign tasks that are in the Team's unassigned container
     * @default false
     */
    enableSelfAssignment?: boolean | Expression<boolean>;
  };
};

export type OnfleetV1TeamCreateNode = {
  type: 'n8n-nodes-base.onfleet';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<OnfleetV1TeamCreateParams>;
};
