// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/keap/v1/resource_email/operation_create_record.schema.js
/**
 * Keap Node - Version 1 - Zod Schema
 * Discriminator: resource=email, operation=createRecord
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('email'),
      operation: z.literal('createRecord'),
      sentToAddress: stringOrExpression.optional(),
      sentFromAddress: stringOrExpression.optional(),
      additionalFields: z.object({ clickedDate: stringOrExpression.optional(), contactId: numberOrExpression.optional(), headers: stringOrExpression.optional(), htmlContent: stringOrExpression.optional(), openedDate: stringOrExpression.optional(), originalProvider: z.union([z.literal('UNKNOWN'), z.literal('INFUSIONSOFT'), z.literal('MICROSOFT'), z.literal('GOOGLE'), expressionSchema]).optional(), originalProviderId: stringOrExpression.optional(), plainContent: stringOrExpression.optional(), providerSourceId: stringOrExpression.optional(), receivedDate: stringOrExpression.optional(), sentDate: stringOrExpression.optional(), sentFromReplyAddress: stringOrExpression.optional(), sentToBccAddresses: stringOrExpression.optional(), sentToCCAddresses: stringOrExpression.optional(), subject: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
