// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/keap/v1/resource_email/operation_get_all.ts
/**
 * Keap Node - Version 1
 * Discriminator: resource=email, operation=getAll
 */


interface Credentials {
  keapOAuth2Api: CredentialReference;
}

/** Retrieve many companies */
export type KeapV1EmailGetAllParams = {
  resource: 'email';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Contact ID
     * @default 0
     */
    contactId?: number | Expression<number>;
    /** Email
     */
    email?: string | Expression<string> | PlaceholderValue;
    /** Emails sent since the provided date, must be present if untilDate is provided
     */
    sinceSentDate?: string | Expression<string>;
    /** Email sent until the provided date
     */
    untilSentDate?: string | Expression<string>;
  };
};

export type KeapV1EmailGetAllNode = {
  type: 'n8n-nodes-base.keap';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<KeapV1EmailGetAllParams>;
};
