// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/keap/v1/resource_email/operation_create_record.ts
/**
 * Keap Node - Version 1
 * Discriminator: resource=email, operation=createRecord
 */


interface Credentials {
  keapOAuth2Api: CredentialReference;
}

/** Create a record of an email sent to a contact */
export type KeapV1EmailCreateRecordParams = {
  resource: 'email';
  operation: 'createRecord';
/**
 * Sent To Address
 */
    sentToAddress?: string | Expression<string> | PlaceholderValue;
/**
 * Sent From Address
 */
    sentFromAddress?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Clicked Date
     */
    clickedDate?: string | Expression<string>;
    /** Contact ID
     * @default 0
     */
    contactId?: number | Expression<number>;
    /** Headers
     */
    headers?: string | Expression<string> | PlaceholderValue;
    /** Base64 encoded HTML
     */
    htmlContent?: string | Expression<string> | PlaceholderValue;
    /** Opened Date
     */
    openedDate?: string | Expression<string>;
    /** Provider that sent the email case insensitive, must be in list
     * @default UNKNOWN
     */
    originalProvider?: 'UNKNOWN' | 'INFUSIONSOFT' | 'MICROSOFT' | 'GOOGLE' | Expression<string>;
    /** Provider ID that sent the email, must be unique when combined with provider. If omitted a UUID without dashes is autogenerated for the record.
     */
    originalProviderId?: string | Expression<string> | PlaceholderValue;
    /** Base64 encoded text
     */
    plainContent?: string | Expression<string> | PlaceholderValue;
    /** Provider Source ID
     * @default The email address of the synced email account that generated this message.
     */
    providerSourceId?: string | Expression<string> | PlaceholderValue;
    /** Received Date
     */
    receivedDate?: string | Expression<string>;
    /** Sent Date
     */
    sentDate?: string | Expression<string>;
    /** Sent From Reply Address
     */
    sentFromReplyAddress?: string | Expression<string> | PlaceholderValue;
    /** Sent To Bcc Addresses
     */
    sentToBccAddresses?: string | Expression<string> | PlaceholderValue;
    /** Sent To CC Addresses
     */
    sentToCCAddresses?: string | Expression<string> | PlaceholderValue;
    /** Subject
     */
    subject?: string | Expression<string> | PlaceholderValue;
  };
};

export type KeapV1EmailCreateRecordNode = {
  type: 'n8n-nodes-base.keap';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<KeapV1EmailCreateRecordParams>;
};
