// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/keap/v1/resource_email/operation_send.schema.js
/**
 * Keap Node - Version 1 - Zod Schema
 * Discriminator: resource=email, operation=send
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('email'),
      operation: z.literal('send'),
      userId: stringOrExpression.optional(),
      contactIds: stringOrExpression.optional(),
      subject: stringOrExpression.optional(),
      additionalFields: z.object({ addressField: stringOrExpression.optional(), htmlContent: stringOrExpression.optional(), plainContent: stringOrExpression.optional() }).optional(),
      attachmentsUi: z.object({ attachmentsValues: z.array(z.object({ fileData: stringOrExpression.optional(), fileName: stringOrExpression.optional() })).optional(), attachmentsBinary: z.array(z.object({ property: stringOrExpression.optional() })).optional() }).optional(),
    }).optional(),
  });
};
