// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/keap/v1/resource_email/operation_send.ts
/**
 * Keap Node - Version 1
 * Discriminator: resource=email, operation=send
 */


interface Credentials {
  keapOAuth2Api: CredentialReference;
}

/** Send Email */
export type KeapV1EmailSendParams = {
  resource: 'email';
  operation: 'send';
/**
 * The infusionsoft user to send the email on behalf of. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    userId?: string | Expression<string>;
/**
 * Contact IDs to receive the email. Multiple can be added seperated by comma.
 */
    contactIds?: string | Expression<string> | PlaceholderValue;
/**
 * The subject line of the email
 */
    subject?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Email field of each Contact record to address the email to, such as 'EmailAddress1', 'EmailAddress2', 'EmailAddress3', defaulting to the contact's primary email
     */
    addressField?: string | Expression<string> | PlaceholderValue;
    /** The HTML-formatted content of the email, encoded in Base64
     */
    htmlContent?: string | Expression<string> | PlaceholderValue;
    /** The plain-text content of the email, encoded in Base64
     */
    plainContent?: string | Expression<string> | PlaceholderValue;
  };
/**
 * Attachments to be sent with each copy of the email, maximum of 10 with size of 1MB each
 * @default {}
 */
    attachmentsUi?: {
        /** Attachments Values
     */
    attachmentsValues?: Array<{
      /** The content of the attachment, encoded in Base64
       */
      fileData?: string | Expression<string> | PlaceholderValue;
      /** The filename of the attached file, including extension
       */
      fileName?: string | Expression<string> | PlaceholderValue;
    }>;
        /** Attachments Binary
     */
    attachmentsBinary?: Array<{
      /** Name of the binary properties which contain data which should be added to email as attachment
       */
      property?: string | Expression<string> | PlaceholderValue;
    }>;
  };
};

export type KeapV1EmailSendNode = {
  type: 'n8n-nodes-base.keap';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<KeapV1EmailSendParams>;
};
