// @n8n/n8n-nodes-langchain/dist/node-definitions/nodes/n8n-nodes-langchain/openAi/v16/resource_assistant/operation_message.schema.js
/**
 * OpenAI Node - Version 1.6 - Zod Schema
 * Discriminator: resource=assistant, operation=message
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema, toolInstanceSchema, memoryInstanceSchema }) {

  // Static subnode schema
  const subnodesSchema = z.object({
    tools: z.array(toolInstanceSchema).optional(),
    memory: memoryInstanceSchema.optional(),
  }).strict();

  return z.object({
    parameters: z.object({
      resource: z.literal('assistant'),
      operation: z.literal('message').default('message'),
      assistantId: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      prompt: z.union([z.literal('auto'), z.literal('guardrails'), z.literal('define'), expressionSchema]).optional(),
      text: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"prompt":["define"]}}, defaults: {"prompt":"auto"} }),
      memory: z.union([z.literal('connector'), z.literal('threadId'), expressionSchema]).optional(),
      threadId: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"memory":["threadId"]}}, defaults: {"memory":"connector"} }),
      options: z.object({ baseURL: stringOrExpression.optional(), maxRetries: numberOrExpression.optional(), timeout: numberOrExpression.optional(), preserveOriginalTools: booleanOrExpression.optional() }).optional(),
    }).optional(),
    subnodes: subnodesSchema.optional(),
  });
};
