// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/matrix/v1/resource_room_member/operation_get_all.schema.js
/**
 * Matrix Node - Version 1 - Zod Schema
 * Discriminator: resource=roomMember, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('roomMember'),
      operation: z.literal('getAll'),
      roomId: stringOrExpression.optional(),
      filters: z.object({ notMembership: z.union([z.literal(''), z.literal('ban'), z.literal('invite'), z.literal('join'), z.literal('leave'), expressionSchema]).optional(), membership: z.union([z.literal(''), z.literal('ban'), z.literal('invite'), z.literal('join'), z.literal('leave'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
