// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/questDb/v1.schema.js
/**
 * QuestDB Node - Version 1 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    operation: z.union([z.literal('executeQuery'), z.literal('insert')]).optional(),
    query: resolveSchema({ parameters, schema: z.string(), required: false, displayOptions: {"show":{"operation":["executeQuery"]}}, defaults: {"operation":"insert"} }),
    schema: resolveSchema({ parameters, schema: z.unknown(), required: false, displayOptions: {"show":{"operation":["insert"]}}, defaults: {"operation":"insert"} }),
    table: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"operation":["insert"]}}, defaults: {"operation":"insert"} }),
    columns: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"operation":["insert"]}}, defaults: {"operation":"insert"} }),
    returnFields: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"operation":["insert"]}}, defaults: {"operation":"insert"} }),
    additionalFields: resolveSchema({ parameters, schema: z.object({ mode: z.union([z.literal('independently'), z.literal('transaction'), expressionSchema]).optional(), queryParams: stringOrExpression.optional() }), required: false, displayOptions: {"show":{"operation":["executeQuery","insert"]}}, defaults: {"operation":"insert"} }),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
