// @n8n/n8n-nodes-langchain/dist/nodes/embeddings/EmbeddingsLemonade/EmbeddingsLemonade.node.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmbeddingsLemonade = void 0;
const openai_1 = require("@langchain/openai");
const n8n_workflow_1 = require("n8n-workflow");
const ai_utilities_1 = require("@n8n/ai-utilities");
const description_1 = require("../../llms/LMLemonade/description");
class EmbeddingsLemonade {
    constructor() {
        this.description = {
            displayName: 'Embeddings Lemonade',
            name: 'embeddingsLemonade',
            icon: 'file:lemonade.svg',
            group: ['transform'],
            version: 1,
            description: 'Use Lemonade Embeddings',
            defaults: {
                name: 'Embeddings Lemonade',
            },
            ...description_1.lemonadeDescription,
            codex: {
                categories: ['AI'],
                subcategories: {
                    AI: ['Embeddings'],
                },
                resources: {
                    primaryDocumentation: [
                        {
                            url: 'https://docs.n8n.io/integrations/builtin/cluster-nodes/sub-nodes/n8n-nodes-langchain.embeddingslemonade/',
                        },
                    ],
                },
            },
            inputs: [],
            outputs: [n8n_workflow_1.NodeConnectionTypes.AiEmbedding],
            outputNames: ['Embeddings'],
            properties: [(0, ai_utilities_1.getConnectionHintNoticeField)([n8n_workflow_1.NodeConnectionTypes.AiVectorStore]), description_1.lemonadeModel],
        };
    }
    async supplyData(itemIndex) {
        const modelName = this.getNodeParameter('model', itemIndex);
        const credentials = (await this.getCredentials('lemonadeApi'));
        const apiKey = credentials.apiKey || 'lemonade-placeholder-key';
        const configuration = {
            baseURL: credentials.baseUrl,
        };
        if (credentials.apiKey) {
            configuration.defaultHeaders = {
                Authorization: `Bearer ${credentials.apiKey}`,
            };
        }
        const embeddings = new openai_1.OpenAIEmbeddings({
            apiKey,
            model: modelName,
            configuration,
        });
        return {
            response: (0, ai_utilities_1.logWrapper)(embeddings, this),
        };
    }
}
exports.EmbeddingsLemonade = EmbeddingsLemonade;
