// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/monicaCrm/v1/resource_reminder/operation_create.schema.js
/**
 * Monica CRM Node - Version 1 - Zod Schema
 * Discriminator: resource=reminder, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('reminder'),
      operation: z.literal('create').default('create'),
      contactId: stringOrExpression.optional(),
      frequencyType: z.union([z.literal('one_time'), z.literal('week'), z.literal('month'), z.literal('year'), expressionSchema]).optional(),
      frequencyNumber: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"frequencyType":["week","month","year"]}}, defaults: {"frequencyType":"one_time"} }),
      initialDate: stringOrExpression.optional(),
      title: stringOrExpression.optional(),
      additionalFields: z.object({ description: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
