// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/monicaCrm/v1/resource_reminder/operation_create.ts
/**
 * Monica CRM Node - Version 1
 * Discriminator: resource=reminder, operation=create
 */


interface Credentials {
  monicaCrmApi: CredentialReference;
}

/** Create an activity */
export type MonicaCrmV1ReminderCreateParams = {
  resource: 'reminder';
  operation: 'create';
/**
 * ID of the contact to associate the reminder with
 */
    contactId?: string | Expression<string> | PlaceholderValue;
/**
 * Type of frequency of the reminder
 * @default one_time
 */
    frequencyType?: 'one_time' | 'week' | 'month' | 'year' | Expression<string>;
/**
 * Interval for the reminder
 * @displayOptions.show { frequencyType: ["week", "month", "year"] }
 * @default 0
 */
    frequencyNumber?: number | Expression<number>;
/**
 * Date of the reminder
 */
    initialDate?: string | Expression<string>;
/**
 * Title of the reminder - max 100,000 characters
 */
    title?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Description about the reminder - Max 100,000 characters
     */
    description?: string | Expression<string> | PlaceholderValue;
  };
};

export type MonicaCrmV1ReminderCreateNode = {
  type: 'n8n-nodes-base.monicaCrm';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MonicaCrmV1ReminderCreateParams>;
};
