// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/monicaCrm/v1/resource_reminder/operation_update.schema.js
/**
 * Monica CRM Node - Version 1 - Zod Schema
 * Discriminator: resource=reminder, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('reminder'),
      operation: z.literal('update'),
      reminderId: stringOrExpression.optional(),
      updateFields: z.object({ contact_id: stringOrExpression.optional(), description: stringOrExpression.optional(), frequency_type: z.union([z.literal('one_time'), z.literal('week'), z.literal('month'), z.literal('year'), expressionSchema]).optional(), initial_data: stringOrExpression.optional(), frequency_number: numberOrExpression.optional(), title: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
