// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/wekan/v1/resource_checklist/operation_get_all.ts
/**
 * Wekan Node - Version 1
 * Discriminator: resource=checklist, operation=getAll
 */


interface Credentials {
  wekanApi: CredentialReference;
}

/** Get many user boards */
export type WekanV1ChecklistGetAllParams = {
  resource: 'checklist';
  operation: 'getAll';
/**
 * The ID of the board that list belongs to. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    boardId?: string | Expression<string>;
/**
 * The ID of the list that card belongs to. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    listId?: string | Expression<string>;
/**
 * The ID of the card to get checklists. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    cardId?: string | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
};

export type WekanV1ChecklistGetAllNode = {
  type: 'n8n-nodes-base.wekan';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<WekanV1ChecklistGetAllParams>;
};
