// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleFirebaseRealtimeDatabase/v1.schema.js
/**
 * Google Cloud Realtime Database Node - Version 1 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    projectId: stringOrExpression.optional(),
    operation: z.union([z.literal('create'), z.literal('delete'), z.literal('get'), z.literal('push'), z.literal('update')]).optional(),
    path: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"operation":["get"]},"hide":{"operation":["get"]}}, defaults: {"operation":"create"} }),
    attributes: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"operation":["create","push","update"]}}, defaults: {"operation":"create"} }),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
