// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/jinaAi/v1/resource_research/operation_deep_research.schema.js
/**
 * Jina AI Node - Version 1 - Zod Schema
 * Discriminator: resource=research, operation=deepResearch
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('research'),
      operation: z.literal('deepResearch'),
      researchQuery: stringOrExpression.optional(),
      simplify: booleanOrExpression.optional(),
      options: z.object({ maxReturnedSources: numberOrExpression.optional(), prioritizeSources: stringOrExpression.optional(), excludeSources: stringOrExpression.optional(), siteFilter: stringOrExpression.optional() }).optional(),
      requestOptions: z.object({ batching: z.unknown().optional(), allowUnauthorizedCerts: z.boolean().optional(), proxy: stringOrExpression.optional(), timeout: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
