// @n8n/n8n-nodes-langchain/dist/node-definitions/nodes/n8n-nodes-langchain/agentTool/v3.schema.js
/**
 * AI Agent Tool Node - Version 3 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema, languageModelInstanceSchema, memoryInstanceSchema, toolInstanceSchema, outputParserInstanceSchema }) {

  // Helper function for conditional subnode schema
  function getSubnodesSchema() {
    return z.object({
      model: z.union([languageModelInstanceSchema, z.array(languageModelInstanceSchema)]),
      memory: memoryInstanceSchema.optional(),
      tools: z.array(toolInstanceSchema).optional(),
      outputParser: resolveSchema({ parameters, schema: outputParserInstanceSchema, required: false, displayOptions: {"show":{"hasOutputParser":[true]}}, defaults: {"hasOutputParser":false} }),
    }).strict();
  }

  // Parameters schema
  const parametersSchema = z.object({
    toolDescription: stringOrExpression.optional(),
    text: stringOrExpression.optional(),
    hasOutputParser: z.boolean().optional(),
    needsFallback: z.boolean().optional(),
    options: z.object({ systemMessage: stringOrExpression.optional(), maxIterations: numberOrExpression.optional(), returnIntermediateSteps: booleanOrExpression.optional(), passthroughBinaryImages: booleanOrExpression.optional(), enableStreaming: booleanOrExpression.optional(), batching: z.unknown().optional(), maxTokensFromMemory: z.unknown().optional() }).optional(),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
    subnodes: getSubnodesSchema(),
  });
};
