// @n8n/n8n-nodes-langchain/dist/node-definitions/nodes/n8n-nodes-langchain/agentTool/v3.ts
/**
 * AI Agent Tool Node - Version 3
 * Generates an action plan and executes it. Can use external tools.
 */


export interface LcAgentToolV3Params {
/**
 * Explain to the LLM what this tool does, a good, specific description would allow LLMs to produce expected results much more often
 * @default AI Agent that can call other tools
 */
    toolDescription?: string | Expression<string> | PlaceholderValue;
/**
 * Prompt (User Message)
 * @builderHint Use expressions to include dynamic data from previous nodes (e.g., "={{ $json.input }}"). Static text prompts ignore incoming data.
 */
    text?: string | Expression<string>;
  hasOutputParser?: boolean;
  needsFallback?: boolean;
  options?: {
    /** The message that will be sent to the agent before the conversation starts
     * @builderHint Must include: agent's purpose, exact names of connected tools, and response instructions
     * @default You are a helpful assistant
     */
    systemMessage?: string | Expression<string> | PlaceholderValue;
    /** The maximum number of iterations the agent will run before stopping
     * @default 10
     */
    maxIterations?: number | Expression<number>;
    /** Whether or not the output should include intermediate steps the agent took
     * @default false
     */
    returnIntermediateSteps?: boolean | Expression<boolean>;
    /** Whether or not binary images should be automatically passed through to the agent as image type messages
     * @default true
     */
    passthroughBinaryImages?: boolean | Expression<boolean>;
    /** Whether this agent will stream the response in real-time as it generates text
     * @default true
     */
    enableStreaming?: boolean | Expression<boolean>;
    /** Batch processing options for rate limiting
     * @default {}
     */
    batching?: {
    /** How many items to process in parallel. This is useful for rate limiting, but might impact the log output ordering.
     * @default 1
     */
    batchSize?: number | Expression<number>;
    /** Delay in milliseconds between batches. This is useful for rate limiting.
     * @default 0
     */
    delayBetweenBatches?: number | Expression<number>;
  };
    /** The maximum number of tokens to read from the chat memory history. Set to 0 to read all history.
     * @default 0
     */
    maxTokensFromMemory?: unknown;
  };
}

export interface LcAgentToolV3SubnodeConfig {
  model: LanguageModelInstance | LanguageModelInstance[];
  memory?: MemoryInstance;
  tools?: ToolInstance[];
  /**
   * @displayOptions.show { hasOutputParser: [true] }
   */
  outputParser?: OutputParserInstance;
}

interface LcAgentToolV3NodeBase {
  type: '@n8n/n8n-nodes-langchain.agentTool';
  version: 3;
}

export type LcAgentToolV3ParamsNode = LcAgentToolV3NodeBase & {
  config: NodeConfig<LcAgentToolV3Params> & { subnodes: LcAgentToolV3SubnodeConfig };
};

export type LcAgentToolV3Node = LcAgentToolV3ParamsNode;
