// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mattermost/v1/resource_message/operation_post_ephemeral.ts
/**
 * Mattermost Node - Version 1
 * Discriminator: resource=message, operation=postEphemeral
 */


interface Credentials {
  mattermostApi: CredentialReference;
}

/** Post an ephemeral message into a channel */
export type MattermostV1MessagePostEphemeralParams = {
  resource: 'message';
  operation: 'postEphemeral';
/**
 * ID of the user to send the ephemeral message to. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    userId?: string | Expression<string>;
/**
 * ID of the channel to send the ephemeral message in. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    channelId?: string | Expression<string>;
/**
 * Text to send in the ephemeral message
 */
    message?: string | Expression<string> | PlaceholderValue;
};

export type MattermostV1MessagePostEphemeralNode = {
  type: 'n8n-nodes-base.mattermost';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MattermostV1MessagePostEphemeralParams>;
};
