// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/stripe/v1/resource_token/operation_create.ts
/**
 * Stripe Node - Version 1
 * Discriminator: resource=token, operation=create
 */


interface Credentials {
  stripeApi: CredentialReference;
}

/** Create a charge */
export type StripeV1TokenCreateParams = {
  resource: 'token';
  operation: 'create';
/**
 * Type of token to create
 * @default cardToken
 */
    type?: 'cardToken' | Expression<string>;
/**
 * Card Number
 * @displayOptions.show { type: ["cardToken"] }
 */
    number?: string | Expression<string> | PlaceholderValue;
/**
 * Security code printed on the back of the card
 * @displayOptions.show { type: ["cardToken"] }
 */
    cvc?: string | Expression<string> | PlaceholderValue;
/**
 * Number of the month when the card will expire
 * @displayOptions.show { type: ["cardToken"] }
 */
    expirationMonth?: string | Expression<string> | PlaceholderValue;
/**
 * Year when the card will expire
 * @displayOptions.show { type: ["cardToken"] }
 */
    expirationYear?: string | Expression<string> | PlaceholderValue;
};

export type StripeV1TokenCreateNode = {
  type: 'n8n-nodes-base.stripe';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<StripeV1TokenCreateParams>;
};
