// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/stripe/v1/resource_token/operation_create.schema.js
/**
 * Stripe Node - Version 1 - Zod Schema
 * Discriminator: resource=token, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('token'),
      operation: z.literal('create'),
      type: z.union([z.literal('cardToken'), expressionSchema]).optional(),
      number: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"type":["cardToken"]}}, defaults: {"type":"cardToken"} }),
      cvc: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"type":["cardToken"]}}, defaults: {"type":"cardToken"} }),
      expirationMonth: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"type":["cardToken"]}}, defaults: {"type":"cardToken"} }),
      expirationYear: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"type":["cardToken"]}}, defaults: {"type":"cardToken"} }),
    }).optional(),
  });
};
