// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/salesforce/v1/resource_account/operation_get.ts
/**
 * Salesforce Node - Version 1
 * Discriminator: resource=account, operation=get
 */


interface Credentials {
  salesforceOAuth2Api: CredentialReference;
  salesforceJwtApi: CredentialReference;
}

/** Represents an individual account, which is an organization or person involved with your business (such as customers, competitors, and partners) */
export type SalesforceV1AccountGetParams = {
  resource: 'account';
  operation: 'get';
/**
 * OAuth Authorization Flow
 * @default oAuth2
 */
    authentication?: 'oAuth2' | 'jwt' | Expression<string>;
/**
 * ID of account that needs to be fetched
 */
    accountId?: string | Expression<string> | PlaceholderValue;
};

export type SalesforceV1AccountGetOutput = {
  attributes?: {
    type?: string;
    url?: string;
  };
  BillingAddress?: {
    geocodeAccuracy?: null;
  };
  BillingGeocodeAccuracy?: null;
  CreatedById?: string;
  CreatedDate?: string;
  error?: string;
  Id?: string;
  IsDeleted?: boolean;
  Jigsaw?: null;
  JigsawCompanyId?: null;
  LastModifiedById?: string;
  LastModifiedDate?: string;
  MasterRecordId?: null;
  Name?: string;
  OwnerId?: string;
  ShippingGeocodeAccuracy?: null;
  SicDesc?: null;
  SystemModstamp?: string;
};

export type SalesforceV1AccountGetNode = {
  type: 'n8n-nodes-base.salesforce';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SalesforceV1AccountGetParams>;
  output?: Items<SalesforceV1AccountGetOutput>;
};
