// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/salesforce/v1/resource_account/operation_get_all.ts
/**
 * Salesforce Node - Version 1
 * Discriminator: resource=account, operation=getAll
 */


interface Credentials {
  salesforceOAuth2Api: CredentialReference;
  salesforceJwtApi: CredentialReference;
}

/** Represents an individual account, which is an organization or person involved with your business (such as customers, competitors, and partners) */
export type SalesforceV1AccountGetAllParams = {
  resource: 'account';
  operation: 'getAll';
/**
 * OAuth Authorization Flow
 * @default oAuth2
 */
    authentication?: 'oAuth2' | 'jwt' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Fields to include separated by ,
     */
    fields?: string | Expression<string> | PlaceholderValue;
    /** The condition to set
     * @default {}
     */
    conditionsUi?: {
        /** Condition
     */
    conditionValues?: Array<{
      /** For date, number, or boolean, please use expressions. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
       */
      field?: string | Expression<string>;
      /** Operation
       * @default equal
       */
      operation?: '<' | '<=' | 'equal' | '>' | '>=';
      /** Value
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
  };
};

export type SalesforceV1AccountGetAllOutput = {
  attributes?: {
    type?: string;
    url?: string;
  };
  Id?: string;
  Name?: string;
};

export type SalesforceV1AccountGetAllNode = {
  type: 'n8n-nodes-base.salesforce';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SalesforceV1AccountGetAllParams>;
  output?: Items<SalesforceV1AccountGetAllOutput>;
};
