// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/helpScout/v1/resource_customer/operation_get_all.schema.js
/**
 * Help Scout Node - Version 1 - Zod Schema
 * Discriminator: resource=customer, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('customer'),
      operation: z.literal('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      options: z.object({ firstName: stringOrExpression.optional(), lastName: stringOrExpression.optional(), mailbox: stringOrExpression.optional(), modifiedSince: stringOrExpression.optional(), sortField: z.union([z.literal('score'), z.literal('firstName'), z.literal('lastName'), z.literal('modifiedAt'), expressionSchema]).optional(), sortOrder: z.union([z.literal('asc'), z.literal('desc'), expressionSchema]).optional(), query: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
