// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/koBoToolbox/v1/resource_file/operation_create.schema.js
/**
 * KoBoToolbox Node - Version 1 - Zod Schema
 * Discriminator: resource=file, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('file'),
      operation: z.literal('create'),
      formId: stringOrExpression.optional(),
      fileMode: z.union([z.literal('binary'), z.literal('url'), expressionSchema]).optional(),
      binaryPropertyName: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"fileMode":["binary"]}}, defaults: {"fileMode":"binary"} }),
      fileUrl: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"fileMode":["url"]}}, defaults: {"fileMode":"binary"} }),
    }).optional(),
  });
};
