// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleCalendar/v1/resource_calendar/operation_availability.ts
/**
 * Google Calendar Node - Version 1
 * Discriminator: resource=calendar, operation=availability
 */


interface Credentials {
  googleCalendarOAuth2Api: CredentialReference;
}

/** If a time-slot is available in a calendar */
export type GoogleCalendarV1CalendarAvailabilityParams = {
  resource: 'calendar';
  operation: 'availability';
/**
 * Google Calendar to operate on
 * @default {"mode":"list","value":""}
 */
    calendar?: { __rl: true; mode: 'list' | 'id'; value: string; cachedResultName?: string };
/**
 * Start of the interval
 */
    timeMin?: string | Expression<string>;
/**
 * End of the interval
 */
    timeMax?: string | Expression<string>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** The format to return the data in
     * @default availability
     */
    outputFormat?: 'availability' | 'bookedSlots' | 'raw' | Expression<string>;
    /** Time zone used in the response. By default n8n timezone is used.
     * @default {"mode":"list","value":""}
     */
    timezone?: { __rl: true; mode: 'list' | 'id'; value: string; cachedResultName?: string };
  };
};

export type GoogleCalendarV1CalendarAvailabilityOutput = {
  available?: boolean;
};

export type GoogleCalendarV1CalendarAvailabilityNode = {
  type: 'n8n-nodes-base.googleCalendar';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GoogleCalendarV1CalendarAvailabilityParams>;
  output?: Items<GoogleCalendarV1CalendarAvailabilityOutput>;
};
