// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/uptimeRobot/v1/resource_monitor/operation_get_all.schema.js
/**
 * UptimeRobot Node - Version 1 - Zod Schema
 * Discriminator: resource=monitor, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('monitor'),
      operation: z.literal('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filters: z.object({ alert_contacts: booleanOrExpression.optional(), logs: booleanOrExpression.optional(), mwindow: booleanOrExpression.optional(), monitors: stringOrExpression.optional(), response_times: booleanOrExpression.optional(), search: stringOrExpression.optional(), statuses: z.array(z.union([z.literal(0), z.literal(1), z.literal(2), z.literal(8), z.literal(9)])).optional(), types: z.array(z.union([z.literal(5), z.literal(1), z.literal(2), z.literal(3), z.literal(4)])).optional() }).optional(),
    }).optional(),
  });
};
