// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/uptimeRobot/v1/resource_monitor/operation_update.schema.js
/**
 * UptimeRobot Node - Version 1 - Zod Schema
 * Discriminator: resource=monitor, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('monitor'),
      operation: z.literal('update'),
      id: stringOrExpression.optional(),
      updateFields: z.object({ friendly_name: stringOrExpression.optional(), http_auth_type: z.union([z.literal(1), z.literal(2), expressionSchema]).optional(), http_method: z.union([z.literal(6), z.literal(2), z.literal(1), z.literal(7), z.literal(5), z.literal(3), z.literal(4), expressionSchema]).optional(), http_password: stringOrExpression.optional(), http_username: stringOrExpression.optional(), interval: numberOrExpression.optional(), port: numberOrExpression.optional(), status: z.union([z.literal(0), z.literal(1), expressionSchema]).optional(), sub_type: z.union([z.literal(99), z.literal(3), z.literal(1), z.literal(2), z.literal(6), z.literal(5), z.literal(4), expressionSchema]).optional(), url: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
