// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/flow/v1/resource_task/operation_get_all.schema.js
/**
 * Flow Node - Version 1 - Zod Schema
 * Discriminator: resource=task, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('task').default('task'),
      operation: z.literal('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filters: z.object({ include: z.array(z.union([z.literal('schedule'), z.literal('files'), z.literal('file_associations'), z.literal('parent')])).optional(), order: z.union([z.literal('account_id'), z.literal('completed_at'), z.literal('created_at'), z.literal('due_on'), z.literal('list_id'), z.literal('name'), z.literal('owner_id'), z.literal('position'), z.literal('section_id'), z.literal('starts_on'), z.literal('updated_at'), expressionSchema]).optional(), workspaceId: stringOrExpression.optional(), createdBefore: stringOrExpression.optional(), createdAfter: stringOrExpression.optional(), updateBefore: stringOrExpression.optional(), updateAfter: stringOrExpression.optional(), deleted: booleanOrExpression.optional(), cleared: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
