// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/flow/v1/resource_task/operation_create.schema.js
/**
 * Flow Node - Version 1 - Zod Schema
 * Discriminator: resource=task, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('task').default('task'),
      operation: z.literal('create').default('create'),
      workspaceId: stringOrExpression.optional(),
      name: stringOrExpression.optional(),
      additionalFields: z.object({ ownerid: stringOrExpression.optional(), listID: stringOrExpression.optional(), startsOn: stringOrExpression.optional(), dueOn: stringOrExpression.optional(), mirrorParentSubscribers: booleanOrExpression.optional(), mirrorParentTags: booleanOrExpression.optional(), noteContent: stringOrExpression.optional(), noteMimeType: z.union([z.literal('text/plain'), z.literal('text/x-markdown'), z.literal('text/html'), expressionSchema]).optional(), parentId: stringOrExpression.optional(), positionList: numberOrExpression.optional(), positionUpcoming: numberOrExpression.optional(), position: numberOrExpression.optional(), sectionId: stringOrExpression.optional(), tags: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
