// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/flow/v1/resource_task/operation_get.schema.js
/**
 * Flow Node - Version 1 - Zod Schema
 * Discriminator: resource=task, operation=get
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('task').default('task'),
      operation: z.literal('get'),
      taskId: stringOrExpression.optional(),
      filters: z.object({ include: z.array(z.union([z.literal('schedule'), z.literal('files'), z.literal('file_associations'), z.literal('parent')])).optional() }).optional(),
    }).optional(),
  });
};
