// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/flow/v1/resource_task/operation_get_all.ts
/**
 * Flow Node - Version 1
 * Discriminator: resource=task, operation=getAll
 */


interface Credentials {
  flowApi: CredentialReference;
}

/** Tasks are units of work that can be private or assigned to a list. Through this endpoint, you can manipulate your tasks in Flow, including creating new ones. */
export type FlowV1TaskGetAllParams = {
  resource: 'task';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Include
     * @default []
     */
    include?: Array<'schedule' | 'files' | 'file_associations' | 'parent'>;
    /** Order
     * @default created_at
     */
    order?: 'account_id' | 'completed_at' | 'created_at' | 'due_on' | 'list_id' | 'name' | 'owner_id' | 'position' | 'section_id' | 'starts_on' | 'updated_at' | Expression<string>;
    /** Create resources under the given workspace
     */
    workspaceId?: string | Expression<string> | PlaceholderValue;
    /** Select resources created before a certain time
     */
    createdBefore?: string | Expression<string>;
    /** Select resources created after a certain time
     */
    createdAfter?: string | Expression<string>;
    /** Select resources updated before a certain time
     */
    updateBefore?: string | Expression<string>;
    /** Select resources updated after a certain time
     */
    updateAfter?: string | Expression<string>;
    /** Whether to select deleted resources
     * @default false
     */
    deleted?: boolean | Expression<boolean>;
    /** Whether to select cleared resources
     * @default false
     */
    cleared?: boolean | Expression<boolean>;
  };
};

export type FlowV1TaskGetAllNode = {
  type: 'n8n-nodes-base.flow';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<FlowV1TaskGetAllParams>;
};
