// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/intercom/v1/resource_company/operation_get_all.ts
/**
 * Intercom Node - Version 1
 * Discriminator: resource=company, operation=getAll
 */


interface Credentials {
  intercomApi: CredentialReference;
}

/** Companies allow you to represent commercial organizations using your product */
export type IntercomV1CompanyGetAllParams = {
  resource: 'company';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Segment representing the Lead
     */
    segment_id?: string | Expression<string> | PlaceholderValue;
    /** Tag representing the Lead
     */
    tag_id?: string | Expression<string> | PlaceholderValue;
  };
};

export type IntercomV1CompanyGetAllNode = {
  type: 'n8n-nodes-base.intercom';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<IntercomV1CompanyGetAllParams>;
};
