// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/telegram/v12/resource_message/operation_delete_message.ts
/**
 * Telegram Node - Version 1.2
 * Discriminator: resource=message, operation=deleteMessage
 */


interface Credentials {
  telegramApi: CredentialReference;
}

/** Delete a chat message */
export type TelegramV12MessageDeleteMessageParams = {
  resource: 'message';
  operation: 'deleteMessage';
/**
 * Unique identifier for the target chat or username, To find your chat ID ask @get_id_bot
 */
    chatId?: string | Expression<string> | PlaceholderValue;
/**
 * Unique identifier of the message to delete
 */
    messageId?: string | Expression<string> | PlaceholderValue;
/**
 * Force Reply
 * @displayOptions.show { replyMarkup: ["forceReply"] }
 * @default {}
 */
    forceReply?: {
    /** Whether to show reply interface to the user, as if they manually selected the bot‘s message and tapped ’Reply
     * @default false
     */
    force_reply?: boolean | Expression<boolean>;
    /** Whether to force reply from specific users only
     * @default false
     */
    selective?: boolean | Expression<boolean>;
  };
/**
 * Adds an inline keyboard that appears right next to the message it belongs to
 * @displayOptions.show { replyMarkup: ["inlineKeyboard"] }
 * @default {}
 */
    inlineKeyboard?: {
        /** Rows
     */
    rows?: Array<{
      /** The value to set
       * @default {}
       */
      row?: {
        /** Buttons
     */
    buttons?: Array<{
      /** Label text on the button
       */
      text?: string | Expression<string> | PlaceholderValue;
      /** Additional Fields
       * @default {}
       */
      additionalFields?: {
    /** Data to be sent in a callback query to the bot when button is pressed, 1-64 bytes
     */
    callback_data?: string | Expression<string> | PlaceholderValue;
    /** Whether to send a Pay button
     * @default false
     */
    pay?: boolean | Expression<boolean>;
    /** If set, pressing the button will insert the bot‘s username and the specified inline query in the current chat's input field.Can be empty, in which case only the bot’s username will be inserted
     */
    switch_inline_query_current_chat?: string | Expression<string> | PlaceholderValue;
    /** If set, pressing the button will prompt the user to select one of their chats, open that chat and insert the bot‘s username and the specified inline query in the input field. Can be empty, in which case just the bot’s username will be inserted.
     */
    switch_inline_query?: string | Expression<string> | PlaceholderValue;
    /** HTTP or tg:// URL to be opened when button is pressed
     */
    url?: string | Expression<string> | PlaceholderValue;
    /** Launch the Telegram Web App
     * @default {}
     */
    web_app?: {
    /** An HTTPS URL of a Web App to be opened
     */
    url?: string | Expression<string> | PlaceholderValue;
  };
  };
    }>;
  };
    }>;
  };
/**
 * Adds a custom keyboard with reply options
 * @displayOptions.show { replyMarkup: ["replyKeyboard"] }
 * @default {}
 */
    replyKeyboard?: {
        /** Rows
     */
    rows?: Array<{
      /** The value to set
       * @default {}
       */
      row?: {
        /** Buttons
     */
    buttons?: Array<{
      /** Text of the button. If none of the optional fields are used, it will be sent as a message when the button is pressed.
       */
      text?: string | Expression<string> | PlaceholderValue;
      /** Additional Fields
       * @default {}
       */
      additionalFields?: {
    /** Whether the user's phone number will be sent as a contact when the button is pressed.Available in private chats only
     * @default false
     */
    request_contact?: boolean | Expression<boolean>;
    /** Whether the user's request_location
     * @default false
     */
    request_location?: boolean | Expression<boolean>;
    /** Launch the Telegram Web App
     * @default {}
     */
    web_app?: {
    /** An HTTPS URL of a Web App to be opened
     */
    url?: string | Expression<string> | PlaceholderValue;
  };
  };
    }>;
  };
    }>;
  };
/**
 * Reply Keyboard Options
 * @displayOptions.show { replyMarkup: ["replyKeyboard"] }
 * @default {}
 */
    replyKeyboardOptions?: {
    /** Whether to request clients to resize the keyboard vertically for optimal fit
     * @default false
     */
    resize_keyboard?: boolean | Expression<boolean>;
    /** Whether to request clients to hide the keyboard as soon as it's been used
     * @default false
     */
    one_time_keyboard?: boolean | Expression<boolean>;
    /** Whether to show the keyboard to specific users only
     * @default false
     */
    selective?: boolean | Expression<boolean>;
  };
/**
 * Reply Keyboard Remove
 * @displayOptions.show { replyMarkup: ["replyKeyboardRemove"] }
 * @default {}
 */
    replyKeyboardRemove?: {
    /** Whether to request clients to remove the custom keyboard
     * @default false
     */
    remove_keyboard?: boolean | Expression<boolean>;
    /** Whether to force reply from specific users only
     * @default false
     */
    selective?: boolean | Expression<boolean>;
  };
};

export type TelegramV12MessageDeleteMessageOutput = {
  ok?: boolean;
  result?: boolean;
};

export type TelegramV12MessageDeleteMessageNode = {
  type: 'n8n-nodes-base.telegram';
  version: 1.2;
  credentials?: Credentials;
  config: NodeConfig<TelegramV12MessageDeleteMessageParams>;
  output?: Items<TelegramV12MessageDeleteMessageOutput>;
};
