// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleCloudStorage/v1/resource_object/operation_get_all.ts
/**
 * Google Cloud Storage Node - Version 1
 * Discriminator: resource=object, operation=getAll
 */


interface Credentials {
  googleCloudStorageOAuth2Api: CredentialReference;
}

/** Get list of Buckets */
export type GoogleCloudStorageV1ObjectGetAllParams = {
  resource: 'object';
  operation: 'getAll';
/**
 * Bucket Name
 */
    bucketName?: string | Expression<string> | PlaceholderValue;
/**
 * Projection
 * @default noAcl
 */
    projection?: 'full' | 'noAcl';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    maxResults?: number | Expression<number>;
/**
 * Additional Parameters
 * @default {}
 */
    listFilters?: {
    /** Delimiter
     * @default /
     */
    delimiter?: string | Expression<string> | PlaceholderValue;
    /** End Offset
     */
    endOffset?: string | Expression<string> | PlaceholderValue;
    /** Include Trailing Delimiter
     * @default false
     */
    includeTrailingDelimiter?: boolean | Expression<boolean>;
    /** Prefix
     */
    prefix?: string | Expression<string> | PlaceholderValue;
    /** Start Offset
     */
    startOffset?: string | Expression<string> | PlaceholderValue;
    /** Versions
     * @default false
     */
    versions?: boolean | Expression<boolean>;
  };
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type GoogleCloudStorageV1ObjectGetAllOutput = {
  bucket?: string;
  contentType?: string;
  crc32c?: string;
  etag?: string;
  eventBasedHold?: boolean;
  generation?: string;
  id?: string;
  kind?: string;
  md5Hash?: string;
  mediaLink?: string;
  metageneration?: string;
  name?: string;
  selfLink?: string;
  size?: string;
  storageClass?: string;
  temporaryHold?: boolean;
  timeCreated?: string;
  timeFinalized?: string;
  timeStorageClassUpdated?: string;
  updated?: string;
};

export type GoogleCloudStorageV1ObjectGetAllNode = {
  type: 'n8n-nodes-base.googleCloudStorage';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GoogleCloudStorageV1ObjectGetAllParams>;
  output?: Items<GoogleCloudStorageV1ObjectGetAllOutput>;
};
