// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleCloudStorage/v1/resource_object/operation_get_all.schema.js
/**
 * Google Cloud Storage Node - Version 1 - Zod Schema
 * Discriminator: resource=object, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('object'),
      operation: z.literal('getAll').default('getAll'),
      bucketName: stringOrExpression.optional(),
      projection: z.union([z.literal('full'), z.literal('noAcl')]).optional(),
      returnAll: booleanOrExpression.optional(),
      maxResults: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      listFilters: z.object({ delimiter: stringOrExpression.optional(), endOffset: stringOrExpression.optional(), includeTrailingDelimiter: booleanOrExpression.optional(), prefix: stringOrExpression.optional(), startOffset: stringOrExpression.optional(), versions: booleanOrExpression.optional() }).optional(),
      requestOptions: z.object({ batching: z.unknown().optional(), allowUnauthorizedCerts: z.boolean().optional(), proxy: stringOrExpression.optional(), timeout: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
