// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleCloudStorage/v1/resource_object/operation_get.schema.js
/**
 * Google Cloud Storage Node - Version 1 - Zod Schema
 * Discriminator: resource=object, operation=get
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('object'),
      operation: z.literal('get'),
      bucketName: stringOrExpression.optional(),
      objectName: stringOrExpression.optional(),
      projection: z.union([z.literal('full'), z.literal('noAcl')]).optional(),
      alt: z.union([z.literal('json'), z.literal('media'), expressionSchema]).optional(),
      binaryPropertyName: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"alt":["media"]}}, defaults: {"alt":"json"} }),
      getParameters: z.object({ generation: numberOrExpression.optional(), ifGenerationMatch: numberOrExpression.optional(), ifGenerationNotMatch: numberOrExpression.optional(), ifMetagenerationMatch: numberOrExpression.optional(), ifMetagenerationNotMatch: numberOrExpression.optional() }).optional(),
      encryptionHeaders: z.object({ 'X-Goog-Encryption-Algorithm': z.union([z.literal('AES256'), expressionSchema]).optional(), 'X-Goog-Encryption-Key': stringOrExpression.optional(), 'X-Goog-Encryption-Key-Sha256': stringOrExpression.optional() }).optional(),
      requestOptions: z.object({ batching: z.unknown().optional(), allowUnauthorizedCerts: z.boolean().optional(), proxy: stringOrExpression.optional(), timeout: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
