// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleCloudStorage/v1/resource_object/operation_delete.ts
/**
 * Google Cloud Storage Node - Version 1
 * Discriminator: resource=object, operation=delete
 */


interface Credentials {
  googleCloudStorageOAuth2Api: CredentialReference;
}

/** Delete an empty Bucket */
export type GoogleCloudStorageV1ObjectDeleteParams = {
  resource: 'object';
  operation: 'delete';
/**
 * Bucket Name
 */
    bucketName?: string | Expression<string> | PlaceholderValue;
/**
 * Object Name
 */
    objectName?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Parameters
 * @default {}
 */
    getParameters?: {
    /** Generation
     * @default -1
     */
    generation?: number | Expression<number>;
    /** Generation Match
     * @default -1
     */
    ifGenerationMatch?: number | Expression<number>;
    /** Generation Exclude
     * @default -1
     */
    ifGenerationNotMatch?: number | Expression<number>;
    /** Metageneration Match
     * @default -1
     */
    ifMetagenerationMatch?: number | Expression<number>;
    /** Metageneration Exclude
     * @default -1
     */
    ifMetagenerationNotMatch?: number | Expression<number>;
  };
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type GoogleCloudStorageV1ObjectDeleteOutput = {
  error?: string;
};

export type GoogleCloudStorageV1ObjectDeleteNode = {
  type: 'n8n-nodes-base.googleCloudStorage';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GoogleCloudStorageV1ObjectDeleteParams>;
  output?: Items<GoogleCloudStorageV1ObjectDeleteOutput>;
};
