// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/facebookLeadAdsTrigger/v1.ts
/**
 * Facebook Lead Ads Trigger Node - Version 1
 * Handle Facebook Lead Ads events via webhooks
 */


export interface FacebookLeadAdsTriggerV1Params {
  event?: 'newLead' | Expression<string>;
/**
 * The page linked to the form for retrieving new leads
 * @default {"mode":"list","value":""}
 */
    page?: { __rl: true; mode: 'list' | 'id'; value: string; cachedResultName?: string };
/**
 * The form to monitor for fetching lead details upon submission
 * @default {"mode":"list","value":""}
 */
    form?: { __rl: true; mode: 'list' | 'id'; value: string; cachedResultName?: string };
  options?: {
    /** Whether to return a simplified version of the webhook event instead of all fields
     * @default true
     */
    simplifyOutput?: boolean | Expression<boolean>;
  };
}

export interface FacebookLeadAdsTriggerV1Credentials {
  facebookLeadAdsOAuth2Api: CredentialReference;
}

interface FacebookLeadAdsTriggerV1NodeBase {
  type: 'n8n-nodes-base.facebookLeadAdsTrigger';
  version: 1;
  credentials?: FacebookLeadAdsTriggerV1Credentials;
  isTrigger: true;
}

export type FacebookLeadAdsTriggerV1ParamsNode = FacebookLeadAdsTriggerV1NodeBase & {
  config: NodeConfig<FacebookLeadAdsTriggerV1Params>;
};

export type FacebookLeadAdsTriggerV1Node = FacebookLeadAdsTriggerV1ParamsNode;
