// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/dropbox/v1/resource_folder/operation_list.schema.js
/**
 * Dropbox Node - Version 1 - Zod Schema
 * Discriminator: resource=folder, operation=list
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('folder'),
      operation: z.literal('list'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      path: stringOrExpression.optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filters: z.object({ include_deleted: booleanOrExpression.optional(), include_has_explicit_shared_members: booleanOrExpression.optional(), include_mounted_folders: booleanOrExpression.optional(), include_non_downloadable_files: booleanOrExpression.optional(), recursive: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
