// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/dropbox/v1/resource_folder/operation_delete.ts
/**
 * Dropbox Node - Version 1
 * Discriminator: resource=folder, operation=delete
 */


interface Credentials {
  dropboxApi: CredentialReference;
  dropboxOAuth2Api: CredentialReference;
}

/** Delete a file */
export type DropboxV1FolderDeleteParams = {
  resource: 'folder';
  operation: 'delete';
/**
 * Means of authenticating with the service
 * @default accessToken
 */
    authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * The path to delete. Can be a single file or a whole folder.
 */
    path?: string | Expression<string> | PlaceholderValue;
};

export type DropboxV1FolderDeleteOutput = {
  metadata?: {
    '.tag'?: string;
    id?: string;
    name?: string;
    parent_shared_folder_id?: string;
    path_display?: string;
    path_lower?: string;
    sharing_info?: {
      no_access?: boolean;
      parent_shared_folder_id?: string;
      read_only?: boolean;
      traverse_only?: boolean;
    };
  };
};

export type DropboxV1FolderDeleteNode = {
  type: 'n8n-nodes-base.dropbox';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<DropboxV1FolderDeleteParams>;
  output?: Items<DropboxV1FolderDeleteOutput>;
};
