// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/metabase/v1/resource_questions/operation_result_data.ts
/**
 * Metabase Node - Version 1
 * Discriminator: resource=questions, operation=resultData
 */


interface Credentials {
  metabaseApi: CredentialReference;
}

/** Return the result of the question to a specific file format */
export type MetabaseV1QuestionsResultDataParams = {
  resource: 'questions';
  operation: 'resultData';
/**
 * Question ID
 */
    questionId?: string | Expression<string> | PlaceholderValue;
/**
 * Format
 * @default csv
 */
    format?: 'csv' | 'json' | 'xlsx' | Expression<string>;
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type MetabaseV1QuestionsResultDataNode = {
  type: 'n8n-nodes-base.metabase';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MetabaseV1QuestionsResultDataParams>;
};
