// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/metabase/v1/resource_questions/operation_get.ts
/**
 * Metabase Node - Version 1
 * Discriminator: resource=questions, operation=get
 */


interface Credentials {
  metabaseApi: CredentialReference;
}

/** Get a specific question */
export type MetabaseV1QuestionsGetParams = {
  resource: 'questions';
  operation: 'get';
/**
 * Question ID
 */
    questionId?: string | Expression<string> | PlaceholderValue;
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type MetabaseV1QuestionsGetOutput = {
  archived?: boolean;
  archived_directly?: boolean;
  cache_ttl?: null;
  can_delete?: boolean;
  can_manage_db?: boolean;
  can_restore?: boolean;
  can_run_adhoc_query?: boolean;
  can_write?: boolean;
  collection?: {
    archive_operation_id?: null;
    archived?: boolean;
    archived_directly?: null;
    authority_level?: null;
    created_at?: string;
    is_personal?: boolean;
    is_sample?: boolean;
    location?: string;
    name?: string;
    namespace?: null;
    slug?: string;
    type?: null;
  };
  collection_preview?: boolean;
  created_at?: string;
  creator?: {
    common_name?: string;
    date_joined?: string;
    email?: string;
    id?: number;
    is_qbnewb?: boolean;
    is_superuser?: boolean;
    tenant_id?: null;
  };
  creator_id?: number;
  dashboard_count?: number;
  database_id?: number;
  dataset_query?: {
    database?: number;
    native?: {
      query?: string;
    };
    type?: string;
  };
  display?: string;
  embedding_params?: null;
  enable_embedding?: boolean;
  id?: number;
  initially_published_at?: null;
  last_used_at?: string;
  'last-edit-info'?: {
    email?: string;
    id?: number;
    timestamp?: string;
  };
  moderation_reviews?: Array<{
    created_at?: string;
    id?: number;
    moderated_item_id?: number;
    moderated_item_type?: string;
    moderator_id?: number;
    most_recent?: boolean;
    status?: string;
    text?: null;
    updated_at?: string;
    user?: {
      common_name?: string;
      date_joined?: string;
      email?: string;
      first_name?: string;
      id?: number;
      is_qbnewb?: boolean;
      is_superuser?: boolean;
      last_login?: string;
      last_name?: string;
    };
  }>;
  name?: string;
  parameter_usage_count?: number;
  query_type?: string;
  type?: string;
  updated_at?: string;
  view_count?: number;
  visualization_settings?: {
    'table.cell_column'?: string;
    'table.pivot_column'?: string;
  };
};

export type MetabaseV1QuestionsGetNode = {
  type: 'n8n-nodes-base.metabase';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MetabaseV1QuestionsGetParams>;
  output?: Items<MetabaseV1QuestionsGetOutput>;
};
