// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/vero/v1/resource_event/operation_track.schema.js
/**
 * Vero Node - Version 1 - Zod Schema
 * Discriminator: resource=event, operation=track
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('event'),
      operation: z.literal('track'),
      id: stringOrExpression.optional(),
      email: stringOrExpression.optional(),
      eventName: stringOrExpression.optional(),
      jsonParameters: booleanOrExpression.optional(),
      dataAttributesUi: resolveSchema({ parameters, schema: z.object({ dataAttributesValues: z.array(z.object({ key: stringOrExpression.optional(), value: stringOrExpression.optional() })).optional() }), required: false, displayOptions: {"show":{"jsonParameters":[false]}}, defaults: {"jsonParameters":false} }),
      extraAttributesUi: resolveSchema({ parameters, schema: z.object({ extraAttributesValues: z.array(z.object({ key: stringOrExpression.optional(), value: stringOrExpression.optional() })).optional() }), required: false, displayOptions: {"show":{"jsonParameters":[false]}}, defaults: {"jsonParameters":false} }),
      dataAttributesJson: resolveSchema({ parameters, schema: z.union([iDataObjectSchema, z.string()]), required: false, displayOptions: {"show":{"jsonParameters":[true]}}, defaults: {"jsonParameters":false} }),
      extraAttributesJson: resolveSchema({ parameters, schema: z.union([iDataObjectSchema, z.string()]), required: false, displayOptions: {"show":{"jsonParameters":[true]}}, defaults: {"jsonParameters":false} }),
    }).optional(),
  });
};
