// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleSheets/v2/resource_sheet/operation_remove.schema.js
/**
 * Google Sheets  Node - Version 2 - Zod Schema
 * Discriminator: resource=sheet, operation=remove
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('sheet').default('sheet'),
      operation: z.literal('remove'),
      authentication: z.union([z.literal('oAuth2'), z.literal('serviceAccount'), expressionSchema]).optional(),
      sheetId: stringOrExpression.optional(),
      range: stringOrExpression.optional(),
      dataStartRow: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"hide":{"rawData":[true]}} }),
      keyRow: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"hide":{"rawData":[true]}} }),
      id: stringOrExpression.optional(),
    }).optional(),
  });
};
