// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleDocs/v2/resource_document/operation_get.ts
/**
 * Google Docs Node - Version 2
 * Discriminator: resource=document, operation=get
 */


interface Credentials {
  googleApi: CredentialReference;
  googleDocsOAuth2Api: CredentialReference;
}

export type GoogleDocsV2DocumentGetParams = {
  resource: 'document';
  operation: 'get';
/**
 * Authentication
 * @default oAuth2
 */
    authentication?: 'oAuth2' | 'serviceAccount' | Expression<string>;
/**
 * The ID in the document URL (or just paste the whole URL)
 */
    documentURL?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to return a simplified version of the response instead of the raw data
 * @default true
 */
    simple?: boolean | Expression<boolean>;
};

export type GoogleDocsV2DocumentGetOutput = {
  content?: string;
  documentId?: string;
};

export type GoogleDocsV2DocumentGetNode = {
  type: 'n8n-nodes-base.googleDocs';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<GoogleDocsV2DocumentGetParams>;
  output?: Items<GoogleDocsV2DocumentGetOutput>;
};
