// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/reddit/v1/resource_subreddit/operation_get_all.ts
/**
 * Reddit Node - Version 1
 * Discriminator: resource=subreddit, operation=getAll
 */


interface Credentials {
  redditOAuth2Api: CredentialReference;
}

/** Retrieve many comments in a post */
export type RedditV1SubredditGetAllParams = {
  resource: 'subreddit';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** The keyword for the subreddit search
     */
    keyword?: string | Expression<string> | PlaceholderValue;
    /** Whether to fetch currently trending subreddits in all of Reddit
     * @default false
     */
    trending?: boolean | Expression<boolean>;
  };
};

export type RedditV1SubredditGetAllNode = {
  type: 'n8n-nodes-base.reddit';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<RedditV1SubredditGetAllParams>;
};
