// @n8n/n8n-nodes-langchain/dist/node-definitions/nodes/n8n-nodes-langchain/openAi/v11/resource_assistant/operation_delete_assistant.ts
/**
 * OpenAI Node - Version 1.1
 * Discriminator: resource=assistant, operation=deleteAssistant
 */


interface Credentials {
  openAiApi: CredentialReference;
}

/** Delete an assistant from the account */
export type LcOpenAiV11AssistantDeleteAssistantParams = {
  resource: 'assistant';
  operation: 'deleteAssistant';
/**
 * Assistant to respond to the message. You can add, modify or remove assistants in the &lt;a href="https://platform.openai.com/playground?mode=assistant" target="_blank"&gt;playground&lt;/a&gt;.
 * @default {"mode":"list","value":""}
 */
    assistantId?: { __rl: true; mode: 'list' | 'id'; value: string; cachedResultName?: string };
};

export interface LcOpenAiV11AssistantDeleteAssistantSubnodeConfig {
  tools?: ToolInstance[];
  memory?: MemoryInstance;
}

export type LcOpenAiV11AssistantDeleteAssistantNode = {
  type: '@n8n/n8n-nodes-langchain.openAi';
  version: 1.1;
  credentials?: Credentials;
  config: NodeConfig<LcOpenAiV11AssistantDeleteAssistantParams> & { subnodes?: LcOpenAiV11AssistantDeleteAssistantSubnodeConfig };
};
