// @n8n/api-types/dist/chat-hub.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChatHubUpdateToolRequest = exports.ChatHubCreateToolRequest = exports.CHAT_USER_BLOCKED_CHAT_HUB_TOOL_TYPES = exports.ALWAYS_BLOCKED_CHAT_HUB_TOOL_TYPES = exports.chatHubMessageWithButtonsSchema = exports.chatHubMessageButtonSchema = exports.ChatReconnectRequest = exports.ChatHubSemanticSearchSettings = exports.UpdateChatSettingsRequest = exports.ChatHubUpdateAgentRequest = exports.ChatHubCreateAgentRequest = exports.suggestedPromptsSchema = exports.suggestedPromptSchema = exports.ChatHubConversationsRequest = exports.ChatHubUpdateConversationRequest = exports.ChatHubEditMessageRequest = exports.ChatHubRegenerateMessageRequest = exports.ChatHubSendMessageRequest = exports.TimeZoneSchema = exports.StrictTimeZoneSchema = exports.isValidTimeZone = exports.chatAttachmentSchema = exports.emptyChatModelsResponse = exports.chatModelsRequestSchema = exports.chatHubConversationModelSchema = exports.VECTOR_STORE_PROVIDER_CREDENTIAL_TYPE_MAP = exports.PROVIDER_CREDENTIAL_TYPE_MAP = exports.chatHubProviderSchema = exports.agentIconOrEmojiSchema = exports.chatHubVectorStoreProviderSchema = exports.chatHubLLMProviderSchema = void 0;
const n8n_workflow_1 = require("n8n-workflow");
const zod_1 = require("zod");
const zod_class_1 = require("./zod-class");
exports.chatHubLLMProviderSchema = zod_1.z.enum([
    'openai',
    'anthropic',
    'google',
    'azureOpenAi',
    'azureEntraId',
    'ollama',
    'awsBedrock',
    'vercelAiGateway',
    'xAiGrok',
    'groq',
    'openRouter',
    'deepSeek',
    'cohere',
    'mistralCloud',
]);
exports.chatHubVectorStoreProviderSchema = zod_1.z.enum(['pgvector', 'qdrant', 'pinecone']);
exports.agentIconOrEmojiSchema = zod_1.z.discriminatedUnion('type', [
    zod_1.z.object({
        type: zod_1.z.literal('icon'),
        value: zod_1.z.string(),
    }),
    zod_1.z.object({
        type: zod_1.z.literal('emoji'),
        value: zod_1.z.string(),
    }),
]);
exports.chatHubProviderSchema = zod_1.z.enum([
    ...exports.chatHubLLMProviderSchema.options,
    'n8n',
    'custom-agent',
]);
exports.PROVIDER_CREDENTIAL_TYPE_MAP = {
    openai: 'openAiApi',
    anthropic: 'anthropicApi',
    google: 'googlePalmApi',
    ollama: 'ollamaApi',
    azureOpenAi: 'azureOpenAiApi',
    azureEntraId: 'azureEntraCognitiveServicesOAuth2Api',
    awsBedrock: 'aws',
    vercelAiGateway: 'vercelAiGatewayApi',
    xAiGrok: 'xAiApi',
    groq: 'groqApi',
    openRouter: 'openRouterApi',
    deepSeek: 'deepSeekApi',
    cohere: 'cohereApi',
    mistralCloud: 'mistralCloudApi',
};
exports.VECTOR_STORE_PROVIDER_CREDENTIAL_TYPE_MAP = {
    pgvector: 'chatHubVectorStorePGVectorApi',
    qdrant: 'chatHubVectorStoreQdrantApi',
    pinecone: 'chatHubVectorStorePineconeApi',
};
const openAIModelSchema = zod_1.z.object({
    provider: zod_1.z.literal('openai'),
    model: zod_1.z.string(),
});
const anthropicModelSchema = zod_1.z.object({
    provider: zod_1.z.literal('anthropic'),
    model: zod_1.z.string(),
});
const googleModelSchema = zod_1.z.object({
    provider: zod_1.z.literal('google'),
    model: zod_1.z.string(),
});
const azureOpenAIModelSchema = zod_1.z.object({
    provider: zod_1.z.literal('azureOpenAi'),
    model: zod_1.z.string(),
});
const azureEntraIdModelSchema = zod_1.z.object({
    provider: zod_1.z.literal('azureEntraId'),
    model: zod_1.z.string(),
});
const ollamaModelSchema = zod_1.z.object({
    provider: zod_1.z.literal('ollama'),
    model: zod_1.z.string(),
});
const awsBedrockModelSchema = zod_1.z.object({
    provider: zod_1.z.literal('awsBedrock'),
    model: zod_1.z.string(),
});
const vercelAiGatewaySchema = zod_1.z.object({
    provider: zod_1.z.literal('vercelAiGateway'),
    model: zod_1.z.string(),
});
const xAiGrokModelSchema = zod_1.z.object({
    provider: zod_1.z.literal('xAiGrok'),
    model: zod_1.z.string(),
});
const groqModelSchema = zod_1.z.object({
    provider: zod_1.z.literal('groq'),
    model: zod_1.z.string(),
});
const openRouterModelSchema = zod_1.z.object({
    provider: zod_1.z.literal('openRouter'),
    model: zod_1.z.string(),
});
const deepSeekModelSchema = zod_1.z.object({
    provider: zod_1.z.literal('deepSeek'),
    model: zod_1.z.string(),
});
const cohereModelSchema = zod_1.z.object({
    provider: zod_1.z.literal('cohere'),
    model: zod_1.z.string(),
});
const mistralCloudModelSchema = zod_1.z.object({
    provider: zod_1.z.literal('mistralCloud'),
    model: zod_1.z.string(),
});
const n8nModelSchema = zod_1.z.object({
    provider: zod_1.z.literal('n8n'),
    workflowId: zod_1.z.string(),
});
const chatAgentSchema = zod_1.z.object({
    provider: zod_1.z.literal('custom-agent'),
    agentId: zod_1.z.string(),
});
exports.chatHubConversationModelSchema = zod_1.z.discriminatedUnion('provider', [
    openAIModelSchema,
    anthropicModelSchema,
    googleModelSchema,
    azureOpenAIModelSchema,
    azureEntraIdModelSchema,
    ollamaModelSchema,
    awsBedrockModelSchema,
    vercelAiGatewaySchema,
    xAiGrokModelSchema,
    groqModelSchema,
    openRouterModelSchema,
    deepSeekModelSchema,
    cohereModelSchema,
    mistralCloudModelSchema,
    n8nModelSchema,
    chatAgentSchema,
]);
exports.chatModelsRequestSchema = zod_1.z.object({
    credentials: zod_1.z.record(exports.chatHubProviderSchema, zod_1.z.string().nullable()),
});
exports.emptyChatModelsResponse = {
    openai: { models: [] },
    anthropic: { models: [] },
    google: { models: [] },
    azureOpenAi: { models: [] },
    azureEntraId: { models: [] },
    ollama: { models: [] },
    awsBedrock: { models: [] },
    vercelAiGateway: { models: [] },
    xAiGrok: { models: [] },
    groq: { models: [] },
    openRouter: { models: [] },
    deepSeek: { models: [] },
    cohere: { models: [] },
    mistralCloud: { models: [] },
    n8n: { models: [] },
    'custom-agent': { models: [] },
};
exports.chatAttachmentSchema = zod_1.z.object({
    data: zod_1.z.string(),
    mimeType: zod_1.z.string(),
    fileName: zod_1.z.string(),
});
const isValidTimeZone = (tz) => {
    try {
        new Intl.DateTimeFormat('en-US', { timeZone: tz });
        return true;
    }
    catch {
        return false;
    }
};
exports.isValidTimeZone = isValidTimeZone;
exports.StrictTimeZoneSchema = zod_1.z
    .string()
    .min(1)
    .max(50)
    .regex(/^[A-Za-z0-9_/+-]+$/)
    .refine(exports.isValidTimeZone, {
    message: 'Unknown or invalid time zone',
});
exports.TimeZoneSchema = exports.StrictTimeZoneSchema.optional().catch(undefined);
class ChatHubSendMessageRequest extends zod_class_1.Z.class({
    messageId: zod_1.z.string().uuid(),
    sessionId: zod_1.z.string().uuid(),
    message: zod_1.z.string(),
    model: exports.chatHubConversationModelSchema,
    previousMessageId: zod_1.z.string().uuid().nullable(),
    credentials: zod_1.z.record(zod_1.z.object({
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })),
    attachments: zod_1.z.array(exports.chatAttachmentSchema),
    agentName: zod_1.z.string().optional(),
    timeZone: exports.TimeZoneSchema,
}) {
}
exports.ChatHubSendMessageRequest = ChatHubSendMessageRequest;
class ChatHubRegenerateMessageRequest extends zod_class_1.Z.class({
    model: exports.chatHubConversationModelSchema,
    credentials: zod_1.z.record(zod_1.z.object({
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })),
    timeZone: exports.TimeZoneSchema,
}) {
}
exports.ChatHubRegenerateMessageRequest = ChatHubRegenerateMessageRequest;
class ChatHubEditMessageRequest extends zod_class_1.Z.class({
    message: zod_1.z.string(),
    messageId: zod_1.z.string().uuid(),
    model: exports.chatHubConversationModelSchema,
    credentials: zod_1.z.record(zod_1.z.object({
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })),
    newAttachments: zod_1.z.array(exports.chatAttachmentSchema),
    keepAttachmentIndices: zod_1.z.array(zod_1.z.number()),
    timeZone: exports.TimeZoneSchema,
}) {
}
exports.ChatHubEditMessageRequest = ChatHubEditMessageRequest;
class ChatHubUpdateConversationRequest extends zod_class_1.Z.class({
    title: zod_1.z.string().optional(),
    credentialId: zod_1.z.string().max(36).optional(),
    agent: zod_1.z
        .object({
        model: exports.chatHubConversationModelSchema,
        name: zod_1.z.string(),
    })
        .optional(),
    toolIds: zod_1.z.array(zod_1.z.string().uuid()).optional(),
}) {
}
exports.ChatHubUpdateConversationRequest = ChatHubUpdateConversationRequest;
class ChatHubConversationsRequest extends zod_class_1.Z.class({
    limit: zod_1.z.coerce.number().int().min(1).max(100),
    cursor: zod_1.z.string().uuid().optional(),
}) {
}
exports.ChatHubConversationsRequest = ChatHubConversationsRequest;
exports.suggestedPromptSchema = zod_1.z.object({
    text: zod_1.z.string().min(1).max(256),
    icon: exports.agentIconOrEmojiSchema.optional(),
});
exports.suggestedPromptsSchema = zod_1.z.array(exports.suggestedPromptSchema).max(6);
class ChatHubCreateAgentRequest extends zod_class_1.Z.class({
    name: zod_1.z.string().min(1).max(128),
    description: zod_1.z.string().max(512).optional(),
    icon: exports.agentIconOrEmojiSchema,
    systemPrompt: zod_1.z.string().default(''),
    suggestedPrompts: exports.suggestedPromptsSchema.optional(),
    credentialId: zod_1.z.string(),
    provider: exports.chatHubLLMProviderSchema,
    model: zod_1.z.string().max(64),
    toolIds: zod_1.z.array(zod_1.z.string().uuid()),
}) {
}
exports.ChatHubCreateAgentRequest = ChatHubCreateAgentRequest;
class ChatHubUpdateAgentRequest extends zod_class_1.Z.class({
    name: zod_1.z.string().min(1).max(128).optional(),
    description: zod_1.z.string().max(512).optional(),
    icon: exports.agentIconOrEmojiSchema.optional(),
    systemPrompt: zod_1.z.string().optional(),
    suggestedPrompts: exports.suggestedPromptsSchema.optional(),
    credentialId: zod_1.z.string().optional(),
    provider: exports.chatHubLLMProviderSchema.optional(),
    model: zod_1.z.string().max(64).optional(),
    toolIds: zod_1.z.array(zod_1.z.string().uuid()).optional(),
}) {
}
exports.ChatHubUpdateAgentRequest = ChatHubUpdateAgentRequest;
const chatProviderSettingsSchema = zod_1.z.object({
    provider: exports.chatHubLLMProviderSchema,
    enabled: zod_1.z.boolean().optional(),
    credentialId: zod_1.z.string().nullable(),
    allowedModels: zod_1.z.array(zod_1.z.object({
        displayName: zod_1.z.string(),
        model: zod_1.z.string(),
        isManual: zod_1.z.boolean().optional(),
    })),
    createdAt: zod_1.z.string(),
    updatedAt: zod_1.z.string().nullable(),
});
class UpdateChatSettingsRequest extends zod_class_1.Z.class({
    payload: chatProviderSettingsSchema,
}) {
}
exports.UpdateChatSettingsRequest = UpdateChatSettingsRequest;
class ChatHubSemanticSearchSettings extends zod_class_1.Z.class({
    vectorStore: zod_1.z.object({
        provider: exports.chatHubVectorStoreProviderSchema,
        credentialId: zod_1.z.string().nullable(),
    }),
    embeddingModel: zod_1.z.object({
        provider: exports.chatHubLLMProviderSchema,
        credentialId: zod_1.z.string().nullable(),
    }),
}) {
}
exports.ChatHubSemanticSearchSettings = ChatHubSemanticSearchSettings;
class ChatReconnectRequest extends zod_class_1.Z.class({
    lastSequence: zod_1.z.coerce.number().int().min(0).optional(),
}) {
}
exports.ChatReconnectRequest = ChatReconnectRequest;
exports.chatHubMessageButtonSchema = zod_1.z.object({
    text: zod_1.z.string(),
    link: zod_1.z.string(),
    type: zod_1.z.enum(['primary', 'secondary']),
});
exports.chatHubMessageWithButtonsSchema = zod_1.z.object({
    type: zod_1.z.literal('with-buttons'),
    text: zod_1.z.string(),
    blockUserInput: zod_1.z.boolean(),
    buttons: zod_1.z.array(exports.chatHubMessageButtonSchema).min(1),
});
exports.ALWAYS_BLOCKED_CHAT_HUB_TOOL_TYPES = [n8n_workflow_1.CHAT_TOOL_NODE_TYPE];
exports.CHAT_USER_BLOCKED_CHAT_HUB_TOOL_TYPES = [
    n8n_workflow_1.WORKFLOW_TOOL_LANGCHAIN_NODE_TYPE,
    n8n_workflow_1.DATA_TABLE_TOOL_NODE_TYPE,
];
class ChatHubCreateToolRequest extends zod_class_1.Z.class({
    definition: n8n_workflow_1.INodeSchema,
}) {
}
exports.ChatHubCreateToolRequest = ChatHubCreateToolRequest;
class ChatHubUpdateToolRequest extends zod_class_1.Z.class({
    definition: n8n_workflow_1.INodeSchema.optional(),
    enabled: zod_1.z.boolean().optional(),
}) {
}
exports.ChatHubUpdateToolRequest = ChatHubUpdateToolRequest;
