// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/storyblok/v1/resource_story/operation_get_all.ts
/**
 * Storyblok Node - Version 1
 * Discriminator: resource=story, operation=getAll
 */


interface Credentials {
  storyblokContentApi: CredentialReference;
  storyblokManagementApi: CredentialReference;
}

/** Get many stories */
export type StoryblokV1StoryGetAllParams = {
  resource: 'story';
  operation: 'getAll';
/**
 * Pick where your data comes from, Content or Management API
 * @default contentApi
 */
    source?: 'contentApi' | 'managementApi' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @displayOptions.show { source: ["contentApi"] }
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { source: ["contentApi"], returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @displayOptions.show { source: ["contentApi"] }
 * @default {}
 */
    filters?: {
    /** Filter by slug
     */
    starts_with?: string | Expression<string> | PlaceholderValue;
  };
/**
 * The name of the space. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 * @displayOptions.show { source: ["managementApi"] }
 */
    space?: string | Expression<string>;
};

export type StoryblokV1StoryGetAllOutput = {
  content_type?: string;
  created_at?: string;
  default_root?: null;
  deleted_at?: null;
  disable_fe_editor?: boolean;
  disble_fe_editor?: boolean;
  expire_at?: null;
  full_slug?: string;
  group_id?: string;
  id?: number;
  is_folder?: boolean;
  is_startpage?: boolean;
  last_author?: {
    friendly_name?: string;
    id?: number;
    userid?: string;
  };
  last_author_id?: number;
  name?: string;
  path?: null;
  pinned?: boolean;
  position?: number;
  publish_at?: null;
  published?: boolean;
  scheduled_dates?: null;
  slug?: string;
  sort_by_date?: null;
  tag_list?: Array<string>;
  unpublished_changes?: boolean;
  updated_at?: string;
  uuid?: string;
};

export type StoryblokV1StoryGetAllNode = {
  type: 'n8n-nodes-base.storyblok';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<StoryblokV1StoryGetAllParams>;
  output?: Items<StoryblokV1StoryGetAllOutput>;
};
