// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/wordpress/v1/resource_post/operation_get_all.schema.js
/**
 * Wordpress Node - Version 1 - Zod Schema
 * Discriminator: resource=post, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('post').default('post'),
      operation: z.literal('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      options: z.object({ after: stringOrExpression.optional(), author: z.array(z.string()).optional(), before: stringOrExpression.optional(), categories: z.array(z.string()).optional(), context: z.union([z.literal('view'), z.literal('embed'), z.literal('edit'), expressionSchema]).optional(), excludedCategories: z.array(z.string()).optional(), excludedTags: z.array(z.string()).optional(), order: z.union([z.literal('asc'), z.literal('desc'), expressionSchema]).optional(), orderBy: z.union([z.literal('author'), z.literal('date'), z.literal('id'), z.literal('include'), z.literal('include_slugs'), z.literal('modified'), z.literal('parent'), z.literal('relevance'), z.literal('slug'), z.literal('title'), expressionSchema]).optional(), search: stringOrExpression.optional(), status: z.union([z.literal('draft'), z.literal('future'), z.literal('pending'), z.literal('private'), z.literal('publish'), expressionSchema]).optional(), sticky: booleanOrExpression.optional(), tags: z.array(z.string()).optional() }).optional(),
    }).optional(),
  });
};
