// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/wordpress/v1/resource_post/operation_update.schema.js
/**
 * Wordpress Node - Version 1 - Zod Schema
 * Discriminator: resource=post, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('post').default('post'),
      operation: z.literal('update'),
      postId: stringOrExpression.optional(),
      updateFields: z.object({ authorId: stringOrExpression.optional(), title: stringOrExpression.optional(), content: stringOrExpression.optional(), slug: stringOrExpression.optional(), password: stringOrExpression.optional(), status: z.union([z.literal('draft'), z.literal('future'), z.literal('pending'), z.literal('private'), z.literal('publish'), expressionSchema]).optional(), date: stringOrExpression.optional(), commentStatus: z.union([z.literal('open'), z.literal('closed'), expressionSchema]).optional(), pingStatus: z.union([z.literal('open'), z.literal('closed'), expressionSchema]).optional(), format: z.union([z.literal('aside'), z.literal('audio'), z.literal('chat'), z.literal('gallery'), z.literal('image'), z.literal('link'), z.literal('quote'), z.literal('standard'), z.literal('status'), z.literal('video'), expressionSchema]).optional(), sticky: booleanOrExpression.optional(), categories: z.array(z.string()).optional(), tags: z.array(z.string()).optional(), postTemplate: z.unknown().optional() }).optional(),
    }).optional(),
  });
};
