// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/wordpress/v1/resource_post/operation_get.schema.js
/**
 * Wordpress Node - Version 1 - Zod Schema
 * Discriminator: resource=post, operation=get
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('post').default('post'),
      operation: z.literal('get'),
      postId: stringOrExpression.optional(),
      options: z.object({ password: stringOrExpression.optional(), context: z.union([z.literal('view'), z.literal('embed'), z.literal('edit'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
