// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/customerIo/v1/resource_event/operation_track_anonymous.ts
/**
 * Customer.io Node - Version 1
 * Discriminator: resource=event, operation=trackAnonymous
 */


interface Credentials {
  customerIoApi: CredentialReference;
}

/** Track an anonymous event */
export type CustomerIoV1EventTrackAnonymousParams = {
  resource: 'event';
  operation: 'trackAnonymous';
/**
 * The unique identifier for the customer
 */
    eventName?: string | Expression<string> | PlaceholderValue;
/**
 * JSON Parameters
 * @default false
 */
    jsonParameters?: boolean | Expression<boolean>;
/**
 * Object of values to set as described &lt;a href="https://customer.io/docs/api-triggered-data-format#basic-data-formatting"&gt;here&lt;/a&gt;
 * @displayOptions.show { jsonParameters: [true] }
 */
    additionalFieldsJson?: IDataObject | string | Expression<string>;
/**
 * Additional Fields
 * @displayOptions.show { jsonParameters: [false] }
 * @default {}
 */
    additionalFields?: {
    /** Custom Properties
     * @default {}
     */
    customAttributes?: {
        /** Attribute
     */
    customAttribute?: Array<{
      /** Attribute name
       */
      key?: string | Expression<string> | PlaceholderValue;
      /** Attribute value
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
  };
};

export type CustomerIoV1EventTrackAnonymousNode = {
  type: 'n8n-nodes-base.customerIo';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<CustomerIoV1EventTrackAnonymousParams>;
};
