// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/nocoDb/v2/resource_row/operation_create.schema.js
/**
 * NocoDB Node - Version 2 - Zod Schema
 * Discriminator: resource=row, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('row').default('row'),
      operation: z.literal('create'),
      authentication: z.union([z.literal('nocoDbApiToken'), z.literal('nocoDb'), expressionSchema]).optional(),
      version: z.union([z.literal(1), z.literal(2), z.literal(3), expressionSchema]).optional(),
      workspaceId: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"version":[3]}}, defaults: {"version":2} }),
      projectId: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"version":[3,1,2]}}, defaults: {"version":2} }),
      table: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"version":[2,3,1]}}, defaults: {"version":2} }),
      dataToSend: z.union([z.literal('autoMapInputData'), z.literal('defineBelow'), expressionSchema]).optional(),
      inputsToIgnore: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"dataToSend":["autoMapInputData"]}}, defaults: {"dataToSend":"defineBelow"} }),
      fieldsUi: resolveSchema({ parameters, schema: z.object({ fieldValues: z.array(z.object({ fieldName: stringOrExpression.optional(), binaryData: booleanOrExpression.optional(), fieldValue: stringOrExpression.optional(), binaryProperty: stringOrExpression.optional() })).optional() }), required: false, displayOptions: {"show":{"dataToSend":["defineBelow"]}}, defaults: {"dataToSend":"defineBelow"} }),
    }).optional(),
  });
};
