// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/haloPSA/v1/resource_client/operation_create.schema.js
/**
 * HaloPSA Node - Version 1 - Zod Schema
 * Discriminator: resource=client, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('client').default('client'),
      operation: z.literal('create').default('create'),
      clientName: stringOrExpression.optional(),
      additionalFields: z.object({ inactive: z.union([z.literal(false), z.literal(true), expressionSchema]).optional(), notes: stringOrExpression.optional(), is_vip: booleanOrExpression.optional(), website: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
