// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleBooks/v1/resource_bookshelf/operation_get.schema.js
/**
 * Google Books Node - Version 1 - Zod Schema
 * Discriminator: resource=bookshelf, operation=get
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('bookshelf').default('bookshelf'),
      operation: z.literal('get').default('get'),
      authentication: z.union([z.literal('serviceAccount'), z.literal('oAuth2'), expressionSchema]).optional(),
      myLibrary: booleanOrExpression.optional(),
      userId: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"hide":{"myLibrary":[true]}}, defaults: {"myLibrary":false} }),
      shelfId: stringOrExpression.optional(),
    }).optional(),
  });
};
