// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/taiga/v1/resource_issue/operation_get_all.schema.js
/**
 * Taiga Node - Version 1 - Zod Schema
 * Discriminator: resource=issue, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('issue').default('issue'),
      operation: z.literal('getAll'),
      projectId: stringOrExpression.optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filters: z.object({ assigned_to: stringOrExpression.optional(), orderBy: z.union([z.literal('assigned_to'), z.literal('created_date'), z.literal('modified_date'), z.literal('owner'), z.literal('priority'), z.literal('severity'), z.literal('status'), z.literal('subject'), z.literal('type'), expressionSchema]).optional(), owner: stringOrExpression.optional(), priority: stringOrExpression.optional(), role: stringOrExpression.optional(), severity: stringOrExpression.optional(), status: stringOrExpression.optional(), tags: z.array(z.string()).optional(), type: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
