// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/oneSimpleApi/v1/resource_utility/operation_qr_code.schema.js
/**
 * One Simple API Node - Version 1 - Zod Schema
 * Discriminator: resource=utility, operation=qrCode
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('utility'),
      operation: z.literal('qrCode'),
      message: stringOrExpression.optional(),
      download: booleanOrExpression.optional(),
      output: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"download":[true]}}, defaults: {"download":false} }),
      options: z.object({ size: z.union([z.literal('Small'), z.literal('Medium'), z.literal('Large'), expressionSchema]).optional(), format: z.union([z.literal('PNG'), z.literal('SVG'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
