// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/oneSimpleApi/v1/resource_utility/operation_qr_code.ts
/**
 * One Simple API Node - Version 1
 * Discriminator: resource=utility, operation=qrCode
 */


interface Credentials {
  oneSimpleApi: CredentialReference;
}

/** Generate a QR Code */
export type OneSimpleApiV1UtilityQrCodeParams = {
  resource: 'utility';
  operation: 'qrCode';
/**
 * The text that should be turned into a QR code - like a website URL
 */
    message?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to download the QR code or return a link to it
 * @default false
 */
    download?: boolean | Expression<boolean>;
/**
 * The name of the output field to put the binary file data in
 * @displayOptions.show { download: [true] }
 * @default data
 */
    output?: string | Expression<string> | PlaceholderValue;
/**
 * Options
 * @default {}
 */
    options?: {
    /** The QR Code size
     * @default Small
     */
    size?: 'Small' | 'Medium' | 'Large' | Expression<string>;
    /** The QR Code format
     * @default PNG
     */
    format?: 'PNG' | 'SVG' | Expression<string>;
  };
};

export type OneSimpleApiV1UtilityQrCodeNode = {
  type: 'n8n-nodes-base.oneSimpleApi';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<OneSimpleApiV1UtilityQrCodeParams>;
};
