// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/customerIo/v1/resource_customer/operation_upsert.ts
/**
 * Customer.io Node - Version 1
 * Discriminator: resource=customer, operation=upsert
 */


interface Credentials {
  customerIoApi: CredentialReference;
}

/** Create a new customer, or update the current one if it already exists (upsert) */
export type CustomerIoV1CustomerUpsertParams = {
  resource: 'customer';
  operation: 'upsert';
/**
 * The unique identifier for the customer
 */
    id?: string | Expression<string> | PlaceholderValue;
/**
 * JSON Parameters
 * @default false
 */
    jsonParameters?: boolean | Expression<boolean>;
/**
 * Object of values to set as described &lt;a href="https://github.com/agilecrm/rest-api#1-companys---companies-api"&gt;here&lt;/a&gt;
 * @displayOptions.show { jsonParameters: [true] }
 */
    additionalFieldsJson?: IDataObject | string | Expression<string>;
/**
 * Additional Fields
 * @displayOptions.show { jsonParameters: [false] }
 * @default {}
 */
    additionalFields?: {
    /** Custom Properties
     * @default {}
     */
    customProperties?: {
        /** Property
     */
    customProperty?: Array<{
      /** Property name
       */
      key?: string | Expression<string> | PlaceholderValue;
      /** Property value
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
    /** The email address of the user
     */
    email?: string | Expression<string> | PlaceholderValue;
    /** The UNIX timestamp from when the user was created
     */
    createdAt?: string | Expression<string>;
  };
};

export type CustomerIoV1CustomerUpsertOutput = {
  email?: string;
};

export type CustomerIoV1CustomerUpsertNode = {
  type: 'n8n-nodes-base.customerIo';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<CustomerIoV1CustomerUpsertParams>;
  output?: Items<CustomerIoV1CustomerUpsertOutput>;
};
