// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftOutlook/v1/resource_draft/operation_get.ts
/**
 * Microsoft Outlook Node - Version 1
 * Discriminator: resource=draft, operation=get
 */


interface Credentials {
  microsoftOutlookOAuth2Api: CredentialReference;
}

/** Get a single draft */
export type MicrosoftOutlookV1DraftGetParams = {
  resource: 'draft';
  operation: 'get';
/**
 * Message ID
 */
    messageId?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Prefix for name of the binary property to which to write the attachments. An index starting with 0 will be added. So if name is "attachment_" the first attachment is saved to "attachment_0"
     * @default attachment_
     */
    dataPropertyAttachmentsPrefixName?: string | Expression<string> | PlaceholderValue;
    /** Fields the response will contain. Multiple can be added separated by comma.
     */
    fields?: string | Expression<string> | PlaceholderValue;
    /** Microsoft Graph API OData $filter query. Information about the syntax can be found &lt;a href="https://docs.microsoft.com/en-us/graph/query-parameters#filter-parameter"&gt;here&lt;/a&gt;.
     */
    filter?: string | Expression<string> | PlaceholderValue;
  };
};

export type MicrosoftOutlookV1DraftGetNode = {
  type: 'n8n-nodes-base.microsoftOutlook';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MicrosoftOutlookV1DraftGetParams>;
};
