// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftOutlook/v1/resource_draft/operation_create.ts
/**
 * Microsoft Outlook Node - Version 1
 * Discriminator: resource=draft, operation=create
 */


interface Credentials {
  microsoftOutlookOAuth2Api: CredentialReference;
}

/** Create a new email draft */
export type MicrosoftOutlookV1DraftCreateParams = {
  resource: 'draft';
  operation: 'create';
/**
 * The subject of the message
 */
    subject?: string | Expression<string> | PlaceholderValue;
/**
 * Message body content
 */
    bodyContent?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Attachments
     * @default {}
     */
    attachments?: {
        /** Attachment
     */
    attachments?: Array<{
      /** Name of the binary property containing the data to be added to the email as an attachment
       */
      binaryPropertyName?: string | Expression<string> | PlaceholderValue;
    }>;
  };
    /** Email addresses of BCC recipients
     */
    bccRecipients?: string | Expression<string> | PlaceholderValue;
    /** Message body content type
     * @default html
     */
    bodyContentType?: 'html' | 'Text' | Expression<string>;
    /** Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     * @default []
     */
    categories?: string[];
    /** Email addresses of CC recipients
     */
    ccRecipients?: string | Expression<string> | PlaceholderValue;
    /** Custom Headers
     * @default {}
     */
    internetMessageHeaders?: {
        /** Header
     */
    headers?: Array<{
      /** Name of the header
       */
      name?: string | Expression<string> | PlaceholderValue;
      /** Value to set for the header
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
    /** The owner of the mailbox which the message is sent. Must correspond to the actual mailbox used.
     */
    from?: string | Expression<string> | PlaceholderValue;
    /** The importance of the message
     * @default Low
     */
    importance?: 'Low' | 'Normal' | 'High' | Expression<string>;
    /** Whether a read receipt is requested for the message
     * @default false
     */
    isReadReceiptRequested?: boolean | Expression<boolean>;
    /** Email addresses of recipients. Multiple can be added separated by comma.
     */
    toRecipients?: string | Expression<string> | PlaceholderValue;
    /** Email addresses to use when replying
     */
    replyTo?: string | Expression<string> | PlaceholderValue;
  };
};

export type MicrosoftOutlookV1DraftCreateNode = {
  type: 'n8n-nodes-base.microsoftOutlook';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MicrosoftOutlookV1DraftCreateParams>;
};
