// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftOutlook/v1/resource_draft/operation_create.schema.js
/**
 * Microsoft Outlook Node - Version 1 - Zod Schema
 * Discriminator: resource=draft, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('draft'),
      operation: z.literal('create').default('create'),
      subject: stringOrExpression.optional(),
      bodyContent: stringOrExpression.optional(),
      additionalFields: z.object({ attachments: z.unknown().optional(), bccRecipients: stringOrExpression.optional(), bodyContentType: z.union([z.literal('html'), z.literal('Text'), expressionSchema]).optional(), categories: z.array(z.string()).optional(), ccRecipients: stringOrExpression.optional(), internetMessageHeaders: z.unknown().optional(), from: stringOrExpression.optional(), importance: z.union([z.literal('Low'), z.literal('Normal'), z.literal('High'), expressionSchema]).optional(), isReadReceiptRequested: booleanOrExpression.optional(), toRecipients: stringOrExpression.optional(), replyTo: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
