// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/strava/v11/resource_activity/operation_create.ts
/**
 * Strava Node - Version 1.1
 * Discriminator: resource=activity, operation=create
 */


interface Credentials {
  stravaOAuth2Api: CredentialReference;
}

/** Create a new activity */
export type StravaV11ActivityCreateParams = {
  resource: 'activity';
  operation: 'create';
/**
 * The name of the activity
 */
    name?: string | Expression<string> | PlaceholderValue;
/**
 * Type of sport
 * @default Run
 */
    sport_type?: 'AlpineSki' | 'BackcountrySki' | 'Badminton' | 'Canoeing' | 'Crossfit' | 'EBikeRide' | 'Elliptical' | 'EMountainBikeRide' | 'Golf' | 'GravelRide' | 'Handcycle' | 'HighIntensityIntervalTraining' | 'Hike' | 'IceSkate' | 'InlineSkate' | 'Kayaking' | 'Kitesurf' | 'MountainBikeRide' | 'NordicSki' | 'Pickleball' | 'Pilates' | 'Racquetball' | 'Ride' | 'RockClimbing' | 'RollerSki' | 'Rowing' | 'Run' | 'Sail' | 'Skateboard' | 'Snowboard' | 'Snowshoe' | 'Soccer' | 'Squash' | 'StairStepper' | 'StandUpPaddling' | 'Surfing' | 'Swim' | 'TableTennis' | 'Tennis' | 'TrailRun' | 'Velomobile' | 'VirtualRide' | 'VirtualRow' | 'VirtualRun' | 'Walk' | 'WeightTraining' | 'Wheelchair' | 'Windsurf' | 'Workout' | 'Yoga' | Expression<string>;
/**
 * ISO 8601 formatted date time
 */
    startDate?: string | Expression<string>;
/**
 * In seconds
 * @default 0
 */
    elapsedTime?: number | Expression<number>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Whether to mark as commute
     * @default false
     */
    commute?: boolean | Expression<boolean>;
    /** Description of the activity
     */
    description?: string | Expression<string> | PlaceholderValue;
    /** In meters
     * @default 0
     */
    distance?: number | Expression<number>;
    /** Whether to mark as a trainer activity
     * @default false
     */
    trainer?: boolean | Expression<boolean>;
  };
};

export type StravaV11ActivityCreateNode = {
  type: 'n8n-nodes-base.strava';
  version: 1.1;
  credentials?: Credentials;
  config: NodeConfig<StravaV11ActivityCreateParams>;
};
