// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/strava/v11/resource_activity/operation_get_streams.schema.js
/**
 * Strava Node - Version 1.1 - Zod Schema
 * Discriminator: resource=activity, operation=getStreams
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('activity').default('activity'),
      operation: z.literal('getStreams'),
      activityId: stringOrExpression.optional(),
      keys: z.array(z.union([z.literal('altitude'), z.literal('cadence'), z.literal('distance'), z.literal('grade_smooth'), z.literal('heartrate'), z.literal('latlng'), z.literal('moving'), z.literal('temp'), z.literal('time'), z.literal('velocity_smooth'), z.literal('watts')])).optional(),
    }).optional(),
  });
};
