// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/salesmate/v1/resource_company/operation_update.schema.js
/**
 * Salesmate Node - Version 1 - Zod Schema
 * Discriminator: resource=company, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('company'),
      operation: z.literal('update'),
      id: stringOrExpression.optional(),
      rawData: booleanOrExpression.optional(),
      updateFields: z.object({ name: stringOrExpression.optional(), owner: stringOrExpression.optional(), website: stringOrExpression.optional(), phone: stringOrExpression.optional(), otherPhone: stringOrExpression.optional(), facebookHandle: stringOrExpression.optional(), googlePlusHandle: stringOrExpression.optional(), linkedInHandle: stringOrExpression.optional(), skypeId: stringOrExpression.optional(), twitterHandle: stringOrExpression.optional(), currency: stringOrExpression.optional(), billingAddressLine1: stringOrExpression.optional(), billingAddressLine2: stringOrExpression.optional(), billingCity: stringOrExpression.optional(), billingZipCode: stringOrExpression.optional(), billingState: stringOrExpression.optional(), billingState: stringOrExpression.optional(), description: stringOrExpression.optional(), tags: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
