// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedrive/v1/resource_file/operation_create.ts
/**
 * Pipedrive Node - Version 1
 * Discriminator: resource=file, operation=create
 */


interface Credentials {
  pipedriveApi: CredentialReference;
  pipedriveOAuth2Api: CredentialReference;
}

/** Create an activity */
export type PipedriveV1FileCreateParams = {
  resource: 'file';
  operation: 'create';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * Input Binary Field
 * @hint The name of the input binary field containing the file to be written
 * @default data
 */
    binaryPropertyName?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** ID of the activite this file will be associated with
     * @default 0
     */
    activity_id?: number | Expression<number>;
    /** ID of the deal this file will be associated with
     * @default 0
     */
    deal_id?: number | Expression<number>;
    /** ID of the organization this deal will be associated with. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    org_id?: string | Expression<string>;
    /** ID of the person this file will be associated with
     * @default 0
     */
    person_id?: number | Expression<number>;
    /** ID of the person this file will be associated with
     * @default 0
     */
    product_id?: number | Expression<number>;
  };
};

export type PipedriveV1FileCreateNode = {
  type: 'n8n-nodes-base.pipedrive';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PipedriveV1FileCreateParams>;
};
