// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/gitlab/v1/resource_issue/operation_create.schema.js
/**
 * GitLab Node - Version 1 - Zod Schema
 * Discriminator: resource=issue, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('issue').default('issue'),
      operation: z.literal('create').default('create'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      owner: stringOrExpression.optional(),
      repository: stringOrExpression.optional(),
      title: stringOrExpression.optional(),
      body: stringOrExpression.optional(),
      due_date: stringOrExpression.optional(),
      labels: z.object({ label: stringOrExpression.optional() }).optional(),
      assignee_ids: z.object({ assignee: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
