// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/gitlab/v1/resource_issue/operation_lock.schema.js
/**
 * GitLab Node - Version 1 - Zod Schema
 * Discriminator: resource=issue, operation=lock
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('issue').default('issue'),
      operation: z.literal('lock'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      owner: stringOrExpression.optional(),
      repository: stringOrExpression.optional(),
      issueNumber: numberOrExpression.optional(),
      lockReason: z.union([z.literal('off-topic'), z.literal('too heated'), z.literal('resolved'), z.literal('spam'), expressionSchema]).optional(),
    }).optional(),
  });
};
