// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/gitlab/v1/resource_issue/operation_edit.schema.js
/**
 * GitLab Node - Version 1 - Zod Schema
 * Discriminator: resource=issue, operation=edit
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('issue').default('issue'),
      operation: z.literal('edit'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      owner: stringOrExpression.optional(),
      repository: stringOrExpression.optional(),
      issueNumber: numberOrExpression.optional(),
      editFields: z.object({ title: stringOrExpression.optional(), description: stringOrExpression.optional(), state: z.union([z.literal('closed'), z.literal('open'), expressionSchema]).optional(), labels: z.unknown().optional(), assignee_ids: z.unknown().optional(), due_date: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
