// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/gitlab/v1/resource_issue/operation_lock.ts
/**
 * GitLab Node - Version 1
 * Discriminator: resource=issue, operation=lock
 */


interface Credentials {
  gitlabApi: CredentialReference;
  gitlabOAuth2Api: CredentialReference;
}

/** Lock an issue */
export type GitlabV1IssueLockParams = {
  resource: 'issue';
  operation: 'lock';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * User, group or namespace of the project
 */
    owner?: string | Expression<string> | PlaceholderValue;
/**
 * The name of the project
 * @displayOptions.hide { resource: ["user"], operation: ["getRepositories"] }
 */
    repository?: string | Expression<string> | PlaceholderValue;
/**
 * The number of the issue to lock
 * @default 0
 */
    issueNumber?: number | Expression<number>;
/**
 * The reason to lock the issue
 * @default resolved
 */
    lockReason?: 'off-topic' | 'too heated' | 'resolved' | 'spam' | Expression<string>;
};

export type GitlabV1IssueLockNode = {
  type: 'n8n-nodes-base.gitlab';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GitlabV1IssueLockParams>;
};
