// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/salesforce/v1/resource_case/operation_get.ts
/**
 * Salesforce Node - Version 1
 * Discriminator: resource=case, operation=get
 */


interface Credentials {
  salesforceOAuth2Api: CredentialReference;
  salesforceJwtApi: CredentialReference;
}

/** Represents a case, which is a customer issue or problem */
export type SalesforceV1CaseGetParams = {
  resource: 'case';
  operation: 'get';
/**
 * OAuth Authorization Flow
 * @default oAuth2
 */
    authentication?: 'oAuth2' | 'jwt' | Expression<string>;
/**
 * ID of case that needs to be fetched
 */
    caseId?: string | Expression<string> | PlaceholderValue;
};

export type SalesforceV1CaseGetOutput = {
  AssetId?: null;
  attributes?: {
    type?: string;
    url?: string;
  };
  BusinessHoursId?: string;
  CaseNumber?: string;
  Comments?: null;
  CreatedById?: string;
  CreatedDate?: string;
  Id?: string;
  IsClosed?: boolean;
  IsDeleted?: boolean;
  IsEscalated?: boolean;
  Language?: null;
  LastModifiedById?: string;
  LastModifiedDate?: string;
  MasterRecordId?: null;
  OwnerId?: string;
  RecordTypeId?: string;
  Status?: string;
  SystemModstamp?: string;
};

export type SalesforceV1CaseGetNode = {
  type: 'n8n-nodes-base.salesforce';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SalesforceV1CaseGetParams>;
  output?: Items<SalesforceV1CaseGetOutput>;
};
