// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/salesforce/v1/resource_case/operation_create.schema.js
/**
 * Salesforce Node - Version 1 - Zod Schema
 * Discriminator: resource=case, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('case'),
      operation: z.literal('create').default('create'),
      authentication: z.union([z.literal('oAuth2'), z.literal('jwt'), expressionSchema]).optional(),
      type: stringOrExpression.optional(),
      additionalFields: z.object({ accountId: stringOrExpression.optional(), contactId: stringOrExpression.optional(), customFieldsUi: z.unknown().optional(), description: stringOrExpression.optional(), isEscalated: booleanOrExpression.optional(), origin: stringOrExpression.optional(), owner: stringOrExpression.optional(), ParentId: stringOrExpression.optional(), priority: stringOrExpression.optional(), reason: stringOrExpression.optional(), recordTypeId: stringOrExpression.optional(), status: stringOrExpression.optional(), subject: stringOrExpression.optional(), suppliedCompany: stringOrExpression.optional(), suppliedEmail: stringOrExpression.optional(), suppliedName: stringOrExpression.optional(), suppliedPhone: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
