// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/paddle/v1/resource_payment/operation_get_all.ts
/**
 * Paddle Node - Version 1
 * Discriminator: resource=payment, operation=getAll
 */


interface Credentials {
  paddleApi: CredentialReference;
}

/** Get many coupons */
export type PaddleV1PaymentGetAllParams = {
  resource: 'payment';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * JSON Parameters
 * @default false
 */
    jsonParameters?: boolean | Expression<boolean>;
/**
 * Attributes in JSON form
 * @displayOptions.show { jsonParameters: [true] }
 */
    additionalFieldsJson?: IDataObject | string | Expression<string>;
/**
 * Additional Fields
 * @displayOptions.show { jsonParameters: [false] }
 * @default {}
 */
    additionalFields?: {
    /** Payment starting from date
     */
    from?: string | Expression<string>;
    /** Payment up until date
     */
    to?: string | Expression<string>;
    /** Whether payment is paid
     * @default false
     */
    isPaid?: boolean | Expression<boolean>;
    /** Filter: The product/plan ID (single or comma-separated values)
     */
    plan?: string | Expression<string> | PlaceholderValue;
    /** A specific user subscription ID
     */
    subscriptionId?: number | Expression<number>;
    /** Filter: The user subscription status. Returns all active, past_due, trialing and paused subscription plans if not specified.
     * @default active
     */
    state?: 'active' | 'past_due' | 'paused' | 'trialing' | Expression<string>;
    /** One Off Charge
     * @default false
     */
    isOneOffCharge?: boolean | Expression<boolean>;
  };
};

export type PaddleV1PaymentGetAllNode = {
  type: 'n8n-nodes-base.paddle';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PaddleV1PaymentGetAllParams>;
};
