// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleSlides/v1/resource_presentation/operation_replace_text.schema.js
/**
 * Google Slides Node - Version 1 - Zod Schema
 * Discriminator: resource=presentation, operation=replaceText
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('presentation').default('presentation'),
      operation: z.literal('replaceText'),
      authentication: z.union([z.literal('oAuth2'), z.literal('serviceAccount'), expressionSchema]).optional(),
      presentationId: stringOrExpression.optional(),
      textUi: z.object({ textValues: z.array(z.object({ matchCase: booleanOrExpression.optional(), pageObjectIds: z.array(z.string()).optional(), text: stringOrExpression.optional(), replaceText: stringOrExpression.optional() })).optional() }).optional(),
      options: z.object({ revisionId: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
