// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleSlides/v1/resource_presentation/operation_get_slides.ts
/**
 * Google Slides Node - Version 1
 * Discriminator: resource=presentation, operation=getSlides
 */


interface Credentials {
  googleApi: CredentialReference;
  googleSlidesOAuth2Api: CredentialReference;
}

/** Get presentation slides */
export type GoogleSlidesV1PresentationGetSlidesParams = {
  resource: 'presentation';
  operation: 'getSlides';
/**
 * Authentication
 * @default serviceAccount
 */
    authentication?: 'oAuth2' | 'serviceAccount' | Expression<string>;
/**
 * ID of the presentation to retrieve. Found in the presentation URL: &lt;code&gt;https://docs.google.com/presentation/d/PRESENTATION_ID/edit&lt;/code&gt;
 */
    presentationId?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
};

export type GoogleSlidesV1PresentationGetSlidesNode = {
  type: 'n8n-nodes-base.googleSlides';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GoogleSlidesV1PresentationGetSlidesParams>;
};
