// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/emelia/v1/resource_campaign/operation_duplicate.ts
/**
 * Emelia Node - Version 1
 * Discriminator: resource=campaign, operation=duplicate
 */


interface Credentials {
  emeliaApi: CredentialReference;
}

export type EmeliaV1CampaignDuplicateParams = {
  resource: 'campaign';
  operation: 'duplicate';
/**
 * The ID of the campaign to duplicate. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    campaignId?: string | Expression<string>;
/**
 * The name of the new campaign to create
 */
    campaignName?: string | Expression<string> | PlaceholderValue;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Whether to copy all the contacts from the original campaign
     * @default false
     */
    copyContacts?: boolean | Expression<boolean>;
    /** Whether to set the same email provider than the original campaign
     * @default true
     */
    copyProvider?: boolean | Expression<boolean>;
    /** Whether to copy all the steps of the email sequence from the original campaign
     * @default true
     */
    copyMails?: boolean | Expression<boolean>;
    /** Whether to copy all the general settings from the original campaign
     * @default true
     */
    copySettings?: boolean | Expression<boolean>;
  };
};

export type EmeliaV1CampaignDuplicateNode = {
  type: 'n8n-nodes-base.emelia';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<EmeliaV1CampaignDuplicateParams>;
};
