// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftExcel/v22/resource_workbook/operation_add_worksheet.schema.js
/**
 * Microsoft Excel 365 Node - Version 2.2 - Zod Schema
 * Discriminator: resource=workbook, operation=addWorksheet
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('workbook').default('workbook'),
      operation: z.literal('addWorksheet'),
      workbook: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      additionalFields: z.object({ name: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
