// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/intercom/v1/resource_lead/operation_get_all.ts
/**
 * Intercom Node - Version 1
 * Discriminator: resource=lead, operation=getAll
 */


interface Credentials {
  intercomApi: CredentialReference;
}

/** Leads are useful for representing logged-out users of your application */
export type IntercomV1LeadGetAllParams = {
  resource: 'lead';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** The email address of the lead
     */
    email?: string | Expression<string> | PlaceholderValue;
    /** The phone number of the lead
     */
    phone?: string | Expression<string> | PlaceholderValue;
  };
};

export type IntercomV1LeadGetAllNode = {
  type: 'n8n-nodes-base.intercom';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<IntercomV1LeadGetAllParams>;
};
