// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftOutlook/v2/resource_contact/operation_get_all.schema.js
/**
 * Microsoft Outlook Node - Version 2 - Zod Schema
 * Discriminator: resource=contact, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('contact'),
      operation: z.literal('getAll').default('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      output: z.union([z.literal('simple'), z.literal('raw'), z.literal('fields'), expressionSchema]).optional(),
      fields: resolveSchema({ parameters, schema: z.array(z.union([z.literal('createdDateTime'), z.literal('lastModifiedDateTime'), z.literal('changeKey'), z.literal('categories'), z.literal('parentFolderId'), z.literal('birthday'), z.literal('fileAs'), z.literal('displayName'), z.literal('givenName'), z.literal('initials'), z.literal('middleName'), z.literal('nickName'), z.literal('surname'), z.literal('title'), z.literal('yomiGivenName'), z.literal('yomiSurname'), z.literal('yomiCompanyName'), z.literal('generation'), z.literal('imAddresses'), z.literal('jobTitle'), z.literal('companyName'), z.literal('department'), z.literal('officeLocation'), z.literal('profession'), z.literal('businessHomePage'), z.literal('assistantName'), z.literal('manager'), z.literal('homePhones'), z.literal('mobilePhone'), z.literal('businessPhones'), z.literal('spouseName'), z.literal('personalNotes'), z.literal('children'), z.literal('emailAddresses'), z.literal('homeAddress'), z.literal('businessAddress'), z.literal('otherAddress')])), required: false, displayOptions: {"show":{"output":["fields"]}}, defaults: {"output":"simple"} }),
      filters: z.object({ custom: stringOrExpression.optional(), emailAddress: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
