// @microsoft/agents-a365-observability/dist/cjs/tracing/exporter/utils.js
"use strict";
// ------------------------------------------------------------------------------
// Copyright (c) Microsoft Corporation. All rights reserved.
// ------------------------------------------------------------------------------
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.hexTraceId = hexTraceId;
exports.hexSpanId = hexSpanId;
exports.asStr = asStr;
exports.kindName = kindName;
exports.statusName = statusName;
exports.partitionByIdentity = partitionByIdentity;
exports.isAgent365ExporterEnabled = isAgent365ExporterEnabled;
exports.parseIdentityKey = parseIdentityKey;
const api_1 = require("@opentelemetry/api");
const constants_1 = require("../constants");
const logging_1 = __importDefault(require("../../utils/logging"));
/**
 * Convert trace ID to hex string format
 */
function hexTraceId(value) {
    if (typeof value === 'number') {
        // Convert integer to 32 hex chars (128-bit)
        return value.toString(16).padStart(32, '0');
    }
    // Handle hex string input - ensure it's 32 hex characters
    return value.replace(/^0x/, '').padStart(32, '0');
}
/**
 * Convert span ID to hex string format
 */
function hexSpanId(value) {
    if (typeof value === 'number') {
        // Convert integer to 16 hex chars (64-bit)
        return value.toString(16).padStart(16, '0');
    }
    // Handle hex string input - ensure it's 16 hex characters
    return value.replace(/^0x/, '').padStart(16, '0');
}
/**
 * Convert any value to string, handling null/undefined
 */
function asStr(v) {
    if (v === null || v === undefined) {
        return undefined;
    }
    const s = String(v);
    return s.trim() ? s : undefined;
}
/**
 * Get span kind name from SpanKind enum
 */
function kindName(kind) {
    switch (kind) {
        case api_1.SpanKind.INTERNAL:
            return 'INTERNAL';
        case api_1.SpanKind.SERVER:
            return 'SERVER';
        case api_1.SpanKind.CLIENT:
            return 'CLIENT';
        case api_1.SpanKind.PRODUCER:
            return 'PRODUCER';
        case api_1.SpanKind.CONSUMER:
            return 'CONSUMER';
        default:
            return 'UNSPECIFIED';
    }
}
/**
 * Get status name from SpanStatusCode enum
 */
function statusName(code) {
    switch (code) {
        case api_1.SpanStatusCode.UNSET:
            return 'UNSET';
        case api_1.SpanStatusCode.OK:
            return 'OK';
        case api_1.SpanStatusCode.ERROR:
            return 'ERROR';
        default:
            return 'UNSET';
    }
}
/**
 * Partition spans by (tenantId, agentId) identity pairs
 */
function partitionByIdentity(spans) {
    const groups = new Map();
    let skippedCount = 0;
    for (const span of spans) {
        const attrs = span.attributes || {};
        const tenant = asStr(attrs[constants_1.OpenTelemetryConstants.TENANT_ID_KEY]);
        const agent = asStr(attrs[constants_1.OpenTelemetryConstants.GEN_AI_AGENT_ID_KEY]);
        if (!tenant || !agent) {
            skippedCount++;
            logging_1.default.warn(`[Agent365Exporter] Skipping span without tenant or agent ID. Span name: ${span.name}`);
            continue;
        }
        const key = `${tenant}:${agent}`;
        const existing = groups.get(key) || [];
        existing.push(span);
        groups.set(key, existing);
    }
    logging_1.default.info(`[Agent365Exporter] Partitioned into ${groups.size} identity groups (${skippedCount} spans skipped)`);
    return groups;
}
/**
 * Check if Agent 365 exporter is enabled via environment variable
 */
function isAgent365ExporterEnabled() {
    const a365Env = process.env[constants_1.OpenTelemetryConstants.ENABLE_A365_OBSERVABILITY_EXPORTER]?.toLowerCase() || '';
    const validValues = ['true', '1', 'yes', 'on'];
    const enabled = validValues.includes(a365Env);
    logging_1.default.info(`[Agent365Exporter] Agent 365 exporter enabled: ${enabled}`);
    return enabled;
}
/**
 * Parse identity key back to tenant and agent IDs
 */
function parseIdentityKey(key) {
    const [tenantId, agentId] = key.split(':');
    return { tenantId, agentId };
}
