// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/spotify/v1/resource_my_data/operation_get_following_artists.ts
/**
 * Spotify Node - Version 1
 * Discriminator: resource=myData, operation=getFollowingArtists
 */


interface Credentials {
  spotifyOAuth2Api: CredentialReference;
}

/** Get your followed artists */
export type SpotifyV1MyDataGetFollowingArtistsParams = {
  resource: 'myData';
  operation: 'getFollowingArtists';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
};

export type SpotifyV1MyDataGetFollowingArtistsOutput = {
  external_urls?: {
    spotify?: string;
  };
  followers?: {
    href?: null;
    total?: number;
  };
  genres?: Array<string>;
  href?: string;
  id?: string;
  images?: Array<{
    height?: number;
    url?: string;
    width?: number;
  }>;
  name?: string;
  popularity?: number;
  type?: string;
  uri?: string;
};

export type SpotifyV1MyDataGetFollowingArtistsNode = {
  type: 'n8n-nodes-base.spotify';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SpotifyV1MyDataGetFollowingArtistsParams>;
  output?: Items<SpotifyV1MyDataGetFollowingArtistsOutput>;
};
