// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/goToWebinar/v1/resource_coorganizer/operation_delete.ts
/**
 * GoToWebinar Node - Version 1
 * Discriminator: resource=coorganizer, operation=delete
 */


interface Credentials {
  goToWebinarOAuth2Api: CredentialReference;
}

export type GoToWebinarV1CoorganizerDeleteParams = {
  resource: 'coorganizer';
  operation: 'delete';
/**
 * Key of the webinar to delete. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 * @default []
 */
    webinarKey?: string | Expression<string>;
/**
 * Key of the co-organizer to delete
 */
    coorganizerKey?: string | Expression<string> | PlaceholderValue;
/**
 * By default only internal co-organizers (with a GoToWebinar account) can be deleted. If you want to use this call for external co-organizers you have to set this parameter to 'true'.
 * @default false
 */
    isExternal?: boolean | Expression<boolean>;
};

export type GoToWebinarV1CoorganizerDeleteNode = {
  type: 'n8n-nodes-base.goToWebinar';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GoToWebinarV1CoorganizerDeleteParams>;
};
