// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/goToWebinar/v1/resource_coorganizer/operation_create.ts
/**
 * GoToWebinar Node - Version 1
 * Discriminator: resource=coorganizer, operation=create
 */


interface Credentials {
  goToWebinarOAuth2Api: CredentialReference;
}

export type GoToWebinarV1CoorganizerCreateParams = {
  resource: 'coorganizer';
  operation: 'create';
/**
 * Key of the webinar that the co-organizer is hosting. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 * @default []
 */
    webinarKey?: string | Expression<string>;
/**
 * Whether the co-organizer has no GoToWebinar account
 * @default false
 */
    isExternal?: boolean | Expression<boolean>;
/**
 * The co-organizer's organizer key for the webinar
 * @displayOptions.show { isExternal: [false] }
 */
    organizerKey?: string | Expression<string> | PlaceholderValue;
/**
 * The co-organizer's given name
 * @displayOptions.show { isExternal: [true] }
 */
    givenName?: string | Expression<string> | PlaceholderValue;
/**
 * The co-organizer's email address
 * @displayOptions.show { isExternal: [true] }
 */
    email?: string | Expression<string> | PlaceholderValue;
};

export type GoToWebinarV1CoorganizerCreateNode = {
  type: 'n8n-nodes-base.goToWebinar';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GoToWebinarV1CoorganizerCreateParams>;
};
