// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/goToWebinar/v1/resource_coorganizer/operation_create.schema.js
/**
 * GoToWebinar Node - Version 1 - Zod Schema
 * Discriminator: resource=coorganizer, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('coorganizer'),
      operation: z.literal('create'),
      webinarKey: stringOrExpression.optional(),
      isExternal: booleanOrExpression.optional(),
      organizerKey: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"isExternal":[false]}}, defaults: {"isExternal":false} }),
      givenName: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"isExternal":[true]}}, defaults: {"isExternal":false} }),
      email: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"isExternal":[true]}}, defaults: {"isExternal":false} }),
    }).optional(),
  });
};
