// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/freshservice/v1/resource_agent/operation_get_all.schema.js
/**
 * Freshservice Node - Version 1 - Zod Schema
 * Discriminator: resource=agent, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('agent').default('agent'),
      operation: z.literal('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filters: z.object({ department_id: stringOrExpression.optional(), email: stringOrExpression.optional(), first_name: stringOrExpression.optional(), job_title: stringOrExpression.optional(), language: z.union([z.literal('en'), z.literal('ar'), z.literal('ca'), z.literal('cs'), z.literal('cy-GB'), z.literal('da'), z.literal('de'), z.literal('es'), z.literal('es-LA'), z.literal('et'), z.literal('fi'), z.literal('fr'), z.literal('he'), z.literal('hu'), z.literal('id'), z.literal('it'), z.literal('ja-JP'), z.literal('ko'), z.literal('LV'), z.literal('nb-NO'), z.literal('nl'), z.literal('pl'), z.literal('pt'), z.literal('pt-BR'), z.literal('pt-PT'), z.literal('ru-RU'), z.literal('sk'), z.literal('sk-SK'), z.literal('sl'), z.literal('sv-SE'), z.literal('th'), z.literal('tr'), z.literal('UK'), z.literal('vi'), z.literal('zh-CN'), z.literal('zh-TW'), expressionSchema]).optional(), last_name: stringOrExpression.optional(), location_id: stringOrExpression.optional(), mobile_phone_number: stringOrExpression.optional(), work_phone_number: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
