// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/theHiveProject/v1/resource_comment/operation_add.schema.js
/**
 * TheHive 5 Node - Version 1 - Zod Schema
 * Discriminator: resource=comment, operation=add
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('comment'),
      operation: z.literal('add'),
      addTo: z.union([z.literal('alert'), z.literal('case'), expressionSchema]).optional(),
      id: resolveSchema({ parameters, schema: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('url'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]), required: false, displayOptions: {"show":{"addTo":["case","alert"]}}, defaults: {"addTo":"alert"} }),
      message: stringOrExpression.optional(),
    }).optional(),
  });
};
