// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/helpScout/v1/resource_thread/operation_create.schema.js
/**
 * Help Scout Node - Version 1 - Zod Schema
 * Discriminator: resource=thread, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('thread'),
      operation: z.literal('create').default('create'),
      conversationId: stringOrExpression.optional(),
      type: z.union([z.literal('chat'), z.literal('customer'), z.literal('note'), z.literal('phone'), z.literal('reply'), expressionSchema]).optional(),
      text: stringOrExpression.optional(),
      additionalFields: z.object({ createdAt: stringOrExpression.optional(), customerEmail: stringOrExpression.optional(), customerId: numberOrExpression.optional(), draft: booleanOrExpression.optional(), imported: booleanOrExpression.optional() }).optional(),
      attachmentsUi: z.object({ attachmentsValues: z.array(z.object({ fileName: stringOrExpression.optional(), mimeType: stringOrExpression.optional(), data: stringOrExpression.optional() })).optional(), attachmentsBinary: z.array(z.object({ property: stringOrExpression.optional() })).optional() }).optional(),
    }).optional(),
  });
};
