// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/youTube/v1/resource_video/operation_update.schema.js
/**
 * YouTube Node - Version 1 - Zod Schema
 * Discriminator: resource=video, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('video'),
      operation: z.literal('update'),
      videoId: stringOrExpression.optional(),
      title: stringOrExpression.optional(),
      regionCode: stringOrExpression.optional(),
      categoryId: stringOrExpression.optional(),
      updateFields: z.object({ defaultLanguage: stringOrExpression.optional(), description: stringOrExpression.optional(), embeddable: booleanOrExpression.optional(), license: z.union([z.literal('creativeCommon'), z.literal('youtube'), expressionSchema]).optional(), notifySubscribers: booleanOrExpression.optional(), privacyStatus: z.union([z.literal('private'), z.literal('public'), z.literal('unlistef'), expressionSchema]).optional(), publicStatsViewable: booleanOrExpression.optional(), publishAt: stringOrExpression.optional(), recordingDate: stringOrExpression.optional(), selfDeclaredMadeForKids: booleanOrExpression.optional(), tags: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
