// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/youTube/v1/resource_video/operation_get_all.schema.js
/**
 * YouTube Node - Version 1 - Zod Schema
 * Discriminator: resource=video, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('video'),
      operation: z.literal('getAll').default('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filters: z.object({ channelId: stringOrExpression.optional(), forDeveloper: booleanOrExpression.optional(), publishedAfter: stringOrExpression.optional(), publishedBefore: stringOrExpression.optional(), q: stringOrExpression.optional(), regionCode: stringOrExpression.optional(), relatedToVideoId: stringOrExpression.optional(), videoCategoryId: stringOrExpression.optional(), videoSyndicated: booleanOrExpression.optional(), videoType: z.union([z.literal('any'), z.literal('episode'), z.literal('movie'), expressionSchema]).optional() }).optional(),
      options: z.object({ order: z.union([z.literal('date'), z.literal('relevance'), expressionSchema]).optional(), safeSearch: z.union([z.literal('moderate'), z.literal('none'), z.literal('strict'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
