// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/getResponseTrigger/v1.schema.js
/**
 * GetResponse Trigger Node - Version 1 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    authentication: z.union([z.literal('apiKey'), z.literal('oAuth2'), expressionSchema]).optional(),
    events: z.array(z.union([z.literal('subscribe'), z.literal('unsubscribe'), z.literal('click'), z.literal('open'), z.literal('survey')])).optional(),
    listIds: z.array(z.string()).optional(),
    options: z.object({ 'delete': booleanOrExpression.optional() }).optional(),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
