// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/getResponseTrigger/v1.ts
/**
 * GetResponse Trigger Node - Version 1
 * Starts the workflow when GetResponse events occur
 */


export interface GetResponseTriggerV1Params {
  authentication?: 'apiKey' | 'oAuth2' | Expression<string>;
  events?: Array<'subscribe' | 'unsubscribe' | 'click' | 'open' | 'survey'>;
/**
 * Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 * @default []
 */
    listIds?: string[];
  options?: {
    /** Whether to delete the current subscription
     * @default false
     */
    'delete'?: boolean | Expression<boolean>;
  };
}

export interface GetResponseTriggerV1Credentials {
  getResponseApi: CredentialReference;
  getResponseOAuth2Api: CredentialReference;
}

interface GetResponseTriggerV1NodeBase {
  type: 'n8n-nodes-base.getResponseTrigger';
  version: 1;
  credentials?: GetResponseTriggerV1Credentials;
  isTrigger: true;
}

export type GetResponseTriggerV1ParamsNode = GetResponseTriggerV1NodeBase & {
  config: NodeConfig<GetResponseTriggerV1Params>;
};

export type GetResponseTriggerV1Node = GetResponseTriggerV1ParamsNode;
