// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/gmail/v22/resource_label/operation_create.schema.js
/**
 * Gmail Node - Version 2.2 - Zod Schema
 * Discriminator: resource=label, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('label'),
      operation: z.literal('create').default('create'),
      authentication: z.union([z.literal('oAuth2'), z.literal('serviceAccount'), expressionSchema]).optional(),
      name: stringOrExpression.optional(),
      options: z.object({ labelListVisibility: z.union([z.literal('labelHide'), z.literal('labelShow'), z.literal('labelShowIfUnread'), expressionSchema]).optional(), messageListVisibility: z.union([z.literal('hide'), z.literal('show'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
