// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/shopify/v1/resource_product/operation_get_all.ts
/**
 * Shopify Node - Version 1
 * Discriminator: resource=product, operation=getAll
 */


interface Credentials {
  shopifyApi: CredentialReference;
  shopifyAccessTokenApi: CredentialReference;
  shopifyOAuth2Api: CredentialReference;
}

/** Get many orders */
export type ShopifyV1ProductGetAllParams = {
  resource: 'product';
  operation: 'getAll';
  authentication?: 'accessToken' | 'oAuth2' | 'apiKey' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Filter results by product collection ID
     */
    collection_id?: string | Expression<string> | PlaceholderValue;
    /** Show products created before date
     */
    created_at_max?: string | Expression<string>;
    /** Show products created after date
     */
    created_at_min?: string | Expression<string>;
    /** Show only certain fields, specified by a comma-separated list of field names
     */
    fields?: string | Expression<string> | PlaceholderValue;
    /** Filter results by product handle
     */
    handle?: string | Expression<string> | PlaceholderValue;
    /** Return only products specified by a comma-separated list of product IDs
     */
    ids?: string | Expression<string> | PlaceholderValue;
    /** Return presentment prices in only certain currencies, specified by a comma-separated list of ISO 4217 currency codes
     */
    presentment_currencies?: string | Expression<string> | PlaceholderValue;
    /** Filter results by product type
     */
    product_type?: string | Expression<string> | PlaceholderValue;
    /** Show products published before date
     */
    published_at_max?: string | Expression<string>;
    /** Show products published after date
     */
    published_at_min?: string | Expression<string>;
    /** Return products by their published status
     * @default any
     */
    published_status?: 'any' | 'published' | 'unpublished' | Expression<string>;
    /** Filter results by product title
     */
    title?: string | Expression<string> | PlaceholderValue;
    /** Show products last updated before date
     */
    updated_at_max?: string | Expression<string>;
    /** Show products last updated after date
     */
    updated_at_min?: string | Expression<string>;
    /** Filter results by product vendor
     */
    vendor?: string | Expression<string> | PlaceholderValue;
  };
};

export type ShopifyV1ProductGetAllOutput = {
  admin_graphql_api_id?: string;
  created_at?: string;
  handle?: string;
  id?: number;
  images?: Array<{
    admin_graphql_api_id?: string;
    created_at?: string;
    height?: number;
    id?: number;
    position?: number;
    product_id?: number;
    src?: string;
    updated_at?: string;
    variant_ids?: Array<number>;
    width?: number;
  }>;
  options?: Array<{
    id?: number;
    name?: string;
    position?: number;
    product_id?: number;
    values?: Array<string>;
  }>;
  product_type?: string;
  published_scope?: string;
  status?: string;
  tags?: string;
  title?: string;
  updated_at?: string;
  variants?: Array<{
    admin_graphql_api_id?: string;
    created_at?: string;
    fulfillment_service?: string;
    grams?: number;
    id?: number;
    inventory_item_id?: number;
    inventory_policy?: string;
    inventory_quantity?: number;
    old_inventory_quantity?: number;
    option1?: string;
    position?: number;
    price?: string;
    product_id?: number;
    requires_shipping?: boolean;
    taxable?: boolean;
    title?: string;
    updated_at?: string;
    weight_unit?: string;
  }>;
  vendor?: string;
};

export type ShopifyV1ProductGetAllNode = {
  type: 'n8n-nodes-base.shopify';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ShopifyV1ProductGetAllParams>;
  output?: Items<ShopifyV1ProductGetAllOutput>;
};
