// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/shopify/v1/resource_product/operation_get_all.schema.js
/**
 * Shopify Node - Version 1 - Zod Schema
 * Discriminator: resource=product, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('product'),
      operation: z.literal('getAll'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), z.literal('apiKey'), expressionSchema]).optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      additionalFields: z.object({ collection_id: stringOrExpression.optional(), created_at_max: stringOrExpression.optional(), created_at_min: stringOrExpression.optional(), fields: stringOrExpression.optional(), handle: stringOrExpression.optional(), ids: stringOrExpression.optional(), presentment_currencies: stringOrExpression.optional(), product_type: stringOrExpression.optional(), published_at_max: stringOrExpression.optional(), published_at_min: stringOrExpression.optional(), published_status: z.union([z.literal('any'), z.literal('published'), z.literal('unpublished'), expressionSchema]).optional(), title: stringOrExpression.optional(), updated_at_max: stringOrExpression.optional(), updated_at_min: stringOrExpression.optional(), vendor: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
