// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/shopify/v1/resource_product/operation_get.ts
/**
 * Shopify Node - Version 1
 * Discriminator: resource=product, operation=get
 */


interface Credentials {
  shopifyApi: CredentialReference;
  shopifyAccessTokenApi: CredentialReference;
  shopifyOAuth2Api: CredentialReference;
}

/** Get an order */
export type ShopifyV1ProductGetParams = {
  resource: 'product';
  operation: 'get';
  authentication?: 'accessToken' | 'oAuth2' | 'apiKey' | Expression<string>;
/**
 * Product ID
 */
    productId?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Fields the product will return, formatted as a string of comma-separated values. By default all the fields are returned.
     */
    fields?: string | Expression<string> | PlaceholderValue;
  };
};

export type ShopifyV1ProductGetOutput = {
  admin_graphql_api_id?: string;
  created_at?: string;
  handle?: string;
  id?: number;
  image?: {
    admin_graphql_api_id?: string;
    created_at?: string;
    height?: number;
    id?: number;
    position?: number;
    product_id?: number;
    src?: string;
    updated_at?: string;
    variant_ids?: Array<number>;
    width?: number;
  };
  images?: Array<{
    admin_graphql_api_id?: string;
    created_at?: string;
    height?: number;
    id?: number;
    position?: number;
    product_id?: number;
    src?: string;
    updated_at?: string;
    variant_ids?: Array<number>;
    width?: number;
  }>;
  options?: Array<{
    id?: number;
    name?: string;
    position?: number;
    product_id?: number;
    values?: Array<string>;
  }>;
  product_type?: string;
  published_scope?: string;
  status?: string;
  tags?: string;
  title?: string;
  updated_at?: string;
  variants?: Array<{
    admin_graphql_api_id?: string;
    created_at?: string;
    fulfillment_service?: string;
    grams?: number;
    id?: number;
    inventory_item_id?: number;
    inventory_policy?: string;
    inventory_quantity?: number;
    old_inventory_quantity?: number;
    option1?: string;
    position?: number;
    price?: string;
    product_id?: number;
    requires_shipping?: boolean;
    taxable?: boolean;
    title?: string;
    updated_at?: string;
    weight_unit?: string;
  }>;
  vendor?: string;
};

export type ShopifyV1ProductGetNode = {
  type: 'n8n-nodes-base.shopify';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ShopifyV1ProductGetParams>;
  output?: Items<ShopifyV1ProductGetOutput>;
};
