// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/shopify/v1/resource_product/operation_create.ts
/**
 * Shopify Node - Version 1
 * Discriminator: resource=product, operation=create
 */


interface Credentials {
  shopifyApi: CredentialReference;
  shopifyAccessTokenApi: CredentialReference;
  shopifyOAuth2Api: CredentialReference;
}

/** Create an order */
export type ShopifyV1ProductCreateParams = {
  resource: 'product';
  operation: 'create';
  authentication?: 'accessToken' | 'oAuth2' | 'apiKey' | Expression<string>;
/**
 * The name of the product
 */
    title?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** A description of the product. Supports HTML formatting.
     */
    body_html?: string | Expression<string> | PlaceholderValue;
    /** A unique human-friendly string for the product. Automatically generated from the product's title. Used by the Liquid templating language to refer to objects.
     */
    handle?: string | Expression<string> | PlaceholderValue;
    /** A list of product image objects, each one representing an image associated with the product
     * @default {}
     */
    images?: {
    /** The date and time when the product image was created
     */
    created_at?: string | Expression<string>;
    /** A unique numeric identifier for the product image
     */
    id?: number | Expression<number>;
    /** The order of the product image in the list. The first product image is at position 1 and is the "main" image for the product.
     */
    position?: number | Expression<number>;
    /** The ID of the product associated with the image
     */
    product_id?: number | Expression<number>;
    /** An array of variant IDs associated with the image
     */
    variant_ids?: number | Expression<number>;
    /** &lt;p&gt;Specifies the location of the product image. This parameter supports URL filters that you can use to retrieve modified copies of the image.&lt;/p&gt;&lt;p&gt;For example, add _small, to the filename to retrieve a scaled copy of the image at 100 x 100 px (for example, ipod-nano_small.png), or add _2048x2048 to retrieve a copy of the image constrained at 2048 x 2048 px resolution (for example, ipod-nano_2048x2048.png).&lt;/p&gt;.
     */
    src?: string | Expression<string> | PlaceholderValue;
    /** Width dimension of the image which is determined on upload
     */
    width?: number | Expression<number>;
    /** Height dimension of the image which is determined on upload
     */
    height?: number | Expression<number>;
    /** The date and time when the product image was last modified
     */
    updated_at?: string | Expression<string>;
  };
    /** The custom product property names like Size, Color, and Material. You can add up to 3 options of up to 255 characters each.
     * @default {}
     */
    productOptions?: {
        /** Option
     */
    option?: Array<{
      /** Option's name
       */
      name?: string | Expression<string> | PlaceholderValue;
      /** Option's values
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
    /** A categorization for the product used for filtering and searching products
     */
    product_type?: string | Expression<string> | PlaceholderValue;
    /** The date and time (ISO 8601 format) when the product was published. Can be set to null to unpublish the product from the Online Store channel.
     */
    published_at?: string | Expression<string>;
    /** Published Scope
     */
    published_scope?: 'global' | 'web' | Expression<string>;
    /** A string of comma-separated tags that are used for filtering and search. A product can have up to 250 tags. Each tag can have up to 255 characters.
     */
    tags?: string | Expression<string> | PlaceholderValue;
    /** The suffix of the Liquid template used for the product page. If this property is specified, then the product page uses a template called "product.suffix.liquid", where "suffix" is the value of this property. If this property is "" or null, then the product page uses the default template "product.liquid". (default: null)
     */
    template_suffix?: string | Expression<string> | PlaceholderValue;
    /** The name of the product's vendor
     */
    vendor?: string | Expression<string> | PlaceholderValue;
  };
};

export type ShopifyV1ProductCreateOutput = {
  admin_graphql_api_id?: string;
  created_at?: string;
  handle?: string;
  id?: number;
  images?: Array<{
    admin_graphql_api_id?: string;
    alt?: null;
    created_at?: string;
    height?: number;
    id?: number;
    position?: number;
    product_id?: number;
    src?: string;
    updated_at?: string;
    width?: number;
  }>;
  options?: Array<{
    id?: number;
    name?: string;
    position?: number;
    product_id?: number;
    values?: Array<string>;
  }>;
  product_type?: string;
  published_scope?: string;
  status?: string;
  tags?: string;
  title?: string;
  updated_at?: string;
  variants?: Array<{
    admin_graphql_api_id?: string;
    barcode?: null;
    compare_at_price?: null;
    created_at?: string;
    fulfillment_service?: string;
    grams?: number;
    id?: number;
    image_id?: null;
    inventory_item_id?: number;
    inventory_management?: null;
    inventory_policy?: string;
    inventory_quantity?: number;
    old_inventory_quantity?: number;
    option1?: string;
    option2?: null;
    option3?: null;
    position?: number;
    price?: string;
    product_id?: number;
    requires_shipping?: boolean;
    sku?: string;
    taxable?: boolean;
    title?: string;
    updated_at?: string;
    weight?: number;
    weight_unit?: string;
  }>;
  vendor?: string;
};

export type ShopifyV1ProductCreateNode = {
  type: 'n8n-nodes-base.shopify';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ShopifyV1ProductCreateParams>;
  output?: Items<ShopifyV1ProductCreateOutput>;
};
