// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/gmail/v22/resource_message/operation_get.schema.js
/**
 * Gmail Node - Version 2.2 - Zod Schema
 * Discriminator: resource=message, operation=get
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('message').default('message'),
      operation: z.literal('get'),
      authentication: z.union([z.literal('oAuth2'), z.literal('serviceAccount'), expressionSchema]).optional(),
      messageId: stringOrExpression.optional(),
      simple: booleanOrExpression.optional(),
      options: resolveSchema({ parameters, schema: z.object({ dataPropertyAttachmentsPrefixName: stringOrExpression.optional(), downloadAttachments: booleanOrExpression.optional() }), required: false, displayOptions: {"hide":{"simple":[true]}}, defaults: {"simple":true} }),
    }).optional(),
  });
};
