// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/discourse/v1/resource_post/operation_update.schema.js
/**
 * Discourse Node - Version 1 - Zod Schema
 * Discriminator: resource=post, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('post').default('post'),
      operation: z.literal('update'),
      postId: stringOrExpression.optional(),
      content: stringOrExpression.optional(),
      updateFields: z.object({ edit_reason: stringOrExpression.optional(), cooked: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
