// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/discourse/v1/resource_user/operation_get.schema.js
/**
 * Discourse Node - Version 1 - Zod Schema
 * Discriminator: resource=user, operation=get
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('user'),
      operation: z.literal('get'),
      by: z.union([z.literal('username'), z.literal('externalId'), expressionSchema]).optional(),
      username: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"by":["username"]}}, defaults: {"by":"username"} }),
      externalId: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"by":["externalId"]}}, defaults: {"by":"username"} }),
    }).optional(),
  });
};
