// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/discourse/v1/resource_user/operation_get_all.schema.js
/**
 * Discourse Node - Version 1 - Zod Schema
 * Discriminator: resource=user, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('user'),
      operation: z.literal('getAll'),
      flag: z.union([z.literal('active'), z.literal('blocked'), z.literal('new'), z.literal('staff'), z.literal('suspect'), z.literal('suspended'), expressionSchema]).optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      options: z.object({ asc: booleanOrExpression.optional(), order: z.union([z.literal('created'), z.literal('days_visited'), z.literal('email'), z.literal('last_emailed'), z.literal('posts'), z.literal('posts_read'), z.literal('read_time'), z.literal('seen'), z.literal('topics_viewed'), z.literal('trust_level'), z.literal('username'), expressionSchema]).optional(), showEmails: booleanOrExpression.optional(), stats: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
