// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/n8n/v1/resource_audit/operation_generate.ts
/**
 * n8n Node - Version 1
 * Discriminator: resource=audit, operation=generate
 */


interface Credentials {
  n8nApi: CredentialReference;
}

/** Generate a security audit for this n8n instance */
export type N8nV1AuditGenerateParams = {
  resource: 'audit';
  operation: 'generate';
/**
 * Additional Options
 * @default {}
 */
    additionalOptions?: {
    /** Risk categories to include in the audit
     * @default []
     */
    categories?: Array<'credentials' | 'database' | 'filesystem' | 'instance' | 'nodes'>;
    /** Days for a workflow to be considered abandoned if not executed
     * @default 90
     */
    daysAbandonedWorkflow?: number | Expression<number>;
  };
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type N8nV1AuditGenerateNode = {
  type: 'n8n-nodes-base.n8n';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<N8nV1AuditGenerateParams>;
};
