// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/clockify/v1/resource_time_entry/operation_update.schema.js
/**
 * Clockify Node - Version 1 - Zod Schema
 * Discriminator: resource=timeEntry, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('timeEntry'),
      operation: z.literal('update'),
      workspaceId: stringOrExpression.optional(),
      timeEntryId: stringOrExpression.optional(),
      updateFields: z.object({ billable: booleanOrExpression.optional(), customFieldsUi: z.unknown().optional(), description: stringOrExpression.optional(), end: stringOrExpression.optional(), projectId: stringOrExpression.optional(), start: stringOrExpression.optional(), tagIds: z.array(z.string()).optional(), taskId: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
