// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/awsCertificateManager/v1/resource_certificate/operation_renew.schema.js
/**
 * AWS Certificate Manager Node - Version 1 - Zod Schema
 * Discriminator: resource=certificate, operation=renew
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('certificate').default('certificate'),
      operation: z.literal('renew').default('renew'),
      authentication: z.union([z.literal('iam'), z.literal('assumeRole'), expressionSchema]).optional(),
      certificateArn: stringOrExpression.optional(),
    }).optional(),
  });
};
