// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/serviceNow/v1/resource_configuration_items/operation_get_all.schema.js
/**
 * ServiceNow Node - Version 1 - Zod Schema
 * Discriminator: resource=configurationItems, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('configurationItems'),
      operation: z.literal('getAll'),
      authentication: z.union([z.literal('basicAuth'), z.literal('oAuth2'), expressionSchema]).optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      options: z.object({ sysparm_exclude_reference_link: booleanOrExpression.optional(), sysparm_fields: z.array(z.string()).optional(), sysparm_query: stringOrExpression.optional(), sysparm_display_value: z.union([z.literal('false'), z.literal('all'), z.literal('true'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
