// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/awsS3/v2/resource_file/operation_delete.schema.js
/**
 * AWS S3 Node - Version 2 - Zod Schema
 * Discriminator: resource=file, operation=delete
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('file').default('file'),
      operation: z.literal('delete'),
      authentication: z.union([z.literal('iam'), z.literal('assumeRole'), expressionSchema]).optional(),
      bucketName: stringOrExpression.optional(),
      fileKey: stringOrExpression.optional(),
      options: z.object({ versionId: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
