// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/highLevel/v2/resource_calendar/operation_get_free_slots.ts
/**
 * HighLevel Node - Version 2
 * Discriminator: resource=calendar, operation=getFreeSlots
 */


interface Credentials {
  highLevelOAuth2Api: CredentialReference;
}

export type HighLevelV2CalendarGetFreeSlotsParams = {
  resource: 'calendar';
  operation: 'getFreeSlots';
/**
 * Calendar ID
 */
    calendarId?: string | Expression<string> | PlaceholderValue;
/**
 * The start date for fetching free calendar slots. Example: 1548898600000.
 */
    startDate?: number | Expression<number>;
/**
 * The end date for fetching free calendar slots. Example: 1601490599999.
 */
    endDate?: number | Expression<number>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** The timezone to use for the returned slots. Example: America/Chihuahua.
     */
    timezone?: string | Expression<string> | PlaceholderValue;
    /** User ID to filter the slots (optional)
     */
    userId?: string | Expression<string> | PlaceholderValue;
    /** User IDs
     * @default {}
     */
    userIds?: {
    /** Comma-separated list of user IDs to filter the slots
     */
    userIds?: string | Expression<string> | PlaceholderValue;
  };
    /** Apply Look Busy to the slots
     * @default false
     */
    enableLookBusy?: boolean | Expression<boolean>;
  };
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type HighLevelV2CalendarGetFreeSlotsOutput = {
  '2026-02-19'?: {
    slots?: Array<string>;
  };
  traceId?: string;
};

export type HighLevelV2CalendarGetFreeSlotsNode = {
  type: 'n8n-nodes-base.highLevel';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<HighLevelV2CalendarGetFreeSlotsParams>;
  output?: Items<HighLevelV2CalendarGetFreeSlotsOutput>;
};
