// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/highLevel/v2/resource_calendar/operation_book_appointment.ts
/**
 * HighLevel Node - Version 2
 * Discriminator: resource=calendar, operation=bookAppointment
 */


interface Credentials {
  highLevelOAuth2Api: CredentialReference;
}

export type HighLevelV2CalendarBookAppointmentParams = {
  resource: 'calendar';
  operation: 'bookAppointment';
/**
 * Calendar ID
 */
    calendarId?: string | Expression<string> | PlaceholderValue;
/**
 * Location ID
 */
    locationId?: string | Expression<string> | PlaceholderValue;
/**
 * Contact ID
 */
    contactId?: string | Expression<string> | PlaceholderValue;
/**
 * Example: 2021-06-23T03:30:00+05:30
 */
    startTime?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Example: 2021-06-23T04:30:00+05:30
     */
    endTime?: string | Expression<string> | PlaceholderValue;
    /** Title
     */
    title?: string | Expression<string> | PlaceholderValue;
    /** The status of the appointment. Allowed values: new, confirmed, cancelled, showed, noshow, invalid.
     * @default new
     */
    appointmentStatus?: 'cancelled' | 'confirmed' | 'invalid' | 'new' | 'noshow' | 'showed' | Expression<string>;
    /** Assigned User ID
     */
    assignedUserId?: string | Expression<string> | PlaceholderValue;
    /** Address
     */
    address?: string | Expression<string> | PlaceholderValue;
    /** Ignore Date Range
     * @default false
     */
    ignoreDateRange?: boolean | Expression<boolean>;
    /** Notify
     * @default true
     */
    toNotify?: boolean | Expression<boolean>;
  };
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type HighLevelV2CalendarBookAppointmentOutput = {
  address?: string;
  appoinmentStatus?: string;
  assignedUserId?: string;
  calendarId?: string;
  contactId?: string;
  id?: string;
  isRecurring?: boolean;
  status?: string;
  title?: string;
  traceId?: string;
};

export type HighLevelV2CalendarBookAppointmentNode = {
  type: 'n8n-nodes-base.highLevel';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<HighLevelV2CalendarBookAppointmentParams>;
  output?: Items<HighLevelV2CalendarBookAppointmentOutput>;
};
