// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/highLevel/v2/resource_calendar/operation_get_free_slots.schema.js
/**
 * HighLevel Node - Version 2 - Zod Schema
 * Discriminator: resource=calendar, operation=getFreeSlots
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('calendar'),
      operation: z.literal('getFreeSlots'),
      calendarId: stringOrExpression.optional(),
      startDate: numberOrExpression.optional(),
      endDate: numberOrExpression.optional(),
      additionalFields: z.object({ timezone: stringOrExpression.optional(), userId: stringOrExpression.optional(), userIds: z.unknown().optional(), enableLookBusy: booleanOrExpression.optional() }).optional(),
      requestOptions: z.object({ batching: z.unknown().optional(), allowUnauthorizedCerts: z.boolean().optional(), proxy: stringOrExpression.optional(), timeout: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
