// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/splunk/v1/resource_search_job/operation_get_all.ts
/**
 * Splunk Node - Version 1
 * Discriminator: resource=searchJob, operation=getAll
 */


interface Credentials {
  splunkApi: CredentialReference;
}

/** Retrieve many search configurations */
export type SplunkV1SearchJobGetAllParams = {
  resource: 'searchJob';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Sort Direction
     * @default asc
     */
    sort_dir?: 'asc' | 'desc' | Expression<string>;
    /** Key name to use for sorting
     */
    sort_key?: string | Expression<string> | PlaceholderValue;
    /** Sort Mode
     * @default auto
     */
    sort_mode?: 'auto' | 'alpha' | 'alpha_case' | 'num' | Expression<string>;
  };
};

export type SplunkV1SearchJobGetAllNode = {
  type: 'n8n-nodes-base.splunk';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SplunkV1SearchJobGetAllParams>;
};
