// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/splunk/v1/resource_search_job/operation_get_all.schema.js
/**
 * Splunk Node - Version 1 - Zod Schema
 * Discriminator: resource=searchJob, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('searchJob').default('searchJob'),
      operation: z.literal('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      options: z.object({ sort_dir: z.union([z.literal('asc'), z.literal('desc'), expressionSchema]).optional(), sort_key: stringOrExpression.optional(), sort_mode: z.union([z.literal('auto'), z.literal('alpha'), z.literal('alpha_case'), z.literal('num'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
