// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/invoiceNinja/v2/resource_payment/operation_create.ts
/**
 * Invoice Ninja Node - Version 2
 * Discriminator: resource=payment, operation=create
 */


interface Credentials {
  invoiceNinjaApi: CredentialReference;
}

/** Create a new client */
export type InvoiceNinjaV2PaymentCreateParams = {
  resource: 'payment';
  operation: 'create';
/**
 * API Version
 * @default v5
 */
    apiVersion?: 'v4' | 'v5' | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    invoice?: string | Expression<string>;
/**
 * Amount
 * @default 0
 */
    amount?: number | Expression<number>;
/**
 * Additional Fields
 * @displayOptions.show { apiVersion: ["v4", "v5"] }
 * @default {}
 */
    additionalFields?: {
    /** Payment Type
     * @default 1
     */
    paymentType?: 5 | 28 | 8 | 1 | 2 | 32 | 17 | 3 | 16 | 13 | 4 | 10 | 9 | 11 | 31 | 15 | 24 | 19 | 20 | 21 | 7 | 27 | 12 | 14 | 30 | 29 | 22 | 23 | 25 | 18 | 26 | 6 | Expression<number>;
    /** Transfer Reference
     */
    transferReference?: string | Expression<string> | PlaceholderValue;
    /** Private Notes
     */
    privateNotes?: string | Expression<string> | PlaceholderValue;
  };
};

export type InvoiceNinjaV2PaymentCreateNode = {
  type: 'n8n-nodes-base.invoiceNinja';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<InvoiceNinjaV2PaymentCreateParams>;
};
