// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/highLevel/v1/resource_opportunity/operation_update.schema.js
/**
 * HighLevel Node - Version 1 - Zod Schema
 * Discriminator: resource=opportunity, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('opportunity'),
      operation: z.literal('update'),
      pipelineId: stringOrExpression.optional(),
      opportunityId: stringOrExpression.optional(),
      updateFields: z.object({ assignedTo: stringOrExpression.optional(), companyName: stringOrExpression.optional(), contactIdentifier: stringOrExpression.optional(), monetaryValue: numberOrExpression.optional(), name: stringOrExpression.optional(), stageId: stringOrExpression.optional(), status: z.union([z.literal('open'), z.literal('won'), z.literal('lost'), z.literal('abandoned'), expressionSchema]).optional(), title: stringOrExpression.optional(), tags: stringOrExpression.optional() }).optional(),
      requestOptions: z.object({ batching: z.unknown().optional(), allowUnauthorizedCerts: z.boolean().optional(), proxy: stringOrExpression.optional(), timeout: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
