// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/highLevel/v1/resource_opportunity/operation_update.ts
/**
 * HighLevel Node - Version 1
 * Discriminator: resource=opportunity, operation=update
 */


interface Credentials {
  highLevelApi: CredentialReference;
}

export type HighLevelV1OpportunityUpdateParams = {
  resource: 'opportunity';
  operation: 'update';
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    pipelineId?: string | Expression<string>;
/**
 * Opportunity ID
 * @hint You cannot update an opportunity's pipeline ID.
 */
    opportunityId?: string | Expression<string> | PlaceholderValue;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** Choose staff member from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     */
    assignedTo?: string | Expression<string>;
    /** Company Name
     */
    companyName?: string | Expression<string> | PlaceholderValue;
    /** Either Email, Phone or Contact ID
     * @hint There can only be one opportunity for each contact.
     */
    contactIdentifier?: string | Expression<string> | PlaceholderValue;
    /** Monetary value of lead opportunity
     */
    monetaryValue?: number | Expression<number>;
    /** Name
     */
    name?: string | Expression<string> | PlaceholderValue;
    /** Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     */
    stageId?: string | Expression<string>;
    /** Status
     * @default open
     */
    status?: 'open' | 'won' | 'lost' | 'abandoned' | Expression<string>;
    /** Title
     */
    title?: string | Expression<string> | PlaceholderValue;
    /** Tags
     * @hint Comma separated list of tags, array of strings can be set in expression
     */
    tags?: string | Expression<string> | PlaceholderValue;
  };
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type HighLevelV1OpportunityUpdateNode = {
  type: 'n8n-nodes-base.highLevel';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<HighLevelV1OpportunityUpdateParams>;
};
