// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/onfleet/v1/resource_recipient/operation_create.schema.js
/**
 * Onfleet Node - Version 1 - Zod Schema
 * Discriminator: resource=recipient, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('recipient'),
      operation: z.literal('create'),
      recipientName: stringOrExpression.optional(),
      recipientPhone: stringOrExpression.optional(),
      additionalFields: z.object({ recipientNotes: stringOrExpression.optional(), recipientSkipSMSNotifications: booleanOrExpression.optional() }).optional(),
      options: z.object({ recipientSkipPhoneNumberValidation: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
