// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/onfleet/v1/resource_recipient/operation_update.ts
/**
 * Onfleet Node - Version 1
 * Discriminator: resource=recipient, operation=update
 */


interface Credentials {
  onfleetApi: CredentialReference;
}

/** Update an Onfleet admin */
export type OnfleetV1RecipientUpdateParams = {
  resource: 'recipient';
  operation: 'update';
/**
 * The ID of the recipient object for lookup
 */
    id?: string | Expression<string> | PlaceholderValue;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** The recipient's complete name
     */
    recipientName?: string | Expression<string> | PlaceholderValue;
    /** Notes for this recipient: these are global notes that should not be task- or destination-specific
     */
    notes?: string | Expression<string> | PlaceholderValue;
    /** A unique, valid phone number as per the organization's country if there's no leading + sign. If a phone number has a leading + sign, it will disregard the organization's country setting.
     */
    recipientPhone?: string | Expression<string> | PlaceholderValue;
    /** Whether this recipient has requested to skip SMS notifications
     * @default false
     */
    skipSMSNotifications?: boolean | Expression<boolean>;
  };
};

export type OnfleetV1RecipientUpdateNode = {
  type: 'n8n-nodes-base.onfleet';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<OnfleetV1RecipientUpdateParams>;
};
