// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/onfleet/v1/resource_recipient/operation_get.schema.js
/**
 * Onfleet Node - Version 1 - Zod Schema
 * Discriminator: resource=recipient, operation=get
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('recipient'),
      operation: z.literal('get'),
      getBy: z.union([z.literal('id'), z.literal('phone'), z.literal('name'), expressionSchema]).optional(),
      id: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"getBy":["id"]}}, defaults: {"getBy":"id"} }),
      name: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"getBy":["name"]}}, defaults: {"getBy":"id"} }),
      phone: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"getBy":["phone"]}}, defaults: {"getBy":"id"} }),
    }).optional(),
  });
};
