// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/messageBird/v1/resource_sms/operation_send.schema.js
/**
 * MessageBird Node - Version 1 - Zod Schema
 * Discriminator: resource=sms, operation=send
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('sms').default('sms'),
      operation: z.literal('send').default('send'),
      originator: stringOrExpression.optional(),
      recipients: stringOrExpression.optional(),
      message: stringOrExpression.optional(),
      additionalFields: z.object({ createdDatetime: stringOrExpression.optional(), datacoding: z.union([z.literal('auto'), z.literal('plain'), z.literal('unicode'), expressionSchema]).optional(), gateway: numberOrExpression.optional(), groupIds: stringOrExpression.optional(), mclass: z.union([z.literal(1), z.literal(0), expressionSchema]).optional(), reference: stringOrExpression.optional(), reportUrl: stringOrExpression.optional(), scheduledDatetime: stringOrExpression.optional(), type: z.union([z.literal('binary'), z.literal('flash'), z.literal('sms'), expressionSchema]).optional(), typeDetails: stringOrExpression.optional(), validity: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
