// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pushcut/v1/resource_notification/operation_send.ts
/**
 * Pushcut Node - Version 1
 * Discriminator: resource=notification, operation=send
 */


interface Credentials {
  pushcutApi: CredentialReference;
}

/** Send a notification */
export type PushcutV1NotificationSendParams = {
  resource: 'notification';
  operation: 'send';
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    notificationName?: string | Expression<string>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** List of devices this notification is sent to. (default is all devices). Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     * @default []
     */
    devices?: string[];
    /** Value that is passed as input to the notification action
     */
    input?: string | Expression<string> | PlaceholderValue;
    /** Text that is used instead of the one defined in the app
     */
    text?: string | Expression<string> | PlaceholderValue;
    /** Title that is used instead of the one defined in the app
     */
    title?: string | Expression<string> | PlaceholderValue;
  };
};

export type PushcutV1NotificationSendOutput = {
  id?: string;
  message?: string;
  notificationId?: string;
};

export type PushcutV1NotificationSendNode = {
  type: 'n8n-nodes-base.pushcut';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PushcutV1NotificationSendParams>;
  output?: Items<PushcutV1NotificationSendOutput>;
};
