// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/magento2/v1/resource_product/operation_create.schema.js
/**
 * Magento 2 Node - Version 1 - Zod Schema
 * Discriminator: resource=product, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('product'),
      operation: z.literal('create').default('create'),
      sku: stringOrExpression.optional(),
      name: stringOrExpression.optional(),
      attributeSetId: stringOrExpression.optional(),
      price: numberOrExpression.optional(),
      additionalFields: z.object({ attribute_set_id: stringOrExpression.optional(), name: stringOrExpression.optional(), price: numberOrExpression.optional(), status: z.union([z.literal(1), z.literal(2), expressionSchema]).optional(), type_id: stringOrExpression.optional(), visibility: z.union([z.literal(1), z.literal(2), z.literal(3), z.literal(4), expressionSchema]).optional(), weight: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
