// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/magento2/v1/resource_product/operation_get_all.schema.js
/**
 * Magento 2 Node - Version 1 - Zod Schema
 * Discriminator: resource=product, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('product'),
      operation: z.literal('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filterType: z.union([z.literal('none'), z.literal('manual'), z.literal('json'), expressionSchema]).optional(),
      matchType: resolveSchema({ parameters, schema: z.union([z.literal('anyFilter'), z.literal('allFilters'), expressionSchema]), required: false, displayOptions: {"show":{"filterType":["manual"]}}, defaults: {"filterType":"none"} }),
      filters: resolveSchema({ parameters, schema: z.object({ conditions: z.array(z.object({ field: stringOrExpression.optional(), condition_type: z.union([z.literal('eq'), z.literal('gt'), z.literal('gteq'), z.literal('in'), z.literal('lt'), z.literal('lte'), z.literal('like'), z.literal('moreq'), z.literal('neq'), z.literal('nin'), z.literal('notnull'), z.literal('null'), expressionSchema]).optional(), value: stringOrExpression.optional() })).optional() }), required: false, displayOptions: {"show":{"filterType":["manual"]}}, defaults: {"filterType":"none"} }),
      filterJson: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"filterType":["json"]}}, defaults: {"filterType":"none"} }),
      options: z.object({ sort: z.unknown().optional() }).optional(),
    }).optional(),
  });
};
