// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/spotify/v1/resource_playlist/operation_get_tracks.ts
/**
 * Spotify Node - Version 1
 * Discriminator: resource=playlist, operation=getTracks
 */


interface Credentials {
  spotifyOAuth2Api: CredentialReference;
}

/** Get an album's tracks by URI or ID */
export type SpotifyV1PlaylistGetTracksParams = {
  resource: 'playlist';
  operation: 'getTracks';
/**
 * The playlist's Spotify URI or its ID
 */
    id?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
};

export type SpotifyV1PlaylistGetTracksOutput = {
  added_at?: string;
  added_by?: {
    external_urls?: {
      spotify?: string;
    };
    href?: string;
    type?: string;
  };
  is_local?: boolean;
  primary_color?: null;
  track?: {
    album?: {
      album_type?: string;
      artists?: Array<{
        external_urls?: {
          spotify?: string;
        };
        href?: string;
        id?: string;
        name?: string;
        type?: string;
        uri?: string;
      }>;
      available_markets?: Array<string>;
      external_urls?: {
        spotify?: string;
      };
      href?: string;
      id?: string;
      images?: Array<{
        height?: number;
        url?: string;
        width?: number;
      }>;
      name?: string;
      release_date?: string;
      release_date_precision?: string;
      total_tracks?: number;
      type?: string;
      uri?: string;
    };
    artists?: Array<{
      external_urls?: {
        spotify?: string;
      };
      href?: string;
      id?: string;
      name?: string;
      type?: string;
      uri?: string;
    }>;
    available_markets?: Array<string>;
    disc_number?: number;
    duration_ms?: number;
    episode?: boolean;
    explicit?: boolean;
    external_ids?: {
      isrc?: string;
    };
    external_urls?: {
      spotify?: string;
    };
    href?: string;
    id?: string;
    is_local?: boolean;
    name?: string;
    popularity?: number;
    track?: boolean;
    track_number?: number;
    type?: string;
    uri?: string;
  };
  video_thumbnail?: {
    url?: null;
  };
};

export type SpotifyV1PlaylistGetTracksNode = {
  type: 'n8n-nodes-base.spotify';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SpotifyV1PlaylistGetTracksParams>;
  output?: Items<SpotifyV1PlaylistGetTracksOutput>;
};
