// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/slack/v22/resource_channel/operation_create.schema.js
/**
 * Slack Node - Version 2.2 - Zod Schema
 * Discriminator: resource=channel, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('channel'),
      operation: z.literal('create').default('create'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      channelId: stringOrExpression.optional(),
      channelVisibility: z.union([z.literal('public'), z.literal('private'), expressionSchema]).optional(),
    }).optional(),
  });
};
