// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/slack/v1/resource_file/operation_get_all.ts
/**
 * Slack Node - Version 1
 * Discriminator: resource=file, operation=getAll
 */


interface Credentials {
  slackApi: CredentialReference;
  slackOAuth2Api: CredentialReference;
}

/** Get many channels in a Slack team */
export type SlackV1FileGetAllParams = {
  resource: 'file';
  operation: 'getAll';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Channel containing the file to be listed. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    channelId?: string | Expression<string>;
    /** Whether to show truncated file info for files hidden due to being too old, and the team who owns the file being over the file limit
     * @default false
     */
    showFilesHidden?: boolean | Expression<boolean>;
    /** Filter files created after this timestamp (inclusive)
     */
    tsFrom?: string | Expression<string> | PlaceholderValue;
    /** Filter files created before this timestamp (inclusive)
     */
    tsTo?: string | Expression<string> | PlaceholderValue;
    /** Filter files by type
     * @default ["all"]
     */
    types?: Array<'all' | 'gdocs' | 'images' | 'pdfs' | 'snippets' | 'spaces' | 'zips'>;
    /** Filter files created by a single user. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    userId?: string | Expression<string>;
  };
};

export type SlackV1FileGetAllNode = {
  type: 'n8n-nodes-base.slack';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SlackV1FileGetAllParams>;
};
