// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleSheets/v4/resource_sheet/operation_delete.schema.js
/**
 * Google Sheets Node - Version 4 - Zod Schema
 * Discriminator: resource=sheet, operation=delete
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('sheet').default('sheet'),
      operation: z.literal('delete'),
      authentication: z.union([z.literal('serviceAccount'), z.literal('oAuth2'), expressionSchema]).optional(),
      documentId: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('url'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      sheetName: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('url'), z.literal('id'), z.literal('name')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      toDelete: resolveSchema({ parameters, schema: z.union([z.literal('rows'), z.literal('columns'), expressionSchema]), required: false, displayOptions: {"hide":{"sheetName":[""]}}, defaults: {"sheetName":{"mode":"list","value":""}} }),
      startIndex: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"toDelete":["rows","columns"]},"hide":{"sheetName":[""]}}, defaults: {"toDelete":"rows","sheetName":{"mode":"list","value":""}} }),
      numberToDelete: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"toDelete":["rows","columns"]},"hide":{"sheetName":[""]}}, defaults: {"toDelete":"rows","sheetName":{"mode":"list","value":""}} }),
    }).optional(),
  });
};
