// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/slack/v1/resource_user_profile/operation_update.schema.js
/**
 * Slack Node - Version 1 - Zod Schema
 * Discriminator: resource=userProfile, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('userProfile'),
      operation: z.literal('update'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      additionalFields: z.object({ customFieldUi: z.unknown().optional(), email: stringOrExpression.optional(), first_name: stringOrExpression.optional(), last_name: stringOrExpression.optional(), status_emoji: stringOrExpression.optional(), status_expiration: stringOrExpression.optional(), status_text: stringOrExpression.optional(), user: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
