// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/slack/v1/resource_user_profile/operation_update.ts
/**
 * Slack Node - Version 1
 * Discriminator: resource=userProfile, operation=update
 */


interface Credentials {
  slackApi: CredentialReference;
  slackOAuth2Api: CredentialReference;
}

/** Updates a message */
export type SlackV1UserProfileUpdateParams = {
  resource: 'userProfile';
  operation: 'update';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Custom Fields
     * @default {}
     */
    customFieldUi?: {
        /** Custom Field
     */
    customFieldValues?: Array<{
      /** ID of the field to set. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
       */
      id?: string | Expression<string>;
      /** Value of the field to set
       */
      value?: string | Expression<string> | PlaceholderValue;
      /** Alt
       */
      alt?: string | Expression<string> | PlaceholderValue;
    }>;
  };
    /** This field can only be changed by admins for users on paid teams
     */
    email?: string | Expression<string> | PlaceholderValue;
    /** First Name
     */
    first_name?: string | Expression<string> | PlaceholderValue;
    /** Last Name
     */
    last_name?: string | Expression<string> | PlaceholderValue;
    /** Is a string referencing an emoji enabled for the Slack team, such as :mountain_railway:
     */
    status_emoji?: string | Expression<string> | PlaceholderValue;
    /** Is an integer specifying seconds since the epoch, more commonly known as "UNIX time". Providing 0 or omitting this field results in a custom status that will not expire.
     */
    status_expiration?: string | Expression<string>;
    /** Allows up to 100 characters, though we strongly encourage brevity
     */
    status_text?: string | Expression<string> | PlaceholderValue;
    /** ID of user to change. This argument may only be specified by team admins on paid teams.
     */
    user?: string | Expression<string> | PlaceholderValue;
  };
};

export type SlackV1UserProfileUpdateNode = {
  type: 'n8n-nodes-base.slack';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SlackV1UserProfileUpdateParams>;
};
