// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/haloPSA/v1/resource_site/operation_update.ts
/**
 * HaloPSA Node - Version 1
 * Discriminator: resource=site, operation=update
 */


interface Credentials {
  haloPSAApi: CredentialReference;
}

/** Update a client */
export type HaloPSAV1SiteUpdateParams = {
  resource: 'site';
  operation: 'update';
/**
 * Site ID
 */
    siteId?: string | Expression<string> | PlaceholderValue;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** Client ID
     */
    client_id?: string | Expression<string> | PlaceholderValue;
    /** Main Contact
     */
    maincontact_name?: string | Expression<string> | PlaceholderValue;
    /** Enter site name
     */
    name?: string | Expression<string> | PlaceholderValue;
    /** Notes
     */
    notes?: string | Expression<string> | PlaceholderValue;
    /** Phone Number
     */
    phonenumber?: string | Expression<string> | PlaceholderValue;
  };
};

export type HaloPSAV1SiteUpdateNode = {
  type: 'n8n-nodes-base.haloPSA';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<HaloPSAV1SiteUpdateParams>;
};
