// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/quickbooks/v1/resource_estimate/operation_update.schema.js
/**
 * QuickBooks Online Node - Version 1 - Zod Schema
 * Discriminator: resource=estimate, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('estimate'),
      operation: z.literal('update'),
      estimateId: stringOrExpression.optional(),
      updateFields: z.object({ ApplyTaxAfterDiscount: booleanOrExpression.optional(), BillAddr: z.unknown().optional(), BillEmail: stringOrExpression.optional(), CustomFields: z.unknown().optional(), CustomerMemo: stringOrExpression.optional(), DocNumber: stringOrExpression.optional(), EmailStatus: z.union([z.literal('NotSet'), z.literal('NeedToSend'), z.literal('EmailSent'), expressionSchema]).optional(), PrintStatus: z.union([z.literal('NotSet'), z.literal('NeedToPrint'), z.literal('PrintComplete'), expressionSchema]).optional(), ShipAddr: z.unknown().optional(), TxnDate: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
