// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pushbullet/v1/resource_push/operation_get_all.ts
/**
 * Pushbullet Node - Version 1
 * Discriminator: resource=push, operation=getAll
 */


interface Credentials {
  pushbulletOAuth2Api: CredentialReference;
}

/** Get many pushes */
export type PushbulletV1PushGetAllParams = {
  resource: 'push';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Don't return deleted pushes
     * @default false
     */
    active?: boolean | Expression<boolean>;
    /** Request pushes modified after this timestamp
     */
    modified_after?: string | Expression<string>;
  };
};

export type PushbulletV1PushGetAllNode = {
  type: 'n8n-nodes-base.pushbullet';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PushbulletV1PushGetAllParams>;
};
