// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pushbullet/v1/resource_push/operation_create.schema.js
/**
 * Pushbullet Node - Version 1 - Zod Schema
 * Discriminator: resource=push, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('push').default('push'),
      operation: z.literal('create').default('create'),
      type: z.union([z.literal('file'), z.literal('link'), z.literal('note'), expressionSchema]).optional(),
      title: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"type":["note","link"]}}, defaults: {"type":"note"} }),
      body: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"type":["note","link","file"]}}, defaults: {"type":"note"} }),
      url: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"type":["link"]}}, defaults: {"type":"note"} }),
      binaryPropertyName: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"type":["file"]}}, defaults: {"type":"note"} }),
      target: z.union([z.literal('channel_tag'), z.literal('default'), z.literal('device_iden'), z.literal('email'), expressionSchema]).optional(),
      value: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"target":["device_iden"]},"hide":{"target":["default","device_iden"]}}, defaults: {"target":"default"} }),
    }).optional(),
  });
};
