// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/coda/v11/resource_table/operation_create_row.ts
/**
 * Coda Node - Version 1.1
 * Discriminator: resource=table, operation=createRow
 */


interface Credentials {
  codaApi: CredentialReference;
}

/** Access data of tables in documents */
export type CodaV11TableCreateRowParams = {
  resource: 'table';
  operation: 'createRow';
/**
 * ID of the doc. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    docId?: string | Expression<string>;
/**
 * The table to create the row in. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    tableId?: string | Expression<string>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Whether the API will not attempt to parse the data in any way
     * @default false
     */
    disableParsing?: boolean | Expression<boolean>;
    /** Optional column IDs, URLs, or names (fragile and discouraged), specifying columns to be used as upsert keys. If more than one separate by a comma (,).
     */
    keyColumns?: string | Expression<string> | PlaceholderValue;
  };
};

export type CodaV11TableCreateRowOutput = {
  InboundMessage?: string;
  LeadEmail?: string;
  LeadID?: string;
  LeadLinkedInURL?: string;
  LeadName?: string;
  MsgStatus?: string;
};

export type CodaV11TableCreateRowNode = {
  type: 'n8n-nodes-base.coda';
  version: 1.1;
  credentials?: Credentials;
  config: NodeConfig<CodaV11TableCreateRowParams>;
  output?: Items<CodaV11TableCreateRowOutput>;
};
