// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/splunk/v1/resource_search_result/operation_get_all.schema.js
/**
 * Splunk Node - Version 1 - Zod Schema
 * Discriminator: resource=searchResult, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('searchResult'),
      operation: z.literal('getAll'),
      searchJobId: stringOrExpression.optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filters: z.object({ keyValueMatch: z.unknown().optional() }).optional(),
      options: z.object({ add_summary_to_metadata: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
