// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/splunk/v1/resource_search_result/operation_get_all.ts
/**
 * Splunk Node - Version 1
 * Discriminator: resource=searchResult, operation=getAll
 */


interface Credentials {
  splunkApi: CredentialReference;
}

/** Retrieve many search configurations */
export type SplunkV1SearchResultGetAllParams = {
  resource: 'searchResult';
  operation: 'getAll';
/**
 * ID of the search whose results to retrieve
 */
    searchJobId?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Key-value pair to match against. Example: if "Key" is set to &lt;code&gt;user&lt;/code&gt; and "Field" is set to &lt;code&gt;john&lt;/code&gt;, only the results where &lt;code&gt;user&lt;/code&gt; is &lt;code&gt;john&lt;/code&gt; will be returned.
     * @default {}
     */
    keyValueMatch?: {
        /** Key-Value Pair
     */
    keyValuePair?: {
      /** Key to match against
       */
      key?: string | Expression<string> | PlaceholderValue;
      /** Value to match against
       */
      value?: string | Expression<string> | PlaceholderValue;
    };
  };
  };
/**
 * Options
 * @default {}
 */
    options?: {
    /** Whether to include field summary statistics in the response
     * @default false
     */
    add_summary_to_metadata?: boolean | Expression<boolean>;
  };
};

export type SplunkV1SearchResultGetAllNode = {
  type: 'n8n-nodes-base.splunk';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SplunkV1SearchResultGetAllParams>;
};
