// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleCalendar/v11/resource_event/operation_delete.schema.js
/**
 * Google Calendar Node - Version 1.1 - Zod Schema
 * Discriminator: resource=event, operation=delete
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('event').default('event'),
      operation: z.literal('delete'),
      calendar: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      eventId: stringOrExpression.optional(),
      options: z.object({ sendUpdates: z.union([z.literal('all'), z.literal('externalOnly'), z.literal('none'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
