// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/helpScout/v1/resource_conversation/operation_get_all.schema.js
/**
 * Help Scout Node - Version 1 - Zod Schema
 * Discriminator: resource=conversation, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('conversation').default('conversation'),
      operation: z.literal('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      options: z.object({ assignTo: numberOrExpression.optional(), embed: z.union([z.literal('threads'), expressionSchema]).optional(), folder: stringOrExpression.optional(), mailbox: stringOrExpression.optional(), modifiedSince: stringOrExpression.optional(), number: numberOrExpression.optional(), query: stringOrExpression.optional(), sortField: z.union([z.literal('createdAt'), z.literal('customerEmail'), z.literal('customerName'), z.literal('mailboxid'), z.literal('modifiedAt'), z.literal('number'), z.literal('score'), z.literal('status'), z.literal('subject'), expressionSchema]).optional(), sortOrder: z.union([z.literal('asc'), z.literal('desc'), expressionSchema]).optional(), status: z.union([z.literal('active'), z.literal('all'), z.literal('closed'), z.literal('open'), z.literal('pending'), z.literal('spam'), expressionSchema]).optional(), tags: z.array(z.string()).optional() }).optional(),
    }).optional(),
  });
};
