// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/helpScout/v1/resource_conversation/operation_create.schema.js
/**
 * Help Scout Node - Version 1 - Zod Schema
 * Discriminator: resource=conversation, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('conversation').default('conversation'),
      operation: z.literal('create').default('create'),
      mailboxId: stringOrExpression.optional(),
      status: z.union([z.literal('active'), z.literal('closed'), z.literal('pending'), expressionSchema]).optional(),
      subject: stringOrExpression.optional(),
      type: z.union([z.literal('chat'), z.literal('email'), z.literal('phone'), expressionSchema]).optional(),
      resolveData: booleanOrExpression.optional(),
      additionalFields: z.object({ assignTo: numberOrExpression.optional(), autoReply: booleanOrExpression.optional(), closedAt: stringOrExpression.optional(), createdAt: stringOrExpression.optional(), customerEmail: stringOrExpression.optional(), customerId: numberOrExpression.optional(), imported: booleanOrExpression.optional(), tags: z.array(z.string()).optional(), user: numberOrExpression.optional() }).optional(),
      threadsUi: z.object({ threadsValues: z.array(z.object({ type: z.union([z.literal('chat'), z.literal('customer'), z.literal('note'), z.literal('phone'), z.literal('reply'), expressionSchema]).optional(), text: stringOrExpression.optional(), bcc: stringOrExpression.optional(), cc: stringOrExpression.optional(), draft: booleanOrExpression.optional() })).optional() }).optional(),
    }).optional(),
  });
};
