// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/twist/v1/resource_message_conversation/operation_get_all.schema.js
/**
 * Twist Node - Version 1 - Zod Schema
 * Discriminator: resource=messageConversation, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('messageConversation').default('messageConversation'),
      operation: z.literal('getAll'),
      workspaceId: stringOrExpression.optional(),
      conversationId: stringOrExpression.optional(),
      additionalFields: z.object({ to_obj_index: numberOrExpression.optional(), limit: numberOrExpression.optional(), order_by: z.union([z.literal('ASC'), z.literal('DESC'), expressionSchema]).optional(), from_obj_index: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
