// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/twist/v1/resource_message_conversation/operation_get_all.ts
/**
 * Twist Node - Version 1
 * Discriminator: resource=messageConversation, operation=getAll
 */


interface Credentials {
  twistOAuth2Api: CredentialReference;
}

/** Get many channels */
export type TwistV1MessageConversationGetAllParams = {
  resource: 'messageConversation';
  operation: 'getAll';
/**
 * The ID of the workspace. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    workspaceId?: string | Expression<string>;
/**
 * The ID of the conversation. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    conversationId?: string | Expression<string>;
/**
 * Other options to set
 * @default {}
 */
    additionalFields?: {
    /** Limit messages ending at the specified object index
     * @default 50
     */
    to_obj_index?: number | Expression<number>;
    /** Max number of results to return
     * @default 50
     */
    limit?: number | Expression<number>;
    /** The order of the conversations returned - one of DESC or ASC
     * @default ASC
     */
    order_by?: 'ASC' | 'DESC' | Expression<string>;
    /** Limit messages starting at the specified object index
     * @default 0
     */
    from_obj_index?: number | Expression<number>;
  };
};

export type TwistV1MessageConversationGetAllNode = {
  type: 'n8n-nodes-base.twist';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<TwistV1MessageConversationGetAllParams>;
};
