// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/intercom/v1/resource_user/operation_update.schema.js
/**
 * Intercom Node - Version 1 - Zod Schema
 * Discriminator: resource=user, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('user').default('user'),
      operation: z.literal('update'),
      updateBy: z.union([z.literal('id'), z.literal('email'), z.literal('userId'), expressionSchema]).optional(),
      value: stringOrExpression.optional(),
      jsonParameters: booleanOrExpression.optional(),
      additionalFields: z.object({ avatar: stringOrExpression.optional(), companies: z.array(z.string()).optional(), email: stringOrExpression.optional(), name: stringOrExpression.optional(), phone: stringOrExpression.optional(), sessionCount: numberOrExpression.optional(), userId: stringOrExpression.optional(), unsubscribedFromEmails: booleanOrExpression.optional(), updateLastRequestAt: booleanOrExpression.optional(), utmCampaign: stringOrExpression.optional(), utmContent: stringOrExpression.optional(), utmMedium: stringOrExpression.optional(), utmSource: stringOrExpression.optional(), utmTerm: stringOrExpression.optional() }).optional(),
      customAttributesJson: resolveSchema({ parameters, schema: z.union([iDataObjectSchema, z.string()]), required: false, displayOptions: {"show":{"jsonParameters":[true]}}, defaults: {"jsonParameters":false} }),
      customAttributesUi: resolveSchema({ parameters, schema: z.object({ customAttributesValues: z.array(z.object({ name: stringOrExpression.optional(), value: stringOrExpression.optional() })).optional() }), required: false, displayOptions: {"show":{"jsonParameters":[false]}}, defaults: {"jsonParameters":false} }),
    }).optional(),
  });
};
