// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/intercom/v1/resource_user/operation_get_all.schema.js
/**
 * Intercom Node - Version 1 - Zod Schema
 * Discriminator: resource=user, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('user').default('user'),
      operation: z.literal('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filters: z.object({ company_id: stringOrExpression.optional(), email: stringOrExpression.optional(), segment_id: stringOrExpression.optional(), tag_id: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
