// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/harvest/v1/resource_expense/operation_update.ts
/**
 * Harvest Node - Version 1
 * Discriminator: resource=expense, operation=update
 */


interface Credentials {
  harvestApi: CredentialReference;
  harvestOAuth2Api: CredentialReference;
}

/** Update a client */
export type HarvestV1ExpenseUpdateParams = {
  resource: 'expense';
  operation: 'update';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    accountId?: string | Expression<string>;
/**
 * The ID of the invoice want to update
 */
    id?: string | Expression<string> | PlaceholderValue;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** Whether this expense is billable or not. Defaults to true.
     * @default true
     */
    billable?: boolean | Expression<boolean>;
    /** The ID of the expense category this expense is being tracked against
     */
    expense_category_id?: string | Expression<string> | PlaceholderValue;
    /** Notes about the expense
     */
    notes?: string | Expression<string> | PlaceholderValue;
    /** The ID of the project associated with this expense
     */
    project_id?: string | Expression<string> | PlaceholderValue;
    /** Date the expense occurred
     */
    spent_date?: string | Expression<string>;
    /** The total amount of the expense
     */
    total_cost?: string | Expression<string> | PlaceholderValue;
    /** The quantity of units to use in calculating the total_cost of the expense
     */
    units?: string | Expression<string> | PlaceholderValue;
    /** The ID of the user associated with this expense. Defaults to the ID of the currently authenticated user.
     * @default true
     */
    user_id?: boolean | Expression<boolean>;
  };
};

export type HarvestV1ExpenseUpdateNode = {
  type: 'n8n-nodes-base.harvest';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<HarvestV1ExpenseUpdateParams>;
};
