// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/harvest/v1/resource_expense/operation_get_all.ts
/**
 * Harvest Node - Version 1
 * Discriminator: resource=expense, operation=getAll
 */


interface Credentials {
  harvestApi: CredentialReference;
  harvestOAuth2Api: CredentialReference;
}

/** Get data of many clients */
export type HarvestV1ExpenseGetAllParams = {
  resource: 'expense';
  operation: 'getAll';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    accountId?: string | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Only return time entries belonging to the client with the given ID
     */
    client_id?: string | Expression<string> | PlaceholderValue;
    /** Only return time entries with a spent_date on or after the given date
     */
    from?: string | Expression<string>;
    /** Whether to only return time entries that have been invoiced and false to return time entries that have not been invoiced
     * @default false
     */
    is_billed?: boolean | Expression<boolean>;
    /** The page number to use in pagination. For instance, if you make a list request and receive 100 records, your subsequent call can include page=2 to retrieve the next page of the list. (Default: 1)
     * @default 1
     */
    page?: number | Expression<number>;
    /** Only return time entries belonging to the client with the given ID
     */
    project_id?: string | Expression<string> | PlaceholderValue;
    /** Only return time entries with a spent_date on or before the given date
     */
    to?: string | Expression<string>;
    /** Only return time entries that have been updated since the given date and time
     */
    updated_since?: string | Expression<string>;
    /** Only return time entries belonging to the user with the given ID
     */
    user_id?: string | Expression<string> | PlaceholderValue;
  };
};

export type HarvestV1ExpenseGetAllNode = {
  type: 'n8n-nodes-base.harvest';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<HarvestV1ExpenseGetAllParams>;
};
