// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/harvest/v1/resource_expense/operation_update.schema.js
/**
 * Harvest Node - Version 1 - Zod Schema
 * Discriminator: resource=expense, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('expense'),
      operation: z.literal('update'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      accountId: stringOrExpression.optional(),
      id: stringOrExpression.optional(),
      updateFields: z.object({ billable: booleanOrExpression.optional(), expense_category_id: stringOrExpression.optional(), notes: stringOrExpression.optional(), project_id: stringOrExpression.optional(), spent_date: stringOrExpression.optional(), total_cost: stringOrExpression.optional(), units: stringOrExpression.optional(), user_id: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
