// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/ghost/v1/resource_post/operation_get.schema.js
/**
 * Ghost Node - Version 1 - Zod Schema
 * Discriminator: resource=post, operation=get
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('post').default('post'),
      operation: z.literal('get').default('get'),
      source: z.union([z.literal('adminApi'), z.literal('contentApi'), expressionSchema]).optional(),
      by: resolveSchema({ parameters, schema: z.union([z.literal('id'), z.literal('slug'), expressionSchema]), required: false, displayOptions: {"show":{"source":["contentApi","adminApi"]}}, defaults: {"source":"contentApi"} }),
      identifier: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"source":["contentApi","adminApi"]}}, defaults: {"source":"contentApi"} }),
      options: resolveSchema({ parameters, schema: z.object({ fields: stringOrExpression.optional(), formats: z.array(z.union([z.literal('html'), z.literal('mobiledoc'), z.literal('lexical')])).optional() }), required: false, displayOptions: {"show":{"source":["adminApi","contentApi"]}}, defaults: {"source":"contentApi"} }),
    }).optional(),
  });
};
