// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/ghost/v1/resource_post/operation_get_all.schema.js
/**
 * Ghost Node - Version 1 - Zod Schema
 * Discriminator: resource=post, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('post').default('post'),
      operation: z.literal('getAll'),
      source: z.union([z.literal('adminApi'), z.literal('contentApi'), expressionSchema]).optional(),
      returnAll: resolveSchema({ parameters, schema: booleanOrExpression, required: false, displayOptions: {"show":{"source":["contentApi","adminApi"]}}, defaults: {"source":"contentApi"} }),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"source":["adminApi","contentApi"],"returnAll":[false]}}, defaults: {"source":"contentApi","returnAll":false} }),
      options: resolveSchema({ parameters, schema: z.object({ include: z.array(z.union([z.literal('authors'), z.literal('tags')])).optional(), fields: stringOrExpression.optional(), formats: z.array(z.union([z.literal('html'), z.literal('plaintext'), z.literal('lexical')])).optional() }), required: false, displayOptions: {"show":{"source":["contentApi","adminApi"]}}, defaults: {"source":"contentApi"} }),
    }).optional(),
  });
};
