// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/apiTemplateIo/v1/resource_image/operation_create.schema.js
/**
 * APITemplate.io Node - Version 1 - Zod Schema
 * Discriminator: resource=image, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('image').default('image'),
      operation: z.literal('create').default('create'),
      imageTemplateId: stringOrExpression.optional(),
      jsonParameters: booleanOrExpression.optional(),
      download: booleanOrExpression.optional(),
      binaryProperty: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"download":[true]}}, defaults: {"download":false} }),
      overridesJson: resolveSchema({ parameters, schema: z.union([iDataObjectSchema, z.string()]), required: false, displayOptions: {"show":{"jsonParameters":[true]}}, defaults: {"jsonParameters":false} }),
      overridesUi: resolveSchema({ parameters, schema: z.object({ overrideValues: z.array(z.object({ propertiesUi: z.unknown().optional() })).optional() }), required: false, displayOptions: {"show":{"jsonParameters":[false]}}, defaults: {"jsonParameters":false} }),
      options: resolveSchema({ parameters, schema: z.object({ fileName: stringOrExpression.optional() }), required: false, displayOptions: {"show":{"download":[true]}}, defaults: {"download":false} }),
    }).optional(),
  });
};
