// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/monicaCrm/v1/resource_activity/operation_update.ts
/**
 * Monica CRM Node - Version 1
 * Discriminator: resource=activity, operation=update
 */


interface Credentials {
  monicaCrmApi: CredentialReference;
}

/** Update an activity */
export type MonicaCrmV1ActivityUpdateParams = {
  resource: 'activity';
  operation: 'update';
/**
 * ID of the activity to update
 */
    activityId?: string | Expression<string> | PlaceholderValue;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     */
    activity_type_id?: string | Expression<string>;
    /** IDs of the contacts to associate the activity with
     */
    contacts?: string | Expression<string> | PlaceholderValue;
    /** Description to add more details on the activity - max 100,000 characters
     */
    description?: string | Expression<string> | PlaceholderValue;
    /** Date when the activity happened
     */
    happened_at?: string | Expression<string>;
    /** Brief description of the activity - max 255 characters
     */
    summary?: string | Expression<string> | PlaceholderValue;
  };
};

export type MonicaCrmV1ActivityUpdateNode = {
  type: 'n8n-nodes-base.monicaCrm';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MonicaCrmV1ActivityUpdateParams>;
};
