// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/monicaCrm/v1/resource_activity/operation_create.ts
/**
 * Monica CRM Node - Version 1
 * Discriminator: resource=activity, operation=create
 */


interface Credentials {
  monicaCrmApi: CredentialReference;
}

/** Create an activity */
export type MonicaCrmV1ActivityCreateParams = {
  resource: 'activity';
  operation: 'create';
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    activityTypeId?: string | Expression<string>;
/**
 * Comma-separated list of IDs of the contacts to associate the activity with
 */
    contacts?: string | Expression<string> | PlaceholderValue;
/**
 * Date when the activity happened
 */
    happenedAt?: string | Expression<string>;
/**
 * Brief description of the activity - max 255 characters
 */
    summary?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Description of the activity - max 100,000 characters
     */
    description?: string | Expression<string> | PlaceholderValue;
  };
};

export type MonicaCrmV1ActivityCreateNode = {
  type: 'n8n-nodes-base.monicaCrm';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MonicaCrmV1ActivityCreateParams>;
};
