// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/monicaCrm/v1/resource_activity/operation_update.schema.js
/**
 * Monica CRM Node - Version 1 - Zod Schema
 * Discriminator: resource=activity, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('activity'),
      operation: z.literal('update'),
      activityId: stringOrExpression.optional(),
      updateFields: z.object({ activity_type_id: stringOrExpression.optional(), contacts: stringOrExpression.optional(), description: stringOrExpression.optional(), happened_at: stringOrExpression.optional(), summary: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
