// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/demio/v1/resource_event/operation_get_all.schema.js
/**
 * Demio Node - Version 1 - Zod Schema
 * Discriminator: resource=event, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('event').default('event'),
      operation: z.literal('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filters: z.object({ type: z.union([z.literal('automated'), z.literal('past'), z.literal('upcoming'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
