// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/demio/v1/resource_event/operation_register.schema.js
/**
 * Demio Node - Version 1 - Zod Schema
 * Discriminator: resource=event, operation=register
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('event').default('event'),
      operation: z.literal('register'),
      eventId: stringOrExpression.optional(),
      firstName: stringOrExpression.optional(),
      email: stringOrExpression.optional(),
      additionalFields: z.object({ company: stringOrExpression.optional(), customFieldsUi: z.unknown().optional(), ref_url: stringOrExpression.optional(), gdpr: stringOrExpression.optional(), last_name: stringOrExpression.optional(), phone_number: stringOrExpression.optional(), date_id: stringOrExpression.optional(), website: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
