// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/zohoCrm/v1/resource_sales_order/operation_get_all.ts
/**
 * Zoho CRM Node - Version 1
 * Discriminator: resource=salesOrder, operation=getAll
 */


interface Credentials {
  zohoOAuth2Api: CredentialReference;
}

/** Get many accounts */
export type ZohoCrmV1SalesOrderGetAllParams = {
  resource: 'salesOrder';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 5
 */
    limit?: number | Expression<number>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Whether to retrieve only approved records. Defaults to true.
     * @default true
     */
    approved?: boolean | Expression<boolean>;
    /** Whether to retrieve only converted records. Defaults to false.
     * @default false
     */
    converted?: boolean | Expression<boolean>;
    /** Return only these fields
     * @default []
     */
    fields?: string[];
    /** Whether to retrieve only records from child territories
     * @default false
     */
    include_child?: boolean | Expression<boolean>;
    /** Field to sort records by
     * @default []
     */
    sort_by?: string | Expression<string>;
    /** Ascending or descending order sort order
     * @default desc
     */
    sort_order?: 'asc' | 'desc' | Expression<string>;
    /** Retrieve only records from this territory
     */
    territory_id?: string | Expression<string> | PlaceholderValue;
  };
};

export type ZohoCrmV1SalesOrderGetAllNode = {
  type: 'n8n-nodes-base.zohoCrm';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ZohoCrmV1SalesOrderGetAllParams>;
};
