// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/elasticsearch/v1/resource_document/operation_create.schema.js
/**
 * Elasticsearch Node - Version 1 - Zod Schema
 * Discriminator: resource=document, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('document').default('document'),
      operation: z.literal('create'),
      indexId: stringOrExpression.optional(),
      dataToSend: z.union([z.literal('defineBelow'), z.literal('autoMapInputData'), expressionSchema]).optional(),
      inputsToIgnore: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"dataToSend":["autoMapInputData"]}}, defaults: {"dataToSend":"defineBelow"} }),
      fieldsUi: resolveSchema({ parameters, schema: z.object({ fieldValues: z.array(z.object({ fieldId: stringOrExpression.optional(), fieldValue: stringOrExpression.optional() })).optional() }), required: false, displayOptions: {"show":{"dataToSend":["defineBelow"]}}, defaults: {"dataToSend":"defineBelow"} }),
      additionalFields: z.object({ documentId: stringOrExpression.optional(), routing: stringOrExpression.optional(), timeout: stringOrExpression.optional() }).optional(),
      options: z.object({ bulkOperation: booleanOrExpression.optional(), pipeline: stringOrExpression.optional(), refresh: z.union([z.literal('true'), z.literal('wait_for'), z.literal('false'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
