// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/elasticsearch/v1/resource_document/operation_get.schema.js
/**
 * Elasticsearch Node - Version 1 - Zod Schema
 * Discriminator: resource=document, operation=get
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('document').default('document'),
      operation: z.literal('get').default('get'),
      indexId: stringOrExpression.optional(),
      documentId: stringOrExpression.optional(),
      simple: booleanOrExpression.optional(),
      options: z.object({ _source_excludes: stringOrExpression.optional(), _source_includes: stringOrExpression.optional(), stored_fields: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
