// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/elasticsearch/v1/resource_document/operation_get.ts
/**
 * Elasticsearch Node - Version 1
 * Discriminator: resource=document, operation=get
 */


interface Credentials {
  elasticsearchApi: CredentialReference;
}

/** Get a document */
export type ElasticsearchV1DocumentGetParams = {
  resource: 'document';
  operation: 'get';
/**
 * ID of the index containing the document to retrieve
 */
    indexId?: string | Expression<string> | PlaceholderValue;
/**
 * ID of the document to retrieve
 */
    documentId?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to return a simplified version of the response instead of the raw data
 * @default true
 */
    simple?: boolean | Expression<boolean>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Comma-separated list of source fields to exclude from the response
     */
    _source_excludes?: string | Expression<string> | PlaceholderValue;
    /** Comma-separated list of source fields to include in the response
     */
    _source_includes?: string | Expression<string> | PlaceholderValue;
    /** Whether to retrieve the document fields stored in the index rather than the document &lt;code&gt;_source&lt;/code&gt;. Defaults to false.
     * @default false
     */
    stored_fields?: boolean | Expression<boolean>;
  };
};

export type ElasticsearchV1DocumentGetNode = {
  type: 'n8n-nodes-base.elasticsearch';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ElasticsearchV1DocumentGetParams>;
};
