// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/wooCommerce/v1/resource_product/operation_get_all.ts
/**
 * WooCommerce Node - Version 1
 * Discriminator: resource=product, operation=getAll
 */


interface Credentials {
  wooCommerceApi: CredentialReference;
}

/** Retrieve many customers */
export type WooCommerceV1ProductGetAllParams = {
  resource: 'product';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Limit response to resources published after a given ISO8601 compliant date
     */
    after?: string | Expression<string>;
    /** Limit response to resources published before a given ISO8601 compliant date
     */
    before?: string | Expression<string>;
    /** Limit result set to products assigned a specific category ID. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    category?: string | Expression<string>;
    /** Scope under which the request is made; determines fields present in response
     * @default view
     */
    context?: 'view' | 'embed' | 'edit' | Expression<string>;
    /** Whether to limit the result set to featured products
     * @default false
     */
    featured?: boolean | Expression<boolean>;
    /** Limit result set to products based on a maximun price
     */
    maxPrice?: string | Expression<string> | PlaceholderValue;
    /** Limit result set to products based on a minimum price
     */
    minPrice?: string | Expression<string> | PlaceholderValue;
    /** Order sort attribute ascending or descending
     * @default desc
     */
    order?: 'asc' | 'desc' | Expression<string>;
    /** Sort collection by object attribute
     * @default id
     */
    orderBy?: 'date' | 'id' | 'include' | 'slug' | 'title' | Expression<string>;
    /** Limit results to those matching a string
     */
    search?: string | Expression<string> | PlaceholderValue;
    /** Limit result set to products with a specific SKU
     */
    sku?: string | Expression<string> | PlaceholderValue;
    /** Limit result set to products with a specific slug
     */
    slug?: string | Expression<string> | PlaceholderValue;
    /** Limit result set to products assigned a specific status
     * @default any
     */
    status?: 'any' | 'draft' | 'pending' | 'private' | 'publish' | Expression<string>;
    /** Controls the stock status of the product
     */
    stockStatus?: 'instock' | 'outofstock' | 'onbackorder' | Expression<string>;
    /** Limit result set to products assigned a specific tag ID. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     * @default []
     */
    tag?: string | Expression<string>;
    /** Limit result set to products with a specific tax class
     */
    taxClass?: 'standard' | 'reduced-rate' | 'zero-rate.' | Expression<string>;
    /** Product type
     * @default simple
     */
    type?: 'simple' | 'grouped' | 'external' | 'variable' | Expression<string>;
  };
};

export type WooCommerceV1ProductGetAllOutput = {
  _links?: {
    collection?: Array<{
      href?: string;
    }>;
    self?: Array<{
      href?: string;
      targetHints?: {
        allow?: Array<string>;
      };
    }>;
  };
  attributes?: Array<{
    id?: number;
    name?: string;
    options?: Array<string>;
    position?: number;
    slug?: string;
    variation?: boolean;
    visible?: boolean;
  }>;
  average_rating?: string;
  backordered?: boolean;
  backorders?: string;
  backorders_allowed?: boolean;
  brands?: Array<{
    id?: number;
    name?: string;
    slug?: string;
  }>;
  button_text?: string;
  catalog_visibility?: string;
  categories?: Array<{
    id?: number;
    name?: string;
    slug?: string;
  }>;
  cross_sell_ids?: Array<number>;
  date_modified?: string;
  date_modified_gmt?: string;
  date_on_sale_from?: null;
  date_on_sale_from_gmt?: null;
  date_on_sale_to?: null;
  date_on_sale_to_gmt?: null;
  default_attributes?: Array<{
    id?: number;
    name?: string;
    option?: string;
  }>;
  description?: string;
  dimensions?: {
    height?: string;
    length?: string;
    width?: string;
  };
  download_expiry?: number;
  download_limit?: number;
  downloadable?: boolean;
  downloads?: Array<{
    file?: string;
    id?: string;
    name?: string;
  }>;
  external_url?: string;
  featured?: boolean;
  global_unique_id?: string;
  grouped_products?: Array<number>;
  has_options?: boolean;
  id?: number;
  images?: Array<{
    alt?: string;
    date_created?: string;
    date_created_gmt?: string;
    date_modified?: string;
    date_modified_gmt?: string;
    id?: number;
    name?: string;
    src?: string;
  }>;
  manage_stock?: boolean;
  menu_order?: number;
  meta_data?: Array<{
    id?: number;
    key?: string;
  }>;
  name?: string;
  on_sale?: boolean;
  parent_id?: number;
  permalink?: string;
  post_password?: string;
  price_html?: string;
  purchasable?: boolean;
  purchase_note?: string;
  rating_count?: number;
  related_ids?: Array<number>;
  reviews_allowed?: boolean;
  shipping_class?: string;
  shipping_class_id?: number;
  shipping_required?: boolean;
  shipping_taxable?: boolean;
  short_description?: string;
  sku?: string;
  slug?: string;
  sold_individually?: boolean;
  status?: string;
  stock_status?: string;
  tags?: Array<{
    id?: number;
    name?: string;
    slug?: string;
  }>;
  tax_class?: string;
  tax_status?: string;
  type?: string;
  upsell_ids?: Array<number>;
  variations?: Array<number>;
  virtual?: boolean;
  weight?: string;
};

export type WooCommerceV1ProductGetAllNode = {
  type: 'n8n-nodes-base.wooCommerce';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<WooCommerceV1ProductGetAllParams>;
  output?: Items<WooCommerceV1ProductGetAllOutput>;
};
