// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/wooCommerce/v1/resource_product/operation_get.ts
/**
 * WooCommerce Node - Version 1
 * Discriminator: resource=product, operation=get
 */


interface Credentials {
  wooCommerceApi: CredentialReference;
}

/** Retrieve a customer */
export type WooCommerceV1ProductGetParams = {
  resource: 'product';
  operation: 'get';
/**
 * Product ID
 */
    productId?: string | Expression<string> | PlaceholderValue;
};

export type WooCommerceV1ProductGetOutput = {
  _links?: {
    collection?: Array<{
      href?: string;
    }>;
    self?: Array<{
      href?: string;
      targetHints?: {
        allow?: Array<string>;
      };
    }>;
  };
  attributes?: Array<{
    id?: number;
    name?: string;
    options?: Array<string>;
    position?: number;
    slug?: string;
    variation?: boolean;
    visible?: boolean;
  }>;
  average_rating?: string;
  backordered?: boolean;
  backorders?: string;
  backorders_allowed?: boolean;
  brands?: Array<{
    id?: number;
    name?: string;
    slug?: string;
  }>;
  button_text?: string;
  catalog_visibility?: string;
  categories?: Array<{
    id?: number;
    name?: string;
    slug?: string;
  }>;
  cross_sell_ids?: Array<number>;
  date_modified?: string;
  date_modified_gmt?: string;
  default_attributes?: Array<{
    id?: number;
    name?: string;
    option?: string;
  }>;
  description?: string;
  dimensions?: {
    height?: string;
    length?: string;
    width?: string;
  };
  download_expiry?: number;
  download_limit?: number;
  downloadable?: boolean;
  downloads?: Array<{
    file?: string;
    id?: string;
    name?: string;
  }>;
  external_url?: string;
  featured?: boolean;
  global_unique_id?: string;
  has_options?: boolean;
  id?: number;
  images?: Array<{
    alt?: string;
    date_created?: string;
    date_created_gmt?: string;
    date_modified?: string;
    date_modified_gmt?: string;
    id?: number;
    name?: string;
    src?: string;
  }>;
  manage_stock?: boolean;
  menu_order?: number;
  meta_data?: Array<{
    id?: number;
    key?: string;
  }>;
  name?: string;
  on_sale?: boolean;
  parent_id?: number;
  permalink?: string;
  post_password?: string;
  price_html?: string;
  purchasable?: boolean;
  purchase_note?: string;
  rating_count?: number;
  related_ids?: Array<number>;
  reviews_allowed?: boolean;
  sale_price?: string;
  shipping_class?: string;
  shipping_class_id?: number;
  shipping_required?: boolean;
  shipping_taxable?: boolean;
  short_description?: string;
  sku?: string;
  slug?: string;
  sold_individually?: boolean;
  status?: string;
  stock_status?: string;
  tags?: Array<{
    id?: number;
    name?: string;
    slug?: string;
  }>;
  tax_class?: string;
  tax_status?: string;
  type?: string;
  upsell_ids?: Array<number>;
  variations?: Array<number>;
  virtual?: boolean;
  weight?: string;
};

export type WooCommerceV1ProductGetNode = {
  type: 'n8n-nodes-base.wooCommerce';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<WooCommerceV1ProductGetParams>;
  output?: Items<WooCommerceV1ProductGetOutput>;
};
