// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/wooCommerce/v1/resource_product/operation_get_all.schema.js
/**
 * WooCommerce Node - Version 1 - Zod Schema
 * Discriminator: resource=product, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('product').default('product'),
      operation: z.literal('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      options: z.object({ after: stringOrExpression.optional(), before: stringOrExpression.optional(), category: stringOrExpression.optional(), context: z.union([z.literal('view'), z.literal('embed'), z.literal('edit'), expressionSchema]).optional(), featured: booleanOrExpression.optional(), maxPrice: stringOrExpression.optional(), minPrice: stringOrExpression.optional(), order: z.union([z.literal('asc'), z.literal('desc'), expressionSchema]).optional(), orderBy: z.union([z.literal('date'), z.literal('id'), z.literal('include'), z.literal('slug'), z.literal('title'), expressionSchema]).optional(), search: stringOrExpression.optional(), sku: stringOrExpression.optional(), slug: stringOrExpression.optional(), status: z.union([z.literal('any'), z.literal('draft'), z.literal('pending'), z.literal('private'), z.literal('publish'), expressionSchema]).optional(), stockStatus: z.union([z.literal('instock'), z.literal('outofstock'), z.literal('onbackorder'), expressionSchema]).optional(), tag: stringOrExpression.optional(), taxClass: z.union([z.literal('standard'), z.literal('reduced-rate'), z.literal('zero-rate.'), expressionSchema]).optional(), type: z.union([z.literal('simple'), z.literal('grouped'), z.literal('external'), z.literal('variable'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
