// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/venafiTlsProtectCloud/v1/resource_certificate_request/operation_create.schema.js
/**
 * Venafi TLS Protect Cloud Node - Version 1 - Zod Schema
 * Discriminator: resource=certificateRequest, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('certificateRequest').default('certificateRequest'),
      operation: z.literal('create'),
      applicationId: stringOrExpression.optional(),
      certificateIssuingTemplateId: stringOrExpression.optional(),
      generateCsr: booleanOrExpression.optional(),
      commonName: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"generateCsr":[true]}}, defaults: {"generateCsr":false} }),
      additionalFields: resolveSchema({ parameters, schema: z.object({ keyType: z.union([z.literal('EC'), z.literal('RSA'), expressionSchema]).optional(), keyCurve: z.union([z.literal('ED25519'), z.literal('P256'), z.literal('P384'), z.literal('P521'), z.literal('UNKNOWN'), expressionSchema]).optional(), keyLength: numberOrExpression.optional(), organization: stringOrExpression.optional(), organizationalUnits: stringOrExpression.optional(), locality: stringOrExpression.optional(), state: stringOrExpression.optional(), country: stringOrExpression.optional(), SubjectAltNamesUi: z.unknown().optional() }), required: false, displayOptions: {"show":{"generateCsr":[true]}}, defaults: {"generateCsr":false} }),
      certificateSigningRequest: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"generateCsr":[false]}}, defaults: {"generateCsr":false} }),
      options: z.object({ validityPeriod: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
