// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/venafiTlsProtectCloud/v1/resource_certificate_request/operation_create.ts
/**
 * Venafi TLS Protect Cloud Node - Version 1
 * Discriminator: resource=certificateRequest, operation=create
 */


interface Credentials {
  venafiTlsProtectCloudApi: CredentialReference;
}

/** Create a new certificate request */
export type VenafiTlsProtectCloudV1CertificateRequestCreateParams = {
  resource: 'certificateRequest';
  operation: 'create';
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    applicationId?: string | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    certificateIssuingTemplateId?: string | Expression<string>;
/**
 * Generate CSR
 * @default false
 */
    generateCsr?: boolean | Expression<boolean>;
/**
 * The Common Name field for the certificate Subject (CN)
 * @displayOptions.show { generateCsr: [true] }
 * @default n8n.io
 */
    commonName?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @displayOptions.show { generateCsr: [true] }
 * @default {}
 */
    additionalFields?: {
    /** The encryption algorithm for the public key
     * @default RSA
     */
    keyType?: 'EC' | 'RSA' | Expression<string>;
    /** Key Curve
     * @default ED25519
     */
    keyCurve?: 'ED25519' | 'P256' | 'P384' | 'P521' | 'UNKNOWN' | Expression<string>;
    /** The number of bits to allow for key generation
     * @default 2048
     */
    keyLength?: number | Expression<number>;
    /** The name of a company or organization
     */
    organization?: string | Expression<string> | PlaceholderValue;
    /** The name of a department or section
     */
    organizationalUnits?: string | Expression<string> | PlaceholderValue;
    /** The name of a city or town
     */
    locality?: string | Expression<string> | PlaceholderValue;
    /** The name of a state or province
     */
    state?: string | Expression<string> | PlaceholderValue;
    /** A 2 letter country code
     */
    country?: string | Expression<string> | PlaceholderValue;
    /** Subject Alt Names
     * @default {}
     */
    SubjectAltNamesUi?: {
        /** Subject Alt Name
     */
    SubjectAltNamesValues?: Array<{
      /** What type of SAN is being used
       * @default dnsNames
       */
      Typename?: 'dnsNames' | Expression<string>;
      /** The SAN friendly name that corresponds to the Type or TypeName parameter. For example, if a TypeName is IPAddress, the Name value is a valid IP address.
       * @default community.n8n.io
       */
      name?: string | Expression<string> | PlaceholderValue;
    }>;
  };
  };
/**
 * Certificate Signing Request
 * @displayOptions.show { generateCsr: [false] }
 */
    certificateSigningRequest?: string | Expression<string> | PlaceholderValue;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Specify how long the issued certificate should be valid for. Use ISO8601 format.
     * @hint e.g. 1 year -&gt; P1Y
     * @default P1Y
     */
    validityPeriod?: string | Expression<string> | PlaceholderValue;
  };
};

export type VenafiTlsProtectCloudV1CertificateRequestCreateNode = {
  type: 'n8n-nodes-base.venafiTlsProtectCloud';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<VenafiTlsProtectCloudV1CertificateRequestCreateParams>;
};
