// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/invoiceNinja/v1/resource_quote/operation_get_all.ts
/**
 * Invoice Ninja Node - Version 1
 * Discriminator: resource=quote, operation=getAll
 */


interface Credentials {
  invoiceNinjaApi: CredentialReference;
}

/** Get data of many clients */
export type InvoiceNinjaV1QuoteGetAllParams = {
  resource: 'quote';
  operation: 'getAll';
/**
 * API Version
 * @default v4
 */
    apiVersion?: 'v4' | 'v5' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Quote Number
     */
    quoteNumber?: string | Expression<string> | PlaceholderValue;
    /** Include
     * @default client
     */
    include?: 'client' | Expression<string>;
    /** Status
     * @default active
     */
    status?: 'active' | 'archived' | 'deleted' | Expression<string>;
    /** Created At
     */
    createdAt?: string | Expression<string>;
    /** Updated At
     */
    updatedAt?: string | Expression<string>;
    /** Is Deleted
     * @default false
     */
    isDeleted?: boolean | Expression<boolean>;
  };
};

export type InvoiceNinjaV1QuoteGetAllNode = {
  type: 'n8n-nodes-base.invoiceNinja';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<InvoiceNinjaV1QuoteGetAllParams>;
};
