// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/github/v11/resource_issue/operation_get.ts
/**
 * GitHub Node - Version 1.1
 * Discriminator: resource=issue, operation=get
 */


interface Credentials {
  githubApi: CredentialReference;
  githubOAuth2Api: CredentialReference;
}

/** Get the data of a single issue */
export type GithubV11IssueGetParams = {
  resource: 'issue';
  operation: 'get';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Repository Owner
 * @displayOptions.hide { operation: ["invite", "getUserIssues"] }
 * @default {"mode":"list","value":""}
 */
    owner?: { __rl: true; mode: 'list' | 'url' | 'name'; value: string; cachedResultName?: string };
/**
 * Repository Name
 * @displayOptions.hide { resource: ["user", "organization"], operation: ["getRepositories"] }
 * @default {"mode":"list","value":""}
 */
    repository?: { __rl: true; mode: 'list' | 'url' | 'name'; value: string; cachedResultName?: string };
/**
 * The issue number to get data for
 * @default 0
 */
    issueNumber?: number | Expression<number>;
};

export type GithubV11IssueGetOutput = {
  active_lock_reason?: null;
  assignee?: {
    avatar_url?: string;
    events_url?: string;
    followers_url?: string;
    following_url?: string;
    gists_url?: string;
    gravatar_id?: string;
    html_url?: string;
    id?: number;
    login?: string;
    node_id?: string;
    organizations_url?: string;
    received_events_url?: string;
    repos_url?: string;
    site_admin?: boolean;
    starred_url?: string;
    subscriptions_url?: string;
    type?: string;
    url?: string;
    user_view_type?: string;
  };
  assignees?: Array<{
    avatar_url?: string;
    events_url?: string;
    followers_url?: string;
    following_url?: string;
    gists_url?: string;
    gravatar_id?: string;
    html_url?: string;
    id?: number;
    login?: string;
    node_id?: string;
    organizations_url?: string;
    received_events_url?: string;
    repos_url?: string;
    site_admin?: boolean;
    starred_url?: string;
    subscriptions_url?: string;
    type?: string;
    url?: string;
    user_view_type?: string;
  }>;
  author_association?: string;
  comments?: number;
  comments_url?: string;
  created_at?: string;
  events_url?: string;
  html_url?: string;
  id?: number;
  labels?: Array<{
    color?: string;
    'default'?: boolean;
    id?: number;
    name?: string;
    node_id?: string;
    url?: string;
  }>;
  labels_url?: string;
  locked?: boolean;
  node_id?: string;
  number?: number;
  performed_via_github_app?: null;
  reactions?: {
    '-1'?: number;
    '+1'?: number;
    confused?: number;
    eyes?: number;
    heart?: number;
    hooray?: number;
    laugh?: number;
    rocket?: number;
    total_count?: number;
    url?: string;
  };
  repository_url?: string;
  state?: string;
  timeline_url?: string;
  title?: string;
  updated_at?: string;
  url?: string;
  user?: {
    avatar_url?: string;
    events_url?: string;
    followers_url?: string;
    following_url?: string;
    gists_url?: string;
    gravatar_id?: string;
    html_url?: string;
    id?: number;
    login?: string;
    node_id?: string;
    organizations_url?: string;
    received_events_url?: string;
    repos_url?: string;
    site_admin?: boolean;
    starred_url?: string;
    subscriptions_url?: string;
    type?: string;
    url?: string;
    user_view_type?: string;
  };
};

export type GithubV11IssueGetNode = {
  type: 'n8n-nodes-base.github';
  version: 1.1;
  credentials?: Credentials;
  config: NodeConfig<GithubV11IssueGetParams>;
  output?: Items<GithubV11IssueGetOutput>;
};
