// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleBusinessProfile/v1/resource_review/operation_reply.ts
/**
 * Google Business Profile Node - Version 1
 * Discriminator: resource=review, operation=reply
 */


interface Credentials {
  googleBusinessProfileOAuth2Api: CredentialReference;
}

/** Reply to a review */
export type GoogleBusinessProfileV1ReviewReplyParams = {
  resource: 'review';
  operation: 'reply';
/**
 * The Google Business Profile account
 * @default {"mode":"list","value":""}
 */
    account?: { __rl: true; mode: 'list' | 'name'; value: string; cachedResultName?: string };
/**
 * The specific location or business associated with the account
 * @default {"mode":"list","value":""}
 */
    location?: { __rl: true; mode: 'list' | 'name'; value: string; cachedResultName?: string };
/**
 * Select the review to retrieve its details
 * @default {"mode":"list","value":""}
 */
    review?: { __rl: true; mode: 'list' | 'id' | 'name'; value: string; cachedResultName?: string };
/**
 * The body of the reply (up to 4096 characters)
 */
    reply?: string | Expression<string> | PlaceholderValue;
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type GoogleBusinessProfileV1ReviewReplyOutput = {
  comment?: string;
  updateTime?: string;
};

export type GoogleBusinessProfileV1ReviewReplyNode = {
  type: 'n8n-nodes-base.googleBusinessProfile';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GoogleBusinessProfileV1ReviewReplyParams>;
  output?: Items<GoogleBusinessProfileV1ReviewReplyOutput>;
};
