// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/deepL/v1/resource_language/operation_translate.schema.js
/**
 * DeepL Node - Version 1 - Zod Schema
 * Discriminator: resource=language, operation=translate
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('language').default('language'),
      operation: z.literal('translate').default('translate'),
      text: stringOrExpression.optional(),
      translateTo: stringOrExpression.optional(),
      additionalFields: z.object({ sourceLang: stringOrExpression.optional(), splitSentences: z.union([z.literal('nonewlines'), z.literal('0'), z.literal('1'), expressionSchema]).optional(), preserveFormatting: z.union([z.literal('0'), z.literal('1'), expressionSchema]).optional(), formality: z.union([z.literal('more'), z.literal('less'), z.literal('default'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
