// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/keap/v1/resource_ecommerce_product/operation_create.schema.js
/**
 * Keap Node - Version 1 - Zod Schema
 * Discriminator: resource=ecommerceProduct, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('ecommerceProduct'),
      operation: z.literal('create').default('create'),
      productName: stringOrExpression.optional(),
      additionalFields: z.object({ active: booleanOrExpression.optional(), productDesc: stringOrExpression.optional(), productPrice: numberOrExpression.optional(), productShortDesc: stringOrExpression.optional(), sku: stringOrExpression.optional(), subscriptionOnly: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
