// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/nextCloud/v1/resource_file/operation_upload.ts
/**
 * Nextcloud Node - Version 1
 * Discriminator: resource=file, operation=upload
 */


interface Credentials {
  nextCloudApi: CredentialReference;
  nextCloudOAuth2Api: CredentialReference;
}

/** Upload a file */
export type NextCloudV1FileUploadParams = {
  resource: 'file';
  operation: 'upload';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * The absolute file path of the file to upload. Has to contain the full path. The parent folder has to exist. Existing files get overwritten.
 */
    path?: string | Expression<string> | PlaceholderValue;
/**
 * Binary File
 * @default false
 */
    binaryDataUpload?: boolean | Expression<boolean>;
/**
 * The text content of the file to upload
 * @displayOptions.show { binaryDataUpload: [false] }
 */
    fileContent?: string | Expression<string> | PlaceholderValue;
/**
 * Input Binary Field
 * @hint The name of the input binary field containing the file to be uploaded
 * @displayOptions.show { binaryDataUpload: [true] }
 * @default data
 */
    binaryPropertyName?: string | Expression<string> | PlaceholderValue;
};

export type NextCloudV1FileUploadNode = {
  type: 'n8n-nodes-base.nextCloud';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<NextCloudV1FileUploadParams>;
};
