// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/nextCloud/v1/resource_file/operation_share.schema.js
/**
 * Nextcloud Node - Version 1 - Zod Schema
 * Discriminator: resource=file, operation=share
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('file').default('file'),
      operation: z.literal('share'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      path: stringOrExpression.optional(),
      shareType: z.union([z.literal(7), z.literal(4), z.literal(1), z.literal(3), z.literal(0), expressionSchema]).optional(),
      circleId: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"shareType":[7]}}, defaults: {"shareType":0} }),
      email: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"shareType":[4]}}, defaults: {"shareType":0} }),
      groupId: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"shareType":[1]}}, defaults: {"shareType":0} }),
      user: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"shareType":[0]}}, defaults: {"shareType":0} }),
      options: z.object({ password: stringOrExpression.optional(), permissions: z.union([z.literal(31), z.literal(4), z.literal(8), z.literal(1), z.literal(2), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
