// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/awsSes/v1/resource_custom_verification_email/operation_update.ts
/**
 * AWS SES Node - Version 1
 * Discriminator: resource=customVerificationEmail, operation=update
 */


interface Credentials {
  aws: CredentialReference;
  awsAssumeRole: CredentialReference;
}

/** Update an existing custom verification email template */
export type AwsSesV1CustomVerificationEmailUpdateParams = {
  resource: 'customVerificationEmail';
  operation: 'update';
  authentication?: 'iam' | 'assumeRole' | Expression<string>;
/**
 * The name of the custom verification email template
 */
    templateName?: string | Expression<string> | PlaceholderValue;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** The URL that the recipient of the verification email is sent to if his or her address is not successfully verified
     */
    failureRedirectionURL?: string | Expression<string> | PlaceholderValue;
    /** The email address that the custom verification email is sent from
     */
    fromEmailAddress?: string | Expression<string> | PlaceholderValue;
    /** The URL that the recipient of the verification email is sent to if his or her address is successfully verified
     */
    successRedirectionURL?: string | Expression<string> | PlaceholderValue;
    /** The content of the custom verification email. The total size of the email must be less than 10 MB. The message body may contain HTML
     */
    templateContent?: string | Expression<string> | PlaceholderValue;
    /** The subject line of the custom verification email
     */
    templateSubject?: string | Expression<string> | PlaceholderValue;
  };
};

export type AwsSesV1CustomVerificationEmailUpdateNode = {
  type: 'n8n-nodes-base.awsSes';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<AwsSesV1CustomVerificationEmailUpdateParams>;
};
