// @n8n/n8n-nodes-langchain/dist/nodes/vector_store/ChatHubVectorStorePGVector/ChatHubVectorStorePGVector.node.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChatHubVectorStorePGVector = void 0;
const pgvector_1 = require("@langchain/community/vectorstores/pgvector");
const ai_utilities_1 = require("@n8n/ai-utilities");
const index_1 = require("n8n-nodes-base/dist/nodes/Postgres/transport/index");
const credentialTest_1 = require("n8n-nodes-base/dist/nodes/Postgres/v2/methods/credentialTest");
const n8n_workflow_1 = require("n8n-workflow");
const userScoped_1 = require("../shared/userScoped");
const VectorStorePGVector_node_1 = require("../VectorStorePGVector/VectorStorePGVector.node");
const retrieveFields = [
    {
        displayName: 'Options',
        name: 'options',
        type: 'collection',
        placeholder: 'Add Option',
        default: {},
        options: [ai_utilities_1.metadataFilterField],
    },
];
async function deleteDocuments(payload) {
    const { filter } = (typeof payload === 'string' ? (0, n8n_workflow_1.jsonParse)(payload) : (payload ?? {}));
    const credentials = await this.getCredentials('chatHubVectorStorePGVectorApi');
    const tableName = (0, userScoped_1.getUserScopedSlot)(this, credentials.tableNamePrefix);
    const pgConf = await index_1.configurePostgres.call(this, credentials);
    const pool = pgConf.db.$pool;
    if (!filter || Object.keys(filter).length === 0) {
        await pool.query(`DROP TABLE IF EXISTS "${tableName}"`);
        return null;
    }
    const conditions = [];
    const values = [];
    let paramIndex = 1;
    for (const [key, value] of Object.entries(filter)) {
        if (Array.isArray(value)) {
            conditions.push(`metadata->>($${paramIndex}::text) = ANY($${paramIndex + 1}::text[])`);
        }
        else {
            conditions.push(`metadata->>($${paramIndex}::text) = $${paramIndex + 1}`);
        }
        values.push(key, value);
        paramIndex += 2;
    }
    await pool.query(`DELETE FROM "${tableName}" WHERE ${conditions.join(' AND ')}`, values);
    return null;
}
async function chatHubVectorStorePGVectorApiConnectionTest(credential) {
    const credentialWithSsh = {
        ...credential,
        data: { ...(credential.data ?? {}), sshTunnel: false },
    };
    const connectionResult = await credentialTest_1.postgresConnectionTest.call(this, credentialWithSsh);
    if (connectionResult.status === 'Error') {
        return connectionResult;
    }
    const credentials = credential.data;
    try {
        const pgConf = await index_1.configurePostgres.call(this, credentials);
        const pool = pgConf.db.$pool;
        const result = await pool.query("SELECT EXISTS(SELECT 1 FROM pg_extension WHERE extname = 'vector') AS exists");
        const extensionExists = result.rows[0]?.exists;
        if (!extensionExists) {
            return {
                status: 'Error',
                message: 'The pgvector extension is not enabled. Please install and enable the pgvector extension in your PostgreSQL database.',
            };
        }
    }
    catch (error) {
        return {
            status: 'Error',
            message: error.message,
        };
    }
    return connectionResult;
}
class ChatHubVectorStorePGVector extends (0, ai_utilities_1.createVectorStoreNode)({
    hidden: true,
    methods: {
        credentialTest: { chatHubVectorStorePGVectorApiConnectionTest },
        actionHandler: { deleteDocuments },
    },
    meta: {
        description: 'Internal-use vector store for ChatHub',
        icon: 'file:../VectorStorePGVector/postgres.svg',
        displayName: 'ChatHub PGVector Store',
        docsUrl: 'https://docs.n8n.io',
        name: 'chatHubVectorStorePGVector',
        credentials: [
            {
                name: 'chatHubVectorStorePGVectorApi',
                required: true,
                testedBy: 'chatHubVectorStorePGVectorApiConnectionTest',
            },
        ],
        operationModes: ['load', 'insert', 'retrieve', 'retrieve-as-tool'],
    },
    sharedFields: [],
    insertFields: [],
    loadFields: retrieveFields,
    retrieveFields,
    async getVectorStoreClient(context, filter, embeddings, itemIndex) {
        const credentials = await context.getCredentials('chatHubVectorStorePGVectorApi');
        const tableName = (0, userScoped_1.getUserScopedSlot)(context, credentials.tableNamePrefix, itemIndex);
        const pgConf = await index_1.configurePostgres.call(context, credentials);
        const pool = pgConf.db.$pool;
        const config = {
            pool,
            tableName,
            filter,
        };
        return await VectorStorePGVector_node_1.ExtendedPGVectorStore.initialize(embeddings, config);
    },
    async populateVectorStore(context, embeddings, documents, itemIndex) {
        const credentials = await context.getCredentials('chatHubVectorStorePGVectorApi');
        const tableName = (0, userScoped_1.getUserScopedSlot)(context, credentials.tableNamePrefix, itemIndex);
        const pgConf = await index_1.configurePostgres.call(context, credentials);
        const pool = pgConf.db.$pool;
        const config = {
            pool,
            tableName,
        };
        const vectorStore = await pgvector_1.PGVectorStore.fromDocuments(documents, embeddings, config);
        vectorStore.client?.release();
    },
    releaseVectorStoreClient(vectorStore) {
        vectorStore.client?.release();
    },
}) {
}
exports.ChatHubVectorStorePGVector = ChatHubVectorStorePGVector;
