// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedrive/v1/resource_organization/operation_update.ts
/**
 * Pipedrive Node - Version 1
 * Discriminator: resource=organization, operation=update
 */


interface Credentials {
  pipedriveApi: CredentialReference;
  pipedriveOAuth2Api: CredentialReference;
}

/** Update an activity */
export type PipedriveV1OrganizationUpdateParams = {
  resource: 'organization';
  operation: 'update';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * The ID of the organization to create
 */
    organizationId?: number | Expression<number>;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** Adds a custom property to set also values which have not been predefined
     * @default {}
     */
    customProperties?: {
        /** Property
     */
    property?: Array<{
      /** Name of the property to set
       */
      name?: string | Expression<string> | PlaceholderValue;
      /** Value of the property to set
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
    /** Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     */
    label?: string | Expression<string>;
    /** Organization name
     */
    name?: string | Expression<string> | PlaceholderValue;
    /** The ID of the user who will be marked as the owner of this Organization. When omitted, the authorized User ID will be used.
     * @default 0
     */
    owner_id?: number | Expression<number>;
    /** Visibility of the person. If omitted, visibility will be set to the default visibility setting of this item type for the authorized user.
     * @default 3
     */
    visible_to?: '1' | '3' | Expression<string>;
  };
/**
 * By default do custom properties have to be set as ID instead of their actual name. Also option fields have to be set as ID instead of their actual value. If this option gets set they get automatically encoded.
 * @default false
 */
    encodeProperties?: boolean | Expression<boolean>;
};

export type PipedriveV1OrganizationUpdateOutput = {
  active_flag?: boolean;
  activities_count?: number;
  add_time?: string;
  annual_revenue?: null;
  cc_email?: string;
  closed_deals_count?: number;
  country_code?: null;
  delete_time?: null;
  done_activities_count?: number;
  email_messages_count?: number;
  employee_count?: null;
  files_count?: number;
  first_char?: string;
  followers_count?: number;
  id?: number;
  industry?: null;
  label_ids?: Array<number>;
  linkedin?: null;
  lost_deals_count?: number;
  name?: string;
  notes_count?: number;
  open_deals_count?: number;
  owner_id?: {
    active_flag?: boolean;
    email?: string;
    has_pic?: number;
    id?: number;
    name?: string;
    value?: number;
  };
  owner_name?: string;
  people_count?: number;
  picture_id?: null;
  related_closed_deals_count?: number;
  related_lost_deals_count?: number;
  related_open_deals_count?: number;
  related_won_deals_count?: number;
  undone_activities_count?: number;
  update_time?: string;
  visible_to?: string;
  website?: null;
  won_deals_count?: number;
};

export type PipedriveV1OrganizationUpdateNode = {
  type: 'n8n-nodes-base.pipedrive';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PipedriveV1OrganizationUpdateParams>;
  output?: Items<PipedriveV1OrganizationUpdateOutput>;
};
