// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedrive/v1/resource_organization/operation_create.ts
/**
 * Pipedrive Node - Version 1
 * Discriminator: resource=organization, operation=create
 */


interface Credentials {
  pipedriveApi: CredentialReference;
  pipedriveOAuth2Api: CredentialReference;
}

/** Create an activity */
export type PipedriveV1OrganizationCreateParams = {
  resource: 'organization';
  operation: 'create';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * The name of the organization to create
 */
    name?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Adds a custom property to set also values which have not been predefined
     * @default {}
     */
    customProperties?: {
        /** Property
     */
    property?: Array<{
      /** Name of the property to set
       */
      name?: string | Expression<string> | PlaceholderValue;
      /** Value of the property to set
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
    /** Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     */
    label?: string | Expression<string>;
    /** Visibility of the person. If omitted, visibility will be set to the default visibility setting of this item type for the authorized user.
     * @default 3
     */
    visible_to?: '1' | '3' | Expression<string>;
  };
};

export type PipedriveV1OrganizationCreateOutput = {
  active_flag?: boolean;
  activities_count?: number;
  add_time?: string;
  address_admin_area_level_1?: null;
  address_admin_area_level_2?: null;
  address_country?: null;
  address_formatted_address?: null;
  address_route?: null;
  address_sublocality?: null;
  address_subpremise?: null;
  cc_email?: string;
  closed_deals_count?: number;
  country_code?: null;
  delete_time?: null;
  done_activities_count?: number;
  edit_name?: boolean;
  email_messages_count?: number;
  files_count?: number;
  first_char?: string;
  followers_count?: number;
  id?: number;
  label_ids?: Array<number>;
  last_activity_date?: null;
  last_activity_id?: null;
  lost_deals_count?: number;
  name?: string;
  next_activity_date?: null;
  next_activity_id?: null;
  next_activity_time?: null;
  notes_count?: number;
  open_deals_count?: number;
  owner_id?: {
    active_flag?: boolean;
    email?: string;
    has_pic?: number;
    id?: number;
    name?: string;
    value?: number;
  };
  owner_name?: string;
  people_count?: number;
  picture_id?: null;
  related_closed_deals_count?: number;
  related_lost_deals_count?: number;
  related_open_deals_count?: number;
  related_won_deals_count?: number;
  undone_activities_count?: number;
  update_time?: string;
  visible_to?: string;
  won_deals_count?: number;
};

export type PipedriveV1OrganizationCreateNode = {
  type: 'n8n-nodes-base.pipedrive';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PipedriveV1OrganizationCreateParams>;
  output?: Items<PipedriveV1OrganizationCreateOutput>;
};
