// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedrive/v1/resource_organization/operation_get_all.ts
/**
 * Pipedrive Node - Version 1
 * Discriminator: resource=organization, operation=getAll
 */


interface Credentials {
  pipedriveApi: CredentialReference;
  pipedriveOAuth2Api: CredentialReference;
}

/** Get data of many activities */
export type PipedriveV1OrganizationGetAllParams = {
  resource: 'organization';
  operation: 'getAll';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * By default do custom properties get returned only as ID instead of their actual name. Also option fields contain only the ID instead of their actual value. If this option gets set they get automatically resolved.
 * @default false
 */
    resolveProperties?: boolean | Expression<boolean>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** If supplied, only organizations whose name starts with the specified letter will be returned
     */
    firstChar?: string | Expression<string> | PlaceholderValue;
    /** ID of the filter to use. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    filterId?: string | Expression<string>;
  };
};

export type PipedriveV1OrganizationGetAllOutput = {
  active_flag?: boolean;
  activities_count?: number;
  add_time?: string;
  cc_email?: string;
  closed_deals_count?: number;
  company_id?: number;
  country_code?: null;
  delete_time?: null;
  done_activities_count?: number;
  email_messages_count?: number;
  files_count?: number;
  first_char?: string;
  followers_count?: number;
  id?: number;
  label_ids?: Array<number>;
  lost_deals_count?: number;
  name?: string;
  notes_count?: number;
  open_deals_count?: number;
  owner_id?: {
    active_flag?: boolean;
    email?: string;
    has_pic?: number;
    id?: number;
    name?: string;
    value?: number;
  };
  owner_name?: string;
  people_count?: number;
  picture_id?: null;
  related_closed_deals_count?: number;
  related_lost_deals_count?: number;
  related_open_deals_count?: number;
  related_won_deals_count?: number;
  undone_activities_count?: number;
  update_time?: string;
  visible_to?: string;
  won_deals_count?: number;
};

export type PipedriveV1OrganizationGetAllNode = {
  type: 'n8n-nodes-base.pipedrive';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PipedriveV1OrganizationGetAllParams>;
  output?: Items<PipedriveV1OrganizationGetAllOutput>;
};
