// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedrive/v1/resource_organization/operation_get.ts
/**
 * Pipedrive Node - Version 1
 * Discriminator: resource=organization, operation=get
 */


interface Credentials {
  pipedriveApi: CredentialReference;
  pipedriveOAuth2Api: CredentialReference;
}

/** Get data of an activity */
export type PipedriveV1OrganizationGetParams = {
  resource: 'organization';
  operation: 'get';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * ID of the organization to get
 * @default 0
 */
    organizationId?: number | Expression<number>;
/**
 * By default do custom properties get returned only as ID instead of their actual name. Also option fields contain only the ID instead of their actual value. If this option gets set they get automatically resolved.
 * @default false
 */
    resolveProperties?: boolean | Expression<boolean>;
};

export type PipedriveV1OrganizationGetOutput = {
  active_flag?: boolean;
  activities_count?: number;
  add_time?: string;
  category_id?: null;
  cc_email?: string;
  closed_deals_count?: number;
  country_code?: null;
  done_activities_count?: number;
  edit_name?: boolean;
  email_messages_count?: number;
  files_count?: number;
  first_char?: string;
  followers_count?: number;
  id?: number;
  label_ids?: Array<number>;
  lost_deals_count?: number;
  name?: string;
  next_activity_time?: null;
  notes_count?: number;
  open_deals_count?: number;
  owner_id?: {
    active_flag?: boolean;
    email?: string;
    has_pic?: number;
    id?: number;
    name?: string;
    value?: number;
  };
  owner_name?: string;
  people_count?: number;
  picture_id?: null;
  related_closed_deals_count?: number;
  related_lost_deals_count?: number;
  related_open_deals_count?: number;
  related_won_deals_count?: number;
  undone_activities_count?: number;
  update_time?: string;
  visible_to?: string;
  won_deals_count?: number;
};

export type PipedriveV1OrganizationGetNode = {
  type: 'n8n-nodes-base.pipedrive';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PipedriveV1OrganizationGetParams>;
  output?: Items<PipedriveV1OrganizationGetOutput>;
};
