// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedrive/v1/resource_organization/operation_update.schema.js
/**
 * Pipedrive Node - Version 1 - Zod Schema
 * Discriminator: resource=organization, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('organization'),
      operation: z.literal('update'),
      authentication: z.union([z.literal('apiToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      organizationId: numberOrExpression.optional(),
      updateFields: z.object({ customProperties: z.unknown().optional(), label: stringOrExpression.optional(), name: stringOrExpression.optional(), owner_id: numberOrExpression.optional(), visible_to: z.union([z.literal('1'), z.literal('3'), expressionSchema]).optional() }).optional(),
      encodeProperties: booleanOrExpression.optional(),
    }).optional(),
  });
};
