// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedriveTrigger/v11.schema.js
/**
 * Pipedrive Trigger Node - Version 1.1 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    authentication: z.union([z.literal('apiToken'), z.literal('oAuth2'), expressionSchema]).optional(),
    incomingAuthentication: z.union([z.literal('basicAuth'), z.literal('none'), expressionSchema]).optional(),
    action: z.union([z.literal('*'), z.literal('create'), z.literal('delete'), z.literal('change'), expressionSchema]).optional(),
    entity: z.union([z.literal('activity'), z.literal('activityType'), z.literal('*'), z.literal('deal'), z.literal('note'), z.literal('organization'), z.literal('person'), z.literal('pipeline'), z.literal('product'), z.literal('stage'), z.literal('user'), expressionSchema]).optional(),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
