// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/freshworksCrm/v1/resource_search/operation_query.schema.js
/**
 * Freshworks CRM Node - Version 1 - Zod Schema
 * Discriminator: resource=search, operation=query
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('search'),
      operation: z.literal('query'),
      query: stringOrExpression.optional(),
      entities: z.array(z.union([z.literal('contact'), z.literal('deal'), z.literal('sales_account'), z.literal('user')])).optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
    }).optional(),
  });
};
