// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/freshworksCrm/v1/resource_search/operation_lookup.schema.js
/**
 * Freshworks CRM Node - Version 1 - Zod Schema
 * Discriminator: resource=search, operation=lookup
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('search'),
      operation: z.literal('lookup'),
      searchField: z.union([z.literal('email'), z.literal('name'), z.literal('customField'), expressionSchema]).optional(),
      customFieldName: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"searchField":["customField"]}}, defaults: {"searchField":""} }),
      customFieldValue: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"searchField":["customField"]}}, defaults: {"searchField":""} }),
      fieldValue: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"searchField":["email","name"]}}, defaults: {"searchField":""} }),
      options: z.object({ entities: z.array(z.union([z.literal('contact'), z.literal('deal'), z.literal('sales_account')])).optional() }).optional(),
    }).optional(),
  });
};
