// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/taiga/v1/resource_user_story/operation_get_all.ts
/**
 * Taiga Node - Version 1
 * Discriminator: resource=userStory, operation=getAll
 */


interface Credentials {
  taigaApi: CredentialReference;
}

/** Get many epics */
export type TaigaV1UserStoryGetAllParams = {
  resource: 'userStory';
  operation: 'getAll';
/**
 * ID of the project to which the user story belongs. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    projectId?: string | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** ID of the user whom the user story is assigned to. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    assigned_to?: string | Expression<string>;
    /** ID of the epic to which the user story belongs. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    epic?: string | Expression<string>;
    /** Whether the user story is closed
     * @default false
     */
    statusIsClosed?: boolean | Expression<boolean>;
    /** Whether the user story has been archived
     * @default false
     */
    statusIsArchived?: boolean | Expression<boolean>;
    /** ID of the milestone of the user story. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    milestone?: string | Expression<string>;
    /** Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     */
    role?: string | Expression<string>;
    /** ID of the status of the user story. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    status?: string | Expression<string>;
    /** Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     * @default []
     */
    tags?: string[];
  };
};

export type TaigaV1UserStoryGetAllNode = {
  type: 'n8n-nodes-base.taiga';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<TaigaV1UserStoryGetAllParams>;
};
