// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/taiga/v1/resource_user_story/operation_get_all.schema.js
/**
 * Taiga Node - Version 1 - Zod Schema
 * Discriminator: resource=userStory, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('userStory'),
      operation: z.literal('getAll'),
      projectId: stringOrExpression.optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filters: z.object({ assigned_to: stringOrExpression.optional(), epic: stringOrExpression.optional(), statusIsClosed: booleanOrExpression.optional(), statusIsArchived: booleanOrExpression.optional(), milestone: stringOrExpression.optional(), role: stringOrExpression.optional(), status: stringOrExpression.optional(), tags: z.array(z.string()).optional() }).optional(),
    }).optional(),
  });
};
