// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/dropbox/v1/resource_file/operation_move.ts
/**
 * Dropbox Node - Version 1
 * Discriminator: resource=file, operation=move
 */


interface Credentials {
  dropboxApi: CredentialReference;
  dropboxOAuth2Api: CredentialReference;
}

/** Move a file */
export type DropboxV1FileMoveParams = {
  resource: 'file';
  operation: 'move';
/**
 * Means of authenticating with the service
 * @default accessToken
 */
    authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * The path of file or folder to move
 */
    path?: string | Expression<string> | PlaceholderValue;
/**
 * The new path of file or folder
 */
    toPath?: string | Expression<string> | PlaceholderValue;
};

export type DropboxV1FileMoveOutput = {
  metadata?: {
    '.tag'?: string;
    client_modified?: string;
    content_hash?: string;
    id?: string;
    is_downloadable?: boolean;
    name?: string;
    path_display?: string;
    path_lower?: string;
    rev?: string;
    server_modified?: string;
    size?: number;
  };
};

export type DropboxV1FileMoveNode = {
  type: 'n8n-nodes-base.dropbox';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<DropboxV1FileMoveParams>;
  output?: Items<DropboxV1FileMoveOutput>;
};
