// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/dropbox/v1/resource_file/operation_delete.ts
/**
 * Dropbox Node - Version 1
 * Discriminator: resource=file, operation=delete
 */


interface Credentials {
  dropboxApi: CredentialReference;
  dropboxOAuth2Api: CredentialReference;
}

/** Delete a file */
export type DropboxV1FileDeleteParams = {
  resource: 'file';
  operation: 'delete';
/**
 * Means of authenticating with the service
 * @default accessToken
 */
    authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * The path to delete. Can be a single file or a whole folder.
 */
    path?: string | Expression<string> | PlaceholderValue;
};

export type DropboxV1FileDeleteOutput = {
  metadata?: {
    '.tag'?: string;
    client_modified?: string;
    content_hash?: string;
    id?: string;
    is_downloadable?: boolean;
    name?: string;
    path_display?: string;
    path_lower?: string;
    rev?: string;
    server_modified?: string;
    size?: number;
  };
};

export type DropboxV1FileDeleteNode = {
  type: 'n8n-nodes-base.dropbox';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<DropboxV1FileDeleteParams>;
  output?: Items<DropboxV1FileDeleteOutput>;
};
