// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftOutlook/v1/resource_folder/operation_get.ts
/**
 * Microsoft Outlook Node - Version 1
 * Discriminator: resource=folder, operation=get
 */


interface Credentials {
  microsoftOutlookOAuth2Api: CredentialReference;
}

/** Get a single draft */
export type MicrosoftOutlookV1FolderGetParams = {
  resource: 'folder';
  operation: 'get';
/**
 * Folder ID
 */
    folderId?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Fields the response will contain. Multiple can be added separated by ,.
     */
    fields?: string | Expression<string> | PlaceholderValue;
    /** Microsoft Graph API OData $filter query. Information about the syntax can be found &lt;a href="https://docs.microsoft.com/en-us/graph/query-parameters#filter-parameter"&gt;here&lt;/a&gt;.
     */
    filter?: string | Expression<string> | PlaceholderValue;
  };
};

export type MicrosoftOutlookV1FolderGetNode = {
  type: 'n8n-nodes-base.microsoftOutlook';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MicrosoftOutlookV1FolderGetParams>;
};
