// @microsoft/agents-a365-observability/dist/esm/tracing/exporter/Agent365ExporterOptions.js
// ------------------------------------------------------------------------------
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
// ------------------------------------------------------------------------------
/**
 * Options controlling the behavior of the Agent365 OpenTelemetry span exporter.
 *
 * These values tune batching, timeouts, token acquisition and endpoint shape. All properties have sensible
 * defaults so callers can usually construct without arguments and override selectively.
 *
 * @property {ClusterCategory | string} clusterCategory Environment / cluster category (e.g. "preprod", "prod", default to "prod").
 * @property {TokenResolver} [tokenResolver] Optional delegate to obtain an auth token. If omitted the exporter will
 *           fall back to reading the cached token (AgenticTokenCacheInstance.getObservabilityToken).
 * @property {number} maxQueueSize Maximum span queue size before drops occur (passed to BatchSpanProcessor).
 * @property {number} scheduledDelayMilliseconds Delay between automatic batch flush attempts.
 * @property {number} exporterTimeoutMilliseconds Per-export timeout (abort if exceeded).
 * @property {number} maxExportBatchSize Maximum number of spans per export batch.
 */
export class Agent365ExporterOptions {
    constructor() {
        /** Environment / cluster category (e.g. "preprod", "prod"). */
        this.clusterCategory = 'prod';
        /** Maximum span queue size before new spans are dropped. */
        this.maxQueueSize = 2048;
        /** Delay (ms) between automatic batch flush attempts. */
        this.scheduledDelayMilliseconds = 5000;
        /** Per-export timeout in milliseconds. */
        this.exporterTimeoutMilliseconds = 30000;
        /** Maximum number of spans per export batch. */
        this.maxExportBatchSize = 512;
    }
}
