// @microsoft/agents-a365-observability/dist/esm/tracing/exporter/utils.js
// ------------------------------------------------------------------------------
// Copyright (c) Microsoft Corporation. All rights reserved.
// ------------------------------------------------------------------------------
import { SpanKind, SpanStatusCode } from '@opentelemetry/api';
import { OpenTelemetryConstants } from '../constants';
import logger from '../../utils/logging';
/**
 * Convert trace ID to hex string format
 */
export function hexTraceId(value) {
    if (typeof value === 'number') {
        // Convert integer to 32 hex chars (128-bit)
        return value.toString(16).padStart(32, '0');
    }
    // Handle hex string input - ensure it's 32 hex characters
    return value.replace(/^0x/, '').padStart(32, '0');
}
/**
 * Convert span ID to hex string format
 */
export function hexSpanId(value) {
    if (typeof value === 'number') {
        // Convert integer to 16 hex chars (64-bit)
        return value.toString(16).padStart(16, '0');
    }
    // Handle hex string input - ensure it's 16 hex characters
    return value.replace(/^0x/, '').padStart(16, '0');
}
/**
 * Convert any value to string, handling null/undefined
 */
export function asStr(v) {
    if (v === null || v === undefined) {
        return undefined;
    }
    const s = String(v);
    return s.trim() ? s : undefined;
}
/**
 * Get span kind name from SpanKind enum
 */
export function kindName(kind) {
    switch (kind) {
        case SpanKind.INTERNAL:
            return 'INTERNAL';
        case SpanKind.SERVER:
            return 'SERVER';
        case SpanKind.CLIENT:
            return 'CLIENT';
        case SpanKind.PRODUCER:
            return 'PRODUCER';
        case SpanKind.CONSUMER:
            return 'CONSUMER';
        default:
            return 'UNSPECIFIED';
    }
}
/**
 * Get status name from SpanStatusCode enum
 */
export function statusName(code) {
    switch (code) {
        case SpanStatusCode.UNSET:
            return 'UNSET';
        case SpanStatusCode.OK:
            return 'OK';
        case SpanStatusCode.ERROR:
            return 'ERROR';
        default:
            return 'UNSET';
    }
}
/**
 * Partition spans by (tenantId, agentId) identity pairs
 */
export function partitionByIdentity(spans) {
    const groups = new Map();
    let skippedCount = 0;
    for (const span of spans) {
        const attrs = span.attributes || {};
        const tenant = asStr(attrs[OpenTelemetryConstants.TENANT_ID_KEY]);
        const agent = asStr(attrs[OpenTelemetryConstants.GEN_AI_AGENT_ID_KEY]);
        if (!tenant || !agent) {
            skippedCount++;
            logger.warn(`[Agent365Exporter] Skipping span without tenant or agent ID. Span name: ${span.name}`);
            continue;
        }
        const key = `${tenant}:${agent}`;
        const existing = groups.get(key) || [];
        existing.push(span);
        groups.set(key, existing);
    }
    logger.info(`[Agent365Exporter] Partitioned into ${groups.size} identity groups (${skippedCount} spans skipped)`);
    return groups;
}
/**
 * Check if Agent 365 exporter is enabled via environment variable
 */
export function isAgent365ExporterEnabled() {
    const a365Env = process.env[OpenTelemetryConstants.ENABLE_A365_OBSERVABILITY_EXPORTER]?.toLowerCase() || '';
    const validValues = ['true', '1', 'yes', 'on'];
    const enabled = validValues.includes(a365Env);
    logger.info(`[Agent365Exporter] Agent 365 exporter enabled: ${enabled}`);
    return enabled;
}
/**
 * Parse identity key back to tenant and agent IDs
 */
export function parseIdentityKey(key) {
    const [tenantId, agentId] = key.split(':');
    return { tenantId, agentId };
}
