// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pagerDuty/v1/resource_log_entry/operation_get_all.ts
/**
 * PagerDuty Node - Version 1
 * Discriminator: resource=logEntry, operation=getAll
 */


interface Credentials {
  pagerDutyApi: CredentialReference;
  pagerDutyOAuth2Api: CredentialReference;
}

/** Get many incidents */
export type PagerDutyV1LogEntryGetAllParams = {
  resource: 'logEntry';
  operation: 'getAll';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Additional details to include
     * @default []
     */
    include?: Array<'channels' | 'incidents' | 'services' | 'teams'>;
    /** Whether to return a subset of log entries that show only the most important changes to the incident
     * @default false
     */
    isOverview?: boolean | Expression<boolean>;
    /** The start of the date range over which you want to search. (the limit on date ranges is 6 months).
     */
    since?: string | Expression<string>;
    /** Time zone in which dates in the result will be rendered. If not set dates will return UTC. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    timeZone?: string | Expression<string>;
    /** The end of the date range over which you want to search. (the limit on date ranges is 6 months).
     */
    until?: string | Expression<string>;
  };
};

export type PagerDutyV1LogEntryGetAllNode = {
  type: 'n8n-nodes-base.pagerDuty';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PagerDutyV1LogEntryGetAllParams>;
};
