// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mailjet/v1/resource_sms/operation_send.ts
/**
 * Mailjet Node - Version 1
 * Discriminator: resource=sms, operation=send
 */


interface Credentials {
  mailjetEmailApi: CredentialReference;
  mailjetSmsApi: CredentialReference;
}

/** Send a email */
export type MailjetV1SmsSendParams = {
  resource: 'sms';
  operation: 'send';
/**
 * Subject line of the email
 */
    subject?: string | Expression<string> | PlaceholderValue;
/**
 * Customizable sender name. Should be between 3 and 11 characters in length, only alphanumeric characters are allowed.
 */
    from?: string | Expression<string> | PlaceholderValue;
/**
 * Message recipient. Should be between 3 and 15 characters in length. The number always starts with a plus sign followed by a country code, followed by the number. Phone numbers are expected to comply with the E.164 format.
 */
    to?: string | Expression<string> | PlaceholderValue;
/**
 * Text
 */
    text?: string | Expression<string> | PlaceholderValue;
};

export type MailjetV1SmsSendNode = {
  type: 'n8n-nodes-base.mailjet';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MailjetV1SmsSendParams>;
};
