// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/gong/v1/resource_user/operation_get.ts
/**
 * Gong Node - Version 1
 * Discriminator: resource=user, operation=get
 */


interface Credentials {
  gongApi: CredentialReference;
  gongOAuth2Api: CredentialReference;
}

/** Retrieve data for a specific call */
export type GongV1UserGetParams = {
  resource: 'user';
  operation: 'get';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * User to Get
 * @default {"mode":"list","value":""}
 */
    user?: { __rl: true; mode: 'list' | 'id'; value: string; cachedResultName?: string };
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type GongV1UserGetOutput = {
  active?: boolean;
  created?: string;
  emailAddress?: string;
  emailAliases?: Array<string>;
  extension?: null;
  firstName?: string;
  id?: string;
  lastName?: string;
  settings?: {
    emailsImported?: boolean;
    gongConnectEnabled?: boolean;
    nonRecordedMeetingsImported?: boolean;
    preventEmailImport?: boolean;
    preventWebConferenceRecording?: boolean;
    telephonyCallsImported?: boolean;
    webConferencesRecorded?: boolean;
  };
  spokenLanguages?: Array<{
    language?: string;
    primary?: boolean;
  }>;
  trustedEmailAddress?: null;
};

export type GongV1UserGetNode = {
  type: 'n8n-nodes-base.gong';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GongV1UserGetParams>;
  output?: Items<GongV1UserGetOutput>;
};
