// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/gong/v1/resource_user/operation_get_all.ts
/**
 * Gong Node - Version 1
 * Discriminator: resource=user, operation=getAll
 */


interface Credentials {
  gongApi: CredentialReference;
  gongOAuth2Api: CredentialReference;
}

/** Retrieve a list of calls */
export type GongV1UserGetAllParams = {
  resource: 'user';
  operation: 'getAll';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** An optional user creation time lower limit, if supplied the API will return only the users created at or after this time
     */
    createdFromDateTime?: string | Expression<string>;
    /** An optional user creation time upper limit, if supplied the API will return only the users created before this time
     */
    createdToDateTime?: string | Expression<string>;
    /** Set of Gong's unique numeric identifiers for the users (up to 20 digits)
     * @hint Comma separated list of IDs, array of strings can be set in expression
     */
    userIds?: string | Expression<string> | PlaceholderValue;
  };
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type GongV1UserGetAllNode = {
  type: 'n8n-nodes-base.gong';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GongV1UserGetAllParams>;
};
