// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/gSuiteAdmin/v1/resource_user/operation_add_to_group.ts
/**
 * Google Workspace Admin Node - Version 1
 * Discriminator: resource=user, operation=addToGroup
 */


interface Credentials {
  gSuiteAdminOAuth2Api: CredentialReference;
}

/** Add an existing user to a group */
export type GSuiteAdminV1UserAddToGroupParams = {
  resource: 'user';
  operation: 'addToGroup';
/**
 * Select the user to perform the operation on
 * @default {"mode":"list","value":""}
 */
    userId?: { __rl: true; mode: 'list' | 'userEmail' | 'userId'; value: string; cachedResultName?: string };
/**
 * Select the group to perform the operation on
 * @default {"mode":"list","value":""}
 */
    groupId?: { __rl: true; mode: 'list' | 'groupId'; value: string; cachedResultName?: string };
};

export type GSuiteAdminV1UserAddToGroupOutput = {
  added?: boolean;
};

export type GSuiteAdminV1UserAddToGroupNode = {
  type: 'n8n-nodes-base.gSuiteAdmin';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GSuiteAdminV1UserAddToGroupParams>;
  output?: Items<GSuiteAdminV1UserAddToGroupOutput>;
};
