// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/gSuiteAdmin/v1/resource_user/operation_get_all.schema.js
/**
 * Google Workspace Admin Node - Version 1 - Zod Schema
 * Discriminator: resource=user, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('user').default('user'),
      operation: z.literal('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      output: z.union([z.literal('simplified'), z.literal('raw'), z.literal('select'), expressionSchema]).optional(),
      fields: resolveSchema({ parameters, schema: z.array(z.union([z.literal('creationTime'), z.literal('isAdmin'), z.literal('kind'), z.literal('lastLoginTime'), z.literal('name'), z.literal('primaryEmail'), z.literal('suspended')])), required: false, displayOptions: {"show":{"output":["select"]}}, defaults: {"output":"simplified"} }),
      projection: z.union([z.literal('basic'), z.literal('custom'), z.literal('full'), expressionSchema]).optional(),
      customFieldMask: resolveSchema({ parameters, schema: z.array(z.string()), required: false, displayOptions: {"show":{"/projection":["custom"]}} }),
      filter: z.object({ customer: stringOrExpression.optional(), domain: stringOrExpression.optional(), query: stringOrExpression.optional(), showDeleted: booleanOrExpression.optional() }).optional(),
      sort: z.object({ sortRules: z.object({ orderBy: z.union([z.literal('email'), z.literal('familyName'), z.literal('givenName'), expressionSchema]).optional(), sortOrder: z.union([z.literal('ASCENDING'), z.literal('DESCENDING'), expressionSchema]).optional() }).optional() }).optional(),
    }).optional(),
  });
};
