// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftExcel/v1/resource_worksheet/operation_get_content.ts
/**
 * Microsoft Excel Node - Version 1
 * Discriminator: resource=worksheet, operation=getContent
 */


interface Credentials {
  microsoftExcelOAuth2Api: CredentialReference;
}

/** An Excel worksheet is a grid of cells. It can contain data, tables, charts, etc. */
export type MicrosoftExcelV1WorksheetGetContentParams = {
  resource: 'worksheet';
  operation: 'getContent';
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    workbook?: string | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    worksheet?: string | Expression<string>;
/**
 * The address or the name of the range. If not specified, the entire worksheet range is returned.
 * @default A1:C3
 */
    range?: string | Expression<string> | PlaceholderValue;
/**
 * Whether the data should be returned RAW instead of parsed into keys according to their header
 * @default false
 */
    rawData?: boolean | Expression<boolean>;
/**
 * The name of the property into which to write the RAW data
 * @displayOptions.show { rawData: [true] }
 * @default data
 */
    dataProperty?: string | Expression<string> | PlaceholderValue;
/**
 * Index of the first row which contains the actual data and not the keys. Starts with 0.
 * @displayOptions.hide { rawData: [true] }
 * @default 1
 */
    dataStartRow?: number | Expression<number>;
/**
 * Index of the row which contains the keys. Starts at 0. The incoming node data is matched to the keys for assignment. The matching is case sensitve.
 * @displayOptions.hide { rawData: [true] }
 * @default 0
 */
    keyRow?: number | Expression<number>;
/**
 * Filters
 * @displayOptions.show { rawData: [true] }
 * @default {}
 */
    filters?: {
    /** Fields the response will containt. Multiple can be added separated by ,.
     */
    fields?: string | Expression<string> | PlaceholderValue;
  };
};

export type MicrosoftExcelV1WorksheetGetContentNode = {
  type: 'n8n-nodes-base.microsoftExcel';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MicrosoftExcelV1WorksheetGetContentParams>;
};
