// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftExcel/v1/resource_worksheet/operation_get_content.schema.js
/**
 * Microsoft Excel Node - Version 1 - Zod Schema
 * Discriminator: resource=worksheet, operation=getContent
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('worksheet'),
      operation: z.literal('getContent'),
      workbook: stringOrExpression.optional(),
      worksheet: stringOrExpression.optional(),
      range: stringOrExpression.optional(),
      rawData: booleanOrExpression.optional(),
      dataProperty: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"rawData":[true]}}, defaults: {"rawData":false} }),
      dataStartRow: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"hide":{"rawData":[true]}}, defaults: {"rawData":false} }),
      keyRow: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"hide":{"rawData":[true]}}, defaults: {"rawData":false} }),
      filters: resolveSchema({ parameters, schema: z.object({ fields: stringOrExpression.optional() }), required: false, displayOptions: {"show":{"rawData":[true]}}, defaults: {"rawData":false} }),
    }).optional(),
  });
};
