// n8n/dist/modules/quick-connect/quick-connect.config.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.QuickConnectConfig = void 0;
const config_1 = require("@n8n/config");
const zod_1 = require("zod");
const baseQuickConnectOptionSchema = zod_1.z.object({
    packageName: zod_1.z.string(),
    credentialType: zod_1.z.string(),
    text: zod_1.z.string(),
    quickConnectType: zod_1.z.string(),
    consentText: zod_1.z.string().optional(),
    consentCheckbox: zod_1.z.string().optional(),
    config: zod_1.z.never().optional(),
    backendFlowConfig: zod_1.z.never().optional(),
});
const pineconeQuickConnectOptionSchema = baseQuickConnectOptionSchema.extend({
    quickConnectType: zod_1.z.literal('pinecone'),
    config: zod_1.z.object({
        integrationId: zod_1.z.string(),
    }),
});
const firecrawlQuickConnectSchema = baseQuickConnectOptionSchema.extend({
    quickConnectType: zod_1.z.literal('firecrawl'),
    consentText: zod_1.z.string(),
    backendFlowConfig: zod_1.z.object({
        secret: zod_1.z.string(),
    }),
});
const quickConnectOptionSchema = zod_1.z.union([
    firecrawlQuickConnectSchema,
    pineconeQuickConnectOptionSchema,
    baseQuickConnectOptionSchema,
]);
const quickConnectOptionsSchema = zod_1.z.string().pipe(zod_1.z.preprocess((input) => {
    try {
        return JSON.parse(input);
    }
    catch {
        return [];
    }
}, zod_1.z.array(quickConnectOptionSchema)));
let QuickConnectConfig = class QuickConnectConfig {
    constructor() {
        this.options = [];
    }
};
exports.QuickConnectConfig = QuickConnectConfig;
__decorate([
    (0, config_1.Env)('N8N_QUICK_CONNECT_OPTIONS', quickConnectOptionsSchema),
    __metadata("design:type", Array)
], QuickConnectConfig.prototype, "options", void 0);
exports.QuickConnectConfig = QuickConnectConfig = __decorate([
    config_1.Config
], QuickConnectConfig);
