// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/hubspot/v22/resource_ticket/operation_get_all.ts
/**
 * HubSpot Node - Version 2.2
 * Discriminator: resource=ticket, operation=getAll
 */


interface Credentials {
  hubspotApi: CredentialReference;
  hubspotAppToken: CredentialReference;
  hubspotOAuth2Api: CredentialReference;
}

/** Get many contacts */
export type HubspotV22TicketGetAllParams = {
  resource: 'ticket';
  operation: 'getAll';
  authentication?: 'apiKey' | 'appToken' | 'oAuth2' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Options
 * @default {}
 */
    additionalFields?: {
    /** Whether to include specific Ticket properties in the returned results. Choose from a list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;. Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     * @default []
     */
    properties?: string[];
    /** Works similarly to properties=, but this parameter will include the history for the specified property, instead of just including the current value. Use this parameter when you need the full history of changes to a property's value.
     */
    propertiesWithHistory?: string | Expression<string> | PlaceholderValue;
  };
};

export type HubspotV22TicketGetAllOutput = {
  isDeleted?: boolean;
  objectId?: number;
  objectType?: string;
  portalId?: number;
  properties?: {
    subject?: {
      isEncrypted?: boolean;
      sensitivityLevel?: string;
      source?: string;
      timestamp?: number;
      value?: string;
      versions?: Array<{
        name?: string;
        requestId?: string;
        source?: string;
        sourceId?: string;
        sourceUpstreamDeployable?: string;
        timestamp?: number;
        updatedByUserId?: number;
        useTimestampAsPersistenceTimestamp?: boolean;
        value?: string;
      }>;
    };
  };
};

export type HubspotV22TicketGetAllNode = {
  type: 'n8n-nodes-base.hubspot';
  version: 2.2;
  credentials?: Credentials;
  config: NodeConfig<HubspotV22TicketGetAllParams>;
  output?: Items<HubspotV22TicketGetAllOutput>;
};
