// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/freshdesk/v1/resource_ticket/operation_create.schema.js
/**
 * Freshdesk Node - Version 1 - Zod Schema
 * Discriminator: resource=ticket, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('ticket').default('ticket'),
      operation: z.literal('create').default('create'),
      requester: z.union([z.literal('email'), z.literal('facebookId'), z.literal('phone'), z.literal('requesterId'), z.literal('twitterId'), z.literal('uniqueExternalId'), expressionSchema]).optional(),
      requesterIdentificationValue: stringOrExpression.optional(),
      status: z.union([z.literal('closed'), z.literal('open'), z.literal('pending'), z.literal('resolved'), expressionSchema]).optional(),
      priority: z.union([z.literal('low'), z.literal('medium'), z.literal('high'), z.literal('urgent'), expressionSchema]).optional(),
      source: z.union([z.literal('chat'), z.literal('email'), z.literal('feedbackWidget'), z.literal('mobileHelp'), z.literal('OutboundEmail'), z.literal('phone'), z.literal('portal'), expressionSchema]).optional(),
      options: z.object({ agent: stringOrExpression.optional(), ccEmails: stringOrExpression.optional(), company: stringOrExpression.optional(), description: stringOrExpression.optional(), dueBy: stringOrExpression.optional(), emailConfigId: numberOrExpression.optional(), frDueBy: stringOrExpression.optional(), group: stringOrExpression.optional(), name: stringOrExpression.optional(), product: stringOrExpression.optional(), subject: stringOrExpression.optional(), tags: stringOrExpression.optional(), type: z.union([z.literal('Feature Request'), z.literal('Incident'), z.literal('Problem'), z.literal('Question'), z.literal('Refund'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
