// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/freshdesk/v1/resource_ticket/operation_create.ts
/**
 * Freshdesk Node - Version 1
 * Discriminator: resource=ticket, operation=create
 */


interface Credentials {
  freshdeskApi: CredentialReference;
}

/** Create a new ticket */
export type FreshdeskV1TicketCreateParams = {
  resource: 'ticket';
  operation: 'create';
/**
 * Requester Identification
 * @default requesterId
 */
    requester?: 'email' | 'facebookId' | 'phone' | 'requesterId' | 'twitterId' | 'uniqueExternalId' | Expression<string>;
/**
 * Value of the identification selected
 */
    requesterIdentificationValue?: string | Expression<string> | PlaceholderValue;
/**
 * Status
 * @default pending
 */
    status?: 'closed' | 'open' | 'pending' | 'resolved' | Expression<string>;
/**
 * Priority
 * @default low
 */
    priority?: 'low' | 'medium' | 'high' | 'urgent' | Expression<string>;
/**
 * The channel through which the ticket was created
 * @default portal
 */
    source?: 'chat' | 'email' | 'feedbackWidget' | 'mobileHelp' | 'OutboundEmail' | 'phone' | 'portal' | Expression<string>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** ID of the agent to whom the ticket has been assigned. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    agent?: string | Expression<string>;
    /** Separated by a comma (,) email addresses added in the 'cc' field of the incoming ticket email
     */
    ccEmails?: string | Expression<string> | PlaceholderValue;
    /** Company ID of the requester. This attribute can only be set if the Multiple Companies feature is enabled (Estate plan and above). Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    company?: string | Expression<string>;
    /** HTML content of the ticket
     */
    description?: string | Expression<string> | PlaceholderValue;
    /** Timestamp that denotes when the ticket is due to be resolved
     */
    dueBy?: string | Expression<string>;
    /** ID of email config which is used for this ticket. (i.e., support@yourcompany.com/sales@yourcompany.com) If product_id is given and email_config_id is not given, product's primary email_config_id will be set.
     */
    emailConfigId?: number | Expression<number>;
    /** Timestamp that denotes when the first response is due
     */
    frDueBy?: string | Expression<string>;
    /** ID of the group to which the ticket has been assigned. The default value is the ID of the group that is associated with the given email_config_id. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    group?: string | Expression<string>;
    /** Name of the requester
     */
    name?: string | Expression<string> | PlaceholderValue;
    /** ID of the product to which the ticket is associated. It will be ignored if the email_config_id attribute is set in the request. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    product?: string | Expression<string>;
    /** Subject of the ticket
     */
    subject?: string | Expression<string> | PlaceholderValue;
    /** Separated by a comma (,) tags that have been associated with the ticket
     */
    tags?: string | Expression<string> | PlaceholderValue;
    /** Helps categorize the ticket according to the different kinds of issues your support team deals with
     * @default Question
     */
    type?: 'Feature Request' | 'Incident' | 'Problem' | 'Question' | 'Refund' | Expression<string>;
  };
};

export type FreshdeskV1TicketCreateOutput = {
  created_at?: string;
  custom_fields?: {
    cf_reference_number?: null;
  };
  description?: string;
  description_text?: string;
  fr_escalated?: boolean;
  id?: number;
  is_escalated?: boolean;
  nr_due_by?: null;
  nr_escalated?: boolean;
  priority?: number;
  requester_id?: number;
  source?: number;
  spam?: boolean;
  status?: number;
  subject?: string;
  support_email?: null;
  tags?: Array<string>;
  to_emails?: null;
  updated_at?: string;
};

export type FreshdeskV1TicketCreateNode = {
  type: 'n8n-nodes-base.freshdesk';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<FreshdeskV1TicketCreateParams>;
  output?: Items<FreshdeskV1TicketCreateOutput>;
};
