// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/freshdesk/v1/resource_ticket/operation_get_all.schema.js
/**
 * Freshdesk Node - Version 1 - Zod Schema
 * Discriminator: resource=ticket, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('ticket').default('ticket'),
      operation: z.literal('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      options: z.object({ companyId: stringOrExpression.optional(), include: z.array(z.union([z.literal('company'), z.literal('description'), z.literal('requester'), z.literal('stats')])).optional(), order: z.union([z.literal('asc'), z.literal('desc'), expressionSchema]).optional(), orderBy: z.union([z.literal('createdAt'), z.literal('dueBy'), z.literal('updatedAt'), expressionSchema]).optional(), requesterEmail: stringOrExpression.optional(), requesterId: stringOrExpression.optional(), updatedSince: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
