// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/freshdesk/v1/resource_ticket/operation_update.ts
/**
 * Freshdesk Node - Version 1
 * Discriminator: resource=ticket, operation=update
 */


interface Credentials {
  freshdeskApi: CredentialReference;
}

/** Update a ticket */
export type FreshdeskV1TicketUpdateParams = {
  resource: 'ticket';
  operation: 'update';
/**
 * Ticket ID
 */
    ticketId?: string | Expression<string> | PlaceholderValue;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** ID of the agent to whom the ticket has been assigned. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    agent?: string | Expression<string>;
    /** Separated by a comma (,) email addresses added in the 'cc' field of the incoming ticket email
     */
    ccEmails?: string | Expression<string> | PlaceholderValue;
    /** Company ID of the requester. This attribute can only be set if the Multiple Companies feature is enabled (Estate plan and above). Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    company?: string | Expression<string>;
    /** Timestamp that denotes when the ticket is due to be resolved
     */
    dueBy?: string | Expression<string>;
    /** ID of email config which is used for this ticket. (i.e., support@yourcompany.com/sales@yourcompany.com) If product_id is given and email_config_id is not given, product's primary email_config_id will be set.
     */
    emailConfigId?: number | Expression<number>;
    /** Timestamp that denotes when the first response is due
     */
    frDueBy?: string | Expression<string>;
    /** ID of the group to which the ticket has been assigned. The default value is the ID of the group that is associated with the given email_config_id. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    group?: string | Expression<string>;
    /** Name of the requester
     */
    name?: string | Expression<string> | PlaceholderValue;
    /** ID of the product to which the ticket is associated. It will be ignored if the email_config_id attribute is set in the request. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    product?: string | Expression<string>;
    /** Priority
     * @default low
     */
    priority?: 'low' | 'medium' | 'high' | 'urgent' | Expression<string>;
    /** Requester Identification
     * @default requesterId
     */
    requester?: 'email' | 'facebookId' | 'phone' | 'requesterId' | 'twitterId' | 'uniqueExternalId' | Expression<string>;
    /** Value of the identification selected
     */
    requesterIdentificationValue?: string | Expression<string> | PlaceholderValue;
    /** Status
     * @default pending
     */
    status?: 'open' | 'pending' | 'resolved' | 'closed' | Expression<string>;
    /** The channel through which the ticket was created
     * @default portal
     */
    source?: 'chat' | 'email' | 'feedbackWidget' | 'mobileHelp' | 'OutboundEmail' | 'phone' | 'portal' | Expression<string>;
    /** Separated by a comma (,) tags that have been associated with the ticket
     */
    tags?: string | Expression<string> | PlaceholderValue;
    /** Helps categorize the ticket according to the different kinds of issues your support team deals with
     * @default Question
     */
    type?: 'Feature Request' | 'Incident' | 'Problem' | 'Question' | 'Refund' | Expression<string>;
  };
};

export type FreshdeskV1TicketUpdateNode = {
  type: 'n8n-nodes-base.freshdesk';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<FreshdeskV1TicketUpdateParams>;
};
