// @n8n/n8n-nodes-langchain/dist/nodes/vendors/GoogleGemini/actions/versionDescription.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.versionDescription = void 0;
const n8n_workflow_1 = require("n8n-workflow");
const audio = __importStar(require("./audio"));
const document = __importStar(require("./document"));
const file = __importStar(require("./file"));
const fileSearch = __importStar(require("./fileSearch"));
const image = __importStar(require("./image"));
const text = __importStar(require("./text"));
const video = __importStar(require("./video"));
exports.versionDescription = {
    displayName: 'Google Gemini',
    name: 'googleGemini',
    icon: 'file:gemini.svg',
    group: ['transform'],
    version: [1, 1.1],
    defaultVersion: 1.1,
    subtitle: '={{ $parameter["operation"] + ": " + $parameter["resource"] }}',
    description: 'Interact with Google Gemini AI models',
    defaults: {
        name: 'Google Gemini',
    },
    usableAsTool: true,
    codex: {
        alias: ['LangChain', 'video', 'document', 'audio', 'transcribe', 'assistant'],
        categories: ['AI'],
        subcategories: {
            AI: ['Agents', 'Miscellaneous', 'Root Nodes'],
        },
        resources: {
            primaryDocumentation: [
                {
                    url: 'https://docs.n8n.io/integrations/builtin/app-nodes/n8n-nodes-langchain.googlegemini/',
                },
            ],
        },
    },
    inputs: `={{
		(() => {
			const resource = $parameter.resource;
	  	const operation = $parameter.operation;
			if (resource === 'text' && operation === 'message') {
				return [{ type: 'main' }, { type: 'ai_tool', displayName: 'Tools' }];
			}

			return ['main'];
		})()
	}}`,
    outputs: [n8n_workflow_1.NodeConnectionTypes.Main],
    credentials: [
        {
            name: 'googlePalmApi',
            required: true,
        },
    ],
    properties: [
        {
            displayName: 'Resource',
            name: 'resource',
            type: 'options',
            noDataExpression: true,
            options: [
                {
                    name: 'Audio',
                    value: 'audio',
                },
                {
                    name: 'Document',
                    value: 'document',
                },
                {
                    name: 'File Search',
                    value: 'fileSearch',
                },
                {
                    name: 'Image',
                    value: 'image',
                },
                {
                    name: 'Media File',
                    value: 'file',
                },
                {
                    name: 'Text',
                    value: 'text',
                },
                {
                    name: 'Video',
                    value: 'video',
                },
            ],
            default: 'text',
        },
        ...audio.description,
        ...document.description,
        ...file.description,
        ...fileSearch.description,
        ...image.description,
        ...text.description,
        ...video.description,
    ],
};
