// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/harvest/v1/resource_task/operation_get_all.ts
/**
 * Harvest Node - Version 1
 * Discriminator: resource=task, operation=getAll
 */


interface Credentials {
  harvestApi: CredentialReference;
  harvestOAuth2Api: CredentialReference;
}

/** Get data of many clients */
export type HarvestV1TaskGetAllParams = {
  resource: 'task';
  operation: 'getAll';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    accountId?: string | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Whether to only return active tasks and false to return inactive tasks
     * @default true
     */
    is_active?: boolean | Expression<boolean>;
    /** The page number to use in pagination
     * @default 1
     */
    page?: number | Expression<number>;
    /** Only return tasks belonging to the task with the given ID
     */
    updated_since?: string | Expression<string>;
  };
};

export type HarvestV1TaskGetAllNode = {
  type: 'n8n-nodes-base.harvest';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<HarvestV1TaskGetAllParams>;
};
