// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/harvest/v1/resource_task/operation_update.ts
/**
 * Harvest Node - Version 1
 * Discriminator: resource=task, operation=update
 */


interface Credentials {
  harvestApi: CredentialReference;
  harvestOAuth2Api: CredentialReference;
}

/** Update a client */
export type HarvestV1TaskUpdateParams = {
  resource: 'task';
  operation: 'update';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    accountId?: string | Expression<string>;
/**
 * The ID of the task you want to update
 */
    id?: string | Expression<string> | PlaceholderValue;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** Whether default tasks should be marked billable when creating a new project. Defaults to true.
     * @default false
     */
    billable_by_default?: boolean | Expression<boolean>;
    /** The default hourly rate to use for this task when it is added to a project. Defaults to 0.
     * @default 0
     */
    default_hourly_rate?: number | Expression<number>;
    /** Whether this task is active or archived. Defaults to true.
     * @default true
     */
    is_active?: boolean | Expression<boolean>;
    /** Whether this task should be automatically added to future projects. Defaults to false.
     * @default false
     */
    is_default?: boolean | Expression<boolean>;
    /** Name of the task
     */
    name?: string | Expression<string> | PlaceholderValue;
  };
};

export type HarvestV1TaskUpdateNode = {
  type: 'n8n-nodes-base.harvest';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<HarvestV1TaskUpdateParams>;
};
