// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/freshworksCrm/v1/resource_deal/operation_create.schema.js
/**
 * Freshworks CRM Node - Version 1 - Zod Schema
 * Discriminator: resource=deal, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('deal'),
      operation: z.literal('create').default('create'),
      amount: numberOrExpression.optional(),
      name: stringOrExpression.optional(),
      additionalFields: z.object({ base_currency_amount: numberOrExpression.optional(), campaign_id: stringOrExpression.optional(), currency_id: stringOrExpression.optional(), deal_payment_status_id: stringOrExpression.optional(), deal_pipeline_id: stringOrExpression.optional(), deal_product_id: stringOrExpression.optional(), deal_reason_id: stringOrExpression.optional(), deal_stage_id: stringOrExpression.optional(), deal_type_id: stringOrExpression.optional(), lead_source_id: stringOrExpression.optional(), owner_id: stringOrExpression.optional(), probability: numberOrExpression.optional(), sales_account_id: stringOrExpression.optional(), territory_id: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
