// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleSlides/v2/resource_page/operation_get_thumbnail.ts
/**
 * Google Slides Node - Version 2
 * Discriminator: resource=page, operation=getThumbnail
 */


interface Credentials {
  googleApi: CredentialReference;
  googleSlidesOAuth2Api: CredentialReference;
}

/** Get a thumbnail */
export type GoogleSlidesV2PageGetThumbnailParams = {
  resource: 'page';
  operation: 'getThumbnail';
/**
 * Authentication
 * @default oAuth2
 */
    authentication?: 'oAuth2' | 'serviceAccount' | Expression<string>;
/**
 * ID of the presentation to retrieve. Found in the presentation URL: &lt;code&gt;https://docs.google.com/presentation/d/PRESENTATION_ID/edit&lt;/code&gt;
 */
    presentationId?: string | Expression<string> | PlaceholderValue;
/**
 * ID of the page object to retrieve
 */
    pageObjectId?: string | Expression<string> | PlaceholderValue;
/**
 * Name of the binary property to which to write the data of the read page
 * @default false
 */
    download?: boolean | Expression<boolean>;
/**
 * Put Output File in Field
 * @hint The name of the output binary field to put the file in
 * @displayOptions.show { download: [true] }
 * @default data
 */
    binaryProperty?: string | Expression<string> | PlaceholderValue;
};

export type GoogleSlidesV2PageGetThumbnailOutput = {
  contentUrl?: string;
  height?: number;
  width?: number;
};

export type GoogleSlidesV2PageGetThumbnailNode = {
  type: 'n8n-nodes-base.googleSlides';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<GoogleSlidesV2PageGetThumbnailParams>;
  output?: Items<GoogleSlidesV2PageGetThumbnailOutput>;
};
