// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/salesforce/v1/resource_lead/operation_get_all.ts
/**
 * Salesforce Node - Version 1
 * Discriminator: resource=lead, operation=getAll
 */


interface Credentials {
  salesforceOAuth2Api: CredentialReference;
  salesforceJwtApi: CredentialReference;
}

/** Represents a prospect or potential */
export type SalesforceV1LeadGetAllParams = {
  resource: 'lead';
  operation: 'getAll';
/**
 * OAuth Authorization Flow
 * @default oAuth2
 */
    authentication?: 'oAuth2' | 'jwt' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** The condition to set
     * @default {}
     */
    conditionsUi?: {
        /** Condition
     */
    conditionValues?: Array<{
      /** For date, number, or boolean, please use expressions. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
       */
      field?: string | Expression<string>;
      /** Operation
       * @default equal
       */
      operation?: '<' | '<=' | 'equal' | '>' | '>=' | Expression<string>;
      /** Value
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
    /** Fields to include separated by ,
     */
    fields?: string | Expression<string> | PlaceholderValue;
  };
};

export type SalesforceV1LeadGetAllOutput = {
  attributes?: {
    type?: string;
    url?: string;
  };
  Id?: string;
  Status?: string;
};

export type SalesforceV1LeadGetAllNode = {
  type: 'n8n-nodes-base.salesforce';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SalesforceV1LeadGetAllParams>;
  output?: Items<SalesforceV1LeadGetAllOutput>;
};
