// @n8n/n8n-nodes-langchain/dist/node-definitions/nodes/n8n-nodes-langchain/chainRetrievalQa/v16.schema.js
/**
 * Question and Answer Chain Node - Version 1.6 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema, languageModelInstanceSchema, retrieverInstanceSchema }) {

  // Static subnode schema
  const subnodesSchema = z.object({
    model: z.union([languageModelInstanceSchema, z.array(languageModelInstanceSchema)]),
    retriever: retrieverInstanceSchema,
  }).strict();

  // Parameters schema
  const parametersSchema = z.object({
    promptType: z.union([z.literal('auto'), z.literal('guardrails'), z.literal('define'), expressionSchema]).optional(),
    text: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"promptType":["guardrails","auto","define"]}}, defaults: {"promptType":"auto"} }),
    options: z.object({ systemPromptTemplate: stringOrExpression.optional(), systemPromptTemplate: stringOrExpression.optional(), batching: z.unknown().optional() }).optional(),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
    subnodes: subnodesSchema,
  });
};
