// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/syncroMsp/v1/resource_rmm/operation_mute.schema.js
/**
 * SyncroMSP Node - Version 1 - Zod Schema
 * Discriminator: resource=rmm, operation=mute
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('rmm'),
      operation: z.literal('mute'),
      alertId: stringOrExpression.optional(),
      muteFor: z.union([z.literal('1-hour'), z.literal('1-day'), z.literal('2-days'), z.literal('1-week'), z.literal('2-weeks'), z.literal('1-month'), z.literal('forever'), expressionSchema]).optional(),
    }).optional(),
  });
};
