// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/wooCommerce/v1/resource_order/operation_get_all.schema.js
/**
 * WooCommerce Node - Version 1 - Zod Schema
 * Discriminator: resource=order, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('order'),
      operation: z.literal('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      options: z.object({ after: stringOrExpression.optional(), before: stringOrExpression.optional(), customer: stringOrExpression.optional(), decimalPoints: numberOrExpression.optional(), order: z.union([z.literal('asc'), z.literal('desc'), expressionSchema]).optional(), product: stringOrExpression.optional(), orderBy: z.union([z.literal('date'), z.literal('id'), z.literal('include'), z.literal('slug'), z.literal('title'), expressionSchema]).optional(), search: stringOrExpression.optional(), status: z.union([z.literal('any'), z.literal('cancelled'), z.literal('completed'), z.literal('failed'), z.literal('on-hold'), z.literal('pending'), z.literal('processing'), z.literal('refunded'), z.literal('trash'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
