// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/twitter/v2/resource_user/operation_search_user.schema.js
/**
 * X (Formerly Twitter) Node - Version 2 - Zod Schema
 * Discriminator: resource=user, operation=searchUser
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('user'),
      operation: z.literal('searchUser'),
      user: resolveSchema({ parameters, schema: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('username'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]), required: false, displayOptions: {"hide":{"me":[true]}}, defaults: {"me":false} }),
      me: booleanOrExpression.optional(),
    }).optional(),
  });
};
