// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/bubble/v1/resource_object/operation_update.schema.js
/**
 * Bubble Node - Version 1 - Zod Schema
 * Discriminator: resource=object, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('object').default('object'),
      operation: z.literal('update'),
      typeName: stringOrExpression.optional(),
      objectId: stringOrExpression.optional(),
      properties: z.object({ property: z.array(z.object({ key: stringOrExpression.optional(), value: stringOrExpression.optional() })).optional() }).optional(),
    }).optional(),
  });
};
