// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/monicaCrm/v1/resource_conversation_message/operation_update.schema.js
/**
 * Monica CRM Node - Version 1 - Zod Schema
 * Discriminator: resource=conversationMessage, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('conversationMessage'),
      operation: z.literal('update'),
      messageId: stringOrExpression.optional(),
      conversationId: stringOrExpression.optional(),
      updateFields: z.object({ contact_id: stringOrExpression.optional(), content: stringOrExpression.optional(), written_at: stringOrExpression.optional(), written_by_me: z.union([z.literal(true), z.literal(false), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
