// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pagerDuty/v1/resource_incident_note/operation_create.schema.js
/**
 * PagerDuty Node - Version 1 - Zod Schema
 * Discriminator: resource=incidentNote, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('incidentNote'),
      operation: z.literal('create').default('create'),
      authentication: z.union([z.literal('apiToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      incidentId: stringOrExpression.optional(),
      content: stringOrExpression.optional(),
      email: stringOrExpression.optional(),
    }).optional(),
  });
};
