// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/airtop/v1/resource_file/operation_get_many.ts
/**
 * Airtop Node - Version 1
 * Discriminator: resource=file, operation=getMany
 */


interface Credentials {
  airtopApi: CredentialReference;
}

/** Get details of multiple uploaded files */
export type AirtopV1FileGetManyParams = {
  resource: 'file';
  operation: 'getMany';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 10
 */
    limit?: number | Expression<number>;
/**
 * Comma-separated list of &lt;a href="https://docs.airtop.ai/api-reference/airtop-api/sessions/create" target="_blank"&gt;Session IDs&lt;/a&gt; to filter files by. When empty, all files from all sessions will be returned.
 */
    sessionIds?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to output one item containing all files or output each file as a separate item
 * @default true
 */
    outputSingleItem?: boolean | Expression<boolean>;
};

export type AirtopV1FileGetManyNode = {
  type: 'n8n-nodes-base.airtop';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<AirtopV1FileGetManyParams>;
};
