// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/harvest/v1/resource_contact/operation_create.ts
/**
 * Harvest Node - Version 1
 * Discriminator: resource=contact, operation=create
 */


interface Credentials {
  harvestApi: CredentialReference;
  harvestOAuth2Api: CredentialReference;
}

/** Create a client */
export type HarvestV1ContactCreateParams = {
  resource: 'contact';
  operation: 'create';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    accountId?: string | Expression<string>;
/**
 * The first name of the contact
 */
    firstName?: string | Expression<string> | PlaceholderValue;
/**
 * The ID of the client associated with this contact
 */
    clientId?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** The contact’s email address
     */
    email?: string | Expression<string> | PlaceholderValue;
    /** The contact’s fax number
     */
    fax?: string | Expression<string> | PlaceholderValue;
    /** The last name of the contact
     */
    last_name?: string | Expression<string> | PlaceholderValue;
    /** The contact’s mobile phone number
     */
    phone_mobile?: string | Expression<string> | PlaceholderValue;
    /** The contact’s office phone number
     */
    phone_office?: string | Expression<string> | PlaceholderValue;
    /** The title of the contact
     */
    title?: string | Expression<string> | PlaceholderValue;
  };
};

export type HarvestV1ContactCreateNode = {
  type: 'n8n-nodes-base.harvest';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<HarvestV1ContactCreateParams>;
};
