// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/quickbase/v1/resource_file/operation_download.schema.js
/**
 * Quick Base Node - Version 1 - Zod Schema
 * Discriminator: resource=file, operation=download
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('file'),
      operation: z.literal('download'),
      tableId: stringOrExpression.optional(),
      recordId: stringOrExpression.optional(),
      fieldId: stringOrExpression.optional(),
      versionNumber: numberOrExpression.optional(),
      binaryPropertyName: stringOrExpression.optional(),
    }).optional(),
  });
};
