// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/quickbase/v1/resource_file/operation_download.ts
/**
 * Quick Base Node - Version 1
 * Discriminator: resource=file, operation=download
 */


interface Credentials {
  quickbaseApi: CredentialReference;
}

/** Download a file */
export type QuickbaseV1FileDownloadParams = {
  resource: 'file';
  operation: 'download';
/**
 * The table identifier
 */
    tableId?: string | Expression<string> | PlaceholderValue;
/**
 * The unique identifier of the record
 */
    recordId?: string | Expression<string> | PlaceholderValue;
/**
 * The unique identifier of the field
 */
    fieldId?: string | Expression<string> | PlaceholderValue;
/**
 * The file attachment version number
 * @default 1
 */
    versionNumber?: number | Expression<number>;
/**
 * Input Binary Field
 * @hint The name of the input binary field containing the file to be written
 * @default data
 */
    binaryPropertyName?: string | Expression<string> | PlaceholderValue;
};

export type QuickbaseV1FileDownloadNode = {
  type: 'n8n-nodes-base.quickbase';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<QuickbaseV1FileDownloadParams>;
};
