// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleChat/v1/resource_message/operation_update.ts
/**
 * Google Chat Node - Version 1
 * Discriminator: resource=message, operation=update
 */


interface Credentials {
  googleApi: CredentialReference;
  googleChatOAuth2Api: CredentialReference;
}

/** Update a message */
export type GoogleChatV1MessageUpdateParams = {
  resource: 'message';
  operation: 'update';
  authentication?: 'oAuth2' | 'serviceAccount' | Expression<string>;
/**
 * Resource name of the message to be updated, in the form "spaces//messages/"
 */
    messageId?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to pass the update fields object as JSON
 * @default false
 */
    jsonParameters?: boolean | Expression<boolean>;
/**
 * Update Fields
 * @displayOptions.show { jsonParameters: [false] }
 * @default {}
 */
    updateFieldsUi?: {
    /** Text
     */
    text?: string | Expression<string> | PlaceholderValue;
  };
/**
 * Message input as JSON Object or JSON String
 * @displayOptions.show { jsonParameters: [true] }
 */
    updateFieldsJson?: IDataObject | string | Expression<string>;
};

export type GoogleChatV1MessageUpdateNode = {
  type: 'n8n-nodes-base.googleChat';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GoogleChatV1MessageUpdateParams>;
};
