// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleChat/v1/resource_message/operation_create.schema.js
/**
 * Google Chat Node - Version 1 - Zod Schema
 * Discriminator: resource=message, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('message').default('message'),
      operation: z.literal('create'),
      authentication: z.union([z.literal('oAuth2'), z.literal('serviceAccount'), expressionSchema]).optional(),
      spaceId: stringOrExpression.optional(),
      jsonParameters: booleanOrExpression.optional(),
      messageUi: resolveSchema({ parameters, schema: z.object({ text: stringOrExpression.optional() }), required: false, displayOptions: {"show":{"jsonParameters":[false]}}, defaults: {"jsonParameters":false} }),
      messageJson: resolveSchema({ parameters, schema: z.union([iDataObjectSchema, z.string()]), required: false, displayOptions: {"show":{"jsonParameters":[true]}}, defaults: {"jsonParameters":false} }),
      additionalFields: z.object({ requestId: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
