// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/youTube/v1/resource_video_category/operation_get_all.ts
/**
 * YouTube Node - Version 1
 * Discriminator: resource=videoCategory, operation=getAll
 */


interface Credentials {
  youTubeOAuth2Api: CredentialReference;
}

/** Retrieve many channels */
export type YouTubeV1VideoCategoryGetAllParams = {
  resource: 'videoCategory';
  operation: 'getAll';
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    regionCode?: string | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 25
 */
    limit?: number | Expression<number>;
};

export type YouTubeV1VideoCategoryGetAllOutput = {
  etag?: string;
  id?: string;
  kind?: string;
  snippet?: {
    assignable?: boolean;
    channelId?: string;
    title?: string;
  };
};

export type YouTubeV1VideoCategoryGetAllNode = {
  type: 'n8n-nodes-base.youTube';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<YouTubeV1VideoCategoryGetAllParams>;
  output?: Items<YouTubeV1VideoCategoryGetAllOutput>;
};
