// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/s3/v1/resource_folder/operation_get_all.ts
/**
 * S3 Node - Version 1
 * Discriminator: resource=folder, operation=getAll
 */


interface Credentials {
  s3: CredentialReference;
}

/** Get many buckets */
export type S3V1FolderGetAllParams = {
  resource: 'folder';
  operation: 'getAll';
/**
 * Bucket Name
 */
    bucketName?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** The owner field is not present in listV2 by default, if you want to return owner field with each key in the result then set the fetch owner field to true
     * @default false
     */
    fetchOwner?: boolean | Expression<boolean>;
    /** Folder Key
     */
    folderKey?: string | Expression<string> | PlaceholderValue;
  };
};

export type S3V1FolderGetAllOutput = {
  ETag?: string;
  Key?: string;
  LastModified?: string;
  Size?: string;
  StorageClass?: string;
  Type?: string;
};

export type S3V1FolderGetAllNode = {
  type: 'n8n-nodes-base.s3';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<S3V1FolderGetAllParams>;
  output?: Items<S3V1FolderGetAllOutput>;
};
