// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/zulip/v1/resource_message/operation_update.schema.js
/**
 * Zulip Node - Version 1 - Zod Schema
 * Discriminator: resource=message, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('message').default('message'),
      operation: z.literal('update'),
      messageId: stringOrExpression.optional(),
      updateFields: z.object({ content: stringOrExpression.optional(), propagateMode: z.union([z.literal('changeOne'), z.literal('changeLater'), z.literal('changeAll'), expressionSchema]).optional(), topic: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
