// @microsoft/agents-a365-observability/dist/cjs/utils/logging.js
"use strict";
// ------------------------------------------------------------------------------
// Copyright (c) Microsoft Corporation. All rights reserved.
// ------------------------------------------------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.logger = void 0;
exports.formatError = formatError;
/**
 * Format error object for logging with message and stack trace
 */
function formatError(error) {
    if (error instanceof Error) {
        return `${error.message}\nStack: ${error.stack || 'No stack trace'}`;
    }
    return String(error);
}
/**
 * Simple logger for Agent 365 observability
 *
 * Usage:
 *   import logger from './logging';
 *   logger.info('Info message');    // Shows when A365ObservabilityLogLevel includes 'info'
 *   logger.warn('Warning');         // Shows when A365ObservabilityLogLevel includes 'warn'
 *   logger.error('Error');          // Shows when A365ObservabilityLogLevel includes 'error'
 *
 * Environment Variable:
 *   A365ObservabilityLogLevel=none|info|warn|error|info|warn|info|error  (default: none)
 *
 *   Single values:
 *   none = no logging (default)
 *   info = info messages only
 *   warn = warn messages only
 *   error = error messages only
 *
 *   Multiple values (pipe-separated):
 *   info|warn = info and warn messages
 *   warn|error = warn and error messages
 *   info|warn|error = all message types
 */
const LOG_LEVELS = {
    none: 0,
    info: 1,
    warn: 2,
    error: 3
};
function parseLogLevel(level) {
    const levels = new Set();
    // Split by | to support multiple levels like "info|warn|error"
    const levelStrings = level.toLowerCase().trim().split('|');
    for (const levelString of levelStrings) {
        const normalizedLevel = levelString.trim();
        const levelValue = LOG_LEVELS[normalizedLevel];
        if (levelValue !== undefined) {
            levels.add(levelValue);
        }
    }
    // If no valid levels found, default to none
    if (levels.size === 0) {
        levels.add(LOG_LEVELS.none);
    }
    return levels;
}
const enabledLogLevels = parseLogLevel(process.env.A365_OBSERVABILITY_LOG_LEVEL || 'none');
exports.logger = {
    info: (message, ...args) => {
        if (enabledLogLevels.has(LOG_LEVELS.info)) {
            console.log('[INFO]', message, ...args);
        }
    },
    warn: (message, ...args) => {
        if (enabledLogLevels.has(LOG_LEVELS.warn)) {
            console.warn('[WARN]', message, ...args);
        }
    },
    error: (message, ...args) => {
        if (enabledLogLevels.has(LOG_LEVELS.error)) {
            console.error('[ERROR]', message, ...args);
        }
    }
};
exports.default = exports.logger;
