// @n8n/ai-workflow-builder/dist/utils/stream-processor.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DEFAULT_WORKFLOW_UPDATE_TOOLS = void 0;
exports.cleanContextTags = cleanContextTags;
exports.processStreamChunk = processStreamChunk;
exports.createStreamProcessor = createStreamProcessor;
exports.formatMessages = formatMessages;
const messages_1 = require("@langchain/core/messages");
exports.DEFAULT_WORKFLOW_UPDATE_TOOLS = [
    'add_nodes',
    'connect_nodes',
    'update_node_parameters',
    'remove_node',
];
const EMITTING_NODES = ['responder'];
const SKIPPED_NODES = [
    'supervisor',
    'tools',
    'cleanup_dangling_tool_calls',
    'create_workflow_name',
    'auto_compact_messages',
    'builder_subgraph',
    'discovery_subgraph',
    'assistant_subgraph',
];
const SKIPPED_SUBGRAPH_PREFIXES = ['discovery_subgraph', 'builder_subgraph'];
function isFromSkippedSubgraph(namespace) {
    return namespace.some((ns) => SKIPPED_SUBGRAPH_PREFIXES.some((prefix) => ns.startsWith(prefix)));
}
function shouldSkipNode(nodeName) {
    return SKIPPED_NODES.includes(nodeName);
}
function shouldEmitFromNode(nodeName) {
    return EMITTING_NODES.includes(nodeName);
}
function hasMessageInUpdate(update) {
    const typed = update;
    return Array.isArray(typed?.messages) && typed.messages.length > 0;
}
function shouldFilterSubgraphUpdate(namespace, data) {
    if (!isFromSkippedSubgraph(namespace))
        return false;
    return Object.entries(data).some(([nodeName, update]) => {
        if (shouldSkipNode(nodeName))
            return false;
        return hasMessageInUpdate(update);
    });
}
function isSubgraphEvent(event) {
    return Array.isArray(event) && event.length === 3 && Array.isArray(event[0]);
}
function isParentEvent(event) {
    return Array.isArray(event) && event.length === 2 && typeof event[0] === 'string';
}
function extractMessageContent(messages) {
    if (messages.length === 0)
        return null;
    const lastMessage = messages[messages.length - 1];
    if (!lastMessage.content)
        return null;
    if (Array.isArray(lastMessage.content)) {
        const textContent = lastMessage.content
            .filter((c) => c.type === 'text')
            .map((c) => c.text)
            .join('\n');
        return textContent || null;
    }
    return lastMessage.content;
}
function cleanContextTags(text) {
    let cleaned = text.replace(/\n*<current_workflow_json>[\s\S]*?<\/current_execution_nodes_schemas>/, '');
    const userRequestMatch = cleaned.match(/<user_request>\n?([\s\S]*?)\n?<\/user_request>/);
    if (userRequestMatch) {
        return userRequestMatch[1].trim();
    }
    cleaned = cleaned.replace(/<conversation_summary>[\s\S]*?<\/conversation_summary>\s*/g, '');
    cleaned = cleaned.replace(/<previous_requests>[\s\S]*?<\/previous_requests>\s*/g, '');
    cleaned = cleaned.replace(/<workflow_file[^>]*>[\s\S]*?<\/workflow_file>\s*/g, '');
    return cleaned.trim();
}
function isRecord(value) {
    return typeof value === 'object' && value !== null;
}
function isUnknownArray(value) {
    return Array.isArray(value);
}
function isQuestionsInterruptValue(value) {
    if (!isRecord(value))
        return false;
    return value.type === 'questions' && Array.isArray(value.questions);
}
function isPlanInterruptValue(value) {
    if (!isRecord(value))
        return false;
    return value.type === 'plan' && isRecord(value.plan);
}
function extractInterruptPayload(update) {
    if (!isRecord(update))
        return null;
    const rawInterrupts = update.__interrupt__;
    if (!isUnknownArray(rawInterrupts) || rawInterrupts.length === 0)
        return null;
    const first = rawInterrupts[0];
    if (!isRecord(first))
        return null;
    const value = first.value;
    if (!isRecord(value))
        return null;
    const id = typeof first.id === 'string' ? first.id : undefined;
    if (isQuestionsInterruptValue(value) || isPlanInterruptValue(value)) {
        return { value, id };
    }
    return null;
}
function processInterrupt(interruptValue, id) {
    if (interruptValue.type === 'questions') {
        const chunk = {
            role: 'assistant',
            type: 'questions',
            introMessage: interruptValue.introMessage,
            questions: interruptValue.questions,
        };
        return { messages: [chunk], ...(id ? { interruptId: id } : {}) };
    }
    const chunk = {
        role: 'assistant',
        type: 'plan',
        plan: interruptValue.plan,
    };
    return { messages: [chunk], ...(id ? { interruptId: id } : {}) };
}
function processOperationsUpdate(update) {
    const typed = update;
    if (!typed?.workflowJSON || typed.workflowOperations === undefined)
        return null;
    const workflowUpdateChunk = {
        role: 'assistant',
        type: 'workflow-updated',
        codeSnippet: JSON.stringify(typed.workflowJSON, null, 2),
    };
    return { messages: [workflowUpdateChunk] };
}
function processWorkflowNameUpdate(update) {
    const typed = update;
    if (!typed?.workflowJSON?.name)
        return null;
    const workflowUpdateChunk = {
        role: 'assistant',
        type: 'workflow-updated',
        codeSnippet: JSON.stringify(typed.workflowJSON, null, 2),
    };
    return { messages: [workflowUpdateChunk] };
}
function processAgentNodeUpdate(nodeName, update) {
    if (!shouldEmitFromNode(nodeName))
        return null;
    const typed = update;
    if (!typed?.messages?.length)
        return null;
    const content = extractMessageContent(typed.messages);
    if (!content?.trim() || content.includes('<current_workflow_json>'))
        return null;
    const messageChunk = {
        role: 'assistant',
        type: 'message',
        text: content,
    };
    return { messages: [messageChunk] };
}
function processCustomChunk(chunk) {
    if (!chunk || typeof chunk !== 'object')
        return null;
    const typed = chunk;
    if (typed.type === 'tool') {
        return { messages: [typed] };
    }
    if (typed.type === 'message' && 'role' in typed && 'text' in typed) {
        return { messages: [typed] };
    }
    return null;
}
function processUpdatesChunk(nodeUpdate) {
    if (!nodeUpdate || typeof nodeUpdate !== 'object')
        return null;
    if (nodeUpdate.compact_messages) {
        const compactedChunk = { type: 'messages-compacted' };
        return { messages: [compactedChunk] };
    }
    if (nodeUpdate.delete_messages) {
        return null;
    }
    const interruptPayload = extractInterruptPayload(nodeUpdate);
    if (interruptPayload) {
        return processInterrupt(interruptPayload.value, interruptPayload.id);
    }
    if (nodeUpdate.process_operations) {
        return processOperationsUpdate(nodeUpdate.process_operations);
    }
    if (nodeUpdate.create_workflow_name) {
        return processWorkflowNameUpdate(nodeUpdate.create_workflow_name);
    }
    for (const [nodeName, update] of Object.entries(nodeUpdate)) {
        if (shouldSkipNode(nodeName))
            continue;
        const result = processAgentNodeUpdate(nodeName, update);
        if (result)
            return result;
    }
    return null;
}
function processStreamChunk(streamMode, chunk) {
    if (streamMode === 'updates') {
        return processUpdatesChunk(chunk);
    }
    if (streamMode === 'custom') {
        return processCustomChunk(chunk);
    }
    return null;
}
function processSubgraphEvent(event) {
    const [namespace, streamMode, data] = event;
    if (streamMode === 'updates') {
        const interruptPayload = extractInterruptPayload(data);
        if (interruptPayload) {
            const shouldEmit = namespace.length > 0;
            if (shouldEmit) {
                return processInterrupt(interruptPayload.value, interruptPayload.id);
            }
            return null;
        }
    }
    if (streamMode === 'updates' &&
        shouldFilterSubgraphUpdate(namespace, data)) {
        return null;
    }
    return processStreamChunk(streamMode, data);
}
function processParentEvent(event) {
    const [streamMode, chunk] = event;
    if (!streamMode || typeof streamMode !== 'string')
        return null;
    return processStreamChunk(streamMode, chunk);
}
function processEvent(event) {
    if (isSubgraphEvent(event)) {
        return processSubgraphEvent(event);
    }
    if (isParentEvent(event)) {
        return processParentEvent(event);
    }
    return null;
}
async function* createStreamProcessor(stream) {
    const seenInterruptIds = new Set();
    for await (const event of stream) {
        const result = processEvent(event);
        if (!result)
            continue;
        if (result.interruptId) {
            if (seenInterruptIds.has(result.interruptId)) {
                continue;
            }
            seenInterruptIds.add(result.interruptId);
        }
        yield result;
    }
}
function extractHumanMessageText(content) {
    if (typeof content === 'string') {
        return content;
    }
    if (Array.isArray(content)) {
        return content
            .filter((c) => typeof c === 'object' && c !== null && 'type' in c && c.type === 'text' && 'text' in c)
            .map((c) => c.text)
            .join('\n');
    }
    return '';
}
function isUserAnswersPayload(value) {
    if (!Array.isArray(value))
        return false;
    return value.every((item) => isRecord(item) &&
        typeof item.questionId === 'string' &&
        typeof item.question === 'string' &&
        Array.isArray(item.selectedOptions));
}
function formatHumanMessage(msg) {
    const rawText = extractHumanMessageText(msg.content);
    const cleanedText = cleanContextTags(rawText);
    const resumeData = msg.additional_kwargs?.resumeData;
    if (isUserAnswersPayload(resumeData)) {
        const result = {
            role: 'user',
            type: 'user_answers',
            answers: resumeData,
        };
        const messageId = msg.additional_kwargs?.messageId;
        if (typeof messageId === 'string') {
            result.id = messageId;
        }
        return result;
    }
    const result = {
        role: 'user',
        type: 'message',
        text: cleanedText,
    };
    const versionId = msg.additional_kwargs?.versionId;
    if (typeof versionId === 'string') {
        result.revertVersionId = versionId;
    }
    const messageId = msg.additional_kwargs?.messageId;
    if (typeof messageId === 'string') {
        result.id = messageId;
    }
    return result;
}
function processArrayContent(content) {
    const textMessages = content.filter((c) => typeof c === 'object' && c !== null && 'type' in c && c.type === 'text' && 'text' in c);
    return textMessages.map((textMessage) => ({
        role: 'assistant',
        type: 'message',
        text: textMessage.text,
    }));
}
function processAIMessageContent(msg) {
    if (!msg.content) {
        return [];
    }
    if (Array.isArray(msg.content)) {
        return processArrayContent(msg.content);
    }
    return [
        {
            role: 'assistant',
            type: 'message',
            text: msg.content,
        },
    ];
}
function tryFormatHitlMessage(msg) {
    const messageType = msg.additional_kwargs?.messageType;
    if (messageType !== 'questions' && messageType !== 'plan')
        return null;
    if (typeof msg.content !== 'string')
        return null;
    let parsed;
    try {
        parsed = JSON.parse(msg.content);
    }
    catch {
        return null;
    }
    if (!isRecord(parsed) || parsed.type !== messageType)
        return null;
    if (messageType === 'questions') {
        const questions = parsed.questions;
        if (!Array.isArray(questions))
            return null;
        const introMessage = typeof parsed.introMessage === 'string' ? parsed.introMessage : undefined;
        return {
            role: 'assistant',
            type: 'questions',
            questions,
            ...(introMessage ? { introMessage } : {}),
        };
    }
    const plan = parsed.plan;
    if (!isRecord(plan))
        return null;
    return {
        role: 'assistant',
        type: 'plan',
        plan,
    };
}
function createToolCallMessage(toolCall, builderTool) {
    return {
        id: toolCall.id,
        toolCallId: toolCall.id,
        role: 'assistant',
        type: 'tool',
        toolName: toolCall.name,
        displayTitle: builderTool?.displayTitle,
        customDisplayTitle: toolCall.args && builderTool?.getCustomDisplayTitle?.(toolCall.args),
        status: 'completed',
        updates: [
            {
                type: 'input',
                data: toolCall.args || {},
            },
        ],
    };
}
function processToolCalls(toolCalls, builderTools) {
    return toolCalls.map((toolCall) => {
        const builderTool = builderTools?.find((bt) => bt.toolName === toolCall.name);
        return createToolCallMessage(toolCall, builderTool);
    });
}
function processToolMessage(msg, formattedMessages) {
    const toolCallId = msg.tool_call_id;
    for (let i = formattedMessages.length - 1; i >= 0; i--) {
        const m = formattedMessages[i];
        if (m.type === 'tool' && m.id === toolCallId) {
            m.updates ??= [];
            m.updates.push({
                type: 'output',
                data: typeof msg.content === 'string' ? { result: msg.content } : msg.content,
            });
            break;
        }
    }
}
function formatMessages(messages, builderTools) {
    const formattedMessages = [];
    for (const msg of messages) {
        if (msg instanceof messages_1.HumanMessage) {
            formattedMessages.push(formatHumanMessage(msg));
        }
        else if (msg instanceof messages_1.AIMessage) {
            const hitlMessage = tryFormatHitlMessage(msg);
            if (hitlMessage) {
                formattedMessages.push(hitlMessage);
                continue;
            }
            if (msg.tool_calls?.length) {
                formattedMessages.push(...processToolCalls(msg.tool_calls, builderTools));
            }
            else {
                formattedMessages.push(...processAIMessageContent(msg));
            }
        }
        else if (msg instanceof messages_1.ToolMessage) {
            processToolMessage(msg, formattedMessages);
        }
    }
    return formattedMessages;
}
