// @n8n/ai-workflow-builder/dist/utils/subgraph-helpers.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.executeSubgraphTools = executeSubgraphTools;
exports.extractUserRequest = extractUserRequest;
exports.createStandardShouldContinue = createStandardShouldContinue;
exports.extractToolMessagesForPersistence = extractToolMessagesForPersistence;
exports.filterOutSubgraphToolMessages = filterOutSubgraphToolMessages;
const messages_1 = require("@langchain/core/messages");
const langgraph_1 = require("@langchain/langgraph");
const cache_control_1 = require("./cache-control");
const langchain_1 = require("../types/langchain");
function isCommandUpdate(value) {
    if (typeof value !== 'object' || value === null) {
        return false;
    }
    const obj = value;
    if ('messages' in obj && obj.messages !== undefined && !Array.isArray(obj.messages)) {
        return false;
    }
    if ('workflowOperations' in obj &&
        obj.workflowOperations !== undefined &&
        !Array.isArray(obj.workflowOperations)) {
        return false;
    }
    if ('templateIds' in obj && obj.templateIds !== undefined && !Array.isArray(obj.templateIds)) {
        return false;
    }
    if ('cachedTemplates' in obj &&
        obj.cachedTemplates !== undefined &&
        !Array.isArray(obj.cachedTemplates)) {
        return false;
    }
    if ('bestPractices' in obj &&
        obj.bestPractices !== undefined &&
        typeof obj.bestPractices !== 'string') {
        return false;
    }
    return true;
}
async function executeSubgraphTools(state, toolMap) {
    const lastMessage = state.messages[state.messages.length - 1];
    if (!lastMessage || !messages_1.AIMessage.isInstance(lastMessage) || !lastMessage.tool_calls?.length) {
        return {};
    }
    const toolResults = await Promise.all(lastMessage.tool_calls.map(async (toolCall) => {
        const tool = toolMap.get(toolCall.name);
        if (!tool) {
            return new messages_1.ToolMessage({
                content: `Tool ${toolCall.name} not found`,
                tool_call_id: toolCall.id ?? '',
            });
        }
        try {
            const result = await tool.invoke(toolCall.args ?? {}, {
                toolCall: {
                    id: toolCall.id,
                    name: toolCall.name,
                    args: toolCall.args ?? {},
                },
            });
            return result;
        }
        catch (error) {
            if ((0, langgraph_1.isGraphInterrupt)(error)) {
                throw error;
            }
            return new messages_1.ToolMessage({
                content: `Tool failed: ${error instanceof Error ? error.message : 'Unknown error'}`,
                tool_call_id: toolCall.id ?? '',
            });
        }
    }));
    const messages = [];
    const operations = [];
    const templateIds = [];
    const cachedTemplates = [];
    let bestPractices;
    for (const result of toolResults) {
        if ((0, langgraph_1.isCommand)(result)) {
            if (isCommandUpdate(result.update)) {
                if (result.update.messages) {
                    messages.push(...result.update.messages);
                }
                if (result.update.workflowOperations) {
                    operations.push(...result.update.workflowOperations);
                }
                if (result.update.templateIds) {
                    templateIds.push(...result.update.templateIds);
                }
                if (result.update.cachedTemplates) {
                    cachedTemplates.push(...result.update.cachedTemplates);
                }
                if (result.update.bestPractices) {
                    bestPractices = result.update.bestPractices;
                }
            }
        }
        else if ((0, langchain_1.isBaseMessage)(result)) {
            messages.push(result);
        }
    }
    const stateUpdate = {};
    if (messages.length > 0) {
        stateUpdate.messages = messages;
    }
    if (operations.length > 0) {
        stateUpdate.workflowOperations = operations;
    }
    if (templateIds.length > 0) {
        stateUpdate.templateIds = templateIds;
    }
    if (cachedTemplates.length > 0) {
        stateUpdate.cachedTemplates = cachedTemplates;
    }
    if (bestPractices) {
        stateUpdate.bestPractices = bestPractices;
    }
    return stateUpdate;
}
function extractUserRequest(messages, defaultValue = '') {
    const humanMessages = messages.filter((m) => m instanceof messages_1.HumanMessage);
    const lastNonResumeMessage = [...humanMessages]
        .reverse()
        .find((msg) => msg.additional_kwargs?.resumeData === undefined);
    const lastUserMessage = lastNonResumeMessage ?? humanMessages[humanMessages.length - 1];
    return typeof lastUserMessage?.content === 'string' ? lastUserMessage.content : defaultValue;
}
function createStandardShouldContinue() {
    return (state) => {
        const lastMessage = state.messages[state.messages.length - 1];
        const hasToolCalls = lastMessage &&
            'tool_calls' in lastMessage &&
            Array.isArray(lastMessage.tool_calls) &&
            lastMessage.tool_calls.length > 0;
        return hasToolCalls ? 'tools' : langgraph_1.END;
    };
}
function extractToolMessagesForPersistence(messages) {
    const completedToolCallIds = new Set();
    for (const msg of messages) {
        if (messages_1.ToolMessage.isInstance(msg) && msg.tool_call_id) {
            completedToolCallIds.add(msg.tool_call_id);
        }
    }
    const filtered = messages.filter((msg) => {
        if (messages_1.ToolMessage.isInstance(msg)) {
            return true;
        }
        if (messages_1.AIMessage.isInstance(msg) && msg.tool_calls && msg.tool_calls.length > 0) {
            return msg.tool_calls.every((tc) => tc.id && completedToolCallIds.has(tc.id));
        }
        return false;
    });
    (0, cache_control_1.stripAllCacheControlMarkers)(filtered);
    return filtered;
}
function filterOutSubgraphToolMessages(messages) {
    return messages.filter((msg) => {
        if (messages_1.ToolMessage.isInstance(msg)) {
            return false;
        }
        if (messages_1.AIMessage.isInstance(msg) && msg.tool_calls && msg.tool_calls.length > 0) {
            return false;
        }
        return true;
    });
}
