// @n8n/ai-workflow-builder/dist/utils/token-usage-tracking-handler.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TokenUsageTrackingHandler = void 0;
const base_1 = require("@langchain/core/callbacks/base");
function isRecord(value) {
    return typeof value === 'object' && value !== null && !Array.isArray(value);
}
function extractInputTokens(usage) {
    if (typeof usage.input_tokens === 'number') {
        return usage.input_tokens;
    }
    if (typeof usage.prompt_tokens === 'number') {
        return usage.prompt_tokens;
    }
    return 0;
}
function extractOutputTokens(usage) {
    if (typeof usage.output_tokens === 'number') {
        return usage.output_tokens;
    }
    if (typeof usage.completion_tokens === 'number') {
        return usage.completion_tokens;
    }
    return 0;
}
class TokenUsageTrackingHandler extends base_1.BaseCallbackHandler {
    name = 'TokenUsageTrackingHandler';
    totalInputTokens = 0;
    totalOutputTokens = 0;
    async handleLLMEnd(output) {
        if (isRecord(output.llmOutput) && isRecord(output.llmOutput.usage)) {
            this.totalInputTokens += extractInputTokens(output.llmOutput.usage);
            this.totalOutputTokens += extractOutputTokens(output.llmOutput.usage);
            return;
        }
        for (const generation of output.generations.flat()) {
            if (isRecord(generation.generationInfo) && isRecord(generation.generationInfo.usage)) {
                this.totalInputTokens += extractInputTokens(generation.generationInfo.usage);
                this.totalOutputTokens += extractOutputTokens(generation.generationInfo.usage);
            }
        }
    }
    getUsage() {
        return {
            inputTokens: this.totalInputTokens,
            outputTokens: this.totalOutputTokens,
        };
    }
    reset() {
        this.totalInputTokens = 0;
        this.totalOutputTokens = 0;
    }
}
exports.TokenUsageTrackingHandler = TokenUsageTrackingHandler;
