// @n8n/ai-workflow-builder/dist/utils/tool-executor.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.executeToolsInParallel = executeToolsInParallel;
const messages_1 = require("@langchain/core/messages");
const tools_1 = require("@langchain/core/tools");
const langgraph_1 = require("@langchain/langgraph");
const errors_1 = require("../errors");
function isArray(value) {
    return Array.isArray(value);
}
function collectArrayFromUpdates(updates, key) {
    const result = [];
    for (const update of updates) {
        const value = update[key];
        if (isArray(value)) {
            for (const item of value) {
                result.push(item);
            }
        }
    }
    return result;
}
function createToolErrorMessage(toolName, toolCallId, error) {
    const errorMessage = error instanceof Error ? error.message : 'Unknown error occurred';
    const isParsingError = error instanceof tools_1.ToolInputParsingException || errorMessage.includes('expected schema');
    const errorContent = isParsingError
        ? `Invalid input for tool ${toolName}: ${errorMessage}`
        : `Tool ${toolName} failed: ${errorMessage}`;
    return new messages_1.ToolMessage({
        content: errorContent,
        tool_call_id: toolCallId,
        name: toolName,
        additional_kwargs: { error: true },
    });
}
async function executeToolsInParallel(options) {
    const { state, toolMap } = options;
    const lastMessage = state.messages.at(-1);
    if (!lastMessage || !(0, messages_1.isAIMessage)(lastMessage)) {
        const error = new errors_1.WorkflowStateError('Most recent message must be an AIMessage with tool calls');
        throw error;
    }
    const aiMessage = lastMessage;
    if (!aiMessage.tool_calls?.length) {
        const error = new errors_1.WorkflowStateError('AIMessage must have tool calls');
        throw error;
    }
    const toolResults = await Promise.all(aiMessage.tool_calls.map(async (toolCall) => {
        try {
            const tool = toolMap.get(toolCall.name);
            if (!tool) {
                throw new errors_1.ToolExecutionError(`Tool ${toolCall.name} not found`, {
                    toolName: toolCall.name,
                });
            }
            const result = await tool.invoke(toolCall.args ?? {}, {
                toolCall: {
                    id: toolCall.id,
                    name: toolCall.name,
                    args: toolCall.args ?? {},
                },
            });
            return result;
        }
        catch (error) {
            return createToolErrorMessage(toolCall.name, toolCall.id ?? '', error);
        }
    }));
    const allMessages = [];
    const stateUpdates = [];
    toolResults.forEach((result) => {
        if ((0, langgraph_1.isCommand)(result)) {
            const update = result.update;
            if (update) {
                stateUpdates.push(update);
            }
        }
        else {
            allMessages.push(result);
        }
    });
    allMessages.push(...collectArrayFromUpdates(stateUpdates, 'messages'));
    const allOperations = collectArrayFromUpdates(stateUpdates, 'workflowOperations');
    const allTechniqueCategories = collectArrayFromUpdates(stateUpdates, 'techniqueCategories');
    const allValidationHistory = collectArrayFromUpdates(stateUpdates, 'validationHistory');
    const allTemplateIds = collectArrayFromUpdates(stateUpdates, 'templateIds');
    const allCachedTemplates = collectArrayFromUpdates(stateUpdates, 'cachedTemplates');
    const finalUpdate = {
        messages: allMessages,
    };
    if (allOperations.length > 0) {
        finalUpdate.workflowOperations = allOperations;
    }
    if (allTechniqueCategories.length > 0) {
        finalUpdate.techniqueCategories = allTechniqueCategories;
    }
    if (allValidationHistory.length > 0) {
        finalUpdate.validationHistory = allValidationHistory;
    }
    if (allTemplateIds.length > 0) {
        finalUpdate.templateIds = allTemplateIds;
    }
    if (allCachedTemplates.length > 0) {
        finalUpdate.cachedTemplates = allCachedTemplates;
    }
    return finalUpdate;
}
