// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/github/v11/resource_repository/operation_get_issues.ts
/**
 * GitHub Node - Version 1.1
 * Discriminator: resource=repository, operation=getIssues
 */


interface Credentials {
  githubApi: CredentialReference;
  githubOAuth2Api: CredentialReference;
}

/** Returns issues of a repository */
export type GithubV11RepositoryGetIssuesParams = {
  resource: 'repository';
  operation: 'getIssues';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Repository Owner
 * @displayOptions.hide { operation: ["invite", "getUserIssues"] }
 * @default {"mode":"list","value":""}
 */
    owner?: { __rl: true; mode: 'list' | 'url' | 'name'; value: string; cachedResultName?: string };
/**
 * Repository Name
 * @displayOptions.hide { resource: ["user", "organization"], operation: ["getRepositories"] }
 * @default {"mode":"list","value":""}
 */
    repository?: { __rl: true; mode: 'list' | 'url' | 'name'; value: string; cachedResultName?: string };
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    getRepositoryIssuesFilters?: {
    /** Return only issues which are assigned to a specific user
     */
    assignee?: string | Expression<string> | PlaceholderValue;
    /** Return only issues which were created by a specific user
     */
    creator?: string | Expression<string> | PlaceholderValue;
    /** Return only issues in which a specific user was mentioned
     */
    mentioned?: string | Expression<string> | PlaceholderValue;
    /** Return only issues with the given labels. Multiple labels can be separated by comma.
     */
    labels?: string | Expression<string> | PlaceholderValue;
    /** Return only issues updated at or after this time
     */
    since?: string | Expression<string>;
    /** The state to set
     * @default open
     */
    state?: 'all' | 'closed' | 'open' | Expression<string>;
    /** The order the issues should be returned in
     * @default created
     */
    sort?: 'created' | 'updated' | 'comments' | Expression<string>;
    /** The sort order
     * @default desc
     */
    direction?: 'asc' | 'desc' | Expression<string>;
  };
};

export type GithubV11RepositoryGetIssuesOutput = {
  active_lock_reason?: null;
  assignees?: Array<{
    avatar_url?: string;
    events_url?: string;
    followers_url?: string;
    following_url?: string;
    gists_url?: string;
    gravatar_id?: string;
    html_url?: string;
    id?: number;
    login?: string;
    node_id?: string;
    organizations_url?: string;
    received_events_url?: string;
    repos_url?: string;
    site_admin?: boolean;
    starred_url?: string;
    subscriptions_url?: string;
    type?: string;
    url?: string;
    user_view_type?: string;
  }>;
  author_association?: string;
  comments?: number;
  comments_url?: string;
  created_at?: string;
  events_url?: string;
  html_url?: string;
  id?: number;
  labels?: Array<{
    color?: string;
    'default'?: boolean;
    id?: number;
    name?: string;
    node_id?: string;
    url?: string;
  }>;
  labels_url?: string;
  locked?: boolean;
  node_id?: string;
  number?: number;
  performed_via_github_app?: null;
  reactions?: {
    '-1'?: number;
    '+1'?: number;
    confused?: number;
    eyes?: number;
    heart?: number;
    hooray?: number;
    laugh?: number;
    rocket?: number;
    total_count?: number;
    url?: string;
  };
  repository_url?: string;
  state?: string;
  timeline_url?: string;
  title?: string;
  updated_at?: string;
  url?: string;
  user?: {
    avatar_url?: string;
    events_url?: string;
    followers_url?: string;
    following_url?: string;
    gists_url?: string;
    gravatar_id?: string;
    html_url?: string;
    id?: number;
    login?: string;
    node_id?: string;
    organizations_url?: string;
    received_events_url?: string;
    repos_url?: string;
    site_admin?: boolean;
    starred_url?: string;
    subscriptions_url?: string;
    type?: string;
    url?: string;
    user_view_type?: string;
  };
};

export type GithubV11RepositoryGetIssuesNode = {
  type: 'n8n-nodes-base.github';
  version: 1.1;
  credentials?: Credentials;
  config: NodeConfig<GithubV11RepositoryGetIssuesParams>;
  output?: Items<GithubV11RepositoryGetIssuesOutput>;
};
