// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/github/v11/resource_repository/operation_get_pull_requests.ts
/**
 * GitHub Node - Version 1.1
 * Discriminator: resource=repository, operation=getPullRequests
 */


interface Credentials {
  githubApi: CredentialReference;
  githubOAuth2Api: CredentialReference;
}

/** Returns pull requests of a repository */
export type GithubV11RepositoryGetPullRequestsParams = {
  resource: 'repository';
  operation: 'getPullRequests';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Repository Owner
 * @displayOptions.hide { operation: ["invite", "getUserIssues"] }
 * @default {"mode":"list","value":""}
 */
    owner?: { __rl: true; mode: 'list' | 'url' | 'name'; value: string; cachedResultName?: string };
/**
 * Repository Name
 * @displayOptions.hide { resource: ["user", "organization"], operation: ["getRepositories"] }
 * @default {"mode":"list","value":""}
 */
    repository?: { __rl: true; mode: 'list' | 'url' | 'name'; value: string; cachedResultName?: string };
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return. Maximum value is &lt;a href="https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#list-pull-requests"&gt;100&lt;/a&gt;.
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    getRepositoryPullRequestsFilters?: {
    /** The state to set
     * @default open
     */
    state?: 'all' | 'closed' | 'open' | Expression<string>;
    /** The order the pull requests should be returned in
     * @default created
     */
    sort?: 'created' | 'updated' | 'popularity' | 'long-running' | Expression<string>;
    /** The sort order
     * @default desc
     */
    direction?: 'asc' | 'desc' | Expression<string>;
  };
};

export type GithubV11RepositoryGetPullRequestsNode = {
  type: 'n8n-nodes-base.github';
  version: 1.1;
  credentials?: Credentials;
  config: NodeConfig<GithubV11RepositoryGetPullRequestsParams>;
};
