// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/github/v11/resource_repository/operation_get_profile.ts
/**
 * GitHub Node - Version 1.1
 * Discriminator: resource=repository, operation=getProfile
 */


interface Credentials {
  githubApi: CredentialReference;
  githubOAuth2Api: CredentialReference;
}

/** Get the community profile of a repository with metrics, health score, description, license, etc */
export type GithubV11RepositoryGetProfileParams = {
  resource: 'repository';
  operation: 'getProfile';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Repository Owner
 * @displayOptions.hide { operation: ["invite", "getUserIssues"] }
 * @default {"mode":"list","value":""}
 */
    owner?: { __rl: true; mode: 'list' | 'url' | 'name'; value: string; cachedResultName?: string };
/**
 * Repository Name
 * @displayOptions.hide { resource: ["user", "organization"], operation: ["getRepositories"] }
 * @default {"mode":"list","value":""}
 */
    repository?: { __rl: true; mode: 'list' | 'url' | 'name'; value: string; cachedResultName?: string };
};

export type GithubV11RepositoryGetProfileNode = {
  type: 'n8n-nodes-base.github';
  version: 1.1;
  credentials?: Credentials;
  config: NodeConfig<GithubV11RepositoryGetProfileParams>;
};
