// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/onfleet/v1/resource_destination/operation_create.schema.js
/**
 * Onfleet Node - Version 1 - Zod Schema
 * Discriminator: resource=destination, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('destination'),
      operation: z.literal('create'),
      id: stringOrExpression.optional(),
      unparsed: booleanOrExpression.optional(),
      address: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"unparsed":[true]}}, defaults: {"unparsed":false} }),
      addressNumber: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"unparsed":[false]}}, defaults: {"unparsed":false} }),
      addressStreet: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"unparsed":[false]}}, defaults: {"unparsed":false} }),
      addressCity: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"unparsed":[false]}}, defaults: {"unparsed":false} }),
      addressCountry: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"unparsed":[false]}}, defaults: {"unparsed":false} }),
      additionalFields: resolveSchema({ parameters, schema: z.object({ addressApartment: stringOrExpression.optional(), addressName: stringOrExpression.optional(), addressNotes: stringOrExpression.optional(), addressPostalCode: stringOrExpression.optional() }), required: false, displayOptions: {"show":{"unparsed":[true,false]}}, defaults: {"unparsed":false} }),
    }).optional(),
  });
};
