// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/sort/v1.schema.js
/**
 * Sort Node - Version 1 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    type: z.union([z.literal('simple'), z.literal('random'), z.literal('code'), expressionSchema]).optional(),
    sortFieldsUi: resolveSchema({ parameters, schema: z.object({ sortField: z.array(z.object({ fieldName: stringOrExpression.optional(), order: z.union([z.literal('ascending'), z.literal('descending'), expressionSchema]).optional() })).optional() }), required: false, displayOptions: {"show":{"type":["simple"]}}, defaults: {"type":"simple"} }),
    code: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"type":["code"]}}, defaults: {"type":"simple"} }),
    options: resolveSchema({ parameters, schema: z.object({ disableDotNotation: booleanOrExpression.optional() }), required: false, displayOptions: {"show":{"type":["simple"]}}, defaults: {"type":"simple"} }),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
