// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mailchimp/v1/resource_member_tag/operation_create.ts
/**
 * Mailchimp Node - Version 1
 * Discriminator: resource=memberTag, operation=create
 */


interface Credentials {
  mailchimpApi: CredentialReference;
  mailchimpOAuth2Api: CredentialReference;
}

/** Create a new member on list */
export type MailchimpV1MemberTagCreateParams = {
  resource: 'memberTag';
  operation: 'create';
  authentication?: 'apiKey' | 'oAuth2' | Expression<string>;
/**
 * List of lists. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    list?: string | Expression<string>;
/**
 * Email address of the subscriber
 */
    email?: string | Expression<string> | PlaceholderValue;
/**
 * Tags
 * @default []
 */
    tags?: string | Expression<string> | PlaceholderValue;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Whether automations based on the tags in the request will not fire
     * @default false
     */
    isSyncing?: boolean | Expression<boolean>;
  };
};

export type MailchimpV1MemberTagCreateOutput = {
  success?: boolean;
};

export type MailchimpV1MemberTagCreateNode = {
  type: 'n8n-nodes-base.mailchimp';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MailchimpV1MemberTagCreateParams>;
  output?: Items<MailchimpV1MemberTagCreateOutput>;
};
