// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/coda/v1/resource_table/operation_get_row.schema.js
/**
 * Coda Node - Version 1 - Zod Schema
 * Discriminator: resource=table, operation=getRow
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('table').default('table'),
      operation: z.literal('getRow'),
      docId: stringOrExpression.optional(),
      tableId: stringOrExpression.optional(),
      rowId: stringOrExpression.optional(),
      options: z.object({ rawData: booleanOrExpression.optional(), useColumnNames: booleanOrExpression.optional(), valueFormat: z.union([z.literal('simple'), z.literal('simpleWithArrays'), z.literal('rich'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
