// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/strava/v1/resource_activity/operation_create.schema.js
/**
 * Strava Node - Version 1 - Zod Schema
 * Discriminator: resource=activity, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('activity').default('activity'),
      operation: z.literal('create').default('create'),
      name: stringOrExpression.optional(),
      type: stringOrExpression.optional(),
      startDate: stringOrExpression.optional(),
      elapsedTime: numberOrExpression.optional(),
      additionalFields: z.object({ commute: booleanOrExpression.optional(), description: stringOrExpression.optional(), distance: numberOrExpression.optional(), trainer: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
