// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/wordpress/v1/resource_page/operation_update.schema.js
/**
 * Wordpress Node - Version 1 - Zod Schema
 * Discriminator: resource=page, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('page'),
      operation: z.literal('update'),
      pageId: stringOrExpression.optional(),
      updateFields: z.object({ authorId: stringOrExpression.optional(), parent: numberOrExpression.optional(), title: stringOrExpression.optional(), content: stringOrExpression.optional(), slug: stringOrExpression.optional(), password: stringOrExpression.optional(), status: z.union([z.literal('draft'), z.literal('future'), z.literal('pending'), z.literal('private'), z.literal('publish'), expressionSchema]).optional(), commentStatus: z.union([z.literal('open'), z.literal('closed'), expressionSchema]).optional(), pingStatus: z.union([z.literal('open'), z.literal('closed'), expressionSchema]).optional(), pageTemplate: z.unknown().optional(), menuOrder: numberOrExpression.optional(), commentStatus: z.union([z.literal('open'), z.literal('closed'), expressionSchema]).optional(), featuredMediaId: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
