// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/wordpress/v1/resource_page/operation_get.ts
/**
 * Wordpress Node - Version 1
 * Discriminator: resource=page, operation=get
 */


interface Credentials {
  wordpressApi: CredentialReference;
}

/** Get a post */
export type WordpressV1PageGetParams = {
  resource: 'page';
  operation: 'get';
/**
 * Unique identifier for the object
 */
    pageId?: string | Expression<string> | PlaceholderValue;
/**
 * Options
 * @default {}
 */
    options?: {
    /** The password for the page if it is password protected
     */
    password?: string | Expression<string> | PlaceholderValue;
    /** Scope under which the request is made; determines fields present in response
     * @default view
     */
    context?: 'view' | 'embed' | 'edit' | Expression<string>;
  };
};

export type WordpressV1PageGetOutput = {
  _links?: {
    about?: Array<{
      href?: string;
    }>;
    author?: Array<{
      embeddable?: boolean;
      href?: string;
    }>;
    collection?: Array<{
      href?: string;
    }>;
    curies?: Array<{
      href?: string;
      name?: string;
      templated?: boolean;
    }>;
    'predecessor-version'?: Array<{
      href?: string;
      id?: number;
    }>;
    replies?: Array<{
      embeddable?: boolean;
      href?: string;
    }>;
    self?: Array<{
      href?: string;
      targetHints?: {
        allow?: Array<string>;
      };
    }>;
    'version-history'?: Array<{
      count?: number;
      href?: string;
    }>;
    'wp:attachment'?: Array<{
      href?: string;
    }>;
  };
  author?: number;
  class_list?: Array<string>;
  comment_status?: string;
  content?: {
    protected?: boolean;
    rendered?: string;
  };
  date?: string;
  date_gmt?: string;
  excerpt?: {
    protected?: boolean;
    rendered?: string;
  };
  featured_media?: number;
  guid?: {
    rendered?: string;
  };
  link?: string;
  menu_order?: number;
  meta?: {
    footnotes?: string;
  };
  modified?: string;
  modified_gmt?: string;
  parent?: number;
  ping_status?: string;
  slug?: string;
  status?: string;
  template?: string;
  title?: {
    rendered?: string;
  };
  type?: string;
};

export type WordpressV1PageGetNode = {
  type: 'n8n-nodes-base.wordpress';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<WordpressV1PageGetParams>;
  output?: Items<WordpressV1PageGetOutput>;
};
