// @n8n/ai-workflow-builder/dist/validation/checks/credentials.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.validateCredentials = validateCredentials;
const n8n_workflow_1 = require("n8n-workflow");
const CREDENTIAL_FIELD_PATTERNS = [
    /api[_-]?key/i,
    /access[_-]?token/i,
    /auth[_-]?token/i,
    /bearer[_-]?token/i,
    /secret[_-]?key/i,
    /private[_-]?key/i,
    /client[_-]?secret/i,
    /password/i,
    /credentials?/i,
    /^token$/i,
    /^secret$/i,
    /^auth$/i,
];
const SENSITIVE_HEADERS = new Set([
    'authorization',
    'x-api-key',
    'x-auth-token',
    'x-access-token',
    'api-key',
    'apikey',
]);
function isCredentialFieldName(name) {
    return CREDENTIAL_FIELD_PATTERNS.some((pattern) => pattern.test(name));
}
function isSensitiveHeader(headerName) {
    return SENSITIVE_HEADERS.has(headerName.toLowerCase());
}
function isParameterArray(arr) {
    return (Array.isArray(arr) &&
        arr.every((item) => typeof item === 'object' && item !== null && !Array.isArray(item)));
}
function isParametersContainer(value) {
    if (typeof value !== 'object' || value === null)
        return false;
    const obj = value;
    return !('parameters' in obj) || isParameterArray(obj.parameters);
}
function isAssignmentsContainer(value) {
    if (typeof value !== 'object' || value === null)
        return false;
    const obj = value;
    return !('assignments' in obj) || isParameterArray(obj.assignments);
}
function getHeaderParameters(parameters) {
    const headerParams = parameters.headerParameters;
    if (!isParametersContainer(headerParams))
        return [];
    return headerParams.parameters ?? [];
}
function getQueryParameters(parameters) {
    const queryParams = parameters.queryParameters;
    if (!isParametersContainer(queryParams))
        return [];
    return queryParams.parameters ?? [];
}
function getSetNodeAssignments(parameters) {
    const assignments = parameters.assignments;
    if (!isAssignmentsContainer(assignments))
        return [];
    return assignments.assignments ?? [];
}
function hasHardcodedCredentialValue(param) {
    return Boolean(param.value) && !(0, n8n_workflow_1.isExpression)(param.value);
}
function validateHttpRequestNode(node, violations) {
    const params = node.parameters ?? {};
    for (const header of getHeaderParameters(params)) {
        if (header.name && isSensitiveHeader(header.name) && hasHardcodedCredentialValue(header)) {
            violations.push({
                name: 'http-request-hardcoded-credentials',
                type: 'minor',
                description: `HTTP Request node "${node.name}" has a hardcoded value for sensitive header "${header.name}". Use n8n credentials instead (e.g., httpHeaderAuth, httpBearerAuth).`,
                pointsDeducted: 5,
            });
        }
    }
    for (const param of getQueryParameters(params)) {
        if (param.name && isCredentialFieldName(param.name) && hasHardcodedCredentialValue(param)) {
            violations.push({
                name: 'http-request-hardcoded-credentials',
                type: 'minor',
                description: `HTTP Request node "${node.name}" has a hardcoded value for credential-like query parameter "${param.name}". Use n8n credentials instead (e.g., httpQueryAuth).`,
                pointsDeducted: 5,
            });
        }
    }
}
function validateSetNode(node, violations) {
    const params = node.parameters ?? {};
    for (const assignment of getSetNodeAssignments(params)) {
        if (assignment.name && isCredentialFieldName(assignment.name)) {
            violations.push({
                name: 'set-node-credential-field',
                type: 'minor',
                description: `Set node "${node.name}" has a field named "${assignment.name}" which appears to be storing credentials. Credentials should be stored securely using n8n's credential system, not in workflow data.`,
                pointsDeducted: 5,
            });
        }
    }
}
function validateCredentials(workflow) {
    const violations = [];
    if (!workflow.nodes || workflow.nodes.length === 0) {
        return violations;
    }
    for (const node of workflow.nodes) {
        if (node.type === 'n8n-nodes-base.httpRequest') {
            validateHttpRequestNode(node, violations);
        }
        else if (node.type === 'n8n-nodes-base.set') {
            validateSetNode(node, violations);
        }
    }
    return violations;
}
