// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/twilio/v1/resource_call/operation_make.schema.js
/**
 * Twilio Node - Version 1 - Zod Schema
 * Discriminator: resource=call, operation=make
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('call'),
      operation: z.literal('make'),
      from: stringOrExpression.optional(),
      to: stringOrExpression.optional(),
      twiml: booleanOrExpression.optional(),
      message: stringOrExpression.optional(),
      options: z.object({ statusCallback: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
