// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/discord/v2/resource_channel/operation_update.ts
/**
 * Discord Node - Version 2
 * Discriminator: resource=channel, operation=update
 */


interface Credentials {
  discordBotApi: CredentialReference;
  discordOAuth2Api: CredentialReference;
  discordWebhookApi?: CredentialReference;
}

/** Update a channel */
export type DiscordV2ChannelUpdateParams = {
  resource: 'channel';
  operation: 'update';
  authentication?: 'botToken' | 'oAuth2' | 'webhook' | Expression<string>;
/**
 * Select the server (guild) that your bot is connected to
 * @displayOptions.show { authentication: ["botToken", "oAuth2"] }
 * @default {"mode":"list","value":""}
 */
    guildId?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
/**
 * Select the channel by name, URL, or ID
 * @displayOptions.hide { authentication: ["webhook"] }
 * @default {"mode":"list","value":""}
 */
    channelId?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
/**
 * The new name of the channel. Fill this field only if you want to change the channel's name.
 * @displayOptions.hide { authentication: ["webhook"] }
 */
    name?: string | Expression<string> | PlaceholderValue;
/**
 * Options
 * @displayOptions.hide { authentication: ["webhook"] }
 * @default {}
 */
    options?: {
    /** Whether the content of the channel might be nsfw (not safe for work)
     * @default false
     */
    nsfw?: boolean | Expression<boolean>;
    /** The bitrate (in bits) of the voice channel
     * @hint Only applicable to voice channels
     * @default 8000
     */
    bitrate?: number | Expression<number>;
    /** The parent category where you want the channel to appear
     * @default {"mode":"list","value":""}
     */
    categoryId?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
    /** Position
     * @default 1
     */
    position?: number | Expression<number>;
    /** Amount of seconds a user has to wait before sending another message
     * @default 0
     */
    rate_limit_per_user?: number | Expression<number>;
    /** The channel topic description (0-1024 characters)
     */
    topic?: string | Expression<string> | PlaceholderValue;
    /** The limit for the number of members that can be in the channel (0 refers to no limit)
     * @hint Only applicable to voice channels
     * @default 0
     */
    user_limit?: number | Expression<number>;
  };
};

export type DiscordV2ChannelUpdateNode = {
  type: 'n8n-nodes-base.discord';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<DiscordV2ChannelUpdateParams>;
};
