// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/discord/v2/resource_channel/operation_create.ts
/**
 * Discord Node - Version 2
 * Discriminator: resource=channel, operation=create
 */


interface Credentials {
  discordBotApi: CredentialReference;
  discordOAuth2Api: CredentialReference;
  discordWebhookApi?: CredentialReference;
}

/** Create a new channel */
export type DiscordV2ChannelCreateParams = {
  resource: 'channel';
  operation: 'create';
  authentication?: 'botToken' | 'oAuth2' | 'webhook' | Expression<string>;
/**
 * Select the server (guild) that your bot is connected to
 * @displayOptions.show { authentication: ["botToken", "oAuth2"] }
 * @default {"mode":"list","value":""}
 */
    guildId?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
/**
 * The name of the channel
 * @displayOptions.hide { authentication: ["webhook"] }
 */
    name?: string | Expression<string> | PlaceholderValue;
/**
 * The type of channel to create
 * @displayOptions.hide { authentication: ["webhook"] }
 * @default 0
 */
    type?: '0' | '2' | '4' | Expression<string>;
/**
 * Options
 * @displayOptions.hide { authentication: ["webhook"] }
 * @default {}
 */
    options?: {
    /** Whether the content of the channel might be nsfw (not safe for work)
     * @displayOptions.hide { /type: ["4"] }
     * @default false
     */
    nsfw?: boolean | Expression<boolean>;
    /** The bitrate (in bits) of the voice channel
     * @displayOptions.show { /type: ["2"] }
     * @default 8000
     */
    bitrate?: number | Expression<number>;
    /** The parent category where you want the channel to appear
     * @displayOptions.hide { /type: ["4"] }
     * @default {"mode":"list","value":""}
     */
    categoryId?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
    /** Position
     * @default 1
     */
    position?: number | Expression<number>;
    /** Amount of seconds a user has to wait before sending another message
     * @displayOptions.hide { /type: ["4"] }
     * @default 0
     */
    rate_limit_per_user?: number | Expression<number>;
    /** The channel topic description (0-1024 characters)
     * @displayOptions.hide { /type: ["4"] }
     */
    topic?: string | Expression<string> | PlaceholderValue;
    /** The limit for the number of members that can be in the channel (0 refers to no limit)
     * @displayOptions.show { /type: ["2"] }
     * @default 0
     */
    user_limit?: number | Expression<number>;
  };
};

export type DiscordV2ChannelCreateOutput = {
  flags?: number;
  guild_id?: string;
  id?: string;
  last_message_id?: null;
  name?: string;
  nsfw?: boolean;
  permission_overwrites?: Array<{
    allow?: string;
    deny?: string;
    id?: string;
    type?: number;
  }>;
  position?: number;
  rate_limit_per_user?: number;
  type?: number;
};

export type DiscordV2ChannelCreateNode = {
  type: 'n8n-nodes-base.discord';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<DiscordV2ChannelCreateParams>;
  output?: Items<DiscordV2ChannelCreateOutput>;
};
