// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/discord/v2/resource_channel/operation_get.ts
/**
 * Discord Node - Version 2
 * Discriminator: resource=channel, operation=get
 */


interface Credentials {
  discordBotApi: CredentialReference;
  discordOAuth2Api: CredentialReference;
  discordWebhookApi?: CredentialReference;
}

/** Get a message in a channel */
export type DiscordV2ChannelGetParams = {
  resource: 'channel';
  operation: 'get';
  authentication?: 'botToken' | 'oAuth2' | 'webhook' | Expression<string>;
/**
 * Select the server (guild) that your bot is connected to
 * @displayOptions.show { authentication: ["botToken", "oAuth2"] }
 * @default {"mode":"list","value":""}
 */
    guildId?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
/**
 * Select the channel by name, URL, or ID
 * @displayOptions.hide { authentication: ["webhook"] }
 * @default {"mode":"list","value":""}
 */
    channelId?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
};

export type DiscordV2ChannelGetOutput = {
  flags?: number;
  guild_id?: string;
  id?: string;
  name?: string;
  nsfw?: boolean;
  permission_overwrites?: Array<{
    allow?: string;
    deny?: string;
    id?: string;
    type?: number;
  }>;
  position?: number;
  rate_limit_per_user?: number;
  type?: number;
};

export type DiscordV2ChannelGetNode = {
  type: 'n8n-nodes-base.discord';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<DiscordV2ChannelGetParams>;
  output?: Items<DiscordV2ChannelGetOutput>;
};
