// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/discord/v2/resource_channel/operation_get_all.ts
/**
 * Discord Node - Version 2
 * Discriminator: resource=channel, operation=getAll
 */


interface Credentials {
  discordBotApi: CredentialReference;
  discordOAuth2Api: CredentialReference;
  discordWebhookApi?: CredentialReference;
}

/** Retrieve the latest messages in a channel */
export type DiscordV2ChannelGetAllParams = {
  resource: 'channel';
  operation: 'getAll';
  authentication?: 'botToken' | 'oAuth2' | 'webhook' | Expression<string>;
/**
 * Select the server (guild) that your bot is connected to
 * @displayOptions.show { authentication: ["botToken", "oAuth2"] }
 * @default {"mode":"list","value":""}
 */
    guildId?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
/**
 * Whether to return all results or only up to a given limit
 * @displayOptions.hide { authentication: ["webhook"] }
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @displayOptions.hide { authentication: ["webhook"] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Options
 * @displayOptions.hide { authentication: ["webhook"] }
 * @default {}
 */
    options?: {
    /** Filter by Type
     * @default []
     */
    filter?: Array<0 | 2 | 4>;
  };
};

export type DiscordV2ChannelGetAllOutput = {
  flags?: number;
  guild_id?: string;
  icon_emoji?: {
    id?: null;
    name?: string;
  };
  id?: string;
  name?: string;
  nsfw?: boolean;
  permission_overwrites?: Array<{
    allow?: string;
    deny?: string;
    id?: string;
    type?: number;
  }>;
  position?: number;
  rate_limit_per_user?: number;
  theme_color?: null;
  type?: number;
};

export type DiscordV2ChannelGetAllNode = {
  type: 'n8n-nodes-base.discord';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<DiscordV2ChannelGetAllParams>;
  output?: Items<DiscordV2ChannelGetAllOutput>;
};
