// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/sentryIo/v1/resource_project/operation_update.ts
/**
 * Sentry.io Node - Version 1
 * Discriminator: resource=project, operation=update
 */


interface Credentials {
  sentryIoOAuth2Api: CredentialReference;
  sentryIoApi: CredentialReference;
  sentryIoServerApi: CredentialReference;
}

/** Update an issue */
export type SentryIoV1ProjectUpdateParams = {
  resource: 'project';
  operation: 'update';
  authentication?: 'accessToken' | 'oAuth2' | 'accessTokenServer' | Expression<string>;
/**
 * The slug of the organization the project belong to. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    organizationSlug?: string | Expression<string>;
/**
 * The slug of the project to update. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    projectSlug?: string | Expression<string>;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** The new platform for the updated project
     * @default false
     */
    isBookmarked?: boolean | Expression<boolean>;
    /** Maximum interval to digest alerts
     * @default 1800
     */
    digestsMaxDelay?: number | Expression<number>;
    /** Minium interval to digest alerts
     * @default 60
     */
    digestsMinDelay?: number | Expression<number>;
    /** The new name for the updated project
     */
    name?: string | Expression<string> | PlaceholderValue;
    /** The new slug for the updated project
     */
    slug?: string | Expression<string> | PlaceholderValue;
    /** The new team name
     */
    team?: string | Expression<string> | PlaceholderValue;
    /** The new platform for the updated project
     */
    platform?: string | Expression<string> | PlaceholderValue;
  };
};

export type SentryIoV1ProjectUpdateNode = {
  type: 'n8n-nodes-base.sentryIo';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SentryIoV1ProjectUpdateParams>;
};
