// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/awsIam/v1/resource_group/operation_get_all.ts
/**
 * AWS IAM Node - Version 1
 * Discriminator: resource=group, operation=getAll
 */


interface Credentials {
  aws: CredentialReference;
}

/** Retrieve a list of users */
export type AwsIamV1GroupGetAllParams = {
  resource: 'group';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.hide { returnAll: [true] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Whether to include a list of users in the group
 * @default false
 */
    includeUsers?: boolean | Expression<boolean>;
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type AwsIamV1GroupGetAllNode = {
  type: 'n8n-nodes-base.awsIam';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<AwsIamV1GroupGetAllParams>;
};
