// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/quickbooks/v1/resource_vendor/operation_update.schema.js
/**
 * QuickBooks Online Node - Version 1 - Zod Schema
 * Discriminator: resource=vendor, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('vendor'),
      operation: z.literal('update'),
      vendorId: stringOrExpression.optional(),
      updateFields: z.object({ AcctNum: stringOrExpression.optional(), Active: booleanOrExpression.optional(), Balance: numberOrExpression.optional(), BillAddr: z.unknown().optional(), CompanyName: stringOrExpression.optional(), FamilyName: stringOrExpression.optional(), GivenName: stringOrExpression.optional(), PrimaryEmailAddr: stringOrExpression.optional(), PrimaryPhone: stringOrExpression.optional(), PrintOnCheckName: stringOrExpression.optional(), Vendor1099: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
