// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/freshservice/v1/resource_agent_group/operation_update.ts
/**
 * Freshservice Node - Version 1
 * Discriminator: resource=agentGroup, operation=update
 */


interface Credentials {
  freshserviceApi: CredentialReference;
}

/** Update an agent */
export type FreshserviceV1AgentGroupUpdateParams = {
  resource: 'agentGroup';
  operation: 'update';
/**
 * ID of the agent group to update
 */
    agentGroupId?: string | Expression<string> | PlaceholderValue;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** Description
     */
    description?: string | Expression<string> | PlaceholderValue;
    /** ID of the agent to whom an escalation email is sent if a ticket in this group is unassigned. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    escalate_to?: string | Expression<string>;
    /** Comma-separated IDs of agents who are members of this group. Choose from the list or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;. Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     * @default []
     */
    members?: string[];
    /** Name
     */
    name?: string | Expression<string> | PlaceholderValue;
    /** Comma-separated agent user IDs who are observers of this group. Choose from the list or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;. Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     * @default []
     */
    observers?: string[];
    /** Time after which an escalation email is sent if a ticket in the group remains unassigned
     * @default 30m
     */
    unassigned_for?: '1d' | '1h' | '12h' | '2d' | '2h' | '3d' | '30m' | '8h' | Expression<string>;
  };
};

export type FreshserviceV1AgentGroupUpdateNode = {
  type: 'n8n-nodes-base.freshservice';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<FreshserviceV1AgentGroupUpdateParams>;
};
