// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/freshservice/v1/resource_agent_group/operation_update.schema.js
/**
 * Freshservice Node - Version 1 - Zod Schema
 * Discriminator: resource=agentGroup, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('agentGroup'),
      operation: z.literal('update'),
      agentGroupId: stringOrExpression.optional(),
      updateFields: z.object({ description: stringOrExpression.optional(), escalate_to: stringOrExpression.optional(), members: z.array(z.string()).optional(), name: stringOrExpression.optional(), observers: z.array(z.string()).optional(), unassigned_for: z.union([z.literal('1d'), z.literal('1h'), z.literal('12h'), z.literal('2d'), z.literal('2h'), z.literal('3d'), z.literal('30m'), z.literal('8h'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
