// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/hubspot/v1/resource_form/operation_submit.schema.js
/**
 * HubSpot Node - Version 1 - Zod Schema
 * Discriminator: resource=form, operation=submit
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('form'),
      operation: z.literal('submit'),
      authentication: z.union([z.literal('apiKey'), z.literal('appToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      formId: stringOrExpression.optional(),
      additionalFields: z.object({ skipValidation: booleanOrExpression.optional(), submittedAt: stringOrExpression.optional() }).optional(),
      contextUi: z.object({ contextValue: z.object({ hutk: stringOrExpression.optional(), ipAddress: stringOrExpression.optional(), pageUri: stringOrExpression.optional(), pageName: stringOrExpression.optional(), pageId: stringOrExpression.optional(), sfdcCampaignId: stringOrExpression.optional(), goToWebinarWebinarKey: stringOrExpression.optional() }).optional() }).optional(),
      lengalConsentUi: z.object({ lengalConsentValues: z.object({ consentToProcess: booleanOrExpression.optional(), text: stringOrExpression.optional(), communicationsUi: z.unknown().optional() }).optional(), legitimateInterestValues: z.object({ subscriptionTypeId: stringOrExpression.optional(), value: booleanOrExpression.optional(), legalBasis: z.union([z.literal('CUSTOMER'), z.literal('LEAD'), expressionSchema]).optional(), text: stringOrExpression.optional() }).optional() }).optional(),
    }).optional(),
  });
};
