// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/freshservice/v1/resource_problem/operation_create.ts
/**
 * Freshservice Node - Version 1
 * Discriminator: resource=problem, operation=create
 */


interface Credentials {
  freshserviceApi: CredentialReference;
}

/** Create an agent */
export type FreshserviceV1ProblemCreateParams = {
  resource: 'problem';
  operation: 'create';
/**
 * Subject
 */
    subject?: string | Expression<string> | PlaceholderValue;
/**
 * ID of the initiator of the problem. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    requesterId?: string | Expression<string>;
/**
 * Date when the problem is due to be solved
 */
    dueBy?: string | Expression<string>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** ID of the agent to whom the problem is assigned. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    agent_id?: string | Expression<string>;
    /** ID of the department initiating the problem. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    department_id?: string | Expression<string>;
    /** HTML supported
     */
    description?: string | Expression<string> | PlaceholderValue;
    /** ID of the agent group to which the problem is assigned. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    group_id?: string | Expression<string>;
    /** Impact
     * @default 1
     */
    impact?: 1 | 2 | 3 | Expression<number>;
    /** Priority
     * @default 1
     */
    priority?: 1 | 2 | 3 | 4 | Expression<number>;
    /** Status
     * @default 1
     */
    status?: 1 | 2 | 3 | Expression<number>;
  };
};

export type FreshserviceV1ProblemCreateNode = {
  type: 'n8n-nodes-base.freshservice';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<FreshserviceV1ProblemCreateParams>;
};
