// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/twist/v1/resource_thread/operation_get_all.ts
/**
 * Twist Node - Version 1
 * Discriminator: resource=thread, operation=getAll
 */


interface Credentials {
  twistOAuth2Api: CredentialReference;
}

/** Get many channels */
export type TwistV1ThreadGetAllParams = {
  resource: 'thread';
  operation: 'getAll';
/**
 * The ID of the channel
 */
    channelId?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Whether only the IDs of the threads are returned
     * @default false
     */
    as_ids?: boolean | Expression<boolean>;
    /** A filter can be one of &lt;code&gt;attached_to_me&lt;/code&gt;, &lt;code&gt;everyone&lt;/code&gt; and &lt;code&gt;is_starred&lt;/code&gt;
     */
    filter_by?: 'attached_to_me' | 'everyone' | 'is_starred' | Expression<string>;
    /** Limits threads to those newer when the specified Unix time
     */
    newer_than_ts?: string | Expression<string>;
    /** Limits threads to those older than the specified Unix time
     */
    older_than_ts?: string | Expression<string>;
  };
};

export type TwistV1ThreadGetAllNode = {
  type: 'n8n-nodes-base.twist';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<TwistV1ThreadGetAllParams>;
};
