// @n8n/n8n-nodes-langchain/dist/nodes/llms/LmChatGoogleGemini/LmChatGoogleGemini.node.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LmChatGoogleGemini = void 0;
const google_genai_1 = require("@langchain/google-genai");
const n8n_workflow_1 = require("n8n-workflow");
const additional_options_1 = require("../gemini-common/additional-options");
const ai_utilities_1 = require("@n8n/ai-utilities");
function errorDescriptionMapper(error) {
    if (error.description?.includes('properties: should be non-empty for OBJECT type')) {
        return 'Google Gemini requires at least one <a href="https://docs.n8n.io/advanced-ai/examples/using-the-fromai-function/" target="_blank">dynamic parameter</a> when using tools';
    }
    return error.description ?? 'Unknown error';
}
class LmChatGoogleGemini {
    constructor() {
        this.description = {
            displayName: 'Google Gemini Chat Model',
            name: 'lmChatGoogleGemini',
            icon: 'file:google.svg',
            group: ['transform'],
            version: 1,
            description: 'Chat Model Google Gemini',
            defaults: {
                name: 'Google Gemini Chat Model',
            },
            codex: {
                categories: ['AI'],
                subcategories: {
                    AI: ['Language Models', 'Root Nodes'],
                    'Language Models': ['Chat Models (Recommended)'],
                },
                resources: {
                    primaryDocumentation: [
                        {
                            url: 'https://docs.n8n.io/integrations/builtin/cluster-nodes/sub-nodes/n8n-nodes-langchain.lmchatgooglegemini/',
                        },
                    ],
                },
            },
            inputs: [],
            outputs: [n8n_workflow_1.NodeConnectionTypes.AiLanguageModel],
            outputNames: ['Model'],
            credentials: [
                {
                    name: 'googlePalmApi',
                    required: true,
                },
            ],
            requestDefaults: {
                ignoreHttpStatusErrors: true,
                baseURL: '={{ $credentials.host }}',
            },
            properties: [
                (0, ai_utilities_1.getConnectionHintNoticeField)([n8n_workflow_1.NodeConnectionTypes.AiChain, n8n_workflow_1.NodeConnectionTypes.AiAgent]),
                {
                    displayName: 'Model',
                    name: 'modelName',
                    type: 'options',
                    description: 'The model which will generate the completion. <a href="https://developers.generativeai.google/api/rest/generativelanguage/models/list">Learn more</a>.',
                    typeOptions: {
                        loadOptions: {
                            routing: {
                                request: {
                                    method: 'GET',
                                    url: '/v1beta/models',
                                },
                                output: {
                                    postReceive: [
                                        {
                                            type: 'rootProperty',
                                            properties: {
                                                property: 'models',
                                            },
                                        },
                                        {
                                            type: 'filter',
                                            properties: {
                                                pass: "={{ !$responseItem.name.includes('embedding') }}",
                                            },
                                        },
                                        {
                                            type: 'setKeyValue',
                                            properties: {
                                                name: '={{$responseItem.name}}',
                                                value: '={{$responseItem.name}}',
                                                description: '={{$responseItem.description}}',
                                            },
                                        },
                                        {
                                            type: 'sort',
                                            properties: {
                                                key: 'name',
                                            },
                                        },
                                    ],
                                },
                            },
                        },
                    },
                    routing: {
                        send: {
                            type: 'body',
                            property: 'model',
                        },
                    },
                    default: 'models/gemini-2.5-flash',
                },
                (0, additional_options_1.getAdditionalOptions)({ supportsThinkingBudget: false }),
            ],
        };
    }
    async supplyData(itemIndex) {
        const credentials = await this.getCredentials('googlePalmApi');
        const modelName = this.getNodeParameter('modelName', itemIndex);
        const options = this.getNodeParameter('options', itemIndex, {
            maxOutputTokens: 1024,
            temperature: 0.7,
            topK: 40,
            topP: 0.9,
        });
        const safetySettings = this.getNodeParameter('options.safetySettings.values', itemIndex, null);
        const model = new google_genai_1.ChatGoogleGenerativeAI({
            apiKey: credentials.apiKey,
            baseUrl: credentials.host,
            model: modelName,
            topK: options.topK,
            topP: options.topP,
            temperature: options.temperature,
            maxOutputTokens: options.maxOutputTokens,
            safetySettings,
            callbacks: [new ai_utilities_1.N8nLlmTracing(this, { errorDescriptionMapper })],
            onFailedAttempt: (0, ai_utilities_1.makeN8nLlmFailedAttemptHandler)(this),
        });
        return {
            response: model,
        };
    }
}
exports.LmChatGoogleGemini = LmChatGoogleGemini;
