// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/gitlab/v1/resource_release/operation_update.schema.js
/**
 * GitLab Node - Version 1 - Zod Schema
 * Discriminator: resource=release, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('release'),
      operation: z.literal('update'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      owner: stringOrExpression.optional(),
      repository: stringOrExpression.optional(),
      projectId: stringOrExpression.optional(),
      tag_name: stringOrExpression.optional(),
      additionalFields: z.object({ name: stringOrExpression.optional(), description: stringOrExpression.optional(), milestones: stringOrExpression.optional(), released_at: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
