// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/gitlab/v1/resource_release/operation_update.ts
/**
 * GitLab Node - Version 1
 * Discriminator: resource=release, operation=update
 */


interface Credentials {
  gitlabApi: CredentialReference;
  gitlabOAuth2Api: CredentialReference;
}

/** Update a release */
export type GitlabV1ReleaseUpdateParams = {
  resource: 'release';
  operation: 'update';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * User, group or namespace of the project
 */
    owner?: string | Expression<string> | PlaceholderValue;
/**
 * The name of the project
 * @displayOptions.hide { resource: ["user"], operation: ["getRepositories"] }
 */
    repository?: string | Expression<string> | PlaceholderValue;
/**
 * The ID or URL-encoded path of the project
 */
    projectId?: string | Expression<string> | PlaceholderValue;
/**
 * The Git tag the release is associated with
 */
    tag_name?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** The release name
     */
    name?: string | Expression<string> | PlaceholderValue;
    /** The description of the release. You can use Markdown.
     */
    description?: string | Expression<string> | PlaceholderValue;
    /** The title of each milestone to associate with the release (provide a titles list spearated with comma)
     */
    milestones?: string | Expression<string> | PlaceholderValue;
    /** The date when the release is/was ready
     */
    released_at?: string | Expression<string>;
  };
};

export type GitlabV1ReleaseUpdateNode = {
  type: 'n8n-nodes-base.gitlab';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GitlabV1ReleaseUpdateParams>;
};
