// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/gitlab/v1/resource_release/operation_get_all.ts
/**
 * GitLab Node - Version 1
 * Discriminator: resource=release, operation=getAll
 */


interface Credentials {
  gitlabApi: CredentialReference;
  gitlabOAuth2Api: CredentialReference;
}

/** Get many releases */
export type GitlabV1ReleaseGetAllParams = {
  resource: 'release';
  operation: 'getAll';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * User, group or namespace of the project
 */
    owner?: string | Expression<string> | PlaceholderValue;
/**
 * The name of the project
 * @displayOptions.hide { resource: ["user"], operation: ["getRepositories"] }
 */
    repository?: string | Expression<string> | PlaceholderValue;
/**
 * The ID or URL-encoded path of the project
 */
    projectId?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 20
 */
    limit?: number | Expression<number>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** The field to use as order
     * @default released_at
     */
    order_by?: 'created_at' | 'released_at' | Expression<string>;
    /** The direction of the order. .
     * @default desc
     */
    sort?: 'asc' | 'desc' | Expression<string>;
  };
};

export type GitlabV1ReleaseGetAllNode = {
  type: 'n8n-nodes-base.gitlab';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GitlabV1ReleaseGetAllParams>;
};
