// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/misp/v1/resource_feed/operation_update.ts
/**
 * MISP Node - Version 1
 * Discriminator: resource=feed, operation=update
 */


interface Credentials {
  mispApi: CredentialReference;
}

export type MispV1FeedUpdateParams = {
  resource: 'feed';
  operation: 'update';
/**
 * ID of the feed to update
 */
    feedId?: string | Expression<string> | PlaceholderValue;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** Who will be able to see this event once published
     * @default 0
     */
    distribution?: 3 | 2 | 5 | 4 | 1 | 0 | Expression<number>;
    /** Name
     */
    name?: string | Expression<string> | PlaceholderValue;
    /** Provider
     */
    provider?: string | Expression<string> | PlaceholderValue;
    /** Filter rules for the feed
     */
    rules?: IDataObject | string | Expression<string>;
    /** URL
     */
    url?: string | Expression<string> | PlaceholderValue;
  };
};

export type MispV1FeedUpdateNode = {
  type: 'n8n-nodes-base.misp';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MispV1FeedUpdateParams>;
};
