// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/odoo/v1/resource_opportunity/operation_update.schema.js
/**
 * Odoo Node - Version 1 - Zod Schema
 * Discriminator: resource=opportunity, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('opportunity'),
      operation: z.literal('update'),
      opportunityId: stringOrExpression.optional(),
      updateFields: z.object({ email_from: stringOrExpression.optional(), expected_revenue: numberOrExpression.optional(), description: stringOrExpression.optional(), name: stringOrExpression.optional(), phone: stringOrExpression.optional(), priority: z.union([z.literal('1'), z.literal('2'), z.literal('3'), expressionSchema]).optional(), probability: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
