// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/odoo/v1/resource_opportunity/operation_get.ts
/**
 * Odoo Node - Version 1
 * Discriminator: resource=opportunity, operation=get
 */


interface Credentials {
  odooApi: CredentialReference;
}

/** Get an item */
export type OdooV1OpportunityGetParams = {
  resource: 'opportunity';
  operation: 'get';
/**
 * Opportunity ID
 */
    opportunityId?: string | Expression<string> | PlaceholderValue;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     * @default []
     */
    fieldsList?: string[];
  };
};

export type OdooV1OpportunityGetOutput = {
  __last_update?: string;
  active?: boolean;
  id?: number;
  kanban_state?: string;
  lost_reason?: boolean;
  name?: string;
  order_ids?: Array<number>;
  priority?: string;
  sale_amount_total?: number;
  sale_order_count?: number;
  type?: string;
  website_message_ids?: Array<number>;
  won_status?: string;
  write_date?: string;
};

export type OdooV1OpportunityGetNode = {
  type: 'n8n-nodes-base.odoo';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<OdooV1OpportunityGetParams>;
  output?: Items<OdooV1OpportunityGetOutput>;
};
