// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/odoo/v1/resource_opportunity/operation_get_all.ts
/**
 * Odoo Node - Version 1
 * Discriminator: resource=opportunity, operation=getAll
 */


interface Credentials {
  odooApi: CredentialReference;
}

/** Get many items */
export type OdooV1OpportunityGetAllParams = {
  resource: 'opportunity';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     * @default []
     */
    fieldsList?: string[];
  };
};

export type OdooV1OpportunityGetAllOutput = {
  create_date?: string;
  display_name?: string;
  id?: number;
  name?: string;
};

export type OdooV1OpportunityGetAllNode = {
  type: 'n8n-nodes-base.odoo';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<OdooV1OpportunityGetAllParams>;
  output?: Items<OdooV1OpportunityGetAllOutput>;
};
