// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/clockify/v1/resource_client/operation_get_all.ts
/**
 * Clockify Node - Version 1
 * Discriminator: resource=client, operation=getAll
 */


interface Credentials {
  clockifyApi: CredentialReference;
}

/** Get many clients */
export type ClockifyV1ClientGetAllParams = {
  resource: 'client';
  operation: 'getAll';
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 * @displayOptions.hide { resource: ["workspace"] }
 * @default []
 */
    workspaceId?: string | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Archived
     * @default false
     */
    archived?: boolean | Expression<boolean>;
    /** If provided, clients will be filtered by name
     */
    name?: string | Expression<string> | PlaceholderValue;
    /** Sort Order
     */
    'sort-order'?: 'ASCENDING' | 'DESCENDING' | Expression<string>;
  };
};

export type ClockifyV1ClientGetAllOutput = {
  archived?: boolean;
  currencyCode?: string;
  currencyId?: string;
  id?: string;
  name?: string;
  workspaceId?: string;
};

export type ClockifyV1ClientGetAllNode = {
  type: 'n8n-nodes-base.clockify';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ClockifyV1ClientGetAllParams>;
  output?: Items<ClockifyV1ClientGetAllOutput>;
};
