// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/freshworksCrm/v1/resource_task/operation_create.ts
/**
 * Freshworks CRM Node - Version 1
 * Discriminator: resource=task, operation=create
 */


interface Credentials {
  freshworksCrmApi: CredentialReference;
}

/** Create an account */
export type FreshworksCrmV1TaskCreateParams = {
  resource: 'task';
  operation: 'create';
/**
 * Title of the task
 */
    title?: string | Expression<string> | PlaceholderValue;
/**
 * Timestamp that denotes when the task is due to be completed
 */
    dueDate?: string | Expression<string>;
/**
 * ID of the user to whom the task is assigned. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    ownerId?: string | Expression<string>;
/**
 * Type of the entity for which the task is updated
 * @default Contact
 */
    targetableType?: 'Contact' | 'Deal' | 'SalesAccount' | Expression<string>;
/**
 * ID of the entity for which the task is created. The type of entity is selected in "Target Type".
 */
    targetable_id?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** ID of the user who created the task. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    creater_id?: string | Expression<string>;
    /** ID of the outcome of the task. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    outcome_id?: string | Expression<string>;
    /** ID of the type of task
     */
    task_type_id?: string | Expression<string> | PlaceholderValue;
  };
};

export type FreshworksCrmV1TaskCreateNode = {
  type: 'n8n-nodes-base.freshworksCrm';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<FreshworksCrmV1TaskCreateParams>;
};
