// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/freshworksCrm/v1/resource_task/operation_get_all.ts
/**
 * Freshworks CRM Node - Version 1
 * Discriminator: resource=task, operation=getAll
 */


interface Credentials {
  freshworksCrmApi: CredentialReference;
}

/** Retrieve many accounts */
export type FreshworksCrmV1TaskGetAllParams = {
  resource: 'task';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Include
     * @default owner
     */
    include?: 'owner' | 'targetable' | 'users' | Expression<string>;
    /** Status
     * @default open
     */
    filter?: 'completed' | 'due_today' | 'due_tomorrow' | 'open' | 'overdue' | Expression<string>;
  };
};

export type FreshworksCrmV1TaskGetAllNode = {
  type: 'n8n-nodes-base.freshworksCrm';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<FreshworksCrmV1TaskGetAllParams>;
};
