// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/freshworksCrm/v1/resource_task/operation_update.ts
/**
 * Freshworks CRM Node - Version 1
 * Discriminator: resource=task, operation=update
 */


interface Credentials {
  freshworksCrmApi: CredentialReference;
}

/** Update an account */
export type FreshworksCrmV1TaskUpdateParams = {
  resource: 'task';
  operation: 'update';
/**
 * ID of the task to update
 */
    taskId?: string | Expression<string> | PlaceholderValue;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** ID of the user who created the sales activity. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    creater_id?: string | Expression<string>;
    /** Timestamp that denotes when the task is due to be completed
     */
    dueDate?: string | Expression<string>;
    /** ID of the outcome of the task. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    outcome_id?: string | Expression<string>;
    /** ID of the user to whom the task is assigned. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    owner_id?: string | Expression<string>;
    /** ID of the entity for which the task is updated. The type of entity is selected in "Target Type".
     */
    targetable_id?: string | Expression<string> | PlaceholderValue;
    /** Type of the entity for which the task is updated
     * @default Contact
     */
    targetable_type?: 'Contact' | 'Deal' | 'SalesAccount' | Expression<string>;
    /** ID of the type of task
     */
    task_type_id?: string | Expression<string> | PlaceholderValue;
    /** Title of the task
     */
    title?: string | Expression<string> | PlaceholderValue;
  };
};

export type FreshworksCrmV1TaskUpdateNode = {
  type: 'n8n-nodes-base.freshworksCrm';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<FreshworksCrmV1TaskUpdateParams>;
};
