// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/freshworksCrm/v1/resource_task/operation_update.schema.js
/**
 * Freshworks CRM Node - Version 1 - Zod Schema
 * Discriminator: resource=task, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('task'),
      operation: z.literal('update'),
      taskId: stringOrExpression.optional(),
      updateFields: z.object({ creater_id: stringOrExpression.optional(), dueDate: stringOrExpression.optional(), outcome_id: stringOrExpression.optional(), owner_id: stringOrExpression.optional(), targetable_id: stringOrExpression.optional(), targetable_type: z.union([z.literal('Contact'), z.literal('Deal'), z.literal('SalesAccount'), expressionSchema]).optional(), task_type_id: stringOrExpression.optional(), title: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
