// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/salesforce/v1/resource_contact/operation_get.ts
/**
 * Salesforce Node - Version 1
 * Discriminator: resource=contact, operation=get
 */


interface Credentials {
  salesforceOAuth2Api: CredentialReference;
  salesforceJwtApi: CredentialReference;
}

/** Represents a contact, which is an individual associated with an account */
export type SalesforceV1ContactGetParams = {
  resource: 'contact';
  operation: 'get';
/**
 * OAuth Authorization Flow
 * @default oAuth2
 */
    authentication?: 'oAuth2' | 'jwt' | Expression<string>;
/**
 * ID of contact that needs to be fetched
 */
    contactId?: string | Expression<string> | PlaceholderValue;
};

export type SalesforceV1ContactGetOutput = {
  attributes?: {
    type?: string;
    url?: string;
  };
  CreatedById?: string;
  CreatedDate?: string;
  HomePhone?: null;
  Id?: string;
  IsDeleted?: boolean;
  IsEmailBounced?: boolean;
  IsPriorityRecord?: boolean;
  Jigsaw?: null;
  JigsawContactId?: null;
  LastCURequestDate?: null;
  LastCUUpdateDate?: null;
  LastModifiedById?: string;
  LastModifiedDate?: string;
  LastName?: string;
  MasterRecordId?: null;
  MiddleName?: null;
  Name?: string;
  OtherPhone?: null;
  OwnerId?: string;
  Suffix?: null;
  SystemModstamp?: string;
};

export type SalesforceV1ContactGetNode = {
  type: 'n8n-nodes-base.salesforce';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SalesforceV1ContactGetParams>;
  output?: Items<SalesforceV1ContactGetOutput>;
};
