// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/sentryIo/v1/resource_organization/operation_update.ts
/**
 * Sentry.io Node - Version 1
 * Discriminator: resource=organization, operation=update
 */


interface Credentials {
  sentryIoOAuth2Api: CredentialReference;
  sentryIoApi: CredentialReference;
  sentryIoServerApi: CredentialReference;
}

/** Update an issue */
export type SentryIoV1OrganizationUpdateParams = {
  resource: 'organization';
  operation: 'update';
  authentication?: 'accessToken' | 'oAuth2' | 'accessTokenServer' | Expression<string>;
/**
 * The slug of the organization to update. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    organization_slug?: string | Expression<string>;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** The new name of the organization
     */
    name?: string | Expression<string> | PlaceholderValue;
    /** The new URL slug for this organization
     */
    slug?: string | Expression<string> | PlaceholderValue;
  };
};

export type SentryIoV1OrganizationUpdateNode = {
  type: 'n8n-nodes-base.sentryIo';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SentryIoV1OrganizationUpdateParams>;
};
