// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/sentryIo/v1/resource_organization/operation_create.ts
/**
 * Sentry.io Node - Version 1
 * Discriminator: resource=organization, operation=create
 */


interface Credentials {
  sentryIoOAuth2Api: CredentialReference;
  sentryIoApi: CredentialReference;
  sentryIoServerApi: CredentialReference;
}

/** Create an organization */
export type SentryIoV1OrganizationCreateParams = {
  resource: 'organization';
  operation: 'create';
  authentication?: 'accessToken' | 'oAuth2' | 'accessTokenServer' | Expression<string>;
/**
 * The slug of the organization the team should be created for
 */
    name?: string | Expression<string> | PlaceholderValue;
/**
 * Whether you agree to the applicable terms of service and privacy policy of Sentry.io
 * @default false
 */
    agreeTerms?: boolean | Expression<boolean>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** The unique URL slug for this organization. If this is not provided a slug is automatically generated based on the name.
     */
    slug?: string | Expression<string> | PlaceholderValue;
  };
};

export type SentryIoV1OrganizationCreateNode = {
  type: 'n8n-nodes-base.sentryIo';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SentryIoV1OrganizationCreateParams>;
};
