// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/sentryIo/v1/resource_organization/operation_get_all.ts
/**
 * Sentry.io Node - Version 1
 * Discriminator: resource=organization, operation=getAll
 */


interface Credentials {
  sentryIoOAuth2Api: CredentialReference;
  sentryIoApi: CredentialReference;
  sentryIoServerApi: CredentialReference;
}

/** Get many events */
export type SentryIoV1OrganizationGetAllParams = {
  resource: 'organization';
  operation: 'getAll';
  authentication?: 'accessToken' | 'oAuth2' | 'accessTokenServer' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Whether to restrict results to organizations which you have membership
     * @default true
     */
    member?: boolean | Expression<boolean>;
    /** Whether to restrict results to organizations which you are the owner
     * @default true
     */
    owner?: boolean | Expression<boolean>;
  };
};

export type SentryIoV1OrganizationGetAllNode = {
  type: 'n8n-nodes-base.sentryIo';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SentryIoV1OrganizationGetAllParams>;
};
