// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/misp/v1/resource_organisation/operation_create.schema.js
/**
 * MISP Node - Version 1 - Zod Schema
 * Discriminator: resource=organisation, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('organisation'),
      operation: z.literal('create').default('create'),
      name: stringOrExpression.optional(),
      additionalFields: z.object({ created_by_email: stringOrExpression.optional(), description: stringOrExpression.optional(), nationality: stringOrExpression.optional(), sector: stringOrExpression.optional(), type: stringOrExpression.optional(), usercount: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
