// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/slack/v22/resource_reaction/operation_remove.ts
/**
 * Slack Node - Version 2.2
 * Discriminator: resource=reaction, operation=remove
 */


interface Credentials {
  slackApi: CredentialReference;
  slackOAuth2Api: CredentialReference;
}

/** Remove a reaction of a message */
export type SlackV22ReactionRemoveParams = {
  resource: 'reaction';
  operation: 'remove';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * The Slack channel to get the reactions from
 * @default {"mode":"list","value":""}
 */
    channelId?: { __rl: true; mode: 'list' | 'id' | 'url'; value: string; cachedResultName?: string };
/**
 * Timestamp of the message to add, get or remove
 */
    timestamp: number | Expression<number>;
/**
 * Emoji code to use for the message reaction. Use emoji codes like +1, not an actual emoji like 👍. &lt;a target="_blank" href=" https://www.webfx.com/tools/emoji-cheat-sheet/"&gt;List of common emoji codes&lt;/a&gt;
 */
    name?: string | Expression<string> | PlaceholderValue;
};

export type SlackV22ReactionRemoveOutput = {
  ok?: boolean;
};

export type SlackV22ReactionRemoveNode = {
  type: 'n8n-nodes-base.slack';
  version: 2.2;
  credentials?: Credentials;
  config: NodeConfig<SlackV22ReactionRemoveParams>;
  output?: Items<SlackV22ReactionRemoveOutput>;
};
