// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/facebookTrigger/v1.schema.js
/**
 * Facebook Trigger Node - Version 1 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    appId: stringOrExpression.optional(),
    object: z.union([z.literal('adAccount'), z.literal('application'), z.literal('certificateTransparency'), z.literal('group'), z.literal('instagram'), z.literal('link'), z.literal('page'), z.literal('permissions'), z.literal('user'), z.literal('whatsappBusinessAccount'), z.literal('workplaceSecurity'), expressionSchema]).optional(),
    fields: z.array(z.string()).optional(),
    options: z.object({ includeValues: booleanOrExpression.optional() }).optional(),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
