// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/serviceNow/v1/resource_user/operation_create.schema.js
/**
 * ServiceNow Node - Version 1 - Zod Schema
 * Discriminator: resource=user, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('user').default('user'),
      operation: z.literal('create'),
      authentication: z.union([z.literal('basicAuth'), z.literal('oAuth2'), expressionSchema]).optional(),
      short_description: stringOrExpression.optional(),
      additionalFields: z.object({ active: booleanOrExpression.optional(), building: stringOrExpression.optional(), city: stringOrExpression.optional(), company: stringOrExpression.optional(), country: stringOrExpression.optional(), department: stringOrExpression.optional(), email: stringOrExpression.optional(), first_name: stringOrExpression.optional(), gender: stringOrExpression.optional(), home_phone: stringOrExpression.optional(), last_name: stringOrExpression.optional(), location: stringOrExpression.optional(), manager: stringOrExpression.optional(), middle_name: stringOrExpression.optional(), mobile_phone: stringOrExpression.optional(), user_password: stringOrExpression.optional(), password_needs_reset: booleanOrExpression.optional(), phone: stringOrExpression.optional(), roles: z.array(z.string()).optional(), source: stringOrExpression.optional(), state: stringOrExpression.optional(), street: stringOrExpression.optional(), user_name: stringOrExpression.optional(), zip: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
