// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/serviceNow/v1/resource_user/operation_get.schema.js
/**
 * ServiceNow Node - Version 1 - Zod Schema
 * Discriminator: resource=user, operation=get
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('user').default('user'),
      operation: z.literal('get'),
      authentication: z.union([z.literal('basicAuth'), z.literal('oAuth2'), expressionSchema]).optional(),
      getOption: z.union([z.literal('id'), z.literal('user_name'), expressionSchema]).optional(),
      user_name: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"getOption":["user_name"]}}, defaults: {"getOption":"id"} }),
      id: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"getOption":["id"]}}, defaults: {"getOption":"id"} }),
      options: z.object({ sysparm_exclude_reference_link: booleanOrExpression.optional(), sysparm_fields: z.array(z.string()).optional(), sysparm_display_value: z.union([z.literal('false'), z.literal('all'), z.literal('true'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
