// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mongoDb/v12/resource_document/operation_find_one_and_replace.schema.js
/**
 * MongoDB Node - Version 1.2 - Zod Schema
 * Discriminator: resource=document, operation=findOneAndReplace
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('document').default('document'),
      operation: z.literal('findOneAndReplace'),
      collection: stringOrExpression.optional(),
      updateKey: stringOrExpression.optional(),
      fields: stringOrExpression.optional(),
      upsert: booleanOrExpression.optional(),
      options: z.object({ dateFields: stringOrExpression.optional(), useDotNotation: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
