// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mongoDb/v12/resource_document/operation_find.schema.js
/**
 * MongoDB Node - Version 1.2 - Zod Schema
 * Discriminator: resource=document, operation=find
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('document').default('document'),
      operation: z.literal('find').default('find'),
      collection: stringOrExpression.optional(),
      options: z.object({ limit: numberOrExpression.optional(), skip: numberOrExpression.optional(), sort: z.union([iDataObjectSchema, z.string()]).optional(), projection: z.union([iDataObjectSchema, z.string()]).optional() }).optional(),
      query: z.union([iDataObjectSchema, z.string()]).optional(),
    }).optional(),
  });
};
