// @n8n/ai-workflow-builder/dist/tools/get-node-context.tool.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GET_NODE_CONTEXT_TOOL = void 0;
exports.createGetNodeContextTool = createGetNodeContextTool;
const tools_1 = require("@langchain/core/tools");
const n8n_workflow_1 = require("n8n-workflow");
const zod_1 = require("zod");
const errors_1 = require("../errors");
const node_helpers_1 = require("../utils/node-helpers");
const truncate_json_1 = require("../utils/truncate-json");
const progress_1 = require("./helpers/progress");
const response_1 = require("./helpers/response");
const state_1 = require("./helpers/state");
const DISPLAY_TITLE = 'Getting node context';
function isRecord(value) {
    return typeof value === 'object' && value !== null && !Array.isArray(value);
}
function getNodeRunData(executionData, nodeName) {
    if (!executionData || !isRecord(executionData.runData)) {
        return undefined;
    }
    return executionData.runData[nodeName];
}
const getNodeContextSchema = zod_1.z.object({
    nodeName: zod_1.z.string().describe('The name of the node to get context for'),
    includeExecutionData: zod_1.z
        .boolean()
        .optional()
        .default(true)
        .describe('Include execution data for this node if available'),
});
function classifyNode(nodeType, hasAiInputs, hasAiOutputs) {
    if ((0, node_helpers_1.isTriggerNodeType)(nodeType)) {
        return 'trigger';
    }
    if (hasAiInputs) {
        return 'ai_parent';
    }
    if (hasAiOutputs) {
        return 'ai_subnode';
    }
    return 'regular';
}
function getParentConnections(nodeName, connectionsByDestination) {
    const nodeConns = connectionsByDestination[nodeName];
    if (!nodeConns)
        return [];
    const parents = [];
    for (const [type, typeConns] of Object.entries(nodeConns)) {
        if (!(0, n8n_workflow_1.isNodeConnectionType)(type) || !typeConns)
            continue;
        for (const connArray of typeConns) {
            if (!connArray)
                continue;
            for (const conn of connArray) {
                parents.push({ node: conn.node, type });
            }
        }
    }
    return parents;
}
function getChildConnections(nodeName, connections) {
    const nodeConns = connections[nodeName];
    if (!nodeConns)
        return [];
    const children = [];
    for (const [type, typeConns] of Object.entries(nodeConns)) {
        if (!(0, n8n_workflow_1.isNodeConnectionType)(type) || !typeConns)
            continue;
        for (const connArray of typeConns) {
            if (!connArray)
                continue;
            for (const conn of connArray) {
                children.push({ node: conn.node, type });
            }
        }
    }
    return children;
}
function buildNodeContext(node, workflow, includeExecutionData, executionSchema, executionData) {
    const parts = [`<node_context name="${node.name}" id="${node.id}">`];
    parts.push(`ID: ${node.id}`);
    parts.push(`Type: ${node.type}`);
    if (node.typeVersion) {
        parts.push(`Version: ${node.typeVersion}`);
    }
    const connectionsByDestination = (0, n8n_workflow_1.mapConnectionsByDestination)(workflow.connections);
    const parents = getParentConnections(node.name, connectionsByDestination);
    const children = getChildConnections(node.name, workflow.connections);
    const hasAiInputs = parents.some((p) => p.type !== 'main');
    const hasAiOutputs = children.some((c) => c.type !== 'main');
    const classification = classifyNode(node.type, hasAiInputs, hasAiOutputs);
    parts.push(`Classification: ${classification}`);
    parts.push('');
    if (parents.length > 0) {
        parts.push('Parent nodes (upstream):');
        for (const parent of parents) {
            if (parent.type === 'main') {
                parts.push(`  ← ${parent.node}`);
            }
            else {
                parts.push(`  ←[${parent.type}] ${parent.node}`);
            }
        }
    }
    else {
        parts.push('Parent nodes: None (this is a start node)');
    }
    parts.push('');
    if (children.length > 0) {
        parts.push('Child nodes (downstream):');
        for (const child of children) {
            if (child.type === 'main') {
                parts.push(`  → ${child.node}`);
            }
            else {
                parts.push(`  -[${child.type}]-> ${child.node}`);
            }
        }
    }
    else {
        parts.push('Child nodes: None (this is an end node)');
    }
    parts.push('');
    parts.push('Parameters:');
    if (node.parameters && Object.keys(node.parameters).length > 0) {
        parts.push((0, truncate_json_1.truncateJson)(node.parameters));
    }
    else {
        parts.push('  (no parameters set)');
    }
    if (includeExecutionData) {
        parts.push('');
        const nodeSchema = executionSchema.find((s) => s.nodeName === node.name);
        if (nodeSchema) {
            parts.push('Output schema (from last execution):');
            parts.push((0, truncate_json_1.truncateJson)(nodeSchema.schema));
        }
        const nodeRunData = getNodeRunData(executionData, node.name);
        if (nodeRunData) {
            parts.push('');
            parts.push('Execution data (from last run):');
            parts.push((0, truncate_json_1.truncateJson)(nodeRunData));
        }
        if (!nodeSchema && !nodeRunData) {
            parts.push('No execution data available for this node');
        }
    }
    parts.push('</node_context>');
    return parts.join('\n');
}
exports.GET_NODE_CONTEXT_TOOL = {
    toolName: 'get_node_context',
    displayTitle: DISPLAY_TITLE,
};
function createGetNodeContextTool(logger) {
    const dynamicTool = (0, tools_1.tool)((input, config) => {
        const reporter = (0, progress_1.createProgressReporter)(config, exports.GET_NODE_CONTEXT_TOOL.toolName, DISPLAY_TITLE);
        try {
            const validatedInput = getNodeContextSchema.parse(input);
            const { nodeName, includeExecutionData } = validatedInput;
            reporter.start(validatedInput);
            const workflow = (0, state_1.getEffectiveWorkflow)();
            const state = (0, state_1.getWorkflowState)();
            const node = workflow.nodes.find((n) => n.name === nodeName);
            if (!node) {
                const output = {
                    found: false,
                    nodeName,
                    parentCount: 0,
                    childCount: 0,
                    hasExecutionData: false,
                    message: `Node "${nodeName}" not found in workflow`,
                };
                reporter.complete(output);
                return (0, response_1.createSuccessResponse)(config, `Node "${nodeName}" not found. Available nodes: ${workflow.nodes.map((n) => n.name).join(', ')}`);
            }
            const executionSchema = state.workflowContext?.executionSchema ?? [];
            const executionData = state.workflowContext?.executionData;
            logger?.debug(`Getting context for node "${nodeName}" (${node.type})`);
            const formattedContext = buildNodeContext(node, workflow, includeExecutionData, executionSchema, executionData);
            const connectionsByDestination = (0, n8n_workflow_1.mapConnectionsByDestination)(workflow.connections);
            const parents = getParentConnections(nodeName, connectionsByDestination);
            const children = getChildConnections(nodeName, workflow.connections);
            const nodeSchema = executionSchema.find((s) => s.nodeName === nodeName);
            const nodeRunData = getNodeRunData(executionData, nodeName);
            const output = {
                found: true,
                nodeName,
                nodeType: node.type,
                parentCount: parents.length,
                childCount: children.length,
                hasExecutionData: nodeSchema !== undefined || nodeRunData !== undefined,
                message: `Found node "${nodeName}" with ${parents.length} parents and ${children.length} children`,
            };
            reporter.complete(output);
            return (0, response_1.createSuccessResponse)(config, formattedContext);
        }
        catch (error) {
            if (error instanceof zod_1.z.ZodError) {
                const validationError = new errors_1.ValidationError('Invalid input parameters', {
                    extra: { errors: error.errors },
                });
                reporter.error(validationError);
                return (0, response_1.createErrorResponse)(config, validationError);
            }
            const errorMessage = error instanceof Error
                ? `${error.message}${error.stack ? `\n${error.stack}` : ''}`
                : 'Unknown error occurred';
            const toolError = new errors_1.ToolExecutionError(errorMessage, {
                toolName: exports.GET_NODE_CONTEXT_TOOL.toolName,
                cause: error instanceof Error ? error : undefined,
            });
            reporter.error(toolError);
            return (0, response_1.createErrorResponse)(config, toolError);
        }
    }, {
        name: exports.GET_NODE_CONTEXT_TOOL.toolName,
        description: 'Get full context for a specific node including its parameters, parent nodes (upstream), child nodes (downstream), classification, and execution data. Use this before configuring a node to understand its current state and connections.',
        schema: getNodeContextSchema,
    });
    return {
        tool: dynamicTool,
        ...exports.GET_NODE_CONTEXT_TOOL,
    };
}
