// @n8n/ai-workflow-builder/dist/tools/get-resource-locator-options.tool.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GET_RESOURCE_LOCATOR_OPTIONS_TOOL = void 0;
exports.createGetResourceLocatorOptionsTool = createGetResourceLocatorOptionsTool;
const tools_1 = require("@langchain/core/tools");
const zod_1 = require("zod");
const errors_1 = require("../errors");
const progress_1 = require("./helpers/progress");
const response_1 = require("./helpers/response");
const state_1 = require("./helpers/state");
const validation_1 = require("./helpers/validation");
const TOOL_NAME = 'get_resource_locator_options';
const DISPLAY_TITLE = 'Fetching resource options';
const DEFAULT_TIMEOUT_MS = 10000;
const getResourceLocatorOptionsSchema = zod_1.z.object({
    nodeId: zod_1.z.string().describe('The ID of the node to fetch resource locator options for'),
    parameterPath: zod_1.z
        .string()
        .describe('The path to the resource locator parameter (e.g., "calendarId", "model", "boardId")'),
    filter: zod_1.z.string().optional().describe('Optional filter string to narrow down results'),
});
exports.GET_RESOURCE_LOCATOR_OPTIONS_TOOL = {
    toolName: TOOL_NAME,
    displayTitle: DISPLAY_TITLE,
};
function extractSearchMethod(nodeType, parameterPath) {
    const property = nodeType.properties.find((p) => p.name === parameterPath && p.type === 'resourceLocator');
    if (!property?.modes) {
        return null;
    }
    const listMode = property.modes.find((mode) => mode.type === 'list' && mode.typeOptions?.searchListMethod);
    return listMode?.typeOptions?.searchListMethod ?? null;
}
function formatOptionsForLLM(options, parameterPath) {
    if (options.length === 0) {
        return `No options available for parameter "${parameterPath}". The resource may require credentials to be set up first or the external service returned no results.`;
    }
    const parts = [
        `<resource_locator_options parameter="${parameterPath}">`,
        `<total_count>${options.length}</total_count>`,
        '<options>',
    ];
    options.forEach((opt, index) => {
        parts.push(`  <option index="${index}">`);
        parts.push(`    <display_name>${opt.name}</display_name>`);
        parts.push(`    <id>${String(opt.value)}</id>`);
        parts.push('  </option>');
    });
    parts.push('</options>');
    parts.push('</resource_locator_options>');
    return parts.join('\n');
}
function createGetResourceLocatorOptionsTool(nodeTypes, resourceLocatorCallback, logger) {
    const dynamicTool = (0, tools_1.tool)(async (input, config) => {
        const reporter = (0, progress_1.createProgressReporter)(config, TOOL_NAME, DISPLAY_TITLE);
        try {
            const validatedInput = getResourceLocatorOptionsSchema.parse(input);
            const { nodeId, parameterPath, filter } = validatedInput;
            reporter.start(validatedInput);
            if (!resourceLocatorCallback) {
                const error = new errors_1.ValidationError('Resource locator fetching is not available. The workflow builder service may not have been configured with credentials support.');
                reporter.error(error);
                return (0, response_1.createErrorResponse)(config, error);
            }
            const state = (0, state_1.getWorkflowState)();
            const workflow = (0, state_1.getCurrentWorkflow)(state);
            const node = (0, validation_1.validateNodeExists)(nodeId, workflow.nodes);
            if (!node) {
                const error = (0, validation_1.createNodeNotFoundError)(nodeId);
                reporter.error(error);
                return (0, response_1.createErrorResponse)(config, error);
            }
            const nodeType = (0, validation_1.findNodeType)(node.type, node.typeVersion, nodeTypes);
            if (!nodeType) {
                const error = (0, validation_1.createNodeTypeNotFoundError)(node.type);
                reporter.error(error);
                return (0, response_1.createErrorResponse)(config, error);
            }
            const searchMethod = extractSearchMethod(nodeType, parameterPath);
            if (!searchMethod) {
                const error = new errors_1.ValidationError(`Parameter "${parameterPath}" on node "${node.name}" is not a resource locator with list search capability`, { extra: { nodeId, parameterPath } });
                reporter.error(error);
                return (0, response_1.createErrorResponse)(config, error);
            }
            const nodeTypeRequiresCredentials = Array.isArray(nodeType.credentials) && nodeType.credentials.length > 0;
            const credentials = node.credentials;
            const hasCredentials = credentials && Object.keys(credentials).length > 0;
            if (nodeTypeRequiresCredentials && !hasCredentials) {
                const error = new errors_1.ValidationError(`Node "${node.name}" does not have credentials configured. Set up credentials before fetching resource options.`);
                reporter.error(error);
                return (0, response_1.createErrorResponse)(config, error);
            }
            (0, progress_1.reportProgress)(reporter, `Fetching options for "${parameterPath}" on node "${node.name}"`, {
                nodeId,
                parameterPath,
                searchMethod,
            });
            try {
                const result = await Promise.race([
                    resourceLocatorCallback(searchMethod, `parameters.${parameterPath}`, { name: node.type, version: node.typeVersion }, node.parameters ?? {}, credentials, filter),
                    new Promise((_, reject) => setTimeout(() => reject(new Error('Request timed out')), DEFAULT_TIMEOUT_MS)),
                ]);
                const formattedOptions = formatOptionsForLLM(result.results, parameterPath);
                reporter.complete({
                    parameterPath,
                    optionsCount: result.results.length,
                    hasPagination: !!result.paginationToken,
                });
                return (0, response_1.createSuccessResponse)(config, formattedOptions);
            }
            catch (callbackError) {
                const errorMessage = callbackError instanceof Error
                    ? callbackError.message
                    : 'Unknown error fetching options';
                logger?.warn('Resource locator callback failed', {
                    nodeId,
                    parameterPath,
                    searchMethod,
                    error: errorMessage,
                });
                const error = new errors_1.ToolExecutionError(`Failed to fetch options for "${parameterPath}": ${errorMessage}`, {
                    toolName: TOOL_NAME,
                    cause: callbackError instanceof Error ? callbackError : undefined,
                });
                reporter.error(error);
                return (0, response_1.createErrorResponse)(config, error);
            }
        }
        catch (error) {
            if (error instanceof zod_1.z.ZodError) {
                const validationError = new errors_1.ValidationError('Invalid input parameters', {
                    extra: { errors: error.errors },
                });
                reporter.error(validationError);
                return (0, response_1.createErrorResponse)(config, validationError);
            }
            const toolError = new errors_1.ToolExecutionError(error instanceof Error ? error.message : 'Unknown error occurred', {
                toolName: TOOL_NAME,
                cause: error instanceof Error ? error : undefined,
            });
            reporter.error(toolError);
            return (0, response_1.createErrorResponse)(config, toolError);
        }
    }, {
        name: TOOL_NAME,
        description: 'Fetch available options for a resource locator parameter. Use this when a node parameter requires selecting from a dynamic list (e.g., calendars, boards, models, channels). Returns the available options that can be used to configure the parameter.',
        schema: getResourceLocatorOptionsSchema,
    });
    return {
        tool: dynamicTool,
        ...exports.GET_RESOURCE_LOCATOR_OPTIONS_TOOL,
    };
}
