// @n8n/ai-workflow-builder/dist/tools/introspect.tool.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.INTROSPECT_TOOL = void 0;
exports.extractIntrospectionEventsFromMessages = extractIntrospectionEventsFromMessages;
exports.createIntrospectTool = createIntrospectTool;
const tools_1 = require("@langchain/core/tools");
const zod_1 = require("zod");
const errors_1 = require("../errors");
const progress_1 = require("../tools/helpers/progress");
const response_1 = require("../tools/helpers/response");
function safeString(value, defaultValue = '') {
    if (value === null || value === undefined) {
        return defaultValue;
    }
    if (typeof value === 'string') {
        return value;
    }
    if (typeof value === 'number' || typeof value === 'boolean') {
        return String(value);
    }
    try {
        return JSON.stringify(value);
    }
    catch {
        return defaultValue;
    }
}
function extractIntrospectionEventsFromMessages(messages) {
    const events = [];
    for (const msg of messages) {
        const aiMsg = msg;
        if (aiMsg._getType?.() === 'ai' && aiMsg.tool_calls && Array.isArray(aiMsg.tool_calls)) {
            for (const toolCall of aiMsg.tool_calls) {
                if (toolCall.name === 'introspect' && toolCall.args) {
                    const args = toolCall.args;
                    events.push({
                        timestamp: new Date().toISOString(),
                        category: safeString(args.category, 'other'),
                        issue: safeString(args.issue, ''),
                        source: args.source !== undefined ? safeString(args.source) : undefined,
                    });
                }
            }
        }
    }
    return events;
}
const introspectSchema = zod_1.z.object({
    issue: zod_1.z.string().min(1).describe('Describe the problem with your instructions or documentation'),
    category: zod_1.z
        .enum([
        'conflicting_instructions',
        'missing_guidance',
        'unclear_node_description',
        'incomplete_example',
        'other',
    ])
        .describe('Categorize the type of issue with your instructions'),
    source: zod_1.z
        .string()
        .optional()
        .describe('Which instruction, documentation section, or node description caused this issue'),
});
exports.INTROSPECT_TOOL = {
    toolName: 'introspect',
    displayTitle: 'Introspecting',
};
function createIntrospectTool(logger) {
    const dynamicTool = (0, tools_1.tool)((input, config) => {
        const reporter = (0, progress_1.createProgressReporter)(config, exports.INTROSPECT_TOOL.toolName, exports.INTROSPECT_TOOL.displayTitle);
        try {
            const validatedInput = introspectSchema.parse(input);
            const { issue, category, source } = validatedInput;
            const timestamp = new Date().toISOString();
            logger?.debug('Introspection event', {
                tool: 'introspect',
                timestamp,
                category,
                issue,
                source,
            });
            reporter.start(validatedInput);
            reporter.progress('Processing diagnostic report...');
            const acknowledgment = `Diagnostic report received. Category: ${category}. Please proceed with your best judgment based on the available information. If the issue persists, prioritize user safety and explicit user requests over implicit assumptions.`;
            reporter.complete({ acknowledged: true, category, issueLength: issue.length });
            return (0, response_1.createSuccessResponse)(config, acknowledgment);
        }
        catch (error) {
            if (error instanceof zod_1.z.ZodError) {
                const validationError = new errors_1.ValidationError('Invalid input parameters', {
                    extra: { errors: error.errors },
                });
                reporter.error(validationError);
                return (0, response_1.createErrorResponse)(config, validationError);
            }
            const toolError = new errors_1.ToolExecutionError(error instanceof Error ? error.message : 'Unknown error occurred', {
                toolName: exports.INTROSPECT_TOOL.toolName,
                cause: error instanceof Error ? error : undefined,
            });
            reporter.error(toolError);
            return (0, response_1.createErrorResponse)(config, toolError);
        }
    }, {
        name: exports.INTROSPECT_TOOL.toolName,
        description: `MANDATORY diagnostic tool - you MUST call this before responding to report issues with your instructions.

Report issues with YOUR instructions and documentation (not the user's request):
- System prompt sections that conflict, are unclear, or hard to follow
- Best practices documentation that lacks guidance for your situation
- Node descriptions from discovery that are confusing or incomplete
- Template patterns or examples that don't apply to the current task
- Missing guidance for common workflow scenarios

Be specific: identify WHICH instruction section or documentation caused the issue.
If everything was clear, report category "other" with issue describing what worked well.

This tool is REQUIRED as part of your execution sequence. Do not skip it.`,
        schema: introspectSchema,
    });
    return {
        tool: dynamicTool,
        ...exports.INTROSPECT_TOOL,
    };
}
