// @n8n/ai-workflow-builder/dist/tools/node-details.tool.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NODE_DETAILS_TOOL = void 0;
exports.createNodeDetailsTool = createNodeDetailsTool;
const tools_1 = require("@langchain/core/tools");
const zod_1 = require("zod");
const constants_1 = require("../constants");
const resource_operation_extractor_1 = require("../utils/resource-operation-extractor");
const errors_1 = require("../errors");
const progress_1 = require("./helpers/progress");
const response_1 = require("./helpers/response");
const state_1 = require("./helpers/state");
const validation_1 = require("./helpers/validation");
const node_configuration_utils_1 = require("./utils/node-configuration.utils");
const templates_1 = require("./web/templates");
const MAX_NODE_EXAMPLES = 5;
const nodeDetailsSchema = zod_1.z.object({
    nodeName: zod_1.z.string().describe('The exact node type name (e.g., n8n-nodes-base.httpRequest)'),
    nodeVersion: zod_1.z.number().describe('The exact node version'),
    withParameters: zod_1.z
        .boolean()
        .optional()
        .default(false)
        .describe('Whether to include node parameters in the output'),
    withConnections: zod_1.z
        .boolean()
        .optional()
        .default(true)
        .describe('Whether to include node supported connections in the output'),
});
function formatInputs(inputs) {
    if (!inputs || inputs.length === 0) {
        return '<inputs>none</inputs>';
    }
    if (typeof inputs === 'string') {
        return `<input>${inputs}</input>`;
    }
    const formattedInputs = inputs.map((input) => {
        if (typeof input === 'string') {
            return `<input>${input}</input>`;
        }
        return `<input>${JSON.stringify(input)}</input>`;
    });
    return formattedInputs.join('\n');
}
function formatOutputs(outputs) {
    if (!outputs || outputs.length === 0) {
        return '<outputs>none</outputs>';
    }
    if (typeof outputs === 'string') {
        return `<output>${outputs}</output>`;
    }
    const formattedOutputs = outputs.map((output) => {
        if (typeof output === 'string') {
            return `<output>${output}</output>`;
        }
        return `<output>${JSON.stringify(output)}</output>`;
    });
    return formattedOutputs.join('\n');
}
function formatNodeDetails(details, withParameters = false, withConnections = true, examples = [], resourceOperationInfo) {
    const parts = [];
    parts.push('<node_details>');
    parts.push(`<name>${details.name}</name>`);
    parts.push(`<display_name>${details.displayName}</display_name>`);
    parts.push(`<description>${details.description}</description>`);
    if (details.subtitle) {
        parts.push(`<subtitle>${details.subtitle}</subtitle>`);
    }
    if (resourceOperationInfo) {
        parts.push((0, resource_operation_extractor_1.formatResourceOperationsForPrompt)(resourceOperationInfo));
    }
    if (withParameters && details.properties.length > 0) {
        const stringifiedProperties = JSON.stringify(details.properties, null, 2);
        parts.push(`<properties>
			${stringifiedProperties.length > 1000 ? stringifiedProperties.slice(0, 1000) + '... Rest of properties omitted' : stringifiedProperties}
			</properties>`);
    }
    if (withConnections) {
        parts.push('<connections>');
        parts.push(formatInputs(details.inputs));
        parts.push(formatOutputs(details.outputs));
        parts.push('</connections>');
    }
    if (examples.length > 0) {
        const limitedExamples = examples.slice(0, MAX_NODE_EXAMPLES);
        const { parts: exampleParts } = limitedExamples.reduce((acc, config) => {
            const exampleStr = JSON.stringify(config.parameters, null, 2);
            if (acc.chars + exampleStr.length <= constants_1.MAX_NODE_EXAMPLE_CHARS) {
                acc.parts.push(`<example>\n${exampleStr}\n</example>`);
                acc.chars += exampleStr.length;
            }
            return acc;
        }, { parts: [], chars: 0 });
        if (exampleParts.length > 0) {
            parts.push('<node_examples>');
            parts.push(...exampleParts);
            parts.push('</node_examples>');
        }
    }
    parts.push('</node_details>');
    return parts.join('\n');
}
function extractNodeDetails(nodeType) {
    return {
        name: nodeType.name,
        displayName: nodeType.displayName,
        description: nodeType.description,
        properties: nodeType.properties,
        subtitle: nodeType.subtitle,
        inputs: nodeType.inputs,
        outputs: nodeType.outputs,
    };
}
exports.NODE_DETAILS_TOOL = {
    toolName: 'get_node_details',
    displayTitle: 'Getting node details',
};
async function getNodeExamples(nodeName, nodeVersion, logger, onProgress) {
    try {
        const state = (0, state_1.getWorkflowState)();
        const cachedTemplates = state?.cachedTemplates ?? [];
        const filteredConfigs = (0, node_configuration_utils_1.getNodeConfigurationsFromTemplates)(cachedTemplates, nodeName, nodeVersion);
        if (filteredConfigs.length > 0) {
            logger?.debug('Found node configurations in cached templates', {
                nodeName,
                nodeVersion,
                count: filteredConfigs.length,
            });
            return { examples: filteredConfigs };
        }
    }
    catch {
    }
    onProgress?.(`Fetching examples for ${nodeName}...`);
    try {
        const result = await (0, templates_1.fetchWorkflowsFromTemplates)({ nodes: nodeName, rows: 10 }, { maxTemplates: 5, logger });
        if (result.workflows.length > 0) {
            const nodeConfigs = (0, node_configuration_utils_1.getNodeConfigurationsFromTemplates)(result.workflows, nodeName, nodeVersion);
            logger?.debug('Fetched node configurations from templates API', {
                nodeName,
                nodeVersion,
                count: nodeConfigs.length,
                workflowCount: result.workflows.length,
            });
            return { examples: nodeConfigs, newTemplates: result.workflows };
        }
    }
    catch (error) {
        logger?.warn('Failed to fetch node examples from templates', { nodeName, error });
    }
    return { examples: [] };
}
function createNodeDetailsTool(nodeTypes, logger) {
    const dynamicTool = (0, tools_1.tool)(async (input, config) => {
        const reporter = (0, progress_1.createProgressReporter)(config, exports.NODE_DETAILS_TOOL.toolName, exports.NODE_DETAILS_TOOL.displayTitle);
        try {
            const validatedInput = nodeDetailsSchema.parse(input);
            const { nodeName, nodeVersion, withParameters, withConnections } = validatedInput;
            reporter.start(validatedInput);
            (0, progress_1.reportProgress)(reporter, `Looking up details for ${nodeName}...`);
            const nodeType = (0, validation_1.findNodeType)(nodeName, nodeVersion, nodeTypes);
            if (!nodeType) {
                const error = (0, validation_1.createNodeTypeNotFoundError)(nodeName);
                reporter.error(error);
                return (0, response_1.createErrorResponse)(config, error);
            }
            const details = extractNodeDetails(nodeType);
            const resourceOperationInfo = (0, resource_operation_extractor_1.extractResourceOperations)(nodeType, nodeVersion, logger);
            const { examples, newTemplates } = await getNodeExamples(nodeName, nodeVersion, logger, (msg) => (0, progress_1.reportProgress)(reporter, msg));
            const message = formatNodeDetails(details, withParameters, withConnections, examples, resourceOperationInfo);
            const output = {
                details,
                found: true,
                message,
            };
            reporter.complete(output);
            const cacheKey = (0, resource_operation_extractor_1.createResourceCacheKey)(nodeName, nodeVersion);
            const stateUpdates = {
                resourceOperationCache: { [cacheKey]: resourceOperationInfo },
            };
            if (newTemplates) {
                stateUpdates.cachedTemplates = newTemplates;
            }
            return (0, response_1.createSuccessResponse)(config, message, stateUpdates);
        }
        catch (error) {
            if (error instanceof zod_1.z.ZodError) {
                const validationError = new errors_1.ValidationError('Invalid input parameters', {
                    extra: { errors: error.errors },
                });
                reporter.error(validationError);
                return (0, response_1.createErrorResponse)(config, validationError);
            }
            const toolError = new errors_1.ToolExecutionError(error instanceof Error ? error.message : 'Unknown error occurred', {
                toolName: exports.NODE_DETAILS_TOOL.toolName,
                cause: error instanceof Error ? error : undefined,
            });
            reporter.error(toolError);
            return (0, response_1.createErrorResponse)(config, toolError);
        }
    }, {
        name: exports.NODE_DETAILS_TOOL.toolName,
        description: 'Get detailed information about a specific n8n node type including properties, available connections, and up to 5 example configurations. Use this before adding nodes to understand their input/output structure.',
        schema: nodeDetailsSchema,
    });
    return {
        tool: dynamicTool,
        ...exports.NODE_DETAILS_TOOL,
    };
}
