// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/sms77/v1/resource_sms/operation_send.schema.js
/**
 * seven Node - Version 1 - Zod Schema
 * Discriminator: resource=sms, operation=send
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('sms').default('sms'),
      operation: z.literal('send').default('send'),
      from: stringOrExpression.optional(),
      to: stringOrExpression.optional(),
      message: stringOrExpression.optional(),
      options: z.object({ delay: stringOrExpression.optional(), foreign_id: stringOrExpression.optional(), flash: booleanOrExpression.optional(), label: stringOrExpression.optional(), performance_tracking: booleanOrExpression.optional(), ttl: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
