// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedrive/v1/resource_deal/operation_update.ts
/**
 * Pipedrive Node - Version 1
 * Discriminator: resource=deal, operation=update
 */


interface Credentials {
  pipedriveApi: CredentialReference;
  pipedriveOAuth2Api: CredentialReference;
}

/** Update an activity */
export type PipedriveV1DealUpdateParams = {
  resource: 'deal';
  operation: 'update';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * ID of the deal to update
 * @default 0
 */
    dealId?: number | Expression<number>;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** Currency of the deal. Accepts a 3-character currency code. Like EUR, USD, ...
     * @default USD
     */
    currency?: string | Expression<string> | PlaceholderValue;
    /** Adds a custom property to set also values which have not been predefined
     * @default {}
     */
    customProperties?: {
        /** Property
     */
    property?: Array<{
      /** Name of the custom field to set. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
       */
      name?: string | Expression<string>;
      /** Value of the property to set
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
    /** ID of the active user whom the activity will be assigned to. If omitted, the activity will be assigned to the authorized user. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    user_id?: string | Expression<string>;
    /** Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     */
    label?: string | Expression<string>;
    /** Reason why the deal was lost
     */
    lost_reason?: string | Expression<string> | PlaceholderValue;
    /** ID of the organization this deal will be associated with. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    org_id?: string | Expression<string>;
    /** ID of the person this deal will be associated with
     * @default 0
     */
    person_id?: number | Expression<number>;
    /** Deal success probability percentage
     * @default 0
     */
    probability?: number | Expression<number>;
    /** ID of the stage this deal will be placed in a pipeline. If omitted, the deal will be placed in the first stage of the default pipeline. (PIPELINE &gt; STAGE). Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    stage_id?: string | Expression<string>;
    /** The status of the deal. If not provided it will automatically be set to "open".
     * @default open
     */
    status?: 'open' | 'won' | 'lost' | 'deleted' | Expression<string>;
    /** The title of the deal
     */
    title?: string | Expression<string> | PlaceholderValue;
    /** Value of the deal. If not set it will automatically be set to 0.
     * @default 0
     */
    value?: number | Expression<number>;
    /** Visibility of the deal. If omitted, visibility will be set to the default visibility setting of this item type for the authorized user.
     * @default 3
     */
    visible_to?: '1' | '3' | Expression<string>;
  };
/**
 * By default do custom properties have to be set as ID instead of their actual name. Also option fields have to be set as ID instead of their actual value. If this option gets set they get automatically encoded.
 * @default false
 */
    encodeProperties?: boolean | Expression<boolean>;
};

export type PipedriveV1DealUpdateOutput = {
  active?: boolean;
  activities_count?: number;
  add_time?: string;
  cc_email?: string;
  creator_user_id?: {
    active_flag?: boolean;
    email?: string;
    has_pic?: number;
    id?: number;
    name?: string;
    value?: number;
  };
  currency?: string;
  deleted?: boolean;
  done_activities_count?: number;
  email_messages_count?: number;
  files_count?: number;
  followers_count?: number;
  formatted_value?: string;
  formatted_weighted_value?: string;
  id?: number;
  notes_count?: number;
  org_hidden?: boolean;
  owner_name?: string;
  participants_count?: number;
  person_hidden?: boolean;
  person_id?: {
    active_flag?: boolean;
    email?: Array<{
      label?: string;
      primary?: boolean;
      value?: string;
    }>;
    name?: string;
    owner_id?: number;
    phone?: Array<{
      label?: string;
      primary?: boolean;
      value?: string;
    }>;
    value?: number;
  };
  pipeline_id?: number;
  products_count?: number;
  stage_id?: number;
  stage_order_nr?: number;
  status?: string;
  title?: string;
  undone_activities_count?: number;
  update_time?: string;
  user_id?: {
    active_flag?: boolean;
    email?: string;
    has_pic?: number;
    id?: number;
    name?: string;
    value?: number;
  };
  visible_to?: string;
  weighted_value_currency?: string;
};

export type PipedriveV1DealUpdateNode = {
  type: 'n8n-nodes-base.pipedrive';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PipedriveV1DealUpdateParams>;
  output?: Items<PipedriveV1DealUpdateOutput>;
};
