// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedrive/v1/resource_deal/operation_search.schema.js
/**
 * Pipedrive Node - Version 1 - Zod Schema
 * Discriminator: resource=deal, operation=search
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('deal').default('deal'),
      operation: z.literal('search'),
      authentication: z.union([z.literal('apiToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      term: stringOrExpression.optional(),
      exactMatch: booleanOrExpression.optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      additionalFields: z.object({ includeFields: stringOrExpression.optional(), organizationId: stringOrExpression.optional(), personId: stringOrExpression.optional(), fields: z.array(z.union([z.literal('custom_fields'), z.literal('notes'), z.literal('title')])).optional(), status: z.union([z.literal('open'), z.literal('won'), z.literal('lost'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
