// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedrive/v1/resource_deal/operation_get_all.schema.js
/**
 * Pipedrive Node - Version 1 - Zod Schema
 * Discriminator: resource=deal, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('deal').default('deal'),
      operation: z.literal('getAll'),
      authentication: z.union([z.literal('apiToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      resolveProperties: booleanOrExpression.optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filters: z.object({ filter_id: stringOrExpression.optional(), stage_id: stringOrExpression.optional(), status: z.union([z.literal('all_not_deleted'), z.literal('deleted'), z.literal('lost'), z.literal('open'), z.literal('won'), expressionSchema]).optional(), user_id: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
