// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedrive/v1/resource_deal/operation_get_all.ts
/**
 * Pipedrive Node - Version 1
 * Discriminator: resource=deal, operation=getAll
 */


interface Credentials {
  pipedriveApi: CredentialReference;
  pipedriveOAuth2Api: CredentialReference;
}

/** Get data of many activities */
export type PipedriveV1DealGetAllParams = {
  resource: 'deal';
  operation: 'getAll';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * By default do custom properties get returned only as ID instead of their actual name. Also option fields contain only the ID instead of their actual value. If this option gets set they get automatically resolved.
 * @default false
 */
    resolveProperties?: boolean | Expression<boolean>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Predefined filter to apply to the deals to retrieve. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    filter_id?: string | Expression<string>;
    /** ID of the stage to filter deals by. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    stage_id?: string | Expression<string>;
    /** Status to filter deals by. Defaults to &lt;code&gt;all_not_deleted&lt;/code&gt;
     * @default all_not_deleted
     */
    status?: 'all_not_deleted' | 'deleted' | 'lost' | 'open' | 'won' | Expression<string>;
    /** ID of the user to filter deals by. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    user_id?: string | Expression<string>;
  };
};

export type PipedriveV1DealGetAllOutput = {
  active?: boolean;
  activities_count?: number;
  add_time?: string;
  archive_time?: null;
  cc_email?: string;
  creator_user_id?: {
    active_flag?: boolean;
    email?: string;
    has_pic?: number;
    id?: number;
    name?: string;
    value?: number;
  };
  currency?: string;
  deleted?: boolean;
  done_activities_count?: number;
  email_messages_count?: number;
  files_count?: number;
  followers_count?: number;
  formatted_value?: string;
  formatted_weighted_value?: string;
  id?: number;
  is_archived?: boolean;
  notes_count?: number;
  org_hidden?: boolean;
  org_id?: {
    active_flag?: boolean;
    cc_email?: string;
    label_ids?: Array<number>;
    name?: string;
    owner_id?: number;
    owner_name?: string;
    people_count?: number;
    value?: number;
  };
  origin?: string;
  owner_name?: string;
  participants_count?: number;
  person_hidden?: boolean;
  pipeline_id?: number;
  products_count?: number;
  stage_id?: number;
  stage_order_nr?: number;
  status?: string;
  title?: string;
  undone_activities_count?: number;
  update_time?: string;
  user_id?: {
    active_flag?: boolean;
    email?: string;
    has_pic?: number;
    id?: number;
    name?: string;
    value?: number;
  };
  visible_to?: string;
  weighted_value_currency?: string;
};

export type PipedriveV1DealGetAllNode = {
  type: 'n8n-nodes-base.pipedrive';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PipedriveV1DealGetAllParams>;
  output?: Items<PipedriveV1DealGetAllOutput>;
};
