// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedrive/v1/resource_deal/operation_get.ts
/**
 * Pipedrive Node - Version 1
 * Discriminator: resource=deal, operation=get
 */


interface Credentials {
  pipedriveApi: CredentialReference;
  pipedriveOAuth2Api: CredentialReference;
}

/** Get data of an activity */
export type PipedriveV1DealGetParams = {
  resource: 'deal';
  operation: 'get';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * ID of the deal to get
 * @default 0
 */
    dealId?: number | Expression<number>;
/**
 * By default do custom properties get returned only as ID instead of their actual name. Also option fields contain only the ID instead of their actual value. If this option gets set they get automatically resolved.
 * @default false
 */
    resolveProperties?: boolean | Expression<boolean>;
};

export type PipedriveV1DealGetOutput = {
  '0bb28fa193c6b091813e9ba57a42aae322b1f098'?: string;
  '0d2381bc2f18042052f532661380b2526d5d1418'?: string;
  '2b88250c8fe6de6a9c18a843012c3be4120bc977'?: string;
  '509080cf228cf9363d8d822ae4e7391e42b46bd0'?: string;
  '5a3096c3afe73de34f661fb2f39fd1730b8b9a14'?: number;
  '5bf837804980758d9b007751a08a0282c400d7d8'?: string;
  '6978a0bb41b5952de4d88d432af93fcd1cee3c48'?: string;
  '6ed765e95e07cb884c5ca83bed7f9129687b06f1'?: string;
  '861f700a2a00ee759b696c61c68ad81d6add71b8'?: string;
  '943fc1f0be467988db62ed6e53b7662a9e03da71'?: string;
  active?: boolean;
  activities_count?: number;
  add_time?: string;
  age?: {
    d?: number;
    h?: number;
    i?: number;
    m?: number;
    s?: number;
    total_seconds?: number;
    y?: number;
  };
  average_stage_progress?: number;
  average_time_to_won?: {
    d?: number;
    h?: number;
    i?: number;
    m?: number;
    s?: number;
    total_seconds?: number;
    y?: number;
  };
  cc_email?: string;
  creator_user_id?: {
    active_flag?: boolean;
    email?: string;
    has_pic?: number;
    id?: number;
    name?: string;
    value?: number;
  };
  currency?: string;
  deleted?: boolean;
  done_activities_count?: number;
  email_messages_count?: number;
  files_count?: number;
  followers_count?: number;
  formatted_value?: string;
  formatted_weighted_value?: string;
  id?: number;
  next_activity_duration?: null;
  next_activity_note?: null;
  next_activity_time?: null;
  notes_count?: number;
  org_hidden?: boolean;
  owner_name?: string;
  participants_count?: number;
  person_hidden?: boolean;
  person_id?: {
    active_flag?: boolean;
    email?: Array<{
      label?: string;
      primary?: boolean;
      value?: string;
    }>;
    name?: string;
    owner_id?: number;
    phone?: Array<{
      primary?: boolean;
      value?: string;
    }>;
    value?: number;
  };
  pipeline_id?: number;
  products_count?: number;
  stage_id?: number;
  stage_order_nr?: number;
  status?: string;
  stay_in_pipeline_stages?: {
    order_of_stages?: Array<number>;
  };
  title?: string;
  undone_activities_count?: number;
  update_time?: string;
  user_id?: {
    active_flag?: boolean;
    email?: string;
    has_pic?: number;
    id?: number;
    name?: string;
    value?: number;
  };
  visible_to?: string;
  weighted_value_currency?: string;
};

export type PipedriveV1DealGetNode = {
  type: 'n8n-nodes-base.pipedrive';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PipedriveV1DealGetParams>;
  output?: Items<PipedriveV1DealGetOutput>;
};
