// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedrive/v1/resource_deal/operation_create.schema.js
/**
 * Pipedrive Node - Version 1 - Zod Schema
 * Discriminator: resource=deal, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('deal').default('deal'),
      operation: z.literal('create').default('create'),
      authentication: z.union([z.literal('apiToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      title: stringOrExpression.optional(),
      associateWith: z.union([z.literal('organization'), z.literal('person'), expressionSchema]).optional(),
      org_id: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"associateWith":["organization"]}}, defaults: {"associateWith":"organization"} }),
      person_id: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"associateWith":["person"]}}, defaults: {"associateWith":"organization"} }),
      additionalFields: z.object({ currency: stringOrExpression.optional(), customProperties: z.unknown().optional(), label: stringOrExpression.optional(), lost_reason: stringOrExpression.optional(), org_id: numberOrExpression.optional(), person_id: numberOrExpression.optional(), probability: numberOrExpression.optional(), stage_id: stringOrExpression.optional(), status: z.union([z.literal('open'), z.literal('won'), z.literal('lost'), z.literal('deleted'), expressionSchema]).optional(), user_id: stringOrExpression.optional(), value: numberOrExpression.optional(), visible_to: z.union([z.literal('1'), z.literal('3'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
