// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/hubspot/v2/resource_company/operation_get_recently_created_updated.ts
/**
 * HubSpot Node - Version 2
 * Discriminator: resource=company, operation=getRecentlyCreatedUpdated
 */


interface Credentials {
  hubspotApi: CredentialReference;
  hubspotAppToken: CredentialReference;
  hubspotOAuth2Api: CredentialReference;
}

/** Get recently created/updated contacts */
export type HubspotV2CompanyGetRecentlyCreatedUpdatedParams = {
  resource: 'company';
  operation: 'getRecentlyCreatedUpdated';
  authentication?: 'apiKey' | 'appToken' | 'oAuth2' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Options
 * @default {}
 */
    additionalFields?: {
    /** Only return companys created after timestamp x. When using expressions, the time should be specified in YYYY-MM-DD hh-mm-ss format.
     */
    since?: string | Expression<string>;
    /** Whether to include specific Company properties in the returned results. Choose from a list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     * @default {}
     */
    propertiesCollection?: {
        /** Companies Properties to Include
     */
    propertiesValues?: {
      /** Whether to include specific Company properties in the returned results. Choose from a list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;. Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
       * @default []
       */
      properties?: string[];
      /** Specify if the current value for a property should be fetched, or the value and all the historical values for that property
       * @default valueAndHistory
       */
      propertyMode?: 'valueAndHistory' | 'valueOnly' | Expression<string>;
    };
  };
  };
};

export type HubspotV2CompanyGetRecentlyCreatedUpdatedNode = {
  type: 'n8n-nodes-base.hubspot';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<HubspotV2CompanyGetRecentlyCreatedUpdatedParams>;
};
