// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/hubspot/v2/resource_company/operation_get.ts
/**
 * HubSpot Node - Version 2
 * Discriminator: resource=company, operation=get
 */


interface Credentials {
  hubspotApi: CredentialReference;
  hubspotAppToken: CredentialReference;
  hubspotOAuth2Api: CredentialReference;
}

/** Get a contact */
export type HubspotV2CompanyGetParams = {
  resource: 'company';
  operation: 'get';
  authentication?: 'apiKey' | 'appToken' | 'oAuth2' | Expression<string>;
/**
 * Company to Get
 * @default {"mode":"list","value":""}
 */
    companyId?: { __rl: true; mode: 'list' | 'id'; value: string; cachedResultName?: string };
/**
 * Options
 * @default {}
 */
    additionalFields?: {
    /** Whether to return any merge history if the company has been previously merged with another company record. Defaults to false.
     * @default false
     */
    includeMergeAudits?: boolean | Expression<boolean>;
  };
};

export type HubspotV2CompanyGetNode = {
  type: 'n8n-nodes-base.hubspot';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<HubspotV2CompanyGetParams>;
};
