// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/notion/v1/resource_block/operation_get_all.ts
/**
 * Notion Node - Version 1
 * Discriminator: resource=block, operation=getAll
 */


interface Credentials {
  notionApi: CredentialReference;
}

/** Get many child blocks */
export type NotionV1BlockGetAllParams = {
  resource: 'block';
  operation: 'getAll';
/**
 * The Notion Block to get all children from
 * @default {"mode":"url","value":""}
 */
    blockId?: { __rl: true; mode: 'url' | 'id'; value: string; cachedResultName?: string };
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Also Fetch Nested Blocks
 * @default false
 */
    fetchNestedBlocks?: boolean | Expression<boolean>;
};

export type NotionV1BlockGetAllNode = {
  type: 'n8n-nodes-base.notion';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<NotionV1BlockGetAllParams>;
};
