// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mySql/v1.schema.js
/**
 * MySQL Node - Version 1 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    operation: z.union([z.literal('executeQuery'), z.literal('insert'), z.literal('update')]).optional(),
    query: resolveSchema({ parameters, schema: z.string(), required: false, displayOptions: {"show":{"operation":["executeQuery"]}}, defaults: {"operation":"insert"} }),
    table: resolveSchema({ parameters, schema: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('name')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]), required: false, displayOptions: {"show":{"operation":["insert","update"]}}, defaults: {"operation":"insert"} }),
    columns: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"operation":["insert","update"]}}, defaults: {"operation":"insert"} }),
    options: resolveSchema({ parameters, schema: z.object({ ignore: booleanOrExpression.optional(), priority: z.union([z.literal('LOW_PRIORITY'), z.literal('HIGH_PRIORITY'), expressionSchema]).optional() }), required: false, displayOptions: {"show":{"operation":["insert"]}}, defaults: {"operation":"insert"} }),
    updateKey: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"operation":["update"]}}, defaults: {"operation":"insert"} }),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
