// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/openWeatherMap/v1.schema.js
/**
 * OpenWeatherMap Node - Version 1 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    operation: z.union([z.literal('currentWeather'), z.literal('5DayForecast')]).optional(),
    format: z.union([z.literal('imperial'), z.literal('metric'), z.literal('standard'), expressionSchema]).optional(),
    locationSelection: z.union([z.literal('cityName'), z.literal('cityId'), z.literal('coordinates'), z.literal('zipCode'), expressionSchema]).optional(),
    cityName: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"locationSelection":["cityName"]}}, defaults: {"locationSelection":"cityName"} }),
    cityId: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"locationSelection":["cityId"]}}, defaults: {"locationSelection":"cityName"} }),
    latitude: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"locationSelection":["coordinates"]}}, defaults: {"locationSelection":"cityName"} }),
    longitude: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"locationSelection":["coordinates"]}}, defaults: {"locationSelection":"cityName"} }),
    zipCode: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"locationSelection":["zipCode"]}}, defaults: {"locationSelection":"cityName"} }),
    language: stringOrExpression.optional(),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
