// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/lemlist/v1/resource_lead/operation_create.ts
/**
 * Lemlist Node - Version 1
 * Discriminator: resource=lead, operation=create
 */


interface Credentials {
  lemlistApi: CredentialReference;
}

export type LemlistV1LeadCreateParams = {
  resource: 'lead';
  operation: 'create';
/**
 * ID of the campaign to create the lead under. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 * @default []
 */
    campaignId?: string | Expression<string>;
/**
 * Email of the lead to create
 */
    email?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Company name of the lead to create
     */
    companyName?: string | Expression<string> | PlaceholderValue;
    /** Whether to do not insert if this email is already present in another campaign
     * @default false
     */
    deduplicate?: boolean | Expression<boolean>;
    /** First name of the lead to create
     */
    firstName?: string | Expression<string> | PlaceholderValue;
    /** Last name of the lead to create
     */
    lastName?: string | Expression<string> | PlaceholderValue;
    /** Icebreaker of the lead to create
     */
    icebreaker?: string | Expression<string> | PlaceholderValue;
    /** Phone number of the lead to create
     */
    phone?: string | Expression<string> | PlaceholderValue;
    /** Picture URL of the lead to create
     */
    picture?: string | Expression<string> | PlaceholderValue;
    /** LinkedIn URL of the lead to create
     */
    linkedinUrl?: string | Expression<string> | PlaceholderValue;
  };
};

export type LemlistV1LeadCreateNode = {
  type: 'n8n-nodes-base.lemlist';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<LemlistV1LeadCreateParams>;
};
