// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/monicaCrm/v1/resource_contact_field/operation_update.ts
/**
 * Monica CRM Node - Version 1
 * Discriminator: resource=contactField, operation=update
 */


interface Credentials {
  monicaCrmApi: CredentialReference;
}

/** Update an activity */
export type MonicaCrmV1ContactFieldUpdateParams = {
  resource: 'contactField';
  operation: 'update';
/**
 * ID of the contact to associate the contact field with
 */
    contactId?: string | Expression<string> | PlaceholderValue;
/**
 * ID of the contact field to update
 */
    contactFieldId?: string | Expression<string> | PlaceholderValue;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    contactFieldTypeId?: string | Expression<string>;
/**
 * Content of the contact field - max 255 characters
 */
    data?: string | Expression<string> | PlaceholderValue;
};

export type MonicaCrmV1ContactFieldUpdateNode = {
  type: 'n8n-nodes-base.monicaCrm';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MonicaCrmV1ContactFieldUpdateParams>;
};
