// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/demio/v1/resource_report/operation_get.schema.js
/**
 * Demio Node - Version 1 - Zod Schema
 * Discriminator: resource=report, operation=get
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('report'),
      operation: z.literal('get').default('get'),
      eventId: stringOrExpression.optional(),
      dateId: stringOrExpression.optional(),
      filters: z.object({ status: z.union([z.literal('attended'), z.literal('banned'), z.literal('completed'), z.literal('did-not-attend'), z.literal('left-early'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
