// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/n8n/v1/resource_execution/operation_get_all.ts
/**
 * n8n Node - Version 1
 * Discriminator: resource=execution, operation=getAll
 */


interface Credentials {
  n8nApi: CredentialReference;
}

export type N8nV1ExecutionGetAllParams = {
  resource: 'execution';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Workflow to filter the executions by
     * @default {"mode":"list","value":""}
     */
    workflowId?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
    /** Status to filter the executions by
     * @default success
     */
    status?: 'error' | 'success' | 'waiting' | Expression<string>;
  };
/**
 * Options
 * @default {}
 */
    options?: {
    /** Whether to include the detailed execution data
     * @default false
     */
    activeWorkflows?: boolean | Expression<boolean>;
  };
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type N8nV1ExecutionGetAllNode = {
  type: 'n8n-nodes-base.n8n';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<N8nV1ExecutionGetAllParams>;
};
