// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mandrill/v1/resource_message/operation_send_template.ts
/**
 * Mandrill Node - Version 1
 * Discriminator: resource=message, operation=sendTemplate
 */


interface Credentials {
  mandrillApi: CredentialReference;
}

/** Send a message */
export type MandrillV1MessageSendTemplateParams = {
  resource: 'message';
  operation: 'sendTemplate';
/**
 * The template you want to send. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    template?: string | Expression<string>;
/**
 * Email address of the sender optional with name
 */
    fromEmail?: string | Expression<string> | PlaceholderValue;
/**
 * Email address of the recipient. Multiple ones can be separated by comma.
 */
    toEmail?: string | Expression<string> | PlaceholderValue;
/**
 * JSON Parameters
 * @default false
 */
    jsonParameters?: boolean | Expression<boolean>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Whether to enable a background sending mode that is optimized for bulk sending. In async mode, messages/send will immediately return a status of "queued" for every recipient. To handle rejections when sending in async mode, set up a webhook for the 'reject' event. Defaults to false for messages with no more than 10 recipients; messages with more than 10 recipients are always sent asynchronously, regardless of the value of async.
     * @default false
     */
    async?: boolean | Expression<boolean>;
    /** Whether or not to automatically generate a text part for messages that are not given text
     * @default false
     */
    autoText?: boolean | Expression<boolean>;
    /** Whether or not to automatically generate an HTML part for messages that are not given HTML
     * @default false
     */
    autoHtml?: boolean | Expression<boolean>;
    /** An optional address to receive an exact copy of each recipient's email
     */
    bccAddress?: string | Expression<string> | PlaceholderValue;
    /** Optional from name to be used
     */
    fromName?: string | Expression<string> | PlaceholderValue;
    /** Optional string indicating the value to set for the utm_campaign tracking parameter. If this isn't provided the email's from address will be used instead.
     */
    googleAnalyticsCampaign?: string | Expression<string> | PlaceholderValue;
    /** An array of strings separated by a comma (,) indicating for which any matching URLs will automatically have Google Analytics parameters appended to their query string automatically
     */
    googleAnalyticsDomains?: string | Expression<string> | PlaceholderValue;
    /** The html you want to send
     */
    html?: string | Expression<string> | PlaceholderValue;
    /** Whether or not this message is important, and should be delivered ahead of non-important messages
     * @default false
     */
    important?: boolean | Expression<boolean>;
    /** Whether or not to automatically inline all CSS styles provided in the message HTML - only for HTML documents less than 256KB in size
     * @default false
     */
    inlineCss?: boolean | Expression<boolean>;
    /** The name of the dedicated ip pool that should be used to send the message. If you do not have any dedicated IPs, this parameter has no effect. If you specify a pool that does not exist, your default pool will be used instead.
     */
    ipPool?: string | Expression<string> | PlaceholderValue;
    /** Whether or not to expose all recipients in to "To" header for each email
     * @default false
     */
    preserveRecipients?: boolean | Expression<boolean>;
    /** A custom domain to use for the messages's return-path
     */
    returnPathDomain?: string | Expression<string> | PlaceholderValue;
    /** When this message should be sent as a UTC timestamp in YYYY-MM-DD HH:MM:SS format. If you specify a time in the past, the message will be sent immediately. An additional fee applies for scheduled email, and this feature is only available to accounts with a positive balance.
     */
    sendAt?: string | Expression<string>;
    /** A custom domain to use for SPF/DKIM signing instead of mandrill(for "via" or "on behalf of" in email clients)
     */
    signingDomain?: string | Expression<string> | PlaceholderValue;
    /** The unique ID of a subaccount for this message - must already exist or will fail with an error
     */
    subAccount?: string | Expression<string> | PlaceholderValue;
    /** Subject line of the email
     */
    subject?: string | Expression<string> | PlaceholderValue;
    /** An array of string separated by a comma (,) to tag the message with. Stats are accumulated using tags, though we only store the first 100 we see, so this should not be unique or change frequently. Tags should be 50 characters or less. Any tags starting with an underscore are reserved for internal use and will cause errors.
     */
    tags?: string | Expression<string> | PlaceholderValue;
    /** Example text content
     */
    text?: string | Expression<string> | PlaceholderValue;
    /** Whether or not to turn on click tracking for the message
     * @default false
     */
    trackClicks?: boolean | Expression<boolean>;
    /** Whether or not to turn on open tracking for the message
     * @default false
     */
    trackOpens?: boolean | Expression<boolean>;
    /** A custom domain to use for tracking opens and clicks instead of mandrillapp.com
     */
    trackingDomain?: string | Expression<string> | PlaceholderValue;
    /** Whether or not to strip the query string from URLs when aggregating tracked URL data
     * @default false
     */
    urlStripQs?: boolean | Expression<boolean>;
    /** Whether to remove content logging for sensitive emails
     * @default false
     */
    viewContentLink?: boolean | Expression<boolean>;
  };
/**
 * Global merge variables
 * @displayOptions.show { jsonParameters: [true] }
 */
    mergeVarsJson?: IDataObject | string | Expression<string>;
/**
 * Per-recipient merge variables
 * @displayOptions.show { jsonParameters: [false] }
 * @default {}
 */
    mergeVarsUi?: {
        /** Vars
     */
    mergeVarsValues?: Array<{
      /** Name
       */
      name?: string | Expression<string> | PlaceholderValue;
      /** Content
       */
      content?: string | Expression<string> | PlaceholderValue;
    }>;
  };
/**
 * Metadata an associative array of user metadata. Mandrill will store this metadata and make it available for retrieval. In addition, you can select up to 10 metadata fields to index and make searchable using the Mandrill search api.
 * @displayOptions.show { jsonParameters: [false] }
 * @default {}
 */
    metadataUi?: {
        /** Metadata
     */
    metadataValues?: Array<{
      /** Name
       * @default Name of the metadata key to add.
       */
      name?: string | Expression<string> | PlaceholderValue;
      /** Value to set for the metadata key
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
/**
 * Metadata an associative array of user metadata. Mandrill will store this metadata and make it available for retrieval. In addition, you can select up to 10 metadata fields to index and make searchable using the Mandrill search api.
 * @displayOptions.show { jsonParameters: [true] }
 */
    metadataJson?: IDataObject | string | Expression<string>;
/**
 * An array of supported attachments to add to the message
 * @displayOptions.show { jsonParameters: [true] }
 */
    attachmentsJson?: IDataObject | string | Expression<string>;
/**
 * Array of supported attachments to add to the message
 * @displayOptions.show { jsonParameters: [false] }
 * @default {}
 */
    attachmentsUi?: {
        /** Attachments Values
     */
    attachmentsValues?: Array<{
      /** The MIME type of the attachment
       */
      type?: string | Expression<string> | PlaceholderValue;
      /** The file name of the attachment
       */
      name?: string | Expression<string> | PlaceholderValue;
      /** The content of the attachment as a base64-encoded string
       */
      content?: string | Expression<string> | PlaceholderValue;
    }>;
        /** Attachments Binary
     */
    attachmentsBinary?: Array<{
      /** Name of the binary properties which contain data which should be added to email as attachment
       */
      property?: string | Expression<string> | PlaceholderValue;
    }>;
  };
/**
 * Optional extra headers to add to the message (most headers are allowed)
 * @displayOptions.show { jsonParameters: [true] }
 */
    headersJson?: IDataObject | string | Expression<string>;
/**
 * Optional extra headers to add to the message (most headers are allowed)
 * @displayOptions.show { jsonParameters: [false] }
 * @default {}
 */
    headersUi?: {
        /** Values
     */
    headersValues?: Array<{
      /** Name
       */
      name?: string | Expression<string> | PlaceholderValue;
      /** Value
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
};

export type MandrillV1MessageSendTemplateOutput = {
  _id?: string;
  email?: string;
  status?: string;
};

export type MandrillV1MessageSendTemplateNode = {
  type: 'n8n-nodes-base.mandrill';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MandrillV1MessageSendTemplateParams>;
  output?: Items<MandrillV1MessageSendTemplateOutput>;
};
