// @n8n/api-types/dist/schemas/breaking-changes.schema.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MIGRATION_REPORT_TARGET_VERSION = exports.breakingChangeIssueLevelSchema = exports.breakingChangeRuleSeveritySchema = void 0;
const zod_1 = require("zod");
exports.breakingChangeRuleSeveritySchema = zod_1.z.enum(['low', 'medium', 'critical']);
exports.breakingChangeIssueLevelSchema = zod_1.z.enum(['info', 'warning', 'error']);
const breakingChangeVersionSchema = zod_1.z.enum(['v2']);
exports.MIGRATION_REPORT_TARGET_VERSION = null;
const recommendationSchema = zod_1.z.object({
    action: zod_1.z.string(),
    description: zod_1.z.string(),
});
const instanceIssueSchema = zod_1.z.object({
    title: zod_1.z.string(),
    description: zod_1.z.string(),
    level: exports.breakingChangeIssueLevelSchema,
});
const workflowIssueSchema = instanceIssueSchema.extend({
    nodeId: zod_1.z.string().optional(),
    nodeName: zod_1.z.string().optional(),
});
const affectedWorkflowSchema = zod_1.z.object({
    id: zod_1.z.string(),
    name: zod_1.z.string(),
    active: zod_1.z.boolean(),
    numberOfExecutions: zod_1.z.number(),
    lastUpdatedAt: zod_1.z.date(),
    lastExecutedAt: zod_1.z.date().optional(),
    issues: zod_1.z.array(workflowIssueSchema),
});
const ruleResultBaseSchema = zod_1.z.object({
    ruleId: zod_1.z.string(),
    ruleTitle: zod_1.z.string(),
    ruleDescription: zod_1.z.string(),
    ruleSeverity: exports.breakingChangeRuleSeveritySchema,
    ruleDocumentationUrl: zod_1.z.string().optional(),
    recommendations: zod_1.z.array(recommendationSchema),
});
const instanceRuleResultsSchema = ruleResultBaseSchema.extend({
    instanceIssues: zod_1.z.array(instanceIssueSchema),
});
const workflowRuleResultsSchema = ruleResultBaseSchema.extend({
    affectedWorkflows: zod_1.z.array(affectedWorkflowSchema),
});
const breakingChangeReportDataSchema = {
    generatedAt: zod_1.z.date(),
    targetVersion: zod_1.z.string(),
    currentVersion: zod_1.z.string(),
    instanceResults: zod_1.z.array(instanceRuleResultsSchema),
    workflowResults: zod_1.z.array(workflowRuleResultsSchema),
};
const breakingChangeReportSchema = zod_1.z.object(breakingChangeReportDataSchema).strict();
const breakingChangeLightReportDataSchema = {
    generatedAt: zod_1.z.date(),
    targetVersion: zod_1.z.string(),
    currentVersion: zod_1.z.string(),
    instanceResults: zod_1.z.array(instanceRuleResultsSchema),
    workflowResults: zod_1.z.array(workflowRuleResultsSchema.omit({ affectedWorkflows: true }).extend({
        nbAffectedWorkflows: zod_1.z.number(),
    })),
};
const breakingChangeLightReportSchema = zod_1.z.object(breakingChangeLightReportDataSchema).strict();
const breakingChangeReportResultDataSchema = zod_1.z.object({
    report: breakingChangeReportSchema,
    totalWorkflows: zod_1.z.number(),
    shouldCache: zod_1.z.boolean(),
});
const breakingChangeLightReportResultDataSchema = zod_1.z.object({
    report: breakingChangeLightReportSchema,
    totalWorkflows: zod_1.z.number(),
    shouldCache: zod_1.z.boolean(),
});
