// @n8n/api-types/dist/schemas/credential-resolver.schema.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.credentialResolversSchema = exports.credentialResolverTypesSchema = exports.credentialResolverTypeSchema = exports.credentialResolverSchema = exports.credentialResolverConfigSchema = exports.credentialResolverTypeNameSchema = exports.credentialResolverNameSchema = exports.credentialResolverIdSchema = void 0;
const zod_1 = require("zod");
exports.credentialResolverIdSchema = zod_1.z.string().max(36);
exports.credentialResolverNameSchema = zod_1.z.string().trim().min(1).max(255);
exports.credentialResolverTypeNameSchema = zod_1.z.string().trim().min(1).max(255);
exports.credentialResolverConfigSchema = zod_1.z.record(zod_1.z.unknown());
exports.credentialResolverSchema = zod_1.z.object({
    id: exports.credentialResolverIdSchema,
    name: exports.credentialResolverNameSchema,
    type: exports.credentialResolverTypeNameSchema,
    config: zod_1.z.string(),
    decryptedConfig: exports.credentialResolverConfigSchema.optional(),
    createdAt: zod_1.z.coerce.date(),
    updatedAt: zod_1.z.coerce.date(),
});
exports.credentialResolverTypeSchema = zod_1.z.object({
    name: exports.credentialResolverTypeNameSchema,
    displayName: zod_1.z.string().trim().min(1).max(255),
    description: zod_1.z.string().trim().max(1024).optional(),
    options: zod_1.z.array(zod_1.z.record(zod_1.z.unknown())).optional(),
});
exports.credentialResolverTypesSchema = zod_1.z.array(exports.credentialResolverTypeSchema);
exports.credentialResolversSchema = zod_1.z.array(exports.credentialResolverSchema);
