// @n8n/api-types/dist/schemas/secrets-provider.schema.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.reloadSecretProviderConnectionResponseSchema = exports.testSecretProviderConnectionResponseSchema = exports.secretCompletionsResponseSchema = exports.secretProviderTypeResponseSchema = exports.secretProviderConnectionSchema = exports.secretsProviderConnectionTestStateSchema = exports.secretsProviderStateSchema = exports.secretsProviderTypeSchema = void 0;
const zod_1 = require("zod");
exports.secretsProviderTypeSchema = zod_1.z.enum([
    'awsSecretsManager',
    'gcpSecretsManager',
    'vault',
    'azureKeyVault',
    'infisical',
    'onePassword',
]);
exports.secretsProviderStateSchema = zod_1.z.enum([
    'initializing',
    'initialized',
    'connecting',
    'connected',
    'error',
    'retrying',
]);
exports.secretsProviderConnectionTestStateSchema = zod_1.z.enum(['connected', 'tested', 'error']);
const projectSummarySchema = zod_1.z.object({
    id: zod_1.z.string(),
    name: zod_1.z.string(),
});
const secretSummarySchema = zod_1.z.object({
    name: zod_1.z.string(),
    credentialsCount: zod_1.z.number().optional(),
});
exports.secretProviderConnectionSchema = zod_1.z.object({
    id: zod_1.z.string(),
    name: zod_1.z.string(),
    type: exports.secretsProviderTypeSchema,
    state: exports.secretsProviderStateSchema,
    isEnabled: zod_1.z.boolean(),
    projects: zod_1.z.array(projectSummarySchema),
    settings: zod_1.z.object({}).catchall(zod_1.z.any()),
    secretsCount: zod_1.z.number(),
    secrets: zod_1.z.array(secretSummarySchema).optional(),
    createdAt: zod_1.z.string(),
    updatedAt: zod_1.z.string(),
});
exports.secretProviderTypeResponseSchema = zod_1.z.object({
    type: exports.secretsProviderTypeSchema,
    displayName: zod_1.z.string(),
    icon: zod_1.z.string(),
    properties: zod_1.z.array(zod_1.z.any()),
});
exports.secretCompletionsResponseSchema = zod_1.z.record(zod_1.z.string(), zod_1.z.array(zod_1.z.string()));
exports.testSecretProviderConnectionResponseSchema = zod_1.z.object({
    success: zod_1.z.boolean(),
    testState: exports.secretsProviderConnectionTestStateSchema,
    error: zod_1.z.string().optional(),
});
exports.reloadSecretProviderConnectionResponseSchema = zod_1.z.object({
    success: zod_1.z.boolean(),
    providers: zod_1.z.record(zod_1.z.string(), zod_1.z.object({ success: zod_1.z.boolean() })).optional(),
});
