// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/slack/v21/resource_channel/operation_join.ts
/**
 * Slack Node - Version 2.1
 * Discriminator: resource=channel, operation=join
 */


interface Credentials {
  slackApi: CredentialReference;
  slackOAuth2Api: CredentialReference;
}

/** Joins an existing conversation */
export type SlackV21ChannelJoinParams = {
  resource: 'channel';
  operation: 'join';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * The Slack channel to join
 * @default {"mode":"list","value":""}
 */
    channelId?: { __rl: true; mode: 'list' | 'id' | 'url'; value: string; cachedResultName?: string };
};

export type SlackV21ChannelJoinOutput = {
  context_team_id?: string;
  created?: number;
  creator?: string;
  id?: string;
  is_archived?: boolean;
  is_channel?: boolean;
  is_ext_shared?: boolean;
  is_general?: boolean;
  is_group?: boolean;
  is_im?: boolean;
  is_member?: boolean;
  is_mpim?: boolean;
  is_org_shared?: boolean;
  is_pending_ext_shared?: boolean;
  is_private?: boolean;
  is_shared?: boolean;
  name?: string;
  name_normalized?: string;
  parent_conversation?: null;
  previous_names?: Array<string>;
  purpose?: {
    creator?: string;
    last_set?: number;
    value?: string;
  };
  shared_team_ids?: Array<string>;
  topic?: {
    creator?: string;
    last_set?: number;
    value?: string;
  };
  unlinked?: number;
  updated?: number;
};

export type SlackV21ChannelJoinNode = {
  type: 'n8n-nodes-base.slack';
  version: 2.1;
  credentials?: Credentials;
  config: NodeConfig<SlackV21ChannelJoinParams>;
  output?: Items<SlackV21ChannelJoinOutput>;
};
