// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/slack/v21/resource_channel/operation_create.ts
/**
 * Slack Node - Version 2.1
 * Discriminator: resource=channel, operation=create
 */


interface Credentials {
  slackApi: CredentialReference;
  slackOAuth2Api: CredentialReference;
}

/** Initiates a public or private channel-based conversation */
export type SlackV21ChannelCreateParams = {
  resource: 'channel';
  operation: 'create';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Channel
 */
    channelId?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to create a Public or a Private Slack channel. &lt;a href="https://slack.com/help/articles/360017938993-What-is-a-channel"&gt;More info&lt;/a&gt;.
 * @default public
 */
    channelVisibility?: 'public' | 'private' | Expression<string>;
};

export type SlackV21ChannelCreateOutput = {
  context_team_id?: string;
  created?: number;
  creator?: string;
  id?: string;
  is_archived?: boolean;
  is_channel?: boolean;
  is_ext_shared?: boolean;
  is_general?: boolean;
  is_group?: boolean;
  is_im?: boolean;
  is_member?: boolean;
  is_mpim?: boolean;
  is_org_shared?: boolean;
  is_pending_ext_shared?: boolean;
  is_private?: boolean;
  is_shared?: boolean;
  last_read?: string;
  name?: string;
  name_normalized?: string;
  parent_conversation?: null;
  priority?: number;
  purpose?: {
    creator?: string;
    last_set?: number;
    value?: string;
  };
  shared_team_ids?: Array<string>;
  topic?: {
    creator?: string;
    last_set?: number;
    value?: string;
  };
  unlinked?: number;
  updated?: number;
};

export type SlackV21ChannelCreateNode = {
  type: 'n8n-nodes-base.slack';
  version: 2.1;
  credentials?: Credentials;
  config: NodeConfig<SlackV21ChannelCreateParams>;
  output?: Items<SlackV21ChannelCreateOutput>;
};
