// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/gmail/v21/resource_message/operation_get_all.schema.js
/**
 * Gmail Node - Version 2.1 - Zod Schema
 * Discriminator: resource=message, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('message').default('message'),
      operation: z.literal('getAll'),
      authentication: z.union([z.literal('oAuth2'), z.literal('serviceAccount'), expressionSchema]).optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      simple: booleanOrExpression.optional(),
      filters: z.object({ includeSpamTrash: booleanOrExpression.optional(), labelIds: z.array(z.string()).optional(), q: stringOrExpression.optional(), readStatus: z.union([z.literal('both'), z.literal('unread'), z.literal('read'), expressionSchema]).optional(), receivedAfter: stringOrExpression.optional(), receivedBefore: stringOrExpression.optional(), sender: stringOrExpression.optional() }).optional(),
      options: resolveSchema({ parameters, schema: z.object({ dataPropertyAttachmentsPrefixName: stringOrExpression.optional(), downloadAttachments: booleanOrExpression.optional() }), required: false, displayOptions: {"hide":{"simple":[true]}}, defaults: {"simple":true} }),
    }).optional(),
  });
};
