// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/highLevel/v2/resource_contact/operation_update.schema.js
/**
 * HighLevel Node - Version 2 - Zod Schema
 * Discriminator: resource=contact, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('contact').default('contact'),
      operation: z.literal('update'),
      contactId: stringOrExpression.optional(),
      updateFields: z.object({ address1: stringOrExpression.optional(), city: stringOrExpression.optional(), customFields: z.unknown().optional(), dnd: booleanOrExpression.optional(), email: stringOrExpression.optional(), firstName: stringOrExpression.optional(), lastName: stringOrExpression.optional(), name: stringOrExpression.optional(), phone: stringOrExpression.optional(), postalCode: stringOrExpression.optional(), state: stringOrExpression.optional(), tags: stringOrExpression.optional(), timezone: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(), website: stringOrExpression.optional() }).optional(),
      requestOptions: z.object({ batching: z.unknown().optional(), allowUnauthorizedCerts: z.boolean().optional(), proxy: stringOrExpression.optional(), timeout: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
