// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/highLevel/v2/resource_contact/operation_get_all.ts
/**
 * HighLevel Node - Version 2
 * Discriminator: resource=contact, operation=getAll
 */


interface Credentials {
  highLevelOAuth2Api: CredentialReference;
}

export type HighLevelV2ContactGetAllParams = {
  resource: 'contact';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Query will search on these fields: Name, Phone, Email, Tags, and Company Name
     */
    query?: string | Expression<string> | PlaceholderValue;
  };
/**
 * Options
 * @default {}
 */
    options?: {
    /** Order
     * @default desc
     */
    order?: 'desc' | 'asc' | Expression<string>;
    /** Sort By
     * @default date_added
     */
    sortBy?: 'date_added' | 'date_updated' | Expression<string>;
  };
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type HighLevelV2ContactGetAllOutput = {
  attributions?: Array<{
    isFirst?: boolean;
    medium?: string;
    utmSessionSource?: string;
  }>;
  contactName?: string;
  customFields?: Array<{
    id?: string;
  }>;
  dateAdded?: string;
  dateUpdated?: string;
  dnd?: boolean;
  id?: string;
  locationId?: string;
  tags?: Array<string>;
};

export type HighLevelV2ContactGetAllNode = {
  type: 'n8n-nodes-base.highLevel';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<HighLevelV2ContactGetAllParams>;
  output?: Items<HighLevelV2ContactGetAllOutput>;
};
