// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/trello/v1/resource_label/operation_update.schema.js
/**
 * Trello Node - Version 1 - Zod Schema
 * Discriminator: resource=label, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('label'),
      operation: z.literal('update'),
      id: stringOrExpression.optional(),
      updateFields: z.object({ name: stringOrExpression.optional(), color: z.union([z.literal('black'), z.literal('blue'), z.literal('green'), z.literal('lime'), z.literal('null'), z.literal('orange'), z.literal('pink'), z.literal('purple'), z.literal('red'), z.literal('sky'), z.literal('yellow'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
