// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/koBoToolbox/v1/resource_submission/operation_get_all.schema.js
/**
 * KoBoToolbox Node - Version 1 - Zod Schema
 * Discriminator: resource=submission, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('submission').default('submission'),
      operation: z.literal('getAll'),
      formId: stringOrExpression.optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filterType: z.union([z.literal('none'), z.literal('json'), expressionSchema]).optional(),
      filterJson: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"filterType":["json"]}}, defaults: {"filterType":"none"} }),
      options: z.object({ download: booleanOrExpression.optional(), binaryNamingScheme: z.union([z.literal('sequence'), z.literal('question'), expressionSchema]).optional(), dataPropertyAttachmentsPrefixName: stringOrExpression.optional(), fields: stringOrExpression.optional(), version: z.union([z.literal('download_url'), z.literal('download_small_url'), z.literal('download_medium_url'), z.literal('download_large_url'), expressionSchema]).optional(), selectMask: stringOrExpression.optional(), numberMask: stringOrExpression.optional(), reformat: booleanOrExpression.optional(), sort: z.union([iDataObjectSchema, z.string()]).optional() }).optional(),
    }).optional(),
  });
};
