// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/koBoToolbox/v1/resource_submission/operation_set_validation.schema.js
/**
 * KoBoToolbox Node - Version 1 - Zod Schema
 * Discriminator: resource=submission, operation=setValidation
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('submission').default('submission'),
      operation: z.literal('setValidation'),
      formId: stringOrExpression.optional(),
      submissionId: stringOrExpression.optional(),
      validationStatus: z.union([z.literal('validation_status_approved'), z.literal('validation_status_not_approved'), z.literal('validation_status_on_hold'), expressionSchema]).optional(),
    }).optional(),
  });
};
