// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftTeams/v1/resource_channel_message/operation_create.ts
/**
 * Microsoft Teams Node - Version 1
 * Discriminator: resource=channelMessage, operation=create
 */


interface Credentials {
  microsoftTeamsOAuth2Api: CredentialReference;
}

/** Create a channel */
export type MicrosoftTeamsV1ChannelMessageCreateParams = {
  resource: 'channelMessage';
  operation: 'create';
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    teamId?: string | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    channelId?: string | Expression<string>;
/**
 * The type of the content
 * @default text
 */
    messageType?: 'text' | 'html' | Expression<string>;
/**
 * The content of the item
 */
    message?: string | Expression<string> | PlaceholderValue;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Whether to append a link to this workflow at the end of the message. This is helpful if you have many workflows sending messages.
     * @default true
     */
    includeLinkToWorkflow?: boolean | Expression<boolean>;
    /** An optional ID of the message you want to reply to
     */
    makeReply?: string | Expression<string> | PlaceholderValue;
  };
};

export type MicrosoftTeamsV1ChannelMessageCreateNode = {
  type: 'n8n-nodes-base.microsoftTeams';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MicrosoftTeamsV1ChannelMessageCreateParams>;
};
