// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/currents/v1/resource_project/operation_get_insights.schema.js
/**
 * Currents Node - Version 1 - Zod Schema
 * Discriminator: resource=project, operation=getInsights
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('project'),
      operation: z.literal('getInsights'),
      projectId: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      dateStart: stringOrExpression.optional(),
      dateEnd: stringOrExpression.optional(),
      options: z.object({ authors: stringOrExpression.optional(), branches: stringOrExpression.optional(), groups: stringOrExpression.optional(), resolution: z.union([z.literal('1h'), z.literal('1d'), z.literal('1w'), expressionSchema]).optional(), tags: stringOrExpression.optional() }).optional(),
      requestOptions: z.object({ batching: z.unknown().optional(), allowUnauthorizedCerts: z.boolean().optional(), proxy: stringOrExpression.optional(), timeout: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
