// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/currents/v1/resource_project/operation_get_insights.ts
/**
 * Currents Node - Version 1
 * Discriminator: resource=project, operation=getInsights
 */


interface Credentials {
  currentsApi: CredentialReference;
}

/** Test project */
export type CurrentsV1ProjectGetInsightsParams = {
  resource: 'project';
  operation: 'getInsights';
/**
 * The Currents project
 * @default {"mode":"list","value":""}
 */
    projectId?: { __rl: true; mode: 'list' | 'id'; value: string; cachedResultName?: string };
/**
 * Start date for metrics (ISO 8601 format)
 */
    dateStart?: string | Expression<string>;
/**
 * End date for metrics (ISO 8601 format)
 */
    dateEnd?: string | Expression<string>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Filter by git authors (comma-separated for multiple)
     */
    authors?: string | Expression<string> | PlaceholderValue;
    /** Filter by branches (comma-separated for multiple)
     */
    branches?: string | Expression<string> | PlaceholderValue;
    /** Filter by groups (comma-separated for multiple)
     */
    groups?: string | Expression<string> | PlaceholderValue;
    /** Time resolution for metrics
     * @default 1d
     */
    resolution?: '1h' | '1d' | '1w' | Expression<string>;
    /** Filter by tags (comma-separated for multiple)
     */
    tags?: string | Expression<string> | PlaceholderValue;
  };
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type CurrentsV1ProjectGetInsightsNode = {
  type: 'n8n-nodes-base.currents';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<CurrentsV1ProjectGetInsightsParams>;
};
