// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/slack/v24/resource_star/operation_delete.ts
/**
 * Slack Node - Version 2.4
 * Discriminator: resource=star, operation=delete
 */


interface Credentials {
  slackApi: CredentialReference;
  slackOAuth2Api: CredentialReference;
}

export type SlackV24StarDeleteParams = {
  resource: 'star';
  operation: 'delete';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Options to set
 * @default {}
 */
    options?: {
    /** Channel to add star to, or channel where the message to add star to was posted (used with timestamp). Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    channelId?: string | Expression<string>;
    /** File to add star to
     */
    fileId?: string | Expression<string> | PlaceholderValue;
    /** File comment to add star to
     */
    fileComment?: string | Expression<string> | PlaceholderValue;
    /** Timestamp of the message to delete
     * @default 0
     */
    timestamp?: number | Expression<number>;
  };
};

export type SlackV24StarDeleteNode = {
  type: 'n8n-nodes-base.slack';
  version: 2.4;
  credentials?: Credentials;
  config: NodeConfig<SlackV24StarDeleteParams>;
};
