// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/slack/v24/resource_star/operation_add.ts
/**
 * Slack Node - Version 2.4
 * Discriminator: resource=star, operation=add
 */


interface Credentials {
  slackApi: CredentialReference;
  slackOAuth2Api: CredentialReference;
}

/** Add a star to an item */
export type SlackV24StarAddParams = {
  resource: 'star';
  operation: 'add';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Choose whether to add a star to a message or a file
 */
    target?: 'message' | 'file' | Expression<string>;
/**
 * The Slack channel to add a star to
 * @displayOptions.show { target: ["message", "file"] }
 * @default {"mode":"list","value":""}
 */
    channelId?: { __rl: true; mode: 'list' | 'id' | 'url'; value: string; cachedResultName?: string };
/**
 * File to add star to
 * @displayOptions.show { target: ["file"] }
 */
    fileId?: string | Expression<string> | PlaceholderValue;
/**
 * Timestamp of the message to add
 * @displayOptions.show { target: ["message"] }
 */
    timestamp?: number | Expression<number>;
/**
 * Options to set
 * @default {}
 */
    options?: {
    /** File comment to add star to
     */
    fileComment?: string | Expression<string> | PlaceholderValue;
  };
};

export type SlackV24StarAddNode = {
  type: 'n8n-nodes-base.slack';
  version: 2.4;
  credentials?: Credentials;
  config: NodeConfig<SlackV24StarAddParams>;
};
