// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/sendInBlue/v1/resource_attribute/operation_create.schema.js
/**
 * Brevo Node - Version 1 - Zod Schema
 * Discriminator: resource=attribute, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('attribute'),
      operation: z.literal('create').default('create'),
      attributeCategory: z.union([z.literal('calculated'), z.literal('category'), z.literal('global'), z.literal('normal'), z.literal('transactional'), expressionSchema]).optional(),
      attributeName: stringOrExpression.optional(),
      attributeType: resolveSchema({ parameters, schema: z.union([z.literal('boolean'), z.literal('date'), z.literal('float'), z.literal('text'), expressionSchema]), required: false, displayOptions: {"show":{"attributeCategory":["normal"]}}, defaults: {"attributeCategory":"normal"} }),
      attributeValue: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"attributeCategory":["global","calculated"]}}, defaults: {"attributeCategory":"normal"} }),
      attributeCategoryList: resolveSchema({ parameters, schema: z.object({ categoryEnumeration: z.unknown().optional() }), required: false, displayOptions: {"show":{"attributeCategory":["category"]}}, defaults: {"attributeCategory":"normal"} }),
      requestOptions: z.object({ batching: z.unknown().optional(), allowUnauthorizedCerts: z.boolean().optional(), proxy: stringOrExpression.optional(), timeout: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
