// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/sendInBlue/v1/resource_attribute/operation_delete.schema.js
/**
 * Brevo Node - Version 1 - Zod Schema
 * Discriminator: resource=attribute, operation=delete
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('attribute'),
      operation: z.literal('delete'),
      deleteAttributeCategory: z.union([z.literal('calculated'), z.literal('category'), z.literal('global'), z.literal('normal'), z.literal('transactional'), expressionSchema]).optional(),
      deleteAttributeName: stringOrExpression.optional(),
      requestOptions: z.object({ batching: z.unknown().optional(), allowUnauthorizedCerts: z.boolean().optional(), proxy: stringOrExpression.optional(), timeout: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
