// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/sendInBlue/v1/resource_attribute/operation_get_all.ts
/**
 * Brevo Node - Version 1
 * Discriminator: resource=attribute, operation=getAll
 */


interface Credentials {
  sendInBlueApi: CredentialReference;
}

export type SendInBlueV1AttributeGetAllParams = {
  resource: 'attribute';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type SendInBlueV1AttributeGetAllNode = {
  type: 'n8n-nodes-base.sendInBlue';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SendInBlueV1AttributeGetAllParams>;
};
