// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedrive/v1/resource_person/operation_get_all.ts
/**
 * Pipedrive Node - Version 1
 * Discriminator: resource=person, operation=getAll
 */


interface Credentials {
  pipedriveApi: CredentialReference;
  pipedriveOAuth2Api: CredentialReference;
}

/** Get data of many activities */
export type PipedriveV1PersonGetAllParams = {
  resource: 'person';
  operation: 'getAll';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * By default do custom properties get returned only as ID instead of their actual name. Also option fields contain only the ID instead of their actual value. If this option gets set they get automatically resolved.
 * @default false
 */
    resolveProperties?: boolean | Expression<boolean>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** ID of the filter to use. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    filterId?: string | Expression<string>;
    /** If supplied, only persons whose name starts with the specified letter will be returned
     */
    firstChar?: string | Expression<string> | PlaceholderValue;
    /** The field names and sorting mode separated by a comma (field_name_1 ASC, field_name_2 DESC). Only first-level field keys are supported (no nested keys).
     */
    sort?: string | Expression<string> | PlaceholderValue;
  };
};

export type PipedriveV1PersonGetAllOutput = {
  active_flag?: boolean;
  activities_count?: number;
  add_time?: string;
  birthday?: null;
  cc_email?: string;
  closed_deals_count?: number;
  company_id?: number;
  delete_time?: null;
  done_activities_count?: number;
  email?: Array<{
    label?: string;
    primary?: boolean;
    value?: string;
  }>;
  email_messages_count?: number;
  files_count?: number;
  first_char?: string;
  first_name?: string;
  followers_count?: number;
  id?: number;
  im?: Array<{
    primary?: boolean;
    value?: string;
  }>;
  label_ids?: Array<number>;
  lost_deals_count?: number;
  name?: string;
  notes_count?: number;
  open_deals_count?: number;
  owner_id?: {
    active_flag?: boolean;
    email?: string;
    has_pic?: number;
    id?: number;
    name?: string;
    value?: number;
  };
  owner_name?: string;
  participant_closed_deals_count?: number;
  participant_open_deals_count?: number;
  phone?: Array<{
    label?: string;
    primary?: boolean;
    value?: string;
  }>;
  postal_address_lat?: null;
  postal_address_long?: null;
  postal_address_subpremise?: null;
  related_closed_deals_count?: number;
  related_lost_deals_count?: number;
  related_open_deals_count?: number;
  related_won_deals_count?: number;
  undone_activities_count?: number;
  update_time?: string;
  visible_to?: string;
  won_deals_count?: number;
};

export type PipedriveV1PersonGetAllNode = {
  type: 'n8n-nodes-base.pipedrive';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PipedriveV1PersonGetAllParams>;
  output?: Items<PipedriveV1PersonGetAllOutput>;
};
