// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedrive/v1/resource_person/operation_search.ts
/**
 * Pipedrive Node - Version 1
 * Discriminator: resource=person, operation=search
 */


interface Credentials {
  pipedriveApi: CredentialReference;
  pipedriveOAuth2Api: CredentialReference;
}

/** Search a deal */
export type PipedriveV1PersonSearchParams = {
  resource: 'person';
  operation: 'search';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * The search term to look for. Minimum 2 characters (or 1 if using exact_match).
 */
    term?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Whether only full exact matches against the given term are returned. It is not case sensitive.
     * @default false
     */
    exactMatch?: boolean | Expression<boolean>;
    /** A comma-separated string array. The fields to perform the search from. Defaults to all of them.
     */
    fields?: string | Expression<string> | PlaceholderValue;
    /** Supports including optional fields in the results which are not provided by default
     */
    includeFields?: string | Expression<string> | PlaceholderValue;
    /** Will filter Deals by the provided Organization ID
     */
    organizationId?: string | Expression<string> | PlaceholderValue;
    /** Whether to return the data exactly in the way it got received from the API
     * @default false
     */
    rawData?: boolean | Expression<boolean>;
  };
};

export type PipedriveV1PersonSearchOutput = {
  custom_fields?: Array<string>;
  emails?: Array<string>;
  id?: number;
  name?: string;
  notes?: Array<string>;
  organization?: {
    id?: number;
    name?: string;
  };
  owner?: {
    id?: number;
  };
  phones?: Array<string>;
  result_score?: number;
  type?: string;
  update_time?: string;
  visible_to?: number;
};

export type PipedriveV1PersonSearchNode = {
  type: 'n8n-nodes-base.pipedrive';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PipedriveV1PersonSearchParams>;
  output?: Items<PipedriveV1PersonSearchOutput>;
};
