// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedrive/v1/resource_person/operation_update.ts
/**
 * Pipedrive Node - Version 1
 * Discriminator: resource=person, operation=update
 */


interface Credentials {
  pipedriveApi: CredentialReference;
  pipedriveOAuth2Api: CredentialReference;
}

/** Update an activity */
export type PipedriveV1PersonUpdateParams = {
  resource: 'person';
  operation: 'update';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * ID of the person to update
 * @default 0
 */
    personId?: number | Expression<number>;
/**
 * The fields to update
 * @default {}
 */
    updateFields?: {
    /** Adds a custom property to set also values which have not been predefined
     * @default {}
     */
    customProperties?: {
        /** Property
     */
    property?: Array<{
      /** Name of the custom field to set. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
       */
      name?: string | Expression<string>;
      /** Value of the property to set
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
    /** Email of the person
     */
    email?: string | Expression<string> | PlaceholderValue;
    /** Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     */
    label?: string | Expression<string>;
    /** Please be aware that it is only allowed once to change the marketing status from an old status to a new one
     * @default subscribed
     */
    marketing_status?: 'no_consent' | 'unsubscribed' | 'subscribed' | 'archived' | Expression<string>;
    /** The name of the person
     */
    name?: string | Expression<string> | PlaceholderValue;
    /** ID of the organization this deal will be associated with. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    org_id?: string | Expression<string>;
    /** Phone number of the person
     */
    phone?: string | Expression<string> | PlaceholderValue;
    /** ID of the User this person will be associated with. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    owner_id?: string | Expression<string>;
    /** Visibility of the deal. If omitted, visibility will be set to the default visibility setting of this item type for the authorized user.
     * @default 3
     */
    visible_to?: '1' | '3' | Expression<string>;
  };
/**
 * By default do custom properties have to be set as ID instead of their actual name. Also option fields have to be set as ID instead of their actual value. If this option gets set they get automatically encoded.
 * @default false
 */
    encodeProperties?: boolean | Expression<boolean>;
};

export type PipedriveV1PersonUpdateOutput = {
  active_flag?: boolean;
  activities_count?: number;
  add_time?: string;
  cc_email?: string;
  closed_deals_count?: number;
  company_id?: number;
  delete_time?: null;
  done_activities_count?: number;
  email?: Array<{
    label?: string;
    primary?: boolean;
    value?: string;
  }>;
  email_messages_count?: number;
  files_count?: number;
  first_char?: string;
  first_name?: string;
  followers_count?: number;
  id?: number;
  im?: Array<{
    primary?: boolean;
    value?: string;
  }>;
  label_ids?: Array<number>;
  lost_deals_count?: number;
  name?: string;
  notes_count?: number;
  open_deals_count?: number;
  org_id?: {
    active_flag?: boolean;
    cc_email?: string;
    label_ids?: Array<number>;
    name?: string;
    owner_id?: number;
    owner_name?: string;
    people_count?: number;
    value?: number;
  };
  owner_id?: {
    active_flag?: boolean;
    email?: string;
    has_pic?: number;
    id?: number;
    name?: string;
    value?: number;
  };
  owner_name?: string;
  participant_closed_deals_count?: number;
  participant_open_deals_count?: number;
  phone?: Array<{
    label?: string;
    primary?: boolean;
    value?: string;
  }>;
  postal_address_lat?: null;
  postal_address_long?: null;
  postal_address_subpremise?: null;
  related_closed_deals_count?: number;
  related_lost_deals_count?: number;
  related_open_deals_count?: number;
  related_won_deals_count?: number;
  undone_activities_count?: number;
  update_time?: string;
  visible_to?: string;
  won_deals_count?: number;
};

export type PipedriveV1PersonUpdateNode = {
  type: 'n8n-nodes-base.pipedrive';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PipedriveV1PersonUpdateParams>;
  output?: Items<PipedriveV1PersonUpdateOutput>;
};
