// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedrive/v1/resource_person/operation_get.ts
/**
 * Pipedrive Node - Version 1
 * Discriminator: resource=person, operation=get
 */


interface Credentials {
  pipedriveApi: CredentialReference;
  pipedriveOAuth2Api: CredentialReference;
}

/** Get data of an activity */
export type PipedriveV1PersonGetParams = {
  resource: 'person';
  operation: 'get';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * ID of the person to get
 * @default 0
 */
    personId?: number | Expression<number>;
/**
 * By default do custom properties get returned only as ID instead of their actual name. Also option fields contain only the ID instead of their actual value. If this option gets set they get automatically resolved.
 * @default false
 */
    resolveProperties?: boolean | Expression<boolean>;
};

export type PipedriveV1PersonGetOutput = {
  active_flag?: boolean;
  activities_count?: number;
  add_time?: string;
  birthday?: null;
  cc_email?: string;
  closed_deals_count?: number;
  company_id?: number;
  done_activities_count?: number;
  email?: Array<{
    label?: string;
    primary?: boolean;
    value?: string;
  }>;
  email_messages_count?: number;
  files_count?: number;
  first_char?: string;
  first_name?: string;
  followers_count?: number;
  id?: number;
  im?: Array<{
    primary?: boolean;
    value?: string;
  }>;
  label_ids?: Array<number>;
  lost_deals_count?: number;
  name?: string;
  notes_count?: number;
  open_deals_count?: number;
  org_id?: {
    active_flag?: boolean;
    cc_email?: string;
    label_ids?: Array<number>;
    name?: string;
    owner_id?: number;
    owner_name?: string;
    people_count?: number;
    value?: number;
  };
  owner_id?: {
    active_flag?: boolean;
    email?: string;
    has_pic?: number;
    id?: number;
    name?: string;
    value?: number;
  };
  owner_name?: string;
  participant_closed_deals_count?: number;
  participant_open_deals_count?: number;
  phone?: Array<{
    label?: string;
    primary?: boolean;
    value?: string;
  }>;
  postal_address_admin_area_level_1?: null;
  postal_address_admin_area_level_2?: null;
  postal_address_country?: null;
  postal_address_formatted_address?: null;
  postal_address_lat?: null;
  postal_address_locality?: null;
  postal_address_long?: null;
  postal_address_postal_code?: null;
  postal_address_route?: null;
  postal_address_street_number?: null;
  postal_address_sublocality?: null;
  postal_address_subpremise?: null;
  related_closed_deals_count?: number;
  related_lost_deals_count?: number;
  related_open_deals_count?: number;
  related_won_deals_count?: number;
  undone_activities_count?: number;
  update_time?: string;
  visible_to?: string;
  won_deals_count?: number;
};

export type PipedriveV1PersonGetNode = {
  type: 'n8n-nodes-base.pipedrive';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PipedriveV1PersonGetParams>;
  output?: Items<PipedriveV1PersonGetOutput>;
};
