// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/freshservice/v1/resource_requester/operation_create.schema.js
/**
 * Freshservice Node - Version 1 - Zod Schema
 * Discriminator: resource=requester, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('requester'),
      operation: z.literal('create').default('create'),
      firstName: stringOrExpression.optional(),
      primaryEmail: stringOrExpression.optional(),
      additionalFields: z.object({ address: stringOrExpression.optional(), background_information: stringOrExpression.optional(), department_ids: z.array(z.string()).optional(), job_title: stringOrExpression.optional(), language: z.union([z.literal('en'), z.literal('ar'), z.literal('ca'), z.literal('cs'), z.literal('cy-GB'), z.literal('da'), z.literal('de'), z.literal('es'), z.literal('es-LA'), z.literal('et'), z.literal('fi'), z.literal('fr'), z.literal('he'), z.literal('hu'), z.literal('id'), z.literal('it'), z.literal('ja-JP'), z.literal('ko'), z.literal('LV'), z.literal('nb-NO'), z.literal('nl'), z.literal('pl'), z.literal('pt'), z.literal('pt-BR'), z.literal('pt-PT'), z.literal('ru-RU'), z.literal('sk'), z.literal('sk-SK'), z.literal('sl'), z.literal('sv-SE'), z.literal('th'), z.literal('tr'), z.literal('UK'), z.literal('vi'), z.literal('zh-CN'), z.literal('zh-TW'), expressionSchema]).optional(), last_name: stringOrExpression.optional(), location_id: stringOrExpression.optional(), mobile_phone_number: stringOrExpression.optional(), secondary_emails: stringOrExpression.optional(), time_format: z.union([z.literal('12h'), z.literal('24h'), expressionSchema]).optional(), work_phone_number: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
