// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/erpNext/v1/resource_document/operation_get_all.schema.js
/**
 * ERPNext Node - Version 1 - Zod Schema
 * Discriminator: resource=document, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('document').default('document'),
      operation: z.literal('getAll'),
      docType: stringOrExpression.optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      options: z.object({ fields: z.array(z.string()).optional(), filters: z.unknown().optional() }).optional(),
    }).optional(),
  });
};
