// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/erpNext/v1/resource_document/operation_create.schema.js
/**
 * ERPNext Node - Version 1 - Zod Schema
 * Discriminator: resource=document, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('document').default('document'),
      operation: z.literal('create').default('create'),
      docType: stringOrExpression.optional(),
      properties: z.object({ customProperty: z.array(z.object({ field: stringOrExpression.optional(), value: stringOrExpression.optional() })).optional() }).optional(),
    }).optional(),
  });
};
