// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleCalendar/v12/resource_calendar/operation_availability.schema.js
/**
 * Google Calendar Node - Version 1.2 - Zod Schema
 * Discriminator: resource=calendar, operation=availability
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('calendar'),
      operation: z.literal('availability').default('availability'),
      calendar: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      timeMin: stringOrExpression.optional(),
      timeMax: stringOrExpression.optional(),
      options: z.object({ outputFormat: z.union([z.literal('availability'), z.literal('bookedSlots'), z.literal('raw'), expressionSchema]).optional(), timezone: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
