// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/raindrop/v1/resource_collection/operation_update.schema.js
/**
 * Raindrop Node - Version 1 - Zod Schema
 * Discriminator: resource=collection, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('collection').default('collection'),
      operation: z.literal('update'),
      collectionId: stringOrExpression.optional(),
      updateFields: z.object({ cover: stringOrExpression.optional(), public: booleanOrExpression.optional(), parentId: stringOrExpression.optional(), sort: numberOrExpression.optional(), title: stringOrExpression.optional(), view: z.union([z.literal('list'), z.literal('simple'), z.literal('grid'), z.literal('Masonry'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
