// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/helpScout/v1/resource_mailbox/operation_get_all.ts
/**
 * Help Scout Node - Version 1
 * Discriminator: resource=mailbox, operation=getAll
 */


interface Credentials {
  helpScoutOAuth2Api: CredentialReference;
}

/** Get many conversations */
export type HelpScoutV1MailboxGetAllParams = {
  resource: 'mailbox';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
};

export type HelpScoutV1MailboxGetAllOutput = {
  _links?: {
    fields?: {
      href?: string;
    };
    folders?: {
      href?: string;
    };
    self?: {
      href?: string;
    };
  };
  createdAt?: string;
  email?: string;
  id?: number;
  name?: string;
  slug?: string;
  updatedAt?: string;
};

export type HelpScoutV1MailboxGetAllNode = {
  type: 'n8n-nodes-base.helpScout';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<HelpScoutV1MailboxGetAllParams>;
  output?: Items<HelpScoutV1MailboxGetAllOutput>;
};
