// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/invoiceNinja/v2/resource_client/operation_get_all.schema.js
/**
 * Invoice Ninja Node - Version 2 - Zod Schema
 * Discriminator: resource=client, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('client').default('client'),
      operation: z.literal('getAll'),
      apiVersion: z.union([z.literal('v4'), z.literal('v5'), expressionSchema]).optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      options: z.object({ include: z.union([z.literal('invoices'), expressionSchema]).optional(), status: z.union([z.literal('active'), z.literal('archived'), z.literal('deleted'), expressionSchema]).optional(), createdAt: stringOrExpression.optional(), updatedAt: stringOrExpression.optional(), isDeleted: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
