// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/slack/v24/resource_file/operation_get_all.ts
/**
 * Slack Node - Version 2.4
 * Discriminator: resource=file, operation=getAll
 */


interface Credentials {
  slackApi: CredentialReference;
  slackOAuth2Api: CredentialReference;
}

/** Get many channels in a Slack team */
export type SlackV24FileGetAllParams = {
  resource: 'file';
  operation: 'getAll';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Channel containing the file to be listed. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    channelId?: string | Expression<string>;
    /** Whether to show truncated file info for files hidden due to being too old, and the team who owns the file being over the file limit
     * @default false
     */
    showFilesHidden?: boolean | Expression<boolean>;
    /** Filter files created after this timestamp (inclusive)
     */
    tsFrom?: string | Expression<string> | PlaceholderValue;
    /** Filter files created before this timestamp (inclusive)
     */
    tsTo?: string | Expression<string> | PlaceholderValue;
    /** Filter files by type
     * @default ["all"]
     */
    types?: Array<'all' | 'gdocs' | 'images' | 'pdfs' | 'snippets' | 'spaces' | 'zips'>;
    /** Filter files created by a single user. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    userId?: string | Expression<string>;
  };
};

export type SlackV24FileGetAllOutput = {
  channels?: Array<string>;
  comments_count?: number;
  created?: number;
  display_as_bot?: boolean;
  editable?: boolean;
  external_type?: string;
  filetype?: string;
  groups?: Array<string>;
  id?: string;
  ims?: Array<string>;
  is_external?: boolean;
  is_public?: boolean;
  media_display_type?: string;
  mimetype?: string;
  mode?: string;
  name?: string;
  permalink?: string;
  pretty_type?: string;
  public_url_shared?: boolean;
  size?: number;
  timestamp?: number;
  title?: string;
  url_private?: string;
  url_private_download?: string;
  user?: string;
  user_team?: string;
  username?: string;
};

export type SlackV24FileGetAllNode = {
  type: 'n8n-nodes-base.slack';
  version: 2.4;
  credentials?: Credentials;
  config: NodeConfig<SlackV24FileGetAllParams>;
  output?: Items<SlackV24FileGetAllOutput>;
};
