// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/hubspot/v1/resource_contact/operation_search.schema.js
/**
 * HubSpot Node - Version 1 - Zod Schema
 * Discriminator: resource=contact, operation=search
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('contact'),
      operation: z.literal('search'),
      authentication: z.union([z.literal('apiKey'), z.literal('appToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filterGroupsUi: z.object({ filterGroupsValues: z.array(z.object({ filtersUi: z.unknown().optional() })).optional() }).optional(),
      additionalFields: z.object({ direction: z.union([z.literal('ASCENDING'), z.literal('DESCENDING'), expressionSchema]).optional(), properties: z.array(z.string()).optional(), query: stringOrExpression.optional(), sortBy: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
