// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/hubspot/v1/resource_contact/operation_get_recently_created_updated.ts
/**
 * HubSpot Node - Version 1
 * Discriminator: resource=contact, operation=getRecentlyCreatedUpdated
 */


interface Credentials {
  hubspotApi: CredentialReference;
  hubspotAppToken: CredentialReference;
  hubspotOAuth2Api: CredentialReference;
}

/** Get recently created/updated contacts */
export type HubspotV1ContactGetRecentlyCreatedUpdatedParams = {
  resource: 'contact';
  operation: 'getRecentlyCreatedUpdated';
  authentication?: 'apiKey' | 'appToken' | 'oAuth2' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Specify which form submissions should be fetched
     * @default all
     */
    formSubmissionMode?: 'all' | 'none' | 'newest' | 'oldest' | Expression<string>;
    /** Whether current list memberships should be fetched for the contact
     * @default true
     */
    listMemberships?: boolean | Expression<boolean>;
    /** &lt;p&gt;Used to include specific company properties in the results. By default, the results will only include company ID and will not include the values for any properties for your company.&lt;/p&gt;&lt;p&gt;Including this parameter will include the data for the specified property in the results. You can include this parameter multiple times to request multiple properties separated by a comma: &lt;code&gt;,&lt;/code&gt;.&lt;/p&gt;. Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     * @default []
     */
    properties?: string[];
    /** Specify if the current value for a property should be fetched, or the value and all the historical values for that property
     * @default valueAndHistory
     */
    propertyMode?: 'valueAndHistory' | 'valueOnly' | Expression<string>;
  };
};

export type HubspotV1ContactGetRecentlyCreatedUpdatedNode = {
  type: 'n8n-nodes-base.hubspot';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<HubspotV1ContactGetRecentlyCreatedUpdatedParams>;
};
