// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedrive/v1/resource_note/operation_update.ts
/**
 * Pipedrive Node - Version 1
 * Discriminator: resource=note, operation=update
 */


interface Credentials {
  pipedriveApi: CredentialReference;
  pipedriveOAuth2Api: CredentialReference;
}

/** Update an activity */
export type PipedriveV1NoteUpdateParams = {
  resource: 'note';
  operation: 'update';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * ID of the note to update
 * @default 0
 */
    noteId?: number | Expression<number>;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** The content of the note
     */
    content?: string | Expression<string> | PlaceholderValue;
    /** ID of the deal this note will be associated with
     * @default 0
     */
    deal_id?: number | Expression<number>;
    /** ID of the lead this note will be associated with
     * @default 0
     */
    lead_id?: number | Expression<number>;
    /** ID of the organization this deal will be associated with. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    org_id?: string | Expression<string>;
    /** ID of the person this note will be associated with
     * @default 0
     */
    person_id?: number | Expression<number>;
  };
};

export type PipedriveV1NoteUpdateNode = {
  type: 'n8n-nodes-base.pipedrive';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PipedriveV1NoteUpdateParams>;
};
