// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedrive/v1/resource_lead/operation_get.ts
/**
 * Pipedrive Node - Version 1
 * Discriminator: resource=lead, operation=get
 */


interface Credentials {
  pipedriveApi: CredentialReference;
  pipedriveOAuth2Api: CredentialReference;
}

/** Get data of an activity */
export type PipedriveV1LeadGetParams = {
  resource: 'lead';
  operation: 'get';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * ID of the lead to retrieve
 */
    leadId?: string | Expression<string> | PlaceholderValue;
};

export type PipedriveV1LeadGetOutput = {
  add_time?: string;
  archive_time?: null;
  cc_email?: string;
  creator_id?: number;
  db247a86f77b7a989f9c2834851c946b26718442?: string;
  id?: string;
  is_archived?: boolean;
  label_ids?: Array<string>;
  origin?: string;
  owner_id?: number;
  source_name?: string;
  title?: string;
  update_time?: string;
  visible_to?: string;
  was_seen?: boolean;
};

export type PipedriveV1LeadGetNode = {
  type: 'n8n-nodes-base.pipedrive';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PipedriveV1LeadGetParams>;
  output?: Items<PipedriveV1LeadGetOutput>;
};
