// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedrive/v1/resource_lead/operation_create.ts
/**
 * Pipedrive Node - Version 1
 * Discriminator: resource=lead, operation=create
 */


interface Credentials {
  pipedriveApi: CredentialReference;
  pipedriveOAuth2Api: CredentialReference;
}

/** Create an activity */
export type PipedriveV1LeadCreateParams = {
  resource: 'lead';
  operation: 'create';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * Name of the lead to create
 */
    title?: string | Expression<string> | PlaceholderValue;
/**
 * Type of entity to link to this lead
 * @default organization
 */
    associateWith?: 'organization' | 'person' | Expression<string>;
/**
 * ID of the organization to link to this lead
 * @displayOptions.show { associateWith: ["organization"] }
 * @default 0
 */
    organization_id?: number | Expression<number>;
/**
 * ID of the person to link to this lead
 * @displayOptions.show { associateWith: ["person"] }
 * @default 0
 */
    person_id?: number | Expression<number>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Date when the lead’s deal is expected to be closed, in ISO-8601 format
     */
    expected_close_date?: string | Expression<string>;
    /** ID of the labels to attach to the lead to create. Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     * @default []
     */
    label_ids?: string[];
    /** ID of the organization to link to this lead
     * @displayOptions.show { /associateWith: ["person"] }
     * @default 0
     */
    organization_id?: number | Expression<number>;
    /** ID of the user who will own the lead to create. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    owner_id?: string | Expression<string>;
    /** ID of the person to link to this lead
     * @displayOptions.show { /associateWith: ["organization"] }
     * @default 0
     */
    person_id?: number | Expression<number>;
    /** Potential monetary value associated with the lead
     * @default {}
     */
    value?: {
        /** Value Properties
     */
    valueProperties?: {
      /** Amount
       */
      amount?: number | Expression<number>;
      /** Currency
       * @default USD
       */
      currency?: 'AFN' | 'DZD' | 'ARS' | 'AUD' | 'AZN' | 'BSD' | 'THB' | 'BBD' | 'BZD' | 'BMD' | 'BOB' | 'BRL' | 'BND' | 'BGN' | 'CAD' | 'CLP' | 'COP' | 'NIO' | 'CRC' | 'CZK' | 'DKK' | 'DOP' | 'VND' | 'XCD' | 'EGP' | 'EUR' | 'FJD' | 'HUF' | 'HKD' | 'UAH' | 'INR' | 'JMD' | 'KES' | 'PGK' | 'HRK' | 'MMK' | 'LAK' | 'LBP' | 'ALL' | 'HNL' | 'LRD' | 'MYR' | 'MUR' | 'MXN' | 'MAD' | 'NPR' | 'ILS' | 'TWD' | 'NZD' | 'NOK' | 'MRO' | 'TOP' | 'PKR' | 'MOP' | 'PHP' | 'GBP' | 'BWP' | 'QAR' | 'GTQ' | 'ZAR' | 'RON' | 'MVR' | 'IDR' | 'RUB' | 'SAR' | 'SCR' | 'SGD' | 'PEN' | 'SBD' | 'LKR' | 'SEK' | 'CHF' | 'SYP' | 'BDT' | 'WST' | 'KZT' | 'TTD' | 'TRY' | 'AED' | 'USD' | 'VUV' | 'XOF' | 'KRW' | 'YER' | 'JPY' | 'CNY' | 'PLN' | Expression<string>;
    };
  };
  };
};

export type PipedriveV1LeadCreateOutput = {
  add_time?: string;
  cc_email?: string;
  channel?: null;
  channel_id?: null;
  creator_id?: number;
  id?: string;
  is_archived?: boolean;
  label_ids?: Array<string>;
  next_activity_id?: null;
  origin?: string;
  origin_id?: null;
  owner_id?: number;
  source_name?: string;
  title?: string;
  update_time?: string;
  visible_to?: string;
  was_seen?: boolean;
};

export type PipedriveV1LeadCreateNode = {
  type: 'n8n-nodes-base.pipedrive';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PipedriveV1LeadCreateParams>;
  output?: Items<PipedriveV1LeadCreateOutput>;
};
