// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedrive/v1/resource_lead/operation_update.ts
/**
 * Pipedrive Node - Version 1
 * Discriminator: resource=lead, operation=update
 */


interface Credentials {
  pipedriveApi: CredentialReference;
  pipedriveOAuth2Api: CredentialReference;
}

/** Update an activity */
export type PipedriveV1LeadUpdateParams = {
  resource: 'lead';
  operation: 'update';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * ID of the lead to update
 */
    leadId?: string | Expression<string> | PlaceholderValue;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** Name of the lead to update
     */
    title?: string | Expression<string> | PlaceholderValue;
    /** ID of the user who will own the lead to update. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    owner_id?: string | Expression<string>;
    /** ID of the labels to attach to the lead to update. Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     * @default []
     */
    label_ids?: string[];
    /** ID of the person to link to this lead. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    person_id?: string | Expression<string>;
    /** Potential monetary value associated with the lead
     * @default {}
     */
    value?: {
        /** Value Properties
     */
    valueProperties?: {
      /** Amount
       */
      amount?: number | Expression<number>;
      /** Currency
       * @default USD
       */
      currency?: 'AFN' | 'DZD' | 'ARS' | 'AUD' | 'AZN' | 'BSD' | 'THB' | 'BBD' | 'BZD' | 'BMD' | 'BOB' | 'BRL' | 'BND' | 'BGN' | 'CAD' | 'CLP' | 'COP' | 'NIO' | 'CRC' | 'CZK' | 'DKK' | 'DOP' | 'VND' | 'XCD' | 'EGP' | 'EUR' | 'FJD' | 'HUF' | 'HKD' | 'UAH' | 'INR' | 'JMD' | 'KES' | 'PGK' | 'HRK' | 'MMK' | 'LAK' | 'LBP' | 'ALL' | 'HNL' | 'LRD' | 'MYR' | 'MUR' | 'MXN' | 'MAD' | 'NPR' | 'ILS' | 'TWD' | 'NZD' | 'NOK' | 'MRO' | 'TOP' | 'PKR' | 'MOP' | 'PHP' | 'GBP' | 'BWP' | 'QAR' | 'GTQ' | 'ZAR' | 'RON' | 'MVR' | 'IDR' | 'RUB' | 'SAR' | 'SCR' | 'SGD' | 'PEN' | 'SBD' | 'LKR' | 'SEK' | 'CHF' | 'SYP' | 'BDT' | 'WST' | 'KZT' | 'TTD' | 'TRY' | 'AED' | 'USD' | 'VUV' | 'XOF' | 'KRW' | 'YER' | 'JPY' | 'CNY' | 'PLN' | Expression<string>;
    };
  };
    /** Date when the lead’s deal is expected to be closed, in ISO-8601 format
     */
    expected_close_date?: string | Expression<string>;
  };
};

export type PipedriveV1LeadUpdateNode = {
  type: 'n8n-nodes-base.pipedrive';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PipedriveV1LeadUpdateParams>;
};
