// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/sendy/v1/resource_subscriber/operation_add.ts
/**
 * Sendy Node - Version 1
 * Discriminator: resource=subscriber, operation=add
 */


interface Credentials {
  sendyApi: CredentialReference;
}

/** Add a subscriber to a list */
export type SendyV1SubscriberAddParams = {
  resource: 'subscriber';
  operation: 'add';
/**
 * Email address of the subscriber
 */
    email?: string | Expression<string> | PlaceholderValue;
/**
 * The list ID you want to subscribe a user to. This encrypted & hashed ID can be found under View all lists section named ID.
 */
    listId?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** User's 2 letter country code
     */
    country?: string | Expression<string> | PlaceholderValue;
    /** Whether you're signing up EU users in a GDPR compliant manner
     * @default false
     */
    gdpr?: boolean | Expression<boolean>;
    /** Include this 'honeypot' field to prevent spambots from signing up via this API call. When spambots fills in this field, this API call will exit, preventing them from signing up fake addresses to your form. This parameter is only supported in Sendy 3.0 onwards.
     * @default false
     */
    hp?: boolean | Expression<boolean>;
    /** User's IP address
     */
    ipaddress?: string | Expression<string> | PlaceholderValue;
    /** User's name
     */
    name?: string | Expression<string> | PlaceholderValue;
    /** The URL where the user signed up from
     */
    referrer?: string | Expression<string> | PlaceholderValue;
    /** Set to "true" if your list is 'Double opt-in' but you want to bypass that and signup the user to the list as 'Single Opt-in instead' (optional)
     * @default false
     */
    silent?: boolean | Expression<boolean>;
  };
};

export type SendyV1SubscriberAddNode = {
  type: 'n8n-nodes-base.sendy';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SendyV1SubscriberAddParams>;
};
