// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/sendy/v1/resource_subscriber/operation_add.schema.js
/**
 * Sendy Node - Version 1 - Zod Schema
 * Discriminator: resource=subscriber, operation=add
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('subscriber').default('subscriber'),
      operation: z.literal('add'),
      email: stringOrExpression.optional(),
      listId: stringOrExpression.optional(),
      additionalFields: z.object({ country: stringOrExpression.optional(), gdpr: booleanOrExpression.optional(), hp: booleanOrExpression.optional(), ipaddress: stringOrExpression.optional(), name: stringOrExpression.optional(), referrer: stringOrExpression.optional(), silent: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
