// @n8n/n8n-nodes-langchain/dist/node-definitions/nodes/n8n-nodes-langchain/openAi/v12/resource_file/operation_upload.ts
/**
 * OpenAI Node - Version 1.2
 * Discriminator: resource=file, operation=upload
 */


interface Credentials {
  openAiApi: CredentialReference;
}

/** Upload a file that can be used across various endpoints */
export type LcOpenAiV12FileUploadParams = {
  resource: 'file';
  operation: 'upload';
/**
 * Name of the binary property which contains the file. The size of individual files can be a maximum of 512 MB or 2 million tokens for Assistants.
 * @hint The name of the input field containing the binary file data to be processed
 * @default data
 */
    binaryPropertyName?: string | Expression<string> | PlaceholderValue;
/**
 * Options
 * @default {}
 */
    options?: {
    /** The intended purpose of the uploaded file, the 'Fine-tuning' only supports .jsonl files
     * @default assistants
     */
    purpose?: 'assistants' | 'fine-tune' | Expression<string>;
  };
};

export interface LcOpenAiV12FileUploadSubnodeConfig {
  tools?: ToolInstance[];
  memory?: MemoryInstance;
}

export type LcOpenAiV12FileUploadNode = {
  type: '@n8n/n8n-nodes-langchain.openAi';
  version: 1.2;
  credentials?: Credentials;
  config: NodeConfig<LcOpenAiV12FileUploadParams> & { subnodes?: LcOpenAiV12FileUploadSubnodeConfig };
};
