// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftSharePoint/v1/resource_list/operation_get_all.ts
/**
 * Microsoft SharePoint Node - Version 1
 * Discriminator: resource=list, operation=getAll
 */


interface Credentials {
  microsoftSharePointOAuth2Api: CredentialReference;
}

/** Get specific items in a list or list many items */
export type MicrosoftSharePointV1ListGetAllParams = {
  resource: 'list';
  operation: 'getAll';
/**
 * Select the site to retrieve lists from
 * @default {"mode":"list","value":""}
 */
    site?: { __rl: true; mode: 'list' | 'id'; value: string; cachedResultName?: string };
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Simplify
 * @default true
 */
    simplify?: boolean | Expression<boolean>;
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type MicrosoftSharePointV1ListGetAllOutput = {
  createdDateTime?: string;
  description?: string;
  displayName?: string;
  id?: string;
  lastModifiedDateTime?: string;
  name?: string;
  webUrl?: string;
};

export type MicrosoftSharePointV1ListGetAllNode = {
  type: 'n8n-nodes-base.microsoftSharePoint';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MicrosoftSharePointV1ListGetAllParams>;
  output?: Items<MicrosoftSharePointV1ListGetAllOutput>;
};
