// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/linear/v1/resource_issue/operation_update.schema.js
/**
 * Linear Node - Version 1 - Zod Schema
 * Discriminator: resource=issue, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('issue').default('issue'),
      operation: z.literal('update'),
      authentication: z.union([z.literal('apiToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      issueId: stringOrExpression.optional(),
      updateFields: z.object({ assigneeId: stringOrExpression.optional(), description: stringOrExpression.optional(), priorityId: z.union([z.literal(1), z.literal(2), z.literal(3), z.literal(3), z.literal(0), expressionSchema]).optional(), stateId: stringOrExpression.optional(), teamId: stringOrExpression.optional(), title: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
