// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/linear/v1/resource_issue/operation_get_all.ts
/**
 * Linear Node - Version 1
 * Discriminator: resource=issue, operation=getAll
 */


interface Credentials {
  linearApi: CredentialReference;
  linearOAuth2Api: CredentialReference;
}

/** Get many issues */
export type LinearV1IssueGetAllParams = {
  resource: 'issue';
  operation: 'getAll';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
};

export type LinearV1IssueGetAllOutput = {
  archivedAt?: null;
  createdAt?: string;
  creator?: {
    displayName?: string;
    id?: string;
  };
  id?: string;
  identifier?: string;
  priority?: number;
  state?: {
    id?: string;
    name?: string;
  };
  title?: string;
};

export type LinearV1IssueGetAllNode = {
  type: 'n8n-nodes-base.linear';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<LinearV1IssueGetAllParams>;
  output?: Items<LinearV1IssueGetAllOutput>;
};
