// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/todoist/v2/resource_project/operation_update.ts
/**
 * Todoist Node - Version 2
 * Discriminator: resource=project, operation=update
 */


interface Credentials {
  todoistApi: CredentialReference;
  todoistOAuth2Api: CredentialReference;
}

/** Project resource */
export type TodoistV2ProjectUpdateParams = {
  resource: 'project';
  operation: 'update';
  authentication?: 'apiKey' | 'oAuth2' | Expression<string>;
/**
 * The project ID - can be either a string or number
 */
    projectId?: string | Expression<string> | PlaceholderValue;
/**
 * Update Fields
 * @default {}
 */
    projectUpdateFields?: {
    /** Name of the project
     */
    name?: string | Expression<string> | PlaceholderValue;
    /** The color of the project
     */
    color?: 'berry_red' | 'red' | 'orange' | 'yellow' | 'olive_green' | 'lime_green' | 'green' | 'mint_green' | 'teal' | 'sky_blue' | 'light_blue' | 'blue' | 'grape' | 'violet' | 'lavender' | 'magenta' | 'salmon' | 'charcoal' | 'grey' | 'taupe' | Expression<string>;
    /** Whether the project is a favorite
     * @default false
     */
    is_favorite?: boolean | Expression<boolean>;
    /** The default view style of the project
     * @default list
     */
    view_style?: 'list' | 'board' | Expression<string>;
  };
};

export type TodoistV2ProjectUpdateNode = {
  type: 'n8n-nodes-base.todoist';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<TodoistV2ProjectUpdateParams>;
};
