// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/todoist/v2/resource_project/operation_create.ts
/**
 * Todoist Node - Version 2
 * Discriminator: resource=project, operation=create
 */


interface Credentials {
  todoistApi: CredentialReference;
  todoistOAuth2Api: CredentialReference;
}

/** Project resource */
export type TodoistV2ProjectCreateParams = {
  resource: 'project';
  operation: 'create';
  authentication?: 'apiKey' | 'oAuth2' | Expression<string>;
/**
 * Name of the project
 */
    name?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    projectOptions?: {
    /** The color of the project
     */
    color?: 'berry_red' | 'red' | 'orange' | 'yellow' | 'olive_green' | 'lime_green' | 'green' | 'mint_green' | 'teal' | 'sky_blue' | 'light_blue' | 'blue' | 'grape' | 'violet' | 'lavender' | 'magenta' | 'salmon' | 'charcoal' | 'grey' | 'taupe' | Expression<string>;
    /** Whether the project is a favorite
     * @default false
     */
    is_favorite?: boolean | Expression<boolean>;
    /** Parent project ID
     */
    parent_id?: string | Expression<string> | PlaceholderValue;
    /** The default view style of the project
     * @default list
     */
    view_style?: 'list' | 'board' | Expression<string>;
  };
};

export type TodoistV2ProjectCreateNode = {
  type: 'n8n-nodes-base.todoist';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<TodoistV2ProjectCreateParams>;
};
