// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/zohoCrm/v1/resource_deal/operation_update.schema.js
/**
 * Zoho CRM Node - Version 1 - Zod Schema
 * Discriminator: resource=deal, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('deal'),
      operation: z.literal('update'),
      dealId: stringOrExpression.optional(),
      updateFields: z.object({ Amount: numberOrExpression.optional(), Closing_Date: stringOrExpression.optional(), Currency: stringOrExpression.optional(), customFields: z.unknown().optional(), Deal_Name: stringOrExpression.optional(), Description: stringOrExpression.optional(), Lead_Conversion_Time: numberOrExpression.optional(), Next_Step: stringOrExpression.optional(), Overall_Sales_Duration: numberOrExpression.optional(), Probability: numberOrExpression.optional(), Sales_Cycle_Duration: numberOrExpression.optional(), Stage: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
