// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftTeams/v2/resource_chat_message/operation_get_all.ts
/**
 * Microsoft Teams Node - Version 2
 * Discriminator: resource=chatMessage, operation=getAll
 */


interface Credentials {
  microsoftTeamsOAuth2Api: CredentialReference;
}

/** Get many channels */
export type MicrosoftTeamsV2ChatMessageGetAllParams = {
  resource: 'chatMessage';
  operation: 'getAll';
/**
 * Select the chat from the list, by URL, or by ID (find the chat ID after "conversations/" in the URL)
 * @default {"mode":"list","value":""}
 */
    chatId?: { __rl: true; mode: 'list' | 'id'; value: string; cachedResultName?: string };
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
};

export type MicrosoftTeamsV2ChatMessageGetAllOutput = {
  attachments?: Array<{
    contentType?: string;
    id?: string;
    teamsAppId?: null;
    thumbnailUrl?: null;
  }>;
  body?: {
    content?: string;
    contentType?: string;
  };
  channelIdentity?: null;
  chatId?: string;
  createdDateTime?: string;
  etag?: string;
  id?: string;
  importance?: string;
  lastModifiedDateTime?: string;
  locale?: string;
  mentions?: Array<{
    id?: number;
    mentioned?: {
      device?: null;
      tag?: null;
      user?: {
        '@odata.type'?: string;
        displayName?: string;
        id?: string;
        tenantId?: string;
        userIdentityType?: string;
      };
    };
    mentionText?: string;
  }>;
  messageType?: string;
  policyViolation?: null;
  reactions?: Array<{
    createdDateTime?: string;
    displayName?: string;
    reactionContentUrl?: null;
    reactionType?: string;
    user?: {
      application?: null;
      device?: null;
      user?: {
        '@odata.type'?: string;
        displayName?: null;
        id?: string;
        tenantId?: string;
        userIdentityType?: string;
      };
    };
  }>;
  replyToId?: null;
  webUrl?: null;
};

export type MicrosoftTeamsV2ChatMessageGetAllNode = {
  type: 'n8n-nodes-base.microsoftTeams';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<MicrosoftTeamsV2ChatMessageGetAllParams>;
  output?: Items<MicrosoftTeamsV2ChatMessageGetAllOutput>;
};
