// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/jira/v1/resource_issue/operation_transitions.ts
/**
 * Jira Software Node - Version 1
 * Discriminator: resource=issue, operation=transitions
 */


interface Credentials {
  jiraSoftwareCloudApi: CredentialReference;
  jiraSoftwareServerApi: CredentialReference;
  jiraSoftwareServerPatApi: CredentialReference;
}

/** Creates an issue or, where the option to create subtasks is enabled in Jira, a subtask */
export type JiraV1IssueTransitionsParams = {
  resource: 'issue';
  operation: 'transitions';
  jiraVersion?: 'cloud' | 'server' | 'serverPat' | Expression<string>;
/**
 * Issue Key
 */
    issueKey?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Use expand to include additional information about transitions in the response. This parameter accepts transitions.fields, which returns information about the fields in the transition screen for each transition. Fields hidden from the screen are not returned. Use this information to populate the fields and update fields in Transition issue.
     */
    expand?: string | Expression<string> | PlaceholderValue;
    /** The ID of the transition
     */
    transitionId?: string | Expression<string> | PlaceholderValue;
    /** Whether transitions with the condition Hide From User Condition are included in the response
     * @default false
     */
    skipRemoteOnlyCondition?: boolean | Expression<boolean>;
  };
};

export type JiraV1IssueTransitionsOutput = {
  hasScreen?: boolean;
  id?: string;
  isAvailable?: boolean;
  isConditional?: boolean;
  isGlobal?: boolean;
  isInitial?: boolean;
  isLooped?: boolean;
  name?: string;
  to?: {
    description?: string;
    iconUrl?: string;
    id?: string;
    name?: string;
    self?: string;
    statusCategory?: {
      colorName?: string;
      id?: number;
      key?: string;
      name?: string;
      self?: string;
    };
  };
};

export type JiraV1IssueTransitionsNode = {
  type: 'n8n-nodes-base.jira';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<JiraV1IssueTransitionsParams>;
  output?: Items<JiraV1IssueTransitionsOutput>;
};
