// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/jira/v1/resource_issue/operation_get.ts
/**
 * Jira Software Node - Version 1
 * Discriminator: resource=issue, operation=get
 */


interface Credentials {
  jiraSoftwareCloudApi: CredentialReference;
  jiraSoftwareServerApi: CredentialReference;
  jiraSoftwareServerPatApi: CredentialReference;
}

/** Creates an issue or, where the option to create subtasks is enabled in Jira, a subtask */
export type JiraV1IssueGetParams = {
  resource: 'issue';
  operation: 'get';
  jiraVersion?: 'cloud' | 'server' | 'serverPat' | Expression<string>;
/**
 * Issue Key
 */
    issueKey?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to return a simplified version of the response instead of the raw data
 * @default false
 */
    simplifyOutput?: boolean | Expression<boolean>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** &lt;p&gt;Use expand to include additional information about the issues in the response. This parameter accepts a comma-separated list. Expand options include:
				&lt;ul&gt;
					&lt;li&gt;&lt;code&gt;renderedFields&lt;/code&gt; Returns field values rendered in HTML format.&lt;/li&gt;
					&lt;li&gt;&lt;code&gt;names&lt;/code&gt; Returns the display name of each field.&lt;/li&gt;
					&lt;li&gt;&lt;code&gt;schema&lt;/code&gt; Returns the schema describing a field type.&lt;/li&gt;
					&lt;li&gt;&lt;code&gt;transitions&lt;/code&gt; Returns all possible transitions for the issue.&lt;/li&gt;
					&lt;li&gt;&lt;code&gt;editmeta&lt;/code&gt; Returns information about how each field can be edited.&lt;/li&gt;
					&lt;li&gt;&lt;code&gt;changelog&lt;/code&gt; Returns a list of recent updates to an issue, sorted by date, starting from the most recent.&lt;/li&gt;
					&lt;li&gt;&lt;code&gt;versionedRepresentations&lt;/code&gt; Returns a JSON array for each version of a field's value, with the highest number representing the most recent version. Note: When included in the request, the fields parameter is ignored.&lt;/li&gt;
				&lt;/ul&gt;
     */
    expand?: string | Expression<string> | PlaceholderValue;
    /** A list of fields to return for the issue. This parameter accepts a comma-separated list. Use it to retrieve a subset of fields. Allowed values: &lt;code&gt;*all&lt;/code&gt; Returns all fields. &lt;code&gt;*navigable&lt;/code&gt; Returns navigable fields. Any issue field, prefixed with a minus to exclude.
     */
    fields?: string | Expression<string> | PlaceholderValue;
    /** Whether fields in fields are referenced by keys rather than IDs. This parameter is useful where fields have been added by a connect app and a field's key may differ from its ID.
     * @default false
     */
    fieldsByKey?: boolean | Expression<boolean>;
    /** A list of issue properties to return for the issue. This parameter accepts a comma-separated list. Allowed values: &lt;code&gt;*all&lt;/code&gt; Returns all issue properties. Any issue property key, prefixed with a minus to exclude. Examples: &lt;code&gt;*all&lt;/code&gt; Returns all properties. &lt;code&gt;*all&lt;/code&gt;,-prop1 Returns all properties except prop1. &lt;code&gt;prop1,prop2&lt;/code&gt; Returns prop1 and prop2 properties. This parameter may be specified multiple times. For example, properties=prop1,prop2& properties=prop3.
     */
    properties?: string | Expression<string> | PlaceholderValue;
    /** Whether the project in which the issue is created is added to the user's Recently viewed project list, as shown under Projects in Jira. This also populates the JQL issues search lastViewed field.
     * @default false
     */
    updateHistory?: boolean | Expression<boolean>;
  };
};

export type JiraV1IssueGetOutput = {
  expand?: string;
  fields?: {
    aggregateprogress?: {
      progress?: number;
      total?: number;
    };
    assignee?: {
      accountId?: string;
      accountType?: string;
      active?: boolean;
      avatarUrls?: {
        '16x16'?: string;
        '24x24'?: string;
        '32x32'?: string;
        '48x48'?: string;
      };
      displayName?: string;
      emailAddress?: string;
      self?: string;
      timeZone?: string;
    };
    attachment?: Array<{
      author?: {
        accountId?: string;
        accountType?: string;
        active?: boolean;
        avatarUrls?: {
          '16x16'?: string;
          '24x24'?: string;
          '32x32'?: string;
          '48x48'?: string;
        };
        displayName?: string;
        emailAddress?: string;
        self?: string;
        timeZone?: string;
      };
      content?: string;
      created?: string;
      filename?: string;
      id?: string;
      mimeType?: string;
      self?: string;
      size?: number;
      thumbnail?: string;
    }>;
    comment?: {
      comments?: Array<{
        author?: {
          accountId?: string;
          accountType?: string;
          active?: boolean;
          avatarUrls?: {
            '16x16'?: string;
            '24x24'?: string;
            '32x32'?: string;
            '48x48'?: string;
          };
          displayName?: string;
          emailAddress?: string;
          self?: string;
          timeZone?: string;
        };
        body?: string;
        created?: string;
        id?: string;
        jsdPublic?: boolean;
        self?: string;
        updateAuthor?: {
          accountId?: string;
          accountType?: string;
          active?: boolean;
          avatarUrls?: {
            '16x16'?: string;
            '24x24'?: string;
            '32x32'?: string;
            '48x48'?: string;
          };
          displayName?: string;
          emailAddress?: string;
          self?: string;
          timeZone?: string;
        };
        updated?: string;
      }>;
      maxResults?: number;
      self?: string;
      startAt?: number;
      total?: number;
    };
    created?: string;
    creator?: {
      accountId?: string;
      accountType?: string;
      active?: boolean;
      avatarUrls?: {
        '16x16'?: string;
        '24x24'?: string;
        '32x32'?: string;
        '48x48'?: string;
      };
      displayName?: string;
      emailAddress?: string;
      self?: string;
      timeZone?: string;
    };
    customfield_10001?: null;
    customfield_10019?: string;
    environment?: null;
    issuelinks?: Array<{
      id?: string;
      inwardIssue?: {
        fields?: {
          issuetype?: {
            avatarId?: number;
            description?: string;
            entityId?: string;
            hierarchyLevel?: number;
            iconUrl?: string;
            id?: string;
            name?: string;
            self?: string;
            subtask?: boolean;
          };
          priority?: {
            iconUrl?: string;
            id?: string;
            name?: string;
            self?: string;
          };
          status?: {
            description?: string;
            iconUrl?: string;
            id?: string;
            name?: string;
            self?: string;
            statusCategory?: {
              colorName?: string;
              id?: number;
              key?: string;
              name?: string;
              self?: string;
            };
          };
          summary?: string;
        };
        id?: string;
        key?: string;
        self?: string;
      };
      outwardIssue?: {
        fields?: {
          issuetype?: {
            avatarId?: number;
            description?: string;
            hierarchyLevel?: number;
            iconUrl?: string;
            id?: string;
            name?: string;
            self?: string;
            subtask?: boolean;
          };
          priority?: {
            iconUrl?: string;
            id?: string;
            name?: string;
            self?: string;
          };
          status?: {
            description?: string;
            iconUrl?: string;
            id?: string;
            name?: string;
            self?: string;
            statusCategory?: {
              colorName?: string;
              id?: number;
              key?: string;
              name?: string;
              self?: string;
            };
          };
          summary?: string;
        };
        id?: string;
        key?: string;
        self?: string;
      };
      self?: string;
      type?: {
        id?: string;
        inward?: string;
        name?: string;
        outward?: string;
        self?: string;
      };
    }>;
    issuetype?: {
      avatarId?: number;
      description?: string;
      entityId?: string;
      hierarchyLevel?: number;
      iconUrl?: string;
      id?: string;
      name?: string;
      self?: string;
      subtask?: boolean;
    };
    labels?: Array<string>;
    priority?: {
      iconUrl?: string;
      id?: string;
      name?: string;
      self?: string;
    };
    progress?: {
      progress?: number;
      total?: number;
    };
    project?: {
      avatarUrls?: {
        '16x16'?: string;
        '24x24'?: string;
        '32x32'?: string;
        '48x48'?: string;
      };
      id?: string;
      key?: string;
      name?: string;
      projectTypeKey?: string;
      self?: string;
      simplified?: boolean;
    };
    reporter?: {
      accountId?: string;
      accountType?: string;
      active?: boolean;
      avatarUrls?: {
        '16x16'?: string;
        '24x24'?: string;
        '32x32'?: string;
        '48x48'?: string;
      };
      displayName?: string;
      emailAddress?: string;
      self?: string;
      timeZone?: string;
    };
    security?: null;
    status?: {
      description?: string;
      iconUrl?: string;
      id?: string;
      name?: string;
      self?: string;
      statusCategory?: {
        colorName?: string;
        id?: number;
        key?: string;
        name?: string;
        self?: string;
      };
    };
    statusCategory?: {
      colorName?: string;
      id?: number;
      key?: string;
      name?: string;
      self?: string;
    };
    statuscategorychangedate?: string;
    subtasks?: Array<{
      fields?: {
        issuetype?: {
          avatarId?: number;
          description?: string;
          entityId?: string;
          hierarchyLevel?: number;
          iconUrl?: string;
          id?: string;
          name?: string;
          self?: string;
          subtask?: boolean;
        };
        priority?: {
          iconUrl?: string;
          id?: string;
          name?: string;
          self?: string;
        };
        status?: {
          description?: string;
          iconUrl?: string;
          id?: string;
          name?: string;
          self?: string;
          statusCategory?: {
            colorName?: string;
            id?: number;
            key?: string;
            name?: string;
            self?: string;
          };
        };
        summary?: string;
      };
      id?: string;
      key?: string;
      self?: string;
    }>;
    summary?: string;
    timeestimate?: null;
    timeoriginalestimate?: null;
    timespent?: null;
    updated?: string;
    votes?: {
      hasVoted?: boolean;
      self?: string;
      votes?: number;
    };
    watches?: {
      isWatching?: boolean;
      self?: string;
      watchCount?: number;
    };
    worklog?: {
      maxResults?: number;
      startAt?: number;
      total?: number;
    };
    workratio?: number;
  };
  id?: string;
  key?: string;
  self?: string;
};

export type JiraV1IssueGetNode = {
  type: 'n8n-nodes-base.jira';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<JiraV1IssueGetParams>;
  output?: Items<JiraV1IssueGetOutput>;
};
