// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mautic/v1/resource_company/operation_get_all.ts
/**
 * Mautic Node - Version 1
 * Discriminator: resource=company, operation=getAll
 */


interface Credentials {
  mauticApi: CredentialReference;
  mauticOAuth2Api: CredentialReference;
}

/** Create or modify a company */
export type MauticV1CompanyGetAllParams = {
  resource: 'company';
  operation: 'getAll';
  authentication?: 'credentials' | 'oAuth2' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 30
 */
    limit?: number | Expression<number>;
/**
 * Whether to return a simplified version of the response instead of the raw data
 * @default true
 */
    simple?: boolean | Expression<boolean>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Sort direction: asc or desc
     */
    orderByDir?: 'asc' | 'desc' | Expression<string>;
    /** Column to sort by. Can use any column listed in the response. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    orderBy?: string | Expression<string>;
    /** String or search command to filter entities by
     */
    search?: string | Expression<string> | PlaceholderValue;
  };
};

export type MauticV1CompanyGetAllNode = {
  type: 'n8n-nodes-base.mautic';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MauticV1CompanyGetAllParams>;
};
