// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/freshworksCrm/v1/resource_account/operation_create.schema.js
/**
 * Freshworks CRM Node - Version 1 - Zod Schema
 * Discriminator: resource=account, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('account').default('account'),
      operation: z.literal('create').default('create'),
      name: stringOrExpression.optional(),
      additionalFields: z.object({ address: stringOrExpression.optional(), annual_revenue: numberOrExpression.optional(), business_type_id: stringOrExpression.optional(), city: stringOrExpression.optional(), country: stringOrExpression.optional(), facebook: stringOrExpression.optional(), industry_type_id: stringOrExpression.optional(), linkedin: stringOrExpression.optional(), number_of_employees: numberOrExpression.optional(), owner_id: stringOrExpression.optional(), parent_sales_account_id: stringOrExpression.optional(), phone: stringOrExpression.optional(), state: stringOrExpression.optional(), territory_id: stringOrExpression.optional(), twitter: stringOrExpression.optional(), website: stringOrExpression.optional(), zipcode: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
