// n8n/dist/modules/mcp/tools/get-execution.tool.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createGetExecutionTool = void 0;
const n8n_workflow_1 = require("n8n-workflow");
const zod_1 = __importDefault(require("zod"));
const mcp_constants_1 = require("../mcp.constants");
const mcp_errors_1 = require("../mcp.errors");
const workflow_validation_utils_1 = require("./workflow-validation.utils");
const inputSchema = zod_1.default.object({
    workflowId: zod_1.default.string().describe('The ID of the workflow the execution belongs to'),
    executionId: zod_1.default.string().describe('The ID of the execution to retrieve'),
});
const outputSchema = {
    execution: zod_1.default
        .object({
        id: zod_1.default.string(),
        workflowId: zod_1.default.string(),
        mode: zod_1.default.string(),
        status: zod_1.default.string(),
        startedAt: zod_1.default.string().nullable(),
        stoppedAt: zod_1.default.string().nullable(),
        retryOf: zod_1.default.string().nullable().optional(),
        retrySuccessId: zod_1.default.string().nullable().optional(),
        waitTill: zod_1.default.string().nullable().optional(),
    })
        .passthrough()
        .nullable()
        .describe('Execution metadata, or null if an error occurred'),
    data: zod_1.default.unknown().optional().describe('Full execution result data'),
    error: zod_1.default.string().optional().describe('Error message if the request failed'),
};
const createGetExecutionTool = (user, executionRepository, workflowFinderService, telemetry) => ({
    name: 'get_execution',
    config: {
        description: 'Get full execution details and results using the execution ID and workflow ID',
        inputSchema: inputSchema.shape,
        outputSchema,
        annotations: {
            title: 'Get Execution',
            readOnlyHint: true,
            destructiveHint: false,
            idempotentHint: true,
            openWorldHint: false,
        },
    },
    handler: async ({ workflowId, executionId }) => {
        const telemetryPayload = {
            user_id: user.id,
            tool_name: 'get_execution',
            parameters: { workflowId, executionId },
        };
        try {
            await (0, workflow_validation_utils_1.getMcpWorkflow)(workflowId, user, ['workflow:read'], workflowFinderService);
            const execution = await executionRepository.findWithUnflattenedData(executionId, [
                workflowId,
            ]);
            if (!execution) {
                const executionExists = await executionRepository.existsBy({ id: executionId });
                if (!executionExists) {
                    throw new mcp_errors_1.WorkflowAccessError(`Execution with ID '${executionId}' does not exist`, 'execution_does_not_exist');
                }
                throw new mcp_errors_1.WorkflowAccessError(`Execution '${executionId}' does not belong to workflow '${workflowId}'`, 'execution_workflow_mismatch');
            }
            const output = {
                execution: {
                    id: execution.id,
                    workflowId: execution.workflowId,
                    mode: execution.mode,
                    status: execution.status,
                    startedAt: execution.startedAt?.toISOString() ?? null,
                    stoppedAt: execution.stoppedAt?.toISOString() ?? null,
                    retryOf: execution.retryOf ?? null,
                    retrySuccessId: execution.retrySuccessId ?? null,
                    waitTill: execution.waitTill?.toISOString() ?? null,
                },
                data: execution.data,
            };
            telemetryPayload.results = {
                success: true,
                data: {
                    executionId,
                    status: execution.status,
                },
            };
            telemetry.track(mcp_constants_1.USER_CALLED_MCP_TOOL_EVENT, telemetryPayload);
            return {
                content: [{ type: 'text', text: (0, n8n_workflow_1.jsonStringify)(output) }],
                structuredContent: output,
            };
        }
        catch (er) {
            const error = (0, n8n_workflow_1.ensureError)(er);
            const isAccessError = error instanceof mcp_errors_1.WorkflowAccessError;
            const errorInfo = {
                message: error.message || 'Unknown error',
                name: error.constructor.name,
            };
            if ('extra' in error && error.extra) {
                errorInfo.extra = error.extra;
            }
            if (error.cause) {
                errorInfo.cause =
                    error.cause instanceof Error ? error.cause.message : (0, n8n_workflow_1.jsonStringify)(error.cause);
            }
            const output = {
                execution: null,
                error: error.message ?? `${error.constructor.name}: (no message)`,
            };
            telemetryPayload.results = {
                success: false,
                error: errorInfo,
                error_reason: isAccessError ? error.reason : undefined,
            };
            telemetry.track(mcp_constants_1.USER_CALLED_MCP_TOOL_EVENT, telemetryPayload);
            return {
                content: [{ type: 'text', text: (0, n8n_workflow_1.jsonStringify)(output) }],
                structuredContent: output,
            };
        }
    },
});
exports.createGetExecutionTool = createGetExecutionTool;
