// n8n/dist/modules/mcp/tools/schemas.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.workflowDetailsOutputSchema = exports.workflowMetaSchema = exports.workflowSettingsSchema = exports.tagSchema = exports.nodeSchema = void 0;
const zod_1 = __importDefault(require("zod"));
exports.nodeSchema = zod_1.default
    .object({
    name: zod_1.default.string(),
    type: zod_1.default.string(),
})
    .passthrough();
exports.tagSchema = zod_1.default.object({ id: zod_1.default.string(), name: zod_1.default.string() }).passthrough();
exports.workflowSettingsSchema = zod_1.default
    .custom((_value) => true)
    .nullable();
exports.workflowMetaSchema = zod_1.default
    .custom((_value) => true)
    .nullable();
exports.workflowDetailsOutputSchema = zod_1.default.object({
    workflow: zod_1.default
        .object({
        id: zod_1.default.string(),
        name: zod_1.default.string().nullable(),
        active: zod_1.default.boolean(),
        isArchived: zod_1.default.boolean(),
        versionId: zod_1.default.string().describe('The current workflow version ID'),
        activeVersionId: zod_1.default
            .string()
            .nullable()
            .describe('The active workflow version ID, if available'),
        triggerCount: zod_1.default.number(),
        createdAt: zod_1.default.string().nullable(),
        updatedAt: zod_1.default.string().nullable(),
        settings: exports.workflowSettingsSchema,
        connections: zod_1.default.record(zod_1.default.unknown()),
        nodes: zod_1.default.array(exports.nodeSchema),
        activeVersion: zod_1.default
            .object({
            nodes: zod_1.default.array(exports.nodeSchema),
            connections: zod_1.default.record(zod_1.default.unknown()),
        })
            .nullable()
            .describe('Active workflow graph, if available'),
        tags: zod_1.default.array(exports.tagSchema),
        meta: exports.workflowMetaSchema,
        parentFolderId: zod_1.default.string().nullable(),
        description: zod_1.default.string().optional().describe('The description of the workflow'),
        scopes: zod_1.default.array(zod_1.default.string()).describe('User permissions for this workflow'),
        canExecute: zod_1.default.boolean().describe('Whether the user has permission to execute this workflow'),
    })
        .passthrough()
        .describe('Sanitized workflow data safe for MCP consumption'),
    triggerInfo: zod_1.default
        .string()
        .describe('Human-readable instructions describing how to trigger the workflow'),
});
