// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/freshworksCrm/v1/resource_contact/operation_update.schema.js
/**
 * Freshworks CRM Node - Version 1 - Zod Schema
 * Discriminator: resource=contact, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('contact'),
      operation: z.literal('update'),
      contactId: stringOrExpression.optional(),
      updateFields: z.object({ address: stringOrExpression.optional(), campaign_id: stringOrExpression.optional(), city: stringOrExpression.optional(), contact_status_id: stringOrExpression.optional(), country: stringOrExpression.optional(), external_id: stringOrExpression.optional(), facebook: stringOrExpression.optional(), first_name: stringOrExpression.optional(), job_title: stringOrExpression.optional(), keyword: stringOrExpression.optional(), last_name: stringOrExpression.optional(), lead_source_id: stringOrExpression.optional(), lifecycle_stage_id: stringOrExpression.optional(), linkedin: stringOrExpression.optional(), medium: stringOrExpression.optional(), mobile_number: stringOrExpression.optional(), owner_id: stringOrExpression.optional(), sales_accounts: z.array(z.string()).optional(), state: stringOrExpression.optional(), subscription_status: stringOrExpression.optional(), subscription_types: stringOrExpression.optional(), territory_id: stringOrExpression.optional(), time_zone: stringOrExpression.optional(), twitter: stringOrExpression.optional(), work_number: stringOrExpression.optional(), zipcode: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
