// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/stripe/v1/resource_coupon/operation_create.schema.js
/**
 * Stripe Node - Version 1 - Zod Schema
 * Discriminator: resource=coupon, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('coupon'),
      operation: z.literal('create'),
      duration: z.union([z.literal('forever'), z.literal('once'), expressionSchema]).optional(),
      type: z.union([z.literal('fixedAmount'), z.literal('percent'), expressionSchema]).optional(),
      amountOff: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"type":["fixedAmount"]}}, defaults: {"type":"percent"} }),
      currency: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"type":["fixedAmount"]}}, defaults: {"type":"percent"} }),
      percentOff: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"type":["percent"]}}, defaults: {"type":"percent"} }),
    }).optional(),
  });
};
