// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/slack/v21/resource_message/operation_update.schema.js
/**
 * Slack Node - Version 2.1 - Zod Schema
 * Discriminator: resource=message, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('message').default('message'),
      operation: z.literal('update'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      channelId: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('id'), z.literal('url')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      ts: numberOrExpression,
      messageType: z.union([z.literal('text'), z.literal('block'), z.literal('attachment'), expressionSchema]).optional(),
      blocksUi: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"messageType":["block"]}}, defaults: {"messageType":"text"} }),
      text: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"messageType":["block","text"]}}, defaults: {"messageType":"text"} }),
      updateFields: z.object({ link_names: booleanOrExpression.optional(), parse: z.union([z.literal('client'), z.literal('full'), z.literal('none'), expressionSchema]).optional() }).optional(),
      otherOptions: z.object({ includeLinkToWorkflow: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
