// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/theHive/v1/resource_log/operation_create.ts
/**
 * TheHive Node - Version 1
 * Discriminator: resource=log, operation=create
 */


interface Credentials {
  theHiveApi: CredentialReference;
}

/** Create task log */
export type TheHiveV1LogCreateParams = {
  resource: 'log';
  operation: 'create';
/**
 * ID of the task
 */
    taskId?: string | Expression<string> | PlaceholderValue;
/**
 * Content of the Log
 */
    message?: string | Expression<string> | PlaceholderValue;
/**
 * Date of the log submission default=now
 */
    startDate?: string | Expression<string>;
/**
 * Status of the log (Ok or Deleted) default=Ok
 */
    status?: 'Ok' | 'Deleted' | Expression<string>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** File attached to the log
     * @default {}
     */
    attachmentValues?: {
        /** Attachment
     */
    attachmentValues?: {
      /** The name of the input binary field which holds binary data
       * @default data
       */
      binaryProperty?: string | Expression<string> | PlaceholderValue;
    };
  };
  };
};

export type TheHiveV1LogCreateNode = {
  type: 'n8n-nodes-base.theHive';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<TheHiveV1LogCreateParams>;
};
