// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/theHive/v1/resource_log/operation_execute_responder.schema.js
/**
 * TheHive Node - Version 1 - Zod Schema
 * Discriminator: resource=log, operation=executeResponder
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('log'),
      operation: z.literal('executeResponder'),
      id: stringOrExpression.optional(),
      responder: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"hide":{"id":[""]}}, defaults: {"id":""} }),
    }).optional(),
  });
};
