// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/removeDuplicates/v11.schema.js
/**
 * Remove Duplicates Node - Version 1.1 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    compare: z.union([z.literal('allFields'), z.literal('allFieldsExcept'), z.literal('selectedFields'), expressionSchema]).optional(),
    fieldsToExclude: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"compare":["allFieldsExcept"]}}, defaults: {"compare":"allFields"} }),
    fieldsToCompare: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"compare":["selectedFields"]}}, defaults: {"compare":"allFields"} }),
    options: resolveSchema({ parameters, schema: z.object({ disableDotNotation: booleanOrExpression.optional(), removeOtherFields: booleanOrExpression.optional() }), required: false, displayOptions: {"show":{"compare":["allFieldsExcept","selectedFields"]}}, defaults: {"compare":"allFields"} }),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
