// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/sentryIo/v1/resource_issue/operation_get_all.schema.js
/**
 * Sentry.io Node - Version 1 - Zod Schema
 * Discriminator: resource=issue, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('issue'),
      operation: z.literal('getAll'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), z.literal('accessTokenServer'), expressionSchema]).optional(),
      organizationSlug: stringOrExpression.optional(),
      projectSlug: stringOrExpression.optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      additionalFields: z.object({ query: stringOrExpression.optional(), statsPeriod: z.union([z.literal('14d'), z.literal('24h'), expressionSchema]).optional(), shortIdLookUp: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
