// n8n/dist/modules/external-secrets.ee/secrets-providers-connections.service.ee.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecretsProvidersConnectionsService = void 0;
const api_types_1 = require("@n8n/api-types");
const backend_common_1 = require("@n8n/backend-common");
const db_1 = require("@n8n/db");
const di_1 = require("@n8n/di");
const n8n_core_1 = require("n8n-core");
const n8n_workflow_1 = require("n8n-workflow");
const bad_request_error_1 = require("../../errors/response-errors/bad-request.error");
const not_found_error_1 = require("../../errors/response-errors/not-found.error");
const event_service_1 = require("../../events/event.service");
const external_secrets_manager_ee_1 = require("../../modules/external-secrets.ee/external-secrets-manager.ee");
const redaction_service_ee_1 = require("../../modules/external-secrets.ee/redaction.service.ee");
const provider_registry_service_1 = require("./provider-registry.service");
let SecretsProvidersConnectionsService = class SecretsProvidersConnectionsService {
    constructor(logger, repository, projectAccessRepository, providerRegistry, cipher, externalSecretsManager, redactionService, eventService) {
        this.logger = logger;
        this.repository = repository;
        this.projectAccessRepository = projectAccessRepository;
        this.providerRegistry = providerRegistry;
        this.cipher = cipher;
        this.externalSecretsManager = externalSecretsManager;
        this.redactionService = redactionService;
        this.eventService = eventService;
        this.logger = this.logger.scoped('external-secrets');
    }
    async createConnection(proposedConnection, userId) {
        const existing = await this.repository.findOne({
            where: { providerKey: proposedConnection.providerKey },
        });
        if (existing) {
            throw new bad_request_error_1.BadRequestError(`Connection with key "${proposedConnection.providerKey}" already exists`);
        }
        const encryptedSettings = this.encryptConnectionSettings(proposedConnection.settings);
        const connection = this.repository.create({
            ...proposedConnection,
            encryptedSettings,
            isEnabled: true,
        });
        const savedConnection = await this.repository.save(connection);
        if (proposedConnection.projectIds.length > 0) {
            const entries = proposedConnection.projectIds.map((projectId) => this.projectAccessRepository.create({
                secretsProviderConnectionId: savedConnection.id,
                projectId,
            }));
            await this.projectAccessRepository.save(entries);
        }
        const result = (await this.repository.findOne({
            where: { providerKey: proposedConnection.providerKey },
        }));
        await this.externalSecretsManager.syncProviderConnection(proposedConnection.providerKey);
        this.eventService.emit('external-secrets-connection-created', {
            userId,
            providerKey: result.providerKey,
            vaultType: result.type,
            ...this.extractProjectInfo(result),
        });
        return result;
    }
    async updateConnection(providerKey, updates, userId) {
        const connection = await this.repository.findOne({ where: { providerKey } });
        if (!connection) {
            throw new not_found_error_1.NotFoundError(`Connection with key "${providerKey}" not found`);
        }
        if (updates.type !== undefined) {
            connection.type = updates.type;
            if (!updates.settings) {
                throw new bad_request_error_1.BadRequestError('When changing the connection type, new settings must be provided as well');
            }
        }
        if (updates.settings !== undefined) {
            const savedSettings = this.decryptConnectionSettings(connection.encryptedSettings);
            const unredactedSettings = this.redactionService.unredact(updates.settings, savedSettings);
            connection.encryptedSettings = this.encryptConnectionSettings(unredactedSettings);
        }
        if (updates.isEnabled !== undefined) {
            connection.isEnabled = updates.isEnabled;
        }
        await this.repository.save(connection);
        if (updates.projectIds !== undefined) {
            await this.projectAccessRepository.setProjectAccess(connection.id, updates.projectIds);
        }
        await this.externalSecretsManager.syncProviderConnection(providerKey);
        const result = (await this.repository.findOne({
            where: { providerKey },
        }));
        this.eventService.emit('external-secrets-connection-updated', {
            userId,
            providerKey: result.providerKey,
            vaultType: result.type,
            ...this.extractProjectInfo(result),
        });
        return result;
    }
    async deleteConnection(providerKey, userId) {
        const connection = await this.repository.findOne({ where: { providerKey } });
        if (!connection) {
            throw new not_found_error_1.NotFoundError(`Connection with key "${providerKey}" not found`);
        }
        const projectInfo = this.extractProjectInfo(connection);
        await this.projectAccessRepository.deleteByConnectionId(connection.id);
        await this.repository.remove(connection);
        await this.externalSecretsManager.syncProviderConnection(providerKey);
        this.eventService.emit('external-secrets-connection-deleted', {
            userId,
            providerKey: connection.providerKey,
            vaultType: connection.type,
            ...projectInfo,
        });
        return connection;
    }
    async getConnection(providerKey) {
        const connection = await this.repository.findOne({ where: { providerKey } });
        if (!connection) {
            throw new not_found_error_1.NotFoundError(`Connection with key "${providerKey}" not found`);
        }
        return connection;
    }
    async listConnections() {
        return await this.repository.findAll();
    }
    async getGlobalCompletions() {
        const connectedProviderKeys = this.providerRegistry.getConnectedNames();
        return await this.repository.findGlobalConnections({
            providerKeys: connectedProviderKeys,
        });
    }
    async getProjectCompletions(projectId) {
        const connectedProviderKeys = this.providerRegistry.getConnectedNames();
        return await this.repository.findByProjectId(projectId, {
            providerKeys: connectedProviderKeys,
        });
    }
    async listConnectionsForProject(projectId) {
        return await this.repository.findAllAccessibleByProjectWithProjectAccess(projectId);
    }
    toSecretCompletionsResponse(connections) {
        return Object.fromEntries(connections.map((connection) => [
            connection.providerKey,
            this.externalSecretsManager.getSecretNames(connection.providerKey),
        ]));
    }
    toPublicConnectionListItem(connection) {
        const secretNames = this.externalSecretsManager.getSecretNames(connection.providerKey);
        const connectionInstance = this.externalSecretsManager.getProvider(connection.providerKey);
        return {
            id: String(connection.id),
            name: connection.providerKey,
            type: connection.type,
            isEnabled: connection.isEnabled,
            secretsCount: secretNames.length,
            state: connectionInstance?.state ?? 'initializing',
            projects: connection.projectAccess.map((access) => ({
                id: access.project.id,
                name: access.project.name,
            })),
            createdAt: connection.createdAt.toISOString(),
            updatedAt: connection.updatedAt.toISOString(),
        };
    }
    toPublicConnection(connection) {
        const decryptedSettings = this.decryptConnectionSettings(connection.encryptedSettings);
        const properties = this.externalSecretsManager.getProviderProperties(connection.type);
        const redactedSettings = this.redactionService.redact(decryptedSettings, properties);
        const secretNames = this.externalSecretsManager.getSecretNames(connection.providerKey);
        const connectionInstance = this.externalSecretsManager.getProvider(connection.providerKey);
        return {
            id: String(connection.id),
            name: connection.providerKey,
            type: connection.type,
            isEnabled: connection.isEnabled,
            secretsCount: secretNames.length,
            state: connectionInstance?.state ?? 'initializing',
            secrets: secretNames.map((name) => ({ name })),
            projects: connection.projectAccess.map((access) => ({
                id: access.project.id,
                name: access.project.name,
            })),
            settings: redactedSettings,
            createdAt: connection.createdAt.toISOString(),
            updatedAt: connection.updatedAt.toISOString(),
        };
    }
    async testConnection(providerKey, userId) {
        const connection = await this.getConnection(providerKey);
        const decryptedSettings = this.decryptConnectionSettings(connection.encryptedSettings);
        const result = await this.externalSecretsManager.testProviderSettings(connection.type, decryptedSettings);
        const response = api_types_1.testSecretProviderConnectionResponseSchema.parse(result);
        this.eventService.emit('external-secrets-connection-tested', {
            userId,
            providerKey: connection.providerKey,
            vaultType: connection.type,
            ...this.extractProjectInfo(connection),
            isValid: response.success,
            ...(response.error && { errorMessage: response.error }),
        });
        return response;
    }
    async reloadConnectionSecrets(providerKey, userId) {
        try {
            const connection = await this.getConnection(providerKey);
            await this.externalSecretsManager.updateProvider(providerKey);
            this.eventService.emit('external-secrets-connection-reloaded', {
                userId,
                providerKey: connection.providerKey,
                vaultType: connection.type,
                ...this.extractProjectInfo(connection),
            });
            return api_types_1.reloadSecretProviderConnectionResponseSchema.parse({ success: true });
        }
        catch (error) {
            if (error instanceof not_found_error_1.NotFoundError) {
                throw error;
            }
            this.logger.warn(`Failed to reload provider ${providerKey}`, { providerKey });
            return api_types_1.reloadSecretProviderConnectionResponseSchema.parse({ success: false });
        }
    }
    extractProjectInfo(connection) {
        return {
            projects: connection.projectAccess.map((access) => ({
                id: access.project.id,
                name: access.project.name,
            })),
        };
    }
    async reloadProjectConnectionSecrets(projectId, userId) {
        const projectConnections = await this.repository.findByProjectId(projectId);
        const providers = {};
        await Promise.allSettled(projectConnections.map(async (c) => {
            try {
                await this.externalSecretsManager.updateProvider(c.providerKey);
                providers[c.providerKey] = { success: true };
                this.eventService.emit('external-secrets-connection-reloaded', {
                    userId,
                    providerKey: c.providerKey,
                    vaultType: c.type,
                    ...this.extractProjectInfo(c),
                });
            }
            catch (error) {
                providers[c.providerKey] = { success: false };
                this.logger.warn(`Failed to reload provider ${c.providerKey}`, {
                    projectId,
                    providerKey: c.providerKey,
                });
            }
        }));
        const allSucceeded = Object.values(providers).every((p) => p.success);
        return api_types_1.reloadSecretProviderConnectionResponseSchema.parse({
            success: allSucceeded,
            providers,
        });
    }
    encryptConnectionSettings(settings) {
        return this.cipher.encrypt(settings);
    }
    async getConnectionForProject(providerKey, projectId) {
        const connection = await this.repository.findByProviderKeyAndProjectId(providerKey, projectId);
        if (!connection) {
            throw new not_found_error_1.NotFoundError(`Connection with key "${providerKey}" not found`);
        }
        return connection;
    }
    async getConnectionAccessibleFromProject(providerKey, projectId) {
        const connection = await this.repository.findAccessibleByProviderKeyAndProjectId(providerKey, projectId);
        if (!connection) {
            throw new not_found_error_1.NotFoundError(`Connection with key "${providerKey}" not found`);
        }
        return connection;
    }
    async deleteConnectionForProject(providerKey, projectId) {
        const connection = await this.repository.removeByProviderKeyAndProjectId(providerKey, projectId);
        if (!connection) {
            throw new not_found_error_1.NotFoundError(`Connection with key "${providerKey}" not found`);
        }
        await this.projectAccessRepository.deleteByConnectionId(connection.id);
        await this.externalSecretsManager.syncProviderConnection(providerKey);
        return connection;
    }
    decryptConnectionSettings(encryptedSettings) {
        return (0, n8n_workflow_1.jsonParse)(this.cipher.decrypt(encryptedSettings));
    }
};
exports.SecretsProvidersConnectionsService = SecretsProvidersConnectionsService;
exports.SecretsProvidersConnectionsService = SecretsProvidersConnectionsService = __decorate([
    (0, di_1.Service)(),
    __metadata("design:paramtypes", [backend_common_1.Logger,
        db_1.SecretsProviderConnectionRepository,
        db_1.ProjectSecretsProviderAccessRepository,
        provider_registry_service_1.ExternalSecretsProviderRegistry,
        n8n_core_1.Cipher,
        external_secrets_manager_ee_1.ExternalSecretsManager,
        redaction_service_ee_1.RedactionService,
        event_service_1.EventService])
], SecretsProvidersConnectionsService);
