// @n8n/ai-workflow-builder/dist/code-builder/code-builder-agent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CodeBuilderAgent = void 0;
const manager_1 = require("@langchain/core/callbacks/manager");
const workflow_sdk_1 = require("@n8n/workflow-sdk");
const constants_1 = require("./constants");
const agent_iteration_handler_1 = require("./handlers/agent-iteration-handler");
const auto_finalize_handler_1 = require("./handlers/auto-finalize-handler");
const chat_setup_handler_1 = require("./handlers/chat-setup-handler");
const final_response_handler_1 = require("./handlers/final-response-handler");
const parse_validate_handler_1 = require("./handlers/parse-validate-handler");
const tool_dispatch_handler_1 = require("./handlers/tool-dispatch-handler");
const validate_tool_handler_1 = require("./handlers/validate-tool-handler");
const warning_tracker_1 = require("./state/warning-tracker");
const code_builder_get_tool_1 = require("./tools/code-builder-get.tool");
const code_builder_search_tool_1 = require("./tools/code-builder-search.tool");
const get_suggested_nodes_tool_1 = require("./tools/get-suggested-nodes.tool");
const error_sanitizer_1 = require("../utils/error-sanitizer");
const code_builder_session_1 = require("./utils/code-builder-session");
const content_extractors_1 = require("./utils/content-extractors");
const node_diff_1 = require("./utils/node-diff");
const node_type_parser_1 = require("./utils/node-type-parser");
class CodeBuilderAgent {
    nodeTypeParser;
    logger;
    tools;
    toolsMap;
    parseValidateHandler;
    autoFinalizeHandler;
    validateToolHandler;
    iterationHandler;
    finalResponseHandler;
    chatSetupHandler;
    toolDispatchHandler;
    onTelemetryEvent;
    originalOnTokenUsage;
    accumulatedTokens = {
        inputTokens: 0,
        outputTokens: 0,
        thinkingTokens: 0,
    };
    constructor(config) {
        this.nodeTypeParser = new node_type_parser_1.NodeTypeParser(config.nodeTypes);
        this.logger = config.logger;
        this.onTelemetryEvent = config.onTelemetryEvent;
        this.originalOnTokenUsage = config.onTokenUsage;
        if (config.nodeDefinitionDirs) {
            (0, workflow_sdk_1.setSchemaBaseDirs)(config.nodeDefinitionDirs);
        }
        this.parseValidateHandler = new parse_validate_handler_1.ParseValidateHandler({
            logger: config.logger,
            generatePinData: config.generatePinData,
        });
        this.autoFinalizeHandler = new auto_finalize_handler_1.AutoFinalizeHandler({
            parseAndValidate: async (code, currentWorkflow) => await this.parseValidateHandler.parseAndValidate(code, currentWorkflow),
            getErrorContext: (code, errorMessage) => this.parseValidateHandler.getErrorContext(code, errorMessage),
        });
        this.validateToolHandler = new validate_tool_handler_1.ValidateToolHandler({
            parseAndValidate: async (code, currentWorkflow) => await this.parseValidateHandler.parseAndValidate(code, currentWorkflow),
            getErrorContext: (code, errorMessage) => this.parseValidateHandler.getErrorContext(code, errorMessage),
        });
        this.iterationHandler = new agent_iteration_handler_1.AgentIterationHandler({
            onTokenUsage: (usage) => {
                this.accumulatedTokens.inputTokens += usage.inputTokens;
                this.accumulatedTokens.outputTokens += usage.outputTokens;
                this.accumulatedTokens.thinkingTokens += usage.thinkingTokens;
                this.originalOnTokenUsage?.(usage);
            },
            callbacks: config.callbacks,
            runMetadata: config.runMetadata,
        });
        this.finalResponseHandler = new final_response_handler_1.FinalResponseHandler({
            parseAndValidate: async (code, currentWorkflow) => await this.parseValidateHandler.parseAndValidate(code, currentWorkflow),
        });
        const searchTool = (0, code_builder_search_tool_1.createCodeBuilderSearchTool)(this.nodeTypeParser);
        const getTool = (0, code_builder_get_tool_1.createCodeBuilderGetTool)({ nodeDefinitionDirs: config.nodeDefinitionDirs });
        const suggestedNodesTool = (0, get_suggested_nodes_tool_1.createGetSuggestedNodesTool)(this.nodeTypeParser);
        this.tools = [searchTool, getTool, suggestedNodesTool];
        this.toolsMap = new Map(this.tools.map((t) => [t.name, t]));
        this.chatSetupHandler = new chat_setup_handler_1.ChatSetupHandler({
            llm: config.llm,
            tools: this.tools,
            enableTextEditorConfig: config.enableTextEditor,
            parseAndValidate: async (code, currentWorkflow) => await this.parseValidateHandler.parseAndValidate(code, currentWorkflow),
            getErrorContext: (code, errorMessage) => this.parseValidateHandler.getErrorContext(code, errorMessage),
            nodeTypeParser: this.nodeTypeParser,
            logger: this.logger,
        });
        this.toolDispatchHandler = new tool_dispatch_handler_1.ToolDispatchHandler({
            toolsMap: this.toolsMap,
            toolDisplayTitles: new Map([
                [constants_1.CODE_BUILDER_SEARCH_NODES_TOOL.toolName, constants_1.CODE_BUILDER_SEARCH_NODES_TOOL.displayTitle],
                [constants_1.CODE_BUILDER_GET_NODE_TYPES_TOOL.toolName, constants_1.CODE_BUILDER_GET_NODE_TYPES_TOOL.displayTitle],
                [
                    constants_1.CODE_BUILDER_GET_SUGGESTED_NODES_TOOL.toolName,
                    constants_1.CODE_BUILDER_GET_SUGGESTED_NODES_TOOL.displayTitle,
                ],
            ]),
            validateToolHandler: this.validateToolHandler,
        });
    }
    resetAccumulatedTokens() {
        this.accumulatedTokens = { inputTokens: 0, outputTokens: 0, thinkingTokens: 0 };
    }
    emitTelemetryEvent(params) {
        if (!this.onTelemetryEvent) {
            return;
        }
        const { userId, workflowId, userMessageId, sessionId, status, errorMessage, iterationCount, durationMs, beforeWorkflow, afterWorkflow, } = params;
        const nodeChanges = (0, node_diff_1.calculateNodeChanges)(beforeWorkflow, afterWorkflow);
        const properties = {
            user_id: userId,
            workflow_id: workflowId,
            user_message_id: userMessageId,
            session_id: sessionId,
            status,
            duration_ms: durationMs,
            iteration_count: iterationCount,
            input_tokens: this.accumulatedTokens.inputTokens,
            output_tokens: this.accumulatedTokens.outputTokens,
            thinking_tokens: this.accumulatedTokens.thinkingTokens,
            nodes_added: nodeChanges.nodes_added,
            nodes_removed: nodeChanges.nodes_removed,
            nodes_modified: nodeChanges.nodes_modified,
        };
        if (errorMessage) {
            properties.error_message = errorMessage;
        }
        this.onTelemetryEvent('Code builder agent ran', properties);
    }
    async *chat(payload, userId, abortSignal, historyContext) {
        const startTime = Date.now();
        this.resetAccumulatedTokens();
        const beforeWorkflow = payload.workflowContext?.currentWorkflow;
        const workflowId = beforeWorkflow?.id;
        let iteration = 0;
        try {
            this.logger?.debug('Code builder agent starting', {
                userId,
                messageLength: payload.message.length,
            });
            const setupResult = await this.chatSetupHandler.execute({
                payload,
                historyContext,
            });
            const { llmWithTools, messages, textEditorEnabled, currentWorkflow, textEditorHandler, textEditorToolHandler, } = setupResult;
            const loopResult = yield* this.runAgenticLoop({
                llmWithTools,
                messages,
                textEditorEnabled,
                currentWorkflow,
                textEditorHandler,
                textEditorToolHandler,
                abortSignal,
                payload,
                previousMessages: historyContext?.conversationEntries.map(code_builder_session_1.entryToString) ?? [],
            });
            const { workflow } = loopResult;
            iteration = loopResult.iteration;
            if (workflow) {
                this.logger?.info('Code builder agent generated workflow', {
                    userId,
                    nodeCount: workflow.nodes.length,
                    iterations: iteration,
                });
                yield {
                    messages: [
                        {
                            role: 'assistant',
                            type: 'workflow-updated',
                            codeSnippet: JSON.stringify(workflow, null, 2),
                            iterationCount: iteration,
                        },
                    ],
                };
            }
            else {
                this.logger?.info('Code builder agent generated EMPTY workflow', {
                    userId,
                    nodeCount: 0,
                    iterations: iteration,
                });
            }
            yield {
                messages: [
                    {
                        type: 'session-messages',
                        messages,
                    },
                ],
            };
            const totalDuration = Date.now() - startTime;
            this.emitTelemetryEvent({
                userId,
                workflowId,
                userMessageId: payload.id,
                status: 'success',
                iterationCount: iteration,
                durationMs: totalDuration,
                beforeWorkflow,
                afterWorkflow: workflow,
            });
        }
        catch (error) {
            const totalDuration = Date.now() - startTime;
            const rawErrorMessage = error instanceof Error ? error.message : String(error);
            const errorStack = error instanceof Error ? error.stack : undefined;
            const userFacingMessage = (0, error_sanitizer_1.sanitizeLlmErrorMessage)(error);
            this.logger?.error('Code builder agent failed', {
                userId,
                error: rawErrorMessage,
                stack: errorStack,
            });
            yield {
                messages: [
                    {
                        role: 'assistant',
                        type: 'message',
                        text: `I encountered an error while generating the workflow. ${userFacingMessage}`,
                    },
                ],
            };
            const isAborted = error instanceof Error && (error.name === 'AbortError' || error.message === 'Aborted');
            const status = isAborted ? 'aborted' : 'error';
            this.emitTelemetryEvent({
                userId,
                workflowId,
                userMessageId: payload.id,
                status,
                errorMessage: isAborted ? undefined : rawErrorMessage,
                iterationCount: iteration,
                durationMs: totalDuration,
                beforeWorkflow,
                afterWorkflow: null,
            });
        }
    }
    async *runAgenticLoop(params) {
        const { llmWithTools, messages, textEditorEnabled, currentWorkflow, textEditorHandler, textEditorToolHandler, abortSignal, payload, previousMessages, } = params;
        const state = {
            iteration: 0,
            consecutiveParseErrors: 0,
            workflow: null,
            parseDuration: 0,
            sourceCode: null,
            textEditorValidateAttempts: 0,
            warningTracker: new warning_tracker_1.WarningTracker(),
            outputTrace: [],
            hasUnvalidatedEdits: false,
        };
        if (currentWorkflow) {
            try {
                const preExisting = this.parseValidateHandler.validateExistingWorkflow(currentWorkflow);
                if (preExisting.length > 0) {
                    state.warningTracker.markAsPreExisting(preExisting);
                }
            }
            catch {
            }
        }
        const callbackManager = manager_1.CallbackManager.configure(this.iterationHandler.getCallbacks(), undefined, undefined, undefined, this.iterationHandler.getRunMetadata());
        let parentRunManager;
        if (callbackManager) {
            const isFirstGeneration = !payload.workflowContext?.currentWorkflow?.nodes?.length;
            const isFirstMessage = previousMessages.length === 0;
            parentRunManager = await callbackManager.handleChainStart({ lc: 1, type: 'not_implemented', id: ['CodeBuilderAgent'] }, {
                payload,
                previousMessages,
            }, undefined, undefined, undefined, {
                ...this.iterationHandler.getRunMetadata(),
                first_generation: isFirstGeneration,
                first_message: isFirstMessage,
            }, 'CodeBuilderAgentLoop');
        }
        try {
            while (state.iteration < constants_1.MAX_AGENT_ITERATIONS) {
                this.checkConsecutiveParseErrors(state.consecutiveParseErrors);
                state.iteration++;
                const childCallbacks = parentRunManager?.getChild(`iteration_${state.iteration}`);
                const llmResult = yield* this.iterationHandler.invokeLlm({
                    llmWithTools,
                    messages,
                    abortSignal,
                    iteration: state.iteration,
                    callbacks: childCallbacks,
                });
                if (llmResult.textContent) {
                    state.outputTrace.push({ type: 'text', text: llmResult.textContent });
                }
                if (llmResult.hasToolCalls && llmResult.response.tool_calls) {
                    for (const toolCall of llmResult.response.tool_calls) {
                        state.outputTrace.push({ type: 'tool-call', toolName: toolCall.name });
                    }
                }
                const iterationResult = yield* this.processIteration({
                    llmResult,
                    messages,
                    currentWorkflow,
                    textEditorEnabled,
                    textEditorHandler,
                    textEditorToolHandler,
                    state,
                });
                if (iterationResult.shouldBreak) {
                    break;
                }
            }
            await parentRunManager?.handleChainEnd({
                iterations: state.iteration,
                hasWorkflow: !!state.workflow,
                outputTrace: state.outputTrace,
                output: state.workflow
                    ? { code: state.sourceCode, workflow: JSON.stringify(state.workflow) }
                    : null,
            });
        }
        catch (error) {
            await parentRunManager?.handleChainError(error);
            throw error;
        }
        return {
            workflow: state.workflow,
            parseDuration: state.parseDuration,
            sourceCode: state.sourceCode,
            iteration: state.iteration,
        };
    }
    checkConsecutiveParseErrors(count) {
        if (count >= 3) {
            throw new Error('Failed to parse workflow code after 3 consecutive attempts.');
        }
    }
    async *processIteration(params) {
        const { llmResult, messages, currentWorkflow, textEditorEnabled, textEditorHandler, textEditorToolHandler, state, } = params;
        const response = llmResult.response;
        if (llmResult.hasToolCalls && response.tool_calls) {
            return yield* this.handleToolCalls({
                toolCalls: response.tool_calls,
                messages,
                currentWorkflow,
                textEditorEnabled,
                textEditorHandler,
                textEditorToolHandler,
                state,
            });
        }
        if (textEditorEnabled && textEditorHandler) {
            return yield* this.handleTextEditorAutoFinalize({
                textEditorHandler,
                currentWorkflow,
                messages,
                state,
            });
        }
        return await this.handleFinalResponse({
            response: llmResult.response,
            currentWorkflow,
            messages,
            state,
        });
    }
    async *handleToolCalls(params) {
        const { toolCalls, messages, currentWorkflow, textEditorEnabled, textEditorHandler, textEditorToolHandler, state, } = params;
        state.consecutiveParseErrors = 0;
        const dispatchResult = yield* this.toolDispatchHandler.dispatch({
            toolCalls,
            messages,
            currentWorkflow,
            iteration: state.iteration,
            textEditorHandler,
            textEditorToolHandler,
            warningTracker: state.warningTracker,
        });
        if (dispatchResult.hasUnvalidatedEdits !== undefined) {
            state.hasUnvalidatedEdits = dispatchResult.hasUnvalidatedEdits;
        }
        if (dispatchResult.workflow) {
            state.workflow = dispatchResult.workflow;
        }
        if (dispatchResult.parseDuration !== undefined) {
            state.parseDuration = dispatchResult.parseDuration;
        }
        if (dispatchResult.workflowReady) {
            state.sourceCode = dispatchResult.sourceCode ?? null;
            state.textEditorValidateAttempts = 0;
            return { shouldBreak: true };
        }
        const hadValidateCall = toolCalls.some((tc) => tc.name === 'validate_workflow');
        if (hadValidateCall && !dispatchResult.workflowReady) {
            state.textEditorValidateAttempts++;
        }
        if (textEditorEnabled && state.textEditorValidateAttempts >= constants_1.MAX_VALIDATE_ATTEMPTS) {
            throw new Error(`Failed to generate valid workflow after ${constants_1.MAX_VALIDATE_ATTEMPTS} validate attempts.`);
        }
        return { shouldBreak: false };
    }
    async *handleTextEditorAutoFinalize(params) {
        const { textEditorHandler, currentWorkflow, messages, state } = params;
        const code = textEditorHandler.getWorkflowCode();
        if (!code) {
            return { shouldBreak: true };
        }
        if (!state.hasUnvalidatedEdits && code) {
            if (state.workflow) {
                state.sourceCode = code;
                return { shouldBreak: true };
            }
            state.textEditorValidateAttempts++;
            if (state.textEditorValidateAttempts >= constants_1.MAX_VALIDATE_ATTEMPTS) {
                throw new Error(`Failed to generate valid workflow after ${constants_1.MAX_VALIDATE_ATTEMPTS} validate attempts.`);
            }
            (0, content_extractors_1.pushValidationFeedback)(messages, 'Please use the text editor to fix the validation errors.');
            return { shouldBreak: false };
        }
        state.textEditorValidateAttempts++;
        state.hasUnvalidatedEdits = false;
        if (state.textEditorValidateAttempts >= constants_1.MAX_VALIDATE_ATTEMPTS) {
            throw new Error(`Failed to generate valid workflow after ${constants_1.MAX_VALIDATE_ATTEMPTS} validate attempts.`);
        }
        const autoFinalizeResult = yield* this.autoFinalizeHandler.execute({
            code,
            currentWorkflow,
            messages,
            warningTracker: state.warningTracker,
        });
        if (autoFinalizeResult.success && autoFinalizeResult.workflow) {
            state.workflow = autoFinalizeResult.workflow;
            state.parseDuration = autoFinalizeResult.parseDuration ?? 0;
            state.sourceCode = textEditorHandler.getWorkflowCode() ?? null;
            state.textEditorValidateAttempts = 0;
            return { shouldBreak: true };
        }
        if (autoFinalizeResult.parseDuration) {
            state.parseDuration = autoFinalizeResult.parseDuration;
        }
        return { shouldBreak: false };
    }
    async handleFinalResponse(params) {
        const { response, currentWorkflow, messages, state } = params;
        const finalResult = await this.finalResponseHandler.process({
            response,
            currentWorkflow,
            messages,
            warningTracker: state.warningTracker,
        });
        if (finalResult.parseDuration !== undefined) {
            state.parseDuration = finalResult.parseDuration;
        }
        if (finalResult.isParseError) {
            state.consecutiveParseErrors++;
        }
        if (finalResult.success && finalResult.workflow) {
            state.workflow = finalResult.workflow;
            if (finalResult.sourceCode) {
                state.sourceCode = finalResult.sourceCode;
            }
            return { shouldBreak: true };
        }
        return { shouldBreak: false };
    }
}
exports.CodeBuilderAgent = CodeBuilderAgent;
