// @n8n/ai-workflow-builder/dist/code-builder/code-workflow-builder.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CodeWorkflowBuilder = void 0;
exports.createCodeWorkflowBuilder = createCodeWorkflowBuilder;
const code_builder_agent_1 = require("./code-builder-agent");
const session_chat_handler_1 = require("./handlers/session-chat-handler");
class CodeWorkflowBuilder {
    codeBuilderAgent;
    logger;
    sessionChatHandler;
    onGenerationSuccess;
    constructor(config) {
        this.codeBuilderAgent = new code_builder_agent_1.CodeBuilderAgent({
            llm: config.llm,
            nodeTypes: config.nodeTypes,
            logger: config.logger,
            nodeDefinitionDirs: config.nodeDefinitionDirs,
            enableTextEditor: true,
            onTokenUsage: config.onTokenUsage,
            callbacks: config.callbacks,
            runMetadata: config.runMetadata,
            onTelemetryEvent: config.onTelemetryEvent,
            generatePinData: config.generatePinData,
        });
        this.logger = config.logger;
        this.onGenerationSuccess = config.onGenerationSuccess;
        if (config.checkpointer) {
            this.sessionChatHandler = new session_chat_handler_1.SessionChatHandler({
                checkpointer: config.checkpointer,
                llm: config.llm,
                logger: config.logger,
                onGenerationSuccess: config.onGenerationSuccess,
            });
        }
    }
    async *chat(payload, userId, abortSignal) {
        const workflowId = payload.workflowContext?.currentWorkflow?.id;
        this.logger?.debug('CodeWorkflowBuilder starting', {
            userId,
            workflowId,
            messageLength: payload.message.length,
            hasSessionHandler: !!this.sessionChatHandler,
        });
        if (this.sessionChatHandler) {
            yield* this.sessionChatHandler.execute({
                payload,
                userId,
                abortSignal,
                agentChat: (p, u, s, h) => this.codeBuilderAgent.chat(p, u, s, h),
            });
        }
        else {
            let generationSucceeded = false;
            for await (const chunk of this.codeBuilderAgent.chat(payload, userId, abortSignal)) {
                if (chunk.messages?.some((msg) => msg.type === 'workflow-updated')) {
                    generationSucceeded = true;
                }
                yield chunk;
            }
            if (generationSucceeded && this.onGenerationSuccess) {
                try {
                    await this.onGenerationSuccess();
                }
                catch (error) {
                    this.logger?.warn('Failed to execute onGenerationSuccess callback', { error });
                }
            }
        }
    }
}
exports.CodeWorkflowBuilder = CodeWorkflowBuilder;
function createCodeWorkflowBuilder(llm, nodeTypes, options) {
    return new CodeWorkflowBuilder({
        llm,
        nodeTypes,
        logger: options?.logger,
        nodeDefinitionDirs: options?.nodeDefinitionDirs,
    });
}
