// @n8n/ai-workflow-builder/dist/code-builder/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MCP_UPDATE_WORKFLOW_TOOL = exports.MCP_ARCHIVE_WORKFLOW_TOOL = exports.MCP_CREATE_WORKFLOW_FROM_CODE_TOOL = exports.MCP_GET_SDK_REFERENCE_TOOL = exports.CODE_BUILDER_GET_SUGGESTED_NODES_TOOL = exports.CODE_BUILDER_GET_NODE_TYPES_TOOL = exports.CODE_BUILDER_SEARCH_NODES_TOOL = exports.CODE_BUILDER_VALIDATE_TOOL = exports.CODE_BUILDER_BATCH_STR_REPLACE_TOOL = exports.CODE_BUILDER_TEXT_EDITOR_TOOL = exports.VALIDATE_TOOL = exports.BATCH_STR_REPLACE_TOOL = exports.TEXT_EDITOR_TOOL = exports.FIX_VALIDATION_ERRORS_INSTRUCTION = exports.MAX_VALIDATE_ATTEMPTS = exports.MAX_AGENT_ITERATIONS = void 0;
exports.MAX_AGENT_ITERATIONS = 50;
exports.MAX_VALIDATE_ATTEMPTS = 10;
exports.FIX_VALIDATION_ERRORS_INSTRUCTION = `
Analyze the errors, then use the editor tool to apply fixes:
1. Which errors are relevant to the last user request? If NONE, stop — do not fix unrelated warnings.
2. Use search_nodes and get_node_types to look up the correct node schema (if not already fetched)
3. Use batch_str_replace to fix ALL identified issues atomically in one call (preferred), or use individual str_replace/insert calls
4. Call validate_workflow ONCE after all fixes are applied
Do NOT output explanations — just fix the code. Do not add or edit comments.`;
exports.TEXT_EDITOR_TOOL = {
    type: 'text_editor_20250728',
    name: 'str_replace_based_edit_tool',
};
exports.BATCH_STR_REPLACE_TOOL = {
    type: 'function',
    function: {
        name: 'batch_str_replace',
        description: 'Apply multiple str_replace operations to /workflow.js atomically. All replacements are applied in order. If any replacement fails, all changes are rolled back. Example: {"replacements": [{"old_str": "foo", "new_str": "bar"}, {"old_str": "baz", "new_str": "qux"}]}',
        parameters: {
            type: 'object',
            properties: {
                replacements: {
                    type: 'array',
                    items: {
                        type: 'object',
                        properties: {
                            old_str: { type: 'string', description: 'The exact string to find' },
                            new_str: { type: 'string', description: 'The replacement string' },
                        },
                        required: ['old_str', 'new_str'],
                    },
                    description: 'Ordered list of replacements to apply',
                },
            },
            required: ['replacements'],
        },
    },
};
exports.VALIDATE_TOOL = {
    type: 'function',
    function: {
        name: 'validate_workflow',
        description: 'Validate the current workflow code for errors. Returns validation results - either success or a list of errors to fix.',
        parameters: {
            type: 'object',
            properties: {
                path: {
                    type: 'string',
                    description: 'Path to the workflow file (must be /workflow.js)',
                },
            },
            required: ['path'],
        },
    },
};
exports.CODE_BUILDER_TEXT_EDITOR_TOOL = {
    toolName: 'str_replace_based_edit_tool',
    displayTitle: 'Editing workflow',
};
exports.CODE_BUILDER_BATCH_STR_REPLACE_TOOL = {
    toolName: 'batch_str_replace',
    displayTitle: 'Editing workflow',
};
exports.CODE_BUILDER_VALIDATE_TOOL = {
    toolName: 'validate_workflow',
    displayTitle: 'Validating workflow',
};
exports.CODE_BUILDER_SEARCH_NODES_TOOL = {
    toolName: 'search_nodes',
    displayTitle: 'Searching nodes',
};
exports.CODE_BUILDER_GET_NODE_TYPES_TOOL = {
    toolName: 'get_node_types',
    displayTitle: 'Getting node definitions',
};
exports.CODE_BUILDER_GET_SUGGESTED_NODES_TOOL = {
    toolName: 'get_suggested_nodes',
    displayTitle: 'Getting suggested nodes',
};
exports.MCP_GET_SDK_REFERENCE_TOOL = {
    toolName: 'get_sdk_reference',
    displayTitle: 'Getting SDK reference',
};
exports.MCP_CREATE_WORKFLOW_FROM_CODE_TOOL = {
    toolName: 'create_workflow_from_code',
    displayTitle: 'Creating workflow from code',
};
exports.MCP_ARCHIVE_WORKFLOW_TOOL = {
    toolName: 'archive_workflow',
    displayTitle: 'Archiving workflow',
};
exports.MCP_UPDATE_WORKFLOW_TOOL = {
    toolName: 'update_workflow',
    displayTitle: 'Updating workflow',
};
