// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/matrix/v1/resource_media/operation_upload.schema.js
/**
 * Matrix Node - Version 1 - Zod Schema
 * Discriminator: resource=media, operation=upload
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('media'),
      operation: z.literal('upload'),
      roomId: stringOrExpression.optional(),
      binaryPropertyName: stringOrExpression.optional(),
      mediaType: z.union([z.literal('file'), z.literal('image'), z.literal('audio'), z.literal('video'), expressionSchema]).optional(),
      additionalFields: z.object({ fileName: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
