// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleSheets/v45/resource_sheet/operation_append_or_update.schema.js
/**
 * Google Sheets Node - Version 4.5 - Zod Schema
 * Discriminator: resource=sheet, operation=appendOrUpdate
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('sheet').default('sheet'),
      operation: z.literal('appendOrUpdate'),
      authentication: z.union([z.literal('serviceAccount'), z.literal('oAuth2'), expressionSchema]).optional(),
      documentId: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('url'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      sheetName: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('url'), z.literal('id'), z.literal('name')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      columns: resolveSchema({ parameters, schema: resourceMapperValueSchema, required: false, displayOptions: {"hide":{"sheetName":[""]}}, defaults: {"sheetName":{"mode":"list","value":""}} }),
      options: resolveSchema({ parameters, schema: z.object({ cellFormat: z.union([z.literal('USER_ENTERED'), z.literal('RAW'), expressionSchema]).optional(), locationDefine: z.unknown().optional(), handlingExtraData: z.union([z.literal('insertInNewColumn'), z.literal('ignoreIt'), z.literal('error'), expressionSchema]).optional(), handlingExtraData: z.union([z.literal('insertInNewColumn'), z.literal('ignoreIt'), z.literal('error'), expressionSchema]).optional(), useAppend: booleanOrExpression.optional() }), required: false, displayOptions: {"hide":{"sheetName":[""]}}, defaults: {"sheetName":{"mode":"list","value":""}} }),
    }).optional(),
  });
};
