// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/reddit/v1/resource_post/operation_search.schema.js
/**
 * Reddit Node - Version 1 - Zod Schema
 * Discriminator: resource=post, operation=search
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('post').default('post'),
      operation: z.literal('search'),
      location: z.union([z.literal('allReddit'), z.literal('subreddit'), expressionSchema]).optional(),
      subreddit: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"location":["subreddit"]}}, defaults: {"location":"subreddit"} }),
      keyword: stringOrExpression.optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      additionalFields: z.object({ sort: z.union([z.literal('comments'), z.literal('hot'), z.literal('new'), z.literal('relevance'), z.literal('top'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
