// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/reddit/v1/resource_post/operation_create.schema.js
/**
 * Reddit Node - Version 1 - Zod Schema
 * Discriminator: resource=post, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('post').default('post'),
      operation: z.literal('create').default('create'),
      subreddit: stringOrExpression.optional(),
      kind: z.union([z.literal('self'), z.literal('link'), z.literal('image'), expressionSchema]).optional(),
      title: stringOrExpression.optional(),
      url: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"kind":["link","image"]}}, defaults: {"kind":"self"} }),
      text: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"kind":["self"]}}, defaults: {"kind":"self"} }),
      resubmit: resolveSchema({ parameters, schema: booleanOrExpression, required: false, displayOptions: {"show":{"kind":["link","image"]}}, defaults: {"kind":"self"} }),
    }).optional(),
  });
};
