// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/twitter/v2/resource_tweet/operation_search.schema.js
/**
 * X (Formerly Twitter) Node - Version 2 - Zod Schema
 * Discriminator: resource=tweet, operation=search
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('tweet').default('tweet'),
      operation: z.literal('search'),
      searchText: stringOrExpression.optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      additionalFields: z.object({ sortOrder: z.union([z.literal('recency'), z.literal('relevancy'), expressionSchema]).optional(), startTime: stringOrExpression.optional(), endTime: stringOrExpression.optional(), tweetFieldsObject: z.array(z.union([z.literal('attachments'), z.literal('author_id'), z.literal('context_annotations'), z.literal('conversation_id'), z.literal('created_at'), z.literal('edit_controls'), z.literal('entities'), z.literal('geo'), z.literal('id'), z.literal('in_reply_to_user_id'), z.literal('lang'), z.literal('non_public_metrics'), z.literal('public_metrics'), z.literal('organic_metrics'), z.literal('promoted_metrics'), z.literal('possibly_sensitive'), z.literal('referenced_tweets'), z.literal('reply_settings'), z.literal('source'), z.literal('text'), z.literal('withheld')])).optional() }).optional(),
    }).optional(),
  });
};
