// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/lemlist/v2/resource_campaign/operation_get_stats.ts
/**
 * Lemlist Node - Version 2
 * Discriminator: resource=campaign, operation=getStats
 */


interface Credentials {
  lemlistApi: CredentialReference;
}

export type LemlistV2CampaignGetStatsParams = {
  resource: 'campaign';
  operation: 'getStats';
/**
 * ID of the campaign to get stats for. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 * @default []
 */
    campaignId?: string | Expression<string>;
/**
 * Start Date
 */
    startDate?: string | Expression<string>;
/**
 * End Date
 */
    endDate?: string | Expression<string>;
/**
 * Timezone
 */
    timezone?: string | Expression<string> | PlaceholderValue;
};

export type LemlistV2CampaignGetStatsOutput = {
  clickedCount?: number;
  deliveredCount?: number;
  interestedCount?: number;
  leadCompleted?: number;
  leadInProgress?: number;
  leadReadyToSend?: number;
  leadToLaunch?: number;
  leadTotal?: number;
  openedCount?: number;
  repliedCount?: number;
  sentCount?: number;
};

export type LemlistV2CampaignGetStatsNode = {
  type: 'n8n-nodes-base.lemlist';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<LemlistV2CampaignGetStatsParams>;
  output?: Items<LemlistV2CampaignGetStatsOutput>;
};
