// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/misp/v1/resource_attribute/operation_search.schema.js
/**
 * MISP Node - Version 1 - Zod Schema
 * Discriminator: resource=attribute, operation=search
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('attribute').default('attribute'),
      operation: z.literal('search'),
      useJson: booleanOrExpression.optional(),
      jsonOutput: resolveSchema({ parameters, schema: z.union([iDataObjectSchema, z.string()]), required: false, displayOptions: {"show":{"useJson":[true]}}, defaults: {"useJson":false} }),
      value: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"useJson":[false]}}, defaults: {"useJson":false} }),
      additionalFields: resolveSchema({ parameters, schema: z.object({ category: stringOrExpression.optional(), deleted: booleanOrExpression.optional(), searchall: stringOrExpression.optional(), tags: stringOrExpression.optional(), type: stringOrExpression.optional(), published: booleanOrExpression.optional() }), required: false, displayOptions: {"show":{"useJson":[false]}}, defaults: {"useJson":false} }),
    }).optional(),
  });
};
