// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/misp/v1/resource_attribute/operation_update.ts
/**
 * MISP Node - Version 1
 * Discriminator: resource=attribute, operation=update
 */


interface Credentials {
  mispApi: CredentialReference;
}

export type MispV1AttributeUpdateParams = {
  resource: 'attribute';
  operation: 'update';
/**
 * ID of the attribute to update
 */
    attributeId?: string | Expression<string> | PlaceholderValue;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** Who will be able to see this event once published
     * @default 0
     */
    distribution?: 3 | 2 | 5 | 4 | 1 | 0 | Expression<number>;
    /** Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;. Use only for when &lt;code&gt;Sharing Group&lt;/code&gt; is selected in &lt;code&gt;Distribution&lt;/code&gt;.
     */
    sharing_group_id?: string | Expression<string>;
  };
};

export type MispV1AttributeUpdateNode = {
  type: 'n8n-nodes-base.misp';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MispV1AttributeUpdateParams>;
};
