// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/misp/v1/resource_attribute/operation_create.schema.js
/**
 * MISP Node - Version 1 - Zod Schema
 * Discriminator: resource=attribute, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('attribute').default('attribute'),
      operation: z.literal('create').default('create'),
      eventId: stringOrExpression.optional(),
      type: z.union([z.literal('text'), z.literal('url'), z.literal('comment'), expressionSchema]).optional(),
      value: stringOrExpression.optional(),
      additionalFields: z.object({ distribution: z.union([z.literal(3), z.literal(2), z.literal(5), z.literal(4), z.literal(1), z.literal(0), expressionSchema]).optional(), sharing_group_id: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
