// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/bambooHr/v1/resource_employee_document/operation_upload.ts
/**
 * BambooHR Node - Version 1
 * Discriminator: resource=employeeDocument, operation=upload
 */


interface Credentials {
  bambooHrApi: CredentialReference;
}

/** Upload an employee document */
export type BambooHrV1EmployeeDocumentUploadParams = {
  resource: 'employeeDocument';
  operation: 'upload';
/**
 * ID of the employee
 */
    employeeId?: string | Expression<string> | PlaceholderValue;
/**
 * Employee Document Category ID
 */
    categoryId?: string | Expression<string> | PlaceholderValue;
/**
 * The name of the input field containing the binary file data to be uploaded. Supported file types: PNG, JPEG.
 * @default data
 */
    binaryPropertyName?: string | Expression<string> | PlaceholderValue;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Whether this file is shared or not
     * @default true
     */
    share?: boolean | Expression<boolean>;
  };
};

export type BambooHrV1EmployeeDocumentUploadNode = {
  type: 'n8n-nodes-base.bambooHr';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<BambooHrV1EmployeeDocumentUploadParams>;
};
