// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/securityScorecard/v1/resource_portfolio/operation_create.schema.js
/**
 * SecurityScorecard Node - Version 1 - Zod Schema
 * Discriminator: resource=portfolio, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('portfolio'),
      operation: z.literal('create'),
      name: stringOrExpression.optional(),
      description: stringOrExpression.optional(),
      privacy: z.union([z.literal('private'), z.literal('shared'), z.literal('team'), expressionSchema]).optional(),
    }).optional(),
  });
};
