// @microsoft/agents-a365-observability/dist/cjs/index.js
"use strict";
// ------------------------------------------------------------------------------
// Copyright (c) Microsoft Corporation. All rights reserved.
// ------------------------------------------------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.formatError = exports.logger = exports.InferenceScope = exports.InvokeAgentScope = exports.ExecuteToolScope = exports.OpenTelemetryScope = exports.InferenceOperationType = exports.InvocationRole = exports.ExecutionType = exports.BaggageScope = exports.BaggageBuilder = exports.OpenTelemetryConstants = exports.Agent365ExporterOptions = exports.Builder = exports.ObservabilityManager = void 0;
// Main SDK classes
var ObservabilityManager_1 = require("./ObservabilityManager");
Object.defineProperty(exports, "ObservabilityManager", { enumerable: true, get: function () { return ObservabilityManager_1.ObservabilityManager; } });
var ObservabilityBuilder_1 = require("./ObservabilityBuilder");
Object.defineProperty(exports, "Builder", { enumerable: true, get: function () { return ObservabilityBuilder_1.ObservabilityBuilder; } });
var Agent365ExporterOptions_1 = require("./tracing/exporter/Agent365ExporterOptions");
Object.defineProperty(exports, "Agent365ExporterOptions", { enumerable: true, get: function () { return Agent365ExporterOptions_1.Agent365ExporterOptions; } });
// Tracing constants
var constants_1 = require("./tracing/constants");
Object.defineProperty(exports, "OpenTelemetryConstants", { enumerable: true, get: function () { return constants_1.OpenTelemetryConstants; } });
// Baggage builder
var BaggageBuilder_1 = require("./tracing/middleware/BaggageBuilder");
Object.defineProperty(exports, "BaggageBuilder", { enumerable: true, get: function () { return BaggageBuilder_1.BaggageBuilder; } });
Object.defineProperty(exports, "BaggageScope", { enumerable: true, get: function () { return BaggageBuilder_1.BaggageScope; } });
// Contracts and interfaces
var contracts_1 = require("./tracing/contracts");
Object.defineProperty(exports, "ExecutionType", { enumerable: true, get: function () { return contracts_1.ExecutionType; } });
Object.defineProperty(exports, "InvocationRole", { enumerable: true, get: function () { return contracts_1.InvocationRole; } });
Object.defineProperty(exports, "InferenceOperationType", { enumerable: true, get: function () { return contracts_1.InferenceOperationType; } });
// Scopes
var OpenTelemetryScope_1 = require("./tracing/scopes/OpenTelemetryScope");
Object.defineProperty(exports, "OpenTelemetryScope", { enumerable: true, get: function () { return OpenTelemetryScope_1.OpenTelemetryScope; } });
var ExecuteToolScope_1 = require("./tracing/scopes/ExecuteToolScope");
Object.defineProperty(exports, "ExecuteToolScope", { enumerable: true, get: function () { return ExecuteToolScope_1.ExecuteToolScope; } });
var InvokeAgentScope_1 = require("./tracing/scopes/InvokeAgentScope");
Object.defineProperty(exports, "InvokeAgentScope", { enumerable: true, get: function () { return InvokeAgentScope_1.InvokeAgentScope; } });
var InferenceScope_1 = require("./tracing/scopes/InferenceScope");
Object.defineProperty(exports, "InferenceScope", { enumerable: true, get: function () { return InferenceScope_1.InferenceScope; } });
var logging_1 = require("./utils/logging");
Object.defineProperty(exports, "logger", { enumerable: true, get: function () { return logging_1.logger; } });
Object.defineProperty(exports, "formatError", { enumerable: true, get: function () { return logging_1.formatError; } });
