// @microsoft/agents-a365-observability/dist/cjs/ObservabilityBuilder.js
"use strict";
// ------------------------------------------------------------------------------
// Copyright (c) Microsoft Corporation. All rights reserved.
// ------------------------------------------------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityBuilder = void 0;
const sdk_node_1 = require("@opentelemetry/sdk-node");
const sdk_trace_base_1 = require("@opentelemetry/sdk-trace-base");
const SpanProcessor_1 = require("./tracing/processors/SpanProcessor");
const util_1 = require("./tracing/util");
const Agent365Exporter_1 = require("./tracing/exporter/Agent365Exporter");
const Agent365ExporterOptions_1 = require("./tracing/exporter/Agent365ExporterOptions");
const resources_1 = require("@opentelemetry/resources");
const semantic_conventions_1 = require("@opentelemetry/semantic-conventions");
const api_1 = require("@opentelemetry/api");
/**
 * Builder for configuring Agent 365 with OpenTelemetry tracing
 */
class ObservabilityBuilder {
    constructor() {
        this.options = {};
        this.isBuilt = false;
    }
    /**
     * Configures the service name and version for telemetry
     * @param serviceName The service name
     * @param serviceVersion The service version
     * @returns The builder instance for method chaining
     */
    withService(serviceName, serviceVersion) {
        this.options.serviceName = serviceName;
        this.options.serviceVersion = serviceVersion;
        return this;
    }
    /**
     * Configures the token resolver for Agent 365 exporter
     * @param tokenResolver Function to resolve authentication tokens
     * @returns The builder instance for method chaining
     */
    withTokenResolver(tokenResolver) {
        this.options.tokenResolver = tokenResolver;
        return this;
    }
    /**
     * Configures the cluster category for Agent 365 exporter
     * @param clusterCategory The cluster category (e.g., "preprod", "prod")
     * @returns The builder instance for method chaining
     */
    withClusterCategory(clusterCategory) {
        this.options.clusterCategory = clusterCategory;
        return this;
    }
    /**
     * Provide a partial set of Agent365ExporterOptions. These will be merged with
     * defaults and any explicitly configured clusterCategory/tokenResolver.
     * @param exporterOptions Partial exporter options
     * @returns The builder instance for chaining
     */
    withExporterOptions(exporterOptions) {
        this.options.exporterOptions = {
            ...(this.options.exporterOptions || {}),
            ...exporterOptions
        };
        return this;
    }
    createBatchProcessor() {
        if (!(0, util_1.isAgent365ExporterEnabled)()) {
            return new sdk_trace_base_1.BatchSpanProcessor(new sdk_trace_base_1.ConsoleSpanExporter());
        }
        const opts = new Agent365ExporterOptions_1.Agent365ExporterOptions();
        if (this.options.exporterOptions) {
            Object.assign(opts, this.options.exporterOptions);
        }
        opts.clusterCategory = this.options.clusterCategory || opts.clusterCategory || 'prod';
        if (this.options.tokenResolver) {
            opts.tokenResolver = this.options.tokenResolver;
        }
        return new sdk_trace_base_1.BatchSpanProcessor(new Agent365Exporter_1.Agent365Exporter(opts), {
            maxQueueSize: opts.maxQueueSize,
            scheduledDelayMillis: opts.scheduledDelayMilliseconds,
            exportTimeoutMillis: opts.exporterTimeoutMilliseconds,
            maxExportBatchSize: opts.maxExportBatchSize
        });
    }
    createResource() {
        const serviceName = this.options.serviceVersion
            ? `${this.options.serviceName}-${this.options.serviceVersion}`
            : this.options.serviceName ?? 'Agent365-TypeScript';
        return (0, resources_1.resourceFromAttributes)({
            [semantic_conventions_1.ATTR_SERVICE_NAME]: serviceName,
        });
    }
    /**
     * Builds and initializes the Agent 365 configuration
     * @returns The configured NodeSDK instance
     */
    build() {
        if (this.isBuilt) {
            return this.isBuilt;
        }
        // Create processors in the desired order:
        // 1. baggage enricher (copies baggage -> span attributes)
        const spanProcessor = new SpanProcessor_1.SpanProcessor();
        // 2. batch processor that actually ships spans out
        const batchProcessor = this.createBatchProcessor();
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        const globalProvider = api_1.trace.getTracerProvider();
        const canAddProcessors = typeof globalProvider?.addSpanProcessor === 'function' &&
            typeof globalProvider?.resource !== 'undefined';
        if (canAddProcessors) {
            // Someone else already created a provider (maybe their own NodeSDK).
            // We DO NOT create a new NodeSDK.
            // We just add our baggage enricher + batch exporter to their provider,
            // but only if they aren't already there.
            this.attachProcessorIfMissing(globalProvider, spanProcessor);
            this.attachProcessorIfMissing(globalProvider, batchProcessor);
            this.isBuilt = true;
            this.sdk = undefined; // we didn't create/own one
            return this.isBuilt;
        }
        // Create & configure the NodeSDK manually so we can inject processors + resource.
        this.sdk = new sdk_node_1.NodeSDK({
            resource: this.createResource(),
            spanProcessors: [
                spanProcessor,
                batchProcessor,
            ],
        });
        this.isBuilt = true;
        return this.isBuilt;
    }
    /**
     * Starts the OpenTelemetry SDK
     */
    start() {
        if (!this.isBuilt) {
            this.build();
        }
        if (this.sdk) {
            this.sdk.start();
        }
    }
    /**
     * Shuts down the OpenTelemetry SDK
     */
    async shutdown() {
        if (this.sdk) {
            await this.sdk.shutdown();
        }
    }
    /**
     * Helper to avoid double-registering same processor type.
     */
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    attachProcessorIfMissing(provider, processor) {
        const active = provider._activeSpanProcessor?._spanProcessors;
        const alreadyAdded = Array.isArray(active) &&
            // eslint-disable-next-line @typescript-eslint/no-explicit-any
            active.some((p) => p?.constructor?.name === processor.constructor.name);
        if (!alreadyAdded) {
            provider.addSpanProcessor(processor);
        }
    }
}
exports.ObservabilityBuilder = ObservabilityBuilder;
