// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/harvest/v1/resource_invoice/operation_create.schema.js
/**
 * Harvest Node - Version 1 - Zod Schema
 * Discriminator: resource=invoice, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('invoice'),
      operation: z.literal('create'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      accountId: stringOrExpression.optional(),
      clientId: stringOrExpression.optional(),
      additionalFields: z.object({ currency: stringOrExpression.optional(), over_budget_notification_percentage: stringOrExpression.optional(), ends_on: stringOrExpression.optional(), estimate_id: stringOrExpression.optional(), issue_date: stringOrExpression.optional(), notes: stringOrExpression.optional(), number: stringOrExpression.optional(), payment_term: stringOrExpression.optional(), purchase_order: stringOrExpression.optional(), retainer_id: booleanOrExpression.optional(), subject: stringOrExpression.optional(), tax: stringOrExpression.optional(), tax2: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
