// @n8n/n8n-nodes-langchain/dist/nodes/llms/LmChatOpenRouter/LmChatOpenRouter.node.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LmChatOpenRouter = void 0;
const openai_1 = require("@langchain/openai");
const ai_utilities_1 = require("@n8n/ai-utilities");
const n8n_workflow_1 = require("n8n-workflow");
const error_handling_1 = require("../../vendors/OpenAi/helpers/error-handling");
function isOpenAIResponseWithChoices(json) {
    return (typeof json === 'object' &&
        json !== null &&
        'choices' in json &&
        Array.isArray(json.choices));
}
function createOpenRouterFetch(baseFetch) {
    return async (input, init) => {
        const response = await baseFetch(input, init);
        const contentType = response.headers.get('content-type') ?? '';
        if (!contentType.includes('json'))
            return response;
        const clone = response.clone();
        const json = await response.json();
        if (!isOpenAIResponseWithChoices(json))
            return clone;
        const isInvalidArgs = (args) => typeof args !== 'string' || !args.trim();
        const toolCallsToFix = json.choices
            .flatMap((choice) => choice.message?.tool_calls ?? [])
            .filter((tc) => tc.function && isInvalidArgs(tc.function.arguments));
        if (toolCallsToFix.length === 0)
            return clone;
        for (const tc of toolCallsToFix) {
            if (!tc.function)
                continue;
            const { arguments: args } = tc.function;
            const isPlainObject = typeof args === 'object' && args !== null && !Array.isArray(args);
            tc.function.arguments = isPlainObject ? JSON.stringify(args) : '{}';
        }
        const body = JSON.stringify(json);
        return new Response(body, {
            status: response.status,
            statusText: response.statusText,
            headers: { 'content-type': contentType },
        });
    };
}
class LmChatOpenRouter {
    constructor() {
        this.description = {
            displayName: 'OpenRouter Chat Model',
            name: 'lmChatOpenRouter',
            icon: { light: 'file:openrouter.svg', dark: 'file:openrouter.dark.svg' },
            group: ['transform'],
            version: [1],
            description: 'For advanced usage with an AI chain',
            defaults: {
                name: 'OpenRouter Chat Model',
            },
            codex: {
                categories: ['AI'],
                subcategories: {
                    AI: ['Language Models', 'Root Nodes'],
                    'Language Models': ['Chat Models (Recommended)'],
                },
                resources: {
                    primaryDocumentation: [
                        {
                            url: 'https://docs.n8n.io/integrations/builtin/cluster-nodes/sub-nodes/n8n-nodes-langchain.lmchatopenrouter/',
                        },
                    ],
                },
            },
            inputs: [],
            outputs: [n8n_workflow_1.NodeConnectionTypes.AiLanguageModel],
            outputNames: ['Model'],
            credentials: [
                {
                    name: 'openRouterApi',
                    required: true,
                },
            ],
            requestDefaults: {
                ignoreHttpStatusErrors: true,
                baseURL: '={{ $credentials?.url }}',
            },
            properties: [
                (0, ai_utilities_1.getConnectionHintNoticeField)([n8n_workflow_1.NodeConnectionTypes.AiChain, n8n_workflow_1.NodeConnectionTypes.AiAgent]),
                {
                    displayName: 'If using JSON response format, you must include word "json" in the prompt in your chain or agent. Also, make sure to select latest models released post November 2023.',
                    name: 'notice',
                    type: 'notice',
                    default: '',
                    displayOptions: {
                        show: {
                            '/options.responseFormat': ['json_object'],
                        },
                    },
                },
                {
                    displayName: 'Model',
                    name: 'model',
                    type: 'options',
                    description: 'The model which will generate the completion. <a href="https://openrouter.ai/docs/models">Learn more</a>.',
                    typeOptions: {
                        loadOptions: {
                            routing: {
                                request: {
                                    method: 'GET',
                                    url: '/models',
                                },
                                output: {
                                    postReceive: [
                                        {
                                            type: 'rootProperty',
                                            properties: {
                                                property: 'data',
                                            },
                                        },
                                        {
                                            type: 'setKeyValue',
                                            properties: {
                                                name: '={{$responseItem.id}}',
                                                value: '={{$responseItem.id}}',
                                            },
                                        },
                                        {
                                            type: 'sort',
                                            properties: {
                                                key: 'name',
                                            },
                                        },
                                    ],
                                },
                            },
                        },
                    },
                    routing: {
                        send: {
                            type: 'body',
                            property: 'model',
                        },
                    },
                    default: 'openai/gpt-4.1-mini',
                },
                {
                    displayName: 'Options',
                    name: 'options',
                    placeholder: 'Add Option',
                    description: 'Additional options to add',
                    type: 'collection',
                    default: {},
                    options: [
                        {
                            displayName: 'Frequency Penalty',
                            name: 'frequencyPenalty',
                            default: 0,
                            typeOptions: { maxValue: 2, minValue: -2, numberPrecision: 1 },
                            description: "Positive values penalize new tokens based on their existing frequency in the text so far, decreasing the model's likelihood to repeat the same line verbatim",
                            type: 'number',
                        },
                        {
                            displayName: 'Maximum Number of Tokens',
                            name: 'maxTokens',
                            default: -1,
                            description: 'The maximum number of tokens to generate in the completion. Most models have a context length of 2048 tokens (except for the newest models, which support 32,768).',
                            type: 'number',
                            typeOptions: {
                                maxValue: 32768,
                            },
                        },
                        {
                            displayName: 'Response Format',
                            name: 'responseFormat',
                            default: 'text',
                            type: 'options',
                            options: [
                                {
                                    name: 'Text',
                                    value: 'text',
                                    description: 'Regular text response',
                                },
                                {
                                    name: 'JSON',
                                    value: 'json_object',
                                    description: 'Enables JSON mode, which should guarantee the message the model generates is valid JSON',
                                },
                            ],
                        },
                        {
                            displayName: 'Presence Penalty',
                            name: 'presencePenalty',
                            default: 0,
                            typeOptions: { maxValue: 2, minValue: -2, numberPrecision: 1 },
                            description: "Positive values penalize new tokens based on whether they appear in the text so far, increasing the model's likelihood to talk about new topics",
                            type: 'number',
                        },
                        {
                            displayName: 'Sampling Temperature',
                            name: 'temperature',
                            default: 0.7,
                            typeOptions: { maxValue: 2, minValue: 0, numberPrecision: 1 },
                            description: 'Controls randomness: Lowering results in less random completions. As the temperature approaches zero, the model will become deterministic and repetitive.',
                            type: 'number',
                        },
                        {
                            displayName: 'Timeout',
                            name: 'timeout',
                            default: 360000,
                            description: 'Maximum amount of time a request is allowed to take in milliseconds',
                            type: 'number',
                        },
                        {
                            displayName: 'Max Retries',
                            name: 'maxRetries',
                            default: 2,
                            description: 'Maximum number of retries to attempt',
                            type: 'number',
                        },
                        {
                            displayName: 'Top P',
                            name: 'topP',
                            default: 1,
                            typeOptions: { maxValue: 1, minValue: 0, numberPrecision: 1 },
                            description: 'Controls diversity via nucleus sampling: 0.5 means half of all likelihood-weighted options are considered. We generally recommend altering this or temperature but not both.',
                            type: 'number',
                        },
                    ],
                },
            ],
        };
    }
    async supplyData(itemIndex) {
        const credentials = await this.getCredentials('openRouterApi');
        const modelName = this.getNodeParameter('model', itemIndex);
        const options = this.getNodeParameter('options', itemIndex, {});
        const timeout = options.timeout;
        const configuration = {
            baseURL: credentials.url,
            fetch: createOpenRouterFetch(globalThis.fetch),
            fetchOptions: {
                dispatcher: (0, ai_utilities_1.getProxyAgent)(credentials.url, {
                    headersTimeout: timeout,
                    bodyTimeout: timeout,
                }),
            },
        };
        const model = new openai_1.ChatOpenAI({
            apiKey: credentials.apiKey,
            model: modelName,
            ...options,
            timeout,
            maxRetries: options.maxRetries ?? 2,
            configuration,
            callbacks: [new ai_utilities_1.N8nLlmTracing(this)],
            modelKwargs: options.responseFormat
                ? {
                    response_format: { type: options.responseFormat },
                }
                : undefined,
            onFailedAttempt: (0, ai_utilities_1.makeN8nLlmFailedAttemptHandler)(this, error_handling_1.openAiFailedAttemptHandler),
        });
        return {
            response: model,
        };
    }
}
exports.LmChatOpenRouter = LmChatOpenRouter;
