// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/webflow/v2/resource_item/operation_create.ts
/**
 * Webflow Node - Version 2
 * Discriminator: resource=item, operation=create
 */


interface Credentials {
  webflowOAuth2Api: CredentialReference;
}

export type WebflowV2ItemCreateParams = {
  resource: 'item';
  operation: 'create';
/**
 * ID of the site containing the collection whose items to add to. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    siteId?: string | Expression<string>;
/**
 * ID of the collection to add an item to. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    collectionId?: string | Expression<string>;
/**
 * Whether the item should be published on the live site
 * @default false
 */
    live?: boolean | Expression<boolean>;
/**
 * Fields
 * @default {}
 */
    fieldsUi?: {
        /** Field
     */
    fieldValues?: Array<{
      /** Field to set for the item to create. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
       */
      fieldId?: string | Expression<string>;
      /** Value to set for the item to create
       */
      fieldValue?: string | Expression<string> | PlaceholderValue;
    }>;
  };
};

export type WebflowV2ItemCreateOutput = {
  createdOn?: string;
  fieldData?: {
    name?: string;
    slug?: string;
  };
  id?: string;
  isArchived?: boolean;
  isDraft?: boolean;
  lastUpdated?: string;
};

export type WebflowV2ItemCreateNode = {
  type: 'n8n-nodes-base.webflow';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<WebflowV2ItemCreateParams>;
  output?: Items<WebflowV2ItemCreateOutput>;
};
