// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/splunk/v2/resource_search/operation_get_all.ts
/**
 * Splunk Node - Version 2
 * Discriminator: resource=search, operation=getAll
 */


interface Credentials {
  splunkApi: CredentialReference;
}

/** Retrieve many search reports */
export type SplunkV2SearchGetAllParams = {
  resource: 'search';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Sort
 * @default {}
 */
    sort?: {
        /** Values
     */
    values?: {
      /** Sort Direction
       * @default asc
       */
      sort_dir?: 'asc' | 'desc' | Expression<string>;
      /** Key name to use for sorting
       */
      sort_key?: string | Expression<string> | PlaceholderValue;
      /** Sort Mode
       * @default auto
       */
      sort_mode?: 'auto' | 'alpha' | 'alpha_case' | 'num' | Expression<string>;
    };
  };
};

export type SplunkV2SearchGetAllNode = {
  type: 'n8n-nodes-base.splunk';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<SplunkV2SearchGetAllParams>;
};
