// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/splunk/v2/resource_search/operation_create.schema.js
/**
 * Splunk Node - Version 2 - Zod Schema
 * Discriminator: resource=search, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('search').default('search'),
      operation: z.literal('create'),
      search: stringOrExpression.optional(),
      additionalFields: z.object({ adhoc_search_level: z.union([z.literal('fast'), z.literal('smart'), z.literal('verbose'), expressionSchema]).optional(), auto_cancel: numberOrExpression.optional(), auto_finalize_ec: numberOrExpression.optional(), auto_pause: numberOrExpression.optional(), index_earliest: stringOrExpression.optional(), earliest_time: stringOrExpression.optional(), exec_mode: z.union([z.literal('blocking'), z.literal('normal'), z.literal('oneshot'), expressionSchema]).optional(), indexedRealtimeOffset: numberOrExpression.optional(), index_latest: stringOrExpression.optional(), latest_time: stringOrExpression.optional(), max_time: numberOrExpression.optional(), namespace: stringOrExpression.optional(), reduce_freq: numberOrExpression.optional(), remote_server_list: stringOrExpression.optional(), reuse_max_seconds_ago: numberOrExpression.optional(), rf: stringOrExpression.optional(), search_mode: z.union([z.literal('normal'), z.literal('realtime'), expressionSchema]).optional(), status_buckets: numberOrExpression.optional(), timeout: numberOrExpression.optional(), workload_pool: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
