// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/elasticsearch/v1/resource_index/operation_get.schema.js
/**
 * Elasticsearch Node - Version 1 - Zod Schema
 * Discriminator: resource=index, operation=get
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('index'),
      operation: z.literal('get').default('get'),
      indexId: stringOrExpression.optional(),
      additionalFields: z.object({ allow_no_indices: booleanOrExpression.optional(), expand_wildcards: z.union([z.literal('all'), z.literal('closed'), z.literal('hidden'), z.literal('none'), z.literal('open'), expressionSchema]).optional(), flat_settings: booleanOrExpression.optional(), ignore_unavailable: booleanOrExpression.optional(), include_defaults: booleanOrExpression.optional(), local: booleanOrExpression.optional(), master_timeout: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
