// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/elasticsearch/v1/resource_index/operation_create.schema.js
/**
 * Elasticsearch Node - Version 1 - Zod Schema
 * Discriminator: resource=index, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('index'),
      operation: z.literal('create'),
      indexId: stringOrExpression.optional(),
      additionalFields: z.object({ aliases: z.union([iDataObjectSchema, z.string()]).optional(), include_type_name: booleanOrExpression.optional(), mappings: z.union([iDataObjectSchema, z.string()]).optional(), master_timeout: stringOrExpression.optional(), settings: z.union([iDataObjectSchema, z.string()]).optional(), timeout: stringOrExpression.optional(), wait_for_active_shards: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
