// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/emailSend/v1.schema.js
/**
 * Send Email Node - Version 1 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    fromEmail: stringOrExpression.optional(),
    toEmail: stringOrExpression.optional(),
    ccEmail: stringOrExpression.optional(),
    bccEmail: stringOrExpression.optional(),
    subject: stringOrExpression.optional(),
    text: stringOrExpression.optional(),
    html: stringOrExpression.optional(),
    attachments: stringOrExpression.optional(),
    options: z.object({ allowUnauthorizedCerts: booleanOrExpression.optional(), replyTo: stringOrExpression.optional() }).optional(),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
