// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/emailSend/v1.ts
/**
 * Send Email Node - Version 1
 * Sends an Email
 */


export interface EmailSendV1Params {
/**
 * Email address of the sender optional with name
 */
    fromEmail?: string | Expression<string> | PlaceholderValue;
/**
 * Email address of the recipient
 */
    toEmail?: string | Expression<string> | PlaceholderValue;
/**
 * Email address of CC recipient
 */
    ccEmail?: string | Expression<string> | PlaceholderValue;
/**
 * Email address of BCC recipient
 */
    bccEmail?: string | Expression<string> | PlaceholderValue;
/**
 * Subject line of the email
 */
    subject?: string | Expression<string> | PlaceholderValue;
/**
 * Plain text message of email
 */
    text?: string | Expression<string> | PlaceholderValue;
/**
 * HTML text message of email
 */
    html?: string | Expression<string> | PlaceholderValue;
/**
 * Name of the binary properties that contain data to add to email as attachment. Multiple ones can be comma-separated.
 */
    attachments?: string | Expression<string> | PlaceholderValue;
  options?: {
    /** Whether to connect even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean | Expression<boolean>;
    /** The email address to send the reply to
     */
    replyTo?: string | Expression<string> | PlaceholderValue;
  };
}

export interface EmailSendV1Credentials {
  smtp: CredentialReference;
}

interface EmailSendV1NodeBase {
  type: 'n8n-nodes-base.emailSend';
  version: 1;
  credentials?: EmailSendV1Credentials;
}

export type EmailSendV1ParamsNode = EmailSendV1NodeBase & {
  config: NodeConfig<EmailSendV1Params>;
};

export type EmailSendV1Node = EmailSendV1ParamsNode;
