// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/activeCampaign/v1/resource_ecommerce_customer/operation_create.ts
/**
 * ActiveCampaign Node - Version 1
 * Discriminator: resource=ecommerceCustomer, operation=create
 */


interface Credentials {
  activeCampaignApi: CredentialReference;
}

/** Create an account */
export type ActiveCampaignV1EcommerceCustomerCreateParams = {
  resource: 'ecommerceCustomer';
  operation: 'create';
/**
 * The ID of the connection object for the service where the customer originates
 */
    connectionid?: string | Expression<string> | PlaceholderValue;
/**
 * The ID of the customer in the external service
 */
    externalid?: string | Expression<string> | PlaceholderValue;
/**
 * The email address of the customer
 */
    email?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Whether customer has opt-ed in to marketing communications
     * @default false
     */
    acceptsMarketing?: boolean | Expression<boolean>;
  };
};

export type ActiveCampaignV1EcommerceCustomerCreateNode = {
  type: 'n8n-nodes-base.activeCampaign';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ActiveCampaignV1EcommerceCustomerCreateParams>;
};
