// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/activeCampaign/v1/resource_ecommerce_customer/operation_create.schema.js
/**
 * ActiveCampaign Node - Version 1 - Zod Schema
 * Discriminator: resource=ecommerceCustomer, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('ecommerceCustomer'),
      operation: z.literal('create').default('create'),
      connectionid: stringOrExpression.optional(),
      externalid: stringOrExpression.optional(),
      email: stringOrExpression.optional(),
      additionalFields: z.object({ acceptsMarketing: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
