// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/n8n/v1/resource_credential/operation_create.ts
/**
 * n8n Node - Version 1
 * Discriminator: resource=credential, operation=create
 */


interface Credentials {
  n8nApi: CredentialReference;
}

export type N8nV1CredentialCreateParams = {
  resource: 'credential';
  operation: 'create';
/**
 * Name of the new credential
 */
    name?: string | Expression<string> | PlaceholderValue;
/**
 * The available types depend on nodes installed on the n8n instance. Some built-in types include e.g. 'githubApi', 'notionApi', and 'slackApi'.
 */
    credentialTypeName?: string | Expression<string> | PlaceholderValue;
/**
 * A valid JSON object with properties required for this Credential Type. To see the expected format, you can use 'Get Schema' operation.
 */
    data?: IDataObject | string | Expression<string>;
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type N8nV1CredentialCreateNode = {
  type: 'n8n-nodes-base.n8n';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<N8nV1CredentialCreateParams>;
};
