// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/salesforce/v1/resource_custom_object/operation_get.ts
/**
 * Salesforce Node - Version 1
 * Discriminator: resource=customObject, operation=get
 */


interface Credentials {
  salesforceOAuth2Api: CredentialReference;
  salesforceJwtApi: CredentialReference;
}

/** Represents a custom object */
export type SalesforceV1CustomObjectGetParams = {
  resource: 'customObject';
  operation: 'get';
/**
 * OAuth Authorization Flow
 * @default oAuth2
 */
    authentication?: 'oAuth2' | 'jwt' | Expression<string>;
/**
 * Name of the custom object. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    customObject?: string | Expression<string>;
/**
 * Record ID to be retrieved
 */
    recordId?: string | Expression<string> | PlaceholderValue;
};

export type SalesforceV1CustomObjectGetOutput = {
  attributes?: {
    type?: string;
    url?: string;
  };
  CreatedById?: string;
  CreatedDate?: string;
  CurrencyIsoCode?: string;
  Id?: string;
  IsDeleted?: boolean;
  LastModifiedById?: string;
  LastModifiedDate?: string;
  Name?: string;
  OwnerId?: string;
  SystemModstamp?: string;
};

export type SalesforceV1CustomObjectGetNode = {
  type: 'n8n-nodes-base.salesforce';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SalesforceV1CustomObjectGetParams>;
  output?: Items<SalesforceV1CustomObjectGetOutput>;
};
