// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/salesforce/v1/resource_custom_object/operation_delete.ts
/**
 * Salesforce Node - Version 1
 * Discriminator: resource=customObject, operation=delete
 */


interface Credentials {
  salesforceOAuth2Api: CredentialReference;
  salesforceJwtApi: CredentialReference;
}

/** Represents a custom object */
export type SalesforceV1CustomObjectDeleteParams = {
  resource: 'customObject';
  operation: 'delete';
/**
 * OAuth Authorization Flow
 * @default oAuth2
 */
    authentication?: 'oAuth2' | 'jwt' | Expression<string>;
/**
 * Name of the custom object. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    customObject?: string | Expression<string>;
/**
 * Record ID to be deleted
 */
    recordId?: string | Expression<string> | PlaceholderValue;
};

export type SalesforceV1CustomObjectDeleteOutput = {
  success?: boolean;
};

export type SalesforceV1CustomObjectDeleteNode = {
  type: 'n8n-nodes-base.salesforce';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SalesforceV1CustomObjectDeleteParams>;
  output?: Items<SalesforceV1CustomObjectDeleteOutput>;
};
