// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/salesforce/v1/resource_custom_object/operation_update.schema.js
/**
 * Salesforce Node - Version 1 - Zod Schema
 * Discriminator: resource=customObject, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('customObject'),
      operation: z.literal('update'),
      authentication: z.union([z.literal('oAuth2'), z.literal('jwt'), expressionSchema]).optional(),
      customObject: stringOrExpression.optional(),
      recordId: stringOrExpression.optional(),
      customFieldsUi: z.object({ customFieldsValues: z.array(z.object({ fieldId: stringOrExpression.optional(), value: stringOrExpression.optional() })).optional() }).optional(),
      updateFields: z.object({ recordTypeId: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
