// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/salesforce/v1/resource_custom_object/operation_update.ts
/**
 * Salesforce Node - Version 1
 * Discriminator: resource=customObject, operation=update
 */


interface Credentials {
  salesforceOAuth2Api: CredentialReference;
  salesforceJwtApi: CredentialReference;
}

/** Represents a custom object */
export type SalesforceV1CustomObjectUpdateParams = {
  resource: 'customObject';
  operation: 'update';
/**
 * OAuth Authorization Flow
 * @default oAuth2
 */
    authentication?: 'oAuth2' | 'jwt' | Expression<string>;
/**
 * Name of the custom object. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    customObject?: string | Expression<string>;
/**
 * Record ID to be updated
 */
    recordId?: string | Expression<string> | PlaceholderValue;
/**
 * Filter by custom fields
 * @default {}
 */
    customFieldsUi?: {
        /** Custom Field
     */
    customFieldsValues?: Array<{
      /** The ID of the field. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
       */
      fieldId?: string | Expression<string>;
      /** The value to set on custom field
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     */
    recordTypeId?: string | Expression<string>;
  };
};

export type SalesforceV1CustomObjectUpdateOutput = {
  success?: boolean;
};

export type SalesforceV1CustomObjectUpdateNode = {
  type: 'n8n-nodes-base.salesforce';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SalesforceV1CustomObjectUpdateParams>;
  output?: Items<SalesforceV1CustomObjectUpdateOutput>;
};
