// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/zohoCrm/v1/resource_product/operation_create.schema.js
/**
 * Zoho CRM Node - Version 1 - Zod Schema
 * Discriminator: resource=product, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('product'),
      operation: z.literal('create').default('create'),
      productName: stringOrExpression.optional(),
      additionalFields: z.object({ Commission_Rate: numberOrExpression.optional(), customFields: z.unknown().optional(), Description: stringOrExpression.optional(), Manufacturer: stringOrExpression.optional(), Product_Active: booleanOrExpression.optional(), Product_Category: stringOrExpression.optional(), Qty_in_Demand: numberOrExpression.optional(), Qty_in_Stock: numberOrExpression.optional(), Taxable: booleanOrExpression.optional(), Unit_Price: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
