// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/zohoCrm/v1/resource_product/operation_upsert.ts
/**
 * Zoho CRM Node - Version 1
 * Discriminator: resource=product, operation=upsert
 */


interface Credentials {
  zohoOAuth2Api: CredentialReference;
}

/** Create a new record, or update the current one if it already exists (upsert) */
export type ZohoCrmV1ProductUpsertParams = {
  resource: 'product';
  operation: 'upsert';
/**
 * Name of the product. If a record with this product name exists it will be updated, otherwise a new one will be created.
 */
    productName?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Commission rate for the product. For example, enter 12 for 12%.
     * @default 0
     */
    Commission_Rate?: number | Expression<number>;
    /** Filter by custom fields
     * @default {}
     */
    customFields?: {
        /** Custom Field
     */
    customFields?: Array<{
      /** Custom field to set a value to
       */
      fieldId?: string | Expression<string>;
      /** Value to set on custom field
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
    /** Description
     */
    Description?: string | Expression<string> | PlaceholderValue;
    /** Manufacturer
     */
    Manufacturer?: string | Expression<string> | PlaceholderValue;
    /** Product Active
     * @default false
     */
    Product_Active?: boolean | Expression<boolean>;
    /** Product Category
     */
    Product_Category?: string | Expression<string> | PlaceholderValue;
    /** Quantity in Demand
     * @default 0
     */
    Qty_in_Demand?: number | Expression<number>;
    /** Quantity in Stock
     * @default 0
     */
    Qty_in_Stock?: number | Expression<number>;
    /** Taxable
     * @default false
     */
    Taxable?: boolean | Expression<boolean>;
    /** Unit Price
     * @default 0
     */
    Unit_Price?: number | Expression<number>;
  };
};

export type ZohoCrmV1ProductUpsertNode = {
  type: 'n8n-nodes-base.zohoCrm';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ZohoCrmV1ProductUpsertParams>;
};
