// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/customerIo/v1/resource_campaign/operation_get_metrics.ts
/**
 * Customer.io Node - Version 1
 * Discriminator: resource=campaign, operation=getMetrics
 */


interface Credentials {
  customerIoApi: CredentialReference;
}

export type CustomerIoV1CampaignGetMetricsParams = {
  resource: 'campaign';
  operation: 'getMetrics';
/**
 * The unique identifier for the campaign
 * @default 0
 */
    campaignId?: number | Expression<number>;
/**
 * Specify metric period
 * @default days
 */
    period?: 'hours' | 'days' | 'weeks' | 'months' | Expression<string>;
/**
 * JSON Parameters
 * @default false
 */
    jsonParameters?: boolean | Expression<boolean>;
/**
 * Additional Fields
 * @displayOptions.show { jsonParameters: [false] }
 * @default {}
 */
    additionalFields?: {
    /** Integer specifying how many steps to return. Defaults to the maximum number of timeperiods available, or 12 when using the months period. Maximum timeperiods available are 24 hours, 45 days, 12 weeks and 120 months
     * @default 0
     */
    steps?: number | Expression<number>;
    /** Specify metric type
     * @default empty
     */
    type?: 'email' | 'empty' | 'push' | 'slack' | 'twilio' | 'urbanAirship' | 'webhook' | Expression<string>;
  };
};

export type CustomerIoV1CampaignGetMetricsNode = {
  type: 'n8n-nodes-base.customerIo';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<CustomerIoV1CampaignGetMetricsParams>;
};
