// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/customerIo/v1/resource_campaign/operation_get_metrics.schema.js
/**
 * Customer.io Node - Version 1 - Zod Schema
 * Discriminator: resource=campaign, operation=getMetrics
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('campaign'),
      operation: z.literal('getMetrics'),
      campaignId: numberOrExpression.optional(),
      period: z.union([z.literal('hours'), z.literal('days'), z.literal('weeks'), z.literal('months'), expressionSchema]).optional(),
      jsonParameters: booleanOrExpression.optional(),
      additionalFields: resolveSchema({ parameters, schema: z.object({ steps: numberOrExpression.optional(), type: z.union([z.literal('email'), z.literal('empty'), z.literal('push'), z.literal('slack'), z.literal('twilio'), z.literal('urbanAirship'), z.literal('webhook'), expressionSchema]).optional() }), required: false, displayOptions: {"show":{"jsonParameters":[false]}}, defaults: {"jsonParameters":false} }),
    }).optional(),
  });
};
