// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/s3/v1/resource_bucket/operation_search.ts
/**
 * S3 Node - Version 1
 * Discriminator: resource=bucket, operation=search
 */


interface Credentials {
  s3: CredentialReference;
}

/** Search within a bucket */
export type S3V1BucketSearchParams = {
  resource: 'bucket';
  operation: 'search';
/**
 * Bucket Name
 */
    bucketName?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** A delimiter is a character you use to group keys
     */
    delimiter?: string | Expression<string> | PlaceholderValue;
    /** Encoding type used by Amazon S3 to encode object keys in the response
     */
    encodingType?: 'url' | Expression<string>;
    /** The owner field is not present in listV2 by default, if you want to return owner field with each key in the result then set the fetch owner field to true
     * @default false
     */
    fetchOwner?: boolean | Expression<boolean>;
    /** Limits the response to keys that begin with the specified prefix
     */
    prefix?: string | Expression<string> | PlaceholderValue;
    /** Whether the requester will pay for requests and data transfer. While Requester Pays is enabled, anonymous access to this bucket is disabled.
     * @default false
     */
    requesterPays?: boolean | Expression<boolean>;
    /** StartAfter is where you want Amazon S3 to start listing from. Amazon S3 starts listing after this specified key.
     */
    startAfter?: string | Expression<string> | PlaceholderValue;
  };
};

export type S3V1BucketSearchOutput = {
  ETag?: string;
  Key?: string;
  LastModified?: string;
  Size?: string;
  StorageClass?: string;
};

export type S3V1BucketSearchNode = {
  type: 'n8n-nodes-base.s3';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<S3V1BucketSearchParams>;
  output?: Items<S3V1BucketSearchOutput>;
};
