// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/s3/v1/resource_bucket/operation_create.schema.js
/**
 * S3 Node - Version 1 - Zod Schema
 * Discriminator: resource=bucket, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('bucket'),
      operation: z.literal('create').default('create'),
      name: stringOrExpression.optional(),
      additionalFields: z.object({ acl: z.union([z.literal('authenticatedRead'), z.literal('Private'), z.literal('publicRead'), z.literal('publicReadWrite'), expressionSchema]).optional(), bucketObjectLockEnabled: booleanOrExpression.optional(), grantFullControl: booleanOrExpression.optional(), grantRead: booleanOrExpression.optional(), grantReadAcp: booleanOrExpression.optional(), grantWrite: booleanOrExpression.optional(), grantWriteAcp: booleanOrExpression.optional(), region: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
