// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/airtable/v2/resource_record/operation_get.ts
/**
 * Airtable Node - Version 2
 * Discriminator: resource=record, operation=get
 */


interface Credentials {
  airtableTokenApi: CredentialReference;
  airtableOAuth2Api: CredentialReference;
}

/** Retrieve a record from a table */
export type AirtableV2RecordGetParams = {
  resource: 'record';
  operation: 'get';
  authentication?: 'airtableTokenApi' | 'airtableOAuth2Api' | Expression<string>;
/**
 * Base
 * @default {"mode":"list","value":""}
 */
    base?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
/**
 * Table
 * @default {"mode":"list","value":""}
 */
    table?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
/**
 * ID of the record to get. &lt;a href="https://support.airtable.com/docs/record-id" target="_blank"&gt;More info&lt;/a&gt;.
 */
    id?: string | Expression<string> | PlaceholderValue;
/**
 * Additional options which decide which records should be returned
 * @default {}
 */
    options?: {
    /** The fields of type 'attachment' that should be downloaded
     * @default []
     */
    downloadFields?: string[];
  };
};

export type AirtableV2RecordGetNode = {
  type: 'n8n-nodes-base.airtable';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<AirtableV2RecordGetParams>;
};
