// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/keap/v1/resource_company/operation_get_all.schema.js
/**
 * Keap Node - Version 1 - Zod Schema
 * Discriminator: resource=company, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('company').default('company'),
      operation: z.literal('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      options: z.object({ companyName: stringOrExpression.optional(), order: z.union([z.literal('datecreated'), z.literal('id'), z.literal('name'), expressionSchema]).optional(), orderDirection: z.union([z.literal('ascending'), z.literal('descending'), expressionSchema]).optional(), fields: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
