// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/keap/v1/resource_company/operation_create.schema.js
/**
 * Keap Node - Version 1 - Zod Schema
 * Discriminator: resource=company, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('company').default('company'),
      operation: z.literal('create').default('create'),
      companyName: stringOrExpression.optional(),
      additionalFields: z.object({ emailAddress: stringOrExpression.optional(), notes: stringOrExpression.optional(), optInReason: stringOrExpression.optional(), website: stringOrExpression.optional() }).optional(),
      addressesUi: z.object({ addressesValues: z.object({ countryCode: stringOrExpression.optional(), line1: stringOrExpression.optional(), line2: stringOrExpression.optional(), locality: stringOrExpression.optional(), postalCode: stringOrExpression.optional(), region: stringOrExpression.optional(), zipCode: stringOrExpression.optional(), zipFour: stringOrExpression.optional() }).optional() }).optional(),
      faxesUi: z.object({ faxesValues: z.object({ type: stringOrExpression.optional(), number: stringOrExpression.optional() }).optional() }).optional(),
      phonesUi: z.object({ phonesValues: z.array(z.object({ type: stringOrExpression.optional(), number: stringOrExpression.optional() })).optional() }).optional(),
    }).optional(),
  });
};
