// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/merge/v32/mode_choose_branch.schema.js
/**
 * Merge Node - Version 3.2 - Zod Schema
 * Discriminator: mode=chooseBranch
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      mode: z.literal('chooseBranch'),
      numberInputs: z.union([z.literal(2), z.literal(3), z.literal(4), z.literal(5), z.literal(6), z.literal(7), z.literal(8), z.literal(9), z.literal(10)]).optional(),
      chooseBranchMode: z.union([z.literal('waitForAll'), expressionSchema]).optional(),
      output: resolveSchema({ parameters, schema: z.union([z.literal('specifiedInput'), z.literal('empty'), expressionSchema]), required: false, displayOptions: {"show":{"chooseBranchMode":["waitForAll"]}}, defaults: {"chooseBranchMode":"waitForAll"} }),
      useDataOfInput: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"output":["specifiedInput"]}}, defaults: {"output":"specifiedInput"} }),
    }).optional(),
  });
};
