// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/merge/v32/mode_combine_by_sql.ts
/**
 * Merge Node - Version 3.2
 * Discriminator: mode=combineBySql
 */


/** Write a query to do the merge */
export type MergeV32CombineBySqlParams = {
  mode: 'combineBySql';
/**
 * The number of data inputs you want to merge. The node waits for all connected inputs to be executed.
 * @default 2
 */
    numberInputs?: 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10;
/**
 * Input data available as tables with corresponding number, e.g. input1, input2
 * @hint Supports &lt;a href="https://github.com/alasql/alasql/wiki/Supported-SQL-statements" target="_blank"&gt;most&lt;/a&gt; of the SQL-99 language
 * @default SELECT * FROM input1 LEFT JOIN input2 ON input1.name = input2.id
 */
    query?: string;
/**
 * Options
 * @default {}
 */
    options?: {
    /** What to return if the query executed successfully but returned no results
     * @default empty
     */
    emptyQueryResult?: 'success' | 'empty' | Expression<string>;
  };
};

export type MergeV32CombineBySqlNode = {
  type: 'n8n-nodes-base.merge';
  version: 3.2;
  config: NodeConfig<MergeV32CombineBySqlParams>;
};
