// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleTranslate/v1/resource_language/operation_translate.schema.js
/**
 * Google Translate Node - Version 1 - Zod Schema
 * Discriminator: resource=language, operation=translate
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('language').default('language'),
      operation: z.literal('translate').default('translate'),
      authentication: z.union([z.literal('serviceAccount'), z.literal('oAuth2'), expressionSchema]).optional(),
      text: stringOrExpression.optional(),
      translateTo: stringOrExpression.optional(),
    }).optional(),
  });
};
