// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/wise/v1/resource_account/operation_get_statement.schema.js
/**
 * Wise Node - Version 1 - Zod Schema
 * Discriminator: resource=account, operation=getStatement
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('account').default('account'),
      operation: z.literal('getStatement'),
      profileId: stringOrExpression.optional(),
      borderlessAccountId: stringOrExpression.optional(),
      currency: stringOrExpression.optional(),
      format: z.union([z.literal('json'), z.literal('csv'), z.literal('pdf'), z.literal('xml'), expressionSchema]).optional(),
      binaryProperty: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"format":["csv","pdf","xml"]}}, defaults: {"format":"json"} }),
      fileName: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"format":["csv","pdf","xml"]}}, defaults: {"format":"json"} }),
      additionalFields: z.object({ lineStyle: z.union([z.literal('COMPACT'), z.literal('FLAT'), expressionSchema]).optional(), range: z.unknown().optional() }).optional(),
    }).optional(),
  });
};
