// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/wise/v1/resource_account/operation_get_balances.ts
/**
 * Wise Node - Version 1
 * Discriminator: resource=account, operation=getBalances
 */


interface Credentials {
  wiseApi: CredentialReference;
}

/** Retrieve balances for all account currencies of this user */
export type WiseV1AccountGetBalancesParams = {
  resource: 'account';
  operation: 'getBalances';
/**
 * ID of the user profile to retrieve the balance of. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 * @default []
 */
    profileId?: string | Expression<string>;
};

export type WiseV1AccountGetBalancesOutput = {
  active?: boolean;
  balances?: Array<{
    amount?: {
      currency?: string;
    };
    balanceType?: string;
    currency?: string;
    id?: number;
    reservedAmount?: {
      currency?: string;
    };
  }>;
  creationTime?: string;
  eligible?: boolean;
  id?: number;
  modificationTime?: string;
  profileId?: number;
  recipientId?: number;
};

export type WiseV1AccountGetBalancesNode = {
  type: 'n8n-nodes-base.wise';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<WiseV1AccountGetBalancesParams>;
  output?: Items<WiseV1AccountGetBalancesOutput>;
};
