// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/wise/v1/resource_account/operation_get_statement.ts
/**
 * Wise Node - Version 1
 * Discriminator: resource=account, operation=getStatement
 */


interface Credentials {
  wiseApi: CredentialReference;
}

/** Retrieve the statement for the borderless account of this user */
export type WiseV1AccountGetStatementParams = {
  resource: 'account';
  operation: 'getStatement';
/**
 * ID of the user profile whose account to retrieve the statement of. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 * @default []
 */
    profileId?: string | Expression<string>;
/**
 * ID of the borderless account to retrieve the statement of. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 * @default []
 */
    borderlessAccountId?: string | Expression<string>;
/**
 * Code of the currency of the borderless account to retrieve the statement of
 */
    currency?: string | Expression<string> | PlaceholderValue;
/**
 * File format to retrieve the statement in
 * @default json
 */
    format?: 'json' | 'csv' | 'pdf' | 'xml' | Expression<string>;
/**
 * Put Output File in Field
 * @hint The name of the output binary field to put the file in
 * @displayOptions.show { format: ["csv", "pdf", "xml"] }
 * @default data
 */
    binaryProperty?: string | Expression<string> | PlaceholderValue;
/**
 * Name of the file that will be downloaded
 * @displayOptions.show { format: ["csv", "pdf", "xml"] }
 */
    fileName?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Line style to retrieve the statement in
     * @default COMPACT
     */
    lineStyle?: 'COMPACT' | 'FLAT' | Expression<string>;
    /** Range
     * @default {}
     */
    range?: {
        /** Range Properties
     */
    rangeProperties?: {
      /** Range Start
       */
      intervalStart?: string | Expression<string>;
      /** Range End
       */
      intervalEnd?: string | Expression<string>;
    };
  };
  };
};

export type WiseV1AccountGetStatementNode = {
  type: 'n8n-nodes-base.wise';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<WiseV1AccountGetStatementParams>;
};
