// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/actionNetwork/v1/resource_attendance/operation_get_all.ts
/**
 * Action Network Node - Version 1
 * Discriminator: resource=attendance, operation=getAll
 */


interface Credentials {
  actionNetworkApi: CredentialReference;
}

export type ActionNetworkV1AttendanceGetAllParams = {
  resource: 'attendance';
  operation: 'getAll';
/**
 * ID of the event to create an attendance for
 */
    eventId?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Whether to return a simplified version of the response instead of the raw data
 * @default true
 */
    simple?: boolean | Expression<boolean>;
};

export type ActionNetworkV1AttendanceGetAllNode = {
  type: 'n8n-nodes-base.actionNetwork';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ActionNetworkV1AttendanceGetAllParams>;
};
