// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mondayCom/v1/resource_board_column/operation_create.ts
/**
 * Monday.com Node - Version 1
 * Discriminator: resource=boardColumn, operation=create
 */


interface Credentials {
  mondayComApi: CredentialReference;
  mondayComOAuth2Api: CredentialReference;
}

/** Create a new board */
export type MondayComV1BoardColumnCreateParams = {
  resource: 'boardColumn';
  operation: 'create';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    boardId?: string | Expression<string>;
/**
 * Title
 */
    title?: string | Expression<string> | PlaceholderValue;
/**
 * Column Type
 */
    columnType?: 'checkbox' | 'country' | 'date' | 'dropdown' | 'email' | 'hour' | 'Link' | 'longText' | 'numbers' | 'people' | 'person' | 'phone' | 'rating' | 'status' | 'tags' | 'team' | 'text' | 'timeline' | 'timezone' | 'week' | 'worldClock' | Expression<string>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** The new column's defaults
     */
    defaults?: IDataObject | string | Expression<string>;
  };
};

export type MondayComV1BoardColumnCreateOutput = {
  id?: string;
};

export type MondayComV1BoardColumnCreateNode = {
  type: 'n8n-nodes-base.mondayCom';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MondayComV1BoardColumnCreateParams>;
  output?: Items<MondayComV1BoardColumnCreateOutput>;
};
