// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mondayCom/v1/resource_board_column/operation_create.schema.js
/**
 * Monday.com Node - Version 1 - Zod Schema
 * Discriminator: resource=boardColumn, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('boardColumn'),
      operation: z.literal('create').default('create'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      boardId: stringOrExpression.optional(),
      title: stringOrExpression.optional(),
      columnType: z.union([z.literal('checkbox'), z.literal('country'), z.literal('date'), z.literal('dropdown'), z.literal('email'), z.literal('hour'), z.literal('Link'), z.literal('longText'), z.literal('numbers'), z.literal('people'), z.literal('person'), z.literal('phone'), z.literal('rating'), z.literal('status'), z.literal('tags'), z.literal('team'), z.literal('text'), z.literal('timeline'), z.literal('timezone'), z.literal('week'), z.literal('worldClock'), expressionSchema]).optional(),
      additionalFields: z.object({ defaults: z.union([iDataObjectSchema, z.string()]).optional() }).optional(),
    }).optional(),
  });
};
