// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mocean/v1/resource_voice/operation_send.schema.js
/**
 * Mocean Node - Version 1 - Zod Schema
 * Discriminator: resource=voice, operation=send
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('voice'),
      operation: z.literal('send').default('send'),
      from: stringOrExpression.optional(),
      to: stringOrExpression.optional(),
      language: z.union([z.literal('cmn-CN'), z.literal('en-GB'), z.literal('en-US'), z.literal('ja-JP'), z.literal('ko-KR'), expressionSchema]).optional(),
      message: stringOrExpression.optional(),
    }).optional(),
  });
};
