// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mailjet/v1/resource_email/operation_send.schema.js
/**
 * Mailjet Node - Version 1 - Zod Schema
 * Discriminator: resource=email, operation=send
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('email').default('email'),
      operation: z.literal('send').default('send'),
      fromEmail: stringOrExpression.optional(),
      toEmail: stringOrExpression.optional(),
      subject: stringOrExpression.optional(),
      text: stringOrExpression.optional(),
      html: stringOrExpression.optional(),
      jsonParameters: booleanOrExpression.optional(),
      additionalFields: z.object({ bccEmail: stringOrExpression.optional(), ccAddresses: stringOrExpression.optional(), fromName: stringOrExpression.optional(), priority: numberOrExpression.optional(), replyTo: stringOrExpression.optional(), templateLanguage: booleanOrExpression.optional(), trackClicks: z.union([z.literal('account_default'), z.literal('disabled'), z.literal('enabled'), expressionSchema]).optional(), trackOpens: z.union([z.literal('account_default'), z.literal('disabled'), z.literal('enabled'), expressionSchema]).optional(), customCampaign: stringOrExpression.optional(), deduplicateCampaign: booleanOrExpression.optional() }).optional(),
      variablesJson: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"jsonParameters":[true]}}, defaults: {"jsonParameters":false} }),
      variablesUi: resolveSchema({ parameters, schema: z.object({ variablesValues: z.array(z.object({ name: stringOrExpression.optional(), value: stringOrExpression.optional() })).optional() }), required: false, displayOptions: {"show":{"jsonParameters":[false]}}, defaults: {"jsonParameters":false} }),
    }).optional(),
  });
};
