// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/phantombuster/v1/resource_agent/operation_get.ts
/**
 * Phantombuster Node - Version 1
 * Discriminator: resource=agent, operation=get
 */


interface Credentials {
  phantombusterApi: CredentialReference;
}

/** Get an agent by ID */
export type PhantombusterV1AgentGetParams = {
  resource: 'agent';
  operation: 'get';
/**
 * Agent ID
 */
    agentId?: string | Expression<string> | PlaceholderValue;
};

export type PhantombusterV1AgentGetOutput = {
  argument?: string;
  branch?: string;
  code?: null;
  environment?: string;
  fileMgmt?: string;
  id?: string;
  lastEndType?: string;
  launchType?: string;
  maxParallelism?: number;
  name?: string;
  nbLaunches?: number;
  notifications?: {
    mailAutomaticExitError?: boolean;
    mailAutomaticExitSuccess?: boolean;
    mailAutomaticLaunchError?: boolean;
    mailAutomaticTimeError?: boolean;
    mailManualExitError?: boolean;
    mailManualExitSuccess?: boolean;
    mailManualLaunchError?: boolean;
    mailManualTimeError?: boolean;
  };
  orgS3Folder?: string;
  s3Folder?: string;
  script?: string;
  scriptId?: string;
  scriptOrgName?: string;
  updatedAt?: number;
  wasSetupValidWhenSubmittedByTheFrontend?: boolean;
};

export type PhantombusterV1AgentGetNode = {
  type: 'n8n-nodes-base.phantombuster';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PhantombusterV1AgentGetParams>;
  output?: Items<PhantombusterV1AgentGetOutput>;
};
