// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/phantombuster/v1/resource_agent/operation_launch.ts
/**
 * Phantombuster Node - Version 1
 * Discriminator: resource=agent, operation=launch
 */


interface Credentials {
  phantombusterApi: CredentialReference;
}

/** Add an agent to the launch queue */
export type PhantombusterV1AgentLaunchParams = {
  resource: 'agent';
  operation: 'launch';
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    agentId?: string | Expression<string>;
/**
 * By default the launch just include the container ID. If this option gets activated, it will resolve the data automatically.
 * @default true
 */
    resolveData?: boolean | Expression<boolean>;
/**
 * JSON Parameters
 * @default false
 */
    jsonParameters?: boolean | Expression<boolean>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Agent argument. Can either be a JSON string or a plain object. The argument can be retrieved with buster.argument in the agent’s script.
     * @displayOptions.show { /jsonParameters: [true] }
     */
    argumentsJson?: IDataObject | string | Expression<string>;
    /** Arguments
     * @displayOptions.show { /jsonParameters: [false] }
     * @default {}
     */
    argumentsUi?: {
        /** Argument
     */
    argumentValues?: Array<{
      /** Name of the argument key to add
       */
      key?: string | Expression<string> | PlaceholderValue;
      /** Value to set for the argument key
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
    /** Bonus Argument
     * @displayOptions.show { /jsonParameters: [false] }
     * @default {}
     */
    bonusArgumentUi?: {
        /** Bonus Argument
     */
    bonusArgumentValue?: Array<{
      /** Name of the argument key to add
       */
      key?: string | Expression<string> | PlaceholderValue;
      /** Value to set for the argument key
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
    /** Agent bonus argument. Can either be a JSON string or a plain object. This bonus argument is single-use, it will only be used for the current launch. If present, it will be merged with the original argument, resulting in an effective argument that can be retrieved with buster.argument in the agent’s script.
     * @displayOptions.show { /jsonParameters: [true] }
     */
    bonusArgumentJson?: string | Expression<string> | PlaceholderValue;
    /** Whether the agent will be considered as "launched manually"
     * @default false
     */
    manualLaunch?: boolean | Expression<boolean>;
    /** If set, the agent will only be launched if the number of already running instances is below the specified number
     * @default 0
     */
    maxInstanceCount?: number | Expression<number>;
    /** If true, argument will be saved as the default launch options for the agent
     */
    saveArgument?: string | Expression<string> | PlaceholderValue;
  };
};

export type PhantombusterV1AgentLaunchOutput = {
  createdAt?: number;
  id?: string;
  launchedAt?: number;
  launchType?: string;
  retryNumber?: number;
  status?: string;
};

export type PhantombusterV1AgentLaunchNode = {
  type: 'n8n-nodes-base.phantombuster';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PhantombusterV1AgentLaunchParams>;
  output?: Items<PhantombusterV1AgentLaunchOutput>;
};
