// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/phantombuster/v1/resource_agent/operation_get_output.schema.js
/**
 * Phantombuster Node - Version 1 - Zod Schema
 * Discriminator: resource=agent, operation=getOutput
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('agent').default('agent'),
      operation: z.literal('getOutput'),
      agentId: stringOrExpression.optional(),
      resolveData: booleanOrExpression.optional(),
      additionalFields: z.object({ prevContainerId: stringOrExpression.optional(), prevStatus: z.union([z.literal('finished'), z.literal('lauch error'), z.literal('never launched'), z.literal('running'), z.literal('starting'), z.literal('unknown'), expressionSchema]).optional(), prevRuntimeEventIndex: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
