// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/phantombuster/v1/resource_agent/operation_get_output.ts
/**
 * Phantombuster Node - Version 1
 * Discriminator: resource=agent, operation=getOutput
 */


interface Credentials {
  phantombusterApi: CredentialReference;
}

/** Get the output of the most recent container of an agent */
export type PhantombusterV1AgentGetOutputParams = {
  resource: 'agent';
  operation: 'getOutput';
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    agentId?: string | Expression<string>;
/**
 * By default the outpout is presented as string. If this option gets activated, it will resolve the data automatically.
 * @default true
 */
    resolveData?: boolean | Expression<boolean>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** If set, the output will be retrieved from the container after the specified previous container ID
     */
    prevContainerId?: string | Expression<string> | PlaceholderValue;
    /** If set, allows to define which status was previously retrieved on user-side
     */
    prevStatus?: 'finished' | 'lauch error' | 'never launched' | 'running' | 'starting' | 'unknown' | Expression<string>;
    /** If set, the container's runtime events will be returned in the response starting from the provided previous runtime event index
     * @default 0
     */
    prevRuntimeEventIndex?: number | Expression<number>;
  };
};

export type PhantombusterV1AgentGetOutputOutput = {
  profileUrl?: string;
  timestamp?: string;
};

export type PhantombusterV1AgentGetOutputNode = {
  type: 'n8n-nodes-base.phantombuster';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PhantombusterV1AgentGetOutputParams>;
  output?: Items<PhantombusterV1AgentGetOutputOutput>;
};
