// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/paddle/v1/resource_user/operation_get_all.ts
/**
 * Paddle Node - Version 1
 * Discriminator: resource=user, operation=getAll
 */


interface Credentials {
  paddleApi: CredentialReference;
}

/** Get many coupons */
export type PaddleV1UserGetAllParams = {
  resource: 'user';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * JSON Parameters
 * @default false
 */
    jsonParameters?: boolean | Expression<boolean>;
/**
 * Attributes in JSON form
 * @displayOptions.show { jsonParameters: [true] }
 */
    additionalFieldsJson?: IDataObject | string | Expression<string>;
/**
 * Additional Fields
 * @displayOptions.show { jsonParameters: [false] }
 * @default {}
 */
    additionalFields?: {
    /** Filter: The subscription plan ID
     */
    planId?: string | Expression<string> | PlaceholderValue;
    /** A specific user subscription ID
     */
    subscriptionId?: string | Expression<string> | PlaceholderValue;
    /** Filter: The user subscription status. Returns all active, past_due, trialing and paused subscription plans if not specified.
     * @default active
     */
    state?: 'active' | 'past_due' | 'paused' | 'trialing' | Expression<string>;
  };
};

export type PaddleV1UserGetAllNode = {
  type: 'n8n-nodes-base.paddle';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PaddleV1UserGetAllParams>;
};
