// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/hubspot/v22/resource_deal/operation_get_recently_created_updated.ts
/**
 * HubSpot Node - Version 2.2
 * Discriminator: resource=deal, operation=getRecentlyCreatedUpdated
 */


interface Credentials {
  hubspotApi: CredentialReference;
  hubspotAppToken: CredentialReference;
  hubspotOAuth2Api: CredentialReference;
}

/** Get recently created/updated contacts */
export type HubspotV22DealGetRecentlyCreatedUpdatedParams = {
  resource: 'deal';
  operation: 'getRecentlyCreatedUpdated';
  authentication?: 'apiKey' | 'appToken' | 'oAuth2' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Options
 * @default {}
 */
    filters?: {
    /** Only return deals created after timestamp x. When using expressions, the time should be specified in YYYY-MM-DD hh-mm-ss format.
     */
    since?: string | Expression<string>;
    /** By default, you will only get data for the most recent version of a property in the "versions" data. If you include this parameter, you will get data for all previous versions.
     * @default false
     */
    includePropertyVersions?: boolean | Expression<boolean>;
  };
};

export type HubspotV22DealGetRecentlyCreatedUpdatedNode = {
  type: 'n8n-nodes-base.hubspot';
  version: 2.2;
  credentials?: Credentials;
  config: NodeConfig<HubspotV22DealGetRecentlyCreatedUpdatedParams>;
};
