// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftTeams/v11/resource_chat_message/operation_create.ts
/**
 * Microsoft Teams Node - Version 1.1
 * Discriminator: resource=chatMessage, operation=create
 */


interface Credentials {
  microsoftTeamsOAuth2Api: CredentialReference;
}

/** Create a channel */
export type MicrosoftTeamsV11ChatMessageCreateParams = {
  resource: 'chatMessage';
  operation: 'create';
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    chatId?: string | Expression<string>;
/**
 * The type of the content
 * @default text
 */
    messageType?: 'text' | 'html' | Expression<string>;
/**
 * The content of the item
 */
    message?: string | Expression<string> | PlaceholderValue;
/**
 * Other options to set
 * @default {}
 */
    options?: {
    /** Whether to append a link to this workflow at the end of the message. This is helpful if you have many workflows sending messages.
     * @default true
     */
    includeLinkToWorkflow?: boolean | Expression<boolean>;
  };
};

export type MicrosoftTeamsV11ChatMessageCreateNode = {
  type: 'n8n-nodes-base.microsoftTeams';
  version: 1.1;
  credentials?: Credentials;
  config: NodeConfig<MicrosoftTeamsV11ChatMessageCreateParams>;
};
