// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/seaTable/v2/resource_row/operation_create.ts
/**
 * SeaTable Node - Version 2
 * Discriminator: resource=row, operation=create
 */


interface Credentials {
  seaTableApi: CredentialReference;
}

/** Create a new row */
export type SeaTableV2RowCreateParams = {
  resource: 'row';
  operation: 'create';
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    tableName?: string | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    rowId?: string | Expression<string>;
/**
 * Whether to insert the input data this node receives in the new row
 * @default defineBelow
 */
    fieldsToSend?: 'autoMapInputData' | 'defineBelow' | Expression<string>;
/**
 * Whether to use the column default values to populate new rows during creation (only available for normal backend)
 * @displayOptions.show { bigdata: [false] }
 * @default false
 */
    apply_default?: boolean | Expression<boolean>;
/**
 * List of input properties to avoid sending, separated by commas. Leave empty to send all properties.
 * @displayOptions.show { /fieldsToSend: ["autoMapInputData"] }
 */
    inputsToIgnore?: string | Expression<string> | PlaceholderValue;
/**
 * Add destination column with its value. Provide the value in this way. Date: YYYY-MM-DD or YYYY-MM-DD hh:mm. Duration: time in seconds. Checkbox: true, on or 1. Multi-Select: comma-separated list.
 * @displayOptions.show { /fieldsToSend: ["defineBelow"] }
 * @default {}
 */
    columnsUi?: {
        /** Column
     */
    columnValues?: Array<{
      /** Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
       */
      columnName?: string | Expression<string>;
      /** Column Value
       */
      columnValue?: string | Expression<string> | PlaceholderValue;
    }>;
  };
/**
 * Whether write to Big Data backend (true) or not (false). True requires the activation of the Big Data backend in the base.
 * @default false
 */
    bigdata?: boolean | Expression<boolean>;
};

export type SeaTableV2RowCreateOutput = {
  _creator?: string;
  _ctime?: string;
  _id?: string;
  _last_modifier?: string;
  _mtime?: string;
};

export type SeaTableV2RowCreateNode = {
  type: 'n8n-nodes-base.seaTable';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<SeaTableV2RowCreateParams>;
  output?: Items<SeaTableV2RowCreateOutput>;
};
