// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/seaTable/v2/resource_row/operation_update.schema.js
/**
 * SeaTable Node - Version 2 - Zod Schema
 * Discriminator: resource=row, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('row').default('row'),
      operation: z.literal('update'),
      tableName: stringOrExpression.optional(),
      rowId: stringOrExpression.optional(),
      fieldsToSend: z.union([z.literal('autoMapInputData'), z.literal('defineBelow'), expressionSchema]).optional(),
      inputsToIgnore: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"fieldsToSend":["autoMapInputData"]}}, defaults: {"fieldsToSend":"defineBelow"} }),
      columnsUi: resolveSchema({ parameters, schema: z.object({ columnValues: z.array(z.object({ columnName: stringOrExpression.optional(), columnValue: stringOrExpression.optional() })).optional() }), required: false, displayOptions: {"show":{"fieldsToSend":["defineBelow"]}}, defaults: {"fieldsToSend":"defineBelow"} }),
    }).optional(),
  });
};
