// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/seaTable/v2/resource_row/operation_search.schema.js
/**
 * SeaTable Node - Version 2 - Zod Schema
 * Discriminator: resource=row, operation=search
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('row').default('row'),
      operation: z.literal('search'),
      tableName: stringOrExpression.optional(),
      rowId: stringOrExpression.optional(),
      searchColumn: stringOrExpression.optional(),
      searchTerm: stringOrExpression.optional(),
      options: z.object({ insensitive: booleanOrExpression.optional(), wildcard: booleanOrExpression.optional(), simple: booleanOrExpression.optional(), convert: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
