// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/seaTable/v2/resource_row/operation_list.ts
/**
 * SeaTable Node - Version 2
 * Discriminator: resource=row, operation=list
 */


interface Credentials {
  seaTableApi: CredentialReference;
}

/** Get many rows from a table or a table view */
export type SeaTableV2RowListParams = {
  resource: 'row';
  operation: 'list';
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    tableName?: string | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    rowId?: string | Expression<string>;
/**
 * The name of SeaTable view to access, or specify by using an expression. Provide it in the way "col.name:::col.type".
 */
    viewName?: string | Expression<string>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Whether to return a simplified version of the response instead of the raw data
     * @default true
     */
    simple?: boolean | Expression<boolean>;
    /** Whether to return the column keys (false) or the column names (true)
     * @default true
     */
    convert?: boolean | Expression<boolean>;
  };
};

export type SeaTableV2RowListOutput = {
  _archived?: boolean;
  _creator?: string;
  _ctime?: string;
  _id?: string;
  _last_modifier?: string;
  _locked?: null;
  _locked_by?: null;
  _mtime?: string;
};

export type SeaTableV2RowListNode = {
  type: 'n8n-nodes-base.seaTable';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<SeaTableV2RowListParams>;
  output?: Items<SeaTableV2RowListOutput>;
};
