// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/seaTable/v2/resource_row/operation_search.ts
/**
 * SeaTable Node - Version 2
 * Discriminator: resource=row, operation=search
 */


interface Credentials {
  seaTableApi: CredentialReference;
}

/** Search one or multiple rows */
export type SeaTableV2RowSearchParams = {
  resource: 'row';
  operation: 'search';
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    tableName?: string | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    rowId?: string | Expression<string>;
/**
 * Select the column to be searched. Not all column types are supported for search. Choose from the list, or specify a name using an &lt;a href="https://docs.n8n.io/code-examples/expressions/"&gt;expression&lt;/a&gt;.
 */
    searchColumn?: string | Expression<string>;
/**
 * What to look for?
 */
    searchTerm?: string | Expression<string> | PlaceholderValue;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Whether the search ignores case sensitivity (true). Otherwise, it distinguishes between uppercase and lowercase characters.
     * @default false
     */
    insensitive?: boolean | Expression<boolean>;
    /** Whether the search only results perfect matches (true). Otherwise, it finds a row even if the search value is part of a string (false).
     * @default true
     */
    wildcard?: boolean | Expression<boolean>;
    /** Whether to return a simplified version of the response instead of the raw data
     * @default true
     */
    simple?: boolean | Expression<boolean>;
    /** Whether to return the column keys (false) or the column names (true)
     * @default true
     */
    convert?: boolean | Expression<boolean>;
  };
};

export type SeaTableV2RowSearchNode = {
  type: 'n8n-nodes-base.seaTable';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<SeaTableV2RowSearchParams>;
};
