// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/clickUp/v1/resource_task_tag/operation_remove.schema.js
/**
 * ClickUp Node - Version 1 - Zod Schema
 * Discriminator: resource=taskTag, operation=remove
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('taskTag'),
      operation: z.literal('remove'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      taskId: stringOrExpression.optional(),
      tagName: stringOrExpression.optional(),
      additionalFields: z.object({ custom_task_ids: booleanOrExpression.optional(), team_id: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
