// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/clickUp/v1/resource_task_tag/operation_add.ts
/**
 * ClickUp Node - Version 1
 * Discriminator: resource=taskTag, operation=add
 */


interface Credentials {
  clickUpApi: CredentialReference;
  clickUpOAuth2Api: CredentialReference;
}

/** Add a tag to a task */
export type ClickUpV1TaskTagAddParams = {
  resource: 'taskTag';
  operation: 'add';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Task ID
 */
    taskId?: string | Expression<string> | PlaceholderValue;
/**
 * Tag Name
 */
    tagName?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Whether to reference a task by it's custom task ID
     * @default false
     */
    custom_task_ids?: boolean | Expression<boolean>;
    /** Only used when the parameter is set to custom_task_ids=true. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    team_id?: string | Expression<string>;
  };
};

export type ClickUpV1TaskTagAddOutput = {
  success?: boolean;
};

export type ClickUpV1TaskTagAddNode = {
  type: 'n8n-nodes-base.clickUp';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ClickUpV1TaskTagAddParams>;
  output?: Items<ClickUpV1TaskTagAddOutput>;
};
