// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/highLevel/v1/resource_task/operation_create.schema.js
/**
 * HighLevel Node - Version 1 - Zod Schema
 * Discriminator: resource=task, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('task'),
      operation: z.literal('create').default('create'),
      contactId: stringOrExpression.optional(),
      title: stringOrExpression.optional(),
      dueDate: stringOrExpression.optional(),
      additionalFields: z.object({ assignedTo: stringOrExpression.optional(), description: stringOrExpression.optional(), status: z.union([z.literal('incompleted'), z.literal('completed'), expressionSchema]).optional() }).optional(),
      requestOptions: z.object({ batching: z.unknown().optional(), allowUnauthorizedCerts: z.boolean().optional(), proxy: stringOrExpression.optional(), timeout: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
