// @n8n/db/dist/migrations/common/1763047800000-AddActiveVersionIdColumn.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddActiveVersionIdColumn1763047800000 = void 0;
const WORKFLOWS_TABLE_NAME = 'workflow_entity';
const WORKFLOW_HISTORY_TABLE_NAME = 'workflow_history';
class AddActiveVersionIdColumn1763047800000 {
    async up({ schemaBuilder: { addColumns, column, addForeignKey }, queryRunner, escape, runQuery, }) {
        const workflowsTableName = escape.tableName(WORKFLOWS_TABLE_NAME);
        await addColumns(WORKFLOWS_TABLE_NAME, [column('activeVersionId').varchar(36)]);
        await addForeignKey(WORKFLOWS_TABLE_NAME, 'activeVersionId', [WORKFLOW_HISTORY_TABLE_NAME, 'versionId'], undefined, 'RESTRICT');
        await this.backFillHistoryRecords(runQuery, escape);
        const versionIdColumn = escape.columnName('versionId');
        const activeColumn = escape.columnName('active');
        const activeVersionIdColumn = escape.columnName('activeVersionId');
        await queryRunner.query(`UPDATE ${workflowsTableName}
			 SET ${activeVersionIdColumn} = ${versionIdColumn}
			 WHERE ${activeColumn} = true`);
    }
    async down({ schemaBuilder: { dropColumns, dropForeignKey } }) {
        await dropForeignKey(WORKFLOWS_TABLE_NAME, 'activeVersionId', [
            WORKFLOW_HISTORY_TABLE_NAME,
            'versionId',
        ]);
        await dropColumns(WORKFLOWS_TABLE_NAME, ['activeVersionId']);
    }
    async backFillHistoryRecords(runQuery, escape) {
        const workflowTable = escape.tableName('workflow_entity');
        const historyTable = escape.tableName('workflow_history');
        const versionIdColumn = escape.columnName('versionId');
        const idColumn = escape.columnName('id');
        const workflowIdColumn = escape.columnName('workflowId');
        const nodesColumn = escape.columnName('nodes');
        const connectionsColumn = escape.columnName('connections');
        const authorsColumn = escape.columnName('authors');
        const createdAtColumn = escape.columnName('createdAt');
        const updatedAtColumn = escape.columnName('updatedAt');
        await runQuery(`
			INSERT INTO ${historyTable} (
				${versionIdColumn},
				${workflowIdColumn},
				${authorsColumn},
				${nodesColumn},
				${connectionsColumn},
				${createdAtColumn},
				${updatedAtColumn}
			)
			SELECT
				w.${versionIdColumn},
				w.${idColumn},
				:authors,
				w.${nodesColumn},
				w.${connectionsColumn},
				:createdAt,
				:updatedAt
			FROM ${workflowTable} w
	    	LEFT JOIN ${historyTable} wh
				ON w.${versionIdColumn} = wh.${versionIdColumn}
			WHERE wh.${versionIdColumn} IS NULL
			`, {
            authors: 'system migration',
            createdAt: new Date(),
            updatedAt: new Date(),
        });
    }
}
exports.AddActiveVersionIdColumn1763047800000 = AddActiveVersionIdColumn1763047800000;
