// @n8n/db/dist/migrations/common/1769900001000-AddWorkflowUnpublishScopeToCustomRoles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddWorkflowUnpublishScopeToCustomRoles1769900001000 = void 0;
const PERSONAL_OWNER_ROLE_SLUG = 'project:personalOwner';
function isMySQLOrMariaDB(dbType) {
    return dbType === 'mysqldb' || dbType === 'mariadb';
}
class AddWorkflowUnpublishScopeToCustomRoles1769900001000 {
    async up({ escape, runQuery, dbType }) {
        const scopeTableName = escape.tableName('scope');
        const scopeSlugColumn = escape.columnName('slug');
        const displayNameColumn = escape.columnName('displayName');
        const descriptionColumn = escape.columnName('description');
        const roleTableName = escape.tableName('role');
        const roleScopeTableName = escape.tableName('role_scope');
        const roleSlugColumn = escape.columnName('slug');
        const roleScopeRoleSlugColumn = escape.columnName('roleSlug');
        const roleScopeScopeSlugColumn = escape.columnName('scopeSlug');
        const dbTypeStr = dbType;
        const useInsertIgnore = isMySQLOrMariaDB(dbTypeStr);
        const insertScopeQuery = useInsertIgnore
            ? `INSERT IGNORE INTO ${scopeTableName} (${scopeSlugColumn}, ${displayNameColumn}, ${descriptionColumn})
         VALUES (:slug, :displayName, :description)`
            : `INSERT INTO ${scopeTableName} (${scopeSlugColumn}, ${displayNameColumn}, ${descriptionColumn})
         VALUES (:slug, :displayName, :description)
         ON CONFLICT (${scopeSlugColumn}) DO NOTHING`;
        await runQuery(insertScopeQuery, {
            slug: 'workflow:unpublish',
            displayName: 'Unpublish Workflow',
            description: 'Allows unpublishing workflows.',
        });
        const batchInsertBase = `
		INSERT ${useInsertIgnore ? 'IGNORE ' : ''}INTO ${roleScopeTableName} (${roleScopeRoleSlugColumn}, ${roleScopeScopeSlugColumn})
		SELECT DISTINCT role.${roleSlugColumn}, :unpublishScope
		FROM ${roleTableName} role
		INNER JOIN ${roleScopeTableName} role_scope
			ON role.${roleSlugColumn} = role_scope.${roleScopeRoleSlugColumn}
		WHERE role.${roleSlugColumn} != :personalOwnerSlug
			AND role_scope.${roleScopeScopeSlugColumn} = :publishScope
		`;
        const batchInsertQuery = useInsertIgnore
            ? batchInsertBase
            : `${batchInsertBase}
		ON CONFLICT (${roleScopeRoleSlugColumn}, ${roleScopeScopeSlugColumn}) DO NOTHING
		`;
        await runQuery(batchInsertQuery, {
            personalOwnerSlug: PERSONAL_OWNER_ROLE_SLUG,
            publishScope: 'workflow:publish',
            unpublishScope: 'workflow:unpublish',
        });
    }
    async down({ escape, runQuery }) {
        const roleScopeTableName = escape.tableName('role_scope');
        const roleScopeScopeSlugColumn = escape.columnName('scopeSlug');
        const deleteQuery = `
			DELETE FROM ${roleScopeTableName}
			WHERE ${roleScopeScopeSlugColumn} = :unpublishScope
		`;
        await runQuery(deleteQuery, {
            unpublishScope: 'workflow:unpublish',
            personalOwnerSlug: PERSONAL_OWNER_ROLE_SLUG,
        });
    }
}
exports.AddWorkflowUnpublishScopeToCustomRoles1769900001000 = AddWorkflowUnpublishScopeToCustomRoles1769900001000;
