// @n8n/db/dist/migrations/common/1770000000000-CreateChatHubToolsTable.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateChatHubToolsTable1770000000000 = void 0;
const node_crypto_1 = require("node:crypto");
const table = {
    tools: 'chat_hub_tools',
    sessions: 'chat_hub_sessions',
    agents: 'chat_hub_agents',
    sessionTools: 'chat_hub_session_tools',
    agentTools: 'chat_hub_agent_tools',
    user: 'user',
};
class CreateChatHubToolsTable1770000000000 {
    async up({ schemaBuilder: { createTable, column, dropColumns }, escape, isPostgres, runQuery, runInBatches, parseJson, logger, migrationName, }) {
        await createTable(table.tools)
            .withColumns(column('id').uuid.primary, column('name').varchar(255).notNull, column('type').varchar(255).notNull, column('typeVersion').double.notNull, column('ownerId').uuid.notNull, column('definition').json.notNull, column('enabled').bool.notNull.default(true))
            .withForeignKey('ownerId', {
            tableName: table.user,
            columnName: 'id',
            onDelete: 'CASCADE',
        })
            .withIndexOn(['ownerId', 'name'], true).withTimestamps;
        await createTable(table.sessionTools)
            .withColumns(column('sessionId').uuid.notNull.primary, column('toolId').uuid.notNull.primary)
            .withForeignKey('sessionId', {
            tableName: table.sessions,
            columnName: 'id',
            onDelete: 'CASCADE',
        })
            .withForeignKey('toolId', {
            tableName: table.tools,
            columnName: 'id',
            onDelete: 'CASCADE',
        });
        await createTable(table.agentTools)
            .withColumns(column('agentId').uuid.notNull.primary, column('toolId').uuid.notNull.primary)
            .withForeignKey('agentId', {
            tableName: table.agents,
            columnName: 'id',
            onDelete: 'CASCADE',
        })
            .withForeignKey('toolId', {
            tableName: table.tools,
            columnName: 'id',
            onDelete: 'CASCADE',
        });
        const toolsByUserAndName = new Map();
        const sessionsTable = escape.tableName(table.sessions);
        const agentsTable = escape.tableName(table.agents);
        const toolsTable = escape.tableName(table.tools);
        const sessionToolsTable = escape.tableName(table.sessionTools);
        const agentToolsTable = escape.tableName(table.agentTools);
        const toolsFilter = isPostgres ? '"tools"::text != \'[]\'' : '"tools" != \'[]\'';
        async function ensureTool(ownerId, def) {
            const key = `${ownerId}::${def.name}`;
            const existing = toolsByUserAndName.get(key);
            if (existing)
                return existing;
            const toolId = (0, node_crypto_1.randomUUID)();
            await runQuery(`INSERT INTO ${toolsTable} ("id", "name", "type", "typeVersion", "ownerId", "definition", "enabled")
				 VALUES (:id, :name, :type, :typeVersion, :ownerId, :definition, :enabled)`, {
                id: toolId,
                name: def.name,
                type: def.type,
                typeVersion: def.typeVersion,
                ownerId,
                definition: JSON.stringify({ ...def, id: toolId }),
                enabled: true,
            });
            toolsByUserAndName.set(key, toolId);
            return toolId;
        }
        function isValidTool(tool) {
            return Boolean(tool.id && tool.name && tool.type && typeof tool.typeVersion === 'number');
        }
        function safeParseTools(raw, entityId, entityType) {
            try {
                const tools = parseJson(raw);
                if (!Array.isArray(tools)) {
                    logger.warn(`[${migrationName}] Tools column for ${entityType} ${entityId} is not an array. Skipping.`);
                    return [];
                }
                return tools;
            }
            catch (error) {
                logger.warn(`[${migrationName}] Failed to parse tools for ${entityType} ${entityId}: ${error instanceof Error ? error.message : 'Unknown error'}. Skipping.`);
                return [];
            }
        }
        await runInBatches(`SELECT "id", "ownerId", "tools" FROM ${sessionsTable} WHERE ${toolsFilter}`, async (sessions) => {
            for (const session of sessions) {
                const tools = safeParseTools(session.tools, session.id, 'session');
                const insertedToolIds = new Set();
                for (const tool of tools) {
                    if (!isValidTool(tool))
                        continue;
                    const toolId = await ensureTool(session.ownerId, tool);
                    if (insertedToolIds.has(toolId))
                        continue;
                    insertedToolIds.add(toolId);
                    await runQuery(`INSERT INTO ${sessionToolsTable} ("sessionId", "toolId") VALUES (:sessionId, :toolId)`, { sessionId: session.id, toolId });
                }
            }
        });
        await runInBatches(`SELECT "id", "ownerId", "tools" FROM ${agentsTable} WHERE ${toolsFilter}`, async (agents) => {
            for (const agent of agents) {
                const tools = safeParseTools(agent.tools, agent.id, 'agent');
                const insertedToolIds = new Set();
                for (const tool of tools) {
                    if (!isValidTool(tool))
                        continue;
                    const toolId = await ensureTool(agent.ownerId, tool);
                    if (insertedToolIds.has(toolId))
                        continue;
                    insertedToolIds.add(toolId);
                    await runQuery(`INSERT INTO ${agentToolsTable} ("agentId", "toolId") VALUES (:agentId, :toolId)`, { agentId: agent.id, toolId });
                }
            }
        });
        await dropColumns(table.sessions, ['tools']);
        await dropColumns(table.agents, ['tools']);
    }
    async down({ schemaBuilder: { addColumns, column, dropTable } }) {
        await dropTable(table.sessionTools);
        await dropTable(table.agentTools);
        await dropTable(table.tools);
        await addColumns(table.sessions, [column('tools').json.notNull.default("'[]'")]);
        await addColumns(table.agents, [column('tools').json.notNull.default("'[]'")]);
    }
}
exports.CreateChatHubToolsTable1770000000000 = CreateChatHubToolsTable1770000000000;
