// @n8n/db/dist/migrations/common/1771500000000-MigrateExternalSecretsToEntityStorage.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MigrateExternalSecretsToEntityStorage1771500000000 = void 0;
const di_1 = require("@n8n/di");
const n8n_core_1 = require("n8n-core");
const n8n_workflow_1 = require("n8n-workflow");
const EXTERNAL_SECRETS_DB_KEY = 'feature.externalSecrets';
class MigrateExternalSecretsToEntityStorage1771500000000 {
    constructor() {
        this.cipher = di_1.Container.get(n8n_core_1.Cipher);
    }
    async up(context) {
        const allSettings = await this.readSettings(context);
        if (!allSettings)
            return;
        const providerNames = Object.keys(allSettings);
        if (providerNames.length === 0) {
            context.logger.info(`[${context.migrationName}] External secrets settings blob is empty, skipping`);
            return;
        }
        for (const providerName of providerNames) {
            await this.migrateProvider(context, providerName, allSettings[providerName]);
        }
        context.logger.info(`[${context.migrationName}] Migration complete. Old settings row preserved in settings table.`);
    }
    async readSettings({ escape, runQuery, logger, migrationName, }) {
        const settingsTable = escape.tableName('settings');
        const keyCol = escape.columnName('key');
        const valueCol = escape.columnName('value');
        const rows = await runQuery(`SELECT ${valueCol} FROM ${settingsTable} WHERE ${keyCol} = '${EXTERNAL_SECRETS_DB_KEY}';`);
        if (rows.length === 0) {
            logger.info(`[${migrationName}] No external secrets settings found, skipping`);
            return undefined;
        }
        try {
            const decrypted = this.cipher.decrypt(rows[0].value);
            return (0, n8n_workflow_1.jsonParse)(decrypted);
        }
        catch {
            logger.error(`[${migrationName}] Failed to decrypt external secrets settings, skipping migration`);
            return undefined;
        }
    }
    async migrateProvider({ escape, runQuery, logger, migrationName }, providerName, providerData) {
        if (!providerData.connected) {
            logger.info(`[${migrationName}] Provider "${providerName}" is not connected, skipping`);
            return;
        }
        const connectionTable = escape.tableName('secrets_provider_connection');
        const providerKeyCol = escape.columnName('providerKey');
        const typeCol = escape.columnName('type');
        const encryptedSettingsCol = escape.columnName('encryptedSettings');
        const existing = await runQuery(`SELECT ${providerKeyCol} FROM ${connectionTable} WHERE ${providerKeyCol} = :providerKey;`, { providerKey: providerName });
        if (existing.length > 0) {
            logger.info(`[${migrationName}] Provider "${providerName}" already exists in secrets_provider_connection, skipping`);
            return;
        }
        const encryptedSettings = this.cipher.encrypt(providerData.settings ?? {});
        await runQuery(`INSERT INTO ${connectionTable} (${providerKeyCol}, ${typeCol}, ${encryptedSettingsCol}) VALUES (:providerKey, :type, :encryptedSettings);`, {
            providerKey: providerName,
            type: providerName,
            encryptedSettings,
        });
        logger.info(`[${migrationName}] Migrated provider "${providerName}" to secrets_provider_connection`);
    }
}
exports.MigrateExternalSecretsToEntityStorage1771500000000 = MigrateExternalSecretsToEntityStorage1771500000000;
