// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/hubspot/v1/resource_ticket/operation_get.ts
/**
 * HubSpot Node - Version 1
 * Discriminator: resource=ticket, operation=get
 */


interface Credentials {
  hubspotApi: CredentialReference;
  hubspotAppToken: CredentialReference;
  hubspotOAuth2Api: CredentialReference;
}

/** Get a contact */
export type HubspotV1TicketGetParams = {
  resource: 'ticket';
  operation: 'get';
  authentication?: 'apiKey' | 'appToken' | 'oAuth2' | Expression<string>;
/**
 * Unique identifier for a particular ticket
 */
    ticketId?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Include Deleted
     * @default false
     */
    includeDeleted?: boolean | Expression<boolean>;
    /** &lt;p&gt;Used to include specific ticket properties in the results. By default, the results will only include ticket ID and will not include the values for any properties for your tickets.&lt;/p&gt;&lt;p&gt;Including this parameter will include the data for the specified property in the results. You can include this parameter multiple times to request multiple properties separated by a comma: &lt;code&gt;,&lt;/code&gt;.&lt;/p&gt;. Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     * @default []
     */
    properties?: string[];
    /** Works similarly to properties=, but this parameter will include the history for the specified property, instead of just including the current value. Use this parameter when you need the full history of changes to a property's value.
     */
    propertiesWithHistory?: string | Expression<string> | PlaceholderValue;
  };
};

export type HubspotV1TicketGetNode = {
  type: 'n8n-nodes-base.hubspot';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<HubspotV1TicketGetParams>;
};
