// @n8n/n8n-nodes-langchain/dist/node-definitions/nodes/n8n-nodes-langchain/memoryManager/v11/mode_delete.schema.js
/**
 * Chat Memory Manager Node - Version 1.1 - Zod Schema
 * Discriminator: mode=delete
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema, memoryInstanceSchema }) {

  // Static subnode schema
  const subnodesSchema = z.object({
    memory: memoryInstanceSchema.optional(),
  }).strict();

  return z.object({
    parameters: z.object({
      mode: z.literal('delete'),
      deleteMode: z.union([z.literal('lastN'), z.literal('all')]).optional(),
      lastMessagesCount: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"deleteMode":["lastN"]}}, defaults: {"deleteMode":"lastN"} }),
    }).optional(),
    subnodes: subnodesSchema.optional(),
  });
};
