// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/medium/v1/resource_post/operation_create.schema.js
/**
 * Medium Node - Version 1 - Zod Schema
 * Discriminator: resource=post, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('post').default('post'),
      operation: z.literal('create').default('create'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      publication: booleanOrExpression.optional(),
      publicationId: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"publication":[true]}}, defaults: {"publication":false} }),
      title: stringOrExpression.optional(),
      contentFormat: z.union([z.literal('html'), z.literal('markdown'), expressionSchema]).optional(),
      content: stringOrExpression.optional(),
      additionalFields: z.object({ canonicalUrl: stringOrExpression.optional(), license: z.union([z.literal('all-rights-reserved'), z.literal('cc-40-by'), z.literal('cc-40-by-nc'), z.literal('cc-40-by-nc-nd'), z.literal('cc-40-by-nc-sa'), z.literal('cc-40-by-nd'), z.literal('cc-40-by-sa'), z.literal('cc-40-zero'), z.literal('public-domain'), expressionSchema]).optional(), notifyFollowers: booleanOrExpression.optional(), publishStatus: z.union([z.literal('public'), z.literal('draft'), z.literal('unlisted'), expressionSchema]).optional(), tags: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
