// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/sentryIo/v1/resource_team/operation_update.ts
/**
 * Sentry.io Node - Version 1
 * Discriminator: resource=team, operation=update
 */


interface Credentials {
  sentryIoOAuth2Api: CredentialReference;
  sentryIoApi: CredentialReference;
  sentryIoServerApi: CredentialReference;
}

/** Update an issue */
export type SentryIoV1TeamUpdateParams = {
  resource: 'team';
  operation: 'update';
  authentication?: 'accessToken' | 'oAuth2' | 'accessTokenServer' | Expression<string>;
/**
 * The slug of the organization the team belongs to. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    organizationSlug?: string | Expression<string>;
/**
 * The slug of the team to update. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    teamSlug?: string | Expression<string>;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** The new name of the team
     */
    name?: string | Expression<string> | PlaceholderValue;
    /** The new slug of the team. Must be unique and available.
     */
    slug?: string | Expression<string> | PlaceholderValue;
  };
};

export type SentryIoV1TeamUpdateNode = {
  type: 'n8n-nodes-base.sentryIo';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SentryIoV1TeamUpdateParams>;
};
