// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/sentryIo/v1/resource_team/operation_create.ts
/**
 * Sentry.io Node - Version 1
 * Discriminator: resource=team, operation=create
 */


interface Credentials {
  sentryIoOAuth2Api: CredentialReference;
  sentryIoApi: CredentialReference;
  sentryIoServerApi: CredentialReference;
}

/** Create an organization */
export type SentryIoV1TeamCreateParams = {
  resource: 'team';
  operation: 'create';
  authentication?: 'accessToken' | 'oAuth2' | 'accessTokenServer' | Expression<string>;
/**
 * The slug of the organization the team belongs to. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    organizationSlug?: string | Expression<string>;
/**
 * The name of the team
 */
    name?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** The optional slug for this team. If not provided it will be auto generated from the name.
     */
    slug?: string | Expression<string> | PlaceholderValue;
  };
};

export type SentryIoV1TeamCreateNode = {
  type: 'n8n-nodes-base.sentryIo';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SentryIoV1TeamCreateParams>;
};
