// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/tapfiliate/v1/resource_affiliate_metadata/operation_add.schema.js
/**
 * Tapfiliate Node - Version 1 - Zod Schema
 * Discriminator: resource=affiliateMetadata, operation=add
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('affiliateMetadata'),
      operation: z.literal('add'),
      affiliateId: stringOrExpression.optional(),
      metadataUi: z.object({ metadataValues: z.array(z.object({ key: stringOrExpression.optional(), value: stringOrExpression.optional() })).optional() }).optional(),
    }).optional(),
  });
};
