// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/localFileTrigger/v1.schema.js
/**
 * Local File Trigger Node - Version 1 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    triggerOn: z.union([z.literal('file'), z.literal('folder'), expressionSchema]).optional(),
    path: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"triggerOn":["file","folder"]}}, defaults: {"triggerOn":""} }),
    events: resolveSchema({ parameters, schema: z.array(z.union([z.literal('add'), z.literal('change'), z.literal('unlink'), z.literal('addDir'), z.literal('unlinkDir')])), required: false, displayOptions: {"show":{"triggerOn":["folder"]}}, defaults: {"triggerOn":""} }),
    options: z.object({ awaitWriteFinish: booleanOrExpression.optional(), followSymlinks: booleanOrExpression.optional(), ignored: stringOrExpression.optional(), ignoreInitial: booleanOrExpression.optional(), depth: z.union([z.literal(1), z.literal(2), z.literal(3), z.literal(4), z.literal(5), z.literal(0), z.literal(-1), expressionSchema]).optional(), usePolling: booleanOrExpression.optional(), ignoreMode: z.union([z.literal('match'), z.literal('contain'), expressionSchema]).optional() }).optional(),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
