// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/baserow/v1/resource_row/operation_get_all.schema.js
/**
 * Baserow Node - Version 1 - Zod Schema
 * Discriminator: resource=row, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('row').default('row'),
      operation: z.literal('getAll').default('getAll'),
      databaseId: stringOrExpression.optional(),
      tableId: stringOrExpression.optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      additionalOptions: z.object({ filters: z.unknown().optional(), filterType: z.union([z.literal('AND'), z.literal('OR'), expressionSchema]).optional(), search: stringOrExpression.optional(), order: z.unknown().optional() }).optional(),
    }).optional(),
  });
};
