// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/baserow/v1/resource_row/operation_update.ts
/**
 * Baserow Node - Version 1
 * Discriminator: resource=row, operation=update
 */


interface Credentials {
  baserowApi: CredentialReference;
}

/** Update a row */
export type BaserowV1RowUpdateParams = {
  resource: 'row';
  operation: 'update';
/**
 * Database to operate on. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    databaseId?: string | Expression<string>;
/**
 * Table to operate on. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    tableId?: string | Expression<string>;
/**
 * ID of the row to update
 */
    rowId?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to insert the input data this node receives in the new row
 * @default defineBelow
 */
    dataToSend?: 'autoMapInputData' | 'defineBelow' | Expression<string>;
/**
 * List of input properties to avoid sending, separated by commas. Leave empty to send all properties.
 * @displayOptions.show { dataToSend: ["autoMapInputData"] }
 */
    inputsToIgnore?: string | Expression<string> | PlaceholderValue;
/**
 * Fields to Send
 * @displayOptions.show { dataToSend: ["defineBelow"] }
 * @default {}
 */
    fieldsUi?: {
        /** Field
     */
    fieldValues?: Array<{
      /** Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
       */
      fieldId?: string | Expression<string>;
      /** Field Value
       */
      fieldValue?: string | Expression<string> | PlaceholderValue;
    }>;
  };
};

export type BaserowV1RowUpdateNode = {
  type: 'n8n-nodes-base.baserow';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<BaserowV1RowUpdateParams>;
};
