// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/baserow/v1/resource_row/operation_get_all.ts
/**
 * Baserow Node - Version 1
 * Discriminator: resource=row, operation=getAll
 */


interface Credentials {
  baserowApi: CredentialReference;
}

/** Retrieve many rows */
export type BaserowV1RowGetAllParams = {
  resource: 'row';
  operation: 'getAll';
/**
 * Database to operate on. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    databaseId?: string | Expression<string>;
/**
 * Table to operate on. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    tableId?: string | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Options
 * @default {}
 */
    additionalOptions?: {
    /** Filter rows based on comparison operators
     * @default {}
     */
    filters?: {
        /** Field
     */
    fields?: Array<{
      /** Field to compare. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
       */
      field?: string | Expression<string>;
      /** Operator to compare field and value with
       * @default equal
       */
      operator?: 'contains' | 'contains_not' | 'date_after' | 'date_before' | 'date_equal' | 'date_equals_month' | 'date_equals_today' | 'date_equals_year' | 'date_not_equal' | 'equal' | 'filename_contains' | 'higher_than' | 'empty' | 'not_empty' | 'boolean' | 'link_row_has_not' | 'link_row_has' | 'lower_than' | 'not_equal' | 'single_select_equal' | 'single_select_not_equal' | Expression<string>;
      /** Value to compare to
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
    /** This works only if two or more filters are provided. Defaults to &lt;code&gt;AND&lt;/code&gt;
     * @default AND
     */
    filterType?: 'AND' | 'OR' | Expression<string>;
    /** Text to match (can be in any column)
     */
    search?: string | Expression<string> | PlaceholderValue;
    /** Set the sort order of the result rows
     * @default {}
     */
    order?: {
        /** Field
     */
    fields?: Array<{
      /** Field name to sort by. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
       */
      field?: string | Expression<string>;
      /** Sort direction, either ascending or descending
       */
      direction?: '' | '-' | Expression<string>;
    }>;
  };
  };
};

export type BaserowV1RowGetAllNode = {
  type: 'n8n-nodes-base.baserow';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<BaserowV1RowGetAllParams>;
};
