// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/twist/v1/resource_comment/operation_get_all.schema.js
/**
 * Twist Node - Version 1 - Zod Schema
 * Discriminator: resource=comment, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('comment'),
      operation: z.literal('getAll'),
      threadId: stringOrExpression.optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filters: z.object({ as_ids: booleanOrExpression.optional(), to_obj_index: numberOrExpression.optional(), newer_than_ts: stringOrExpression.optional(), older_than_ts: stringOrExpression.optional(), order_by: z.union([z.literal('ASC'), z.literal('DESC'), expressionSchema]).optional(), from_obj_index: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
