// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/twist/v1/resource_comment/operation_create.ts
/**
 * Twist Node - Version 1
 * Discriminator: resource=comment, operation=create
 */


interface Credentials {
  twistOAuth2Api: CredentialReference;
}

/** Initiates a public or private channel-based conversation */
export type TwistV1CommentCreateParams = {
  resource: 'comment';
  operation: 'create';
/**
 * The ID of the thread
 */
    threadId?: string | Expression<string> | PlaceholderValue;
/**
 * The content of the comment
 */
    content?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Actions
     * @default {}
     */
    actionsUi?: {
        /** Action
     */
    actionValues?: Array<{
      /** The action of the button
       */
      action?: 'open_url' | 'prefill_message' | 'send_reply' | Expression<string>;
      /** The text for the action button
       */
      button_text?: string | Expression<string> | PlaceholderValue;
      /** The text for the action button
       * @displayOptions.show { action: ["send_reply", "prefill_message"] }
       */
      message?: string | Expression<string> | PlaceholderValue;
      /** The type of the button. (Currently only &lt;code&gt;action&lt;/code&gt; is available).
       */
      type?: 'action' | Expression<string>;
      /** URL to redirect
       * @displayOptions.show { action: ["open_url"] }
       */
      url?: string | Expression<string> | PlaceholderValue;
    }>;
  };
    /** Name of the property that holds the binary data. Multiple can be defined separated by comma.
     * @default data
     */
    binaryProperties?: string | Expression<string> | PlaceholderValue;
    /** The users that are directly mentioned. Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     * @default []
     */
    direct_mentions?: string[];
    /** Whether the position of the thread is marked
     * @default true
     */
    mark_thread_position?: boolean | Expression<boolean>;
    /** The users that will attached to the comment. Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     * @default []
     */
    recipients?: string[];
    /** The temporary ID of the comment
     * @default 0
     */
    temp_id?: number | Expression<number>;
    /** Whether to display the integration as the comment creator
     * @default false
     */
    send_as_integration?: boolean | Expression<boolean>;
  };
};

export type TwistV1CommentCreateNode = {
  type: 'n8n-nodes-base.twist';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<TwistV1CommentCreateParams>;
};
