// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/twist/v1/resource_comment/operation_get_all.ts
/**
 * Twist Node - Version 1
 * Discriminator: resource=comment, operation=getAll
 */


interface Credentials {
  twistOAuth2Api: CredentialReference;
}

/** Get many channels */
export type TwistV1CommentGetAllParams = {
  resource: 'comment';
  operation: 'getAll';
/**
 * The ID of the channel
 */
    threadId?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Whether only the IDs of the comments are returned
     * @default false
     */
    as_ids?: boolean | Expression<boolean>;
    /** Limit comments ending at the specified object index
     * @default 50
     */
    to_obj_index?: number | Expression<number>;
    /** Limits comments to those newer when the specified Unix time
     */
    newer_than_ts?: string | Expression<string>;
    /** Limits comments to those older than the specified Unix time
     */
    older_than_ts?: string | Expression<string>;
    /** The order of the comments returned - one of DESC or ASC
     * @default ASC
     */
    order_by?: 'ASC' | 'DESC' | Expression<string>;
    /** Limit comments starting at the specified object index
     * @default 0
     */
    from_obj_index?: number | Expression<number>;
  };
};

export type TwistV1CommentGetAllNode = {
  type: 'n8n-nodes-base.twist';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<TwistV1CommentGetAllParams>;
};
