// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/youTube/v1/resource_channel/operation_get.schema.js
/**
 * YouTube Node - Version 1 - Zod Schema
 * Discriminator: resource=channel, operation=get
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('channel').default('channel'),
      operation: z.literal('get'),
      channelId: stringOrExpression.optional(),
      part: z.array(z.union([z.literal('*'), z.literal('brandingSettings'), z.literal('contentDetails'), z.literal('contentOwnerDetails'), z.literal('id'), z.literal('localizations'), z.literal('snippet'), z.literal('statistics'), z.literal('status'), z.literal('topicDetails')])).optional(),
    }).optional(),
  });
};
