// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/koBoToolbox/v1/resource_hook/operation_get_logs.schema.js
/**
 * KoBoToolbox Node - Version 1 - Zod Schema
 * Discriminator: resource=hook, operation=getLogs
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('hook'),
      operation: z.literal('getLogs'),
      formId: stringOrExpression.optional(),
      hookId: stringOrExpression.optional(),
      status: z.union([z.literal(''), z.literal('0'), z.literal('1'), z.literal('2'), expressionSchema]).optional(),
      startDate: stringOrExpression.optional(),
      endDate: stringOrExpression.optional(),
    }).optional(),
  });
};
