// @n8n/n8n-nodes-langchain/dist/nodes/tools/ToolSerpApi/ToolSerpApi.node.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ToolSerpApi = void 0;
const serpapi_1 = require("@langchain/community/tools/serpapi");
const ai_utilities_1 = require("@n8n/ai-utilities");
const n8n_workflow_1 = require("n8n-workflow");
async function getTool(ctx, itemIndex) {
    const credentials = await ctx.getCredentials('serpApi');
    const options = ctx.getNodeParameter('options', itemIndex);
    return new serpapi_1.SerpAPI(credentials.apiKey, options);
}
class ToolSerpApi {
    constructor() {
        this.description = {
            displayName: 'SerpApi (Google Search)',
            name: 'toolSerpApi',
            icon: 'file:serpApi.svg',
            group: ['transform'],
            version: 1,
            description: 'Search in Google using SerpAPI',
            defaults: {
                name: 'SerpAPI',
            },
            codex: {
                categories: ['AI'],
                subcategories: {
                    AI: ['Tools'],
                    Tools: ['Other Tools'],
                },
                resources: {
                    primaryDocumentation: [
                        {
                            url: 'https://docs.n8n.io/integrations/builtin/cluster-nodes/sub-nodes/n8n-nodes-langchain.toolserpapi/',
                        },
                    ],
                },
            },
            inputs: [],
            outputs: [n8n_workflow_1.NodeConnectionTypes.AiTool],
            outputNames: ['Tool'],
            credentials: [
                {
                    name: 'serpApi',
                    required: true,
                },
            ],
            properties: [
                (0, ai_utilities_1.getConnectionHintNoticeField)([n8n_workflow_1.NodeConnectionTypes.AiAgent]),
                {
                    displayName: 'Options',
                    name: 'options',
                    type: 'collection',
                    placeholder: 'Add Option',
                    default: {},
                    options: [
                        {
                            displayName: 'Country',
                            name: 'gl',
                            type: 'string',
                            default: 'us',
                            description: 'Defines the country to use for search. Head to <a href="https://serpapi.com/google-countries">Google countries page</a> for a full list of supported countries.',
                        },
                        {
                            displayName: 'Device',
                            name: 'device',
                            type: 'options',
                            options: [
                                {
                                    name: 'Desktop',
                                    value: 'desktop',
                                },
                                {
                                    name: 'Mobile',
                                    value: 'mobile',
                                },
                                {
                                    name: 'Tablet',
                                    value: 'tablet',
                                },
                            ],
                            default: 'desktop',
                            description: 'Device to use to get the results',
                        },
                        {
                            displayName: 'Explicit Array',
                            name: 'no_cache',
                            type: 'boolean',
                            default: false,
                            description: 'Whether to force SerpApi to fetch the Google results even if a cached version is already present. Cache expires after 1h. Cached searches are free, and are not counted towards your searches per month.',
                        },
                        {
                            displayName: 'Google Domain',
                            name: 'google_domain',
                            type: 'string',
                            default: 'google.com',
                            description: 'Defines the domain to use for search. Head to <a href="https://serpapi.com/google-domains">Google domains page</a> for a full list of supported domains.',
                        },
                        {
                            displayName: 'Language',
                            name: 'hl',
                            type: 'string',
                            default: 'en',
                            description: 'Defines the language to use. It\'s a two-letter language code. (e.g., `en` for English, `es` for Spanish, or `fr` for French). Head to <a href="https://serpapi.com/google-languages">Google languages page</a> for a full list of supported languages.',
                        },
                    ],
                },
            ],
        };
    }
    async supplyData(itemIndex) {
        return {
            response: (0, ai_utilities_1.logWrapper)(await getTool(this, itemIndex), this),
        };
    }
    async execute() {
        const inputData = this.getInputData();
        const returnData = [];
        for (let itemIndex = 0; itemIndex < inputData.length; itemIndex++) {
            const tool = await getTool(this, itemIndex);
            const item = inputData[itemIndex].json;
            if (typeof item.input !== 'string' || !item.input) {
                throw new n8n_workflow_1.NodeOperationError(this.getNode(), `Missing search query input at itemIndex ${itemIndex}`);
            }
            const result = (await tool.invoke(item));
            returnData.push({
                json: {
                    response: result,
                },
                pairedItem: { item: itemIndex },
            });
        }
        return [returnData];
    }
}
exports.ToolSerpApi = ToolSerpApi;
