// @n8n/n8n-nodes-langchain/dist/node-definitions/nodes/n8n-nodes-langchain/vectorStoreSupabase/v11/mode_insert.schema.js
/**
 * Supabase Vector Store Node - Version 1.1 - Zod Schema
 * Discriminator: mode=insert
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, embeddingInstanceSchema, documentLoaderInstanceSchema }) {

  // Static subnode schema
  const subnodesSchema = z.object({
    embedding: z.union([embeddingInstanceSchema, z.array(embeddingInstanceSchema)]),
    documentLoader: z.union([documentLoaderInstanceSchema, z.array(documentLoaderInstanceSchema)]),
  }).strict();

  return z.object({
    parameters: z.object({
      mode: z.literal('insert'),
      ragStarterCallout: z.unknown().optional(),
      tableName: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      embeddingBatchSize: numberOrExpression.optional(),
      options: z.object({ queryName: stringOrExpression.optional() }).optional(),
    }).optional(),
    subnodes: subnodesSchema,
  });
};
