// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/serviceNow/v1/resource_table_record/operation_create.ts
/**
 * ServiceNow Node - Version 1
 * Discriminator: resource=tableRecord, operation=create
 */


interface Credentials {
  serviceNowOAuth2Api: CredentialReference;
  serviceNowBasicApi: CredentialReference;
}

export type ServiceNowV1TableRecordCreateParams = {
  resource: 'tableRecord';
  operation: 'create';
/**
 * Authentication method to use
 * @default oAuth2
 */
    authentication?: 'basicAuth' | 'oAuth2' | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    tableName?: string | Expression<string>;
/**
 * Data to Send
 * @default columns
 */
    dataToSend?: 'mapInput' | 'columns' | 'nothing' | Expression<string>;
/**
 * List of input properties to avoid sending, separated by commas. Leave empty to send all inputs.
 * @displayOptions.show { dataToSend: ["mapInput"] }
 */
    inputsToIgnore?: string | Expression<string> | PlaceholderValue;
/**
 * Fields to Send
 * @displayOptions.show { dataToSend: ["columns"] }
 * @default {}
 */
    fieldsToSend?: {
        /** Field
     */
    field?: Array<{
      /** Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
       */
      column?: string | Expression<string>;
      /** Field Value
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
};

export type ServiceNowV1TableRecordCreateOutput = {
  active?: string;
  activity_due?: string;
  additional_assignee_list?: string;
  approval?: string;
  approval_history?: string;
  approval_set?: string;
  assigned_to?: string;
  business_duration?: string;
  business_service?: string;
  calendar_duration?: string;
  close_notes?: string;
  closed_at?: string;
  closed_by?: string;
  cmdb_ci?: string;
  comments?: string;
  comments_and_work_notes?: string;
  contact_type?: string;
  contract?: string;
  correlation_display?: string;
  correlation_id?: string;
  description?: string;
  due_date?: string;
  escalation?: string;
  expected_start?: string;
  follow_up?: string;
  group_list?: string;
  impact?: string;
  knowledge?: string;
  location?: string;
  made_sla?: string;
  number?: string;
  opened_at?: string;
  opened_by?: {
    link?: string;
    value?: string;
  };
  order?: string;
  parent?: string;
  priority?: string;
  reassignment_count?: string;
  route_reason?: string;
  service_offering?: string;
  short_description?: string;
  sla_due?: string;
  state?: string;
  sys_class_name?: string;
  sys_created_by?: string;
  sys_created_on?: string;
  sys_domain?: {
    link?: string;
    value?: string;
  };
  sys_domain_path?: string;
  sys_id?: string;
  sys_mod_count?: string;
  sys_tags?: string;
  sys_updated_by?: string;
  sys_updated_on?: string;
  task_effective_number?: string;
  universal_request?: string;
  upon_approval?: string;
  upon_reject?: string;
  urgency?: string;
  user_input?: string;
  watch_list?: string;
  work_end?: string;
  work_notes?: string;
  work_notes_list?: string;
  work_start?: string;
};

export type ServiceNowV1TableRecordCreateNode = {
  type: 'n8n-nodes-base.serviceNow';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ServiceNowV1TableRecordCreateParams>;
  output?: Items<ServiceNowV1TableRecordCreateOutput>;
};
