// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/serviceNow/v1/resource_table_record/operation_get.schema.js
/**
 * ServiceNow Node - Version 1 - Zod Schema
 * Discriminator: resource=tableRecord, operation=get
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('tableRecord'),
      operation: z.literal('get'),
      authentication: z.union([z.literal('basicAuth'), z.literal('oAuth2'), expressionSchema]).optional(),
      tableName: stringOrExpression.optional(),
      id: stringOrExpression.optional(),
      options: z.object({ sysparm_exclude_reference_link: booleanOrExpression.optional(), sysparm_fields: z.array(z.string()).optional(), sysparm_display_value: z.union([z.literal('false'), z.literal('all'), z.literal('true'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
