// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/serviceNow/v1/resource_table_record/operation_create.schema.js
/**
 * ServiceNow Node - Version 1 - Zod Schema
 * Discriminator: resource=tableRecord, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('tableRecord'),
      operation: z.literal('create'),
      authentication: z.union([z.literal('basicAuth'), z.literal('oAuth2'), expressionSchema]).optional(),
      tableName: stringOrExpression.optional(),
      dataToSend: z.union([z.literal('mapInput'), z.literal('columns'), z.literal('nothing'), expressionSchema]).optional(),
      inputsToIgnore: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"dataToSend":["mapInput"]}}, defaults: {"dataToSend":"columns"} }),
      fieldsToSend: resolveSchema({ parameters, schema: z.object({ field: z.array(z.object({ column: stringOrExpression.optional(), value: stringOrExpression.optional() })).optional() }), required: false, displayOptions: {"show":{"dataToSend":["columns"]}}, defaults: {"dataToSend":"columns"} }),
    }).optional(),
  });
};
