// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/serviceNow/v1/resource_table_record/operation_get.ts
/**
 * ServiceNow Node - Version 1
 * Discriminator: resource=tableRecord, operation=get
 */


interface Credentials {
  serviceNowOAuth2Api: CredentialReference;
  serviceNowBasicApi: CredentialReference;
}

/** Get an attachment */
export type ServiceNowV1TableRecordGetParams = {
  resource: 'tableRecord';
  operation: 'get';
/**
 * Authentication method to use
 * @default oAuth2
 */
    authentication?: 'basicAuth' | 'oAuth2' | Expression<string>;
/**
 * Name of the table in which the record exists. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    tableName?: string | Expression<string>;
/**
 * Unique identifier of the record
 */
    id?: string | Expression<string> | PlaceholderValue;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Whether to exclude Table API links for reference fields
     * @default false
     */
    sysparm_exclude_reference_link?: boolean | Expression<boolean>;
    /** A list of fields to return. Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     * @hint String of comma separated values or an array of strings can be set in an expression
     * @default []
     */
    sysparm_fields?: string[];
    /** Choose which values to return
     * @default false
     */
    sysparm_display_value?: 'false' | 'all' | 'true' | Expression<string>;
  };
};

export type ServiceNowV1TableRecordGetNode = {
  type: 'n8n-nodes-base.serviceNow';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ServiceNowV1TableRecordGetParams>;
};
