// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleCloudNaturalLanguage/v1/resource_document/operation_analyze_sentiment.schema.js
/**
 * Google Cloud Natural Language Node - Version 1 - Zod Schema
 * Discriminator: resource=document, operation=analyzeSentiment
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('document').default('document'),
      operation: z.literal('analyzeSentiment').default('analyzeSentiment'),
      source: z.union([z.literal('content'), z.literal('gcsContentUri'), expressionSchema]).optional(),
      content: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"source":["content"]}}, defaults: {"source":"content"} }),
      gcsContentUri: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"source":["gcsContentUri"]}}, defaults: {"source":"content"} }),
      options: z.object({ documentType: z.union([z.literal('HTML'), z.literal('PLAIN_TEXT'), expressionSchema]).optional(), encodingType: z.union([z.literal('NONE'), z.literal('UTF8'), z.literal('UTF16'), z.literal('UTF32'), expressionSchema]).optional(), language: z.union([z.literal('ar'), z.literal('zh'), z.literal('zh-Hant'), z.literal('nl'), z.literal('en'), z.literal('fr'), z.literal('de'), z.literal('id'), z.literal('it'), z.literal('ja'), z.literal('ko'), z.literal('pt'), z.literal('es'), z.literal('th'), z.literal('tr'), z.literal('vi'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
