// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/seaTable/v2/resource_asset/operation_upload.ts
/**
 * SeaTable Node - Version 2
 * Discriminator: resource=asset, operation=upload
 */


interface Credentials {
  seaTableApi: CredentialReference;
}

/** Add a file/image to an existing row */
export type SeaTableV2AssetUploadParams = {
  resource: 'asset';
  operation: 'upload';
/**
 * Choose from the list, or specify a name using an &lt;a href="https://docs.n8n.io/code-examples/expressions/"&gt;expression&lt;/a&gt;
 */
    tableName?: string | Expression<string>;
/**
 * Choose from the list, or specify the name using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    uploadColumn?: string | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    rowId?: string | Expression<string>;
/**
 * Name of the binary property which contains the data for the file to be written
 * @default data
 */
    dataPropertyName?: string | Expression<string> | PlaceholderValue;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Whether to replace the existing asset with the same name (true). Otherwise, a new version with a different name (numeral in parentheses) will be uploaded (false).
     * @default true
     */
    replace?: boolean | Expression<boolean>;
    /** Whether to keep existing files/images in the column and append the new asset (true). Otherwise, the existing files/images are removed from the column (false).
     * @default true
     */
    append?: boolean | Expression<boolean>;
  };
};

export type SeaTableV2AssetUploadNode = {
  type: 'n8n-nodes-base.seaTable';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<SeaTableV2AssetUploadParams>;
};
