// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/ssh/v1/resource_file/operation_upload.ts
/**
 * SSH Node - Version 1
 * Discriminator: resource=file, operation=upload
 */


interface Credentials {
  sshPassword: CredentialReference;
  sshPrivateKey: CredentialReference;
}

/** Upload a file */
export type SshV1FileUploadParams = {
  resource: 'file';
  operation: 'upload';
  authentication?: 'password' | 'privateKey' | Expression<string>;
/**
 * Input Binary Field
 * @hint The name of the input binary field containing the file to be uploaded
 * @default data
 */
    binaryPropertyName?: string | Expression<string> | PlaceholderValue;
/**
 * The directory to upload the file to. The name of the file does not need to be specified, it's taken from the binary data file name. To override this behavior, set the parameter "File Name" under options.
 */
    path?: string | Expression<string> | PlaceholderValue;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Overrides the binary data file name
     */
    fileName?: string | Expression<string> | PlaceholderValue;
  };
};

export type SshV1FileUploadOutput = {
  success?: boolean;
};

export type SshV1FileUploadNode = {
  type: 'n8n-nodes-base.ssh';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SshV1FileUploadParams>;
  output?: Items<SshV1FileUploadOutput>;
};
