// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/humanticAi/v1/resource_profile/operation_create.schema.js
/**
 * Humantic AI Node - Version 1 - Zod Schema
 * Discriminator: resource=profile, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('profile').default('profile'),
      operation: z.literal('create').default('create'),
      userId: stringOrExpression.optional(),
      sendResume: booleanOrExpression.optional(),
      binaryPropertyName: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"sendResume":[true]}}, defaults: {"sendResume":false} }),
    }).optional(),
  });
};
