// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/humanticAi/v1/resource_profile/operation_create.ts
/**
 * Humantic AI Node - Version 1
 * Discriminator: resource=profile, operation=create
 */


interface Credentials {
  humanticAiApi: CredentialReference;
}

/** Create a profile */
export type HumanticAiV1ProfileCreateParams = {
  resource: 'profile';
  operation: 'create';
/**
 * The LinkedIn profile URL or email ID for creating a Humantic profile. If you are sending the resume, this should be a unique string.
 */
    userId?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to send a resume for a resume based analysis
 * @default false
 */
    sendResume?: boolean | Expression<boolean>;
/**
 * Input Binary Field
 * @hint The name of the input binary field containing the resume in PDF or DOCX format
 * @displayOptions.show { sendResume: [true] }
 * @default data
 */
    binaryPropertyName?: string | Expression<string> | PlaceholderValue;
};

export type HumanticAiV1ProfileCreateNode = {
  type: 'n8n-nodes-base.humanticAi';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<HumanticAiV1ProfileCreateParams>;
};
