// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/humanticAi/v1/resource_profile/operation_update.ts
/**
 * Humantic AI Node - Version 1
 * Discriminator: resource=profile, operation=update
 */


interface Credentials {
  humanticAiApi: CredentialReference;
}

/** Update a profile */
export type HumanticAiV1ProfileUpdateParams = {
  resource: 'profile';
  operation: 'update';
/**
 * This value is the same as the User ID that was provided when the analysis was created. Currently only supported for profiles created using LinkedIn URL.
 */
    userId?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to send a resume for a resume of the user
 * @default false
 */
    sendResume?: boolean | Expression<boolean>;
/**
 * Additional text written by the user
 * @displayOptions.show { sendResume: [false] }
 */
    text?: string | Expression<string> | PlaceholderValue;
/**
 * Input Binary Field
 * @hint The name of the input binary field containing the resume in PDF or DOCX format
 * @displayOptions.show { sendResume: [true] }
 * @default data
 */
    binaryPropertyName?: string | Expression<string> | PlaceholderValue;
};

export type HumanticAiV1ProfileUpdateNode = {
  type: 'n8n-nodes-base.humanticAi';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<HumanticAiV1ProfileUpdateParams>;
};
