// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/goToWebinar/v1/resource_webinar/operation_create.schema.js
/**
 * GoToWebinar Node - Version 1 - Zod Schema
 * Discriminator: resource=webinar, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('webinar'),
      operation: z.literal('create'),
      subject: stringOrExpression.optional(),
      times: z.object({ timesProperties: z.array(z.object({ startTime: stringOrExpression.optional(), endTime: stringOrExpression.optional() })).optional() }).optional(),
      additionalFields: z.object({ description: stringOrExpression.optional(), experienceType: z.union([z.literal('CLASSIC'), z.literal('BROADCAST'), z.literal('SIMULIVE'), expressionSchema]).optional(), isOnDemand: booleanOrExpression.optional(), isPasswordProtected: booleanOrExpression.optional(), timezone: stringOrExpression.optional(), type: z.union([z.literal('single_session'), z.literal('series'), z.literal('sequence'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
