// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/goToWebinar/v1/resource_webinar/operation_create.ts
/**
 * GoToWebinar Node - Version 1
 * Discriminator: resource=webinar, operation=create
 */


interface Credentials {
  goToWebinarOAuth2Api: CredentialReference;
}

export type GoToWebinarV1WebinarCreateParams = {
  resource: 'webinar';
  operation: 'create';
/**
 * Subject
 */
    subject?: string | Expression<string> | PlaceholderValue;
/**
 * Time Range
 * @default {}
 */
    times?: {
        /** Times Properties
     */
    timesProperties?: Array<{
      /** Start Time
       */
      startTime?: string | Expression<string>;
      /** End Time
       */
      endTime?: string | Expression<string>;
    }>;
  };
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Description
     */
    description?: string | Expression<string> | PlaceholderValue;
    /** Experience Type
     * @default CLASSIC
     */
    experienceType?: 'CLASSIC' | 'BROADCAST' | 'SIMULIVE' | Expression<string>;
    /** Is On-Demand
     * @default false
     */
    isOnDemand?: boolean | Expression<boolean>;
    /** Is Password Protected
     * @default false
     */
    isPasswordProtected?: boolean | Expression<boolean>;
    /** Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     */
    timezone?: string | Expression<string>;
    /** Webinar Type
     * @default single_session
     */
    type?: 'single_session' | 'series' | 'sequence' | Expression<string>;
  };
};

export type GoToWebinarV1WebinarCreateNode = {
  type: 'n8n-nodes-base.goToWebinar';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GoToWebinarV1WebinarCreateParams>;
};
