// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/goToWebinar/v1/resource_webinar/operation_get_all.ts
/**
 * GoToWebinar Node - Version 1
 * Discriminator: resource=webinar, operation=getAll
 */


interface Credentials {
  goToWebinarOAuth2Api: CredentialReference;
}

export type GoToWebinarV1WebinarGetAllParams = {
  resource: 'webinar';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 10
 */
    limit?: number | Expression<number>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Time Range
     * @default {}
     */
    times?: {
        /** Times Properties
     */
    timesProperties?: {
      /** Start of the datetime range for the webinar
       */
      fromTime?: string | Expression<string>;
      /** End of the datetime range for the webinar
       */
      toTime?: string | Expression<string>;
    };
  };
  };
};

export type GoToWebinarV1WebinarGetAllNode = {
  type: 'n8n-nodes-base.goToWebinar';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GoToWebinarV1WebinarGetAllParams>;
};
