// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/keap/v1/resource_file/operation_get_all.schema.js
/**
 * Keap Node - Version 1 - Zod Schema
 * Discriminator: resource=file, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('file'),
      operation: z.literal('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filters: z.object({ contactId: numberOrExpression.optional(), name: stringOrExpression.optional(), permission: z.union([z.literal('user'), z.literal('company'), z.literal('both'), expressionSchema]).optional(), type: z.union([z.literal('application'), z.literal('attachment'), z.literal('contact'), z.literal('digitalProduct'), z.literal('fax'), z.literal('funnel'), z.literal('hidden'), z.literal('image'), z.literal('import'), z.literal('logoThumnail'), z.literal('reSampledImage'), z.literal('styleCart'), z.literal('templateThumnail'), z.literal('ticket'), z.literal('webform'), expressionSchema]).optional(), viewable: z.union([z.literal('public'), z.literal('private'), z.literal('both'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
