// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/keap/v1/resource_file/operation_upload.schema.js
/**
 * Keap Node - Version 1 - Zod Schema
 * Discriminator: resource=file, operation=upload
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('file'),
      operation: z.literal('upload'),
      binaryData: booleanOrExpression.optional(),
      binaryPropertyName: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"binaryData":[true]}}, defaults: {"binaryData":false} }),
      fileAssociation: z.union([z.literal('company'), z.literal('contact'), z.literal('user'), expressionSchema]).optional(),
      contactId: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"fileAssociation":["contact"]}}, defaults: {"fileAssociation":""} }),
      fileName: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"binaryData":[false]}}, defaults: {"binaryData":false} }),
      fileData: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"binaryData":[false]}}, defaults: {"binaryData":false} }),
      isPublic: booleanOrExpression.optional(),
    }).optional(),
  });
};
