// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/e2eTest/v1.schema.js
/**
 * E2E Test Node - Version 1 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    operation: z.union([z.literal('remoteOptions'), z.literal('resourceLocator'), z.literal('resourceMapper')]).optional(),
    fieldId: stringOrExpression.optional(),
    remoteOptions: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"operation":["remoteOptions"]}}, defaults: {"operation":"remoteOptions"} }),
    rlc: resolveSchema({ parameters, schema: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('url'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]), required: false, displayOptions: {"show":{"operation":["resourceLocator"]}}, defaults: {"operation":"remoteOptions"} }),
    resourceMapper: resolveSchema({ parameters, schema: resourceMapperValueSchema, required: false, displayOptions: {"show":{"operation":["resourceMapper"]}}, defaults: {"operation":"remoteOptions"} }),
    otherField: stringOrExpression.optional(),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
