// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/metabase/v1/resource_databases/operation_add_new_datasource.ts
/**
 * Metabase Node - Version 1
 * Discriminator: resource=databases, operation=addNewDatasource
 */


interface Credentials {
  metabaseApi: CredentialReference;
}

/** Add a new datasource to the metabase instance */
export type MetabaseV1DatabasesAddNewDatasourceParams = {
  resource: 'databases';
  operation: 'addNewDatasource';
/**
 * Engine
 * @default postgres
 */
    engine?: 'h2' | 'mongo' | 'mysql' | 'postgres' | 'redshift' | 'sqlite' | Expression<string>;
/**
 * Host
 * @displayOptions.show { engine: ["postgres", "redshift", "mysql", "mongo"] }
 */
    host?: string | Expression<string> | PlaceholderValue;
/**
 * Name
 */
    name?: string | Expression<string> | PlaceholderValue;
/**
 * Port
 * @displayOptions.show { engine: ["postgres", "redshift", "mysql", "mongo"] }
 * @default 5432
 */
    port?: number | Expression<number>;
/**
 * User
 * @displayOptions.show { engine: ["postgres", "redshift", "mysql", "mongo"] }
 */
    user?: string | Expression<string> | PlaceholderValue;
/**
 * Password
 * @displayOptions.show { engine: ["postgres", "redshift", "mysql", "mongo"] }
 */
    password?: string | Expression<string> | PlaceholderValue;
/**
 * Database Name
 * @displayOptions.show { engine: ["postgres", "redshift", "mysql", "mongo"] }
 */
    dbName?: string | Expression<string> | PlaceholderValue;
/**
 * File Path
 * @displayOptions.show { engine: ["h2", "sqlite"] }
 */
    filePath?: string | Expression<string> | PlaceholderValue;
/**
 * Full Sync
 * @default true
 */
    fullSync?: boolean | Expression<boolean>;
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type MetabaseV1DatabasesAddNewDatasourceNode = {
  type: 'n8n-nodes-base.metabase';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MetabaseV1DatabasesAddNewDatasourceParams>;
};
