// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/metabase/v1/resource_databases/operation_add_new_datasource.schema.js
/**
 * Metabase Node - Version 1 - Zod Schema
 * Discriminator: resource=databases, operation=addNewDatasource
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('databases'),
      operation: z.literal('addNewDatasource'),
      engine: z.union([z.literal('h2'), z.literal('mongo'), z.literal('mysql'), z.literal('postgres'), z.literal('redshift'), z.literal('sqlite'), expressionSchema]).optional(),
      host: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"engine":["postgres","redshift","mysql","mongo"]}}, defaults: {"engine":"postgres"} }),
      name: stringOrExpression.optional(),
      port: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"engine":["postgres","redshift","mysql","mongo"]}}, defaults: {"engine":"postgres"} }),
      user: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"engine":["postgres","redshift","mysql","mongo"]}}, defaults: {"engine":"postgres"} }),
      password: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"engine":["postgres","redshift","mysql","mongo"]}}, defaults: {"engine":"postgres"} }),
      dbName: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"engine":["postgres","redshift","mysql","mongo"]}}, defaults: {"engine":"postgres"} }),
      filePath: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"engine":["h2","sqlite"]}}, defaults: {"engine":"postgres"} }),
      fullSync: booleanOrExpression.optional(),
      requestOptions: z.object({ batching: z.unknown().optional(), allowUnauthorizedCerts: z.boolean().optional(), proxy: stringOrExpression.optional(), timeout: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
