// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/metabase/v1/resource_databases/operation_get_fields.ts
/**
 * Metabase Node - Version 1
 * Discriminator: resource=databases, operation=getFields
 */


interface Credentials {
  metabaseApi: CredentialReference;
}

/** Get fields from database */
export type MetabaseV1DatabasesGetFieldsParams = {
  resource: 'databases';
  operation: 'getFields';
/**
 * Database ID
 */
    databaseId?: string | Expression<string> | PlaceholderValue;
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type MetabaseV1DatabasesGetFieldsOutput = {
  base_type?: string;
  display_name?: string;
  id?: number;
  name?: string;
  table_id?: number;
  table_name?: string;
};

export type MetabaseV1DatabasesGetFieldsNode = {
  type: 'n8n-nodes-base.metabase';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MetabaseV1DatabasesGetFieldsParams>;
  output?: Items<MetabaseV1DatabasesGetFieldsOutput>;
};
