// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/hubspot/v22/resource_contact/operation_get_recently_created_updated.schema.js
/**
 * HubSpot Node - Version 2.2 - Zod Schema
 * Discriminator: resource=contact, operation=getRecentlyCreatedUpdated
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('contact').default('contact'),
      operation: z.literal('getRecentlyCreatedUpdated'),
      authentication: z.union([z.literal('apiKey'), z.literal('appToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      additionalFields: z.object({ formSubmissionMode: z.union([z.literal('all'), z.literal('none'), z.literal('newest'), z.literal('oldest'), expressionSchema]).optional(), listMemberships: booleanOrExpression.optional(), propertiesCollection: z.unknown().optional() }).optional(),
    }).optional(),
  });
};
