// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/hubspot/v22/resource_contact/operation_get.schema.js
/**
 * HubSpot Node - Version 2.2 - Zod Schema
 * Discriminator: resource=contact, operation=get
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('contact').default('contact'),
      operation: z.literal('get'),
      authentication: z.union([z.literal('apiKey'), z.literal('appToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      contactId: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      additionalFields: z.object({ formSubmissionMode: z.union([z.literal('all'), z.literal('none'), z.literal('newest'), z.literal('oldest'), expressionSchema]).optional(), listMemberships: booleanOrExpression.optional(), propertiesCollection: z.unknown().optional() }).optional(),
    }).optional(),
  });
};
