// @n8n/n8n-nodes-langchain/dist/nodes/Guardrails/Guardrails.node.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Guardrails = void 0;
const n8n_workflow_1 = require("n8n-workflow");
const GuardrailsV1_node_1 = require("./v1/GuardrailsV1.node");
const GuardrailsV2_node_1 = require("./v2/GuardrailsV2.node");
class Guardrails extends n8n_workflow_1.VersionedNodeType {
    constructor() {
        const baseDescription = {
            displayName: 'Guardrails',
            name: 'guardrails',
            icon: 'file:guardrails.svg',
            group: ['transform'],
            defaultVersion: 2,
            description: 'Safeguard AI models from malicious input or prevent them from generating undesirable responses',
            codex: {
                alias: ['LangChain', 'Guardrails', 'PII', 'Secret', 'Injection', 'Sanitize'],
                categories: ['AI'],
                subcategories: {
                    AI: ['Agents', 'Miscellaneous', 'Root Nodes'],
                },
                resources: {
                    primaryDocumentation: [
                        {
                            url: 'https://docs.n8n.io/integrations/builtin/core-nodes/n8n-nodes-langchain.guardrails/',
                        },
                    ],
                },
            },
        };
        const nodeVersions = {
            1: new GuardrailsV1_node_1.GuardrailsV1(baseDescription),
            2: new GuardrailsV2_node_1.GuardrailsV2(baseDescription),
        };
        super(nodeVersions, baseDescription);
    }
}
exports.Guardrails = Guardrails;
