// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/coinGecko/v1/resource_event/operation_get_all.ts
/**
 * CoinGecko Node - Version 1
 * Discriminator: resource=event, operation=getAll
 */


/** Get many coins */
export type CoinGeckoV1EventGetAllParams = {
  resource: 'event';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Country code of event. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    country_code?: string | Expression<string>;
    /** Lists events after this date
     */
    from_date?: string | Expression<string>;
    /** Lists events before this date
     */
    to_date?: string | Expression<string>;
    /** Type of event. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    type?: string | Expression<string>;
    /** Whether to list only upcoming events
     * @default true
     */
    upcoming_events_only?: boolean | Expression<boolean>;
  };
};

export type CoinGeckoV1EventGetAllNode = {
  type: 'n8n-nodes-base.coinGecko';
  version: 1;
  config: NodeConfig<CoinGeckoV1EventGetAllParams>;
};
