// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleBusinessProfile/v1/resource_post/operation_create.ts
/**
 * Google Business Profile Node - Version 1
 * Discriminator: resource=post, operation=create
 */


interface Credentials {
  googleBusinessProfileOAuth2Api: CredentialReference;
}

/** Create a new post on Google Business Profile */
export type GoogleBusinessProfileV1PostCreateParams = {
  resource: 'post';
  operation: 'create';
/**
 * The Google Business Profile account
 * @default {"mode":"list","value":""}
 */
    account?: { __rl: true; mode: 'list' | 'name'; value: string; cachedResultName?: string };
/**
 * The specific location or business associated with the account
 * @default {"mode":"list","value":""}
 */
    location?: { __rl: true; mode: 'list' | 'name'; value: string; cachedResultName?: string };
/**
 * The type of post to create (standard, event, offer, or alert)
 * @default STANDARD
 */
    postType?: 'STANDARD' | 'EVENT' | 'OFFER' | 'ALERT' | Expression<string>;
/**
 * The main text of the post
 */
    summary?: string | Expression<string> | PlaceholderValue;
/**
 * E.g. Sales this week.
 * @displayOptions.show { postType: ["EVENT"] }
 */
    title?: string | Expression<string> | PlaceholderValue;
/**
 * The start date and time of the event
 * @displayOptions.show { postType: ["EVENT"] }
 */
    startDateTime?: string | Expression<string>;
/**
 * The end date and time of the event
 * @displayOptions.show { postType: ["EVENT"] }
 */
    endDateTime?: string | Expression<string>;
/**
 * The start date of the offer
 * @displayOptions.show { postType: ["OFFER"] }
 */
    startDate?: string | Expression<string> | PlaceholderValue;
/**
 * The end date of the offer
 * @displayOptions.show { postType: ["OFFER"] }
 */
    endDate?: string | Expression<string> | PlaceholderValue;
/**
 * The sub-type of the alert
 * @displayOptions.show { postType: ["ALERT"] }
 * @default COVID_19
 */
    alertType?: 'COVID_19' | Expression<string>;
/**
 * Options
 * @default {}
 */
    additionalOptions?: {
    /** The language code of the post content. &lt;a href="https://cloud.google.com/translate/docs/languages" target="_blank"&gt;More info&lt;/a&gt;.
     */
    languageCode?: string | Expression<string> | PlaceholderValue;
    /** The type of call to action
     * @displayOptions.show { /postType: ["STANDARD", "EVENT", "ALERT"] }
     * @default ACTION_TYPE_UNSPECIFIED
     */
    callToActionType?: 'ACTION_TYPE_UNSPECIFIED' | 'BOOK' | 'CALL' | 'LEARN_MORE' | 'ORDER' | 'SHOP' | 'SIGN_UP' | Expression<string>;
    /** The URL that users are sent to when clicking through the promotion
     * @displayOptions.show { /postType: ["STANDARD", "EVENT", "ALERT"] }
     */
    url?: string | Expression<string> | PlaceholderValue;
    /** The coupon code for the offer
     * @displayOptions.show { /postType: ["OFFER"] }
     */
    couponCode?: string | Expression<string> | PlaceholderValue;
    /** Link to redeem the offer
     * @displayOptions.show { /postType: ["OFFER"] }
     */
    redeemOnlineUrl?: string | Expression<string> | PlaceholderValue;
    /** The terms and conditions of the offer
     * @displayOptions.show { /postType: ["OFFER"] }
     */
    termsConditions?: string | Expression<string> | PlaceholderValue;
  };
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type GoogleBusinessProfileV1PostCreateNode = {
  type: 'n8n-nodes-base.googleBusinessProfile';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GoogleBusinessProfileV1PostCreateParams>;
};
