// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleBusinessProfile/v1/resource_post/operation_update.schema.js
/**
 * Google Business Profile Node - Version 1 - Zod Schema
 * Discriminator: resource=post, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('post').default('post'),
      operation: z.literal('update'),
      account: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('name')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      location: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('name')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      post: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('name')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      additionalOptions: z.object({ summary: stringOrExpression.optional(), languageCode: stringOrExpression.optional(), callToActionType: z.union([z.literal('ACTION_TYPE_UNSPECIFIED'), z.literal('BOOK'), z.literal('GET_OFFER'), z.literal('LEARN_MORE'), z.literal('ORDER'), z.literal('SHOP'), z.literal('SIGN_UP'), expressionSchema]).optional(), url: stringOrExpression.optional(), startDateTime: stringOrExpression.optional(), endDateTime: stringOrExpression.optional(), title: stringOrExpression.optional(), startDate: stringOrExpression.optional(), endDate: stringOrExpression.optional(), couponCode: stringOrExpression.optional(), redeemOnlineUrl: stringOrExpression.optional(), termsConditions: stringOrExpression.optional() }).optional(),
      requestOptions: z.object({ batching: z.unknown().optional(), allowUnauthorizedCerts: z.boolean().optional(), proxy: stringOrExpression.optional(), timeout: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
