// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleBusinessProfile/v1/resource_post/operation_get.ts
/**
 * Google Business Profile Node - Version 1
 * Discriminator: resource=post, operation=get
 */


interface Credentials {
  googleBusinessProfileOAuth2Api: CredentialReference;
}

/** Retrieve details of a specific post */
export type GoogleBusinessProfileV1PostGetParams = {
  resource: 'post';
  operation: 'get';
/**
 * The Google Business Profile account
 * @default {"mode":"list","value":""}
 */
    account?: { __rl: true; mode: 'list' | 'name'; value: string; cachedResultName?: string };
/**
 * The specific location or business associated with the account
 * @default {"mode":"list","value":""}
 */
    location?: { __rl: true; mode: 'list' | 'name'; value: string; cachedResultName?: string };
/**
 * Select the post to retrieve its details
 * @default {"mode":"list","value":""}
 */
    post?: { __rl: true; mode: 'list' | 'name'; value: string; cachedResultName?: string };
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type GoogleBusinessProfileV1PostGetNode = {
  type: 'n8n-nodes-base.googleBusinessProfile';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GoogleBusinessProfileV1PostGetParams>;
};
