// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/wekan/v1/resource_board/operation_create.ts
/**
 * Wekan Node - Version 1
 * Discriminator: resource=board, operation=create
 */


interface Credentials {
  wekanApi: CredentialReference;
}

/** Create a new board */
export type WekanV1BoardCreateParams = {
  resource: 'board';
  operation: 'create';
/**
 * The title of the board
 */
    title?: string | Expression<string> | PlaceholderValue;
/**
 * The user ID in Wekan. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    owner?: string | Expression<string>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Whether to set the board active
     * @default false
     */
    isActive?: boolean | Expression<boolean>;
    /** Whether to set the owner an admin of the board
     * @default false
     */
    isAdmin?: boolean | Expression<boolean>;
    /** The color of the board
     */
    color?: 'belize' | 'midnight' | 'nephritis' | 'pomegranate' | 'pumpkin' | 'wisteria' | Expression<string>;
    /** Whether to enable comments
     * @default false
     */
    isCommentOnly?: boolean | Expression<boolean>;
    /** Whether to disable comments
     * @default false
     */
    isNoComments?: boolean | Expression<boolean>;
    /** Set the board permission
     * @default private
     */
    permission?: 'private' | 'public' | Expression<string>;
    /** Whether to only move cards, assign himself to card and comment
     * @default false
     */
    isWorker?: boolean | Expression<boolean>;
  };
};

export type WekanV1BoardCreateNode = {
  type: 'n8n-nodes-base.wekan';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<WekanV1BoardCreateParams>;
};
