// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/wekan/v1/resource_board/operation_create.schema.js
/**
 * Wekan Node - Version 1 - Zod Schema
 * Discriminator: resource=board, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('board'),
      operation: z.literal('create').default('create'),
      title: stringOrExpression.optional(),
      owner: stringOrExpression.optional(),
      additionalFields: z.object({ isActive: booleanOrExpression.optional(), isAdmin: booleanOrExpression.optional(), color: z.union([z.literal('belize'), z.literal('midnight'), z.literal('nephritis'), z.literal('pomegranate'), z.literal('pumpkin'), z.literal('wisteria'), expressionSchema]).optional(), isCommentOnly: booleanOrExpression.optional(), isNoComments: booleanOrExpression.optional(), permission: z.union([z.literal('private'), z.literal('public'), expressionSchema]).optional(), isWorker: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
