// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/nasa/v1/resource_earth_imagery/operation_get.ts
/**
 * NASA Node - Version 1
 * Discriminator: resource=earthImagery, operation=get
 */


interface Credentials {
  nasaApi: CredentialReference;
}

/** Get the Astronomy Picture of the Day */
export type NasaV1EarthImageryGetParams = {
  resource: 'earthImagery';
  operation: 'get';
/**
 * Latitude for the location of the image
 */
    lat?: number | Expression<number>;
/**
 * Longitude for the location of the image
 */
    lon?: number | Expression<number>;
/**
 * Put Output File in Field
 * @hint The name of the output binary field to put the file in
 * @default data
 */
    binaryPropertyName?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Date of the image
     */
    date?: string | Expression<string>;
    /** Width and height of the image in degrees
     */
    dim?: number | Expression<number>;
  };
};

export type NasaV1EarthImageryGetNode = {
  type: 'n8n-nodes-base.nasa';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<NasaV1EarthImageryGetParams>;
};
