// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/contentful/v1/resource_entry/operation_get.ts
/**
 * Contentful Node - Version 1
 * Discriminator: resource=entry, operation=get
 */


interface Credentials {
  contentfulApi: CredentialReference;
}

export type ContentfulV1EntryGetParams = {
  resource: 'entry';
  operation: 'get';
/**
 * Pick where your data comes from, delivery or preview API
 * @default deliveryApi
 */
    source?: 'deliveryApi' | 'previewApi' | Expression<string>;
/**
 * The ID for the Contentful environment (e.g. master, staging, etc.). Depending on your plan, you might not have environments. In that case use "master".
 * @default master
 */
    environmentId?: string | Expression<string> | PlaceholderValue;
/**
 * Entry ID
 */
    entryId?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Whether the data should be returned RAW instead of parsed
     * @default false
     */
    rawData?: boolean | Expression<boolean>;
  };
};

export type ContentfulV1EntryGetNode = {
  type: 'n8n-nodes-base.contentful';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ContentfulV1EntryGetParams>;
};
