// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/contentful/v1/resource_entry/operation_get_all.schema.js
/**
 * Contentful Node - Version 1 - Zod Schema
 * Discriminator: resource=entry, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('entry').default('entry'),
      operation: z.literal('getAll'),
      source: z.union([z.literal('deliveryApi'), z.literal('previewApi'), expressionSchema]).optional(),
      environmentId: stringOrExpression.optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      additionalFields: z.object({ content_type: stringOrExpression.optional(), equal: stringOrExpression.optional(), exclude: stringOrExpression.optional(), exist: stringOrExpression.optional(), select: stringOrExpression.optional(), include: stringOrExpression.optional(), notEqual: stringOrExpression.optional(), order: stringOrExpression.optional(), query: stringOrExpression.optional(), rawData: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
