// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftExcel/v1/resource_table/operation_lookup.schema.js
/**
 * Microsoft Excel Node - Version 1 - Zod Schema
 * Discriminator: resource=table, operation=lookup
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('table'),
      operation: z.literal('lookup'),
      workbook: stringOrExpression.optional(),
      worksheet: stringOrExpression.optional(),
      table: stringOrExpression.optional(),
      lookupColumn: stringOrExpression.optional(),
      lookupValue: stringOrExpression.optional(),
      options: z.object({ returnAllMatches: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
