// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/serviceNow/v1/resource_incident/operation_update.schema.js
/**
 * ServiceNow Node - Version 1 - Zod Schema
 * Discriminator: resource=incident, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('incident'),
      operation: z.literal('update'),
      authentication: z.union([z.literal('basicAuth'), z.literal('oAuth2'), expressionSchema]).optional(),
      id: stringOrExpression.optional(),
      updateFields: z.object({ assigned_to: stringOrExpression.optional(), assignment_group: stringOrExpression.optional(), business_service: stringOrExpression.optional(), caller_id: stringOrExpression.optional(), category: stringOrExpression.optional(), close_notes: stringOrExpression.optional(), cmdb_ci: z.array(z.string()).optional(), contact_type: z.union([z.literal('email'), z.literal('phone'), z.literal('self-service'), z.literal('walk-in'), expressionSchema]).optional(), description: stringOrExpression.optional(), impact: z.union([z.literal(3), z.literal(2), z.literal(1), expressionSchema]).optional(), close_code: stringOrExpression.optional(), hold_reason: stringOrExpression.optional(), state: stringOrExpression.optional(), subcategory: stringOrExpression.optional(), urgency: z.union([z.literal(3), z.literal(2), z.literal(1), expressionSchema]).optional(), work_notes: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
