// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/twitter/v1/resource_tweet/operation_create.schema.js
/**
 * X (Formerly Twitter) Node - Version 1 - Zod Schema
 * Discriminator: resource=tweet, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('tweet').default('tweet'),
      operation: z.literal('create').default('create'),
      text: stringOrExpression.optional(),
      additionalFields: z.object({ attachments: stringOrExpression.optional(), displayCoordinates: booleanOrExpression.optional(), inReplyToStatusId: stringOrExpression.optional(), locationFieldsUi: z.unknown().optional(), possiblySensitive: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
