// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/freshservice/v1/resource_announcement/operation_create.schema.js
/**
 * Freshservice Node - Version 1 - Zod Schema
 * Discriminator: resource=announcement, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('announcement'),
      operation: z.literal('create').default('create'),
      title: stringOrExpression.optional(),
      bodyHtml: stringOrExpression.optional(),
      visibility: z.union([z.literal('agents_only'), z.literal('grouped_visibility'), z.literal('everyone'), expressionSchema]).optional(),
      visibleFrom: stringOrExpression.optional(),
      additionalFields: z.object({ additional_emails: stringOrExpression.optional(), departments: z.array(z.string()).optional(), visible_from: stringOrExpression.optional(), visible_till: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
