// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleAnalytics/v2/resource_user_activity/operation_search.ts
/**
 * Google Analytics Node - Version 2
 * Discriminator: resource=userActivity, operation=search
 */


interface Credentials {
  googleAnalyticsOAuth2: CredentialReference;
}

/** Return user activity data */
export type GoogleAnalyticsV2UserActivitySearchParams = {
  resource: 'userActivity';
  operation: 'search';
/**
 * The view from Google Analytics. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 * @hint If there's nothing here, try changing the 'Property type' field above
 */
    viewId?: string | Expression<string>;
/**
 * ID of a user
 */
    userId?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Type of activites requested
     * @default []
     */
    activityTypes?: Array<'ECOMMERCE' | 'EVENT' | 'GOAL' | 'PAGEVIEW' | 'SCREENVIEW'>;
  };
};

export type GoogleAnalyticsV2UserActivitySearchNode = {
  type: 'n8n-nodes-base.googleAnalytics';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<GoogleAnalyticsV2UserActivitySearchParams>;
};
