// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/zendesk/v1/resource_ticket/operation_get_all.schema.js
/**
 * Zendesk Node - Version 1 - Zod Schema
 * Discriminator: resource=ticket, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('ticket').default('ticket'),
      operation: z.literal('getAll'),
      authentication: z.union([z.literal('apiToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      ticketType: z.union([z.literal('regular'), z.literal('suspended'), expressionSchema]).optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      options: z.object({ group: stringOrExpression.optional(), query: stringOrExpression.optional(), sortBy: z.union([z.literal('created_at'), z.literal('priority'), z.literal('status'), z.literal('ticket_type'), z.literal('updated_at'), expressionSchema]).optional(), sortOrder: z.union([z.literal('asc'), z.literal('desc'), expressionSchema]).optional(), status: z.union([z.literal('closed'), z.literal('new'), z.literal('hold'), z.literal('open'), z.literal('pending'), z.literal('solved'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
