// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/zendesk/v1/resource_ticket/operation_create.schema.js
/**
 * Zendesk Node - Version 1 - Zod Schema
 * Discriminator: resource=ticket, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('ticket').default('ticket'),
      operation: z.literal('create').default('create'),
      authentication: z.union([z.literal('apiToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      description: stringOrExpression.optional(),
      jsonParameters: booleanOrExpression.optional(),
      additionalFields: resolveSchema({ parameters, schema: z.object({ customFieldsUi: z.unknown().optional(), externalId: stringOrExpression.optional(), group: stringOrExpression.optional(), recipient: stringOrExpression.optional(), status: z.union([z.literal('closed'), z.literal('new'), z.literal('hold'), z.literal('open'), z.literal('pending'), z.literal('solved'), expressionSchema]).optional(), subject: stringOrExpression.optional(), tags: z.array(z.string()).optional(), type: z.union([z.literal('question'), z.literal('incident'), z.literal('problem'), z.literal('task'), expressionSchema]).optional() }), required: false, displayOptions: {"show":{"jsonParameters":[false]}}, defaults: {"jsonParameters":false} }),
      additionalFieldsJson: resolveSchema({ parameters, schema: z.union([iDataObjectSchema, z.string()]), required: false, displayOptions: {"show":{"jsonParameters":[true]}}, defaults: {"jsonParameters":false} }),
    }).optional(),
  });
};
