// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/zendesk/v1/resource_ticket/operation_get_all.ts
/**
 * Zendesk Node - Version 1
 * Discriminator: resource=ticket, operation=getAll
 */


interface Credentials {
  zendeskApi: CredentialReference;
  zendeskOAuth2Api: CredentialReference;
}

/** Tickets are the means through which your end users (customers) communicate with agents in Zendesk Support */
export type ZendeskV1TicketGetAllParams = {
  resource: 'ticket';
  operation: 'getAll';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * Ticket Type
 * @default regular
 */
    ticketType?: 'regular' | 'suspended' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** The group to search. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     * @displayOptions.show { /ticketType: ["regular"] }
     */
    group?: string | Expression<string>;
    /** &lt;a href="https://developer.zendesk.com/api-reference/ticketing/ticket-management/search/#syntax-examples"&gt;Query syntax&lt;/a&gt; to search tickets
     * @displayOptions.show { /ticketType: ["regular"] }
     */
    query?: string | Expression<string> | PlaceholderValue;
    /** Defaults to sorting by relevance
     * @default updated_at
     */
    sortBy?: 'created_at' | 'priority' | 'status' | 'ticket_type' | 'updated_at' | Expression<string>;
    /** Sort Order
     * @default asc
     */
    sortOrder?: 'asc' | 'desc' | Expression<string>;
    /** The state of the ticket
     * @displayOptions.show { /ticketType: ["regular"] }
     */
    status?: 'closed' | 'new' | 'hold' | 'open' | 'pending' | 'solved' | Expression<string>;
  };
};

export type ZendeskV1TicketGetAllOutput = {
  allow_attachments?: boolean;
  allow_channelback?: boolean;
  brand_id?: number;
  collaborator_ids?: Array<number>;
  created_at?: string;
  custom_fields?: Array<{
    id?: number;
  }>;
  custom_status_id?: number;
  description?: string;
  email_cc_ids?: Array<number>;
  encoded_id?: string;
  fields?: Array<{
    id?: number;
  }>;
  follower_ids?: Array<number>;
  followup_ids?: Array<number>;
  forum_topic_id?: null;
  from_messaging_channel?: boolean;
  generated_timestamp?: number;
  has_incidents?: boolean;
  id?: number;
  is_public?: boolean;
  problem_id?: null;
  requester_id?: number;
  result_type?: string;
  satisfaction_rating?: {
    score?: string;
  };
  status?: string;
  submitter_id?: number;
  tags?: Array<string>;
  updated_at?: string;
  url?: string;
  via?: {
    channel?: string;
  };
};

export type ZendeskV1TicketGetAllNode = {
  type: 'n8n-nodes-base.zendesk';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ZendeskV1TicketGetAllParams>;
  output?: Items<ZendeskV1TicketGetAllOutput>;
};
