// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedrive/v1/resource_activity/operation_update.schema.js
/**
 * Pipedrive Node - Version 1 - Zod Schema
 * Discriminator: resource=activity, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('activity'),
      operation: z.literal('update'),
      authentication: z.union([z.literal('apiToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      activityId: numberOrExpression.optional(),
      updateFields: z.object({ busy_flag: booleanOrExpression.optional(), deal_id: numberOrExpression.optional(), due_date: stringOrExpression.optional(), done: z.union([z.literal('0'), z.literal('1'), expressionSchema]).optional(), note: stringOrExpression.optional(), org_id: stringOrExpression.optional(), person_id: numberOrExpression.optional(), public_description: stringOrExpression.optional(), subject: stringOrExpression.optional(), type: stringOrExpression.optional(), user_id: stringOrExpression.optional(), customProperties: z.unknown().optional() }).optional(),
      encodeProperties: booleanOrExpression.optional(),
    }).optional(),
  });
};
