// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedrive/v1/resource_activity/operation_update.ts
/**
 * Pipedrive Node - Version 1
 * Discriminator: resource=activity, operation=update
 */


interface Credentials {
  pipedriveApi: CredentialReference;
  pipedriveOAuth2Api: CredentialReference;
}

/** Update an activity */
export type PipedriveV1ActivityUpdateParams = {
  resource: 'activity';
  operation: 'update';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * ID of the activity to update
 * @default 0
 */
    activityId?: number | Expression<number>;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** Whether the user is set to busy during the activity
     * @default false
     */
    busy_flag?: boolean | Expression<boolean>;
    /** ID of the deal this activity will be associated with
     * @default 0
     */
    deal_id?: number | Expression<number>;
    /** Due Date to activity be done YYYY-MM-DD
     */
    due_date?: string | Expression<string>;
    /** Whether the activity is done or not
     * @default 0
     */
    done?: '0' | '1' | Expression<string>;
    /** Note of the activity (HTML format)
     */
    note?: string | Expression<string> | PlaceholderValue;
    /** ID of the organization this activity will be associated with. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    org_id?: string | Expression<string>;
    /** ID of the person this activity will be associated with
     * @default 0
     */
    person_id?: number | Expression<number>;
    /** Additional details about the activity that is synced to your external calendar
     */
    public_description?: string | Expression<string> | PlaceholderValue;
    /** The subject of the activity
     */
    subject?: string | Expression<string> | PlaceholderValue;
    /** Type of the activity like "call", "meeting", etc
     */
    type?: string | Expression<string> | PlaceholderValue;
    /** ID of the active user whom the activity will be assigned to. If omitted, the activity will be assigned to the authorized user. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    user_id?: string | Expression<string>;
    /** Adds a custom property to set also values which have not been predefined
     * @default {}
     */
    customProperties?: {
        /** Property
     */
    property?: Array<{
      /** Name of the property to set
       */
      name?: string | Expression<string> | PlaceholderValue;
      /** Value of the property to set
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
  };
/**
 * By default do custom properties have to be set as ID instead of their actual name. Also option fields have to be set as ID instead of their actual value. If this option gets set they get automatically encoded.
 * @default false
 */
    encodeProperties?: boolean | Expression<boolean>;
};

export type PipedriveV1ActivityUpdateOutput = {
  active_flag?: boolean;
  add_time?: string;
  assigned_to_user_id?: number;
  busy_flag?: boolean;
  company_id?: number;
  conference_meeting_id?: null;
  created_by_user_id?: number;
  done?: boolean;
  due_date?: string;
  due_time?: string;
  duration?: string;
  id?: number;
  is_recurring?: null;
  last_notification_time?: null;
  last_notification_user_id?: null;
  lead?: null;
  lead_id?: null;
  lead_title?: null;
  location_admin_area_level_1?: null;
  location_admin_area_level_2?: null;
  location_country?: null;
  location_formatted_address?: null;
  location_locality?: null;
  location_postal_code?: null;
  location_route?: null;
  location_street_number?: null;
  location_sublocality?: null;
  location_subpremise?: null;
  marked_as_done_time?: string;
  notification_language_id?: null;
  original_start_time?: null;
  owner_name?: string;
  private?: boolean;
  rec_master_activity_id?: null;
  rec_rule?: null;
  rec_rule_extension?: null;
  reference_id?: null;
  series?: null;
  subject?: string;
  type?: string;
  type_name?: string;
  update_time?: string;
  update_user_id?: number;
  user_id?: number;
};

export type PipedriveV1ActivityUpdateNode = {
  type: 'n8n-nodes-base.pipedrive';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PipedriveV1ActivityUpdateParams>;
  output?: Items<PipedriveV1ActivityUpdateOutput>;
};
