// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedrive/v1/resource_activity/operation_get.ts
/**
 * Pipedrive Node - Version 1
 * Discriminator: resource=activity, operation=get
 */


interface Credentials {
  pipedriveApi: CredentialReference;
  pipedriveOAuth2Api: CredentialReference;
}

/** Get data of an activity */
export type PipedriveV1ActivityGetParams = {
  resource: 'activity';
  operation: 'get';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * ID of the activity to get
 * @default 0
 */
    activityId?: number | Expression<number>;
/**
 * By default do custom properties get returned only as ID instead of their actual name. Also option fields contain only the ID instead of their actual value. If this option gets set they get automatically resolved.
 * @default false
 */
    resolveProperties?: boolean | Expression<boolean>;
};

export type PipedriveV1ActivityGetOutput = {
  active_flag?: boolean;
  'Add time'?: string;
  'Apartment/suite no of Location'?: null;
  'Assigned to user'?: number;
  assigned_to_user_id?: number;
  'City/town/village/locality of Location'?: null;
  company_id?: number;
  conference_meeting_id?: null;
  'Contact person'?: number;
  'Country of Location'?: null;
  Creator?: number;
  'District/sublocality of Location'?: null;
  Done?: string;
  'Due date'?: string;
  'Due time'?: string;
  Duration?: string;
  'Free/busy'?: string;
  'Full/combined address of Location'?: null;
  'House number of Location'?: null;
  ID?: number;
  is_recurring?: null;
  'Last notification time'?: null;
  last_notification_user_id?: null;
  lead?: null;
  Lead?: null;
  lead_title?: null;
  Location?: null;
  'Marked as done time'?: string;
  Note?: string;
  Organisation?: number;
  original_start_time?: null;
  owner_name?: string;
  Priority?: null;
  private?: boolean;
  rec_master_activity_id?: null;
  rec_rule?: null;
  rec_rule_extension?: null;
  reference_id?: null;
  'Region of Location'?: null;
  series?: null;
  'State/county of Location'?: null;
  'Street/road name of Location'?: null;
  Subject?: string;
  Type?: string;
  type_name?: string;
  'Update time'?: string;
  'ZIP/Postal code of Location'?: null;
};

export type PipedriveV1ActivityGetNode = {
  type: 'n8n-nodes-base.pipedrive';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PipedriveV1ActivityGetParams>;
  output?: Items<PipedriveV1ActivityGetOutput>;
};
