// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedrive/v1/resource_activity/operation_create.ts
/**
 * Pipedrive Node - Version 1
 * Discriminator: resource=activity, operation=create
 */


interface Credentials {
  pipedriveApi: CredentialReference;
  pipedriveOAuth2Api: CredentialReference;
}

/** Create an activity */
export type PipedriveV1ActivityCreateParams = {
  resource: 'activity';
  operation: 'create';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * The subject of the activity to create
 */
    subject?: string | Expression<string> | PlaceholderValue;
/**
 * Whether the activity is done or not
 * @default 0
 */
    done?: '0' | '1' | Expression<string>;
/**
 * Type of the activity like "call", "meeting", etc
 */
    type?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** ID of the deal this activity will be associated with
     * @default 0
     */
    deal_id?: number | Expression<number>;
    /** Due Date to activity be done YYYY-MM-DD
     */
    due_date?: string | Expression<string>;
    /** Note of the activity (HTML format)
     */
    note?: string | Expression<string> | PlaceholderValue;
    /** ID of the organization this activity will be associated with. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    org_id?: string | Expression<string>;
    /** ID of the person this activity will be associated with
     * @default 0
     */
    person_id?: number | Expression<number>;
    /** ID of the active user whom the activity will be assigned to. If omitted, the activity will be assigned to the authorized user. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    user_id?: string | Expression<string>;
    /** Adds a custom property to set also values which have not been predefined
     * @default {}
     */
    customProperties?: {
        /** Property
     */
    property?: Array<{
      /** Name of the property to set
       */
      name?: string | Expression<string> | PlaceholderValue;
      /** Value of the property to set
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
  };
};

export type PipedriveV1ActivityCreateOutput = {
  active_flag?: boolean;
  add_time?: string;
  assigned_to_user_id?: number;
  attendees?: null;
  busy_flag?: boolean;
  calendar_sync_include_context?: null;
  company_id?: number;
  conference_meeting_client?: null;
  conference_meeting_id?: null;
  conference_meeting_url?: null;
  created_by_user_id?: number;
  done?: boolean;
  due_date?: string;
  due_time?: string;
  duration?: string;
  id?: number;
  is_recurring?: null;
  last_notification_time?: null;
  last_notification_user_id?: null;
  location?: null;
  location_admin_area_level_1?: null;
  location_admin_area_level_2?: null;
  location_country?: null;
  location_formatted_address?: null;
  location_locality?: null;
  location_postal_code?: null;
  location_route?: null;
  location_street_number?: null;
  location_sublocality?: null;
  location_subpremise?: null;
  marked_as_done_time?: string;
  notification_language_id?: null;
  original_start_time?: null;
  owner_name?: string;
  priority?: null;
  private?: boolean;
  public_description?: null;
  rec_master_activity_id?: null;
  rec_rule?: null;
  rec_rule_extension?: null;
  reference_id?: null;
  reference_type?: null;
  series?: null;
  source_timezone?: null;
  subject?: string;
  type?: string;
  type_name?: string;
  update_time?: string;
  update_user_id?: null;
  user_id?: number;
};

export type PipedriveV1ActivityCreateNode = {
  type: 'n8n-nodes-base.pipedrive';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PipedriveV1ActivityCreateParams>;
  output?: Items<PipedriveV1ActivityCreateOutput>;
};
