// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedrive/v1/resource_activity/operation_get_all.ts
/**
 * Pipedrive Node - Version 1
 * Discriminator: resource=activity, operation=getAll
 */


interface Credentials {
  pipedriveApi: CredentialReference;
  pipedriveOAuth2Api: CredentialReference;
}

/** Get data of many activities */
export type PipedriveV1ActivityGetAllParams = {
  resource: 'activity';
  operation: 'getAll';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * By default do custom properties get returned only as ID instead of their actual name. Also option fields contain only the ID instead of their actual value. If this option gets set they get automatically resolved.
 * @default false
 */
    resolveProperties?: boolean | Expression<boolean>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Whether the Activity is done or not. 0 = Not done, 1 = Done. If omitted returns both Done and Not done activities.
     * @default false
     */
    done?: boolean | Expression<boolean>;
    /** Use the Activity due date where you wish to stop fetching Activities from. Insert due date in YYYY-MM-DD format.
     */
    end_date?: string | Expression<string>;
    /** The ID of the Filter to use (will narrow down results if used together with user_id parameter). Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    filterId?: string | Expression<string>;
    /** Use the Activity due date where you wish to begin fetching Activities from. Insert due date in YYYY-MM-DD format.
     */
    start_date?: string | Expression<string>;
    /** Type of the Activity. Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     * @default []
     */
    type?: string[];
    /** The ID of the User whose Activities will be fetched. If omitted, the User associated with the API token will be used. If 0, Activities for all company Users will be fetched based on the permission sets. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    user_id?: string | Expression<string>;
  };
};

export type PipedriveV1ActivityGetAllOutput = {
  active_flag?: boolean;
  add_time?: string;
  assigned_to_user_id?: number;
  busy_flag?: boolean;
  company_id?: number;
  created_by_user_id?: number;
  done?: boolean;
  due_date?: string;
  due_time?: string;
  duration?: string;
  id?: number;
  is_recurring?: null;
  last_notification_time?: null;
  last_notification_user_id?: null;
  location_admin_area_level_1?: null;
  location_admin_area_level_2?: null;
  location_country?: null;
  location_formatted_address?: null;
  location_locality?: null;
  location_postal_code?: null;
  location_route?: null;
  location_street_number?: null;
  location_sublocality?: null;
  location_subpremise?: null;
  notification_language_id?: null;
  original_start_time?: null;
  owner_name?: string;
  private?: boolean;
  reference_id?: null;
  series?: null;
  subject?: string;
  type?: string;
  update_time?: string;
  user_id?: number;
};

export type PipedriveV1ActivityGetAllNode = {
  type: 'n8n-nodes-base.pipedrive';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PipedriveV1ActivityGetAllParams>;
  output?: Items<PipedriveV1ActivityGetAllOutput>;
};
