// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/venafiTlsProtectDatacenter/v1/resource_certificate/operation_get_many.ts
/**
 * Venafi TLS Protect Datacenter Node - Version 1
 * Discriminator: resource=certificate, operation=getMany
 */


interface Credentials {
  venafiTlsProtectDatacenterApi: CredentialReference;
}

/** Retrieve many certificates */
export type VenafiTlsProtectDatacenterV1CertificateGetManyParams = {
  resource: 'certificate';
  operation: 'getMany';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Include one or more of the following certificate attributes in the return value
     * @default []
     */
    fields?: Array<'Issuer' | 'KeyAlgorithm' | 'KeySize' | 'Subject'>;
  };
};

export type VenafiTlsProtectDatacenterV1CertificateGetManyNode = {
  type: 'n8n-nodes-base.venafiTlsProtectDatacenter';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<VenafiTlsProtectDatacenterV1CertificateGetManyParams>;
};
