// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/venafiTlsProtectDatacenter/v1/resource_certificate/operation_create.ts
/**
 * Venafi TLS Protect Datacenter Node - Version 1
 * Discriminator: resource=certificate, operation=create
 */


interface Credentials {
  venafiTlsProtectDatacenterApi: CredentialReference;
}

/** Provision a new certificate */
export type VenafiTlsProtectDatacenterV1CertificateCreateParams = {
  resource: 'certificate';
  operation: 'create';
/**
 * The folder DN for the new certificate. If the value is missing, the folder name is the system default. If no system default is configured
 */
    PolicyDN?: string | Expression<string> | PlaceholderValue;
/**
 * The Common Name field for the certificate Subject (DN)
 */
    Subject?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** An array of one or more identities for certificate workflow approvers
     */
    Approvers?: string | Expression<string> | PlaceholderValue;
    /** Only required when no policy sets a CA template. The Distinguished Name (DN) of the Trust Protection Platform Certificate Authority Template object for enrolling the certificate.
     */
    CADN?: string | Expression<string> | PlaceholderValue;
    /** One of the following Certificate objects. Ignores any other value.
     */
    CertificateType?: 'Code Signing' | 'Device' | 'Server' | 'User' | Expression<string>;
    /** The City field for the certificate Subject DN. Specify a value when requesting a centrally generated CSR.
     */
    City?: string | Expression<string> | PlaceholderValue;
    /** An array of one or more identities for users or groups who receive notifications about events pertaining to the object
     * @default []
     */
    Contacts?: string | Expression<string> | PlaceholderValue;
    /** The Country field for the certificate Subject DN. Specify a value when requesting a centrally generated CSR.
     */
    Country?: string | Expression<string> | PlaceholderValue;
    /** Custom Fields
     * @default {}
     */
    customFieldsUi?: {
        /** Address
     */
    customFielsValues?: {
      /** Name
       */
      Name?: string | Expression<string> | PlaceholderValue;
      /** Values
       */
      Values?: string | Expression<string> | PlaceholderValue;
    };
  };
    /** The person, entity, or caller of this request. The default is Web SDK. Avoid overriding the default unless the caller is a significant enterprise application that is tightly integrated with Trust Protection Platform, such as a custom web portal. To add details, use Origin instead. If you want both attributes to have the same value, set only CreatedBy.
     * @default Web SDK
     */
    CreatedBy?: string | Expression<string> | PlaceholderValue;
    /** Devices
     * @default {}
     */
    Devices?: {
    /** An array of one or more Application objects to allow software, which runs on ObjectName, to use the same certificate
     */
    applications?: string | Expression<string> | PlaceholderValue;
    /** Required for Amazon EC2 provisioning. The unique cloud instance ID.
     */
    CloudInstanceID?: string | Expression<string> | PlaceholderValue;
    /** Required for Amazon EC2 provisioning. The geographic location where the cloud service instance resides. An instance in AWS can only exist in a single region.
     */
    CloudRegion?: string | Expression<string> | PlaceholderValue;
    /** Required for Amazon EC2 provisioning. AWS: An Amazon E2C cloud service. Requires you to install and configure the Cloud Instance Monitoring feature.
     */
    CloudService?: string | Expression<string> | PlaceholderValue;
    /** Maximum number of connections the device will accept from Trust Protection Platform
     * @default 0
     */
    ConcurrentConnectionLimit?: number | Expression<number>;
    /** An array of one or more identities who receive notifications for this device
     * @default []
     */
    Contacts?: string | Expression<string> | PlaceholderValue;
    /** The person or entity that is creating the device and any associated software applications for the device. Any value is accepted. Default is Web SDK.
     * @default Web SDK
     */
    CreatedBy?: string | Expression<string> | PlaceholderValue;
    /** The device credential
     */
    CredentialDN?: string | Expression<string> | PlaceholderValue;
    /** The description for this device
     */
    description?: string | Expression<string> | PlaceholderValue;
    /** For SSH keys. true: Enable known host key enforcement. false: Disable known host key enforcement.
     * @default true
     */
    EnforceKnownHost?: boolean | Expression<boolean>;
    /** The physical Fully Qualified Domain Name (FQDN) for the host or the IP address for a device
     */
    host?: string | Expression<string> | PlaceholderValue;
    /** The device host name or IP address for the certificate object in Trust Protection Platform. If the value is missing, the object name is the Subject.
     */
    ObjectName?: string | Expression<string> | PlaceholderValue;
    /** The port number to communicate with the device
     * @default 0
     */
    port?: number | Expression<number>;
    /** Use in conjunction with UseSudo. The DN that holds the password credential to be used if sudo is configured to prompt for a password when executing a command.
     * @default 0
     */
    SudoCredentialDN?: number | Expression<number>;
    /** The host directory path to hold temporary files during provisioning. For example /tmp/. The folder should have the necessary write permissions.
     */
    TempDirectory?: string | Expression<string> | PlaceholderValue;
    /** For Secure Shell (SSH) keys. The SSH server key fingerprint. If this value is set, and EnforceKnownHost is enabled, Trust Protection Platform will only successfully connect to the device if the hosts fingerprint matches this value.
     */
    TrustedFingerprint?: string | Expression<string> | PlaceholderValue;
    /** Use in conjunction with SudoCredentialDN. For cases where the device credentials require sudo privilege elevation to execute commands when installing the certificate on a Unix or Linux device: true: Execute commands using sudo when provisioning. false: Execute commands directly without using sudo.
     * @default false
     */
    UseSudo?: boolean | Expression<boolean>;
  };
    /** The setting to control whether manual intervention is required for certificate renewal
     * @default false
     */
    DisableAutomaticRenewal?: boolean | Expression<boolean>;
    /** For Elliptic Curve Cryptography (ECC), use this parameter in conjunction with KeyAlgorithm
     */
    EllipticCurve?: 'P256' | 'P384' | 'P521' | Expression<string>;
    /** The encryption algorithm for the public ke:
     */
    KeyAlgorithm?: 'RSA' | 'ECC' | Expression<string>;
    /** Use this parameter when KeyAlgorithm is RSA. The number of bits to allow for key generation.
     * @default 2048
     */
    KeyBitSize?: number | Expression<number>;
    /** The level of management that Trust Protection Platform applies to the certificate
     */
    ManagementType?: 'Enrollment' | 'Monitoring' | 'Provisioning' | 'Unassigned' | Expression<string>;
    /** Additional information, such as the name and version of the calling application, that describes the source of this enrollment, renewal, or provisioning request. The default is Web SDK.
     * @default Web SDK
     */
    origin?: string | Expression<string> | PlaceholderValue;
    /** The Organization field for the certificate Subject DN. Specify a value when the CSR centrally generates.
     */
    Organization?: string | Expression<string> | PlaceholderValue;
    /** The department or division within the organization that is responsible for maintaining the certificate
     */
    OrganizationalUnit?: string | Expression<string> | PlaceholderValue;
    /** The PKCS#10 Certificate Signing Request (CSR). Omit escape characters such as or . If this value is provided, any Subject DN fields and the KeyBitSize in the request are ignored.
     */
    PKCS10?: string | Expression<string> | PlaceholderValue;
    /** The action to control a previously disabled certificate
     * @default false
     */
    Reenable?: boolean | Expression<boolean>;
    /** The setting to control certificate processing
     * @default false
     */
    SetWorkToDo?: boolean | Expression<boolean>;
    /** The State field for the certificate Subject DN. Specify a value when requesting a centrally generated CSR.
     */
    State?: string | Expression<string> | PlaceholderValue;
    /** Subject Alt Names
     * @default {}
     */
    SubjectAltNamesUi?: {
        /** Subject Alt Name
     */
    SubjectAltNamesValues?: Array<{
      /** An integer that represents the kind of SAN
       */
      Typename?: 0 | 1 | 2 | 6 | 7 | Expression<number>;
      /** The SAN friendly name that corresponds to the Type or TypeName parameter. For example, if a TypeName is IPAddress, the Name value is a valid IP address.
       */
      name?: string | Expression<string> | PlaceholderValue;
    }>;
  };
  };
};

export type VenafiTlsProtectDatacenterV1CertificateCreateNode = {
  type: 'n8n-nodes-base.venafiTlsProtectDatacenter';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<VenafiTlsProtectDatacenterV1CertificateCreateParams>;
};
