// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/venafiTlsProtectDatacenter/v1/resource_certificate/operation_renew.ts
/**
 * Venafi TLS Protect Datacenter Node - Version 1
 * Discriminator: resource=certificate, operation=renew
 */


interface Credentials {
  venafiTlsProtectDatacenterApi: CredentialReference;
}

/** Renew a certificate */
export type VenafiTlsProtectDatacenterV1CertificateRenewParams = {
  resource: 'certificate';
  operation: 'renew';
/**
 * The Distinguished Name (DN) of the certificate to renew
 */
    certificateDN?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** The PKCS#10 Certificate Signing Request (CSR). Omit escape characters such as or . If this value is provided, any Subject DN fields and the KeyBitSize in the request are ignored.
     */
    PKCS10?: string | Expression<string> | PlaceholderValue;
    /** The action to control a previously disabled certificate
     * @default false
     */
    Reenable?: boolean | Expression<boolean>;
  };
};

export type VenafiTlsProtectDatacenterV1CertificateRenewNode = {
  type: 'n8n-nodes-base.venafiTlsProtectDatacenter';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<VenafiTlsProtectDatacenterV1CertificateRenewParams>;
};
