// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/venafiTlsProtectDatacenter/v1/resource_certificate/operation_download.schema.js
/**
 * Venafi TLS Protect Datacenter Node - Version 1 - Zod Schema
 * Discriminator: resource=certificate, operation=download
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('certificate').default('certificate'),
      operation: z.literal('download'),
      certificateDn: stringOrExpression.optional(),
      includePrivateKey: booleanOrExpression.optional(),
      password: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"includePrivateKey":[true]}}, defaults: {"includePrivateKey":false} }),
      binaryProperty: stringOrExpression.optional(),
      additionalFields: z.object({ IncludeChain: booleanOrExpression.optional(), RootFirstOrder: stringOrExpression.optional(), KeystorePassword: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
