// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/oura/v1/resource_summary/operation_get_activity.schema.js
/**
 * Oura Node - Version 1 - Zod Schema
 * Discriminator: resource=summary, operation=getActivity
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('summary').default('summary'),
      operation: z.literal('getActivity'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filters: z.object({ end: stringOrExpression.optional(), start: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
