// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/misp/v1/resource_user/operation_create.schema.js
/**
 * MISP Node - Version 1 - Zod Schema
 * Discriminator: resource=user, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('user'),
      operation: z.literal('create').default('create'),
      email: stringOrExpression.optional(),
      role_id: stringOrExpression.optional(),
      additionalFields: z.object({ gpgkey: stringOrExpression.optional(), invited_by: stringOrExpression.optional(), org_id: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
