// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/currents/v1/resource_spec_file/operation_get_all.schema.js
/**
 * Currents Node - Version 1 - Zod Schema
 * Discriminator: resource=specFile, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('specFile'),
      operation: z.literal('getAll').default('getAll'),
      projectId: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      dateStart: stringOrExpression.optional(),
      dateEnd: stringOrExpression.optional(),
      limit: numberOrExpression.optional(),
      filters: z.object({ authors: stringOrExpression.optional(), branches: stringOrExpression.optional(), groups: stringOrExpression.optional(), specNameFilter: stringOrExpression.optional(), tags: stringOrExpression.optional() }).optional(),
      options: z.object({ includeFailedInDuration: booleanOrExpression.optional(), order: z.union([z.literal('avgDuration'), z.literal('failedExecutions'), z.literal('failureRate'), z.literal('flakeRate'), z.literal('flakyExecutions'), z.literal('fullyReported'), z.literal('overallExecutions'), z.literal('suiteSize'), z.literal('timeoutExecutions'), z.literal('timeoutRate'), expressionSchema]).optional(), dir: z.union([z.literal('asc'), z.literal('desc'), expressionSchema]).optional(), page: numberOrExpression.optional() }).optional(),
      requestOptions: z.object({ batching: z.unknown().optional(), allowUnauthorizedCerts: z.boolean().optional(), proxy: stringOrExpression.optional(), timeout: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
