// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/github/v1/resource_repository/operation_list_popular_paths.ts
/**
 * GitHub Node - Version 1
 * Discriminator: resource=repository, operation=listPopularPaths
 */


interface Credentials {
  githubApi: CredentialReference;
  githubOAuth2Api: CredentialReference;
}

/** Get the top 10 popular content paths over the last 14 days */
export type GithubV1RepositoryListPopularPathsParams = {
  resource: 'repository';
  operation: 'listPopularPaths';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Repository Owner
 * @displayOptions.hide { operation: ["invite", "getUserIssues"] }
 * @default {"mode":"list","value":""}
 */
    owner?: { __rl: true; mode: 'list' | 'url' | 'name'; value: string; cachedResultName?: string };
/**
 * Repository Name
 * @displayOptions.hide { resource: ["user", "organization"], operation: ["getRepositories"] }
 * @default {"mode":"list","value":""}
 */
    repository?: { __rl: true; mode: 'list' | 'url' | 'name'; value: string; cachedResultName?: string };
};

export type GithubV1RepositoryListPopularPathsNode = {
  type: 'n8n-nodes-base.github';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GithubV1RepositoryListPopularPathsParams>;
};
