// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/github/v1/resource_repository/operation_get_issues.schema.js
/**
 * GitHub Node - Version 1 - Zod Schema
 * Discriminator: resource=repository, operation=getIssues
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('repository'),
      operation: z.literal('getIssues'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      owner: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('url'), z.literal('name')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      repository: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('url'), z.literal('name')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      getRepositoryIssuesFilters: z.object({ assignee: stringOrExpression.optional(), creator: stringOrExpression.optional(), mentioned: stringOrExpression.optional(), labels: stringOrExpression.optional(), since: stringOrExpression.optional(), state: z.union([z.literal('all'), z.literal('closed'), z.literal('open'), expressionSchema]).optional(), sort: z.union([z.literal('created'), z.literal('updated'), z.literal('comments'), expressionSchema]).optional(), direction: z.union([z.literal('asc'), z.literal('desc'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
