// @n8n/n8n-nodes-langchain/dist/nodes/vector_store/VectorStoreWeaviate/Weaviate.utils.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.createWeaviateClient = createWeaviateClient;
exports.parseCompositeFilter = parseCompositeFilter;
const n8n_workflow_1 = require("n8n-workflow");
const weaviate_client_1 = __importStar(require("weaviate-client"));
async function createWeaviateClient(credentials, timeout, proxies, skipInitChecks = false) {
    if (credentials.weaviate_cloud_endpoint) {
        const weaviateClient = await weaviate_client_1.default.connectToWeaviateCloud(credentials.weaviate_cloud_endpoint, {
            authCredentials: new weaviate_client_1.default.ApiKey(credentials.weaviate_api_key),
            timeout,
            skipInitChecks,
        });
        return weaviateClient;
    }
    else {
        const weaviateClient = await weaviate_client_1.default.connectToCustom({
            httpHost: credentials.custom_connection_http_host,
            httpPort: credentials.custom_connection_http_port,
            grpcHost: credentials.custom_connection_grpc_host,
            grpcPort: credentials.custom_connection_grpc_port,
            grpcSecure: credentials.custom_connection_grpc_secure,
            httpSecure: credentials.custom_connection_http_secure,
            authCredentials: credentials.weaviate_api_key
                ? new weaviate_client_1.default.ApiKey(credentials.weaviate_api_key)
                : undefined,
            timeout,
            proxies,
            skipInitChecks,
        });
        return weaviateClient;
    }
}
function buildFilter(filter) {
    const { path, operator } = filter;
    const property = weaviate_client_1.default.filter.byProperty(path[0]);
    switch (operator.toLowerCase()) {
        case 'equal':
            if (filter.valueString !== undefined)
                return property.equal(filter.valueString);
            if (filter.valueNumber !== undefined)
                return property.equal(filter.valueNumber);
            break;
        case 'like':
            if (filter.valueString === undefined) {
                throw new n8n_workflow_1.OperationalError("Missing 'valueString' for 'like' operator.");
            }
            return property.like(filter.valueString);
        case 'containsany':
            if (filter.valueTextArray === undefined) {
                throw new n8n_workflow_1.OperationalError("Missing 'valueTextArray' for 'containsAny' operator.");
            }
            return property.containsAny(filter.valueTextArray);
        case 'containsall':
            if (filter.valueTextArray === undefined) {
                throw new n8n_workflow_1.OperationalError("Missing 'valueTextArray' for 'containsAll' operator.");
            }
            return property.containsAll(filter.valueTextArray);
        case 'greaterthan':
            if (filter.valueNumber === undefined) {
                throw new n8n_workflow_1.OperationalError("Missing 'valueNumber' for 'greaterThan' operator.");
            }
            return property.greaterThan(filter.valueNumber);
        case 'lessthan':
            if (filter.valueNumber === undefined) {
                throw new n8n_workflow_1.OperationalError("Missing 'valueNumber' for 'lessThan' operator.");
            }
            return property.lessThan(filter.valueNumber);
        case 'isnull':
            if (filter.valueBoolean === undefined) {
                throw new n8n_workflow_1.OperationalError("Missing 'valueBoolean' for 'isNull' operator.");
            }
            return property.isNull(filter.valueBoolean);
        case 'withingeorange':
            if (!filter.valueGeoCoordinates) {
                throw new n8n_workflow_1.OperationalError("Missing 'valueGeoCoordinates' for 'withinGeoRange' operator.");
            }
            return property.withinGeoRange(filter.valueGeoCoordinates);
        default:
            throw new n8n_workflow_1.OperationalError(`Unsupported operator: ${operator}`);
    }
    throw new n8n_workflow_1.OperationalError(`No valid filter value provided for operator: ${operator}`);
}
function parseCompositeFilter(filter) {
    if (typeof filter === 'object' && ('AND' in filter || 'OR' in filter)) {
        if ('AND' in filter) {
            return weaviate_client_1.Filters.and(...filter.AND.map(buildFilter));
        }
        else if ('OR' in filter) {
            return weaviate_client_1.Filters.or(...filter.OR.map(buildFilter));
        }
    }
    return buildFilter(filter);
}
