// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftTeams/v1/resource_channel/operation_create.ts
/**
 * Microsoft Teams Node - Version 1
 * Discriminator: resource=channel, operation=create
 */


interface Credentials {
  microsoftTeamsOAuth2Api: CredentialReference;
}

/** Create a channel */
export type MicrosoftTeamsV1ChannelCreateParams = {
  resource: 'channel';
  operation: 'create';
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    teamId?: string | Expression<string>;
/**
 * Channel name as it will appear to the user in Microsoft Teams
 */
    name?: string | Expression<string> | PlaceholderValue;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Channel's description
     */
    description?: string | Expression<string> | PlaceholderValue;
    /** The type of the channel
     * @default standard
     */
    type?: 'private' | 'standard' | Expression<string>;
  };
};

export type MicrosoftTeamsV1ChannelCreateNode = {
  type: 'n8n-nodes-base.microsoftTeams';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MicrosoftTeamsV1ChannelCreateParams>;
};
