// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftTeams/v1/resource_channel/operation_create.schema.js
/**
 * Microsoft Teams Node - Version 1 - Zod Schema
 * Discriminator: resource=channel, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('channel').default('channel'),
      operation: z.literal('create').default('create'),
      teamId: stringOrExpression.optional(),
      name: stringOrExpression.optional(),
      options: z.object({ description: stringOrExpression.optional(), type: z.union([z.literal('private'), z.literal('standard'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
