// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mondayCom/v1/resource_board_item/operation_move.ts
/**
 * Monday.com Node - Version 1
 * Discriminator: resource=boardItem, operation=move
 */


interface Credentials {
  mondayComApi: CredentialReference;
  mondayComOAuth2Api: CredentialReference;
}

/** Move item to group */
export type MondayComV1BoardItemMoveParams = {
  resource: 'boardItem';
  operation: 'move';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    boardId?: string | Expression<string>;
/**
 * The item's ID
 */
    itemId?: string | Expression<string> | PlaceholderValue;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    groupId?: string | Expression<string>;
};

export type MondayComV1BoardItemMoveOutput = {
  id?: string;
};

export type MondayComV1BoardItemMoveNode = {
  type: 'n8n-nodes-base.mondayCom';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MondayComV1BoardItemMoveParams>;
  output?: Items<MondayComV1BoardItemMoveOutput>;
};
