// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mondayCom/v1/resource_board_item/operation_get.ts
/**
 * Monday.com Node - Version 1
 * Discriminator: resource=boardItem, operation=get
 */


interface Credentials {
  mondayComApi: CredentialReference;
  mondayComOAuth2Api: CredentialReference;
}

/** Get a board */
export type MondayComV1BoardItemGetParams = {
  resource: 'boardItem';
  operation: 'get';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Item's ID (Multiple can be added separated by comma)
 */
    itemId?: string | Expression<string> | PlaceholderValue;
};

export type MondayComV1BoardItemGetOutput = {
  column_values?: Array<{
    column?: {
      archived?: boolean;
      settings_str?: string;
      title?: string;
    };
    id?: string;
    type?: string;
  }>;
  created_at?: string;
  id?: string;
  name?: string;
  state?: string;
};

export type MondayComV1BoardItemGetNode = {
  type: 'n8n-nodes-base.mondayCom';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MondayComV1BoardItemGetParams>;
  output?: Items<MondayComV1BoardItemGetOutput>;
};
