// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mondayCom/v1/resource_board_item/operation_change_column_value.ts
/**
 * Monday.com Node - Version 1
 * Discriminator: resource=boardItem, operation=changeColumnValue
 */


interface Credentials {
  mondayComApi: CredentialReference;
  mondayComOAuth2Api: CredentialReference;
}

/** Change a column value for a board item */
export type MondayComV1BoardItemChangeColumnValueParams = {
  resource: 'boardItem';
  operation: 'changeColumnValue';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * The unique identifier of the board. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    boardId?: string | Expression<string>;
/**
 * The unique identifier of the item to change column of
 */
    itemId?: string | Expression<string> | PlaceholderValue;
/**
 * The column's unique identifier. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    columnId?: string | Expression<string>;
/**
 * The column value in JSON format. Documentation can be found &lt;a href="https://monday.com/developers/v2#mutations-section-columns-change-column-value"&gt;here&lt;/a&gt;.
 */
    value?: IDataObject | string | Expression<string>;
};

export type MondayComV1BoardItemChangeColumnValueOutput = {
  id?: string;
};

export type MondayComV1BoardItemChangeColumnValueNode = {
  type: 'n8n-nodes-base.mondayCom';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MondayComV1BoardItemChangeColumnValueParams>;
  output?: Items<MondayComV1BoardItemChangeColumnValueOutput>;
};
