// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mondayCom/v1/resource_board_item/operation_get_all.ts
/**
 * Monday.com Node - Version 1
 * Discriminator: resource=boardItem, operation=getAll
 */


interface Credentials {
  mondayComApi: CredentialReference;
  mondayComOAuth2Api: CredentialReference;
}

/** Get many boards */
export type MondayComV1BoardItemGetAllParams = {
  resource: 'boardItem';
  operation: 'getAll';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    boardId?: string | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    groupId?: string | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
};

export type MondayComV1BoardItemGetAllOutput = {
  column_values?: Array<{
    column?: {
      archived?: boolean;
      settings_str?: string;
      title?: string;
    };
    id?: string;
    type?: string;
  }>;
  created_at?: string;
  id?: string;
  name?: string;
  state?: string;
};

export type MondayComV1BoardItemGetAllNode = {
  type: 'n8n-nodes-base.mondayCom';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MondayComV1BoardItemGetAllParams>;
  output?: Items<MondayComV1BoardItemGetAllOutput>;
};
