// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mondayCom/v1/resource_board_item/operation_get_by_column_value.ts
/**
 * Monday.com Node - Version 1
 * Discriminator: resource=boardItem, operation=getByColumnValue
 */


interface Credentials {
  mondayComApi: CredentialReference;
  mondayComOAuth2Api: CredentialReference;
}

/** Get items by column value */
export type MondayComV1BoardItemGetByColumnValueParams = {
  resource: 'boardItem';
  operation: 'getByColumnValue';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * The unique identifier of the board. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    boardId?: string | Expression<string>;
/**
 * The column's unique identifier. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    columnId?: string | Expression<string>;
/**
 * The column value to search items by
 */
    columnValue?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
};

export type MondayComV1BoardItemGetByColumnValueOutput = {
  board?: {
    id?: string;
  };
  column_values?: Array<{
    column?: {
      archived?: boolean;
      settings_str?: string;
      title?: string;
    };
    id?: string;
    type?: string;
  }>;
  created_at?: string;
  id?: string;
  name?: string;
  state?: string;
};

export type MondayComV1BoardItemGetByColumnValueNode = {
  type: 'n8n-nodes-base.mondayCom';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MondayComV1BoardItemGetByColumnValueParams>;
  output?: Items<MondayComV1BoardItemGetByColumnValueOutput>;
};
