// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mondayCom/v1/resource_board_item/operation_create.schema.js
/**
 * Monday.com Node - Version 1 - Zod Schema
 * Discriminator: resource=boardItem, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('boardItem'),
      operation: z.literal('create').default('create'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      boardId: stringOrExpression.optional(),
      groupId: stringOrExpression.optional(),
      name: stringOrExpression.optional(),
      additionalFields: z.object({ columnValues: z.union([iDataObjectSchema, z.string()]).optional() }).optional(),
    }).optional(),
  });
};
