// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/discord/v2/resource_message/operation_get_all.schema.js
/**
 * Discord Node - Version 2 - Zod Schema
 * Discriminator: resource=message, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('message'),
      operation: z.literal('getAll'),
      authentication: z.union([z.literal('botToken'), z.literal('oAuth2'), z.literal('webhook'), expressionSchema]).optional(),
      guildId: resolveSchema({ parameters, schema: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('url'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]), required: false, displayOptions: {"show":{"authentication":["botToken","oAuth2"]}}, defaults: {"authentication":"botToken"} }),
      channelId: resolveSchema({ parameters, schema: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('url'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]), required: false, displayOptions: {"hide":{"authentication":["webhook"]}}, defaults: {"authentication":"botToken"} }),
      returnAll: resolveSchema({ parameters, schema: booleanOrExpression, required: false, displayOptions: {"hide":{"authentication":["webhook"]}}, defaults: {"authentication":"botToken"} }),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]},"hide":{"authentication":["webhook"]}}, defaults: {"returnAll":false,"authentication":"botToken"} }),
      options: resolveSchema({ parameters, schema: z.object({ simplify: booleanOrExpression.optional() }), required: false, displayOptions: {"hide":{"authentication":["webhook"]}}, defaults: {"authentication":"botToken"} }),
    }).optional(),
  });
};
