// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/discord/v2/resource_message/operation_get.ts
/**
 * Discord Node - Version 2
 * Discriminator: resource=message, operation=get
 */


interface Credentials {
  discordBotApi: CredentialReference;
  discordOAuth2Api: CredentialReference;
  discordWebhookApi?: CredentialReference;
}

/** Get a message in a channel */
export type DiscordV2MessageGetParams = {
  resource: 'message';
  operation: 'get';
  authentication?: 'botToken' | 'oAuth2' | 'webhook' | Expression<string>;
/**
 * Select the server (guild) that your bot is connected to
 * @displayOptions.show { authentication: ["botToken", "oAuth2"] }
 * @default {"mode":"list","value":""}
 */
    guildId?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
/**
 * Select the channel by name, URL, or ID
 * @displayOptions.hide { authentication: ["webhook"] }
 * @default {"mode":"list","value":""}
 */
    channelId?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
/**
 * The ID of the message
 * @displayOptions.hide { authentication: ["webhook"] }
 */
    messageId?: string | Expression<string> | PlaceholderValue;
/**
 * Options
 * @displayOptions.hide { authentication: ["webhook"] }
 * @default {}
 */
    options?: {
    /** Whether to return a simplified version of the response instead of the raw data
     * @default true
     */
    simplify?: boolean | Expression<boolean>;
  };
};

export type DiscordV2MessageGetOutput = {
  attachments?: Array<{
    content_scan_version?: number;
    content_type?: string;
    filename?: string;
    height?: number;
    id?: string;
    placeholder?: string;
    placeholder_version?: number;
    proxy_url?: string;
    size?: number;
    url?: string;
    width?: number;
  }>;
  author?: {
    accent_color?: null;
    banner?: null;
    banner_color?: null;
    collectibles?: null;
    discriminator?: string;
    flags?: number;
    id?: string;
    public_flags?: number;
    username?: string;
  };
  channel_id?: string;
  components?: Array<{
    components?: Array<{
      custom_id?: string;
      id?: number;
      label?: string;
      style?: number;
      type?: number;
    }>;
    id?: number;
    type?: number;
  }>;
  content?: string;
  embeds?: Array<{
    color?: number;
    content_scan_version?: number;
    description?: string;
    thumbnail?: {
      flags?: number;
      height?: number;
      placeholder?: string;
      placeholder_version?: number;
      proxy_url?: string;
      url?: string;
      width?: number;
    };
    title?: string;
    type?: string;
    url?: string;
  }>;
  flags?: number;
  id?: string;
  mention_everyone?: boolean;
  mention_roles?: Array<string>;
  mentions?: Array<{
    accent_color?: null;
    avatar_decoration_data?: null;
    banner?: null;
    banner_color?: null;
    clan?: null;
    discriminator?: string;
    flags?: number;
    id?: string;
    primary_guild?: null;
    public_flags?: number;
    username?: string;
  }>;
  pinned?: boolean;
  timestamp?: string;
  tts?: boolean;
  type?: number;
};

export type DiscordV2MessageGetNode = {
  type: 'n8n-nodes-base.discord';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<DiscordV2MessageGetParams>;
  output?: Items<DiscordV2MessageGetOutput>;
};
