// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/keap/v1/resource_ecommerce_order/operation_create.schema.js
/**
 * Keap Node - Version 1 - Zod Schema
 * Discriminator: resource=ecommerceOrder, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('ecommerceOrder'),
      operation: z.literal('create').default('create'),
      contactId: stringOrExpression.optional(),
      orderDate: stringOrExpression.optional(),
      orderTitle: stringOrExpression.optional(),
      orderType: z.union([z.literal('offline'), z.literal('online'), expressionSchema]).optional(),
      additionalFields: z.object({ leadAffiliateId: numberOrExpression.optional(), promoCodes: stringOrExpression.optional(), salesAffiliateId: numberOrExpression.optional() }).optional(),
      addressUi: z.object({ addressValues: z.object({ company: stringOrExpression.optional(), countryCode: stringOrExpression.optional(), firstName: stringOrExpression.optional(), middleName: stringOrExpression.optional(), lastName: stringOrExpression.optional(), line1: stringOrExpression.optional(), line2: stringOrExpression.optional(), locality: stringOrExpression.optional(), region: stringOrExpression.optional(), zipCode: stringOrExpression.optional(), zipFour: stringOrExpression.optional(), phone: stringOrExpression.optional() }).optional() }).optional(),
      orderItemsUi: z.object({ orderItemsValues: z.array(z.object({ description: stringOrExpression.optional(), price: numberOrExpression.optional(), 'product ID': numberOrExpression.optional(), quantity: numberOrExpression.optional() })).optional() }).optional(),
    }).optional(),
  });
};
