// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/keap/v1/resource_ecommerce_order/operation_get_all.schema.js
/**
 * Keap Node - Version 1 - Zod Schema
 * Discriminator: resource=ecommerceOrder, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('ecommerceOrder'),
      operation: z.literal('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      options: z.object({ since: stringOrExpression.optional(), until: stringOrExpression.optional(), paid: booleanOrExpression.optional(), order: stringOrExpression.optional(), contactId: numberOrExpression.optional(), productId: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
