// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/keap/v1/resource_ecommerce_order/operation_get_all.ts
/**
 * Keap Node - Version 1
 * Discriminator: resource=ecommerceOrder, operation=getAll
 */


interface Credentials {
  keapOAuth2Api: CredentialReference;
}

/** Retrieve many companies */
export type KeapV1EcommerceOrderGetAllParams = {
  resource: 'ecommerceOrder';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Date to start searching from
     */
    since?: string | Expression<string>;
    /** Date to search to
     */
    until?: string | Expression<string>;
    /** Paid
     * @default false
     */
    paid?: boolean | Expression<boolean>;
    /** Attribute to order items by
     */
    order?: string | Expression<string> | PlaceholderValue;
    /** Contact ID
     * @default 0
     */
    contactId?: number | Expression<number>;
    /** Product ID
     * @default 0
     */
    productId?: number | Expression<number>;
  };
};

export type KeapV1EcommerceOrderGetAllNode = {
  type: 'n8n-nodes-base.keap';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<KeapV1EcommerceOrderGetAllParams>;
};
