// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/cockpit/v1/resource_form/operation_submit.schema.js
/**
 * Cockpit Node - Version 1 - Zod Schema
 * Discriminator: resource=form, operation=submit
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('form'),
      operation: z.literal('submit'),
      form: stringOrExpression.optional(),
      jsonDataFields: booleanOrExpression.optional(),
      dataFieldsJson: resolveSchema({ parameters, schema: z.union([iDataObjectSchema, z.string()]), required: false, displayOptions: {"show":{"jsonDataFields":[true]}}, defaults: {"jsonDataFields":false} }),
      dataFieldsUi: resolveSchema({ parameters, schema: z.object({ field: z.array(z.object({ name: stringOrExpression.optional(), value: stringOrExpression.optional() })).optional() }), required: false, displayOptions: {"show":{"jsonDataFields":[false]}}, defaults: {"jsonDataFields":false} }),
    }).optional(),
  });
};
