// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mindee/v1.ts
/**
 * Mindee Node - Version 1
 * Consume Mindee API
 */


export interface MindeeV1Params {
/**
 * Which Mindee API Version to use
 * @default 1
 */
    apiVersion?: 1 | 3 | 4 | Expression<number>;
  resource?: 'invoice' | 'receipt';
  operation?: 'predict';
/**
 * Input Binary Field
 * @hint The name of the input binary field containing the file to be uploaded
 * @displayOptions.show { operation: ["predict"], resource: ["receipt", "invoice"] }
 * @default data
 */
    binaryPropertyName?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to return the data exactly in the way it got received from the API
 * @default false
 */
    rawData?: boolean | Expression<boolean>;
}

export interface MindeeV1Credentials {
  mindeeReceiptApi: CredentialReference;
  mindeeInvoiceApi: CredentialReference;
}

interface MindeeV1NodeBase {
  type: 'n8n-nodes-base.mindee';
  version: 1;
  credentials?: MindeeV1Credentials;
}

export type MindeeV1ParamsNode = MindeeV1NodeBase & {
  config: NodeConfig<MindeeV1Params>;
};

export type MindeeV1Node = MindeeV1ParamsNode;
