// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mindee/v3.schema.js
/**
 * Mindee Node - Version 3 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    apiVersion: z.union([z.literal(1), z.literal(3), z.literal(4), expressionSchema]).optional(),
    resource: z.union([z.literal('invoice'), z.literal('receipt')]).optional(),
    operation: z.union([z.literal('predict')]).optional(),
    binaryPropertyName: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"operation":["predict"],"resource":["receipt","invoice"]}}, defaults: {"operation":"predict","resource":"receipt"} }),
    rawData: booleanOrExpression.optional(),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
