// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mindee/v3.ts
/**
 * Mindee Node - Version 3
 * Consume Mindee API
 */


export interface MindeeV3Params {
/**
 * Which Mindee API Version to use
 * @default 4
 */
    apiVersion?: 1 | 3 | 4 | Expression<number>;
  resource?: 'invoice' | 'receipt';
  operation?: 'predict';
/**
 * Input Binary Field
 * @hint The name of the input binary field containing the file to be uploaded
 * @displayOptions.show { operation: ["predict"], resource: ["receipt", "invoice"] }
 * @default data
 */
    binaryPropertyName?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to return the data exactly in the way it got received from the API
 * @default false
 */
    rawData?: boolean | Expression<boolean>;
}

export interface MindeeV3Credentials {
  mindeeReceiptApi: CredentialReference;
  mindeeInvoiceApi: CredentialReference;
}

interface MindeeV3NodeBase {
  type: 'n8n-nodes-base.mindee';
  version: 3;
  credentials?: MindeeV3Credentials;
}

export type MindeeV3ParamsNode = MindeeV3NodeBase & {
  config: NodeConfig<MindeeV3Params>;
};

export type MindeeV3Node = MindeeV3ParamsNode;
