// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/sentryIo/v1/resource_event/operation_get.ts
/**
 * Sentry.io Node - Version 1
 * Discriminator: resource=event, operation=get
 */


interface Credentials {
  sentryIoOAuth2Api: CredentialReference;
  sentryIoApi: CredentialReference;
  sentryIoServerApi: CredentialReference;
}

/** Get event by ID */
export type SentryIoV1EventGetParams = {
  resource: 'event';
  operation: 'get';
  authentication?: 'accessToken' | 'oAuth2' | 'accessTokenServer' | Expression<string>;
/**
 * The slug of the organization the events belong to. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    organizationSlug?: string | Expression<string>;
/**
 * The slug of the project the events belong to. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    projectSlug?: string | Expression<string>;
/**
 * The ID of the event to retrieve (either the numeric primary-key or the hexadecimal ID as reported by the raven client)
 */
    eventId?: string | Expression<string> | PlaceholderValue;
};

export type SentryIoV1EventGetNode = {
  type: 'n8n-nodes-base.sentryIo';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SentryIoV1EventGetParams>;
};
