// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/hubspot/v1/resource_deal/operation_get.ts
/**
 * HubSpot Node - Version 1
 * Discriminator: resource=deal, operation=get
 */


interface Credentials {
  hubspotApi: CredentialReference;
  hubspotAppToken: CredentialReference;
  hubspotOAuth2Api: CredentialReference;
}

/** Get a contact */
export type HubspotV1DealGetParams = {
  resource: 'deal';
  operation: 'get';
  authentication?: 'apiKey' | 'appToken' | 'oAuth2' | Expression<string>;
/**
 * Unique identifier for a particular deal
 */
    dealId?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** By default, you will only get data for the most recent version of a property in the "versions" data. If you include this parameter, you will get data for all previous versions.
     * @default false
     */
    includePropertyVersions?: boolean | Expression<boolean>;
  };
};

export type HubspotV1DealGetNode = {
  type: 'n8n-nodes-base.hubspot';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<HubspotV1DealGetParams>;
};
