// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/hubspot/v1/resource_deal/operation_get_recently_created.ts
/**
 * HubSpot Node - Version 1
 * Discriminator: resource=deal, operation=getRecentlyCreated
 */


interface Credentials {
  hubspotApi: CredentialReference;
  hubspotAppToken: CredentialReference;
  hubspotOAuth2Api: CredentialReference;
}

/** Get recently created companies */
export type HubspotV1DealGetRecentlyCreatedParams = {
  resource: 'deal';
  operation: 'getRecentlyCreated';
  authentication?: 'apiKey' | 'appToken' | 'oAuth2' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Only return deals created after timestamp x
     */
    since?: string | Expression<string>;
    /** By default, you will only get data for the most recent version of a property in the "versions" data. If you include this parameter, you will get data for all previous versions.
     * @default false
     */
    includePropertyVersions?: boolean | Expression<boolean>;
  };
};

export type HubspotV1DealGetRecentlyCreatedNode = {
  type: 'n8n-nodes-base.hubspot';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<HubspotV1DealGetRecentlyCreatedParams>;
};
