// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/hubspot/v1/resource_deal/operation_create.schema.js
/**
 * HubSpot Node - Version 1 - Zod Schema
 * Discriminator: resource=deal, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('deal').default('deal'),
      operation: z.literal('create'),
      authentication: z.union([z.literal('apiKey'), z.literal('appToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      stage: stringOrExpression.optional(),
      additionalFields: z.object({ amount: stringOrExpression.optional(), associatedCompany: z.array(z.string()).optional(), associatedVids: z.array(z.string()).optional(), closeDate: stringOrExpression.optional(), customPropertiesUi: z.unknown().optional(), description: stringOrExpression.optional(), dealName: stringOrExpression.optional(), dealType: stringOrExpression.optional(), pipeline: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
