// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/hubspot/v1/resource_deal/operation_get_recently_created.schema.js
/**
 * HubSpot Node - Version 1 - Zod Schema
 * Discriminator: resource=deal, operation=getRecentlyCreated
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('deal').default('deal'),
      operation: z.literal('getRecentlyCreated'),
      authentication: z.union([z.literal('apiKey'), z.literal('appToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filters: z.object({ since: stringOrExpression.optional(), includePropertyVersions: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
