// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/twitter/v1/resource_direct_message/operation_create.ts
/**
 * X (Formerly Twitter) Node - Version 1
 * Discriminator: resource=directMessage, operation=create
 */


interface Credentials {
  twitterOAuth1Api: CredentialReference;
}

/** Create a direct message */
export type TwitterV1DirectMessageCreateParams = {
  resource: 'directMessage';
  operation: 'create';
/**
 * The ID of the user who should receive the direct message
 */
    userId?: string | Expression<string> | PlaceholderValue;
/**
 * The text of your Direct Message. URL encode as necessary. Max length of 10,000 characters.
 */
    text?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Name of the binary property which contain data that should be added to the direct message as attachment
     * @default data
     */
    attachment?: string | Expression<string> | PlaceholderValue;
  };
};

export type TwitterV1DirectMessageCreateNode = {
  type: 'n8n-nodes-base.twitter';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<TwitterV1DirectMessageCreateParams>;
};
