// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleBooks/v1/resource_bookshelf_volume/operation_get_all.schema.js
/**
 * Google Books Node - Version 1 - Zod Schema
 * Discriminator: resource=bookshelfVolume, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('bookshelfVolume'),
      operation: z.literal('getAll'),
      authentication: z.union([z.literal('serviceAccount'), z.literal('oAuth2'), expressionSchema]).optional(),
      myLibrary: booleanOrExpression.optional(),
      userId: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"hide":{"myLibrary":[true]}}, defaults: {"myLibrary":false} }),
      shelfId: stringOrExpression.optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
    }).optional(),
  });
};
