// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/slack/v23/resource_file/operation_get.ts
/**
 * Slack Node - Version 2.3
 * Discriminator: resource=file, operation=get
 */


interface Credentials {
  slackApi: CredentialReference;
  slackOAuth2Api: CredentialReference;
}

/** Get information about a channel */
export type SlackV23FileGetParams = {
  resource: 'file';
  operation: 'get';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * File ID
 */
    fileId?: string | Expression<string> | PlaceholderValue;
};

export type SlackV23FileGetOutput = {
  aac?: string;
  audio_wave_samples?: Array<number>;
  channels?: Array<string>;
  comments_count?: number;
  created?: number;
  display_as_bot?: boolean;
  duration_ms?: number;
  editable?: boolean;
  external_type?: string;
  file_access?: string;
  filetype?: string;
  groups?: Array<string>;
  has_more_shares?: boolean;
  has_rich_preview?: boolean;
  id?: string;
  ims?: Array<string>;
  is_external?: boolean;
  is_public?: boolean;
  is_starred?: boolean;
  media_display_type?: string;
  mimetype?: string;
  mode?: string;
  name?: string;
  permalink?: string;
  permalink_public?: string;
  pretty_type?: string;
  public_url_shared?: boolean;
  size?: number;
  subtype?: string;
  timestamp?: number;
  title?: string;
  transcription?: {
    status?: string;
  };
  url_private?: string;
  url_private_download?: string;
  user?: string;
  user_team?: string;
  username?: string;
};

export type SlackV23FileGetNode = {
  type: 'n8n-nodes-base.slack';
  version: 2.3;
  credentials?: Credentials;
  config: NodeConfig<SlackV23FileGetParams>;
  output?: Items<SlackV23FileGetOutput>;
};
