// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/copper/v1/resource_task/operation_get_all.ts
/**
 * Copper Node - Version 1
 * Discriminator: resource=task, operation=getAll
 */


interface Credentials {
  copperApi: CredentialReference;
}

export type CopperV1TaskGetAllParams = {
  resource: 'task';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 5
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filterFields?: {
    /** Comma-separated IDs of assignee IDs to filter by
     */
    assignee_ids?: string | Expression<string> | PlaceholderValue;
    /** Comma-separated IDs of project IDs to filter by
     */
    project_ids?: string | Expression<string> | PlaceholderValue;
  };
};

export type CopperV1TaskGetAllNode = {
  type: 'n8n-nodes-base.copper';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<CopperV1TaskGetAllParams>;
};
