// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/circleCi/v1/resource_pipeline/operation_get_all.schema.js
/**
 * CircleCI Node - Version 1 - Zod Schema
 * Discriminator: resource=pipeline, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('pipeline').default('pipeline'),
      operation: z.literal('getAll'),
      vcs: z.union([z.literal('bitbucket'), z.literal('github'), expressionSchema]).optional(),
      projectSlug: stringOrExpression.optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filters: z.object({ branch: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
