// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/okta/v1/resource_user/operation_get_all.ts
/**
 * Okta Node - Version 1
 * Discriminator: resource=user, operation=getAll
 */


interface Credentials {
  oktaApi: CredentialReference;
}

/** Get many users */
export type OktaV1UserGetAllParams = {
  resource: 'user';
  operation: 'getAll';
/**
 * Search Query
 * @hint Filter users by using the allowed syntax. &lt;a href="https://developer.okta.com/docs/reference/core-okta-api/#filter" target="_blank"&gt;More info&lt;/a&gt;.
 */
    searchQuery?: string | Expression<string> | PlaceholderValue;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 20
 */
    limit?: number | Expression<number>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Whether to return a simplified version of the response instead of the raw data
 * @default true
 */
    simplify?: boolean | Expression<boolean>;
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type OktaV1UserGetAllOutput = {
  _links?: {
    self?: {
      href?: string;
    };
  };
  created?: string;
  credentials?: {
    provider?: {
      name?: string;
      type?: string;
    };
  };
  id?: string;
  lastUpdated?: string;
  profile?: {
    city?: string;
    countryCode?: string;
    department?: string;
    displayName?: string;
    email?: string;
    firstName?: string;
    lastName?: string;
    login?: string;
    manager?: string;
    managerEmail?: string;
    organization?: string;
    site?: string;
    startDate?: string;
    timezone?: string;
    title?: string;
    userType?: string;
  };
  realmId?: string;
  status?: string;
  type?: {
    id?: string;
  };
};

export type OktaV1UserGetAllNode = {
  type: 'n8n-nodes-base.okta';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<OktaV1UserGetAllParams>;
  output?: Items<OktaV1UserGetAllOutput>;
};
