// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/okta/v1/resource_user/operation_create.schema.js
/**
 * Okta Node - Version 1 - Zod Schema
 * Discriminator: resource=user, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('user').default('user'),
      operation: z.literal('create'),
      firstName: stringOrExpression.optional(),
      lastName: stringOrExpression.optional(),
      login: stringOrExpression.optional(),
      email: stringOrExpression.optional(),
      activate: booleanOrExpression.optional(),
      getCreateFields: z.object({ city: stringOrExpression.optional(), costCenter: stringOrExpression.optional(), countryCode: stringOrExpression.optional(), department: stringOrExpression.optional(), displayName: stringOrExpression.optional(), division: stringOrExpression.optional(), employeeNumber: stringOrExpression.optional(), honorificPrefix: stringOrExpression.optional(), honorificSuffix: stringOrExpression.optional(), locale: stringOrExpression.optional(), manager: stringOrExpression.optional(), managerId: stringOrExpression.optional(), middleName: stringOrExpression.optional(), mobilePhone: stringOrExpression.optional(), nickName: stringOrExpression.optional(), password: stringOrExpression.optional(), organization: stringOrExpression.optional(), postalAddress: stringOrExpression.optional(), preferredLanguage: stringOrExpression.optional(), primaryPhone: stringOrExpression.optional(), profileUrl: stringOrExpression.optional(), recoveryQuestionAnswer: stringOrExpression.optional(), recoveryQuestionQuestion: stringOrExpression.optional(), secondEmail: stringOrExpression.optional(), state: stringOrExpression.optional(), streetAddress: stringOrExpression.optional(), timezone: stringOrExpression.optional(), title: stringOrExpression.optional(), userType: stringOrExpression.optional(), zipCode: stringOrExpression.optional() }).optional(),
      requestOptions: z.object({ batching: z.unknown().optional(), allowUnauthorizedCerts: z.boolean().optional(), proxy: stringOrExpression.optional(), timeout: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
