// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/okta/v1/resource_user/operation_get.ts
/**
 * Okta Node - Version 1
 * Discriminator: resource=user, operation=get
 */


interface Credentials {
  oktaApi: CredentialReference;
}

/** Get details of a user */
export type OktaV1UserGetParams = {
  resource: 'user';
  operation: 'get';
/**
 * The user you want to operate on. Choose from the list, or specify an ID.
 * @default {"mode":"list","value":""}
 */
    userId?: { __rl: true; mode: 'list' | 'login' | 'id'; value: string; cachedResultName?: string };
/**
 * Whether to return a simplified version of the response instead of the raw data
 * @default true
 */
    simplify?: boolean | Expression<boolean>;
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type OktaV1UserGetOutput = {
  _links?: {
    changePassword?: {
      href?: string;
      method?: string;
    };
    changeRecoveryQuestion?: {
      href?: string;
      method?: string;
    };
    deactivate?: {
      href?: string;
      method?: string;
    };
    expirePassword?: {
      href?: string;
      method?: string;
    };
    forgotPassword?: {
      href?: string;
      method?: string;
    };
    reactivate?: {
      href?: string;
      method?: string;
    };
    resetFactors?: {
      href?: string;
      method?: string;
    };
    resetPassword?: {
      href?: string;
      method?: string;
    };
    schema?: {
      href?: string;
    };
    self?: {
      href?: string;
    };
    suspend?: {
      href?: string;
      method?: string;
    };
    type?: {
      href?: string;
    };
  };
  created?: string;
  credentials?: {
    provider?: {
      name?: string;
      type?: string;
    };
  };
  id?: string;
  lastUpdated?: string;
  profile?: {
    email?: string;
    firstName?: string;
    lastName?: string;
    login?: string;
  };
  status?: string;
  type?: {
    id?: string;
  };
};

export type OktaV1UserGetNode = {
  type: 'n8n-nodes-base.okta';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<OktaV1UserGetParams>;
  output?: Items<OktaV1UserGetOutput>;
};
