// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/freshservice/v1/resource_product/operation_update.schema.js
/**
 * Freshservice Node - Version 1 - Zod Schema
 * Discriminator: resource=product, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('product'),
      operation: z.literal('update'),
      productId: stringOrExpression.optional(),
      updateFields: z.object({ asset_type_id: stringOrExpression.optional(), description: stringOrExpression.optional(), manufacturer: stringOrExpression.optional(), mode_of_procurement: z.union([z.literal('Buy'), z.literal('Lease'), z.literal('Both'), expressionSchema]).optional(), name: stringOrExpression.optional(), status: z.union([z.literal('In Production'), z.literal('In Pipeline'), z.literal('Retired'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
