// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/wise/v1/resource_transfer/operation_get_all.schema.js
/**
 * Wise Node - Version 1 - Zod Schema
 * Discriminator: resource=transfer, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('transfer'),
      operation: z.literal('getAll'),
      profileId: stringOrExpression.optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filters: z.object({ range: z.unknown().optional(), sourceCurrency: stringOrExpression.optional(), status: z.union([z.literal('bounced_back'), z.literal('cancelled'), z.literal('charged_back'), z.literal('funds_converted'), z.literal('funds_refunded'), z.literal('incoming_payment_waiting'), z.literal('outgoing_payment_sent'), z.literal('processing'), z.literal('unknown'), z.literal('waiting_recipient_input_to_proceed'), expressionSchema]).optional(), targetCurrency: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
