// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/wise/v1/resource_transfer/operation_create.schema.js
/**
 * Wise Node - Version 1 - Zod Schema
 * Discriminator: resource=transfer, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('transfer'),
      operation: z.literal('create'),
      profileId: stringOrExpression.optional(),
      quoteId: stringOrExpression.optional(),
      targetAccountId: stringOrExpression.optional(),
      additionalFields: z.object({ reference: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
