// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/wise/v1/resource_transfer/operation_get.schema.js
/**
 * Wise Node - Version 1 - Zod Schema
 * Discriminator: resource=transfer, operation=get
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('transfer'),
      operation: z.literal('get'),
      transferId: stringOrExpression.optional(),
      downloadReceipt: booleanOrExpression.optional(),
      binaryProperty: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"downloadReceipt":[true]}}, defaults: {"downloadReceipt":false} }),
      fileName: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"downloadReceipt":[true]}}, defaults: {"downloadReceipt":false} }),
    }).optional(),
  });
};
