// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/wise/v1/resource_transfer/operation_get_all.ts
/**
 * Wise Node - Version 1
 * Discriminator: resource=transfer, operation=getAll
 */


interface Credentials {
  wiseApi: CredentialReference;
}

export type WiseV1TransferGetAllParams = {
  resource: 'transfer';
  operation: 'getAll';
/**
 * ID of the user profile to retrieve. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 * @default []
 */
    profileId?: string | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 5
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Range of time for filtering the transfers
     * @default {}
     */
    range?: {
        /** Range Properties
     */
    rangeProperties?: {
      /** Created Date Start
       */
      createdDateStart?: string | Expression<string>;
      /** Created Date End
       */
      createdDateEnd?: string | Expression<string>;
    };
  };
    /** Code of the source currency for filtering the transfers
     */
    sourceCurrency?: string | Expression<string> | PlaceholderValue;
    /** Status
     * @default processing
     */
    status?: 'bounced_back' | 'cancelled' | 'charged_back' | 'funds_converted' | 'funds_refunded' | 'incoming_payment_waiting' | 'outgoing_payment_sent' | 'processing' | 'unknown' | 'waiting_recipient_input_to_proceed' | Expression<string>;
    /** Code of the target currency for filtering the transfers
     */
    targetCurrency?: string | Expression<string> | PlaceholderValue;
  };
};

export type WiseV1TransferGetAllNode = {
  type: 'n8n-nodes-base.wise';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<WiseV1TransferGetAllParams>;
};
