// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/lemlist/v1/resource_activity/operation_get_all.ts
/**
 * Lemlist Node - Version 1
 * Discriminator: resource=activity, operation=getAll
 */


interface Credentials {
  lemlistApi: CredentialReference;
}

export type LemlistV1ActivityGetAllParams = {
  resource: 'activity';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 5
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** ID of the campaign to retrieve activity for. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    campaignId?: string | Expression<string>;
    /** Type of activity to retrieve
     * @default emailsOpened
     */
    type?: 'emailsBounced' | 'emailsClicked' | 'emailsOpened' | 'emailsReplied' | 'emailsSendFailed' | 'emailsSent' | 'emailsUnsubscribed' | Expression<string>;
  };
};

export type LemlistV1ActivityGetAllNode = {
  type: 'n8n-nodes-base.lemlist';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<LemlistV1ActivityGetAllParams>;
};
