// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/lemlist/v1/resource_activity/operation_get_all.schema.js
/**
 * Lemlist Node - Version 1 - Zod Schema
 * Discriminator: resource=activity, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('activity').default('activity'),
      operation: z.literal('getAll').default('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filters: z.object({ campaignId: stringOrExpression.optional(), type: z.union([z.literal('emailsBounced'), z.literal('emailsClicked'), z.literal('emailsOpened'), z.literal('emailsReplied'), z.literal('emailsSendFailed'), z.literal('emailsSent'), z.literal('emailsUnsubscribed'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
