// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/trello/v1/resource_board/operation_create.schema.js
/**
 * Trello Node - Version 1 - Zod Schema
 * Discriminator: resource=board, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('board'),
      operation: z.literal('create'),
      name: stringOrExpression.optional(),
      description: stringOrExpression.optional(),
      additionalFields: z.object({ prefs_cardAging: z.union([z.literal('pirate'), z.literal('regular'), expressionSchema]).optional(), prefs_background: stringOrExpression.optional(), prefs_comments: z.union([z.literal('disabled'), z.literal('members'), z.literal('observers'), z.literal('org'), z.literal('public'), expressionSchema]).optional(), prefs_cardCovers: booleanOrExpression.optional(), prefs_invitations: z.union([z.literal('admins'), z.literal('members'), expressionSchema]).optional(), keepFromSource: stringOrExpression.optional(), defaultLabels: booleanOrExpression.optional(), defaultLists: booleanOrExpression.optional(), idOrganization: stringOrExpression.optional(), prefs_permissionLevel: z.union([z.literal('org'), z.literal('private'), z.literal('public'), expressionSchema]).optional(), powerUps: z.union([z.literal('all'), z.literal('calendar'), z.literal('cardAging'), z.literal('recap'), z.literal('voting'), expressionSchema]).optional(), prefs_selfJoin: booleanOrExpression.optional(), idBoardSource: stringOrExpression.optional(), prefs_voting: z.union([z.literal('disabled'), z.literal('members'), z.literal('observers'), z.literal('org'), z.literal('public'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
