// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/trello/v1/resource_board/operation_get.ts
/**
 * Trello Node - Version 1
 * Discriminator: resource=board, operation=get
 */


interface Credentials {
  trelloApi: CredentialReference;
}

/** Get the data of an attachment */
export type TrelloV1BoardGetParams = {
  resource: 'board';
  operation: 'get';
/**
 * The ID of the board
 * @default {"mode":"list","value":""}
 */
    id?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Fields to return. Either "all" or a comma-separated list: closed, dateLastActivity, dateLastView, desc, descData, idOrganization, invitations, invited, labelNames, memberships, name, pinned, powerUps, prefs, shortLink, shortUrl, starred, subscribed, URL.
     * @default all
     */
    fields?: string | Expression<string> | PlaceholderValue;
    /** Whether to include pluginData on the card with the response
     * @default false
     */
    pluginData?: boolean | Expression<boolean>;
  };
};

export type TrelloV1BoardGetOutput = {
  closed?: boolean;
  desc?: string;
  id?: string;
  idEnterprise?: null;
  idOrganization?: string;
  labelNames?: {
    black?: string;
    black_dark?: string;
    black_light?: string;
    blue?: string;
    blue_dark?: string;
    blue_light?: string;
    green?: string;
    green_dark?: string;
    green_light?: string;
    lime?: string;
    lime_dark?: string;
    lime_light?: string;
    orange?: string;
    orange_dark?: string;
    orange_light?: string;
    pink?: string;
    pink_dark?: string;
    pink_light?: string;
    purple?: string;
    purple_dark?: string;
    purple_light?: string;
    red?: string;
    red_dark?: string;
    red_light?: string;
    sky?: string;
    sky_dark?: string;
    sky_light?: string;
    yellow?: string;
    yellow_dark?: string;
    yellow_light?: string;
  };
  name?: string;
  pinned?: boolean;
  prefs?: {
    background?: string;
    backgroundBottomColor?: string;
    backgroundBrightness?: string;
    backgroundTile?: boolean;
    backgroundTopColor?: string;
    calendarFeedEnabled?: boolean;
    canBeEnterprise?: boolean;
    canBeOrg?: boolean;
    canBePrivate?: boolean;
    canBePublic?: boolean;
    canInvite?: boolean;
    cardAging?: string;
    cardCounts?: boolean;
    cardCovers?: boolean;
    comments?: string;
    hiddenPluginBoardButtons?: Array<string>;
    hideVotes?: boolean;
    invitations?: string;
    isTemplate?: boolean;
    permissionLevel?: string;
    selfJoin?: boolean;
    showCompleteStatus?: boolean;
    switcherViews?: Array<{
      enabled?: boolean;
      viewType?: string;
    }>;
    voting?: string;
  };
  shortUrl?: string;
  url?: string;
};

export type TrelloV1BoardGetNode = {
  type: 'n8n-nodes-base.trello';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<TrelloV1BoardGetParams>;
  output?: Items<TrelloV1BoardGetOutput>;
};
