// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/currents/v1/resource_test_result/operation_get_all.schema.js
/**
 * Currents Node - Version 1 - Zod Schema
 * Discriminator: resource=testResult, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('testResult'),
      operation: z.literal('getAll').default('getAll'),
      signature: stringOrExpression.optional(),
      dateStart: stringOrExpression.optional(),
      dateEnd: stringOrExpression.optional(),
      limit: numberOrExpression.optional(),
      filters: z.object({ branches: stringOrExpression.optional(), authors: stringOrExpression.optional(), groups: stringOrExpression.optional(), status: z.array(z.union([z.literal('failed'), z.literal('passed'), z.literal('pending'), z.literal('skipped')])).optional(), tags: stringOrExpression.optional() }).optional(),
      options: z.object({ startingAfter: stringOrExpression.optional(), endingBefore: stringOrExpression.optional() }).optional(),
      requestOptions: z.object({ batching: z.unknown().optional(), allowUnauthorizedCerts: z.boolean().optional(), proxy: stringOrExpression.optional(), timeout: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
