// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/currents/v1/resource_test_result/operation_get_all.ts
/**
 * Currents Node - Version 1
 * Discriminator: resource=testResult, operation=getAll
 */


interface Credentials {
  currentsApi: CredentialReference;
}

/** Historical test execution results */
export type CurrentsV1TestResultGetAllParams = {
  resource: 'testResult';
  operation: 'getAll';
/**
 * The unique test signature. Use the Signature resource to generate this from project ID, spec file path, and test title.
 */
    signature?: string | Expression<string> | PlaceholderValue;
/**
 * Start date for results (ISO 8601 format)
 */
    dateStart?: string | Expression<string>;
/**
 * End date for results (ISO 8601 format)
 */
    dateEnd?: string | Expression<string>;
/**
 * Max number of results to return
 * @default 10
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Filter by branches (comma-separated for multiple)
     */
    branches?: string | Expression<string> | PlaceholderValue;
    /** Filter by git authors (comma-separated for multiple)
     */
    authors?: string | Expression<string> | PlaceholderValue;
    /** Filter by groups (comma-separated for multiple)
     */
    groups?: string | Expression<string> | PlaceholderValue;
    /** Filter by test status
     * @default []
     */
    status?: Array<'failed' | 'passed' | 'pending' | 'skipped'>;
    /** Filter by run tags (multiple values supported)
     */
    tags?: string | Expression<string> | PlaceholderValue;
  };
/**
 * Options
 * @default {}
 */
    options?: {
    /** Cursor for forward pagination
     */
    startingAfter?: string | Expression<string> | PlaceholderValue;
    /** Cursor for backward pagination
     */
    endingBefore?: string | Expression<string> | PlaceholderValue;
  };
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type CurrentsV1TestResultGetAllNode = {
  type: 'n8n-nodes-base.currents';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<CurrentsV1TestResultGetAllParams>;
};
