// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/dataTable/v1/resource_table/operation_list.schema.js
/**
 * Data table Node - Version 1 - Zod Schema
 * Discriminator: resource=table, operation=list
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('table'),
      operation: z.literal('list'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":true} }),
      options: z.object({ filterName: stringOrExpression.optional(), sortField: z.union([z.literal('createdAt'), z.literal('name'), z.literal('updatedAt'), expressionSchema]).optional(), sortDirection: z.union([z.literal('asc'), z.literal('desc'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
