// @n8n/n8n-nodes-langchain/dist/node-definitions/nodes/n8n-nodes-langchain/googleGemini/v11/resource_file_search/operation_list_stores.ts
/**
 * Google Gemini Node - Version 1.1
 * Discriminator: resource=fileSearch, operation=listStores
 */


interface Credentials {
  googlePalmApi: CredentialReference;
}

/** List all File Search stores owned by the user */
export type LcGoogleGeminiV11FileSearchListStoresParams = {
  resource: 'fileSearch';
  operation: 'listStores';
/**
 * Maximum number of File Search stores to return per page (max 20)
 * @default 10
 */
    pageSize?: number | Expression<number>;
/**
 * Token from a previous page to retrieve the next page of results
 */
    pageToken?: string | Expression<string> | PlaceholderValue;
};

export interface LcGoogleGeminiV11FileSearchListStoresSubnodeConfig {
  tools?: ToolInstance[];
}

export type LcGoogleGeminiV11FileSearchListStoresNode = {
  type: '@n8n/n8n-nodes-langchain.googleGemini';
  version: 1.1;
  credentials?: Credentials;
  config: NodeConfig<LcGoogleGeminiV11FileSearchListStoresParams> & { subnodes?: LcGoogleGeminiV11FileSearchListStoresSubnodeConfig };
};
