// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/convertKitTrigger/v1.schema.js
/**
 * ConvertKit Trigger Node - Version 1 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    event: z.union([z.literal('formSubscribe'), z.literal('linkClick'), z.literal('productPurchase'), z.literal('purchaseCreate'), z.literal('courseComplete'), z.literal('courseSubscribe'), z.literal('subscriberActivate'), z.literal('subscriberUnsubscribe'), z.literal('tagAdd'), z.literal('tagRemove'), expressionSchema]).optional(),
    formId: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"event":["formSubscribe"]}}, defaults: {"event":""} }),
    courseId: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"event":["courseSubscribe","courseComplete"]}}, defaults: {"event":""} }),
    link: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"event":["linkClick"]}}, defaults: {"event":""} }),
    productId: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"event":["productPurchase"]}}, defaults: {"event":""} }),
    tagId: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"event":["tagAdd","tagRemove"]}}, defaults: {"event":""} }),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
