// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/invoiceNinja/v2/resource_bank_transaction/operation_create.ts
/**
 * Invoice Ninja Node - Version 2
 * Discriminator: resource=bank_transaction, operation=create
 */


interface Credentials {
  invoiceNinjaApi: CredentialReference;
}

/** Create a new client */
export type InvoiceNinjaV2BankTransactionCreateParams = {
  resource: 'bank_transaction';
  operation: 'create';
/**
 * API Version
 * @default v5
 */
    apiVersion?: 'v4' | 'v5' | Expression<string>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Amount
     * @default 0
     */
    amount?: number | Expression<number>;
    /** Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     */
    bankIntegrationId?: string | Expression<string>;
    /** Base Type
     */
    baseType?: 'CREDIT' | 'DEBIT' | Expression<string>;
    /** Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     */
    currencyId?: string | Expression<string>;
    /** Date
     */
    date?: string | Expression<string>;
    /** Description
     */
    description?: string | Expression<string> | PlaceholderValue;
  };
};

export type InvoiceNinjaV2BankTransactionCreateNode = {
  type: 'n8n-nodes-base.invoiceNinja';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<InvoiceNinjaV2BankTransactionCreateParams>;
};
