// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/sendGrid/v1/resource_contact/operation_get.schema.js
/**
 * SendGrid Node - Version 1 - Zod Schema
 * Discriminator: resource=contact, operation=get
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('contact'),
      operation: z.literal('get'),
      by: z.union([z.literal('id'), z.literal('email'), expressionSchema]).optional(),
      contactId: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"by":["id"]}}, defaults: {"by":"id"} }),
      email: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"by":["email"]}}, defaults: {"by":"id"} }),
    }).optional(),
  });
};
