// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/sendGrid/v1/resource_contact/operation_get_all.ts
/**
 * SendGrid Node - Version 1
 * Discriminator: resource=contact, operation=getAll
 */


interface Credentials {
  sendGridApi: CredentialReference;
}

/** Get many lists */
export type SendGridV1ContactGetAllParams = {
  resource: 'contact';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** The query field accepts valid &lt;a href="https://sendgrid.com/docs/for-developers/sending-email/segmentation-query-language/"&gt;SGQL&lt;/a&gt; for searching for a contact
     */
    query?: string | Expression<string> | PlaceholderValue;
  };
};

export type SendGridV1ContactGetAllNode = {
  type: 'n8n-nodes-base.sendGrid';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SendGridV1ContactGetAllParams>;
};
