// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/todoist/v2/resource_task/operation_update.schema.js
/**
 * Todoist Node - Version 2 - Zod Schema
 * Discriminator: resource=task, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('task').default('task'),
      operation: z.literal('update'),
      authentication: z.union([z.literal('apiKey'), z.literal('oAuth2'), expressionSchema]).optional(),
      taskId: stringOrExpression.optional(),
      updateFields: z.object({ content: stringOrExpression.optional(), description: stringOrExpression.optional(), dueDateTime: stringOrExpression.optional(), dueLang: stringOrExpression.optional(), dueString: stringOrExpression.optional(), dueLang: stringOrExpression.optional(), labels: z.array(z.string()).optional(), priority: numberOrExpression.optional(), order: numberOrExpression.optional(), dueDate: stringOrExpression.optional(), assigneeId: stringOrExpression.optional(), duration: numberOrExpression.optional(), durationUnit: z.union([z.literal('minute'), z.literal('day'), expressionSchema]).optional(), deadlineDate: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
