// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/todoist/v2/resource_task/operation_get.ts
/**
 * Todoist Node - Version 2
 * Discriminator: resource=task, operation=get
 */


interface Credentials {
  todoistApi: CredentialReference;
  todoistOAuth2Api: CredentialReference;
}

/** Task resource */
export type TodoistV2TaskGetParams = {
  resource: 'task';
  operation: 'get';
  authentication?: 'apiKey' | 'oAuth2' | Expression<string>;
/**
 * Task ID
 */
    taskId?: string | Expression<string> | PlaceholderValue;
};

export type TodoistV2TaskGetOutput = {
  comment_count?: number;
  content?: string;
  created_at?: string;
  creator_id?: string;
  description?: string;
  due?: {
    date?: string;
    is_recurring?: boolean;
    lang?: string;
    string?: string;
  };
  id?: string;
  is_completed?: boolean;
  labels?: Array<string>;
  order?: number;
  priority?: number;
  project_id?: string;
  url?: string;
};

export type TodoistV2TaskGetNode = {
  type: 'n8n-nodes-base.todoist';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<TodoistV2TaskGetParams>;
  output?: Items<TodoistV2TaskGetOutput>;
};
