// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/todoist/v2/resource_task/operation_create.schema.js
/**
 * Todoist Node - Version 2 - Zod Schema
 * Discriminator: resource=task, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('task').default('task'),
      operation: z.literal('create').default('create'),
      authentication: z.union([z.literal('apiKey'), z.literal('oAuth2'), expressionSchema]).optional(),
      project: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      labels: z.array(z.string()).optional(),
      content: stringOrExpression.optional(),
      options: z.object({ description: stringOrExpression.optional(), dueDateTime: stringOrExpression.optional(), dueLang: stringOrExpression.optional(), dueString: stringOrExpression.optional(), parentId: stringOrExpression.optional(), priority: numberOrExpression.optional(), section: stringOrExpression.optional(), order: numberOrExpression.optional(), dueDate: stringOrExpression.optional(), assigneeId: stringOrExpression.optional(), duration: numberOrExpression.optional(), durationUnit: z.union([z.literal('minute'), z.literal('day'), expressionSchema]).optional(), deadlineDate: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
