// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/salesforce/v1/resource_opportunity/operation_get.ts
/**
 * Salesforce Node - Version 1
 * Discriminator: resource=opportunity, operation=get
 */


interface Credentials {
  salesforceOAuth2Api: CredentialReference;
  salesforceJwtApi: CredentialReference;
}

/** Represents an opportunity, which is a sale or pending deal */
export type SalesforceV1OpportunityGetParams = {
  resource: 'opportunity';
  operation: 'get';
/**
 * OAuth Authorization Flow
 * @default oAuth2
 */
    authentication?: 'oAuth2' | 'jwt' | Expression<string>;
/**
 * ID of opportunity that needs to be fetched
 */
    opportunityId?: string | Expression<string> | PlaceholderValue;
};

export type SalesforceV1OpportunityGetOutput = {
  AccountId?: string;
  attributes?: {
    type?: string;
    url?: string;
  };
  CloseDate?: string;
  CreatedById?: string;
  CreatedDate?: string;
  Fiscal?: string;
  FiscalQuarter?: number;
  FiscalYear?: number;
  ForecastCategory?: string;
  ForecastCategoryName?: string;
  HasOpenActivity?: boolean;
  HasOpportunityLineItem?: boolean;
  HasOverdueTask?: boolean;
  Id?: string;
  IsClosed?: boolean;
  IsDeleted?: boolean;
  IsWon?: boolean;
  LastModifiedById?: string;
  LastModifiedDate?: string;
  Name?: string;
  OwnerId?: string;
  Probability?: number;
  PushCount?: number;
  StageName?: string;
  SystemModstamp?: string;
};

export type SalesforceV1OpportunityGetNode = {
  type: 'n8n-nodes-base.salesforce';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SalesforceV1OpportunityGetParams>;
  output?: Items<SalesforceV1OpportunityGetOutput>;
};
