// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/salesforce/v1/resource_opportunity/operation_get_summary.ts
/**
 * Salesforce Node - Version 1
 * Discriminator: resource=opportunity, operation=getSummary
 */


interface Credentials {
  salesforceOAuth2Api: CredentialReference;
  salesforceJwtApi: CredentialReference;
}

/** Represents an opportunity, which is a sale or pending deal */
export type SalesforceV1OpportunityGetSummaryParams = {
  resource: 'opportunity';
  operation: 'getSummary';
/**
 * OAuth Authorization Flow
 * @default oAuth2
 */
    authentication?: 'oAuth2' | 'jwt' | Expression<string>;
};

export type SalesforceV1OpportunityGetSummaryOutput = {
  objectDescribe?: {
    activateable?: boolean;
    associateEntityType?: null;
    associateParentEntity?: null;
    createable?: boolean;
    custom?: boolean;
    customSetting?: boolean;
    deepCloneable?: boolean;
    deletable?: boolean;
    deprecatedAndHidden?: boolean;
    feedEnabled?: boolean;
    hasSubtypes?: boolean;
    isInterface?: boolean;
    isSubtype?: boolean;
    keyPrefix?: string;
    label?: string;
    labelPlural?: string;
    layoutable?: boolean;
    mergeable?: boolean;
    mruEnabled?: boolean;
    name?: string;
    queryable?: boolean;
    replicateable?: boolean;
    retrieveable?: boolean;
    searchable?: boolean;
    triggerable?: boolean;
    undeletable?: boolean;
    updateable?: boolean;
    urls?: {
      approvalLayouts?: string;
      compactLayouts?: string;
      describe?: string;
      layouts?: string;
      listviews?: string;
      quickActions?: string;
      rowTemplate?: string;
      sobject?: string;
    };
  };
  recentItems?: Array<{
    attributes?: {
      type?: string;
      url?: string;
    };
    Id?: string;
    Name?: string;
  }>;
};

export type SalesforceV1OpportunityGetSummaryNode = {
  type: 'n8n-nodes-base.salesforce';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SalesforceV1OpportunityGetSummaryParams>;
  output?: Items<SalesforceV1OpportunityGetSummaryOutput>;
};
