// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/sentryIo/v1/resource_release/operation_create.schema.js
/**
 * Sentry.io Node - Version 1 - Zod Schema
 * Discriminator: resource=release, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('release'),
      operation: z.literal('create'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), z.literal('accessTokenServer'), expressionSchema]).optional(),
      organizationSlug: stringOrExpression.optional(),
      version: stringOrExpression.optional(),
      url: stringOrExpression.optional(),
      projects: z.array(z.string()).optional(),
      additionalFields: z.object({ dateReleased: stringOrExpression.optional(), commits: z.unknown().optional(), refs: z.unknown().optional() }).optional(),
    }).optional(),
  });
};
