// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleDrive/v3/resource_file/operation_move.ts
/**
 * Google Drive Node - Version 3
 * Discriminator: resource=file, operation=move
 */


interface Credentials {
  googleApi: CredentialReference;
  googleDriveOAuth2Api: CredentialReference;
}

/** Move a file to another folder */
export type GoogleDriveV3FileMoveParams = {
  resource: 'file';
  operation: 'move';
  authentication?: 'oAuth2' | 'serviceAccount' | Expression<string>;
/**
 * The file to move
 * @default {"mode":"list","value":""}
 */
    fileId?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
/**
 * The drive where to move the file
 * @default {"mode":"list","value":"My Drive"}
 */
    driveId?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
/**
 * The folder where to move the file
 * @default {"mode":"list","value":"root","cachedResultName":"/ (Root folder)"}
 */
    folderId?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
};

export type GoogleDriveV3FileMoveOutput = {
  id?: string;
  kind?: string;
  mimeType?: string;
  name?: string;
};

export type GoogleDriveV3FileMoveNode = {
  type: 'n8n-nodes-base.googleDrive';
  version: 3;
  credentials?: Credentials;
  config: NodeConfig<GoogleDriveV3FileMoveParams>;
  output?: Items<GoogleDriveV3FileMoveOutput>;
};
