// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleDrive/v3/resource_file/operation_update.schema.js
/**
 * Google Drive Node - Version 3 - Zod Schema
 * Discriminator: resource=file, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('file').default('file'),
      operation: z.literal('update'),
      authentication: z.union([z.literal('oAuth2'), z.literal('serviceAccount'), expressionSchema]).optional(),
      fileId: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('url'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      changeFileContent: booleanOrExpression.optional(),
      inputDataFieldName: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"changeFileContent":[true]}}, defaults: {"changeFileContent":false} }),
      newUpdatedFileName: stringOrExpression.optional(),
      options: z.object({ appPropertiesUi: z.unknown().optional(), propertiesUi: z.unknown().optional(), keepRevisionForever: booleanOrExpression.optional(), ocrLanguage: stringOrExpression.optional(), useContentAsIndexableText: booleanOrExpression.optional(), trashed: booleanOrExpression.optional(), fields: z.array(z.union([z.literal('*'), z.literal('explicitlyTrashed'), z.literal('exportLinks'), z.literal('hasThumbnail'), z.literal('iconLink'), z.literal('id'), z.literal('kind'), z.literal('mimeType'), z.literal('name'), z.literal('permissions'), z.literal('shared'), z.literal('spaces'), z.literal('starred'), z.literal('thumbnailLink'), z.literal('trashed'), z.literal('version'), z.literal('webViewLink')])).optional() }).optional(),
    }).optional(),
  });
};
