// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleDrive/v3/resource_file/operation_delete_file.ts
/**
 * Google Drive Node - Version 3
 * Discriminator: resource=file, operation=deleteFile
 */


interface Credentials {
  googleApi: CredentialReference;
  googleDriveOAuth2Api: CredentialReference;
}

/** Permanently delete a file */
export type GoogleDriveV3FileDeleteFileParams = {
  resource: 'file';
  operation: 'deleteFile';
  authentication?: 'oAuth2' | 'serviceAccount' | Expression<string>;
/**
 * The file to delete
 * @default {"mode":"list","value":""}
 */
    fileId?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
/**
 * Options
 * @default {}
 */
    options?: {
    /** Whether to delete the file immediately. If false, the file will be moved to the trash.
     * @default false
     */
    deletePermanently?: boolean | Expression<boolean>;
  };
};

export type GoogleDriveV3FileDeleteFileOutput = {
  id?: string;
  success?: boolean;
};

export type GoogleDriveV3FileDeleteFileNode = {
  type: 'n8n-nodes-base.googleDrive';
  version: 3;
  credentials?: Credentials;
  config: NodeConfig<GoogleDriveV3FileDeleteFileParams>;
  output?: Items<GoogleDriveV3FileDeleteFileOutput>;
};
