// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleDrive/v3/resource_file/operation_update.ts
/**
 * Google Drive Node - Version 3
 * Discriminator: resource=file, operation=update
 */


interface Credentials {
  googleApi: CredentialReference;
  googleDriveOAuth2Api: CredentialReference;
}

/** Update a shared drive */
export type GoogleDriveV3FileUpdateParams = {
  resource: 'file';
  operation: 'update';
  authentication?: 'oAuth2' | 'serviceAccount' | Expression<string>;
/**
 * The file to update
 * @default {"mode":"list","value":""}
 */
    fileId?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
/**
 * Whether to send a new binary data to update the file
 * @default false
 */
    changeFileContent?: boolean | Expression<boolean>;
/**
 * Find the name of input field containing the binary data to update the file in the Input panel on the left, in the Binary tab
 * @hint The name of the input field containing the binary file data to update the file
 * @displayOptions.show { changeFileContent: [true] }
 * @default data
 */
    inputDataFieldName?: string | Expression<string> | PlaceholderValue;
/**
 * If not specified, the file name will not be changed
 */
    newUpdatedFileName?: string | Expression<string> | PlaceholderValue;
/**
 * Options
 * @default {}
 */
    options?: {
    /** A collection of arbitrary key-value pairs which are private to the requesting app
     * @default {}
     */
    appPropertiesUi?: {
        /** APP Property
     */
    appPropertyValues?: Array<{
      /** Name of the key to add
       */
      key?: string | Expression<string> | PlaceholderValue;
      /** Value to set for the key
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
    /** A collection of arbitrary key-value pairs which are visible to all apps
     * @default {}
     */
    propertiesUi?: {
        /** Property
     */
    propertyValues?: Array<{
      /** Name of the key to add
       */
      key?: string | Expression<string> | PlaceholderValue;
      /** Value to set for the key
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
    /** Whether to set the 'keepForever' field in the new head revision. This is only applicable to files with binary content in Google Drive. Only 200 revisions for the file can be kept forever. If the limit is reached, try deleting pinned revisions.
     * @default false
     */
    keepRevisionForever?: boolean | Expression<boolean>;
    /** A language hint for OCR processing during image import (ISO 639-1 code)
     */
    ocrLanguage?: string | Expression<string> | PlaceholderValue;
    /** Whether to use the uploaded content as indexable text
     * @default false
     */
    useContentAsIndexableText?: boolean | Expression<boolean>;
    /** Whether to move a file to the trash. Only the owner may trash a file.
     * @default false
     */
    trashed?: boolean | Expression<boolean>;
    /** The fields to return
     * @default []
     */
    fields?: Array<'*' | 'explicitlyTrashed' | 'exportLinks' | 'hasThumbnail' | 'iconLink' | 'id' | 'kind' | 'mimeType' | 'name' | 'permissions' | 'shared' | 'spaces' | 'starred' | 'thumbnailLink' | 'trashed' | 'version' | 'webViewLink'>;
  };
};

export type GoogleDriveV3FileUpdateOutput = {
  id?: string;
  kind?: string;
  mimeType?: string;
  name?: string;
};

export type GoogleDriveV3FileUpdateNode = {
  type: 'n8n-nodes-base.googleDrive';
  version: 3;
  credentials?: Credentials;
  config: NodeConfig<GoogleDriveV3FileUpdateParams>;
  output?: Items<GoogleDriveV3FileUpdateOutput>;
};
