// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/zulip/v1/resource_stream/operation_update.schema.js
/**
 * Zulip Node - Version 1 - Zod Schema
 * Discriminator: resource=stream, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('stream'),
      operation: z.literal('update'),
      streamId: stringOrExpression.optional(),
      jsonParameters: booleanOrExpression.optional(),
      additionalFieldsJson: resolveSchema({ parameters, schema: z.union([iDataObjectSchema, z.string()]), required: false, displayOptions: {"show":{"jsonParameters":[true]}}, defaults: {"jsonParameters":false} }),
      additionalFields: resolveSchema({ parameters, schema: z.object({ isAnnouncementOnly: booleanOrExpression.optional(), description: stringOrExpression.optional(), isPrivate: booleanOrExpression.optional(), historyPublicToSubscribers: booleanOrExpression.optional(), newName: stringOrExpression.optional(), streamPostPolicy: z.union([z.literal(1), z.literal(2), z.literal(3), expressionSchema]).optional() }), required: false, displayOptions: {"show":{"jsonParameters":[false]}}, defaults: {"jsonParameters":false} }),
    }).optional(),
  });
};
