// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/zulip/v1/resource_stream/operation_create.ts
/**
 * Zulip Node - Version 1
 * Discriminator: resource=stream, operation=create
 */


interface Credentials {
  zulipApi: CredentialReference;
}

/** Create a stream */
export type ZulipV1StreamCreateParams = {
  resource: 'stream';
  operation: 'create';
/**
 * JSON Parameters
 * @default false
 */
    jsonParameters?: boolean | Expression<boolean>;
/**
 * JSON format parameters for stream creation
 * @displayOptions.show { jsonParameters: [true] }
 */
    additionalFieldsJson?: IDataObject | string | Expression<string>;
/**
 * A list of dictionaries containing the the key name and value specifying the name of the stream to subscribe. If the stream does not exist a new stream is created.
 * @displayOptions.show { jsonParameters: [false] }
 * @default {}
 */
    subscriptions?: {
        /** Subscription Properties
     */
    properties?: Array<{
      /** Name of Subscription
       */
      name?: string | Expression<string> | PlaceholderValue;
      /** Description of Subscription
       */
      description?: string | Expression<string> | PlaceholderValue;
    }>;
  };
/**
 * Additional Fields
 * @displayOptions.show { jsonParameters: [false] }
 * @default {}
 */
    additionalFields?: {
    /** If announce is True and one of the streams specified in subscriptions has to be created (i.e. doesnt exist to begin with), an announcement will be made notifying that a new stream was created.
     * @default false
     */
    announce?: boolean | Expression<boolean>;
    /** Whether authorization errors (such as when the requesting user is not authorized to access a private stream) should be considered fatal or not. When True, an authorization error is reported as such. When set to False, the returned JSON payload indicates that there was an authorization error, but the response is still considered a successful one.
     * @default false
     */
    authorizationErrorsFatal?: boolean | Expression<boolean>;
    /** Whether the streams message history should be available to newly subscribed members, or users can only access messages they actually received while subscribed to the stream
     * @default false
     */
    historyPublicToSubscribers?: boolean | Expression<boolean>;
    /** Whether the streams specified in subscriptions are invite-only or not
     * @default false
     */
    inviteOnly?: boolean | Expression<boolean>;
    /** A list of email addresses of the users that will be subscribed/unsubscribed to the streams specified in the subscriptions argument. If not provided, then the requesting user/bot is subscribed.
     * @default {}
     */
    principals?: {
        /** Principals Properties
     */
    properties?: Array<{
      /** Principal email address
       */
      email?: string | Expression<string> | PlaceholderValue;
    }>;
  };
    /** Policy for which users can post messages to the stream
     */
    streamPostPolicy?: 1 | 2 | 3 | Expression<number>;
  };
};

export type ZulipV1StreamCreateNode = {
  type: 'n8n-nodes-base.zulip';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ZulipV1StreamCreateParams>;
};
