// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mondayCom/v1/resource_board/operation_create.ts
/**
 * Monday.com Node - Version 1
 * Discriminator: resource=board, operation=create
 */


interface Credentials {
  mondayComApi: CredentialReference;
  mondayComOAuth2Api: CredentialReference;
}

/** Create a new board */
export type MondayComV1BoardCreateParams = {
  resource: 'board';
  operation: 'create';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * The board's name
 */
    name?: string | Expression<string> | PlaceholderValue;
/**
 * The board's kind (public / private / share)
 */
    kind?: 'share' | 'public' | 'private' | Expression<string>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Optional board template ID
     * @default 0
     */
    templateId?: number | Expression<number>;
  };
};

export type MondayComV1BoardCreateOutput = {
  id?: string;
};

export type MondayComV1BoardCreateNode = {
  type: 'n8n-nodes-base.mondayCom';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MondayComV1BoardCreateParams>;
  output?: Items<MondayComV1BoardCreateOutput>;
};
