// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mondayCom/v1/resource_board/operation_get_all.ts
/**
 * Monday.com Node - Version 1
 * Discriminator: resource=board, operation=getAll
 */


interface Credentials {
  mondayComApi: CredentialReference;
  mondayComOAuth2Api: CredentialReference;
}

/** Get many boards */
export type MondayComV1BoardGetAllParams = {
  resource: 'board';
  operation: 'getAll';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
};

export type MondayComV1BoardGetAllOutput = {
  board_kind?: string;
  id?: string;
  name?: string;
  owners?: Array<{
    id?: string;
  }>;
  state?: string;
};

export type MondayComV1BoardGetAllNode = {
  type: 'n8n-nodes-base.mondayCom';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MondayComV1BoardGetAllParams>;
  output?: Items<MondayComV1BoardGetAllOutput>;
};
