// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/urlScanIo/v1/resource_scan/operation_get_all.ts
/**
 * urlscan.io Node - Version 1
 * Discriminator: resource=scan, operation=getAll
 */


interface Credentials {
  urlScanIoApi: CredentialReference;
}

export type UrlScanIoV1ScanGetAllParams = {
  resource: 'scan';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Query using the &lt;a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html#query-dsl-query-string-query"&gt;Elastic Search Query String syntax&lt;/a&gt;. See &lt;a href="https://urlscan.io/docs/search/"&gt;supported fields&lt;/a&gt; in the documentation.
     */
    query?: string | Expression<string> | PlaceholderValue;
  };
};

export type UrlScanIoV1ScanGetAllNode = {
  type: 'n8n-nodes-base.urlScanIo';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<UrlScanIoV1ScanGetAllParams>;
};
