// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/seaTable/v1.schema.js
/**
 * SeaTable Node - Version 1 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    resource: z.union([z.literal('row')]).optional(),
    operation: z.union([z.literal('create'), z.literal('delete'), z.literal('get'), z.literal('getAll'), z.literal('update')]).optional(),
    tableName: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"hide":{"operation":["get"]}}, defaults: {"operation":"create"} }),
    tableId: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"operation":["get"]}}, defaults: {"operation":"create"} }),
    rowId: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"operation":["update","delete","get"]}}, defaults: {"operation":"create"} }),
    fieldsToSend: resolveSchema({ parameters, schema: z.union([z.literal('autoMapInputData'), z.literal('defineBelow'), expressionSchema]), required: false, displayOptions: {"show":{"operation":["create","update"]}}, defaults: {"operation":"create"} }),
    inputsToIgnore: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"operation":["create","update"],"fieldsToSend":["autoMapInputData"]}}, defaults: {"operation":"create","fieldsToSend":"defineBelow"} }),
    columnsUi: resolveSchema({ parameters, schema: z.object({ columnValues: z.array(z.object({ columnName: stringOrExpression.optional(), columnValue: stringOrExpression.optional() })).optional() }), required: false, displayOptions: {"show":{"operation":["create","update"],"fieldsToSend":["defineBelow"]}}, defaults: {"operation":"create","fieldsToSend":"defineBelow"} }),
    returnAll: resolveSchema({ parameters, schema: booleanOrExpression, required: false, displayOptions: {"show":{"operation":["getAll"]}}, defaults: {"operation":"create"} }),
    limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"operation":["getAll"],"returnAll":[false]}}, defaults: {"operation":"create","returnAll":true} }),
    filters: resolveSchema({ parameters, schema: z.object({ view_name: stringOrExpression.optional() }), required: false, displayOptions: {"show":{"operation":["getAll"]}}, defaults: {"operation":"create"} }),
    options: resolveSchema({ parameters, schema: z.object({ convert_link_id: booleanOrExpression.optional(), direction: z.union([z.literal('asc'), z.literal('desc'), expressionSchema]).optional(), order_by: stringOrExpression.optional() }), required: false, displayOptions: {"show":{"operation":["getAll"]}}, defaults: {"operation":"create"} }),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
