// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/todoist/v21/resource_section/operation_create.ts
/**
 * Todoist Node - Version 2.1
 * Discriminator: resource=section, operation=create
 */


interface Credentials {
  todoistApi: CredentialReference;
  todoistOAuth2Api: CredentialReference;
}

/** Section resource */
export type TodoistV21SectionCreateParams = {
  resource: 'section';
  operation: 'create';
  authentication?: 'apiKey' | 'oAuth2' | Expression<string>;
/**
 * The project to add the section to
 * @default {"mode":"list","value":""}
 */
    sectionProject?: { __rl: true; mode: 'list' | 'id'; value: string; cachedResultName?: string };
/**
 * Name of the section
 */
    sectionName?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    sectionOptions?: {
    /** The order of the section
     * @default 0
     */
    order?: number | Expression<number>;
  };
};

export type TodoistV21SectionCreateNode = {
  type: 'n8n-nodes-base.todoist';
  version: 2.1;
  credentials?: Credentials;
  config: NodeConfig<TodoistV21SectionCreateParams>;
};
