// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/cortex/v1/resource_analyzer/operation_execute.schema.js
/**
 * Cortex Node - Version 1 - Zod Schema
 * Discriminator: resource=analyzer, operation=execute
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('analyzer').default('analyzer'),
      operation: z.literal('execute').default('execute'),
      analyzer: stringOrExpression.optional(),
      observableType: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"hide":{"analyzer":[""]}}, defaults: {"analyzer":""} }),
      observableValue: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"hide":{"observableType":["file"],"analyzer":[""]}}, defaults: {"observableType":"","analyzer":""} }),
      binaryPropertyName: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"observableType":["file"]}}, defaults: {"observableType":""} }),
      tlp: resolveSchema({ parameters, schema: z.union([z.literal(0), z.literal(1), z.literal(2), z.literal(3), expressionSchema]), required: false, displayOptions: {"hide":{"observableType":[""],"analyzer":[""]}}, defaults: {"observableType":"","analyzer":""} }),
      additionalFields: z.object({ force: booleanOrExpression.optional(), timeout: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
