// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/cortex/v1/resource_analyzer/operation_execute.ts
/**
 * Cortex Node - Version 1
 * Discriminator: resource=analyzer, operation=execute
 */


interface Credentials {
  cortexApi: CredentialReference;
}

/** Execute Analyzer */
export type CortexV1AnalyzerExecuteParams = {
  resource: 'analyzer';
  operation: 'execute';
/**
 * Choose the analyzer. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    analyzer?: string | Expression<string>;
/**
 * Choose the observable type. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 * @displayOptions.hide { analyzer: [""] }
 */
    observableType?: string | Expression<string>;
/**
 * Enter the observable value
 * @displayOptions.hide { observableType: ["file"], analyzer: [""] }
 */
    observableValue?: string | Expression<string> | PlaceholderValue;
/**
 * Put Output File in Field
 * @hint The name of the output binary field to put the file in
 * @displayOptions.show { observableType: ["file"] }
 * @default data
 */
    binaryPropertyName?: string | Expression<string> | PlaceholderValue;
/**
 * The TLP of the analyzed observable
 * @displayOptions.hide { observableType: [""], analyzer: [""] }
 * @default 2
 */
    tlp?: 0 | 1 | 2 | 3 | Expression<number>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Whether to force bypassing the cache
     * @default false
     */
    force?: boolean | Expression<boolean>;
    /** Timeout to wait for the report in case it is not available at the time the query was made
     * @default 3
     */
    timeout?: number | Expression<number>;
  };
};

export type CortexV1AnalyzerExecuteNode = {
  type: 'n8n-nodes-base.cortex';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<CortexV1AnalyzerExecuteParams>;
};
