// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/gSuiteAdmin/v1/resource_device/operation_get_all.schema.js
/**
 * Google Workspace Admin Node - Version 1 - Zod Schema
 * Discriminator: resource=device, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('device'),
      operation: z.literal('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      projection: z.union([z.literal('basic'), z.literal('full'), expressionSchema]).optional(),
      includeChildOrgunits: booleanOrExpression.optional(),
      filter: z.object({ orgUnitPath: stringOrExpression.optional(), query: stringOrExpression.optional() }).optional(),
      sort: z.object({ sortRules: z.object({ orderBy: z.union([z.literal('annotatedLocation'), z.literal('annotatedUser'), z.literal('lastSync'), z.literal('notes'), z.literal('serialNumber'), z.literal('status'), expressionSchema]).optional(), sortBy: z.union([z.literal('ascending'), z.literal('descending'), expressionSchema]).optional() }).optional() }).optional(),
    }).optional(),
  });
};
