// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/highLevel/v2/resource_opportunity/operation_create.ts
/**
 * HighLevel Node - Version 2
 * Discriminator: resource=opportunity, operation=create
 */


interface Credentials {
  highLevelOAuth2Api: CredentialReference;
}

export type HighLevelV2OpportunityCreateParams = {
  resource: 'opportunity';
  operation: 'create';
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    pipelineId?: string | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 * @hint There can only be one opportunity for each contact
 */
    contactId?: string | Expression<string>;
/**
 * Name
 */
    name?: string | Expression<string> | PlaceholderValue;
/**
 * Status
 * @default open
 */
    status?: 'open' | 'won' | 'lost' | 'abandoned' | Expression<string>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     */
    assignedTo?: string | Expression<string>;
    /** Company Name
     */
    companyName?: string | Expression<string> | PlaceholderValue;
    /** Monetary value of lead opportunity
     */
    monetaryValue?: number | Expression<number>;
    /** Tags
     * @hint Comma separated list of tags, array of strings can be set in expression
     */
    tags?: string | Expression<string> | PlaceholderValue;
    /** Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     */
    stageId?: string | Expression<string>;
  };
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type HighLevelV2OpportunityCreateOutput = {
  opportunity?: {
    contact?: {
      email?: string;
      followers?: Array<string>;
      id?: string;
      name?: string;
      phone?: string;
      tags?: Array<string>;
    };
    contactId?: string;
    createdAt?: string;
    followers?: Array<string>;
    id?: string;
    internalSource?: {
      apiVersion?: string;
      channel?: string;
      id?: string;
      source?: string;
      type?: string;
    };
    isAttribute?: boolean;
    lastActionDate?: string;
    lastStageChangeAt?: string;
    lastStatusChangeAt?: string;
    locationId?: string;
    monetaryValue?: number;
    name?: string;
    pipelineId?: string;
    pipelineStageId?: string;
    status?: string;
    updatedAt?: string;
  };
  traceId?: string;
};

export type HighLevelV2OpportunityCreateNode = {
  type: 'n8n-nodes-base.highLevel';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<HighLevelV2OpportunityCreateParams>;
  output?: Items<HighLevelV2OpportunityCreateOutput>;
};
