// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/highLevel/v2/resource_opportunity/operation_get.ts
/**
 * HighLevel Node - Version 2
 * Discriminator: resource=opportunity, operation=get
 */


interface Credentials {
  highLevelOAuth2Api: CredentialReference;
}

export type HighLevelV2OpportunityGetParams = {
  resource: 'opportunity';
  operation: 'get';
/**
 * Opportunity ID
 */
    opportunityId?: string | Expression<string> | PlaceholderValue;
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type HighLevelV2OpportunityGetOutput = {
  opportunity?: {
    contact?: {
      email?: string;
      followers?: Array<string>;
      id?: string;
      name?: string;
      phone?: string;
      tags?: Array<string>;
    };
    contactId?: string;
    createdAt?: string;
    followers?: Array<string>;
    id?: string;
    indexVersion?: number;
    internalSource?: {
      apiVersion?: string;
      channel?: string;
      id?: string;
      source?: string;
      type?: string;
    };
    isAttribute?: boolean;
    lastActionDate?: string;
    lastStageChangeAt?: string;
    lastStatusChangeAt?: string;
    locationId?: string;
    name?: string;
    pipelineId?: string;
    pipelineStageId?: string;
    status?: string;
    updatedAt?: string;
  };
  traceId?: string;
};

export type HighLevelV2OpportunityGetNode = {
  type: 'n8n-nodes-base.highLevel';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<HighLevelV2OpportunityGetParams>;
  output?: Items<HighLevelV2OpportunityGetOutput>;
};
