// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/highLevel/v2/resource_opportunity/operation_update.ts
/**
 * HighLevel Node - Version 2
 * Discriminator: resource=opportunity, operation=update
 */


interface Credentials {
  highLevelOAuth2Api: CredentialReference;
}

export type HighLevelV2OpportunityUpdateParams = {
  resource: 'opportunity';
  operation: 'update';
/**
 * Opportunity ID
 * @hint You cannot update an opportunity's pipeline ID.
 */
    opportunityId?: string | Expression<string> | PlaceholderValue;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     */
    assignedTo?: string | Expression<string>;
    /** Monetary value of lead opportunity
     */
    monetaryValue?: number | Expression<number>;
    /** Name
     */
    name?: string | Expression<string> | PlaceholderValue;
    /** Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     */
    pipelineId?: string | Expression<string>;
    /** Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     * @hint Select 'Pipeline Name or ID' first to see stages
     */
    stageId?: string | Expression<string>;
    /** Status
     * @default open
     */
    status?: 'open' | 'won' | 'lost' | 'abandoned' | Expression<string>;
  };
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type HighLevelV2OpportunityUpdateOutput = {
  opportunity?: {
    contact?: {
      email?: string;
      id?: string;
      name?: string;
      phone?: string;
      tags?: Array<string>;
    };
    contactId?: string;
    createdAt?: string;
    followers?: Array<string>;
    id?: string;
    indexVersion?: number;
    internalSource?: {
      apiVersion?: string;
      channel?: string;
      id?: string;
      source?: string;
      type?: string;
    };
    isAttribute?: boolean;
    lastActionDate?: string;
    lastStageChangeAt?: string;
    lastStatusChangeAt?: string;
    locationId?: string;
    monetaryValue?: number;
    name?: string;
    pipelineId?: string;
    pipelineStageId?: string;
    source?: string;
    status?: string;
    updatedAt?: string;
  };
  traceId?: string;
};

export type HighLevelV2OpportunityUpdateNode = {
  type: 'n8n-nodes-base.highLevel';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<HighLevelV2OpportunityUpdateParams>;
  output?: Items<HighLevelV2OpportunityUpdateOutput>;
};
