// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mailchimp/v1/resource_member/operation_get_all.schema.js
/**
 * Mailchimp Node - Version 1 - Zod Schema
 * Discriminator: resource=member, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('member').default('member'),
      operation: z.literal('getAll'),
      authentication: z.union([z.literal('apiKey'), z.literal('oAuth2'), expressionSchema]).optional(),
      list: stringOrExpression.optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      options: z.object({ beforeLastChanged: stringOrExpression.optional(), beforeTimestampOpt: stringOrExpression.optional(), emailType: z.union([z.literal('html'), z.literal('text'), expressionSchema]).optional(), status: z.union([z.literal('cleaned'), z.literal('pending'), z.literal('subscribed'), z.literal('transactional'), z.literal('unsubscribed'), expressionSchema]).optional(), sinceLastChanged: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
