// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mailchimp/v1/resource_member/operation_create.schema.js
/**
 * Mailchimp Node - Version 1 - Zod Schema
 * Discriminator: resource=member, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('member').default('member'),
      operation: z.literal('create').default('create'),
      authentication: z.union([z.literal('apiKey'), z.literal('oAuth2'), expressionSchema]).optional(),
      list: stringOrExpression.optional(),
      email: stringOrExpression.optional(),
      status: z.union([z.literal('cleaned'), z.literal('pending'), z.literal('subscribed'), z.literal('transactional'), z.literal('unsubscribed'), expressionSchema]).optional(),
      jsonParameters: booleanOrExpression.optional(),
      options: z.object({ emailType: z.union([z.literal('html'), z.literal('text'), expressionSchema]).optional(), language: stringOrExpression.optional(), ipOptIn: stringOrExpression.optional(), ipSignup: stringOrExpression.optional(), timestampSignup: stringOrExpression.optional(), tags: stringOrExpression.optional(), vip: booleanOrExpression.optional(), timestampOpt: stringOrExpression.optional() }).optional(),
      locationFieldsUi: resolveSchema({ parameters, schema: z.object({ locationFieldsValues: z.object({ latitude: stringOrExpression.optional(), longitude: stringOrExpression.optional() }).optional() }), required: false, displayOptions: {"show":{"jsonParameters":[false]}}, defaults: {"jsonParameters":false} }),
      mergeFieldsUi: resolveSchema({ parameters, schema: z.object({ mergeFieldsValues: z.array(z.object({ name: stringOrExpression.optional(), value: stringOrExpression.optional() })).optional() }), required: false, displayOptions: {"show":{"jsonParameters":[false]}}, defaults: {"jsonParameters":false} }),
      mergeFieldsJson: resolveSchema({ parameters, schema: z.union([iDataObjectSchema, z.string()]), required: false, displayOptions: {"show":{"jsonParameters":[true]}}, defaults: {"jsonParameters":false} }),
      locationJson: resolveSchema({ parameters, schema: z.union([iDataObjectSchema, z.string()]), required: false, displayOptions: {"show":{"jsonParameters":[true]}}, defaults: {"jsonParameters":false} }),
      groupsUi: resolveSchema({ parameters, schema: z.object({ groupsValues: z.array(z.object({ categoryId: stringOrExpression.optional(), categoryFieldId: stringOrExpression.optional(), value: booleanOrExpression.optional() })).optional() }), required: false, displayOptions: {"show":{"jsonParameters":[false]}}, defaults: {"jsonParameters":false} }),
      groupJson: resolveSchema({ parameters, schema: z.union([iDataObjectSchema, z.string()]), required: false, displayOptions: {"show":{"jsonParameters":[true]}}, defaults: {"jsonParameters":false} }),
    }).optional(),
  });
};
