// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/workableTrigger/v1.ts
/**
 * Workable Trigger Node - Version 1
 * Starts the workflow when Workable events occur
 */


export interface WorkableTriggerV1Params {
  triggerOn?: 'candidateCreated' | 'candidateMoved' | Expression<string>;
  filters?: {
    /** Get notifications only for one job. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    job?: string | Expression<string>;
    /** Get notifications for specific stages. e.g. 'hired'. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    stage?: string | Expression<string>;
  };
}

export interface WorkableTriggerV1Credentials {
  workableApi: CredentialReference;
}

interface WorkableTriggerV1NodeBase {
  type: 'n8n-nodes-base.workableTrigger';
  version: 1;
  credentials?: WorkableTriggerV1Credentials;
  isTrigger: true;
}

export type WorkableTriggerV1ParamsNode = WorkableTriggerV1NodeBase & {
  config: NodeConfig<WorkableTriggerV1Params>;
};

export type WorkableTriggerV1Node = WorkableTriggerV1ParamsNode;
