// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/harvest/v1/resource_project/operation_get.ts
/**
 * Harvest Node - Version 1
 * Discriminator: resource=project, operation=get
 */


interface Credentials {
  harvestApi: CredentialReference;
  harvestOAuth2Api: CredentialReference;
}

/** Get data of a client */
export type HarvestV1ProjectGetParams = {
  resource: 'project';
  operation: 'get';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    accountId?: string | Expression<string>;
/**
 * The ID of the project you are retrieving
 */
    id?: string | Expression<string> | PlaceholderValue;
};

export type HarvestV1ProjectGetOutput = {
  bill_by?: string;
  budget_by?: string;
  budget_is_monthly?: boolean;
  client?: {
    currency?: string;
    id?: number;
    name?: string;
  };
  cost_budget_include_expenses?: boolean;
  created_at?: string;
  id?: number;
  is_active?: boolean;
  is_billable?: boolean;
  is_fixed_fee?: boolean;
  name?: string;
  notes?: string;
  notify_when_over_budget?: boolean;
  over_budget_notification_percentage?: number;
  show_budget_to_all?: boolean;
  updated_at?: string;
};

export type HarvestV1ProjectGetNode = {
  type: 'n8n-nodes-base.harvest';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<HarvestV1ProjectGetParams>;
  output?: Items<HarvestV1ProjectGetOutput>;
};
