// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/harvest/v1/resource_project/operation_get_all.ts
/**
 * Harvest Node - Version 1
 * Discriminator: resource=project, operation=getAll
 */


interface Credentials {
  harvestApi: CredentialReference;
  harvestOAuth2Api: CredentialReference;
}

/** Get data of many clients */
export type HarvestV1ProjectGetAllParams = {
  resource: 'project';
  operation: 'getAll';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    accountId?: string | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Only return projects belonging to the client with the given ID
     */
    client_id?: string | Expression<string> | PlaceholderValue;
    /** Whether to only return active projects and false to return inactive projects
     * @default true
     */
    is_active?: boolean | Expression<boolean>;
    /** The page number to use in pagination
     * @default 1
     */
    page?: number | Expression<number>;
    /** Only return projects by updated_since
     */
    updated_since?: string | Expression<string>;
  };
};

export type HarvestV1ProjectGetAllOutput = {
  bill_by?: string;
  budget_by?: string;
  budget_is_monthly?: boolean;
  client?: {
    currency?: string;
    id?: number;
    name?: string;
  };
  cost_budget_include_expenses?: boolean;
  created_at?: string;
  id?: number;
  is_active?: boolean;
  is_billable?: boolean;
  is_fixed_fee?: boolean;
  name?: string;
  notify_when_over_budget?: boolean;
  over_budget_notification_percentage?: number;
  show_budget_to_all?: boolean;
  updated_at?: string;
};

export type HarvestV1ProjectGetAllNode = {
  type: 'n8n-nodes-base.harvest';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<HarvestV1ProjectGetAllParams>;
  output?: Items<HarvestV1ProjectGetAllOutput>;
};
