// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/seaTable/v2/resource_link/operation_remove.ts
/**
 * SeaTable Node - Version 2
 * Discriminator: resource=link, operation=remove
 */


interface Credentials {
  seaTableApi: CredentialReference;
}

/** Delete a row */
export type SeaTableV2LinkRemoveParams = {
  resource: 'link';
  operation: 'remove';
/**
 * Choose from the list, of specify by using an expression. Provide it in the way "table_name:::table_id".
 */
    tableName?: string | Expression<string>;
/**
 * Choose from the list of specify the Link Column by using an expression. You have to provide it in the way "column_name:::link_id:::other_table_id".
 */
    linkColumn?: string | Expression<string>;
/**
 * Provide the row ID of table you selected
 */
    linkColumnSourceId?: string | Expression<string> | PlaceholderValue;
/**
 * Provide the row ID of table you want to link
 */
    linkColumnTargetId?: string | Expression<string> | PlaceholderValue;
};

export type SeaTableV2LinkRemoveNode = {
  type: 'n8n-nodes-base.seaTable';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<SeaTableV2LinkRemoveParams>;
};
