// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/seaTable/v2/resource_link/operation_list.ts
/**
 * SeaTable Node - Version 2
 * Discriminator: resource=link, operation=list
 */


interface Credentials {
  seaTableApi: CredentialReference;
}

/** Get many rows from a table or a table view */
export type SeaTableV2LinkListParams = {
  resource: 'link';
  operation: 'list';
/**
 * Choose from the list, of specify by using an expression. Provide it in the way "table_name:::table_id".
 */
    tableName?: string | Expression<string>;
/**
 * Choose from the list of specify the Link Column by using an expression. You have to provide it in the way "column_name:::link_id:::other_table_id:::column_key".
 */
    linkColumn?: string | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    rowId?: string | Expression<string>;
};

export type SeaTableV2LinkListNode = {
  type: 'n8n-nodes-base.seaTable';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<SeaTableV2LinkListParams>;
};
