// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/salesforce/v1/resource_task/operation_get.ts
/**
 * Salesforce Node - Version 1
 * Discriminator: resource=task, operation=get
 */


interface Credentials {
  salesforceOAuth2Api: CredentialReference;
  salesforceJwtApi: CredentialReference;
}

/** Represents a business activity such as making a phone call or other to-do items. In the user interface, and records are collectively referred to as activities. */
export type SalesforceV1TaskGetParams = {
  resource: 'task';
  operation: 'get';
/**
 * OAuth Authorization Flow
 * @default oAuth2
 */
    authentication?: 'oAuth2' | 'jwt' | Expression<string>;
/**
 * ID of task that needs to be fetched
 */
    taskId?: string | Expression<string> | PlaceholderValue;
};

export type SalesforceV1TaskGetOutput = {
  attributes?: {
    type?: string;
    url?: string;
  };
  CallDisposition?: null;
  CallObject?: null;
  CreatedById?: string;
  CreatedDate?: string;
  Id?: string;
  IsArchived?: boolean;
  IsClosed?: boolean;
  IsDeleted?: boolean;
  IsHighPriority?: boolean;
  IsRecurrence?: boolean;
  IsReminderSet?: boolean;
  LastModifiedById?: string;
  LastModifiedDate?: string;
  OwnerId?: string;
  Priority?: string;
  RecurrenceDayOfMonth?: null;
  RecurrenceInstance?: null;
  RecurrenceMonthOfYear?: null;
  RecurrenceRegeneratedType?: null;
  Status?: string;
  SystemModstamp?: string;
  TaskSubtype?: string;
};

export type SalesforceV1TaskGetNode = {
  type: 'n8n-nodes-base.salesforce';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SalesforceV1TaskGetParams>;
  output?: Items<SalesforceV1TaskGetOutput>;
};
