// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/salesforce/v1/resource_task/operation_create.ts
/**
 * Salesforce Node - Version 1
 * Discriminator: resource=task, operation=create
 */


interface Credentials {
  salesforceOAuth2Api: CredentialReference;
  salesforceJwtApi: CredentialReference;
}

/** Represents a business activity such as making a phone call or other to-do items. In the user interface, and records are collectively referred to as activities. */
export type SalesforceV1TaskCreateParams = {
  resource: 'task';
  operation: 'create';
/**
 * OAuth Authorization Flow
 * @default oAuth2
 */
    authentication?: 'oAuth2' | 'jwt' | Expression<string>;
/**
 * The current status of the task, such as In Progress or Completed. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    status?: string | Expression<string>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Represents the due date of the task. This field has a timestamp that is always set to midnight in the Coordinated Universal Time (UTC) time zone.
     */
    activityDate?: string | Expression<string>;
    /** Represents the result of a given call, for example, “we'll call back,” or “call unsuccessful.” Limit is 255 characters. Not subject to field-level security, available for any user in an organization with Salesforce CRM Call Center.
     */
    callDisposition?: string | Expression<string> | PlaceholderValue;
    /** Duration of the call in seconds. Not subject to field-level security, available for any user in an organization with Salesforce CRM Call Center.
     */
    callDurationInSeconds?: number | Expression<number>;
    /** Name of a call center. Limit is 255 characters. Not subject to field-level security, available for any user in an organization with Salesforce CRM Call Center.
     */
    callObject?: string | Expression<string> | PlaceholderValue;
    /** The type of call being answered: Inbound, Internal, or Outbound. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    callType?: string | Expression<string>;
    /** Filter by custom fields
     * @default {}
     */
    customFieldsUi?: {
        /** Custom Field
     */
    customFieldsValues?: Array<{
      /** The ID of the field to add custom field to. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
       */
      fieldId?: string | Expression<string>;
      /** The value to set on custom field
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
    /** Contains a text description of the task
     */
    description?: string | Expression<string> | PlaceholderValue;
    /** Whether a popup reminder has been set for the task (true) or not (false)
     * @default false
     */
    isReminderSet?: boolean | Expression<boolean>;
    /** ID of the User who owns the record. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    owner?: string | Expression<string>;
    /** Indicates the importance or urgency of a task, such as high or low. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    priority?: string | Expression<string>;
    /** Recurrence Type of the task. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    recurrenceType?: string | Expression<string>;
    /** The frequency of the recurring task. For example, “2nd” or “3rd.”. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    recurrenceInstance?: string | Expression<string>;
    /** The interval between recurring tasks
     */
    recurrenceInterval?: number | Expression<number>;
    /** The day of the month in which the task repeats
     */
    recurrenceDayOfMonth?: number | Expression<number>;
    /** The day or days of the week on which the task repeats. This field contains a bitmask. The values are as follows: Sunday = 1 Monday = 2 Tuesday = 4 Wednesday = 8 Thursday = 16 Friday = 32 Saturday = 64 Multiple days are represented as the sum of their numerical values. For example, Tuesday and Thursday = 4 + 16 = 20.
     */
    recurrenceDayOfWeekMask?: number | Expression<number>;
    /** The last date on which the task repeats. This field has a timestamp that is always set to midnight in the Coordinated Universal Time (UTC) time zone.
     */
    recurrenceEndDateOnly?: string | Expression<string>;
    /** The month of the year in which the task repeats
     */
    recurrenceMonthOfYear?: 'January' | 'February' | 'March' | 'April' | 'May' | 'June' | 'July' | 'August' | 'September' | 'October' | 'November' | 'December' | Expression<string>;
    /** Represents what triggers a repeating task to repeat. Add this field to a page layout together with the RecurrenceInterval field, which determines the number of days between the triggering date (due date or close date) and the due date of the next repeating task in the series. Label is Repeat This Task.
     */
    recurrenceRegeneratedType?: 'RecurrenceRegenerateAfterDueDate' | 'RecurrenceRegenerateAfterToday' | 'RecurrenceRegenerated' | Expression<string>;
    /** The date when the recurring task begins. Must be a date and time before RecurrenceEndDateOnly.
     */
    recurrenceEndDateOnly?: string | Expression<string>;
    /** The time zone associated with the recurring task. For example, “UTC-8:00” for Pacific Standard Time.
     */
    recurrenceTimeZoneSidKey?: string | Expression<string> | PlaceholderValue;
    /** Represents the time when the reminder is scheduled to fire, if IsReminderSet is set to true. If IsReminderSet is set to false, then the user may have deselected the reminder checkbox in the Salesforce user interface, or the reminder has already fired at the time indicated by the value.
     */
    reminderDateTime?: string | Expression<string>;
    /** The subject line of the task, such as “Call” or “Send Quote.” Limit: 255 characters. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    subject?: string | Expression<string>;
    /** Represents Type of the task, such as Call or Meeting. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    type?: string | Expression<string>;
    /** The WhatId represents nonhuman objects such as accounts, opportunities, campaigns, cases, or custom objects. WhatIds are polymorphic. Polymorphic means a WhatId is equivalent to the ID of a related object.
     */
    whatId?: string | Expression<string> | PlaceholderValue;
    /** The WhoId represents a human such as a lead or a contact. WhoIds are polymorphic. Polymorphic means a WhoId is equivalent to a contact’s ID or a lead’s ID.
     */
    whoId?: string | Expression<string> | PlaceholderValue;
  };
};

export type SalesforceV1TaskCreateOutput = {
  id?: string;
  success?: boolean;
};

export type SalesforceV1TaskCreateNode = {
  type: 'n8n-nodes-base.salesforce';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SalesforceV1TaskCreateParams>;
  output?: Items<SalesforceV1TaskCreateOutput>;
};
