// @n8n/db/dist/repositories/project-secrets-provider-access.repository.ee.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProjectSecretsProviderAccessRepository = void 0;
const di_1 = require("@n8n/di");
const typeorm_1 = require("@n8n/typeorm");
const entities_1 = require("../entities");
let ProjectSecretsProviderAccessRepository = class ProjectSecretsProviderAccessRepository extends typeorm_1.Repository {
    constructor(dataSource) {
        super(entities_1.ProjectSecretsProviderAccess, dataSource.manager);
    }
    async findByConnectionId(secretsProviderConnectionId) {
        return await this.find({
            where: { secretsProviderConnectionId },
            relations: ['project'],
        });
    }
    async findByProjectId(projectId) {
        return await this.find({
            where: { projectId },
            relations: ['secretsProviderConnection'],
        });
    }
    async deleteByConnectionId(secretsProviderConnectionId) {
        await this.delete({ secretsProviderConnectionId });
    }
    async setProjectAccess(secretsProviderConnectionId, projectIds) {
        await this.manager.transaction(async (tx) => {
            await tx.delete(entities_1.ProjectSecretsProviderAccess, { secretsProviderConnectionId });
            if (projectIds.length > 0) {
                const entries = projectIds.map((projectId) => this.create({
                    secretsProviderConnectionId,
                    projectId,
                }));
                await tx.insert(entities_1.ProjectSecretsProviderAccess, entries);
            }
        });
    }
};
exports.ProjectSecretsProviderAccessRepository = ProjectSecretsProviderAccessRepository;
exports.ProjectSecretsProviderAccessRepository = ProjectSecretsProviderAccessRepository = __decorate([
    (0, di_1.Service)(),
    __metadata("design:paramtypes", [typeorm_1.DataSource])
], ProjectSecretsProviderAccessRepository);
