// @n8n/db/dist/repositories/role.repository.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RoleRepository = void 0;
const di_1 = require("@n8n/di");
const typeorm_1 = require("@n8n/typeorm");
const n8n_workflow_1 = require("n8n-workflow");
const entities_1 = require("../entities");
let RoleRepository = class RoleRepository extends typeorm_1.Repository {
    constructor(dataSource) {
        super(entities_1.Role, dataSource.manager);
    }
    async findAll(trx) {
        const em = trx ?? this.manager;
        return await em.find(entities_1.Role, { relations: ['scopes'] });
    }
    async countUsersWithRole(role) {
        if (role.roleType === 'global') {
            return await this.manager.getRepository(entities_1.User).count({
                where: {
                    role: {
                        slug: role.slug,
                    },
                },
            });
        }
        else if (role.roleType === 'project') {
            return await this.manager.getRepository(entities_1.ProjectRelation).count({
                where: { role: { slug: role.slug } },
            });
        }
        return 0;
    }
    async findAllRoleCounts() {
        const userCount = await this.manager
            .createQueryBuilder(entities_1.User, 'user')
            .select('user.roleSlug', 'roleSlug')
            .addSelect('COUNT(user.id)', 'count')
            .groupBy('user.roleSlug')
            .getRawMany();
        const projectCount = await this.manager
            .createQueryBuilder(entities_1.ProjectRelation, 'projectRelation')
            .select('projectRelation.role', 'roleSlug')
            .addSelect('COUNT(projectRelation.user)', 'count')
            .groupBy('projectRelation.role')
            .getRawMany();
        return userCount.concat(projectCount).reduce((acc, { roleSlug, count }) => {
            if (!acc[roleSlug]) {
                acc[roleSlug] = 0;
            }
            acc[roleSlug] += parseInt(count, 10);
            return acc;
        }, {});
    }
    async findAllProjectCounts() {
        const results = await this.manager
            .createQueryBuilder(entities_1.ProjectRelation, 'pr')
            .select('pr.role', 'roleSlug')
            .addSelect('COUNT(DISTINCT pr.projectId)', 'count')
            .groupBy('pr.role')
            .getRawMany();
        return results.reduce((acc, { roleSlug, count }) => {
            acc[roleSlug] = parseInt(count, 10);
            return acc;
        }, {});
    }
    async findProjectAssignments(roleSlug) {
        const counts = await this.manager
            .createQueryBuilder(entities_1.ProjectRelation, 'pr')
            .select('pr.projectId', 'projectId')
            .addSelect('COUNT(pr.userId)', 'memberCount')
            .addSelect('MAX(pr.createdAt)', 'lastAssigned')
            .where('pr.role = :roleSlug', { roleSlug })
            .groupBy('pr.projectId')
            .getRawMany();
        if (counts.length === 0)
            return [];
        const projectIds = counts.map((c) => c.projectId);
        const projects = await this.manager.getRepository(entities_1.Project).findBy({ id: (0, typeorm_1.In)(projectIds) });
        const projectMap = new Map(projects.map((p) => [p.id, p]));
        return counts
            .map((c) => {
            const project = projectMap.get(c.projectId);
            if (!project)
                return null;
            return {
                projectId: project.id,
                projectName: project.name,
                projectIcon: project.icon,
                memberCount: parseInt(c.memberCount, 10),
                lastAssigned: c.lastAssigned instanceof Date
                    ? c.lastAssigned.toISOString()
                    : (c.lastAssigned ?? null),
            };
        })
            .filter((r) => r !== null);
    }
    async findAllProjectMembers(projectId, roleSlug) {
        const qb = this.manager
            .createQueryBuilder(entities_1.ProjectRelation, 'pr')
            .innerJoin(entities_1.User, 'user', 'user.id = pr.userId')
            .select('user.id', 'userId')
            .addSelect('user.firstName', 'firstName')
            .addSelect('user.lastName', 'lastName')
            .addSelect('user.email', 'email')
            .addSelect('pr.role', 'role')
            .where('pr.projectId = :projectId', { projectId });
        if (roleSlug) {
            qb.andWhere('pr.role = :roleSlug', { roleSlug });
        }
        return await qb.getRawMany();
    }
    async findBySlug(slug) {
        return await this.findOne({
            where: { slug },
            relations: ['scopes'],
        });
    }
    async findBySlugs(slugs, roleType) {
        return await this.find({
            where: { slug: (0, typeorm_1.In)(slugs), roleType },
            relations: ['scopes'],
        });
    }
    async removeBySlug(slug) {
        const result = await this.delete({ slug });
        if (result.affected !== 1) {
            throw new Error(`Failed to delete role "${slug}"`);
        }
    }
    async updateEntityWithManager(entityManager, slug, newData) {
        const role = await entityManager.findOne(entities_1.Role, {
            where: { slug },
            relations: ['scopes'],
        });
        if (!role) {
            throw new n8n_workflow_1.UserError('Role not found');
        }
        if (role.systemRole) {
            throw new n8n_workflow_1.UserError('Cannot update system roles');
        }
        if (newData.displayName !== undefined) {
            role.displayName = newData.displayName;
        }
        if (newData.description !== undefined) {
            role.description = newData.description;
        }
        if (newData.scopes !== undefined) {
            role.scopes = newData.scopes;
        }
        return await entityManager.save(role);
    }
    async updateRole(slug, newData) {
        return await this.manager.transaction(async (transactionManager) => {
            return await this.updateEntityWithManager(transactionManager, slug, newData);
        });
    }
};
exports.RoleRepository = RoleRepository;
exports.RoleRepository = RoleRepository = __decorate([
    (0, di_1.Service)(),
    __metadata("design:paramtypes", [typeorm_1.DataSource])
], RoleRepository);
