// @n8n/db/dist/repositories/secrets-provider-connection.repository.ee.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecretsProviderConnectionRepository = void 0;
const di_1 = require("@n8n/di");
const typeorm_1 = require("@n8n/typeorm");
const entities_1 = require("../entities");
let SecretsProviderConnectionRepository = class SecretsProviderConnectionRepository extends typeorm_1.Repository {
    constructor(dataSource) {
        super(entities_1.SecretsProviderConnection, dataSource.manager);
    }
    async findAll() {
        return await this.find();
    }
    async hasGlobalProvider(providerKey) {
        const count = await this.manager
            .createQueryBuilder(entities_1.SecretsProviderConnection, 'connection')
            .leftJoin('connection.projectAccess', 'access')
            .where('access.secretsProviderConnectionId IS NULL')
            .andWhere('connection.providerKey = :providerKey', { providerKey })
            .getCount();
        return count > 0;
    }
    async findGlobalConnections(filters = {}) {
        const { providerKeys } = filters;
        if (providerKeys && providerKeys.length === 0) {
            return [];
        }
        const connectionQuery = this.createQueryBuilder('connection')
            .leftJoin('connection.projectAccess', 'access')
            .where('access.secretsProviderConnectionId IS NULL');
        if (providerKeys) {
            connectionQuery.andWhere('connection.providerKey IN (:...providerKeys)', { providerKeys });
        }
        return await connectionQuery.getMany();
    }
    async findByProjectId(projectId, filters = {}) {
        const { providerKeys } = filters;
        if (providerKeys && providerKeys.length === 0) {
            return [];
        }
        const connectionQuery = this.createQueryBuilder('connection')
            .innerJoinAndSelect('connection.projectAccess', 'projectAccess')
            .leftJoinAndSelect('projectAccess.project', 'project')
            .where('projectAccess.projectId = :projectId', { projectId });
        if (providerKeys) {
            connectionQuery.andWhere('connection.providerKey IN (:...providerKeys)', { providerKeys });
        }
        return await connectionQuery.getMany();
    }
    async isProviderAvailableInProject(providerKey, projectId) {
        const count = await this.manager
            .createQueryBuilder(entities_1.SecretsProviderConnection, 'connection')
            .leftJoin('connection.projectAccess', 'access')
            .where('connection.providerKey = :providerKey', { providerKey })
            .andWhere(new typeorm_1.Brackets((qb) => {
            qb.where('access.secretsProviderConnectionId IS NULL')
                .orWhere('access.projectId = :projectId', { projectId });
        }))
            .getCount();
        return count > 0;
    }
    async findAllAccessibleByProjectWithProjectAccess(projectId) {
        const projectConnections = await this.createQueryBuilder('connection')
            .leftJoinAndSelect('connection.projectAccess', 'projectAccess')
            .leftJoinAndSelect('projectAccess.project', 'project')
            .where('projectAccess.projectId = :projectId', { projectId })
            .getMany();
        const globalConnections = await this.createQueryBuilder('connection')
            .leftJoinAndSelect('connection.projectAccess', 'access')
            .where('access.secretsProviderConnectionId IS NULL')
            .getMany();
        return projectConnections.concat(globalConnections);
    }
    async findAllAccessibleProviderKeysByCredentialId(credentialId) {
        if (!(await this.exists()))
            return [];
        const ownerProjectSubquery = this.manager
            .createQueryBuilder()
            .subQuery()
            .select('sc.projectId')
            .from(entities_1.SharedCredentials, 'sc')
            .where('sc.credentialsId = :credentialId')
            .andWhere('sc.role = :ownerRole')
            .getQuery();
        const rows = await this.createQueryBuilder('connection')
            .select('connection.providerKey', 'providerKey')
            .leftJoin('connection.projectAccess', 'access')
            .where(new typeorm_1.Brackets((qb) => {
            qb.where('access.secretsProviderConnectionId IS NULL')
                .orWhere(`access.projectId IN ${ownerProjectSubquery}`);
        }))
            .setParameters({ credentialId, ownerRole: 'credential:owner' })
            .getRawMany();
        return rows.map((row) => row.providerKey);
    }
    async findAccessibleByProviderKeyAndProjectId(providerKey, projectId) {
        return await this.manager
            .createQueryBuilder(entities_1.SecretsProviderConnection, 'connection')
            .leftJoinAndSelect('connection.projectAccess', 'access')
            .leftJoinAndSelect('access.project', 'project')
            .where('connection.providerKey = :providerKey', { providerKey })
            .andWhere(new typeorm_1.Brackets((qb) => {
            qb.where('access.secretsProviderConnectionId IS NULL')
                .orWhere('access.projectId = :projectId', { projectId });
        }))
            .getOne();
    }
    async findByProviderKeyAndProjectId(providerKey, projectId) {
        return await this.createQueryBuilder('connection')
            .innerJoinAndSelect('connection.projectAccess', 'projectAccess')
            .leftJoinAndSelect('projectAccess.project', 'project')
            .where('connection.providerKey = :providerKey', { providerKey })
            .andWhere('projectAccess.projectId = :projectId', { projectId })
            .getOne();
    }
    async removeByProviderKeyAndProjectId(providerKey, projectId) {
        const connection = await this.findByProviderKeyAndProjectId(providerKey, projectId);
        if (!connection) {
            return null;
        }
        return await this.remove(connection);
    }
};
exports.SecretsProviderConnectionRepository = SecretsProviderConnectionRepository;
exports.SecretsProviderConnectionRepository = SecretsProviderConnectionRepository = __decorate([
    (0, di_1.Service)(),
    __metadata("design:paramtypes", [typeorm_1.DataSource])
], SecretsProviderConnectionRepository);
