// @n8n/db/dist/repositories/shared-credentials.repository.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SharedCredentialsRepository = void 0;
const di_1 = require("@n8n/di");
const permissions_1 = require("@n8n/permissions");
const typeorm_1 = require("@n8n/typeorm");
const entities_1 = require("../entities");
let SharedCredentialsRepository = class SharedCredentialsRepository extends typeorm_1.Repository {
    constructor(dataSource) {
        super(entities_1.SharedCredentials, dataSource.manager);
    }
    async findByCredentialIds(credentialIds, role) {
        return await this.find({
            relations: { credentials: true, project: { projectRelations: { user: true, role: true } } },
            where: {
                credentialsId: (0, typeorm_1.In)(credentialIds),
                role,
            },
        });
    }
    async makeOwnerOfAllCredentials(project) {
        return await this.update({
            projectId: (0, typeorm_1.Not)(project.id),
            role: 'credential:owner',
        }, { project });
    }
    async makeOwner(credentialIds, projectId, trx) {
        trx = trx ?? this.manager;
        return await trx.upsert(entities_1.SharedCredentials, credentialIds.map((credentialsId) => ({
            projectId,
            credentialsId,
            role: 'credential:owner',
        })), ['projectId', 'credentialsId']);
    }
    async deleteByIds(sharedCredentialsIds, projectId, trx) {
        trx = trx ?? this.manager;
        return await trx.delete(entities_1.SharedCredentials, {
            projectId,
            credentialsId: (0, typeorm_1.In)(sharedCredentialsIds),
        });
    }
    async getFilteredAccessibleCredentials(projectIds, credentialsIds) {
        return (await this.find({
            where: {
                projectId: (0, typeorm_1.In)(projectIds),
                credentialsId: (0, typeorm_1.In)(credentialsIds),
            },
            select: ['credentialsId'],
        })).map((s) => s.credentialsId);
    }
    async findCredentialOwningProject(credentialsId) {
        return (await this.findOne({
            where: { credentialsId, role: 'credential:owner' },
            relations: { project: true },
        }))?.project;
    }
    async getAllRelationsForCredentials(credentialIds) {
        return await this.find({
            where: {
                credentialsId: (0, typeorm_1.In)(credentialIds),
            },
            relations: ['project'],
        });
    }
    async getSharedPersonalCredentialsCount() {
        return await this.createQueryBuilder('sc')
            .innerJoin('sc.project', 'project')
            .where('sc.role = :role', { role: 'credential:owner' })
            .andWhere('project.type = :type', { type: 'personal' })
            .andWhere((qb) => {
            const subQuery = qb
                .subQuery()
                .select('1')
                .from(entities_1.SharedCredentials, 'other')
                .where('other.credentialsId = sc.credentialsId')
                .andWhere('other.projectId != sc.projectId')
                .getQuery();
            return `EXISTS ${subQuery}`;
        })
            .getCount();
    }
    async findCredentialsWithOptions(where = {}, trx) {
        trx = trx ?? this.manager;
        return await trx.find(entities_1.SharedCredentials, {
            where,
            relations: {
                credentials: {
                    shared: { project: true },
                },
            },
        });
    }
    async findCredentialsByRoles(userIds, projectRoles, credentialRoles, trx) {
        trx = trx ?? this.manager;
        return await trx.find(entities_1.SharedCredentials, {
            where: {
                role: (0, typeorm_1.In)(credentialRoles),
                project: {
                    projectRelations: {
                        userId: (0, typeorm_1.In)(userIds),
                        role: { slug: (0, typeorm_1.In)(projectRoles) },
                    },
                },
            },
        });
    }
    buildSharedCredentialIdsSubquery(user, sharingOptions) {
        const { projectRoles, credentialRoles, isPersonalProject, personalProjectOwnerId, onlySharedWithMe, projectId, } = sharingOptions;
        const subquery = this.manager
            .createQueryBuilder()
            .select('sc.credentialsId')
            .from(entities_1.SharedCredentials, 'sc');
        if (isPersonalProject) {
            const ownerUserId = personalProjectOwnerId ?? user.id;
            subquery
                .innerJoin(entities_1.Project, 'p', 'sc.projectId = p.id')
                .innerJoin(entities_1.ProjectRelation, 'pr', 'pr.projectId = p.id')
                .where('sc.role = :ownerRole', { ownerRole: 'credential:owner' })
                .andWhere('pr.userId = :subqueryUserId', { subqueryUserId: ownerUserId })
                .andWhere('pr.role = :projectOwnerRole', { projectOwnerRole: permissions_1.PROJECT_OWNER_ROLE_SLUG });
            if (projectId && typeof projectId === 'string' && projectId !== '') {
                subquery.andWhere('sc.projectId = :subqueryProjectId', { subqueryProjectId: projectId });
            }
        }
        else if (onlySharedWithMe) {
            subquery
                .innerJoin(entities_1.Project, 'p', 'sc.projectId = p.id')
                .innerJoin(entities_1.ProjectRelation, 'pr', 'pr.projectId = p.id')
                .where('sc.role = :userRole', { userRole: 'credential:user' })
                .andWhere('pr.userId = :subqueryUserId', { subqueryUserId: user.id })
                .andWhere('pr.role = :projectOwnerRole', { projectOwnerRole: permissions_1.PROJECT_OWNER_ROLE_SLUG });
        }
        else if ((0, permissions_1.hasGlobalScope)(user, 'credential:read')) {
            if (projectId && typeof projectId === 'string' && projectId !== '') {
                subquery.where('sc.projectId = :subqueryProjectId', { subqueryProjectId: projectId });
            }
        }
        else {
            if (!credentialRoles || !projectRoles) {
                throw new Error('credentialRoles and projectRoles are required when not using special cases');
            }
            subquery
                .innerJoin(entities_1.Project, 'p', 'sc.projectId = p.id')
                .innerJoin(entities_1.ProjectRelation, 'pr', 'pr.projectId = p.id')
                .where('sc.role IN (:...credentialRoles)', { credentialRoles })
                .andWhere('pr.userId = :subqueryUserId', { subqueryUserId: user.id })
                .andWhere('pr.role IN (:...projectRoles)', { projectRoles });
            if (projectId && typeof projectId === 'string' && projectId !== '') {
                subquery.andWhere('sc.projectId = :subqueryProjectId', { subqueryProjectId: projectId });
            }
        }
        return subquery;
    }
};
exports.SharedCredentialsRepository = SharedCredentialsRepository;
exports.SharedCredentialsRepository = SharedCredentialsRepository = __decorate([
    (0, di_1.Service)(),
    __metadata("design:paramtypes", [typeorm_1.DataSource])
], SharedCredentialsRepository);
