// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/affinity/v1/resource_organization/operation_update.schema.js
/**
 * Affinity Node - Version 1 - Zod Schema
 * Discriminator: resource=organization, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('organization').default('organization'),
      operation: z.literal('update'),
      organizationId: stringOrExpression.optional(),
      updateFields: z.object({ domain: stringOrExpression.optional(), name: stringOrExpression.optional(), persons: z.array(z.string()).optional() }).optional(),
    }).optional(),
  });
};
