// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/azureStorage/v1/resource_blob/operation_get_all.schema.js
/**
 * Azure Storage Node - Version 1 - Zod Schema
 * Discriminator: resource=blob, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('blob'),
      operation: z.literal('getAll').default('getAll'),
      authentication: z.union([z.literal('oAuth2'), z.literal('sharedKey'), expressionSchema]).optional(),
      container: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      options: z.object({ fields: z.array(z.union([z.literal('copy'), z.literal('deleted'), z.literal('deletedwithversions'), z.literal('immutabilitypolicy'), z.literal('legalhold'), z.literal('metadata'), z.literal('permissions'), z.literal('snapshots'), z.literal('tags'), z.literal('uncommittedblobs'), z.literal('versions')])).optional(), filter: z.array(z.union([z.literal('deleted'), z.literal('files'), z.literal('directories')])).optional(), simplify: booleanOrExpression.optional(), upn: booleanOrExpression.optional() }).optional(),
      requestOptions: z.object({ batching: z.unknown().optional(), allowUnauthorizedCerts: z.boolean().optional(), proxy: stringOrExpression.optional(), timeout: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
