// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleSheets/v1/resource_sheet/operation_remove.ts
/**
 * Google Sheets  Node - Version 1
 * Discriminator: resource=sheet, operation=remove
 */


interface Credentials {
  googleApi: CredentialReference;
  googleSheetsOAuth2Api: CredentialReference;
}

/** Remove a sheet */
export type GoogleSheetsV1SheetRemoveParams = {
  resource: 'sheet';
  operation: 'remove';
/**
 * Authentication
 * @default serviceAccount
 */
    authentication?: 'serviceAccount' | 'oAuth2' | Expression<string>;
/**
 * The ID of the Google Spreadsheet. Found as part of the sheet URL https://docs.google.com/spreadsheets/d/{ID}/.
 */
    sheetId?: string | Expression<string> | PlaceholderValue;
/**
 * The table range to read from or to append data to. See the Google &lt;a href="https://developers.google.com/sheets/api/guides/values#writing"&gt;documentation&lt;/a&gt; for the details. If it contains multiple sheets it can also be added like this: "MySheet!A:F"
 * @default A:F
 */
    range?: string | Expression<string> | PlaceholderValue;
/**
 * Index of the first row which contains the actual data and not the keys. Starts with 0.
 * @displayOptions.hide { rawData: [true] }
 * @default 1
 */
    dataStartRow?: number | Expression<number>;
/**
 * Index of the row which contains the keys. Starts at 0. The incoming node data is matched to the keys for assignment. The matching is case sensitive.
 * @displayOptions.hide { rawData: [true] }
 * @default 0
 */
    keyRow?: number | Expression<number>;
/**
 * The ID of the sheet to delete
 */
    id?: string | Expression<string> | PlaceholderValue;
};

export type GoogleSheetsV1SheetRemoveNode = {
  type: 'n8n-nodes-base.googleSheets';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GoogleSheetsV1SheetRemoveParams>;
};
