// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleSheets/v1/resource_sheet/operation_read.schema.js
/**
 * Google Sheets  Node - Version 1 - Zod Schema
 * Discriminator: resource=sheet, operation=read
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('sheet').default('sheet'),
      operation: z.literal('read').default('read'),
      authentication: z.union([z.literal('serviceAccount'), z.literal('oAuth2'), expressionSchema]).optional(),
      sheetId: stringOrExpression.optional(),
      range: stringOrExpression.optional(),
      rawData: booleanOrExpression.optional(),
      dataProperty: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"rawData":[true]}}, defaults: {"rawData":false} }),
      dataStartRow: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"hide":{"rawData":[true]}}, defaults: {"rawData":false} }),
      keyRow: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"hide":{"rawData":[true]}}, defaults: {"rawData":false} }),
      options: z.object({ 'continue': booleanOrExpression.optional(), returnAllMatches: booleanOrExpression.optional(), usePathForKeyRow: booleanOrExpression.optional(), valueInputMode: z.union([z.literal('RAW'), z.literal('USER_ENTERED'), expressionSchema]).optional(), valueRenderMode: z.union([z.literal('FORMATTED_VALUE'), z.literal('FORMULA'), z.literal('UNFORMATTED_VALUE'), expressionSchema]).optional(), valueRenderMode: z.union([z.literal('FORMATTED_VALUE'), z.literal('FORMULA'), z.literal('UNFORMATTED_VALUE'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
