// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleSheets/v1/resource_sheet/operation_delete.schema.js
/**
 * Google Sheets  Node - Version 1 - Zod Schema
 * Discriminator: resource=sheet, operation=delete
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('sheet').default('sheet'),
      operation: z.literal('delete'),
      authentication: z.union([z.literal('serviceAccount'), z.literal('oAuth2'), expressionSchema]).optional(),
      sheetId: stringOrExpression.optional(),
      range: stringOrExpression.optional(),
      toDelete: z.object({ columns: z.array(z.object({ sheetId: stringOrExpression.optional(), startIndex: numberOrExpression.optional(), amount: numberOrExpression.optional() })).optional(), rows: z.array(z.object({ sheetId: stringOrExpression.optional(), startIndex: numberOrExpression.optional(), amount: numberOrExpression.optional() })).optional() }).optional(),
      dataStartRow: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"hide":{"rawData":[true]}} }),
      keyRow: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"hide":{"rawData":[true]}} }),
    }).optional(),
  });
};
