// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/stripe/v1/resource_charge/operation_get_all.ts
/**
 * Stripe Node - Version 1
 * Discriminator: resource=charge, operation=getAll
 */


interface Credentials {
  stripeApi: CredentialReference;
}

/** Get many charges */
export type StripeV1ChargeGetAllParams = {
  resource: 'charge';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
};

export type StripeV1ChargeGetAllOutput = {
  amount?: number;
  amount_captured?: number;
  amount_refunded?: number;
  application_fee?: null;
  captured?: boolean;
  created?: number;
  currency?: string;
  destination?: null;
  dispute?: null;
  disputed?: boolean;
  failure_balance_transaction?: null;
  id?: string;
  livemode?: boolean;
  object?: string;
  on_behalf_of?: null;
  order?: null;
  outcome?: {
    network_status?: string;
    risk_level?: string;
    seller_message?: string;
    type?: string;
  };
  paid?: boolean;
  payment_method_details?: {
    card?: {
      brand?: string;
      country?: string;
      exp_month?: number;
      exp_year?: number;
      extended_authorization?: {
        status?: string;
      };
      fingerprint?: string;
      funding?: string;
      incremental_authorization?: {
        status?: string;
      };
      installments?: null;
      last4?: string;
      mandate?: null;
      multicapture?: {
        status?: string;
      };
      network?: string;
      network_token?: {
        used?: boolean;
      };
      overcapture?: {
        maximum_amount_capturable?: number;
        status?: string;
      };
      regulated_status?: string;
    };
    type?: string;
  };
  refunded?: boolean;
  refunds?: {
    has_more?: boolean;
    object?: string;
    total_count?: number;
    url?: string;
  };
  review?: null;
  source_transfer?: null;
  status?: string;
  transfer_data?: null;
  transfer_group?: null;
};

export type StripeV1ChargeGetAllNode = {
  type: 'n8n-nodes-base.stripe';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<StripeV1ChargeGetAllParams>;
  output?: Items<StripeV1ChargeGetAllOutput>;
};
