// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/oneSimpleApi/v1/resource_website/operation_pdf.schema.js
/**
 * One Simple API Node - Version 1 - Zod Schema
 * Discriminator: resource=website, operation=pdf
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('website').default('website'),
      operation: z.literal('pdf').default('pdf'),
      link: stringOrExpression.optional(),
      download: booleanOrExpression.optional(),
      output: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"download":[true]}}, defaults: {"download":false} }),
      options: z.object({ page: z.union([z.literal('A0'), z.literal('A1'), z.literal('A2'), z.literal('A3'), z.literal('A4'), z.literal('A5'), z.literal('A6'), z.literal('Ledger'), z.literal('Legal'), z.literal('Letter'), z.literal('Tabloid'), expressionSchema]).optional(), force: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
