// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/oneSimpleApi/v1/resource_website/operation_pdf.ts
/**
 * One Simple API Node - Version 1
 * Discriminator: resource=website, operation=pdf
 */


interface Credentials {
  oneSimpleApi: CredentialReference;
}

/** Generate a PDF from a webpage */
export type OneSimpleApiV1WebsitePdfParams = {
  resource: 'website';
  operation: 'pdf';
/**
 * Link to webpage to convert
 */
    link?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to download the PDF or return a link to it
 * @default false
 */
    download?: boolean | Expression<boolean>;
/**
 * The name of the output field to put the binary file data in
 * @displayOptions.show { download: [true] }
 * @default data
 */
    output?: string | Expression<string> | PlaceholderValue;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Page Size
     */
    page?: 'A0' | 'A1' | 'A2' | 'A3' | 'A4' | 'A5' | 'A6' | 'Ledger' | 'Legal' | 'Letter' | 'Tabloid' | Expression<string>;
    /** Normally the API will reuse a previously taken screenshot of the URL to give a faster response. This option allows you to retake the screenshot at that exact time, for those times when it's necessary.
     * @default false
     */
    force?: boolean | Expression<boolean>;
  };
};

export type OneSimpleApiV1WebsitePdfNode = {
  type: 'n8n-nodes-base.oneSimpleApi';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<OneSimpleApiV1WebsitePdfParams>;
};
