// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/oneSimpleApi/v1/resource_website/operation_screenshot.schema.js
/**
 * One Simple API Node - Version 1 - Zod Schema
 * Discriminator: resource=website, operation=screenshot
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('website').default('website'),
      operation: z.literal('screenshot'),
      link: stringOrExpression.optional(),
      download: booleanOrExpression.optional(),
      output: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"download":[true]}}, defaults: {"download":false} }),
      options: z.object({ screen: z.union([z.literal('phone'), z.literal('phone-landscape'), z.literal('retina'), z.literal('tablet'), z.literal('tablet-landscape'), expressionSchema]).optional(), force: booleanOrExpression.optional(), fullpage: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
