// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftEntra/v1/resource_group/operation_get_all.ts
/**
 * Microsoft Entra ID Node - Version 1
 * Discriminator: resource=group, operation=getAll
 */


interface Credentials {
  microsoftEntraOAuth2Api: CredentialReference;
}

/** Retrieve a list of groups */
export type MicrosoftEntraV1GroupGetAllParams = {
  resource: 'group';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * &lt;a href="https://docs.microsoft.com/en-us/graph/query-parameters#filter-parameter"&gt;Query parameter&lt;/a&gt; to filter results by
 * @hint If empty, all the groups will be returned
 */
    filter?: string | Expression<string> | PlaceholderValue;
/**
 * Output
 * @default simple
 */
    output?: 'simple' | 'raw' | 'fields' | Expression<string>;
/**
 * The fields to add to the output
 * @displayOptions.show { output: ["fields"] }
 * @default []
 */
    fields?: string[];
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type MicrosoftEntraV1GroupGetAllOutput = {
  createdDateTime?: string;
  displayName?: string;
  id?: string;
  mailEnabled?: boolean;
  mailNickname?: string;
  securityEnabled?: boolean;
  securityIdentifier?: string;
};

export type MicrosoftEntraV1GroupGetAllNode = {
  type: 'n8n-nodes-base.microsoftEntra';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MicrosoftEntraV1GroupGetAllParams>;
  output?: Items<MicrosoftEntraV1GroupGetAllOutput>;
};
