// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftEntra/v1/resource_group/operation_update.schema.js
/**
 * Microsoft Entra ID Node - Version 1 - Zod Schema
 * Discriminator: resource=group, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('group'),
      operation: z.literal('update'),
      group: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      updateFields: z.object({ allowExternalSenders: booleanOrExpression.optional(), autoSubscribeNewMembers: booleanOrExpression.optional(), description: stringOrExpression.optional(), displayName: stringOrExpression.optional(), mailNickname: stringOrExpression.optional(), membershipRule: stringOrExpression.optional(), membershipRuleProcessingState: z.union([z.literal('On'), z.literal('Paused'), expressionSchema]).optional(), preferredDataLocation: stringOrExpression.optional(), securityEnabled: booleanOrExpression.optional(), uniqueName: stringOrExpression.optional(), visibility: z.union([z.literal('Private'), z.literal('Public'), expressionSchema]).optional() }).optional(),
      requestOptions: z.object({ batching: z.unknown().optional(), allowUnauthorizedCerts: z.boolean().optional(), proxy: stringOrExpression.optional(), timeout: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
