// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftEntra/v1/resource_group/operation_update.ts
/**
 * Microsoft Entra ID Node - Version 1
 * Discriminator: resource=group, operation=update
 */


interface Credentials {
  microsoftEntraOAuth2Api: CredentialReference;
}

/** Update a group */
export type MicrosoftEntraV1GroupUpdateParams = {
  resource: 'group';
  operation: 'update';
/**
 * Group to Update
 * @default {"mode":"list","value":""}
 */
    group?: { __rl: true; mode: 'list' | 'id'; value: string; cachedResultName?: string };
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** Whether people external to the organization can send messages to the group. Wait a few seconds before editing this field in a newly created group.
     * @default false
     */
    allowExternalSenders?: boolean | Expression<boolean>;
    /** Whether new members added to the group will be auto-subscribed to receive email notifications. Wait a few seconds before editing this field in a newly created group.
     * @default false
     */
    autoSubscribeNewMembers?: boolean | Expression<boolean>;
    /** Description for the group
     */
    description?: string | Expression<string> | PlaceholderValue;
    /** The name to display in the address book for the group
     */
    displayName?: string | Expression<string> | PlaceholderValue;
    /** The mail alias for the group. Only enter the local-part without the domain.
     */
    mailNickname?: string | Expression<string> | PlaceholderValue;
    /** The &lt;a href="https://learn.microsoft.com/en-us/entra/identity/users/groups-dynamic-membership"&gt;dynamic membership rules&lt;/a&gt;
     */
    membershipRule?: string | Expression<string> | PlaceholderValue;
    /** Indicates whether the dynamic membership processing is on or paused
     * @default On
     */
    membershipRuleProcessingState?: 'On' | 'Paused' | Expression<string>;
    /** A property set for the group that Office 365 services use to provision the corresponding data-at-rest resources (mailbox, OneDrive, groups sites, and so on)
     */
    preferredDataLocation?: string | Expression<string> | PlaceholderValue;
    /** Whether the group is a security group
     * @default true
     */
    securityEnabled?: boolean | Expression<boolean>;
    /** The unique identifier for the group, can only be updated if null, and is immutable once set
     */
    uniqueName?: string | Expression<string> | PlaceholderValue;
    /** Specifies the visibility of the group
     * @default Public
     */
    visibility?: 'Private' | 'Public' | Expression<string>;
  };
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type MicrosoftEntraV1GroupUpdateNode = {
  type: 'n8n-nodes-base.microsoftEntra';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MicrosoftEntraV1GroupUpdateParams>;
};
