// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftEntra/v1/resource_group/operation_create.schema.js
/**
 * Microsoft Entra ID Node - Version 1 - Zod Schema
 * Discriminator: resource=group, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('group'),
      operation: z.literal('create'),
      groupType: z.union([z.literal('Unified'), z.literal(''), expressionSchema]).optional(),
      displayName: stringOrExpression.optional(),
      mailNickname: stringOrExpression.optional(),
      mailEnabled: resolveSchema({ parameters, schema: booleanOrExpression, required: false, displayOptions: {"show":{"groupType":["Unified"]}}, defaults: {"groupType":""} }),
      membershipType: z.union([z.literal(''), z.literal('DynamicMembership'), expressionSchema]).optional(),
      securityEnabled: resolveSchema({ parameters, schema: booleanOrExpression, required: false, displayOptions: {"show":{"groupType":["Unified"]}}, defaults: {"groupType":""} }),
      additionalFields: z.object({ isAssignableToRole: booleanOrExpression.optional(), description: stringOrExpression.optional(), membershipRule: stringOrExpression.optional(), membershipRuleProcessingState: z.union([z.literal('On'), z.literal('Paused'), expressionSchema]).optional(), preferredDataLocation: stringOrExpression.optional(), uniqueName: stringOrExpression.optional(), visibility: z.union([z.literal('Private'), z.literal('Public'), expressionSchema]).optional() }).optional(),
      requestOptions: z.object({ batching: z.unknown().optional(), allowUnauthorizedCerts: z.boolean().optional(), proxy: stringOrExpression.optional(), timeout: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
