// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/readWriteFile/v11.schema.js
/**
 * Read/Write Files from Disk Node - Version 1.1 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    operation: z.union([z.literal('read'), z.literal('write')]).optional(),
    fileSelector: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"operation":["read"]}}, defaults: {"operation":"read"} }),
    options: resolveSchema({ parameters, schema: z.object({ fileExtension: stringOrExpression.optional(), fileName: stringOrExpression.optional(), mimeType: stringOrExpression.optional(), dataPropertyName: stringOrExpression.optional(), append: booleanOrExpression.optional() }), required: false, displayOptions: {"show":{"operation":["read","write"]}}, defaults: {"operation":"read"} }),
    fileName: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"operation":["write"]}}, defaults: {"operation":"read"} }),
    dataPropertyName: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"operation":["write"]}}, defaults: {"operation":"read"} }),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
