// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/quickbase/v1/resource_report/operation_run.schema.js
/**
 * Quick Base Node - Version 1 - Zod Schema
 * Discriminator: resource=report, operation=run
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('report'),
      operation: z.literal('run'),
      tableId: stringOrExpression.optional(),
      reportId: stringOrExpression.optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"hide":{"returnAll":[true]}}, defaults: {"returnAll":true} }),
    }).optional(),
  });
};
