// @n8n/n8n-nodes-langchain/dist/node-definitions/nodes/n8n-nodes-langchain/googleGemini/v1/resource_file_search/operation_delete_store.ts
/**
 * Google Gemini Node - Version 1
 * Discriminator: resource=fileSearch, operation=deleteStore
 */


interface Credentials {
  googlePalmApi: CredentialReference;
}

/** Delete a File Search store */
export type LcGoogleGeminiV1FileSearchDeleteStoreParams = {
  resource: 'fileSearch';
  operation: 'deleteStore';
/**
 * The full name of the File Search store to delete (format: fileSearchStores/...)
 */
    fileSearchStoreName?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to delete related Documents and objects. If false, deletion will fail if the store contains any Documents.
 * @default false
 */
    force?: boolean | Expression<boolean>;
};

export interface LcGoogleGeminiV1FileSearchDeleteStoreSubnodeConfig {
  tools?: ToolInstance[];
}

export type LcGoogleGeminiV1FileSearchDeleteStoreNode = {
  type: '@n8n/n8n-nodes-langchain.googleGemini';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<LcGoogleGeminiV1FileSearchDeleteStoreParams> & { subnodes?: LcGoogleGeminiV1FileSearchDeleteStoreSubnodeConfig };
};
