// @n8n/n8n-nodes-langchain/dist/node-definitions/nodes/n8n-nodes-langchain/googleGemini/v1/resource_file_search/operation_upload_to_store.ts
/**
 * Google Gemini Node - Version 1
 * Discriminator: resource=fileSearch, operation=uploadToStore
 */


interface Credentials {
  googlePalmApi: CredentialReference;
}

/** Upload a file to a File Search store for RAG (Retrieval Augmented Generation) */
export type LcGoogleGeminiV1FileSearchUploadToStoreParams = {
  resource: 'fileSearch';
  operation: 'uploadToStore';
/**
 * The full name of the File Search store to upload to (format: fileSearchStores/...)
 */
    fileSearchStoreName?: string | Expression<string> | PlaceholderValue;
/**
 * A human-readable name for the file (will be visible in citations)
 */
    displayName?: string | Expression<string> | PlaceholderValue;
/**
 * Input Type
 * @default url
 */
    inputType?: 'url' | 'binary' | Expression<string>;
/**
 * URL of the file to upload
 * @displayOptions.show { inputType: ["url"] }
 */
    fileUrl?: string | Expression<string> | PlaceholderValue;
/**
 * Name of the binary property which contains the file
 * @hint The name of the input field containing the binary file data to be processed
 * @displayOptions.show { inputType: ["binary"] }
 * @default data
 */
    binaryPropertyName?: string | Expression<string> | PlaceholderValue;
};

export interface LcGoogleGeminiV1FileSearchUploadToStoreSubnodeConfig {
  tools?: ToolInstance[];
}

export type LcGoogleGeminiV1FileSearchUploadToStoreNode = {
  type: '@n8n/n8n-nodes-langchain.googleGemini';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<LcGoogleGeminiV1FileSearchUploadToStoreParams> & { subnodes?: LcGoogleGeminiV1FileSearchUploadToStoreSubnodeConfig };
};
