// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/onfleet/v1/resource_container/operation_add_task.schema.js
/**
 * Onfleet Node - Version 1 - Zod Schema
 * Discriminator: resource=container, operation=addTask
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('container'),
      operation: z.literal('addTask'),
      containerType: z.union([z.literal('organizations'), z.literal('teams'), z.literal('workers'), expressionSchema]).optional(),
      containerId: stringOrExpression.optional(),
      type: z.union([z.literal(-1), z.literal(0), z.literal(1), expressionSchema]).optional(),
      index: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"type":[1]}}, defaults: {"type":""} }),
      tasks: stringOrExpression.optional(),
      options: z.object({ considerDependencies: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
