// @n8n/n8n-nodes-langchain/dist/nodes/vendors/Microsoft/microsoft-utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.configureActivityCallback = exports.microsoftMcpServers = void 0;
exports.extractActivityInfo = extractActivityInfo;
exports.createMicrosoftAgentApplication = createMicrosoftAgentApplication;
exports.getMicrosoftMcpTools = getMicrosoftMcpTools;
exports.configureAdapterProcessCallback = configureAdapterProcessCallback;
const agents_hosting_1 = require("@microsoft/agents-hosting");
const n8n_workflow_1 = require("n8n-workflow");
const agents_a365_observability_1 = require("@microsoft/agents-a365-observability");
const agents_a365_runtime_1 = require("@microsoft/agents-a365-runtime");
const agents_activity_1 = require("@microsoft/agents-activity");
const uuid_1 = require("uuid");
const langchain_utils_1 = require("./langchain-utils");
const agents_a365_tooling_1 = require("@microsoft/agents-a365-tooling");
const utils_1 = require("../../mcp/shared/utils");
const utils_2 = require("../../mcp/McpClientTool/utils");
function extractActivityInfo(activity) {
    return {
        id: activity.id,
        type: activity.type,
        channelId: activity.channelId,
        conversationId: activity.conversation?.id,
        from: activity.from
            ? {
                id: activity.from.id,
                name: activity.from.name,
            }
            : undefined,
        recipient: activity.recipient
            ? {
                id: activity.recipient.id,
                name: activity.recipient.name,
            }
            : undefined,
        timestamp: activity.timestamp instanceof Date ? activity.timestamp.toISOString() : activity.timestamp,
        locale: activity.locale,
    };
}
exports.microsoftMcpServers = [
    { name: 'Calendar', value: 'mcp_CalendarTools' },
    { name: 'Mail', value: 'mcp_MailTools' },
    { name: 'Me', value: 'mcp_MeServer' },
    { name: 'OneDrive & SharePoint', value: 'mcp_ODSPRemoteServer' },
    { name: 'SharePoint Lists', value: 'mcp_SharePointListsTools' },
    { name: 'Teams', value: 'mcp_TeamsServer' },
    { name: 'Teams Canary', value: 'mcp_TeamsCanaryServer' },
    { name: 'Word', value: 'mcp_WordServer' },
];
const MS_TENANT_ID_HEADER = 'x-ms-tenant-id';
function isMicrosoftObservabilityEnabled() {
    return (process.env.ENABLE_OBSERVABILITY === 'true' &&
        process.env.ENABLE_A365_OBSERVABILITY_EXPORTER === 'true');
}
function createMicrosoftAgentApplication(credentials) {
    const authConfig = createAuthConfig(credentials);
    const adapter = new agents_hosting_1.CloudAdapter(authConfig);
    const storage = new agents_hosting_1.MemoryStorage();
    const agent = new agents_hosting_1.AgentApplication({
        adapter,
        storage,
        authorization: {
            agentic: {
                type: 'agentic',
                scopes: ['https://graph.microsoft.com/.default'],
            },
        },
    });
    return agent;
}
async function getMicrosoftMcpTools(turnContext, mcpAuthToken, selectedTools) {
    const configService = new agents_a365_tooling_1.McpToolServerConfigurationService();
    agents_a365_tooling_1.Utility.ValidateAuthToken(mcpAuthToken);
    const agenticAppId = agents_a365_runtime_1.Utility.ResolveAgentIdentity(turnContext, mcpAuthToken);
    let servers = await configService.listToolServers(agenticAppId, mcpAuthToken);
    if (servers.length === 0)
        return undefined;
    if (selectedTools?.length) {
        servers = servers.filter((server) => selectedTools.includes(server.mcpServerName));
    }
    const tenantId = turnContext.activity.recipient?.tenantId || turnContext.activity?.channelData?.tenant?.id;
    const tools = [];
    const clients = [];
    const timeout = 60000;
    for (const server of servers) {
        const headers = {};
        if (mcpAuthToken) {
            headers['Authorization'] = `Bearer ${mcpAuthToken}`;
        }
        if (tenantId) {
            headers[MS_TENANT_ID_HEADER] = tenantId;
        }
        const clientResult = await (0, utils_1.connectMcpClient)({
            serverTransport: 'httpStreamable',
            endpointUrl: server.url,
            headers,
            name: 'Microsoft-Agent-365',
            version: 1,
        });
        if (!clientResult.ok) {
            console.error(`Failed to connect to MCP server ${server.mcpServerName}:`, clientResult.error);
            continue;
        }
        const client = clientResult.result;
        clients.push(client);
        const mcpTools = await (0, utils_1.getAllTools)(client);
        for (const tool of mcpTools) {
            const callToolFunc = (0, utils_2.createCallTool)(tool.name, client, timeout, (errorMessage) => {
                console.error(`Tool "${tool.name}" execution error:`, errorMessage);
            });
            const dynamicTool = (0, utils_2.mcpToolToDynamicTool)(tool, callToolFunc);
            tools.push(dynamicTool);
        }
    }
    if (tools.length === 0)
        return undefined;
    return {
        tools,
        client: {
            async close() {
                await Promise.all(clients.map(async (c) => await c.close()));
            },
        },
    };
}
const configureActivityCallback = (nodeContext, credentials, mcpTokenRef) => {
    const systemPrompt = nodeContext.getNodeParameter('systemPrompt');
    const { clientId, tenantId } = credentials;
    return async (turnContext) => {
        const agentId = turnContext.activity.recipient?.agenticAppId ?? clientId;
        const agentName = turnContext.activity.recipient?.name ?? 'Microsoft Agent 365';
        const tenantDetails = {
            tenantId: turnContext.activity.recipient?.tenantId ?? tenantId ?? '',
        };
        const conversationId = turnContext.activity.conversation?.id;
        const inputText = turnContext.activity.text || '';
        const baggageScope = new agents_a365_observability_1.BaggageBuilder()
            .tenantId(tenantDetails.tenantId)
            .agentId(agentId)
            .correlationId((0, uuid_1.v4)())
            .agentName(agentName)
            .conversationId(conversationId)
            .build();
        await baggageScope.run(async () => {
            const invokeAgentDetails = {
                agentId,
                agentName,
                conversationId,
                request: {
                    content: inputText || 'Unknown text',
                    executionType: agents_a365_observability_1.ExecutionType.HumanToAgent,
                    sessionId: conversationId,
                },
            };
            const invokeAgentScope = agents_a365_observability_1.InvokeAgentScope.start(invokeAgentDetails, tenantDetails);
            await invokeAgentScope.withActiveSpanAsync(async () => {
                invokeAgentScope.recordInputMessages([inputText || 'Unknown text']);
                let microsoftMcpTools = undefined;
                let mcpClient = undefined;
                if (mcpTokenRef.token) {
                    try {
                        const useMcpTools = nodeContext.getNodeParameter('useMcpTools', false);
                        if (useMcpTools) {
                            let selectedTools = undefined;
                            const include = nodeContext.getNodeParameter('include', 'all');
                            if (include === 'selected') {
                                const selected = nodeContext.getNodeParameter('includeTools', []);
                                selectedTools = exports.microsoftMcpServers
                                    .filter((server) => selected.includes(server.value))
                                    .map((server) => server.value);
                            }
                            const result = await getMicrosoftMcpTools(turnContext, mcpTokenRef.token, selectedTools);
                            mcpClient = result?.client;
                            microsoftMcpTools = result?.tools;
                        }
                    }
                    catch (error) {
                        console.log('Error retrieving MCP tools');
                    }
                }
                try {
                    const response = await (0, langchain_utils_1.invokeAgent)(nodeContext, inputText, systemPrompt, {
                        configurable: { thread_id: turnContext.activity.conversation.id },
                    }, microsoftMcpTools);
                    invokeAgentScope.recordOutputMessages([`n8n Agent Response: ${response}`]);
                    await turnContext.sendActivity(response);
                }
                finally {
                    if (mcpClient)
                        await mcpClient.close();
                    invokeAgentScope.dispose();
                }
            });
        });
    };
};
exports.configureActivityCallback = configureActivityCallback;
function configureAdapterProcessCallback(nodeContext, agent, credentials, activityCapture) {
    return async (turnContext) => {
        const { token: aauToken } = await agent.authorization.exchangeToken(turnContext, (0, agents_a365_runtime_1.getObservabilityAuthenticationScope)(), 'agentic');
        let observability;
        if (isMicrosoftObservabilityEnabled()) {
            observability = agents_a365_observability_1.ObservabilityManager.configure((builder) => builder
                .withService('TypeScript Sample Agent', '1.0.0')
                .withTokenResolver((_agentId, _tenantId) => aauToken || ''));
            observability.start();
        }
        const mcpTokenRef = { token: undefined };
        try {
            turnContext.turnState.set('AgenticAuthorization/agentic', undefined);
            const tokenResult = await agent.authorization.exchangeToken(turnContext, 'agentic', {
                scopes: [(0, agents_a365_runtime_1.getMcpPlatformAuthenticationScope)()],
            });
            mcpTokenRef.token = tokenResult.token;
        }
        catch (error) {
            console.error('Error getting MCP token');
        }
        try {
            const originalSendActivity = turnContext.sendActivity.bind(turnContext);
            activityCapture.input = turnContext.activity.text || '';
            activityCapture.activity = extractActivityInfo(turnContext.activity);
            const sendActivityWrapper = async (activityOrText) => {
                if (typeof activityOrText === 'string') {
                    activityCapture.output.push(activityOrText);
                }
                else if (activityOrText.text) {
                    activityCapture.output.push(activityOrText.text);
                }
                return await originalSendActivity(activityOrText);
            };
            turnContext.sendActivity = sendActivityWrapper;
            const welcomeMessage = nodeContext.getNodeParameter('options.welcomeMessage', "Hello! I'm here to help you!");
            agent.onConversationUpdate('membersAdded', async (context) => {
                await context.sendActivity(welcomeMessage);
            });
            const onActivity = (0, exports.configureActivityCallback)(nodeContext, credentials, mcpTokenRef);
            agent.onActivity(agents_activity_1.ActivityTypes.Message, onActivity, ['agentic']);
            await agent.run(turnContext);
        }
        catch (error) {
            throw new n8n_workflow_1.NodeOperationError(nodeContext.getNode(), error);
        }
        finally {
            if (observability) {
                await observability.shutdown();
            }
        }
    };
}
const createAuthConfig = (credentials) => {
    const { clientId, tenantId, clientSecret } = credentials;
    const connections = new Map();
    connections.set('serviceConnection', {
        clientId,
        clientSecret,
        tenantId,
        authority: 'https://login.microsoftonline.com',
        issuers: [
            'https://api.botframework.com',
            `https://sts.windows.net/${tenantId}/`,
            `https://login.microsoftonline.com/${tenantId}/v2.0`,
        ],
    });
    const config = {
        clientId,
        clientSecret,
        tenantId,
        authority: 'https://login.microsoftonline.com',
        issuers: [
            'https://api.botframework.com',
            `https://sts.windows.net/${tenantId}/`,
            `https://login.microsoftonline.com/${tenantId}/v2.0`,
        ],
        connections,
        connectionsMap: [
            {
                connection: 'serviceConnection',
                serviceUrl: '*',
            },
        ],
    };
    return config;
};
