// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/github/v11/resource_release/operation_get_all.ts
/**
 * GitHub Node - Version 1.1
 * Discriminator: resource=release, operation=getAll
 */


interface Credentials {
  githubApi: CredentialReference;
  githubOAuth2Api: CredentialReference;
}

/** Get many repository releases */
export type GithubV11ReleaseGetAllParams = {
  resource: 'release';
  operation: 'getAll';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Repository Owner
 * @displayOptions.hide { operation: ["invite", "getUserIssues"] }
 * @default {"mode":"list","value":""}
 */
    owner?: { __rl: true; mode: 'list' | 'url' | 'name'; value: string; cachedResultName?: string };
/**
 * Repository Name
 * @displayOptions.hide { resource: ["user", "organization"], operation: ["getRepositories"] }
 * @default {"mode":"list","value":""}
 */
    repository?: { __rl: true; mode: 'list' | 'url' | 'name'; value: string; cachedResultName?: string };
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
};

export type GithubV11ReleaseGetAllOutput = {
  assets?: Array<{
    browser_download_url?: string;
    content_type?: string;
    created_at?: string;
    download_count?: number;
    id?: number;
    label?: null;
    name?: string;
    node_id?: string;
    size?: number;
    state?: string;
    updated_at?: string;
    uploader?: {
      avatar_url?: string;
      events_url?: string;
      followers_url?: string;
      following_url?: string;
      gists_url?: string;
      gravatar_id?: string;
      html_url?: string;
      id?: number;
      login?: string;
      node_id?: string;
      organizations_url?: string;
      received_events_url?: string;
      repos_url?: string;
      site_admin?: boolean;
      starred_url?: string;
      subscriptions_url?: string;
      type?: string;
      url?: string;
      user_view_type?: string;
    };
    url?: string;
  }>;
  assets_url?: string;
  author?: {
    avatar_url?: string;
    events_url?: string;
    followers_url?: string;
    following_url?: string;
    gists_url?: string;
    gravatar_id?: string;
    html_url?: string;
    id?: number;
    login?: string;
    node_id?: string;
    organizations_url?: string;
    received_events_url?: string;
    repos_url?: string;
    site_admin?: boolean;
    starred_url?: string;
    subscriptions_url?: string;
    type?: string;
    url?: string;
    user_view_type?: string;
  };
  body?: string;
  created_at?: string;
  draft?: boolean;
  html_url?: string;
  id?: number;
  mentions_count?: number;
  name?: string;
  node_id?: string;
  prerelease?: boolean;
  tag_name?: string;
  target_commitish?: string;
  upload_url?: string;
  url?: string;
};

export type GithubV11ReleaseGetAllNode = {
  type: 'n8n-nodes-base.github';
  version: 1.1;
  credentials?: Credentials;
  config: NodeConfig<GithubV11ReleaseGetAllParams>;
  output?: Items<GithubV11ReleaseGetAllOutput>;
};
