// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/stravaTrigger/v1.schema.js
/**
 * Strava Trigger Node - Version 1 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    object: z.union([z.literal('*'), z.literal('activity'), z.literal('athlete'), expressionSchema]).optional(),
    event: z.union([z.literal('*'), z.literal('create'), z.literal('delete'), z.literal('update'), expressionSchema]).optional(),
    resolveData: booleanOrExpression.optional(),
    options: z.object({ deleteIfExist: booleanOrExpression.optional() }).optional(),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
