// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/lemlist/v2/resource_activity/operation_get_all.ts
/**
 * Lemlist Node - Version 2
 * Discriminator: resource=activity, operation=getAll
 */


interface Credentials {
  lemlistApi: CredentialReference;
}

export type LemlistV2ActivityGetAllParams = {
  resource: 'activity';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 5
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** ID of the campaign to retrieve activity for. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    campaignId?: string | Expression<string>;
    /** Is First
     * @default false
     */
    isFirst?: boolean | Expression<boolean>;
    /** Lead ID
     */
    leadId?: string | Expression<string> | PlaceholderValue;
    /** Type of activity to retrieve
     * @default emailsOpened
     */
    type?: 'aircallCreated' | 'aircallDone' | 'aircallEnded' | 'aircallInterested' | 'aircallNotInterested' | 'apiDone' | 'apiFailed' | 'apiInterested' | 'apiNotInterested' | 'attracted' | 'connectionIssue' | 'contacted' | 'customDomainErrors' | 'emailsBounced' | 'emailsClicked' | 'emailsFailed' | 'emailsInterested' | 'emailsNotInterested' | 'emailsOpened' | 'emailsReplied' | 'emailsSendFailed' | 'emailsSent' | 'emailsUnsubscribed' | 'hooked' | 'interested' | 'lemwarmPaused' | 'linkedinInterested' | 'linkedinInviteAccepted' | 'linkedinInviteDone' | 'linkedinInviteFailed' | 'linkedinNotInterested' | 'linkedinReplied' | 'linkedinSendFailed' | 'linkedinSent' | 'linkedinVisitDone' | 'linkedinVisitFailed' | 'linkedinVoiceNoteDone' | 'linkedinVoiceNoteFailed' | 'manualInterested' | 'manualNotInterested' | 'notInterested' | 'opportunitiesDone' | 'paused' | 'resumed' | 'sendLimitReached' | 'skipped' | 'warmed' | Expression<string>;
    /** Version
     * @default v2
     */
    version?: string | Expression<string> | PlaceholderValue;
  };
};

export type LemlistV2ActivityGetAllOutput = {
  _id?: string;
  bot?: boolean;
  campaignId?: string;
  campaignName?: string;
  companyName?: string;
  contactId?: string;
  createdAt?: string;
  createdBy?: string;
  email?: string;
  emailTemplateId?: string;
  emailTemplateName?: string;
  firstName?: string;
  isFirst?: boolean;
  lastName?: string;
  leadCompanyName?: string;
  leadEmail?: string;
  leadFirstName?: string;
  leadId?: string;
  leadLastName?: string;
  leadPhone?: string;
  linkedinUrl?: string;
  metaData?: {
    campaignId?: string;
    createdBy?: string;
    leadId?: string;
    taskId?: string;
    teamId?: string;
    type?: string;
  };
  name?: string;
  phone?: string;
  relatedSentAt?: string;
  sendUserName?: string;
  sequenceId?: string;
  sequenceStep?: number;
  sequenceTested?: string;
  stopped?: boolean;
  teamId?: string;
  totalSequenceStep?: number;
  type?: string;
};

export type LemlistV2ActivityGetAllNode = {
  type: 'n8n-nodes-base.lemlist';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<LemlistV2ActivityGetAllParams>;
  output?: Items<LemlistV2ActivityGetAllOutput>;
};
