// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/beeminder/v1/resource_datapoint/operation_update.ts
/**
 * Beeminder Node - Version 1
 * Discriminator: resource=datapoint, operation=update
 */


interface Credentials {
  beeminderApi: CredentialReference;
  beeminderOAuth2Api: CredentialReference;
}

/** Update a datapoint */
export type BeeminderV1DatapointUpdateParams = {
  resource: 'datapoint';
  operation: 'update';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * The name of the goal. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    goalName?: string | Expression<string>;
/**
 * Datapoint ID
 */
    datapointId?: string | Expression<string> | PlaceholderValue;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** Datapoint value to send
     * @default 1
     */
    value?: number | Expression<number>;
    /** Comment
     */
    comment?: string | Expression<string> | PlaceholderValue;
    /** Defaults to "now" if none is passed in, or the existing timestamp if the datapoint is being updated rather than created
     */
    timestamp?: string | Expression<string>;
  };
};

export type BeeminderV1DatapointUpdateNode = {
  type: 'n8n-nodes-base.beeminder';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<BeeminderV1DatapointUpdateParams>;
};
