// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/wooCommerce/v1/resource_customer/operation_get.ts
/**
 * WooCommerce Node - Version 1
 * Discriminator: resource=customer, operation=get
 */


interface Credentials {
  wooCommerceApi: CredentialReference;
}

/** Retrieve a customer */
export type WooCommerceV1CustomerGetParams = {
  resource: 'customer';
  operation: 'get';
/**
 * ID of the customer to retrieve
 */
    customerId?: string | Expression<string> | PlaceholderValue;
};

export type WooCommerceV1CustomerGetOutput = {
  _links?: {
    collection?: Array<{
      href?: string;
    }>;
    self?: Array<{
      href?: string;
    }>;
  };
  avatar_url?: string;
  billing?: {
    address_1?: string;
    address_2?: string;
    city?: string;
    company?: string;
    country?: string;
    email?: string;
    first_name?: string;
    last_name?: string;
    phone?: string;
    postcode?: string;
    state?: string;
  };
  date_created?: string;
  date_created_gmt?: string;
  date_modified?: string;
  date_modified_gmt?: string;
  email?: string;
  first_name?: string;
  id?: number;
  is_paying_customer?: boolean;
  last_name?: string;
  meta_data?: Array<{
    id?: number;
    key?: string;
  }>;
  role?: string;
  shipping?: {
    address_1?: string;
    address_2?: string;
    city?: string;
    company?: string;
    country?: string;
    first_name?: string;
    last_name?: string;
    phone?: string;
    postcode?: string;
    state?: string;
  };
  username?: string;
};

export type WooCommerceV1CustomerGetNode = {
  type: 'n8n-nodes-base.wooCommerce';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<WooCommerceV1CustomerGetParams>;
  output?: Items<WooCommerceV1CustomerGetOutput>;
};
