// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/disqus/v1/resource_forum/operation_get_posts.schema.js
/**
 * Disqus Node - Version 1 - Zod Schema
 * Discriminator: resource=forum, operation=getPosts
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('forum').default('forum'),
      operation: z.literal('getPosts'),
      id: stringOrExpression.optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      additionalFields: z.object({ filters: z.array(z.union([z.literal('Has_Bad_Word'), z.literal('Has_Link'), z.literal('Has_Low_Rep_Author'), z.literal('Has_Media'), z.literal('Is_Anonymous'), z.literal('Is_At_Flag_Limit'), z.literal('Is_Flagged'), z.literal('Is_Toxic'), z.literal('Modified_By_Rule'), z.literal('No_Issue'), z.literal('Shadow_Banned')])).optional(), include: z.array(z.union([z.literal('approved')])).optional(), order: z.union([z.literal('asc'), z.literal('desc'), expressionSchema]).optional(), query: stringOrExpression.optional(), related: z.array(z.union([z.literal('thread')])).optional(), since: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
