// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/disqus/v1/resource_forum/operation_get_categories.ts
/**
 * Disqus Node - Version 1
 * Discriminator: resource=forum, operation=getCategories
 */


interface Credentials {
  disqusApi: CredentialReference;
}

/** Return a list of categories within a forum */
export type DisqusV1ForumGetCategoriesParams = {
  resource: 'forum';
  operation: 'getCategories';
/**
 * The short name(aka ID) of the forum to get Categories
 */
    id?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** You may specify order to sort your response
     * @default asc
     */
    order?: 'asc' | 'desc' | Expression<string>;
  };
};

export type DisqusV1ForumGetCategoriesNode = {
  type: 'n8n-nodes-base.disqus';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<DisqusV1ForumGetCategoriesParams>;
};
