// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/disqus/v1/resource_forum/operation_get_posts.ts
/**
 * Disqus Node - Version 1
 * Discriminator: resource=forum, operation=getPosts
 */


interface Credentials {
  disqusApi: CredentialReference;
}

/** Return a list of posts within a forum */
export type DisqusV1ForumGetPostsParams = {
  resource: 'forum';
  operation: 'getPosts';
/**
 * The short name(aka ID) of the forum to get
 */
    id?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** You may specify filters for your response
     * @default []
     */
    filters?: Array<'Has_Bad_Word' | 'Has_Link' | 'Has_Low_Rep_Author' | 'Has_Media' | 'Is_Anonymous' | 'Is_At_Flag_Limit' | 'Is_Flagged' | 'Is_Toxic' | 'Modified_By_Rule' | 'No_Issue' | 'Shadow_Banned'>;
    /** You may specify relations to include with your response
     * @default []
     */
    include?: Array<'approved'>;
    /** You may specify order to sort your response
     * @default asc
     */
    order?: 'asc' | 'desc' | Expression<string>;
    /** You may specify query forChoices: asc, desc your response
     */
    query?: string | Expression<string> | PlaceholderValue;
    /** You may specify relations to include with your response
     * @default []
     */
    related?: Array<'thread'>;
    /** Unix timestamp (or ISO datetime standard)
     */
    since?: string | Expression<string>;
  };
};

export type DisqusV1ForumGetPostsNode = {
  type: 'n8n-nodes-base.disqus';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<DisqusV1ForumGetPostsParams>;
};
