// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/disqus/v1/resource_forum/operation_get.schema.js
/**
 * Disqus Node - Version 1 - Zod Schema
 * Discriminator: resource=forum, operation=get
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('forum').default('forum'),
      operation: z.literal('get').default('get'),
      id: stringOrExpression.optional(),
      additionalFields: z.object({ attach: z.array(z.union([z.literal('counters'), z.literal('followsForum'), z.literal('forumCanDisableAds'), z.literal('forumDaysAlive'), z.literal('forumFeatures'), z.literal('forumForumCategory'), z.literal('forumIntegration'), z.literal('forumNewPolicy'), z.literal('forumPermissions')])).optional(), related: z.array(z.union([z.literal('author')])).optional() }).optional(),
    }).optional(),
  });
};
