// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/disqus/v1/resource_forum/operation_get_threads.schema.js
/**
 * Disqus Node - Version 1 - Zod Schema
 * Discriminator: resource=forum, operation=getThreads
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('forum').default('forum'),
      operation: z.literal('getThreads'),
      id: stringOrExpression.optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      additionalFields: z.object({ related: z.array(z.union([z.literal('author'), z.literal('forum')])).optional(), include: z.array(z.union([z.literal('closed'), z.literal('open'), z.literal('killed')])).optional(), order: z.union([z.literal('asc'), z.literal('desc'), expressionSchema]).optional(), since: stringOrExpression.optional(), thread: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
