// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/webflow/v1/resource_item/operation_update.schema.js
/**
 * Webflow Node - Version 1 - Zod Schema
 * Discriminator: resource=item, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('item').default('item'),
      operation: z.literal('update'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      siteId: stringOrExpression.optional(),
      collectionId: stringOrExpression.optional(),
      itemId: stringOrExpression.optional(),
      live: booleanOrExpression.optional(),
      fieldsUi: z.object({ fieldValues: z.array(z.object({ fieldId: stringOrExpression.optional(), fieldValue: stringOrExpression.optional() })).optional() }).optional(),
    }).optional(),
  });
};
