// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/webflow/v1/resource_item/operation_update.ts
/**
 * Webflow Node - Version 1
 * Discriminator: resource=item, operation=update
 */


interface Credentials {
  webflowApi: CredentialReference;
  webflowOAuth2Api: CredentialReference;
}

export type WebflowV1ItemUpdateParams = {
  resource: 'item';
  operation: 'update';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * ID of the site containing the collection whose items to update. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    siteId?: string | Expression<string>;
/**
 * ID of the collection whose items to update. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    collectionId?: string | Expression<string>;
/**
 * ID of the item to update
 */
    itemId?: string | Expression<string> | PlaceholderValue;
/**
 * Whether the item should be published on the live site
 * @default false
 */
    live?: boolean | Expression<boolean>;
/**
 * Fields
 * @default {}
 */
    fieldsUi?: {
        /** Field
     */
    fieldValues?: Array<{
      /** Field to set for the item to update. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
       */
      fieldId?: string | Expression<string>;
      /** Value to set for the item to update
       */
      fieldValue?: string | Expression<string> | PlaceholderValue;
    }>;
  };
};

export type WebflowV1ItemUpdateNode = {
  type: 'n8n-nodes-base.webflow';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<WebflowV1ItemUpdateParams>;
};
