// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/todoist/v21/resource_reminder/operation_create.schema.js
/**
 * Todoist Node - Version 2.1 - Zod Schema
 * Discriminator: resource=reminder, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('reminder'),
      operation: z.literal('create').default('create'),
      authentication: z.union([z.literal('apiKey'), z.literal('oAuth2'), expressionSchema]).optional(),
      itemId: stringOrExpression.optional(),
      dueDateType: z.union([z.literal('natural_language'), z.literal('full_day'), z.literal('floating_time'), z.literal('fixed_timezone'), expressionSchema]).optional(),
      natural_language_representation: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"dueDateType":["natural_language"]}}, defaults: {"dueDateType":"natural_language"} }),
      date: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"dueDateType":["full_day"]}}, defaults: {"dueDateType":"natural_language"} }),
      datetime: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"dueDateType":["floating_time","fixed_timezone"]}}, defaults: {"dueDateType":"natural_language"} }),
      timezone: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"dueDateType":["fixed_timezone"]}}, defaults: {"dueDateType":"natural_language"} }),
      reminderOptions: z.object({ type: z.union([z.literal('absolute'), z.literal('relative'), expressionSchema]).optional(), minute_offset: numberOrExpression.optional(), notify_uid: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
