// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftEntra/v1/resource_user/operation_get.ts
/**
 * Microsoft Entra ID Node - Version 1
 * Discriminator: resource=user, operation=get
 */


interface Credentials {
  microsoftEntraOAuth2Api: CredentialReference;
}

/** Retrieve data for a specific group */
export type MicrosoftEntraV1UserGetParams = {
  resource: 'user';
  operation: 'get';
/**
 * User to Get
 * @default {"mode":"list","value":""}
 */
    user?: { __rl: true; mode: 'list' | 'id'; value: string; cachedResultName?: string };
/**
 * Output
 * @default simple
 */
    output?: 'simple' | 'raw' | 'fields' | Expression<string>;
/**
 * The fields to add to the output
 * @displayOptions.show { output: ["fields"] }
 * @default []
 */
    fields?: string[];
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type MicrosoftEntraV1UserGetOutput = {
  '@odata.context'?: string;
  createdDateTime?: string;
  displayName?: string;
  id?: string;
  mailNickname?: string;
  securityIdentifier?: string;
  userPrincipalName?: string;
};

export type MicrosoftEntraV1UserGetNode = {
  type: 'n8n-nodes-base.microsoftEntra';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MicrosoftEntraV1UserGetParams>;
  output?: Items<MicrosoftEntraV1UserGetOutput>;
};
