// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/dataTable/v11/resource_table/operation_create.schema.js
/**
 * Data table Node - Version 1.1 - Zod Schema
 * Discriminator: resource=table, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('table'),
      operation: z.literal('create'),
      tableName: stringOrExpression.optional(),
      columns: z.object({ column: z.array(z.object({ name: stringOrExpression.optional(), type: z.union([z.literal('boolean'), z.literal('date'), z.literal('number'), z.literal('string'), expressionSchema]).optional() })).optional() }).optional(),
      options: z.object({ createIfNotExists: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
