// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/set/v33/mode_raw.ts
/**
 * Edit Fields (Set) Node - Version 3.3
 * Discriminator: mode=raw
 */


/** Customize item output with JSON */
export type SetV33RawParams = {
  mode: 'raw';
/**
 * Whether this item should be duplicated a set number of times
 * @default false
 */
    duplicateItem?: boolean | Expression<boolean>;
/**
 * How many times the item should be duplicated, mainly used for testing and debugging
 * @displayOptions.show { duplicateItem: [true] }
 * @default 0
 */
    duplicateCount?: number | Expression<number>;
/**
 * JSON
 */
    jsonOutput?: IDataObject | string | Expression<string>;
/**
 * Whether to pass to the output all the input fields (along with the fields set in 'Fields to Set')
 * @default false
 */
    includeOtherFields?: boolean | Expression<boolean>;
/**
 * How to select the fields you want to include in your output items
 * @displayOptions.hide { /includeOtherFields: [false] }
 * @default all
 */
    include?: 'all' | 'selected' | 'except' | Expression<string>;
/**
 * Comma-separated list of the field names you want to include in the output. You can drag the selected fields from the input panel.
 * @displayOptions.show { include: ["selected"], /includeOtherFields: [true] }
 */
    includeFields?: string | Expression<string> | PlaceholderValue;
/**
 * Comma-separated list of the field names you want to exclude from the output. You can drag the selected fields from the input panel.
 * @displayOptions.show { include: ["except"], /includeOtherFields: [true] }
 */
    excludeFields?: string | Expression<string> | PlaceholderValue;
  options?: {
    /** Whether binary data should be included if present in the input item
     * @default true
     */
    includeBinary?: boolean | Expression<boolean>;
    /** Whether binary data should be stripped from the input item. Only applies when "Include Other Input Fields" is enabled.
     * @displayOptions.show { /includeOtherFields: [true] }
     * @default true
     */
    stripBinary?: boolean | Expression<boolean>;
    /** Whether to ignore field type errors and apply a less strict type conversion
     * @displayOptions.show { /mode: ["manual"] }
     * @default false
     */
    ignoreConversionErrors?: boolean | Expression<boolean>;
    /** By default, dot-notation is used in property names. This means that "a.b" will set the property "b" underneath "a" so { "a": { "b": value} }. If that is not intended this can be deactivated, it will then set { "a.b": value } instead.
     * @default true
     */
    dotNotation?: boolean | Expression<boolean>;
  };
};

export type SetV33RawNode = {
  type: 'n8n-nodes-base.set';
  version: 3.3;
  config: NodeConfig<SetV33RawParams>;
};
