// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/set/v33/mode_raw.schema.js
/**
 * Edit Fields (Set) Node - Version 3.3 - Zod Schema
 * Discriminator: mode=raw
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      mode: z.literal('raw'),
      duplicateItem: booleanOrExpression.optional(),
      duplicateCount: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"duplicateItem":[true]}}, defaults: {"duplicateItem":false} }),
      jsonOutput: z.union([iDataObjectSchema, z.string()]).optional(),
      includeOtherFields: booleanOrExpression.optional(),
      include: resolveSchema({ parameters, schema: z.union([z.literal('all'), z.literal('selected'), z.literal('except'), expressionSchema]), required: false, displayOptions: {"hide":{"/includeOtherFields":[false]}} }),
      includeFields: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"include":["selected"],"/includeOtherFields":[true]}}, defaults: {"include":"all"} }),
      excludeFields: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"include":["except"],"/includeOtherFields":[true]}}, defaults: {"include":"all"} }),
      options: z.object({ includeBinary: booleanOrExpression.optional(), stripBinary: booleanOrExpression.optional(), ignoreConversionErrors: booleanOrExpression.optional(), dotNotation: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
