// @n8n/ai-workflow-builder/dist/code-builder/handlers/chat-setup-handler.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChatSetupHandler = void 0;
exports.extractNodeNamesFromPlan = extractNodeNamesFromPlan;
const workflow_sdk_1 = require("@n8n/workflow-sdk");
const constants_1 = require("../constants");
const prompts_1 = require("../prompts");
const text_editor_handler_1 = require("./text-editor-handler");
const text_editor_tool_handler_1 = require("./text-editor-tool-handler");
const code_builder_search_tool_1 = require("../tools/code-builder-search.tool");
const extract_code_1 = require("../utils/extract-code");
class ChatSetupHandler {
    llm;
    tools;
    enableTextEditorConfig;
    parseAndValidate;
    getErrorContext;
    nodeTypeParser;
    logger;
    constructor(config) {
        this.llm = config.llm;
        this.tools = config.tools;
        this.enableTextEditorConfig = config.enableTextEditorConfig;
        this.parseAndValidate = config.parseAndValidate;
        this.getErrorContext = config.getErrorContext;
        this.nodeTypeParser = config.nodeTypeParser;
        this.logger = config.logger;
    }
    async execute(params) {
        const { payload, historyContext } = params;
        const currentWorkflow = payload.workflowContext?.currentWorkflow;
        const workflowForCodeContext = (currentWorkflow?.nodes?.length ?? 0) > 0 ? currentWorkflow : undefined;
        const preGeneratedWorkflowCode = this.preGenerateWorkflowCode(payload, workflowForCodeContext);
        const preSearchResults = payload.planOutput
            ? this.prefetchSearchResults(payload.planOutput)
            : undefined;
        const textEditorEnabled = this.shouldEnableTextEditor();
        const prompt = (0, prompts_1.buildCodeBuilderPrompt)(workflowForCodeContext, historyContext, {
            enableTextEditor: textEditorEnabled,
            executionSchema: payload.workflowContext?.executionSchema,
            executionData: payload.workflowContext?.executionData,
            expressionValues: payload.workflowContext?.expressionValues,
            preGeneratedCode: preGeneratedWorkflowCode,
            valuesExcluded: payload.workflowContext?.valuesExcluded,
            pinnedNodes: payload.workflowContext?.pinnedNodes,
            planOutput: payload.planOutput,
            preSearchResults,
        });
        const llmWithTools = this.bindToolsToLlm(textEditorEnabled, !!payload.planOutput);
        const messages = await this.formatInitialMessages(prompt, payload.message);
        const { textEditorHandler, textEditorToolHandler } = this.initializeTextEditorHandlers(textEditorEnabled, preGeneratedWorkflowCode);
        return {
            llmWithTools,
            messages,
            textEditorEnabled,
            preGeneratedWorkflowCode,
            currentWorkflow,
            textEditorHandler,
            textEditorToolHandler,
        };
    }
    shouldEnableTextEditor() {
        if (this.enableTextEditorConfig !== undefined) {
            return this.enableTextEditorConfig;
        }
        const modelName = this.llm.modelId ?? '';
        return (modelName.includes('claude-4') ||
            modelName.includes('opus-4') ||
            modelName.includes('sonnet-4'));
    }
    preGenerateWorkflowCode(payload, currentWorkflow) {
        if (!currentWorkflow) {
            return undefined;
        }
        return (0, workflow_sdk_1.generateWorkflowCode)({
            workflow: currentWorkflow,
            executionSchema: payload.workflowContext?.executionSchema,
            executionData: payload.workflowContext?.executionData,
            expressionValues: payload.workflowContext?.expressionValues,
            valuesExcluded: payload.workflowContext?.valuesExcluded,
            pinnedNodes: payload.workflowContext?.pinnedNodes,
        });
    }
    bindToolsToLlm(textEditorEnabled, hasPlanOutput) {
        if (!this.llm.bindTools) {
            throw new Error('LLM does not support bindTools - cannot use tools for node discovery');
        }
        let tools = hasPlanOutput ? this.tools.filter((t) => t.name !== 'get_suggested_nodes') : this.tools;
        if (textEditorEnabled) {
            tools = [...tools, constants_1.TEXT_EDITOR_TOOL, constants_1.VALIDATE_TOOL, constants_1.BATCH_STR_REPLACE_TOOL];
        }
        return this.llm.bindTools(tools);
    }
    async formatInitialMessages(prompt, userMessage) {
        const formattedMessages = await prompt.formatMessages({ userMessage });
        return [...formattedMessages];
    }
    initializeTextEditorHandlers(textEditorEnabled, preGeneratedWorkflowCode) {
        if (!textEditorEnabled) {
            return {};
        }
        const textEditorHandler = new text_editor_handler_1.TextEditorHandler();
        const textEditorToolHandler = new text_editor_tool_handler_1.TextEditorToolHandler({
            textEditorExecute: (args) => textEditorHandler.execute(args),
            textEditorGetCode: () => textEditorHandler.getWorkflowCode(),
            parseAndValidate: this.parseAndValidate,
            getErrorContext: this.getErrorContext,
        });
        if (preGeneratedWorkflowCode) {
            const codeWithImport = `${extract_code_1.SDK_IMPORT_STATEMENT}\n\n${preGeneratedWorkflowCode}`;
            textEditorHandler.setWorkflowCode(codeWithImport);
        }
        return { textEditorHandler, textEditorToolHandler };
    }
    prefetchSearchResults(plan) {
        const nodeNames = extractNodeNamesFromPlan(plan);
        if (nodeNames.length === 0)
            return undefined;
        if (!this.nodeTypeParser) {
            this.logger?.warn('nodeTypeParser not available, skipping pre-fetch of plan suggestedNodes');
            return undefined;
        }
        const formattedResults = [];
        for (const nodeName of nodeNames) {
            const result = (0, code_builder_search_tool_1.formatNodeResult)(this.nodeTypeParser, nodeName);
            if (result) {
                formattedResults.push(result);
            }
        }
        if (formattedResults.length === 0)
            return undefined;
        return `Found ${formattedResults.length} nodes:\n\n${formattedResults.join('\n\n')}`;
    }
}
exports.ChatSetupHandler = ChatSetupHandler;
function extractNodeNamesFromPlan(plan) {
    const nodeSet = new Set();
    for (const step of plan.steps) {
        if (step.suggestedNodes) {
            for (const nodeName of step.suggestedNodes) {
                nodeSet.add(nodeName);
            }
        }
    }
    return [...nodeSet];
}
