// @n8n/ai-workflow-builder/dist/code-builder/handlers/tool-dispatch-handler.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ToolDispatchHandler = void 0;
exports.parseReplacements = parseReplacements;
const messages_1 = require("@langchain/core/messages");
function parseReplacements(raw) {
    let parsed = raw;
    if (typeof parsed === 'string') {
        try {
            parsed = JSON.parse(parsed);
        }
        catch {
            throw new Error('replacements must be a JSON array of {old_str, new_str} objects, but received an invalid JSON string.');
        }
    }
    if (!Array.isArray(parsed)) {
        throw new Error('replacements must be an array of {old_str, new_str} objects. Example: {"replacements": [{"old_str": "foo", "new_str": "bar"}]}');
    }
    for (let i = 0; i < parsed.length; i++) {
        const item = parsed[i];
        if (typeof item?.old_str !== 'string') {
            throw new Error(`replacements[${i}] is missing a valid "old_str" string. Each replacement must have {old_str: string, new_str: string}.`);
        }
        if (typeof item?.new_str !== 'string') {
            throw new Error(`replacements[${i}] is missing a valid "new_str" string. Each replacement must have {old_str: string, new_str: string}.`);
        }
    }
    return parsed;
}
class ToolDispatchHandler {
    toolsMap;
    toolDisplayTitles;
    validateToolHandler;
    constructor(config) {
        this.toolsMap = config.toolsMap;
        this.toolDisplayTitles = config.toolDisplayTitles;
        this.validateToolHandler = config.validateToolHandler;
    }
    async *dispatch(params) {
        const { toolCalls, messages, currentWorkflow, iteration, textEditorHandler, textEditorToolHandler, warningTracker, } = params;
        let workflow;
        let workflowReady = false;
        let sourceCode;
        let parseDuration;
        let validatePassedThisIteration = false;
        let hasUnvalidatedEdits;
        for (const toolCall of toolCalls) {
            if (!toolCall.id) {
                continue;
            }
            const result = yield* this.executeToolCall({
                toolCall,
                messages,
                currentWorkflow,
                iteration,
                textEditorHandler,
                textEditorToolHandler,
                warningTracker,
            });
            if (toolCall.name === 'str_replace_based_edit_tool') {
                const command = toolCall.args.command;
                if (command === 'create') {
                    hasUnvalidatedEdits = false;
                }
                else if (command !== 'view') {
                    hasUnvalidatedEdits = true;
                }
            }
            if (toolCall.name === 'batch_str_replace') {
                hasUnvalidatedEdits = true;
            }
            if (toolCall.name === 'validate_workflow') {
                hasUnvalidatedEdits = false;
            }
            if (result.workflow) {
                workflow = result.workflow;
            }
            if (result.parseDuration !== undefined) {
                parseDuration = result.parseDuration;
            }
            if (result.workflowReady) {
                workflowReady = true;
                break;
            }
            if (result.validatePassed) {
                validatePassedThisIteration = true;
            }
        }
        return {
            workflow,
            workflowReady,
            sourceCode,
            parseDuration,
            validatePassedThisIteration,
            hasUnvalidatedEdits,
        };
    }
    async *executeToolCall(params) {
        const { toolCall, messages, currentWorkflow, iteration, textEditorHandler, textEditorToolHandler, warningTracker, } = params;
        if (toolCall.name === 'str_replace_based_edit_tool' && textEditorToolHandler) {
            const result = yield* textEditorToolHandler.execute({
                toolCallId: toolCall.id,
                args: toolCall.args,
                currentWorkflow,
                iteration,
                messages,
                warningTracker,
            });
            if (result) {
                return {
                    workflow: result.workflow,
                    workflowReady: result.workflowReady,
                };
            }
            return {};
        }
        if (toolCall.name === 'batch_str_replace' && textEditorHandler) {
            yield* this.executeBatchStrReplace({
                toolCall,
                textEditorHandler,
                messages,
                textEditorToolHandler,
                currentWorkflow,
            });
            return {};
        }
        if (toolCall.name === 'validate_workflow' && textEditorToolHandler && textEditorHandler) {
            const result = yield* this.validateToolHandler.execute({
                toolCallId: toolCall.id,
                code: textEditorHandler.getWorkflowCode(),
                currentWorkflow,
                iteration,
                messages,
                warningTracker,
            });
            return {
                workflow: result.workflow,
                workflowReady: result.workflowReady,
                parseDuration: result.parseDuration,
                validatePassed: result.workflowReady,
            };
        }
        yield* this.executeGeneralToolCall(toolCall, messages);
        return {};
    }
    async *executeGeneralToolCall(toolCall, messages) {
        const displayTitle = this.toolDisplayTitles?.get(toolCall.name);
        yield {
            messages: [
                {
                    type: 'tool',
                    toolName: toolCall.name,
                    toolCallId: toolCall.id,
                    displayTitle,
                    status: 'running',
                    args: toolCall.args,
                },
            ],
        };
        const tool = this.toolsMap.get(toolCall.name);
        if (!tool) {
            const errorMessage = `Tool '${toolCall.name}' not found`;
            messages.push(new messages_1.ToolMessage({
                tool_call_id: toolCall.id,
                content: errorMessage,
            }));
            yield {
                messages: [
                    {
                        type: 'tool',
                        toolName: toolCall.name,
                        toolCallId: toolCall.id,
                        displayTitle,
                        status: 'error',
                        error: errorMessage,
                    },
                ],
            };
            return;
        }
        try {
            const result = await tool.invoke(toolCall.args);
            const resultStr = typeof result === 'string' ? result : JSON.stringify(result);
            messages.push(new messages_1.ToolMessage({
                tool_call_id: toolCall.id,
                content: resultStr,
            }));
            yield {
                messages: [
                    {
                        type: 'tool',
                        toolName: toolCall.name,
                        toolCallId: toolCall.id,
                        displayTitle,
                        status: 'completed',
                    },
                ],
            };
        }
        catch (error) {
            const errorMessage = error instanceof Error ? error.message : String(error);
            messages.push(new messages_1.ToolMessage({
                tool_call_id: toolCall.id,
                content: `Error: ${errorMessage}`,
            }));
            yield {
                messages: [
                    {
                        type: 'tool',
                        toolName: toolCall.name,
                        toolCallId: toolCall.id,
                        displayTitle,
                        status: 'error',
                        error: errorMessage,
                    },
                ],
            };
        }
    }
    async *executeBatchStrReplace(params) {
        const { toolCall, textEditorHandler, messages, textEditorToolHandler, currentWorkflow } = params;
        const displayTitle = 'Editing workflow';
        yield {
            messages: [
                {
                    type: 'tool',
                    toolName: toolCall.name,
                    toolCallId: toolCall.id,
                    displayTitle,
                    status: 'running',
                    args: toolCall.args,
                },
            ],
        };
        try {
            const replacements = parseReplacements(toolCall.args.replacements);
            const result = textEditorHandler.executeBatch(replacements);
            const content = typeof result === 'string' ? result : JSON.stringify(result);
            messages.push(new messages_1.ToolMessage({
                tool_call_id: toolCall.id,
                content,
            }));
            if (typeof result === 'string' && textEditorToolHandler) {
                const preview = await textEditorToolHandler.tryParseForPreview(currentWorkflow);
                if (preview.chunk) {
                    yield preview.chunk;
                }
                if (preview.parseError) {
                    const lastMsg = messages[messages.length - 1];
                    lastMsg.content = `${lastMsg.content}\n\nParse error: ${preview.parseError}`;
                }
            }
            yield {
                messages: [
                    {
                        type: 'tool',
                        toolName: toolCall.name,
                        toolCallId: toolCall.id,
                        displayTitle,
                        status: 'completed',
                    },
                ],
            };
        }
        catch (error) {
            const errorMessage = error instanceof Error ? error.message : String(error);
            messages.push(new messages_1.ToolMessage({
                tool_call_id: toolCall.id,
                content: `Error: ${errorMessage}`,
            }));
            yield {
                messages: [
                    {
                        type: 'tool',
                        toolName: toolCall.name,
                        toolCallId: toolCall.id,
                        displayTitle,
                        status: 'error',
                        error: errorMessage,
                    },
                ],
            };
        }
    }
}
exports.ToolDispatchHandler = ToolDispatchHandler;
