// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/eventbriteTrigger/v1.schema.js
/**
 * Eventbrite Trigger Node - Version 1 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    authentication: z.union([z.literal('privateKey'), z.literal('oAuth2'), expressionSchema]).optional(),
    organization: stringOrExpression.optional(),
    event: stringOrExpression.optional(),
    actions: z.array(z.union([z.literal('attendee.checked_in'), z.literal('attendee.checked_out'), z.literal('attendee.updated'), z.literal('event.created'), z.literal('event.published'), z.literal('event.unpublished'), z.literal('event.updated'), z.literal('order.placed'), z.literal('order.refunded'), z.literal('order.updated'), z.literal('organizer.updated'), z.literal('ticket_class.created'), z.literal('ticket_class.deleted'), z.literal('ticket_class.updated'), z.literal('venue.updated')])).optional(),
    resolveData: booleanOrExpression.optional(),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
