// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/xero/v1/resource_invoice/operation_get_all.schema.js
/**
 * Xero Node - Version 1 - Zod Schema
 * Discriminator: resource=invoice, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('invoice').default('invoice'),
      operation: z.literal('getAll'),
      organizationId: stringOrExpression.optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      options: z.object({ createdByMyApp: booleanOrExpression.optional(), orderBy: stringOrExpression.optional(), sortOrder: z.union([z.literal('ASC'), z.literal('DESC'), expressionSchema]).optional(), statuses: z.array(z.union([z.literal('DRAFT'), z.literal('SUBMITTED'), z.literal('AUTHORISED')])).optional(), where: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
