// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/xero/v1/resource_invoice/operation_create.schema.js
/**
 * Xero Node - Version 1 - Zod Schema
 * Discriminator: resource=invoice, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('invoice').default('invoice'),
      operation: z.literal('create').default('create'),
      organizationId: stringOrExpression.optional(),
      type: z.union([z.literal('ACCPAY'), z.literal('ACCREC'), expressionSchema]).optional(),
      contactId: stringOrExpression.optional(),
      lineItemsUi: z.object({ lineItemsValues: z.array(z.object({ description: stringOrExpression.optional(), quantity: numberOrExpression.optional(), unitAmount: stringOrExpression.optional(), itemCode: stringOrExpression.optional(), accountCode: stringOrExpression.optional(), taxType: z.union([z.literal('INPUT'), z.literal('NONE'), z.literal('OUTPUT'), z.literal('GSTONIMPORTS'), expressionSchema]).optional(), taxAmount: stringOrExpression.optional(), lineAmount: stringOrExpression.optional(), discountRate: stringOrExpression.optional() })).optional() }).optional(),
      additionalFields: z.object({ brandingThemeId: stringOrExpression.optional(), currency: stringOrExpression.optional(), currencyRate: stringOrExpression.optional(), date: stringOrExpression.optional(), dueDate: stringOrExpression.optional(), expectedPaymentDate: stringOrExpression.optional(), invoiceNumber: stringOrExpression.optional(), lineAmountType: z.union([z.literal('Exclusive'), z.literal('Inclusive'), z.literal('NoTax'), expressionSchema]).optional(), plannedPaymentDate: stringOrExpression.optional(), reference: stringOrExpression.optional(), sendToContact: booleanOrExpression.optional(), status: z.union([z.literal('DRAFT'), z.literal('SUBMITTED'), z.literal('AUTHORISED'), expressionSchema]).optional(), url: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
