// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/misp/v1/resource_event/operation_update.ts
/**
 * MISP Node - Version 1
 * Discriminator: resource=event, operation=update
 */


interface Credentials {
  mispApi: CredentialReference;
}

export type MispV1EventUpdateParams = {
  resource: 'event';
  operation: 'update';
/**
 * UUID or numeric ID of the event
 */
    eventId?: string | Expression<string> | PlaceholderValue;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** Analysis maturity level of the event
     * @default 0
     */
    analysis?: 0 | 1 | 2 | Expression<number>;
    /** Who will be able to see this event once published
     * @default 0
     */
    distribution?: 3 | 2 | 5 | 4 | 1 | 0 | Expression<number>;
    /** Information on the event - max 65535 characters
     */
    information?: string | Expression<string> | PlaceholderValue;
    /** Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;. Use only for when &lt;code&gt;Sharing Group&lt;/code&gt; is selected in &lt;code&gt;Distribution&lt;/code&gt;.
     */
    sharing_group_id?: string | Expression<string>;
    /** Threat Level ID
     * @default 1
     */
    threat_level_id?: 1 | 2 | 3 | 4 | Expression<number>;
  };
};

export type MispV1EventUpdateNode = {
  type: 'n8n-nodes-base.misp';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MispV1EventUpdateParams>;
};
