// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/misp/v1/resource_event/operation_create.ts
/**
 * MISP Node - Version 1
 * Discriminator: resource=event, operation=create
 */


interface Credentials {
  mispApi: CredentialReference;
}

export type MispV1EventCreateParams = {
  resource: 'event';
  operation: 'create';
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    org_id?: string | Expression<string>;
/**
 * Information on the event - max 65535 characters
 */
    information?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Analysis maturity level of the event
     * @default 0
     */
    analysis?: 0 | 1 | 2 | Expression<number>;
    /** Who will be able to see this event once published
     * @default 0
     */
    distribution?: 3 | 2 | 5 | 4 | 1 | 0 | Expression<number>;
    /** Use only for when &lt;code&gt;Sharing Group&lt;/code&gt; is selected in &lt;code&gt;Distribution&lt;/code&gt;. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    sharing_group_id?: string | Expression<string>;
    /** Threat Level ID
     * @default 1
     */
    threat_level_id?: 1 | 2 | 3 | 4 | Expression<number>;
  };
};

export type MispV1EventCreateNode = {
  type: 'n8n-nodes-base.misp';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MispV1EventCreateParams>;
};
