// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/misp/v1/resource_event/operation_create.schema.js
/**
 * MISP Node - Version 1 - Zod Schema
 * Discriminator: resource=event, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('event'),
      operation: z.literal('create').default('create'),
      org_id: stringOrExpression.optional(),
      information: stringOrExpression.optional(),
      additionalFields: z.object({ analysis: z.union([z.literal(0), z.literal(1), z.literal(2), expressionSchema]).optional(), distribution: z.union([z.literal(3), z.literal(2), z.literal(5), z.literal(4), z.literal(1), z.literal(0), expressionSchema]).optional(), sharing_group_id: stringOrExpression.optional(), threat_level_id: z.union([z.literal(1), z.literal(2), z.literal(3), z.literal(4), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
