// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleSheets/v43/resource_spreadsheet/operation_create.schema.js
/**
 * Google Sheets Node - Version 4.3 - Zod Schema
 * Discriminator: resource=spreadsheet, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('spreadsheet'),
      operation: z.literal('create'),
      authentication: z.union([z.literal('serviceAccount'), z.literal('oAuth2'), expressionSchema]).optional(),
      title: stringOrExpression.optional(),
      sheetsUi: z.object({ sheetValues: z.array(z.object({ title: stringOrExpression.optional(), hidden: booleanOrExpression.optional() })).optional() }).optional(),
      options: z.object({ locale: stringOrExpression.optional(), autoRecalc: z.union([z.literal(''), z.literal('ON_CHANGE'), z.literal('MINUTE'), z.literal('HOUR'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
