// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/theHiveProject/v1/resource_observable/operation_create.schema.js
/**
 * TheHive 5 Node - Version 1 - Zod Schema
 * Discriminator: resource=observable, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('observable'),
      operation: z.literal('create').default('create'),
      createIn: z.union([z.literal('case'), z.literal('alert'), expressionSchema]).optional(),
      id: resolveSchema({ parameters, schema: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('url'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]), required: false, displayOptions: {"show":{"createIn":["case","alert"]}}, defaults: {"createIn":"case"} }),
      dataType: stringOrExpression.optional(),
      data: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"hide":{"dataType":["file"]}}, defaults: {"dataType":"file"} }),
      attachmentsUi: resolveSchema({ parameters, schema: z.object({ values: z.array(z.object({ field: stringOrExpression.optional() })).optional() }), required: false, displayOptions: {"show":{"dataType":["file"]}}, defaults: {"dataType":"file"} }),
      observableFields: resourceMapperValueSchema.optional(),
    }).optional(),
  });
};
