// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/theHiveProject/v1/resource_observable/operation_execute_analyzer.ts
/**
 * TheHive 5 Node - Version 1
 * Discriminator: resource=observable, operation=executeAnalyzer
 */


interface Credentials {
  theHiveProjectApi: CredentialReference;
}

export type TheHiveProjectV1ObservableExecuteAnalyzerParams = {
  resource: 'observable';
  operation: 'executeAnalyzer';
/**
 * Observable
 * @default {"mode":"list","value":""}
 */
    observableId?: { __rl: true; mode: 'list' | 'id'; value: string; cachedResultName?: string };
/**
 * Type of the observable. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    dataType?: string | Expression<string>;
/**
 * Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 * @displayOptions.hide { id: [""] }
 * @default []
 */
    analyzers?: string[];
};

export type TheHiveProjectV1ObservableExecuteAnalyzerNode = {
  type: 'n8n-nodes-base.theHiveProject';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<TheHiveProjectV1ObservableExecuteAnalyzerParams>;
};
