// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/currents/v1/resource_test/operation_get_all.ts
/**
 * Currents Node - Version 1
 * Discriminator: resource=test, operation=getAll
 */


interface Credentials {
  currentsApi: CredentialReference;
}

/** Individual test performance metrics */
export type CurrentsV1TestGetAllParams = {
  resource: 'test';
  operation: 'getAll';
/**
 * The Currents project
 * @default {"mode":"list","value":""}
 */
    projectId?: { __rl: true; mode: 'list' | 'id'; value: string; cachedResultName?: string };
/**
 * Start date for metrics (ISO 8601 format)
 */
    dateStart?: string | Expression<string>;
/**
 * End date for metrics (ISO 8601 format)
 */
    dateEnd?: string | Expression<string>;
/**
 * Max number of results to return
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Filter by git authors (comma-separated for multiple)
     */
    authors?: string | Expression<string> | PlaceholderValue;
    /** Filter by branches (comma-separated for multiple)
     */
    branches?: string | Expression<string> | PlaceholderValue;
    /** Filter by groups (comma-separated for multiple)
     */
    groups?: string | Expression<string> | PlaceholderValue;
    /** Minimum number of executions to include a test
     * @default 1
     */
    minExecutions?: number | Expression<number>;
    /** Filter tests by spec file name (partial match)
     */
    spec?: string | Expression<string> | PlaceholderValue;
    /** Filter by tags (comma-separated for multiple)
     */
    tags?: string | Expression<string> | PlaceholderValue;
    /** Filter by test state
     * @default []
     */
    testState?: Array<'failed' | 'passed' | 'pending' | 'skipped'>;
    /** Filter tests by title (partial match)
     */
    title?: string | Expression<string> | PlaceholderValue;
  };
/**
 * Options
 * @default {}
 */
    options?: {
    /** The field to order results by
     * @default title
     */
    order?: 'duration' | 'durationDelta' | 'durationXSamples' | 'executions' | 'failRateXSamples' | 'failureRateDelta' | 'failures' | 'flakiness' | 'flakinessXSamples' | 'flakinessRateDelta' | 'passes' | 'title' | Expression<string>;
    /** The direction to sort results
     * @default desc
     */
    dir?: 'asc' | 'desc' | Expression<string>;
    /** Page number (0-indexed)
     * @default 0
     */
    page?: number | Expression<number>;
    /** Override which test statuses are included in metric calculations. JSON object with optional keys: executions, avgDuration, flakinessRate, failureRate. Each value is an array of status strings: passed, failed, pending, skipped. Example: {"executions":["failed","passed"],"failureRate":["failed"]}
     */
    metric_settings?: string | Expression<string> | PlaceholderValue;
  };
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type CurrentsV1TestGetAllNode = {
  type: 'n8n-nodes-base.currents';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<CurrentsV1TestGetAllParams>;
};
