// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/netlifyTrigger/v1.schema.js
/**
 * Netlify Trigger Node - Version 1 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    siteId: stringOrExpression.optional(),
    event: z.union([z.literal('deployBuilding'), z.literal('deployFailed'), z.literal('deployCreated'), z.literal('submissionCreated'), expressionSchema]).optional(),
    formId: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"event":["submissionCreated"]}}, defaults: {"event":""} }),
    simple: resolveSchema({ parameters, schema: booleanOrExpression, required: false, displayOptions: {"show":{"event":["submissionCreated"]}}, defaults: {"event":""} }),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
