// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/slack/v24/resource_channel/operation_get_all.schema.js
/**
 * Slack Node - Version 2.4 - Zod Schema
 * Discriminator: resource=channel, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('channel'),
      operation: z.literal('getAll'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filters: z.object({ excludeArchived: booleanOrExpression.optional(), types: z.array(z.union([z.literal('public_channel'), z.literal('private_channel'), z.literal('mpim'), z.literal('im')])).optional() }).optional(),
    }).optional(),
  });
};
