// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/slack/v24/resource_channel/operation_get_all.ts
/**
 * Slack Node - Version 2.4
 * Discriminator: resource=channel, operation=getAll
 */


interface Credentials {
  slackApi: CredentialReference;
  slackOAuth2Api: CredentialReference;
}

/** Get many channels in a Slack team */
export type SlackV24ChannelGetAllParams = {
  resource: 'channel';
  operation: 'getAll';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Whether to exclude archived channels from the list
     * @default false
     */
    excludeArchived?: boolean | Expression<boolean>;
    /** Mix and match channel types
     * @default ["public_channel"]
     */
    types?: Array<'public_channel' | 'private_channel' | 'mpim' | 'im'>;
  };
};

export type SlackV24ChannelGetAllOutput = {
  context_team_id?: string;
  created?: number;
  creator?: string;
  id?: string;
  is_archived?: boolean;
  is_channel?: boolean;
  is_ext_shared?: boolean;
  is_general?: boolean;
  is_group?: boolean;
  is_im?: boolean;
  is_member?: boolean;
  is_mpim?: boolean;
  is_org_shared?: boolean;
  is_pending_ext_shared?: boolean;
  is_private?: boolean;
  is_shared?: boolean;
  name?: string;
  name_normalized?: string;
  num_members?: number;
  parent_conversation?: null;
  previous_names?: Array<string>;
  properties?: {
    canvas?: {
      file_id?: string;
      is_empty?: boolean;
      quip_thread_id?: string;
    };
    tabs?: Array<{
      id?: string;
      label?: string;
      type?: string;
    }>;
    tabz?: Array<{
      type?: string;
    }>;
    use_case?: string;
  };
  purpose?: {
    creator?: string;
    last_set?: number;
    value?: string;
  };
  shared_team_ids?: Array<string>;
  topic?: {
    creator?: string;
    last_set?: number;
    value?: string;
  };
  unlinked?: number;
  updated?: number;
};

export type SlackV24ChannelGetAllNode = {
  type: 'n8n-nodes-base.slack';
  version: 2.4;
  credentials?: Credentials;
  config: NodeConfig<SlackV24ChannelGetAllParams>;
  output?: Items<SlackV24ChannelGetAllOutput>;
};
