// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/harvest/v1/resource_user/operation_update.schema.js
/**
 * Harvest Node - Version 1 - Zod Schema
 * Discriminator: resource=user, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('user'),
      operation: z.literal('update'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      accountId: stringOrExpression.optional(),
      id: stringOrExpression.optional(),
      updateFields: z.object({ can_create_invoices: booleanOrExpression.optional(), can_create_projects: booleanOrExpression.optional(), can_see_rates: booleanOrExpression.optional(), cost_rate: numberOrExpression.optional(), default_hourly_rate: stringOrExpression.optional(), email: stringOrExpression.optional(), first_name: stringOrExpression.optional(), has_access_to_all_future_projects: booleanOrExpression.optional(), is_active: booleanOrExpression.optional(), is_admin: booleanOrExpression.optional(), is_contractor: booleanOrExpression.optional(), is_project_manager: booleanOrExpression.optional(), last_name: stringOrExpression.optional(), roles: stringOrExpression.optional(), timezone: stringOrExpression.optional(), weekly_capacity: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
