// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/harvest/v1/resource_user/operation_get_all.ts
/**
 * Harvest Node - Version 1
 * Discriminator: resource=user, operation=getAll
 */


interface Credentials {
  harvestApi: CredentialReference;
  harvestOAuth2Api: CredentialReference;
}

/** Get data of many clients */
export type HarvestV1UserGetAllParams = {
  resource: 'user';
  operation: 'getAll';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    accountId?: string | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Whether to only return active users and false to return inactive users
     * @default true
     */
    is_active?: boolean | Expression<boolean>;
    /** Only return users belonging to the user with the given ID
     */
    updated_since?: string | Expression<string>;
    /** The page number to use in pagination
     * @default 1
     */
    page?: number | Expression<number>;
  };
};

export type HarvestV1UserGetAllOutput = {
  access_roles?: Array<string>;
  avatar_url?: string;
  calendar_integration_enabled?: boolean;
  can_create_projects?: boolean;
  created_at?: string;
  email?: string;
  first_name?: string;
  has_access_to_all_future_projects?: boolean;
  id?: number;
  is_active?: boolean;
  is_contractor?: boolean;
  last_name?: string;
  permissions_claims?: Array<string>;
  roles?: Array<string>;
  telephone?: string;
  timezone?: string;
  updated_at?: string;
  weekly_capacity?: number;
};

export type HarvestV1UserGetAllNode = {
  type: 'n8n-nodes-base.harvest';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<HarvestV1UserGetAllParams>;
  output?: Items<HarvestV1UserGetAllOutput>;
};
