// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/contentful/v1/resource_content_type/operation_get.schema.js
/**
 * Contentful Node - Version 1 - Zod Schema
 * Discriminator: resource=contentType, operation=get
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('contentType'),
      operation: z.literal('get').default('get'),
      source: z.union([z.literal('deliveryApi'), z.literal('previewApi'), expressionSchema]).optional(),
      environmentId: stringOrExpression.optional(),
      contentTypeId: stringOrExpression.optional(),
      additionalFields: z.object({ rawData: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
