// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/contentful/v1/resource_content_type/operation_get.ts
/**
 * Contentful Node - Version 1
 * Discriminator: resource=contentType, operation=get
 */


interface Credentials {
  contentfulApi: CredentialReference;
}

export type ContentfulV1ContentTypeGetParams = {
  resource: 'contentType';
  operation: 'get';
/**
 * Pick where your data comes from, delivery or preview API
 * @default deliveryApi
 */
    source?: 'deliveryApi' | 'previewApi' | Expression<string>;
/**
 * The ID for the Contentful environment (e.g. master, staging, etc.). Depending on your plan, you might not have environments. In that case use "master".
 * @default master
 */
    environmentId?: string | Expression<string> | PlaceholderValue;
/**
 * Content Type ID
 */
    contentTypeId?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Whether the data should be returned RAW instead of parsed
     * @default false
     */
    rawData?: boolean | Expression<boolean>;
  };
};

export type ContentfulV1ContentTypeGetOutput = {
  disabled?: boolean;
  id?: string;
  localized?: boolean;
  name?: string;
  omitted?: boolean;
  required?: boolean;
  type?: string;
};

export type ContentfulV1ContentTypeGetNode = {
  type: 'n8n-nodes-base.contentful';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ContentfulV1ContentTypeGetParams>;
  output?: Items<ContentfulV1ContentTypeGetOutput>;
};
