// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/convertKit/v1/resource_tag_subscriber/operation_get_all.schema.js
/**
 * ConvertKit Node - Version 1 - Zod Schema
 * Discriminator: resource=tagSubscriber, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('tagSubscriber'),
      operation: z.literal('getAll'),
      tagId: stringOrExpression.optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      additionalFields: z.object({ subscriberState: z.union([z.literal('active'), z.literal('cancelled'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
