// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/convertKit/v1/resource_tag_subscriber/operation_add.ts
/**
 * ConvertKit Node - Version 1
 * Discriminator: resource=tagSubscriber, operation=add
 */


interface Credentials {
  convertKitApi: CredentialReference;
}

/** Add a tag to a subscriber */
export type ConvertKitV1TagSubscriberAddParams = {
  resource: 'tagSubscriber';
  operation: 'add';
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    tagId?: string | Expression<string>;
/**
 * Subscriber email address
 */
    email?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Object of key/value pairs for custom fields (the custom field must exist before you can use it here)
     * @default {}
     */
    fields?: {
        /** Custom Field
     */
    field?: Array<{
      /** The field's key
       */
      key?: string | Expression<string> | PlaceholderValue;
      /** Value of the field
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
    /** Subscriber first name
     */
    firstName?: string | Expression<string> | PlaceholderValue;
  };
};

export type ConvertKitV1TagSubscriberAddOutput = {
  created_at?: string;
  id?: number;
  referrer?: null;
  source?: null;
  state?: string;
  subscribable_id?: number;
  subscribable_type?: string;
  subscriber?: {
    created_at?: string;
    email_address?: string;
    id?: number;
    state?: string;
  };
};

export type ConvertKitV1TagSubscriberAddNode = {
  type: 'n8n-nodes-base.convertKit';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ConvertKitV1TagSubscriberAddParams>;
  output?: Items<ConvertKitV1TagSubscriberAddOutput>;
};
