// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/convertKit/v1/resource_tag_subscriber/operation_get_all.ts
/**
 * ConvertKit Node - Version 1
 * Discriminator: resource=tagSubscriber, operation=getAll
 */


interface Credentials {
  convertKitApi: CredentialReference;
}

/** Get many fields */
export type ConvertKitV1TagSubscriberGetAllParams = {
  resource: 'tagSubscriber';
  operation: 'getAll';
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    tagId?: string | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Receive only active subscribers or cancelled subscribers
 * @default {}
 */
    additionalFields?: {
    /** Subscriber State
     * @default active
     */
    subscriberState?: 'active' | 'cancelled' | Expression<string>;
  };
};

export type ConvertKitV1TagSubscriberGetAllNode = {
  type: 'n8n-nodes-base.convertKit';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ConvertKitV1TagSubscriberGetAllParams>;
};
