// @microsoft/agents-a365-observability/dist/cjs/tracing/scopes/OpenTelemetryScope.js
"use strict";
// ------------------------------------------------------------------------------
// Copyright (c) Microsoft Corporation. All rights reserved.
// ------------------------------------------------------------------------------
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OpenTelemetryScope = void 0;
const api_1 = require("@opentelemetry/api");
const constants_1 = require("../constants");
const logging_1 = __importDefault(require("../../utils/logging"));
/**
 * Base class for OpenTelemetry tracing scopes
 */
class OpenTelemetryScope {
    /**
     * Initializes a new instance of the OpenTelemetryScope class
     * @param kind The kind of span (CLIENT, SERVER, INTERNAL, etc.)
     * @param operationName The name of the operation being traced
     * @param spanName The name of the span for display purposes
     * @param agentDetails Optional agent details
     * @param tenantDetails Optional tenant details
     */
    constructor(kind, operationName, spanName, agentDetails, tenantDetails) {
        this.hasEnded = false;
        const currentContext = api_1.context.active();
        logging_1.default.info(`[A365Observability] Starting span: ${spanName}, operation: ${operationName} for tenantId: ${tenantDetails?.tenantId || 'unknown'}, agentId: ${agentDetails?.agentId || 'unknown'}`);
        // Start span with current context to establish parent-child relationship
        this.span = OpenTelemetryScope.tracer.startSpan(spanName, {
            kind,
            attributes: {
                [constants_1.OpenTelemetryConstants.GEN_AI_SYSTEM_KEY]: constants_1.OpenTelemetryConstants.GEN_AI_SYSTEM_VALUE,
                [constants_1.OpenTelemetryConstants.GEN_AI_OPERATION_NAME_KEY]: operationName,
            },
        }, currentContext);
        logging_1.default.info(`[A365Observability] Span[${this.span.spanContext().spanId}] ${spanName}, operation: ${operationName} started successfully`);
        this.startTime = Date.now();
        // Set agent details if provided
        if (agentDetails) {
            this.setTagMaybe(constants_1.OpenTelemetryConstants.GEN_AI_AGENT_ID_KEY, agentDetails.agentId);
            this.setTagMaybe(constants_1.OpenTelemetryConstants.GEN_AI_AGENT_NAME_KEY, agentDetails.agentName);
            this.setTagMaybe(constants_1.OpenTelemetryConstants.GEN_AI_AGENT_DESCRIPTION_KEY, agentDetails.agentDescription);
            this.setTagMaybe(constants_1.OpenTelemetryConstants.GEN_AI_AGENT_PLATFORM_ID_KEY, agentDetails.platformId);
            this.setTagMaybe(constants_1.OpenTelemetryConstants.GEN_AI_CONVERSATION_ID_KEY, agentDetails.conversationId);
            this.setTagMaybe(constants_1.OpenTelemetryConstants.GEN_AI_ICON_URI_KEY, agentDetails.iconUri);
            this.setTagMaybe(constants_1.OpenTelemetryConstants.GEN_AI_AGENT_AUID_KEY, agentDetails.agentAUID);
            this.setTagMaybe(constants_1.OpenTelemetryConstants.GEN_AI_AGENT_UPN_KEY, agentDetails.agentUPN);
            this.setTagMaybe(constants_1.OpenTelemetryConstants.GEN_AI_AGENT_BLUEPRINT_ID_KEY, agentDetails.agentBlueprintId);
        }
        // Set tenant details if provided
        if (tenantDetails) {
            this.setTagMaybe(constants_1.OpenTelemetryConstants.TENANT_ID_KEY, tenantDetails.tenantId);
        }
    }
    /**
     * Makes this span active for the duration of the async callback execution
     */
    withActiveSpanAsync(callback) {
        const newContext = api_1.trace.setSpan(api_1.context.active(), this.span);
        return api_1.context.with(newContext, callback);
    }
    /**
     * Records an error that occurred during the operation
     * @param error The error that occurred
     */
    recordError(error) {
        logging_1.default.error(`[A365Observability] Records an error that occurred during the operation span[${this.span.spanContext().spanId}]: ${error.message}`);
        // Check if it's an HTTP error with status code
        if ('status' in error && typeof error.status === 'number') {
            this.errorType = error.status.toString();
        }
        else {
            this.errorType = error.constructor.name;
        }
        this.exception = error;
        this.span.setStatus({
            code: api_1.SpanStatusCode.ERROR,
            message: error.message
        });
        this.span.recordException(error);
    }
    /**
     * Records multiple attribute key/value pairs for telemetry tracking.
     * @param attributes Collection of attribute key/value pairs (array or iterable of [key, value] or object map).
     */
    recordAttributes(attributes) {
        if (!attributes)
            return;
        // Support both array/iterable of pairs and object maps
        if (Array.isArray(attributes)) {
            for (const [key, value] of attributes) {
                if (!key || typeof key !== 'string' || !key.trim())
                    continue;
                this.span.setAttribute(key, value);
            }
        }
        else if (typeof attributes === 'object' &&
            typeof attributes[Symbol.iterator] === 'function' &&
            !Array.isArray(attributes) &&
            typeof attributes !== 'string') {
            for (const [key, value] of attributes) {
                if (!key || typeof key !== 'string' || !key.trim())
                    continue;
                this.span.setAttribute(key, value);
            }
        }
        else if (typeof attributes === 'object') {
            for (const key of Object.keys(attributes)) {
                if (!key || typeof key !== 'string' || !key.trim())
                    continue;
                this.span.setAttribute(key, attributes[key]);
            }
        }
    }
    /**
     * Sets a tag on the span if telemetry is enabled
     * @param name The tag name
     * @param value The tag value
     */
    setTagMaybe(name, value) {
        if (value != null) {
            this.span.setAttributes({ [name]: value });
        }
    }
    /**
     * Add baggage to the current context
     * @param key The baggage key
     * @param value The baggage value
     */
    addBaggage(key, value) {
        // Note: OpenTelemetry JS doesn't have direct baggage API in span
        // This would typically be handled through the baggage API
        this.span.setAttributes({ [`baggage.${key}`]: value });
    }
    /**
     * Finalizes the scope and records metrics
     */
    end() {
        if (this.hasEnded) {
            logging_1.default.info(`[A365Observability] Span already ended for span[${this.span.spanContext().spanId}]`);
            return;
        }
        const duration = (Date.now() - this.startTime) / 1000; // Convert to seconds
        const finalTags = {};
        if (this.errorType) {
            finalTags[constants_1.OpenTelemetryConstants.ERROR_TYPE_KEY] = this.errorType;
            this.span.setAttributes({ [constants_1.OpenTelemetryConstants.ERROR_TYPE_KEY]: this.errorType });
        }
        // Record duration metric (would typically use a meter here)
        // For now, we'll add it as a span attribute
        this.span.setAttributes({ 'operation.duration': duration });
        this.hasEnded = true;
        logging_1.default.info(`[A365Observability] Ending span[${this.span.spanContext().spanId}], duration: ${duration}s`);
    }
    /**
     * Disposes the scope and finalizes telemetry data collection
     */
    [Symbol.dispose]() {
        if (!this.hasEnded) {
            this.end();
            this.span.end();
        }
    }
    /**
     * Legacy dispose method for compatibility
     */
    dispose() {
        this[Symbol.dispose]();
    }
}
exports.OpenTelemetryScope = OpenTelemetryScope;
OpenTelemetryScope.tracer = api_1.trace.getTracer(constants_1.OpenTelemetryConstants.SOURCE_NAME);
