// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/iterable/v1/resource_event/operation_track.schema.js
/**
 * Iterable Node - Version 1 - Zod Schema
 * Discriminator: resource=event, operation=track
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('event'),
      operation: z.literal('track').default('track'),
      name: stringOrExpression.optional(),
      additionalFields: z.object({ campaignId: stringOrExpression.optional(), createdAt: stringOrExpression.optional(), dataFieldsUi: z.unknown().optional(), email: stringOrExpression.optional(), id: stringOrExpression.optional(), templateId: stringOrExpression.optional(), userId: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
