// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/todoist/v21/resource_task/operation_quick_add.ts
/**
 * Todoist Node - Version 2.1
 * Discriminator: resource=task, operation=quickAdd
 */


interface Credentials {
  todoistApi: CredentialReference;
  todoistOAuth2Api: CredentialReference;
}

/** Task resource */
export type TodoistV21TaskQuickAddParams = {
  resource: 'task';
  operation: 'quickAdd';
  authentication?: 'apiKey' | 'oAuth2' | Expression<string>;
/**
 * Natural language text for quick adding task (e.g., "Buy milk @Grocery #shopping tomorrow"). It can include a due date in free form text, a project name starting with the "#" character (without spaces), a label starting with the "@" character, an assignee starting with the "+" character, a priority (e.g., p1), a deadline between "{}" (e.g. {in 3 days}), or a description starting from "//" until the end of the text.
 */
    text?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    options?: {
    /** The content of the note
     */
    note?: string | Expression<string> | PlaceholderValue;
    /** The date of the reminder, added in free form text
     */
    reminder?: string | Expression<string> | PlaceholderValue;
    /** When this option is enabled, the default reminder will be added to the new item if it has a due date with time set
     * @default false
     */
    auto_reminder?: boolean | Expression<boolean>;
  };
};

export type TodoistV21TaskQuickAddNode = {
  type: 'n8n-nodes-base.todoist';
  version: 2.1;
  credentials?: Credentials;
  config: NodeConfig<TodoistV21TaskQuickAddParams>;
};
