// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/todoist/v21/resource_comment/operation_update.schema.js
/**
 * Todoist Node - Version 2.1 - Zod Schema
 * Discriminator: resource=comment, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('comment'),
      operation: z.literal('update'),
      authentication: z.union([z.literal('apiKey'), z.literal('oAuth2'), expressionSchema]).optional(),
      commentId: stringOrExpression.optional(),
      commentUpdateFields: z.object({ content: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
