// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/gong/v1/resource_call/operation_get.ts
/**
 * Gong Node - Version 1
 * Discriminator: resource=call, operation=get
 */


interface Credentials {
  gongApi: CredentialReference;
  gongOAuth2Api: CredentialReference;
}

/** Retrieve data for a specific call */
export type GongV1CallGetParams = {
  resource: 'call';
  operation: 'get';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Call to Get
 * @default {"mode":"list","value":""}
 */
    call?: { __rl: true; mode: 'list' | 'id' | 'url'; value: string; cachedResultName?: string };
/**
 * Options
 * @default {}
 */
    options?: {
    /** The Call properties to include in the returned results. Choose from a list, or specify IDs using an &lt;a href="https://docs.n8n.io/code-examples/expressions/"&gt;expression&lt;/a&gt;.
     * @default []
     */
    properties?: Array<'pointsOfInterest' | 'media' | 'brief' | 'publicComments' | 'highlights' | 'keyPoints' | 'callOutcome' | 'outline' | 'parties' | 'structure' | 'topics' | 'trackers' | 'transcript'>;
  };
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type GongV1CallGetOutput = {
  metaData?: {
    customData?: null;
    direction?: string;
    duration?: number;
    id?: string;
    isPrivate?: boolean;
    language?: string;
    media?: string;
    meetingUrl?: string;
    primaryUserId?: string;
    purpose?: null;
    scheduled?: string;
    scope?: string;
    started?: string;
    system?: string;
    title?: string;
    url?: string;
    workspaceId?: string;
  };
  parties?: Array<{
    affiliation?: string;
    emailAddress?: string;
    id?: string;
    methods?: Array<string>;
    name?: string;
    phoneNumber?: string;
    title?: string;
    userId?: string;
  }>;
  transcript?: Array<{
    sentences?: Array<{
      end?: number;
      start?: number;
      text?: string;
    }>;
    speakerId?: string;
  }>;
};

export type GongV1CallGetNode = {
  type: 'n8n-nodes-base.gong';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GongV1CallGetParams>;
  output?: Items<GongV1CallGetOutput>;
};
