// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/gong/v1/resource_call/operation_get.schema.js
/**
 * Gong Node - Version 1 - Zod Schema
 * Discriminator: resource=call, operation=get
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('call').default('call'),
      operation: z.literal('get'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      call: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('id'), z.literal('url')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      options: z.object({ properties: z.array(z.union([z.literal('pointsOfInterest'), z.literal('media'), z.literal('brief'), z.literal('publicComments'), z.literal('highlights'), z.literal('keyPoints'), z.literal('callOutcome'), z.literal('outline'), z.literal('parties'), z.literal('structure'), z.literal('topics'), z.literal('trackers'), z.literal('transcript')])).optional() }).optional(),
      requestOptions: z.object({ batching: z.unknown().optional(), allowUnauthorizedCerts: z.boolean().optional(), proxy: stringOrExpression.optional(), timeout: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
