// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/lemlist/v2/resource_lead/operation_create.schema.js
/**
 * Lemlist Node - Version 2 - Zod Schema
 * Discriminator: resource=lead, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('lead'),
      operation: z.literal('create'),
      campaignId: stringOrExpression.optional(),
      email: stringOrExpression.optional(),
      additionalFields: z.object({ companyName: stringOrExpression.optional(), companyDomain: stringOrExpression.optional(), deduplicate: booleanOrExpression.optional(), findEmail: booleanOrExpression.optional(), findPhone: booleanOrExpression.optional(), firstName: stringOrExpression.optional(), icebreaker: stringOrExpression.optional(), jobTitle: stringOrExpression.optional(), lastName: stringOrExpression.optional(), linkedinEnrichment: booleanOrExpression.optional(), linkedinUrl: stringOrExpression.optional(), phone: stringOrExpression.optional(), picture: stringOrExpression.optional(), verifyEmail: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
