// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/youTube/v1/resource_playlist_item/operation_get_all.ts
/**
 * YouTube Node - Version 1
 * Discriminator: resource=playlistItem, operation=getAll
 */


interface Credentials {
  youTubeOAuth2Api: CredentialReference;
}

/** Retrieve many channels */
export type YouTubeV1PlaylistItemGetAllParams = {
  resource: 'playlistItem';
  operation: 'getAll';
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    playlistId?: string | Expression<string>;
/**
 * The fields parameter specifies a comma-separated list of one or more playlistItem resource properties that the API response will include
 * @default ["*"]
 */
    part?: Array<'*' | 'contentDetails' | 'id' | 'snippet' | 'status'>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 25
 */
    limit?: number | Expression<number>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** The onBehalfOfContentOwner parameter indicates that the request's authorization credentials identify a YouTube CMS user who is acting on behalf of the content owner specified in the parameter value
     */
    onBehalfOfContentOwner?: string | Expression<string> | PlaceholderValue;
  };
};

export type YouTubeV1PlaylistItemGetAllOutput = {
  contentDetails?: {
    videoId?: string;
    videoPublishedAt?: string;
  };
  etag?: string;
  id?: string;
  kind?: string;
  snippet?: {
    channelId?: string;
    channelTitle?: string;
    description?: string;
    playlistId?: string;
    position?: number;
    publishedAt?: string;
    resourceId?: {
      kind?: string;
      videoId?: string;
    };
    thumbnails?: {
      'default'?: {
        height?: number;
        url?: string;
        width?: number;
      };
      high?: {
        height?: number;
        url?: string;
        width?: number;
      };
      maxres?: {
        height?: number;
        url?: string;
        width?: number;
      };
      medium?: {
        height?: number;
        url?: string;
        width?: number;
      };
      standard?: {
        height?: number;
        url?: string;
        width?: number;
      };
    };
    title?: string;
    videoOwnerChannelId?: string;
    videoOwnerChannelTitle?: string;
  };
  status?: {
    privacyStatus?: string;
  };
};

export type YouTubeV1PlaylistItemGetAllNode = {
  type: 'n8n-nodes-base.youTube';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<YouTubeV1PlaylistItemGetAllParams>;
  output?: Items<YouTubeV1PlaylistItemGetAllOutput>;
};
