// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/youTube/v1/resource_playlist_item/operation_add.schema.js
/**
 * YouTube Node - Version 1 - Zod Schema
 * Discriminator: resource=playlistItem, operation=add
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('playlistItem'),
      operation: z.literal('add'),
      playlistId: stringOrExpression.optional(),
      videoId: stringOrExpression.optional(),
      options: z.object({ endAt: stringOrExpression.optional(), note: stringOrExpression.optional(), onBehalfOfContentOwner: stringOrExpression.optional(), position: numberOrExpression.optional(), startAt: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
