// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/youTube/v1/resource_playlist_item/operation_add.ts
/**
 * YouTube Node - Version 1
 * Discriminator: resource=playlistItem, operation=add
 */


interface Credentials {
  youTubeOAuth2Api: CredentialReference;
}

/** Add an item to a playlist */
export type YouTubeV1PlaylistItemAddParams = {
  resource: 'playlistItem';
  operation: 'add';
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    playlistId?: string | Expression<string>;
/**
 * Video ID
 */
    videoId?: string | Expression<string> | PlaceholderValue;
/**
 * Options
 * @default {}
 */
    options?: {
    /** The time, measured in seconds from the start of the video, when the video should stop playing
     */
    endAt?: string | Expression<string>;
    /** A user-generated note for this item. The property value has a maximum length of 280 characters.
     */
    note?: string | Expression<string> | PlaceholderValue;
    /** The onBehalfOfContentOwner parameter indicates that the request's authorization credentials identify a YouTube CMS user who is acting on behalf of the content owner specified in the parameter value
     */
    onBehalfOfContentOwner?: string | Expression<string> | PlaceholderValue;
    /** The order in which the item appears in the playlist. The value uses a zero-based index, so the first item has a position of 0, the second item has a position of 1, and so forth.
     */
    position?: number | Expression<number>;
    /** The time, measured in seconds from the start of the video, when the video should start playing
     */
    startAt?: string | Expression<string>;
  };
};

export type YouTubeV1PlaylistItemAddOutput = {
  contentDetails?: {
    videoId?: string;
    videoPublishedAt?: string;
  };
  etag?: string;
  id?: string;
  kind?: string;
  snippet?: {
    channelId?: string;
    channelTitle?: string;
    description?: string;
    playlistId?: string;
    position?: number;
    publishedAt?: string;
    resourceId?: {
      kind?: string;
      videoId?: string;
    };
    thumbnails?: {
      'default'?: {
        height?: number;
        url?: string;
        width?: number;
      };
      high?: {
        height?: number;
        url?: string;
        width?: number;
      };
      medium?: {
        height?: number;
        url?: string;
        width?: number;
      };
    };
    title?: string;
    videoOwnerChannelId?: string;
    videoOwnerChannelTitle?: string;
  };
};

export type YouTubeV1PlaylistItemAddNode = {
  type: 'n8n-nodes-base.youTube';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<YouTubeV1PlaylistItemAddParams>;
  output?: Items<YouTubeV1PlaylistItemAddOutput>;
};
