// @n8n/n8n-nodes-langchain/dist/node-definitions/nodes/n8n-nodes-langchain/textClassifier/v1.schema.js
/**
 * Text Classifier Node - Version 1 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, languageModelInstanceSchema }) {

  // Static subnode schema
  const subnodesSchema = z.object({
    model: z.union([languageModelInstanceSchema, z.array(languageModelInstanceSchema)]),
  }).strict();

  // Parameters schema
  const parametersSchema = z.object({
    inputText: stringOrExpression.optional(),
    categories: z.object({ categories: z.array(z.object({ category: stringOrExpression.optional(), description: stringOrExpression.optional() })).optional() }).optional(),
    options: z.object({ multiClass: booleanOrExpression.optional(), fallback: z.union([z.literal('discard'), z.literal('other'), expressionSchema]).optional(), systemPromptTemplate: stringOrExpression.optional(), enableAutoFixing: booleanOrExpression.optional(), batching: z.unknown().optional() }).optional(),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
    subnodes: subnodesSchema,
  });
};
