// @n8n/ai-workflow-builder/dist/tools/engines/node-search-engine.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NodeSearchEngine = exports.SCORE_WEIGHTS = void 0;
const utils_1 = require("@n8n/utils");
const n8n_workflow_1 = require("n8n-workflow");
const NODE_SEARCH_KEYS = [
    { key: 'displayName', weight: 1.5 },
    { key: 'name', weight: 1.3 },
    { key: 'codex.alias', weight: 1.0 },
    { key: 'description', weight: 0.7 },
];
exports.SCORE_WEIGHTS = {
    CONNECTION_EXACT: 100,
    CONNECTION_IN_EXPRESSION: 50,
};
function getLatestVersion(version) {
    return Array.isArray(version) ? Math.max(...version) : version;
}
function dedupeNodes(nodes) {
    const dedupeCache = {};
    nodes.forEach((node) => {
        const cachedNodeType = dedupeCache[node.name];
        if (!cachedNodeType) {
            dedupeCache[node.name] = node;
            return;
        }
        const cachedVersion = getLatestVersion(cachedNodeType.version);
        const nextVersion = getLatestVersion(node.version);
        if (nextVersion > cachedVersion) {
            dedupeCache[node.name] = node;
        }
    });
    return Object.values(dedupeCache);
}
class NodeSearchEngine {
    nodeTypes;
    constructor(nodeTypes) {
        this.nodeTypes = dedupeNodes(nodeTypes);
    }
    searchByName(query, limit = 20) {
        const searchResults = (0, utils_1.sublimeSearch)(query, this.nodeTypes, NODE_SEARCH_KEYS);
        return searchResults.slice(0, limit).map(({ item, score }) => ({
            name: item.name,
            displayName: item.displayName,
            description: item.description ?? 'No description available',
            version: getLatestVersion(item.version),
            inputs: item.inputs,
            outputs: item.outputs,
            score,
        }));
    }
    searchByConnectionType(connectionType, limit = 20, nameFilter) {
        const nodesWithConnectionType = this.nodeTypes
            .map((nodeType) => {
            const connectionScore = this.getConnectionScore(nodeType, connectionType);
            return connectionScore > 0 ? { nodeType, connectionScore } : null;
        })
            .filter((result) => Boolean(result));
        if (!nameFilter) {
            return nodesWithConnectionType
                .sort((a, b) => b.connectionScore - a.connectionScore)
                .slice(0, limit)
                .map(({ nodeType, connectionScore }) => ({
                name: nodeType.name,
                displayName: nodeType.displayName,
                version: getLatestVersion(nodeType.version),
                description: nodeType.description ?? 'No description available',
                inputs: nodeType.inputs,
                outputs: nodeType.outputs,
                score: connectionScore,
            }));
        }
        const nodeTypesOnly = nodesWithConnectionType.map((result) => result.nodeType);
        const nameFilteredResults = (0, utils_1.sublimeSearch)(nameFilter, nodeTypesOnly, NODE_SEARCH_KEYS);
        return nameFilteredResults
            .slice(0, limit)
            .map(({ item, score: nameScore }) => {
            const connectionResult = nodesWithConnectionType.find((result) => result.nodeType.name === item.name);
            const connectionScore = connectionResult?.connectionScore ?? 0;
            return {
                name: item.name,
                version: getLatestVersion(item.version),
                displayName: item.displayName,
                description: item.description ?? 'No description available',
                inputs: item.inputs,
                outputs: item.outputs,
                score: connectionScore + nameScore,
            };
        });
    }
    formatResult(result) {
        return `
		<node>
			<node_name>${result.name}</node_name>
			<node_version>${result.version}</node_version>
			<node_description>${result.description}</node_description>
			<node_inputs>${typeof result.inputs === 'object' ? JSON.stringify(result.inputs) : result.inputs}</node_inputs>
			<node_outputs>${typeof result.outputs === 'object' ? JSON.stringify(result.outputs) : result.outputs}</node_outputs>
		</node>`;
    }
    getConnectionScore(nodeType, connectionType) {
        const outputs = nodeType.outputs;
        if (Array.isArray(outputs)) {
            if (outputs.includes(connectionType)) {
                return exports.SCORE_WEIGHTS.CONNECTION_EXACT;
            }
        }
        else if (typeof outputs === 'string') {
            if (outputs.includes(connectionType)) {
                return exports.SCORE_WEIGHTS.CONNECTION_IN_EXPRESSION;
            }
        }
        return 0;
    }
    static isAiConnectionType(connectionType) {
        return connectionType.startsWith('ai_');
    }
    static getAiConnectionTypes() {
        return Object.values(n8n_workflow_1.NodeConnectionTypes).filter((type) => NodeSearchEngine.isAiConnectionType(type));
    }
}
exports.NodeSearchEngine = NodeSearchEngine;
