// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/gitlabTrigger/v1.schema.js
/**
 * GitLab Trigger Node - Version 1 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
    owner: stringOrExpression.optional(),
    repository: stringOrExpression.optional(),
    events: z.array(z.union([z.literal('note'), z.literal('confidential_issues'), z.literal('confidential_note'), z.literal('deployment'), z.literal('issues'), z.literal('job'), z.literal('merge_requests'), z.literal('pipeline'), z.literal('push'), z.literal('releases'), z.literal('tag_push'), z.literal('wiki_page'), z.literal('*')])).optional(),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
