// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/clockify/v1/resource_task/operation_update.ts
/**
 * Clockify Node - Version 1
 * Discriminator: resource=task, operation=update
 */


interface Credentials {
  clockifyApi: CredentialReference;
}

/** Update a client */
export type ClockifyV1TaskUpdateParams = {
  resource: 'task';
  operation: 'update';
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 * @displayOptions.hide { resource: ["workspace"] }
 * @default []
 */
    workspaceId?: string | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    projectId?: string | Expression<string>;
/**
 * ID of task to update
 */
    taskId?: string | Expression<string> | PlaceholderValue;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     * @default []
     */
    assigneeIds?: string[];
    /** Estimate time the task will take, e.x: 2:30 (2 hours and 30 minutes)
     */
    estimate?: string | Expression<string> | PlaceholderValue;
    /** Name
     */
    name?: string | Expression<string> | PlaceholderValue;
    /** Status
     * @default ACTIVE
     */
    status?: 'ACTIVE' | 'DONE' | Expression<string>;
  };
};

export type ClockifyV1TaskUpdateNode = {
  type: 'n8n-nodes-base.clockify';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ClockifyV1TaskUpdateParams>;
};
