// @n8n/n8n-nodes-langchain/dist/node-definitions/nodes/n8n-nodes-langchain/vectorStoreMongoDBAtlas/v12/mode_update.schema.js
/**
 * MongoDB Atlas Vector Store Node - Version 1.2 - Zod Schema
 * Discriminator: mode=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, embeddingInstanceSchema }) {

  // Static subnode schema
  const subnodesSchema = z.object({
    embedding: z.union([embeddingInstanceSchema, z.array(embeddingInstanceSchema)]),
  }).strict();

  return z.object({
    parameters: z.object({
      mode: z.literal('update'),
      ragStarterCallout: z.unknown().optional(),
      mongoCollection: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('name')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      embedding: stringOrExpression.optional(),
      metadata_field: stringOrExpression.optional(),
      vectorIndexName: stringOrExpression.optional(),
      id: stringOrExpression.optional(),
    }).optional(),
    subnodes: subnodesSchema,
  });
};
