// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/matrix/v1/resource_message/operation_get_all.ts
/**
 * Matrix Node - Version 1
 * Discriminator: resource=message, operation=getAll
 */


interface Credentials {
  matrixApi: CredentialReference;
}

/** Get many messages from a room */
export type MatrixV1MessageGetAllParams = {
  resource: 'message';
  operation: 'getAll';
/**
 * The token to start returning events from. This token can be obtained from a prev_batch token returned for each room by the sync API. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    roomId?: string | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Other Options
 * @default {}
 */
    otherOptions?: {
    /** A JSON RoomEventFilter to filter returned events with. More information can be found on this &lt;a href="https://matrix.org/docs/spec/client_server/r0.6.0"&gt;page&lt;/a&gt;.
     */
    filter?: string | Expression<string> | PlaceholderValue;
  };
};

export type MatrixV1MessageGetAllOutput = {
  age?: number;
  content?: {
    body?: string;
    msgtype?: string;
  };
  event_id?: string;
  origin_server_ts?: number;
  room_id?: string;
  sender?: string;
  type?: string;
  unsigned?: {
    age?: number;
    membership?: string;
    transaction_id?: string;
  };
  user_id?: string;
};

export type MatrixV1MessageGetAllNode = {
  type: 'n8n-nodes-base.matrix';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MatrixV1MessageGetAllParams>;
  output?: Items<MatrixV1MessageGetAllOutput>;
};
