// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/matrix/v1/resource_message/operation_create.schema.js
/**
 * Matrix Node - Version 1 - Zod Schema
 * Discriminator: resource=message, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('message').default('message'),
      operation: z.literal('create'),
      roomId: stringOrExpression.optional(),
      text: stringOrExpression.optional(),
      messageType: z.union([z.literal('m.emote'), z.literal('m.notice'), z.literal('m.text'), expressionSchema]).optional(),
      messageFormat: z.union([z.literal('plain'), z.literal('org.matrix.custom.html'), expressionSchema]).optional(),
      fallbackText: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"messageFormat":["org.matrix.custom.html"]}}, defaults: {"messageFormat":"plain"} }),
    }).optional(),
  });
};
