// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/highLevel/v1/resource_contact/operation_lookup.ts
/**
 * HighLevel Node - Version 1
 * Discriminator: resource=contact, operation=lookup
 */


interface Credentials {
  highLevelApi: CredentialReference;
}

export type HighLevelV1ContactLookupParams = {
  resource: 'contact';
  operation: 'lookup';
/**
 * Lookup Contact by Email. If Email is not found it will try to find a contact by phone.
 */
    email?: string | Expression<string> | PlaceholderValue;
/**
 * Lookup Contact by Phone. It will first try to find a contact by Email and than by Phone.
 */
    phone?: string | Expression<string> | PlaceholderValue;
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type HighLevelV1ContactLookupOutput = {
  address1?: string;
  assignedTo?: string;
  city?: string;
  country?: string;
  customField?: Array<{
    id?: string;
  }>;
  dateAdded?: string;
  email?: string;
  emailLowerCase?: string;
  fingerprint?: string;
  firstName?: string;
  firstNameLowerCase?: string;
  fullNameLowerCase?: string;
  id?: string;
  lastName?: string;
  lastNameLowerCase?: string;
  locationId?: string;
  phone?: string;
  postalCode?: string;
  source?: string;
  state?: string;
  tags?: Array<string>;
  timezone?: string;
  type?: string;
};

export type HighLevelV1ContactLookupNode = {
  type: 'n8n-nodes-base.highLevel';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<HighLevelV1ContactLookupParams>;
  output?: Items<HighLevelV1ContactLookupOutput>;
};
