// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/highLevel/v1/resource_contact/operation_create.schema.js
/**
 * HighLevel Node - Version 1 - Zod Schema
 * Discriminator: resource=contact, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('contact').default('contact'),
      operation: z.literal('create').default('create'),
      email: stringOrExpression.optional(),
      phone: stringOrExpression.optional(),
      additionalFields: z.object({ address1: stringOrExpression.optional(), city: stringOrExpression.optional(), customFields: z.unknown().optional(), dnd: booleanOrExpression.optional(), firstName: stringOrExpression.optional(), lastName: stringOrExpression.optional(), name: stringOrExpression.optional(), postalCode: stringOrExpression.optional(), source: stringOrExpression.optional(), state: stringOrExpression.optional(), tags: stringOrExpression.optional(), timezone: stringOrExpression.optional(), website: stringOrExpression.optional() }).optional(),
      requestOptions: z.object({ batching: z.unknown().optional(), allowUnauthorizedCerts: z.boolean().optional(), proxy: stringOrExpression.optional(), timeout: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
