// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/n8n/v1/resource_workflow/operation_get_all.ts
/**
 * n8n Node - Version 1
 * Discriminator: resource=workflow, operation=getAll
 */


interface Credentials {
  n8nApi: CredentialReference;
}

export type N8nV1WorkflowGetAllParams = {
  resource: 'workflow';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default true
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Return Only Published Workflows
     * @default true
     */
    activeWorkflows?: boolean | Expression<boolean>;
    /** Include only workflows with these tags
     * @hint Comma separated list of tags (empty value is ignored)
     */
    tags?: string | Expression<string> | PlaceholderValue;
    /** Name
     */
    name?: string | Expression<string> | PlaceholderValue;
    /** Project ID
     */
    projectId?: string | Expression<string> | PlaceholderValue;
    /** Whether to exclude pinned data from the response
     * @default false
     */
    excludePinnedData?: boolean | Expression<boolean>;
  };
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type N8nV1WorkflowGetAllNode = {
  type: 'n8n-nodes-base.n8n';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<N8nV1WorkflowGetAllParams>;
};
