// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/jenkins/v1/resource_job/operation_trigger_params.ts
/**
 * Jenkins Node - Version 1
 * Discriminator: resource=job, operation=triggerParams
 */


interface Credentials {
  jenkinsApi: CredentialReference;
}

/** Trigger a specific job */
export type JenkinsV1JobTriggerParamsParams = {
  resource: 'job';
  operation: 'triggerParams';
/**
 * Name of the job. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    job?: string | Expression<string>;
/**
 * Parameters for Jenkins job
 * @default {}
 */
    param?: {
        /** Parameters
     */
    params?: Array<{
      /** Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
       */
      name?: string | Expression<string>;
      /** Value
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
};

export type JenkinsV1JobTriggerParamsOutput = {
  success?: boolean;
};

export type JenkinsV1JobTriggerParamsNode = {
  type: 'n8n-nodes-base.jenkins';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<JenkinsV1JobTriggerParamsParams>;
  output?: Items<JenkinsV1JobTriggerParamsOutput>;
};
