// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/hubspot/v1/resource_contact_list/operation_add.schema.js
/**
 * HubSpot Node - Version 1 - Zod Schema
 * Discriminator: resource=contactList, operation=add
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('contactList'),
      operation: z.literal('add'),
      authentication: z.union([z.literal('apiKey'), z.literal('appToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      by: z.union([z.literal('id'), z.literal('email'), expressionSchema]).optional(),
      email: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"by":["email"]}}, defaults: {"by":"email"} }),
      id: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"by":["id"]}}, defaults: {"by":"email"} }),
      listId: stringOrExpression.optional(),
    }).optional(),
  });
};
