// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/hubspot/v21/resource_ticket/operation_update.ts
/**
 * HubSpot Node - Version 2.1
 * Discriminator: resource=ticket, operation=update
 */


interface Credentials {
  hubspotApi: CredentialReference;
  hubspotAppToken: CredentialReference;
  hubspotOAuth2Api: CredentialReference;
}

/** Update a company */
export type HubspotV21TicketUpdateParams = {
  resource: 'ticket';
  operation: 'update';
  authentication?: 'apiKey' | 'appToken' | 'oAuth2' | Expression<string>;
/**
 * Ticket to Update
 * @default {"mode":"list","value":""}
 */
    ticketId?: { __rl: true; mode: 'list' | 'id'; value: string; cachedResultName?: string };
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** Whether to include specific Company properties in the returned results. Choose from a list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;. Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     * @default []
     */
    associatedCompanyIds?: string[];
    /** Whether to include specific Contact properties in the returned results. Choose from a list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;. Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     * @default []
     */
    associatedContactIds?: string[];
    /** Main reason customer reached out for help. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    category?: string | Expression<string>;
    /** The date the ticket was closed. When using expressions, the time should be specified in YYYY-MM-DD hh-mm-ss format.
     */
    closeDate?: string | Expression<string>;
    /** The date the ticket was created. When using expressions, the time should be specified in YYYY-MM-DD hh-mm-ss format.
     */
    createDate?: string | Expression<string>;
    /** Description of the ticket
     */
    description?: string | Expression<string> | PlaceholderValue;
    /** The ID of the pipeline the ticket is in. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    pipelineId?: string | Expression<string>;
    /** The level of attention needed on the ticket. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    priority?: string | Expression<string>;
    /** The action taken to resolve the ticket. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    resolution?: string | Expression<string>;
    /** Channel where ticket was originally submitted. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    source?: string | Expression<string>;
    /** The stage ID of the pipeline the ticket is in; depends on Pipeline ID. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    stageId?: string | Expression<string>;
    /** Ticket Name
     */
    ticketName?: string | Expression<string> | PlaceholderValue;
    /** The user from your team that the ticket is assigned to. You can assign additional users to a ticket record by creating a custom HubSpot user property. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    ticketOwnerId?: string | Expression<string>;
  };
};

export type HubspotV21TicketUpdateNode = {
  type: 'n8n-nodes-base.hubspot';
  version: 2.1;
  credentials?: Credentials;
  config: NodeConfig<HubspotV21TicketUpdateParams>;
};
