// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/invoiceNinja/v1/resource_task/operation_create.ts
/**
 * Invoice Ninja Node - Version 1
 * Discriminator: resource=task, operation=create
 */


interface Credentials {
  invoiceNinjaApi: CredentialReference;
}

/** Create a new client */
export type InvoiceNinjaV1TaskCreateParams = {
  resource: 'task';
  operation: 'create';
/**
 * API Version
 * @default v4
 */
    apiVersion?: 'v4' | 'v5' | Expression<string>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     */
    client?: string | Expression<string>;
    /** Custom Value 1
     */
    customValue1?: string | Expression<string> | PlaceholderValue;
    /** Custom Value 2
     */
    customValue2?: string | Expression<string> | PlaceholderValue;
    /** Description
     */
    description?: string | Expression<string> | PlaceholderValue;
    /** Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     */
    project?: string | Expression<string>;
  };
/**
 * Time Logs
 * @default {}
 */
    timeLogsUi?: {
        /** Time Log
     */
    timeLogsValues?: Array<{
      /** Start Date
       */
      startDate?: string | Expression<string>;
      /** End Date
       */
      endDate?: string | Expression<string>;
      /** Duration (Hours)
       * @default 0
       */
      duration?: number | Expression<number>;
    }>;
  };
};

export type InvoiceNinjaV1TaskCreateNode = {
  type: 'n8n-nodes-base.invoiceNinja';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<InvoiceNinjaV1TaskCreateParams>;
};
