// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/keap/v1/resource_contact_note/operation_get_all.ts
/**
 * Keap Node - Version 1
 * Discriminator: resource=contactNote, operation=getAll
 */


interface Credentials {
  keapOAuth2Api: CredentialReference;
}

/** Retrieve many companies */
export type KeapV1ContactNoteGetAllParams = {
  resource: 'contactNote';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Contact ID
     * @default 0
     */
    contactId?: number | Expression<number>;
    /** Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     */
    userId?: string | Expression<string>;
  };
};

export type KeapV1ContactNoteGetAllNode = {
  type: 'n8n-nodes-base.keap';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<KeapV1ContactNoteGetAllParams>;
};
