// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/keap/v1/resource_contact_note/operation_create.ts
/**
 * Keap Node - Version 1
 * Discriminator: resource=contactNote, operation=create
 */


interface Credentials {
  keapOAuth2Api: CredentialReference;
}

/** Create a company */
export type KeapV1ContactNoteCreateParams = {
  resource: 'contactNote';
  operation: 'create';
/**
 * The infusionsoft user to create the note on behalf of. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    userId?: string | Expression<string>;
/**
 * Contact ID
 */
    contactId?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Body
     */
    body?: string | Expression<string> | PlaceholderValue;
    /** Title
     */
    title?: string | Expression<string> | PlaceholderValue;
    /** Type
     */
    type?: 'appointment' | 'call' | 'email' | 'fax' | 'letter' | 'other' | Expression<string>;
  };
};

export type KeapV1ContactNoteCreateNode = {
  type: 'n8n-nodes-base.keap';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<KeapV1ContactNoteCreateParams>;
};
