// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/keap/v1/resource_contact_note/operation_update.schema.js
/**
 * Keap Node - Version 1 - Zod Schema
 * Discriminator: resource=contactNote, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('contactNote'),
      operation: z.literal('update'),
      noteId: stringOrExpression.optional(),
      additionalFields: z.object({ body: stringOrExpression.optional(), contactId: numberOrExpression.optional(), title: stringOrExpression.optional(), type: z.union([z.literal('appointment'), z.literal('call'), z.literal('email'), z.literal('fax'), z.literal('letter'), z.literal('other'), expressionSchema]).optional(), userId: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
