// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/onfleet/v1/resource_worker/operation_create.ts
/**
 * Onfleet Node - Version 1
 * Discriminator: resource=worker, operation=create
 */


interface Credentials {
  onfleetApi: CredentialReference;
}

/** Create a new Onfleet admin */
export type OnfleetV1WorkerCreateParams = {
  resource: 'worker';
  operation: 'create';
/**
 * The worker's name
 */
    name?: string | Expression<string> | PlaceholderValue;
/**
 * A list of worker’s phone numbers
 */
    phone?: string | Expression<string> | PlaceholderValue;
/**
 * One or more teams of which the worker is a member. Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 * @default []
 */
    teams?: string[];
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** The maximum number of units this worker can carry, for route optimization purposes
     * @default 0
     */
    capacity?: number | Expression<number>;
    /** This value is used in place of the worker's actual name within sms notifications, delivery tracking pages, and across organization boundaries
     */
    displayName?: string | Expression<string> | PlaceholderValue;
    /** Vehicle
     * @default {}
     */
    vehicle?: {
        /** Vehicle Properties
     */
    vehicleProperties?: {
      /** Whether the worker has vehicle or not. If it's not provided, this worker will be treated as if on foot.
       */
      type?: 'BICYCLE' | 'CAR' | 'MOTORCYCLE' | 'TRUCK' | Expression<string>;
      /** The vehicle's make, model, year, or any other relevant identifying details
       */
      description?: string | Expression<string> | PlaceholderValue;
      /** The vehicle's license plate number
       */
      licensePlate?: string | Expression<string> | PlaceholderValue;
      /** The vehicle's color
       */
      color?: string | Expression<string> | PlaceholderValue;
    };
  };
  };
};

export type OnfleetV1WorkerCreateNode = {
  type: 'n8n-nodes-base.onfleet';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<OnfleetV1WorkerCreateParams>;
};
