// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/onfleet/v1/resource_worker/operation_get.schema.js
/**
 * Onfleet Node - Version 1 - Zod Schema
 * Discriminator: resource=worker, operation=get
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('worker'),
      operation: z.literal('get'),
      id: stringOrExpression.optional(),
      options: z.object({ analytics: booleanOrExpression.optional(), filter: z.array(z.union([z.literal('accountStatus'), z.literal('activeTask'), z.literal('capacity'), z.literal('delayTime'), z.literal('displayName'), z.literal('imageUrl'), z.literal('location'), z.literal('metadata'), z.literal('name'), z.literal('onDuty'), z.literal('organization'), z.literal('phone'), z.literal('tasks'), z.literal('teams'), z.literal('timeCreated'), z.literal('timeLastModified'), z.literal('timeLastSeen'), z.literal('userData'), z.literal('vehicle'), z.literal('id')])).optional() }).optional(),
    }).optional(),
  });
};
