// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/onfleet/v1/resource_worker/operation_get.ts
/**
 * Onfleet Node - Version 1
 * Discriminator: resource=worker, operation=get
 */


interface Credentials {
  onfleetApi: CredentialReference;
}

/** Get container information */
export type OnfleetV1WorkerGetParams = {
  resource: 'worker';
  operation: 'get';
/**
 * The ID of the worker object for lookup
 */
    id?: string | Expression<string> | PlaceholderValue;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Whether a more detailed response is needed, includes basic worker duty event, traveled distance (meters) and time analytics
     * @default true
     */
    analytics?: boolean | Expression<boolean>;
    /** A list of fields to show in the response, if all are not desired
     * @default []
     */
    filter?: Array<'accountStatus' | 'activeTask' | 'capacity' | 'delayTime' | 'displayName' | 'imageUrl' | 'location' | 'metadata' | 'name' | 'onDuty' | 'organization' | 'phone' | 'tasks' | 'teams' | 'timeCreated' | 'timeLastModified' | 'timeLastSeen' | 'userData' | 'vehicle' | 'id'>;
  };
};

export type OnfleetV1WorkerGetNode = {
  type: 'n8n-nodes-base.onfleet';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<OnfleetV1WorkerGetParams>;
};
