// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/onfleet/v1/resource_worker/operation_update.ts
/**
 * Onfleet Node - Version 1
 * Discriminator: resource=worker, operation=update
 */


interface Credentials {
  onfleetApi: CredentialReference;
}

/** Update an Onfleet admin */
export type OnfleetV1WorkerUpdateParams = {
  resource: 'worker';
  operation: 'update';
/**
 * The ID of the worker object for lookup
 */
    id?: string | Expression<string> | PlaceholderValue;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** The maximum number of units this worker can carry, for route optimization purposes
     * @default 0
     */
    capacity?: number | Expression<number>;
    /** This value is used in place of the worker's actual name within sms notifications, delivery tracking pages, and across organization boundaries
     */
    displayName?: string | Expression<string> | PlaceholderValue;
    /** The worker's name
     */
    name?: string | Expression<string> | PlaceholderValue;
    /** One or more teams of which the worker is a member. Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     * @default []
     */
    teams?: string[];
  };
};

export type OnfleetV1WorkerUpdateNode = {
  type: 'n8n-nodes-base.onfleet';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<OnfleetV1WorkerUpdateParams>;
};
