// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/currents/v1/resource_action/operation_create.ts
/**
 * Currents Node - Version 1
 * Discriminator: resource=action, operation=create
 */


interface Credentials {
  currentsApi: CredentialReference;
}

/** Test action rules (skip, quarantine, tag) */
export type CurrentsV1ActionCreateParams = {
  resource: 'action';
  operation: 'create';
/**
 * The Currents project
 * @default {"mode":"list","value":""}
 */
    projectId?: { __rl: true; mode: 'list' | 'id'; value: string; cachedResultName?: string };
/**
 * The name of the action (1-255 characters)
 */
    name?: string | Expression<string> | PlaceholderValue;
/**
 * Action Type
 * @default quarantine
 */
    actionType?: 'quarantine' | 'skip' | 'tag' | Expression<string>;
/**
 * Comma-separated list of tags to apply
 * @displayOptions.show { actionType: ["tag"] }
 */
    actionTags?: string | Expression<string> | PlaceholderValue;
/**
 * How to match tests for this action
 * @default titleContains
 */
    matcherType?: 'specContains' | 'specEquals' | 'signature' | 'titleContains' | 'titleEquals' | Expression<string>;
/**
 * The value to match against (test title, spec file path, or signature)
 */
    matcherValue?: string | Expression<string> | PlaceholderValue;
/**
 * Options
 * @default {}
 */
    createOptions?: {
    /** A description for the action
     */
    description?: string | Expression<string> | PlaceholderValue;
    /** When the action should expire (ISO 8601 format)
     */
    expiresAfter?: string | Expression<string>;
  };
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type CurrentsV1ActionCreateNode = {
  type: 'n8n-nodes-base.currents';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<CurrentsV1ActionCreateParams>;
};
