// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/currents/v1/resource_action/operation_create.schema.js
/**
 * Currents Node - Version 1 - Zod Schema
 * Discriminator: resource=action, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('action'),
      operation: z.literal('create'),
      projectId: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      name: stringOrExpression.optional(),
      actionType: z.union([z.literal('quarantine'), z.literal('skip'), z.literal('tag'), expressionSchema]).optional(),
      actionTags: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"actionType":["tag"]}}, defaults: {"actionType":"quarantine"} }),
      matcherType: z.union([z.literal('specContains'), z.literal('specEquals'), z.literal('signature'), z.literal('titleContains'), z.literal('titleEquals'), expressionSchema]).optional(),
      matcherValue: stringOrExpression.optional(),
      createOptions: z.object({ description: stringOrExpression.optional(), expiresAfter: stringOrExpression.optional() }).optional(),
      requestOptions: z.object({ batching: z.unknown().optional(), allowUnauthorizedCerts: z.boolean().optional(), proxy: stringOrExpression.optional(), timeout: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
