// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/coinGecko/v1/resource_coin/operation_market.schema.js
/**
 * CoinGecko Node - Version 1 - Zod Schema
 * Discriminator: resource=coin, operation=market
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('coin').default('coin'),
      operation: z.literal('market'),
      baseCurrency: stringOrExpression.optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      options: z.object({ ids: stringOrExpression.optional(), category: z.union([z.literal('decentralized_finance_defi'), expressionSchema]).optional(), order: z.union([z.literal('gecko_asc'), z.literal('gecko_desc'), z.literal('id_asc'), z.literal('id_desc'), z.literal('market_cap_asc'), z.literal('market_cap_desc'), z.literal('volume_asc'), z.literal('volume_desc'), expressionSchema]).optional(), sparkline: booleanOrExpression.optional(), price_change_percentage: z.array(z.union([z.literal('1h'), z.literal('24h'), z.literal('7d'), z.literal('14d'), z.literal('30d'), z.literal('200d'), z.literal('1y')])).optional() }).optional(),
    }).optional(),
  });
};
