// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/coinGecko/v1/resource_coin/operation_market_chart.schema.js
/**
 * CoinGecko Node - Version 1 - Zod Schema
 * Discriminator: resource=coin, operation=marketChart
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('coin').default('coin'),
      operation: z.literal('marketChart'),
      searchBy: z.union([z.literal('coinId'), z.literal('contractAddress'), expressionSchema]).optional(),
      platformId: resolveSchema({ parameters, schema: z.union([z.literal('ethereum'), expressionSchema]), required: false, displayOptions: {"show":{"searchBy":["contractAddress"]}}, defaults: {"searchBy":"coinId"} }),
      contractAddress: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"searchBy":["contractAddress"]}}, defaults: {"searchBy":"coinId"} }),
      baseCurrency: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"searchBy":["coinId"]},"hide":{"searchBy":["contractAddress"]}}, defaults: {"searchBy":"coinId"} }),
      quoteCurrency: stringOrExpression.optional(),
      days: z.union([z.literal('1'), z.literal('7'), z.literal('14'), z.literal('30'), z.literal('90'), z.literal('180'), z.literal('365'), z.literal('max'), expressionSchema]).optional(),
    }).optional(),
  });
};
