// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/coinGecko/v1/resource_coin/operation_history.ts
/**
 * CoinGecko Node - Version 1
 * Discriminator: resource=coin, operation=history
 */


/** Get historical data (name, price, market, stats) at a given date for a coin */
export type CoinGeckoV1CoinHistoryParams = {
  resource: 'coin';
  operation: 'history';
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    coinId?: string | Expression<string>;
/**
 * The date of data snapshot
 */
    date?: string | Expression<string>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Whether to exclude localized languages in response
     * @default true
     */
    localization?: boolean | Expression<boolean>;
  };
};

export type CoinGeckoV1CoinHistoryOutput = {
  community_data?: {
    facebook_likes?: null;
    reddit_average_comments_48h?: number;
    reddit_average_posts_48h?: number;
    reddit_subscribers?: null;
  };
  developer_data?: {
    closed_issues?: null;
    code_additions_deletions_4_weeks?: {
      additions?: null;
      deletions?: null;
    };
    commit_count_4_weeks?: null;
    forks?: null;
    pull_request_contributors?: null;
    pull_requests_merged?: null;
    stars?: null;
    subscribers?: null;
    total_issues?: null;
  };
  id?: string;
  image?: {
    small?: string;
    thumb?: string;
  };
  localization?: {
    ar?: string;
    bg?: string;
    cs?: string;
    da?: string;
    de?: string;
    el?: string;
    en?: string;
    es?: string;
    fi?: string;
    fr?: string;
    he?: string;
    hi?: string;
    hr?: string;
    hu?: string;
    id?: string;
    it?: string;
    ja?: string;
    ko?: string;
    lt?: string;
    nl?: string;
    no?: string;
    pl?: string;
    pt?: string;
    ro?: string;
    ru?: string;
    sk?: string;
    sl?: string;
    sv?: string;
    th?: string;
    tr?: string;
    uk?: string;
    vi?: string;
    zh?: string;
    'zh-tw'?: string;
  };
  market_data?: {
    current_price?: {
      aed?: number;
      ars?: number;
      aud?: number;
      bch?: number;
      bdt?: number;
      bhd?: number;
      bits?: number;
      bmd?: number;
      bnb?: number;
      brl?: number;
      cad?: number;
      chf?: number;
      clp?: number;
      cny?: number;
      czk?: number;
      dkk?: number;
      dot?: number;
      eos?: number;
      eth?: number;
      eur?: number;
      gbp?: number;
      gel?: number;
      hkd?: number;
      huf?: number;
      idr?: number;
      ils?: number;
      inr?: number;
      jpy?: number;
      krw?: number;
      kwd?: number;
      link?: number;
      lkr?: number;
      ltc?: number;
      mmk?: number;
      mxn?: number;
      myr?: number;
      ngn?: number;
      nok?: number;
      nzd?: number;
      php?: number;
      pkr?: number;
      pln?: number;
      rub?: number;
      sar?: number;
      sats?: number;
      sek?: number;
      sgd?: number;
      thb?: number;
      'try'?: number;
      twd?: number;
      uah?: number;
      usd?: number;
      vef?: number;
      vnd?: number;
      xag?: number;
      xau?: number;
      xdr?: number;
      xlm?: number;
      xrp?: number;
      yfi?: number;
      zar?: number;
    };
    total_volume?: {
      aed?: number;
      aud?: number;
      bch?: number;
      bdt?: number;
      bhd?: number;
      bits?: number;
      bmd?: number;
      bnb?: number;
      brl?: number;
      btc?: number;
      cad?: number;
      chf?: number;
      clp?: number;
      cny?: number;
      czk?: number;
      dkk?: number;
      dot?: number;
      eos?: number;
      eth?: number;
      eur?: number;
      gbp?: number;
      gel?: number;
      hkd?: number;
      huf?: number;
      ils?: number;
      inr?: number;
      jpy?: number;
      krw?: number;
      kwd?: number;
      link?: number;
      lkr?: number;
      ltc?: number;
      mmk?: number;
      mxn?: number;
      myr?: number;
      ngn?: number;
      nok?: number;
      nzd?: number;
      php?: number;
      pkr?: number;
      pln?: number;
      rub?: number;
      sar?: number;
      sats?: number;
      sek?: number;
      sgd?: number;
      thb?: number;
      'try'?: number;
      twd?: number;
      uah?: number;
      usd?: number;
      vef?: number;
      xag?: number;
      xau?: number;
      xdr?: number;
      xlm?: number;
      xrp?: number;
      yfi?: number;
      zar?: number;
    };
  };
  name?: string;
  public_interest_stats?: {
    alexa_rank?: null;
    bing_matches?: null;
  };
  symbol?: string;
};

export type CoinGeckoV1CoinHistoryNode = {
  type: 'n8n-nodes-base.coinGecko';
  version: 1;
  config: NodeConfig<CoinGeckoV1CoinHistoryParams>;
  output?: Items<CoinGeckoV1CoinHistoryOutput>;
};
