// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/coinGecko/v1/resource_coin/operation_price.schema.js
/**
 * CoinGecko Node - Version 1 - Zod Schema
 * Discriminator: resource=coin, operation=price
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('coin').default('coin'),
      operation: z.literal('price'),
      searchBy: z.union([z.literal('coinId'), z.literal('contractAddress'), expressionSchema]).optional(),
      baseCurrencies: resolveSchema({ parameters, schema: z.array(z.string()), required: false, displayOptions: {"show":{"searchBy":["coinId"]}}, defaults: {"searchBy":"coinId"} }),
      platformId: resolveSchema({ parameters, schema: z.union([z.literal('ethereum'), expressionSchema]), required: false, displayOptions: {"show":{"searchBy":["contractAddress"]}}, defaults: {"searchBy":"coinId"} }),
      contractAddresses: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"searchBy":["contractAddress"]}}, defaults: {"searchBy":"coinId"} }),
      quoteCurrencies: z.array(z.string()).optional(),
      options: z.object({ include_24hr_change: booleanOrExpression.optional(), include_24hr_vol: booleanOrExpression.optional(), include_last_updated_at: booleanOrExpression.optional(), include_market_cap: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
