// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/coinGecko/v1/resource_coin/operation_get.schema.js
/**
 * CoinGecko Node - Version 1 - Zod Schema
 * Discriminator: resource=coin, operation=get
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('coin').default('coin'),
      operation: z.literal('get'),
      searchBy: z.union([z.literal('coinId'), z.literal('contractAddress'), expressionSchema]).optional(),
      coinId: stringOrExpression.optional(),
      platformId: resolveSchema({ parameters, schema: z.union([z.literal('ethereum'), expressionSchema]), required: false, displayOptions: {"show":{"searchBy":["contractAddress"]}}, defaults: {"searchBy":"coinId"} }),
      contractAddress: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"searchBy":["contractAddress"]}}, defaults: {"searchBy":"coinId"} }),
      options: z.object({ community_data: booleanOrExpression.optional(), developer_data: booleanOrExpression.optional(), localization: booleanOrExpression.optional(), market_data: booleanOrExpression.optional(), sparkline: booleanOrExpression.optional(), tickers: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
