// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/monicaCrm/v1/resource_contact/operation_get_all.ts
/**
 * Monica CRM Node - Version 1
 * Discriminator: resource=contact, operation=getAll
 */


interface Credentials {
  monicaCrmApi: CredentialReference;
}

/** Retrieve many activities */
export type MonicaCrmV1ContactGetAllParams = {
  resource: 'contact';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Search term to filter results by
     */
    query?: string | Expression<string> | PlaceholderValue;
    /** Sort
     */
    sort?: 'created_at' | '-created_at' | 'updated_at' | '-updated_at' | Expression<string>;
  };
};

export type MonicaCrmV1ContactGetAllNode = {
  type: 'n8n-nodes-base.monicaCrm';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MonicaCrmV1ContactGetAllParams>;
};
