// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/monicaCrm/v1/resource_contact/operation_create.ts
/**
 * Monica CRM Node - Version 1
 * Discriminator: resource=contact, operation=create
 */


interface Credentials {
  monicaCrmApi: CredentialReference;
}

/** Create an activity */
export type MonicaCrmV1ContactCreateParams = {
  resource: 'contact';
  operation: 'create';
/**
 * First Name
 */
    firstName?: string | Expression<string> | PlaceholderValue;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    genderId?: string | Expression<string>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Birthdate
     */
    birthdate?: string | Expression<string>;
    /** Deceased Date
     */
    deceasedDate?: string | Expression<string>;
    /** Whether the contact has passed away
     * @default false
     */
    isDeceased?: boolean | Expression<boolean>;
    /** Last Name
     */
    last_name?: string | Expression<string> | PlaceholderValue;
    /** Nickname
     */
    nickname?: string | Expression<string> | PlaceholderValue;
    /** Type
     * @default false
     */
    is_partial?: false | true | Expression<boolean>;
  };
};

export type MonicaCrmV1ContactCreateNode = {
  type: 'n8n-nodes-base.monicaCrm';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MonicaCrmV1ContactCreateParams>;
};
