// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/monicaCrm/v1/resource_contact/operation_update.ts
/**
 * Monica CRM Node - Version 1
 * Discriminator: resource=contact, operation=update
 */


interface Credentials {
  monicaCrmApi: CredentialReference;
}

/** Update an activity */
export type MonicaCrmV1ContactUpdateParams = {
  resource: 'contact';
  operation: 'update';
/**
 * ID of the contact to update
 */
    contactId?: string | Expression<string> | PlaceholderValue;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** Birthdate
     */
    birthdate?: string | Expression<string>;
    /** Deceased Date
     */
    deceased_date?: string | Expression<string>;
    /** First Name
     */
    first_name?: string | Expression<string> | PlaceholderValue;
    /** Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     */
    gender_id?: string | Expression<string>;
    /** Whether the contact has passed away
     * @default false
     */
    is_deceased?: boolean | Expression<boolean>;
    /** Last Name
     */
    last_name?: string | Expression<string> | PlaceholderValue;
    /** Nickname
     */
    nickname?: string | Expression<string> | PlaceholderValue;
    /** Type
     * @default false
     */
    is_partial?: false | true | Expression<boolean>;
  };
};

export type MonicaCrmV1ContactUpdateNode = {
  type: 'n8n-nodes-base.monicaCrm';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MonicaCrmV1ContactUpdateParams>;
};
