// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/monicaCrm/v1/resource_contact/operation_create.schema.js
/**
 * Monica CRM Node - Version 1 - Zod Schema
 * Discriminator: resource=contact, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('contact').default('contact'),
      operation: z.literal('create').default('create'),
      firstName: stringOrExpression.optional(),
      genderId: stringOrExpression.optional(),
      additionalFields: z.object({ birthdate: stringOrExpression.optional(), deceasedDate: stringOrExpression.optional(), isDeceased: booleanOrExpression.optional(), last_name: stringOrExpression.optional(), nickname: stringOrExpression.optional(), is_partial: z.union([z.literal(false), z.literal(true), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
