// @n8n/ai-workflow-builder/dist/subgraphs/builder.subgraph.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BuilderSubgraph = exports.BuilderSubgraphState = void 0;
const prompts_1 = require("@langchain/core/prompts");
const langgraph_1 = require("@langchain/langgraph");
const errors_1 = require("../errors");
const prompts_2 = require("../prompts");
const auto_fix_1 = require("../validation/auto-fix");
const checks_1 = require("../validation/checks");
const subgraph_interface_1 = require("./subgraph-interface");
const add_node_tool_1 = require("../tools/add-node.tool");
const connect_nodes_tool_1 = require("../tools/connect-nodes.tool");
const get_execution_logs_tool_1 = require("../tools/get-execution-logs.tool");
const get_execution_schema_tool_1 = require("../tools/get-execution-schema.tool");
const get_expression_data_mapping_tool_1 = require("../tools/get-expression-data-mapping.tool");
const get_node_context_tool_1 = require("../tools/get-node-context.tool");
const get_node_examples_tool_1 = require("../tools/get-node-examples.tool");
const get_node_parameter_tool_1 = require("../tools/get-node-parameter.tool");
const get_resource_locator_options_tool_1 = require("../tools/get-resource-locator-options.tool");
const get_workflow_overview_tool_1 = require("../tools/get-workflow-overview.tool");
const introspect_tool_1 = require("../tools/introspect.tool");
const remove_connection_tool_1 = require("../tools/remove-connection.tool");
const remove_node_tool_1 = require("../tools/remove-node.tool");
const rename_node_tool_1 = require("../tools/rename-node.tool");
const update_node_parameters_tool_1 = require("../tools/update-node-parameters.tool");
const mermaid_utils_1 = require("../tools/utils/mermaid.utils");
const validate_configuration_tool_1 = require("../tools/validate-configuration.tool");
const validate_structure_tool_1 = require("../tools/validate-structure.tool");
const coordination_1 = require("../types/coordination");
const langchain_1 = require("../types/langchain");
const cache_control_1 = require("../utils/cache-control");
const context_builders_1 = require("../utils/context-builders");
const operations_processor_1 = require("../utils/operations-processor");
const plan_helpers_1 = require("../utils/plan-helpers");
const rlc_prefetch_1 = require("../utils/rlc-prefetch");
const state_reducers_1 = require("../utils/state-reducers");
const subgraph_helpers_1 = require("../utils/subgraph-helpers");
exports.BuilderSubgraphState = langgraph_1.Annotation.Root({
    workflowJSON: (0, langgraph_1.Annotation)({
        reducer: (x, y) => y ?? x,
        default: () => ({ nodes: [], connections: {}, name: '' }),
    }),
    userRequest: (0, langgraph_1.Annotation)({
        reducer: (x, y) => y ?? x,
        default: () => '',
    }),
    workflowContext: (0, langgraph_1.Annotation)({
        reducer: (x, y) => y ?? x,
    }),
    discoveryContext: (0, langgraph_1.Annotation)({
        reducer: (x, y) => y ?? x,
        default: () => null,
    }),
    instanceUrl: (0, langgraph_1.Annotation)({
        reducer: (x, y) => y ?? x,
        default: () => '',
    }),
    messages: (0, langgraph_1.Annotation)({
        reducer: (x, y) => x.concat(y),
        default: () => [],
    }),
    workflowOperations: (0, langgraph_1.Annotation)({
        reducer: (x, y) => {
            if (y === null)
                return [];
            if (!y || y.length === 0)
                return x ?? [];
            return [...(x ?? []), ...y];
        },
        default: () => [],
    }),
    cachedTemplates: (0, langgraph_1.Annotation)({
        reducer: state_reducers_1.cachedTemplatesReducer,
        default: () => [],
    }),
    prefetchedRLCOptions: (0, langgraph_1.Annotation)({
        reducer: (x, y) => y ?? x,
        default: () => [],
    }),
});
class BuilderSubgraph extends subgraph_interface_1.BaseSubgraph {
    name = 'builder_subgraph';
    description = 'Constructs workflow structure and configures node parameters';
    config;
    agent;
    toolMap;
    instanceUrl = '';
    nodeTypes = [];
    resourceLocatorCallback;
    logger;
    create(config) {
        this.config = config;
        this.instanceUrl = config.instanceUrl ?? '';
        this.nodeTypes = config.parsedNodeTypes;
        this.resourceLocatorCallback = config.resourceLocatorCallback;
        this.logger = config.logger;
        const includeExamples = config.featureFlags?.templateExamples === true;
        const enableIntrospection = config.featureFlags?.enableIntrospection === true;
        const parameterUpdaterLLM = config.llmParameterUpdater ?? config.llm;
        const baseTools = [
            (0, add_node_tool_1.createAddNodeTool)(config.parsedNodeTypes),
            (0, connect_nodes_tool_1.createConnectNodesTool)(config.parsedNodeTypes, config.logger),
            (0, remove_node_tool_1.createRemoveNodeTool)(config.logger),
            (0, remove_connection_tool_1.createRemoveConnectionTool)(config.logger),
            (0, rename_node_tool_1.createRenameNodeTool)(config.logger),
            (0, validate_structure_tool_1.createValidateStructureTool)(config.parsedNodeTypes),
            (0, update_node_parameters_tool_1.createUpdateNodeParametersTool)(config.parsedNodeTypes, parameterUpdaterLLM, config.logger, config.instanceUrl),
            (0, get_node_parameter_tool_1.createGetNodeParameterTool)(),
            (0, validate_configuration_tool_1.createValidateConfigurationTool)(config.parsedNodeTypes),
            (0, get_execution_schema_tool_1.createGetExecutionSchemaTool)(config.logger),
            (0, get_execution_logs_tool_1.createGetExecutionLogsTool)(config.logger),
            (0, get_expression_data_mapping_tool_1.createGetExpressionDataMappingTool)(config.logger),
            (0, get_workflow_overview_tool_1.createGetWorkflowOverviewTool)(config.logger),
            (0, get_node_context_tool_1.createGetNodeContextTool)(config.logger),
        ];
        if (config.resourceLocatorCallback) {
            baseTools.push((0, get_resource_locator_options_tool_1.createGetResourceLocatorOptionsTool)(config.parsedNodeTypes, config.resourceLocatorCallback, config.logger));
        }
        if (enableIntrospection) {
            baseTools.push((0, introspect_tool_1.createIntrospectTool)(config.logger));
        }
        const tools = includeExamples
            ? [
                ...baseTools,
                (0, get_node_examples_tool_1.createGetNodeConnectionExamplesTool)(config.logger),
                (0, get_node_examples_tool_1.createGetNodeConfigurationExamplesTool)(config.logger),
            ]
            : baseTools;
        this.toolMap = new Map(tools.map((bt) => [bt.tool.name, bt.tool]));
        const systemPromptTemplate = prompts_1.ChatPromptTemplate.fromMessages([
            [
                'system',
                [
                    {
                        type: 'text',
                        text: (0, prompts_2.buildBuilderPrompt)({ includeExamples, enableIntrospection }),
                    },
                    {
                        type: 'text',
                        text: prompts_2.INSTANCE_URL_PROMPT,
                        cache_control: { type: 'ephemeral' },
                    },
                ],
            ],
            ['placeholder', '{messages}'],
        ]);
        if (typeof config.llm.bindTools !== 'function') {
            throw new errors_1.LLMServiceError('LLM does not support tools', {
                llmModel: config.llm._llmType(),
            });
        }
        this.agent = systemPromptTemplate.pipe(config.llm.bindTools(tools.map((bt) => bt.tool)));
        const prefetchRLCNode = async (state) => {
            if (!this.resourceLocatorCallback) {
                return { prefetchedRLCOptions: [] };
            }
            const parametersToFetch = (0, rlc_prefetch_1.detectRLCParametersForPrefetch)(state.workflowJSON.nodes, this.nodeTypes);
            if (parametersToFetch.length === 0) {
                return { prefetchedRLCOptions: [] };
            }
            this.logger?.debug('Pre-fetching RLC options', {
                parameterCount: parametersToFetch.length,
                parameters: parametersToFetch.map((p) => `${p.nodeName}.${p.parameterPath}`),
            });
            const results = await (0, rlc_prefetch_1.prefetchRLCOptions)(parametersToFetch, state.workflowJSON.nodes, this.resourceLocatorCallback, this.logger);
            this.logger?.debug('RLC prefetch completed', {
                successCount: results.filter((r) => !r.error).length,
                errorCount: results.filter((r) => r.error).length,
            });
            const formattedOptions = (0, rlc_prefetch_1.formatPrefetchedOptionsForLLM)(results);
            if (formattedOptions && state.messages.length > 0) {
                const firstMessage = state.messages[0];
                if (typeof firstMessage.content === 'string') {
                    firstMessage.content = `${firstMessage.content}\n\n${formattedOptions}`;
                }
            }
            return { prefetchedRLCOptions: results };
        };
        const callAgent = async (state) => {
            (0, cache_control_1.applySubgraphCacheMarkers)(state.messages);
            const response = await this.agent.invoke({
                messages: state.messages,
                instanceUrl: state.instanceUrl ?? '',
            });
            if (!(0, langchain_1.isBaseMessage)(response)) {
                throw new errors_1.LLMServiceError('Builder agent did not return a valid message');
            }
            return { messages: [response] };
        };
        const shouldContinue = (0, subgraph_helpers_1.createStandardShouldContinue)();
        const subgraph = new langgraph_1.StateGraph(exports.BuilderSubgraphState)
            .addNode('prefetch_rlc', prefetchRLCNode)
            .addNode('agent', callAgent)
            .addNode('tools', async (state) => await (0, subgraph_helpers_1.executeSubgraphTools)(state, this.toolMap))
            .addNode('process_operations', operations_processor_1.processOperations)
            .addEdge('__start__', 'prefetch_rlc')
            .addEdge('prefetch_rlc', 'agent')
            .addConditionalEdges('agent', shouldContinue)
            .addEdge('tools', 'process_operations')
            .addEdge('process_operations', 'agent');
        return subgraph.compile();
    }
    transformInput(parentState) {
        const userRequest = (0, subgraph_helpers_1.extractUserRequest)(parentState.messages);
        const contextParts = [];
        if (parentState.planOutput) {
            contextParts.push('=== APPROVED PLAN (FOLLOW THIS) ===');
            contextParts.push((0, plan_helpers_1.formatPlanAsText)(parentState.planOutput));
            if (parentState.workflowJSON?.nodes?.length > 0) {
                const overview = (0, mermaid_utils_1.mermaidStringify)({ workflow: parentState.workflowJSON }, { includeNodeType: true, includeNodeParameters: true, includeNodeName: true });
                contextParts.push('=== EXISTING WORKFLOW (do NOT recreate these nodes) ===', overview, 'Only make the changes described in the plan.');
            }
        }
        else {
            const conversationContext = (0, context_builders_1.buildConversationContext)(parentState.messages, parentState.coordinationLog, parentState.previousSummary);
            if (conversationContext) {
                contextParts.push('=== CONVERSATION CONTEXT ===');
                contextParts.push(conversationContext);
            }
            if (userRequest) {
                contextParts.push('=== USER REQUEST ===');
                contextParts.push(userRequest);
            }
        }
        const selectedNodesBlock = (0, context_builders_1.buildSelectedNodesContextBlock)(parentState.workflowContext);
        if (selectedNodesBlock) {
            contextParts.push('=== SELECTED NODES ===');
            contextParts.push(selectedNodesBlock);
        }
        if (parentState.discoveryContext) {
            const includeBestPractices = this.config?.featureFlags?.templateExamples === true;
            contextParts.push('=== DISCOVERY CONTEXT ===');
            contextParts.push((0, context_builders_1.buildDiscoveryContextBlock)(parentState.discoveryContext, includeBestPractices));
        }
        const builderErrorEntry = parentState.coordinationLog?.find((entry) => {
            if (entry.status !== 'error')
                return false;
            if (entry.phase !== 'builder')
                return false;
            return (entry.metadata.phase === 'error' &&
                'partialBuilderData' in entry.metadata &&
                entry.metadata.partialBuilderData);
        });
        if (builderErrorEntry?.metadata.phase === 'error' &&
            builderErrorEntry.metadata.partialBuilderData) {
            const { nodeCount, nodeNames } = builderErrorEntry.metadata.partialBuilderData;
            contextParts.push((0, prompts_2.buildRecoveryModeContext)(nodeCount, nodeNames));
        }
        contextParts.push('=== CURRENT WORKFLOW ===');
        if (parentState.workflowJSON.nodes.length > 0) {
            contextParts.push((0, context_builders_1.buildWorkflowJsonBlock)(parentState.workflowJSON));
        }
        else {
            contextParts.push('Empty workflow - ready to build');
        }
        const schemaBlock = (0, context_builders_1.buildExecutionSchemaBlock)(parentState.workflowContext);
        if (schemaBlock) {
            contextParts.push('=== AVAILABLE DATA SCHEMA ===');
            contextParts.push(schemaBlock);
        }
        contextParts.push('=== EXECUTION CONTEXT ===');
        contextParts.push((0, context_builders_1.buildExecutionContextBlock)(parentState.workflowContext));
        const contextMessage = (0, context_builders_1.createContextMessage)(contextParts);
        return {
            userRequest,
            workflowJSON: parentState.workflowJSON,
            workflowContext: parentState.workflowContext,
            discoveryContext: parentState.discoveryContext,
            instanceUrl: this.instanceUrl,
            messages: [contextMessage],
            cachedTemplates: parentState.cachedTemplates,
        };
    }
    transformOutput(subgraphOutput, _parentState) {
        let workflowJSON = subgraphOutput.workflowJSON;
        let autoFixSummary;
        if (this.config?.parsedNodeTypes) {
            const violations = (0, checks_1.validateConnections)(workflowJSON, this.config.parsedNodeTypes);
            const autoFixResult = (0, auto_fix_1.autoFixConnections)(workflowJSON, this.config.parsedNodeTypes, violations);
            if (autoFixResult.fixed.length > 0) {
                workflowJSON = {
                    ...workflowJSON,
                    connections: autoFixResult.updatedConnections,
                };
                autoFixSummary = `Auto-fixed ${autoFixResult.fixed.length} connection(s): ${autoFixResult.fixed
                    .map((fix) => `${fix.sourceNodeName} → ${fix.targetNodeName}`)
                    .join(', ')}`;
            }
        }
        const nodes = workflowJSON.nodes;
        const connections = workflowJSON.connections;
        const connectionCount = Object.values(connections).flat().length;
        const lastMessage = subgraphOutput.messages[subgraphOutput.messages.length - 1];
        const summaryOrSetupInstructions = typeof lastMessage?.content === 'string'
            ? lastMessage.content
            : 'Build and configuration complete';
        const summary = autoFixSummary
            ? `Created and configured ${nodes.length} nodes with ${connectionCount} connections. ${autoFixSummary}`
            : `Created and configured ${nodes.length} nodes with ${connectionCount} connections`;
        const logEntry = {
            phase: 'builder',
            status: 'completed',
            timestamp: Date.now(),
            summary,
            output: summaryOrSetupInstructions,
            metadata: (0, coordination_1.createBuilderMetadata)({
                nodesCreated: nodes.length,
                connectionsCreated: connectionCount,
                nodeNames: nodes.map((n) => n.name),
            }),
        };
        const toolMessages = (0, subgraph_helpers_1.extractToolMessagesForPersistence)(subgraphOutput.messages);
        const introspectionEvents = (0, introspect_tool_1.extractIntrospectionEventsFromMessages)(subgraphOutput.messages);
        return {
            workflowJSON,
            workflowOperations: subgraphOutput.workflowOperations ?? [],
            coordinationLog: [logEntry],
            cachedTemplates: subgraphOutput.cachedTemplates,
            introspectionEvents,
            planOutput: null,
            planDecision: null,
            planFeedback: null,
            planPrevious: null,
            messages: toolMessages,
        };
    }
}
exports.BuilderSubgraph = BuilderSubgraph;
