// @n8n/ai-workflow-builder/dist/subgraphs/discovery.subgraph.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiscoverySubgraph = exports.DiscoverySubgraphState = void 0;
const messages_1 = require("@langchain/core/messages");
const prompts_1 = require("@langchain/core/prompts");
const tools_1 = require("@langchain/core/tools");
const langgraph_1 = require("@langchain/langgraph");
const zod_1 = require("zod");
const planner_agent_1 = require("../agents/planner.agent");
const errors_1 = require("../errors");
const prompts_2 = require("../prompts");
const get_documentation_tool_1 = require("../tools/get-documentation.tool");
const get_workflow_examples_tool_1 = require("../tools/get-workflow-examples.tool");
const introspect_tool_1 = require("../tools/introspect.tool");
const node_search_tool_1 = require("../tools/node-search.tool");
const submit_questions_tool_1 = require("../tools/submit-questions.tool");
const coordination_1 = require("../types/coordination");
const cache_control_1 = require("../utils/cache-control");
const context_builders_1 = require("../utils/context-builders");
const resource_operation_extractor_1 = require("../utils/resource-operation-extractor");
const state_reducers_1 = require("../utils/state-reducers");
const subgraph_helpers_1 = require("../utils/subgraph-helpers");
const subgraph_interface_1 = require("./subgraph-interface");
const discoveryOutputSchema = zod_1.z.object({
    nodesFound: zod_1.z
        .array(zod_1.z.object({
        nodeName: zod_1.z.string().describe('The internal name of the node (e.g., n8n-nodes-base.gmail)'),
        version: zod_1.z
            .number()
            .describe('The version number of the node (e.g., 1, 1.1, 2, 3, 3.2, etc.)'),
        reasoning: zod_1.z.string().describe('Why this node is relevant for the workflow'),
        connectionChangingParameters: zod_1.z
            .array(zod_1.z.object({
            name: zod_1.z
                .string()
                .describe('Parameter name (e.g., "mode", "operation", "hasOutputParser")'),
            possibleValues: zod_1.z
                .array(zod_1.z.union([zod_1.z.string(), zod_1.z.boolean(), zod_1.z.number()]))
                .describe('Possible values this parameter can take'),
        }))
            .describe('Parameters that affect node connections (inputs/outputs). ONLY include if parameter appears in <input> or <output> expressions'),
    }))
        .describe('List of n8n nodes identified as necessary for the workflow'),
});
exports.DiscoverySubgraphState = langgraph_1.Annotation.Root({
    userRequest: (0, langgraph_1.Annotation)({
        reducer: (x, y) => y ?? x,
        default: () => '',
    }),
    workflowJSON: (0, langgraph_1.Annotation)({
        reducer: (x, y) => y ?? x,
        default: () => ({ nodes: [], connections: {}, name: '' }),
    }),
    mode: (0, langgraph_1.Annotation)({
        reducer: (x, y) => y ?? x,
        default: () => 'build',
    }),
    planOutput: (0, langgraph_1.Annotation)({
        reducer: (x, y) => (y === undefined ? x : y),
        default: () => null,
    }),
    planDecision: (0, langgraph_1.Annotation)({
        reducer: (x, y) => (y === undefined ? x : y),
        default: () => null,
    }),
    planFeedback: (0, langgraph_1.Annotation)({
        reducer: (x, y) => (y === undefined ? x : y),
        default: () => null,
    }),
    planPrevious: (0, langgraph_1.Annotation)({
        reducer: (x, y) => (y === undefined ? x : y),
        default: () => null,
    }),
    planModifyCount: (0, langgraph_1.Annotation)({
        reducer: (x, y) => y ?? x,
        default: () => 0,
    }),
    messages: (0, langgraph_1.Annotation)({
        reducer: (x, y) => x.concat(y),
        default: () => [],
    }),
    nodesFound: (0, langgraph_1.Annotation)({
        reducer: (x, y) => y ?? x,
        default: () => [],
    }),
    bestPractices: (0, langgraph_1.Annotation)({
        reducer: (x, y) => y ?? x,
    }),
    templateIds: (0, langgraph_1.Annotation)({
        reducer: state_reducers_1.appendArrayReducer,
        default: () => [],
    }),
    cachedTemplates: (0, langgraph_1.Annotation)({
        reducer: state_reducers_1.cachedTemplatesReducer,
        default: () => [],
    }),
    resourceOperationCache: (0, langgraph_1.Annotation)({
        reducer: (x, y) => ({ ...x, ...y }),
        default: () => ({}),
    }),
    selectedNodesContext: (0, langgraph_1.Annotation)({
        reducer: (x, y) => y ?? x,
        default: () => '',
    }),
    toolCallRetryCount: (0, langgraph_1.Annotation)({
        reducer: (x, y) => y ?? x,
        default: () => 0,
    }),
});
class DiscoverySubgraph extends subgraph_interface_1.BaseSubgraph {
    name = 'discovery_subgraph';
    description = 'Discovers nodes and context for the workflow';
    agent;
    plannerAgent;
    toolMap;
    logger;
    parsedNodeTypes;
    featureFlags;
    create(config) {
        this.logger = config.logger;
        this.parsedNodeTypes = config.parsedNodeTypes;
        this.featureFlags = config.featureFlags;
        const includeExamples = config.featureFlags?.templateExamples === true;
        const includePlanMode = config.featureFlags?.planMode === true;
        const enableIntrospection = config.featureFlags?.enableIntrospection === true;
        const baseTools = includePlanMode
            ? [(0, node_search_tool_1.createNodeSearchTool)(config.parsedNodeTypes).tool, submit_questions_tool_1.submitQuestionsTool]
            : [(0, node_search_tool_1.createNodeSearchTool)(config.parsedNodeTypes).tool];
        if (enableIntrospection) {
            baseTools.push((0, introspect_tool_1.createIntrospectTool)(config.logger).tool);
        }
        const tools = includeExamples
            ? [
                ...baseTools,
                (0, get_documentation_tool_1.createGetDocumentationTool)().tool,
                (0, get_workflow_examples_tool_1.createGetWorkflowExamplesTool)(config.logger).tool,
            ]
            : baseTools;
        this.toolMap = new Map(tools.map((toolInstance) => [toolInstance.name, toolInstance]));
        const submitTool = (0, tools_1.tool)(() => { }, {
            name: 'submit_discovery_results',
            description: 'Submit the final discovery results',
            schema: discoveryOutputSchema,
        });
        const discoveryPrompt = (0, prompts_2.buildDiscoveryPrompt)({
            includeExamples,
            includeQuestions: includePlanMode,
        });
        const systemPrompt = prompts_1.ChatPromptTemplate.fromMessages([
            [
                'system',
                [
                    {
                        type: 'text',
                        text: discoveryPrompt,
                        cache_control: { type: 'ephemeral' },
                    },
                ],
            ],
            ['human', '{prompt}'],
            ['placeholder', '{messages}'],
        ]);
        if (typeof config.llm.bindTools !== 'function') {
            throw new errors_1.LLMServiceError('LLM does not support tools', {
                llmModel: config.llm._llmType(),
            });
        }
        const allTools = [...tools, submitTool];
        this.agent = systemPrompt.pipe(config.llm.bindTools(allTools));
        this.plannerAgent = (0, planner_agent_1.createPlannerAgent)({
            llm: config.plannerLLM,
            tools: [(0, get_documentation_tool_1.createGetDocumentationTool)().tool],
        });
        const subgraph = new langgraph_1.StateGraph(exports.DiscoverySubgraphState)
            .addNode('discovery_agent', this.callAgent.bind(this))
            .addNode('tools', async (state) => await (0, subgraph_helpers_1.executeSubgraphTools)(state, this.toolMap))
            .addNode('format_output', this.formatOutput.bind(this))
            .addNode('reprompt', this.repromptForToolCall.bind(this))
            .addNode('planner', this.plannerNode.bind(this))
            .addEdge(langgraph_1.START, 'discovery_agent')
            .addConditionalEdges('discovery_agent', this.shouldContinue.bind(this), {
            tools: 'tools',
            format_output: 'format_output',
            reprompt: 'reprompt',
            end: langgraph_1.END,
        })
            .addEdge('tools', 'discovery_agent')
            .addEdge('reprompt', 'discovery_agent')
            .addConditionalEdges('format_output', this.shouldPlan.bind(this))
            .addConditionalEdges('planner', this.shouldLoopPlanner.bind(this));
        return subgraph.compile(config.checkpointer ? { checkpointer: config.checkpointer } : undefined);
    }
    async callAgent(state) {
        if (state.messages.length > 0) {
            (0, cache_control_1.applySubgraphCacheMarkers)(state.messages);
        }
        const response = (await this.agent.invoke({
            messages: state.messages,
            prompt: state.userRequest,
        }));
        return { messages: [response] };
    }
    async plannerNode(state, runnableConfig) {
        if (!this.featureFlags?.planMode || state.mode !== 'plan' || state.planOutput) {
            return {};
        }
        const result = await (0, planner_agent_1.invokePlannerNode)(this.plannerAgent, {
            userRequest: state.userRequest || 'Build a workflow',
            discoveryContext: {
                nodesFound: state.nodesFound ?? [],
                bestPractices: state.bestPractices,
            },
            workflowJSON: state.workflowJSON,
            planPrevious: state.planPrevious,
            planFeedback: state.planFeedback,
            selectedNodesContext: state.selectedNodesContext,
        }, runnableConfig);
        if (result.planDecision === 'modify') {
            return { ...result, planModifyCount: state.planModifyCount + 1 };
        }
        return result;
    }
    shouldPlan(state) {
        if (!this.featureFlags?.planMode)
            return langgraph_1.END;
        if (state.mode !== 'plan')
            return langgraph_1.END;
        return state.planOutput ? langgraph_1.END : 'planner';
    }
    static MAX_PLAN_MODIFY_ITERATIONS = 5;
    shouldLoopPlanner(state) {
        if (state.planDecision !== 'modify')
            return langgraph_1.END;
        if (state.planModifyCount >= DiscoverySubgraph.MAX_PLAN_MODIFY_ITERATIONS) {
            this.logger?.warn(`[Discovery] Plan modify limit reached (${DiscoverySubgraph.MAX_PLAN_MODIFY_ITERATIONS}), proceeding with last plan`);
            return langgraph_1.END;
        }
        return 'discovery_agent';
    }
    BASELINE_NODES = [
        { name: 'n8n-nodes-base.aggregate', reasoning: 'Combines multiple items into a single item' },
        {
            name: 'n8n-nodes-base.if',
            reasoning: 'Routes items to different output paths based on true/false condition evaluation',
        },
        {
            name: 'n8n-nodes-base.switch',
            reasoning: 'Routes items to different output paths based on rules or expression evaluation',
        },
        {
            name: 'n8n-nodes-base.splitOut',
            reasoning: 'Converts a single item containing an array field into multiple separate items',
        },
        {
            name: 'n8n-nodes-base.merge',
            reasoning: 'Combines data from multiple parallel input branches into a single output',
        },
        {
            name: 'n8n-nodes-base.set',
            reasoning: 'Transforms data by adding, modifying, or removing fields from items',
        },
    ];
    formatOutput(state) {
        const lastMessage = state.messages.at(-1);
        let output;
        let submitToolCallId;
        if (lastMessage && (0, messages_1.isAIMessage)(lastMessage) && lastMessage.tool_calls) {
            const submitCall = lastMessage.tool_calls.find((tc) => tc.name === 'submit_discovery_results');
            if (submitCall) {
                submitToolCallId = submitCall.id;
                const parseResult = discoveryOutputSchema.safeParse(submitCall.args);
                if (!parseResult.success) {
                    this.logger?.error('[Discovery] Invalid discovery output schema - returning empty results', {
                        errors: parseResult.error.errors,
                        lastMessageContent: typeof lastMessage?.content === 'string'
                            ? lastMessage.content.substring(0, 200)
                            : JSON.stringify(lastMessage?.content)?.substring(0, 200),
                    });
                    return {
                        nodesFound: [],
                        templateIds: [],
                    };
                }
                output = parseResult.data;
            }
        }
        if (!output) {
            this.logger?.error('[Discovery] No submit_discovery_results tool call found - agent may have stopped early', {
                messageCount: state.messages.length,
                lastMessageType: lastMessage?.getType(),
            });
            return {
                nodesFound: [],
                templateIds: [],
            };
        }
        const latestVersionMap = new Map();
        for (const nt of this.parsedNodeTypes) {
            const ver = Array.isArray(nt.version) ? Math.max(...nt.version) : nt.version;
            const existing = latestVersionMap.get(nt.name);
            if (existing === undefined || ver > existing) {
                latestVersionMap.set(nt.name, ver);
            }
        }
        for (const node of output.nodesFound) {
            const latest = latestVersionMap.get(node.nodeName);
            if (latest !== undefined) {
                node.version = latest;
            }
        }
        const discoveredNames = new Set(output.nodesFound.map((node) => node.nodeName));
        const baselineNodesToAdd = this.BASELINE_NODES.filter((baselineNode) => !discoveredNames.has(baselineNode.name));
        for (const baselineNode of baselineNodesToAdd) {
            const version = latestVersionMap.get(baselineNode.name);
            if (version !== undefined) {
                output.nodesFound.push({
                    nodeName: baselineNode.name,
                    version,
                    reasoning: baselineNode.reasoning,
                    connectionChangingParameters: [],
                });
            }
        }
        const nodeTypeMap = new Map();
        for (const nt of this.parsedNodeTypes) {
            const versions = Array.isArray(nt.version) ? nt.version : [nt.version];
            for (const v of versions) {
                nodeTypeMap.set(`${nt.name}:${v}`, nt);
            }
        }
        const existingCache = state.resourceOperationCache ?? {};
        const hydratedNodesFound = output.nodesFound.map((node) => {
            const cacheKey = (0, resource_operation_extractor_1.createResourceCacheKey)(node.nodeName, node.version);
            if (cacheKey in existingCache) {
                const cached = existingCache[cacheKey];
                if (cached) {
                    return {
                        ...node,
                        availableResources: cached.resources,
                    };
                }
                return node;
            }
            const nodeType = nodeTypeMap.get(cacheKey);
            if (!nodeType) {
                this.logger?.warn(`[Discovery] Node type not found during resource hydration ${node.nodeName}:${node.version}`, {
                    nodeName: node.nodeName,
                    nodeVersion: node.version,
                });
                return node;
            }
            const resourceOpInfo = (0, resource_operation_extractor_1.extractResourceOperations)(nodeType, node.version, this.logger);
            if (!resourceOpInfo) {
                return node;
            }
            return {
                ...node,
                availableResources: resourceOpInfo.resources,
            };
        });
        const toolResponseMessages = submitToolCallId
            ? [
                new messages_1.ToolMessage({
                    content: `Discovery complete: found ${hydratedNodesFound.length} nodes.`,
                    tool_call_id: submitToolCallId,
                }),
            ]
            : [];
        return {
            nodesFound: hydratedNodesFound,
            bestPractices: state.bestPractices,
            templateIds: state.templateIds ?? [],
            messages: toolResponseMessages,
        };
    }
    shouldContinue(state) {
        const lastMessage = state.messages[state.messages.length - 1];
        if (lastMessage &&
            (0, messages_1.isAIMessage)(lastMessage) &&
            lastMessage.tool_calls &&
            lastMessage.tool_calls.length > 0) {
            const submitCall = lastMessage.tool_calls.find((tc) => tc.name === 'submit_discovery_results');
            if (submitCall) {
                return 'format_output';
            }
            return 'tools';
        }
        const MAX_TOOL_CALL_RETRIES = 1;
        if (state.toolCallRetryCount < MAX_TOOL_CALL_RETRIES) {
            this.logger?.warn('[Discovery] Agent stopped without tool calls - will reprompt to use submit_discovery_results tool', {
                retryCount: state.toolCallRetryCount,
                lastMessageContent: typeof lastMessage?.content === 'string'
                    ? lastMessage.content.substring(0, 200)
                    : undefined,
            });
            return 'reprompt';
        }
        this.logger?.error('[Discovery] Agent failed to use tool calls after retry - check if LLM is producing valid tool calls', {
            retryCount: state.toolCallRetryCount,
        });
        return 'end';
    }
    repromptForToolCall(state) {
        const repromptMessage = new messages_1.HumanMessage({
            content: 'You must use the submit_discovery_results tool to submit your results. Do not output the results as text or XML - use the actual tool call. The downstream system can only process results submitted via the tool calling API, not text output. Please call the submit_discovery_results tool now with your nodesFound array.',
        });
        return {
            messages: [repromptMessage],
            toolCallRetryCount: state.toolCallRetryCount + 1,
        };
    }
    transformInput(parentState) {
        const userRequest = (0, subgraph_helpers_1.extractUserRequest)(parentState.messages, 'Build a workflow');
        const contextParts = [];
        contextParts.push('<user_request>');
        contextParts.push(userRequest);
        contextParts.push('</user_request>');
        const selectedNodesSummary = (0, context_builders_1.buildSelectedNodesSummary)(parentState.workflowContext);
        if (selectedNodesSummary) {
            contextParts.push('=== SELECTED NODES ===');
            contextParts.push('<selected_nodes>');
            contextParts.push(selectedNodesSummary);
            contextParts.push('When user says "add X before/after this", find nodes that work well with the selected node(s).');
            contextParts.push('</selected_nodes>');
        }
        if (parentState.workflowJSON.nodes.length > 0) {
            contextParts.push('<existing_workflow_summary>');
            contextParts.push((0, context_builders_1.buildWorkflowSummary)(parentState.workflowJSON));
            contextParts.push('</existing_workflow_summary>');
        }
        const contextMessage = (0, context_builders_1.createContextMessage)(contextParts);
        return {
            userRequest,
            workflowJSON: parentState.workflowJSON,
            mode: parentState.mode,
            planOutput: parentState.planOutput,
            planDecision: null,
            planFeedback: parentState.planFeedback ?? null,
            planPrevious: parentState.planPrevious ?? null,
            selectedNodesContext: selectedNodesSummary ?? '',
            messages: [contextMessage],
            cachedTemplates: parentState.cachedTemplates,
        };
    }
    transformOutput(subgraphOutput, _parentState) {
        const nodesFound = subgraphOutput.nodesFound || [];
        const templateIds = subgraphOutput.templateIds || [];
        const discoveryContext = {
            nodesFound,
            bestPractices: subgraphOutput.bestPractices,
        };
        const logEntry = {
            phase: 'discovery',
            status: 'completed',
            timestamp: Date.now(),
            summary: `Discovered ${nodesFound.length} nodes`,
            metadata: (0, coordination_1.createDiscoveryMetadata)({
                nodesFound: nodesFound.length,
                nodeTypes: nodesFound.map((node) => node.nodeName),
                hasBestPractices: !!subgraphOutput.bestPractices,
            }),
        };
        const toolMessages = (0, subgraph_helpers_1.extractToolMessagesForPersistence)(subgraphOutput.messages);
        const cappedModify = subgraphOutput.planDecision === 'modify' &&
            subgraphOutput.planModifyCount >= DiscoverySubgraph.MAX_PLAN_MODIFY_ITERATIONS;
        const introspectionEvents = (0, introspect_tool_1.extractIntrospectionEventsFromMessages)(subgraphOutput.messages);
        return {
            discoveryContext,
            coordinationLog: [logEntry],
            templateIds,
            cachedTemplates: subgraphOutput.cachedTemplates,
            planOutput: subgraphOutput.planOutput,
            planDecision: cappedModify ? null : subgraphOutput.planDecision,
            planFeedback: subgraphOutput.planFeedback,
            planPrevious: subgraphOutput.planPrevious,
            ...(subgraphOutput.mode ? { mode: subgraphOutput.mode } : {}),
            introspectionEvents,
            messages: toolMessages,
        };
    }
}
exports.DiscoverySubgraph = DiscoverySubgraph;
