// @n8n/db/dist/services/auth.roles.service.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthRolesService = void 0;
const backend_common_1 = require("@n8n/backend-common");
const di_1 = require("@n8n/di");
const permissions_1 = require("@n8n/permissions");
const typeorm_1 = require("@n8n/typeorm");
const entities_1 = require("../entities");
const db_lock_service_1 = require("./db-lock.service");
let AuthRolesService = class AuthRolesService {
    constructor(logger, dbLockService) {
        this.logger = logger;
        this.dbLockService = dbLockService;
    }
    async syncScopes(tx) {
        const scopeRepo = tx.getRepository(entities_1.Scope);
        const roleRepo = tx.getRepository(entities_1.Role);
        const availableScopes = await scopeRepo.find({
            select: {
                slug: true,
                displayName: true,
                description: true,
            },
        });
        const availableScopesMap = new Map(availableScopes.map((scope) => [scope.slug, scope]));
        const scopesToUpdate = permissions_1.ALL_SCOPES.map((slug) => {
            const info = permissions_1.scopeInformation[slug] ?? {
                displayName: slug,
                description: null,
            };
            const existingScope = availableScopesMap.get(slug);
            if (!existingScope) {
                const newScope = new entities_1.Scope();
                newScope.slug = slug;
                newScope.displayName = info.displayName;
                newScope.description = info.description ?? null;
                return newScope;
            }
            const needsUpdate = existingScope.displayName !== info.displayName ||
                existingScope.description !== info.description;
            if (needsUpdate) {
                existingScope.displayName = info.displayName;
                existingScope.description = info.description ?? null;
                return existingScope;
            }
            return null;
        }).filter((scope) => scope !== null);
        if (scopesToUpdate.length > 0) {
            this.logger.debug(`Updating ${scopesToUpdate.length} scopes...`);
            await scopeRepo.save(scopesToUpdate);
            this.logger.debug('Scopes updated successfully.');
        }
        else {
            this.logger.debug('No scopes to update.');
        }
        const scopesToDelete = availableScopes.filter((scope) => !permissions_1.ALL_SCOPES.includes(scope.slug));
        if (scopesToDelete.length > 0) {
            this.logger.debug(`Deleting ${scopesToDelete.length} obsolete scopes: ${scopesToDelete.map((s) => s.slug).join(', ')}`);
            const obsoleteScopeSlugs = scopesToDelete.map((s) => s.slug);
            const rolesWithObsoleteScopes = await roleRepo.find({
                relations: ['scopes'],
                where: { scopes: { slug: (0, typeorm_1.In)(obsoleteScopeSlugs) } },
            });
            const rolesToUpdate = rolesWithObsoleteScopes.map((role) => {
                role.scopes = role.scopes.filter((scope) => !obsoleteScopeSlugs.includes(scope.slug));
                return role;
            });
            if (rolesToUpdate.length > 0) {
                this.logger.debug(`Removing obsolete scopes from ${rolesToUpdate.length} roles...`);
                await roleRepo.save(rolesToUpdate);
            }
            await scopeRepo.remove(scopesToDelete);
            this.logger.debug('Obsolete scopes deleted successfully.');
        }
        else {
            this.logger.debug('No obsolete scopes to delete.');
        }
    }
    async getPersonalOwnerSettingsScopes(tx) {
        const settingKeys = [permissions_1.PERSONAL_SPACE_PUBLISHING_SETTING.key, permissions_1.PERSONAL_SPACE_SHARING_SETTING.key];
        const rows = await tx.findBy(entities_1.Settings, { key: (0, typeorm_1.In)(settingKeys) });
        const personalSpacePublishingValue = rows.find((r) => r.key === permissions_1.PERSONAL_SPACE_PUBLISHING_SETTING.key)?.value;
        const personalSpaceSharingValue = rows.find((r) => r.key === permissions_1.PERSONAL_SPACE_SHARING_SETTING.key)?.value;
        const scopes = [];
        if (personalSpacePublishingValue === 'true' || personalSpacePublishingValue === undefined) {
            scopes.push(...permissions_1.PERSONAL_SPACE_PUBLISHING_SETTING.scopes);
            this.logger.debug(`${permissions_1.PERSONAL_SPACE_PUBLISHING_SETTING.key} is enabled - allowing ${permissions_1.PERSONAL_SPACE_PUBLISHING_SETTING.scopes.join(', ')} scopes to ${permissions_1.PROJECT_OWNER_ROLE_SLUG} role`);
        }
        if (personalSpaceSharingValue === 'true' || personalSpaceSharingValue === undefined) {
            scopes.push(...permissions_1.PERSONAL_SPACE_SHARING_SETTING.scopes);
            this.logger.debug(`${permissions_1.PERSONAL_SPACE_SHARING_SETTING.key} is enabled - allowing ${permissions_1.PERSONAL_SPACE_SHARING_SETTING.scopes.join(', ')} scopes to ${permissions_1.PROJECT_OWNER_ROLE_SLUG} role`);
        }
        return scopes;
    }
    async updateScopesBasedOnSettings(roleSlug, defaultScopes, tx) {
        const scopes = [...defaultScopes];
        if (roleSlug === permissions_1.PROJECT_OWNER_ROLE_SLUG) {
            scopes.push(...(await this.getPersonalOwnerSettingsScopes(tx)));
        }
        return scopes;
    }
    async syncRoles(tx) {
        const roleRepo = tx.getRepository(entities_1.Role);
        const scopeRepo = tx.getRepository(entities_1.Scope);
        const existingRoles = await roleRepo.find({
            select: {
                slug: true,
                displayName: true,
                description: true,
                systemRole: true,
                roleType: true,
            },
            where: {
                systemRole: true,
            },
        });
        const allScopes = await scopeRepo.find({
            select: {
                slug: true,
            },
        });
        const existingRolesMap = new Map(existingRoles.map((role) => [role.slug, role]));
        for (const roleNamespace of Object.keys(permissions_1.ALL_ROLES)) {
            const rolesToUpdate = await Promise.all(permissions_1.ALL_ROLES[roleNamespace].map(async (role) => {
                const existingRole = existingRolesMap.get(role.slug);
                const expectedScopes = await this.updateScopesBasedOnSettings(role.slug, role.scopes, tx);
                if (!existingRole) {
                    const newRole = roleRepo.create({
                        slug: role.slug,
                        displayName: role.displayName,
                        description: role.description ?? null,
                        roleType: roleNamespace,
                        systemRole: true,
                        scopes: allScopes.filter((scope) => expectedScopes.includes(scope.slug)),
                    });
                    return newRole;
                }
                const needsUpdate = existingRole.displayName !== role.displayName ||
                    existingRole.description !== role.description ||
                    existingRole.roleType !== roleNamespace ||
                    existingRole.scopes.some((scope) => !expectedScopes.includes(scope.slug)) ||
                    expectedScopes.some((scope) => !existingRole.scopes.some((s) => s.slug === scope));
                if (needsUpdate) {
                    existingRole.displayName = role.displayName;
                    existingRole.description = role.description ?? null;
                    existingRole.roleType = roleNamespace;
                    existingRole.scopes = allScopes.filter((scope) => expectedScopes.includes(scope.slug));
                    return existingRole;
                }
                return null;
            }));
            const filteredRolesToUpdate = rolesToUpdate.filter((role) => role !== null);
            if (filteredRolesToUpdate.length > 0) {
                this.logger.debug(`Updating ${filteredRolesToUpdate.length} ${roleNamespace} roles...`);
                await roleRepo.save(filteredRolesToUpdate);
                this.logger.debug(`${roleNamespace} roles updated successfully.`);
            }
            else {
                this.logger.debug(`No ${roleNamespace} roles to update.`);
            }
        }
    }
    async init() {
        this.logger.debug('Initializing AuthRolesService...');
        await this.dbLockService.withLock(1001, async (tx) => {
            await this.syncScopes(tx);
            await this.syncRoles(tx);
        });
        this.logger.debug('AuthRolesService initialized successfully.');
    }
};
exports.AuthRolesService = AuthRolesService;
exports.AuthRolesService = AuthRolesService = __decorate([
    (0, di_1.Service)(),
    __metadata("design:paramtypes", [backend_common_1.Logger,
        db_lock_service_1.DbLockService])
], AuthRolesService);
