// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/dataTable/v1/resource_row/operation_get.ts
/**
 * Data table Node - Version 1
 * Discriminator: resource=row, operation=get
 */


/** Get row(s) */
export type DataTableV1RowGetParams = {
  resource: 'row';
  operation: 'get';
/**
 * Data table
 * @builderHint Default to mode: 'list' which is easier for users to set up
 * @default {"mode":"list","value":""}
 */
    dataTableId?: { __rl: true; mode: 'list' | 'name' | 'id'; value: string; cachedResultName?: string };
/**
 * Must Match
 * @default anyCondition
 */
    matchType?: 'anyCondition' | 'allConditions' | Expression<string>;
/**
 * Filter to decide which rows get
 * @default {}
 */
    filters?: {
        /** Conditions
     */
    conditions?: Array<{
      /** Choose from the list, or specify using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
       * @default id
       */
      keyName?: string | Expression<string>;
      /** Condition
       * @default eq
       */
      condition?: string | Expression<string>;
      /** Value
       * @displayOptions.hide { condition: ["isEmpty", "isNotEmpty", "isTrue", "isFalse"] }
       */
      keyValue?: string | Expression<string> | PlaceholderValue;
    }>;
  };
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Whether to sort the results by a column
 * @default false
 */
    orderBy?: boolean | Expression<boolean>;
/**
 * Choose from the list, or specify using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 * @displayOptions.show { orderBy: [true] }
 * @default createdAt
 */
    orderByColumn?: string | Expression<string>;
/**
 * Sort direction for the column
 * @displayOptions.show { orderBy: [true] }
 * @default DESC
 */
    orderByDirection?: 'ASC' | 'DESC' | Expression<string>;
};

export type DataTableV1RowGetOutput = {
  createdAt?: string;
  id?: number;
  updatedAt?: string;
};

export type DataTableV1RowGetNode = {
  type: 'n8n-nodes-base.dataTable';
  version: 1;
  config: NodeConfig<DataTableV1RowGetParams>;
  output?: Items<DataTableV1RowGetOutput>;
};
