// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/dataTable/v1/resource_row/operation_get.schema.js
/**
 * Data table Node - Version 1 - Zod Schema
 * Discriminator: resource=row, operation=get
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('row').default('row'),
      operation: z.literal('get'),
      dataTableId: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('name'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      matchType: z.union([z.literal('anyCondition'), z.literal('allConditions'), expressionSchema]).optional(),
      filters: z.object({ conditions: z.array(z.object({ keyName: stringOrExpression.optional(), condition: stringOrExpression.optional(), keyValue: stringOrExpression.optional() })).optional() }).optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      orderBy: booleanOrExpression.optional(),
      orderByColumn: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"orderBy":[true]}}, defaults: {"orderBy":false} }),
      orderByDirection: resolveSchema({ parameters, schema: z.union([z.literal('ASC'), z.literal('DESC'), expressionSchema]), required: false, displayOptions: {"show":{"orderBy":[true]}}, defaults: {"orderBy":false} }),
    }).optional(),
  });
};
