// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/harvest/v1/resource_client/operation_get_all.ts
/**
 * Harvest Node - Version 1
 * Discriminator: resource=client, operation=getAll
 */


interface Credentials {
  harvestApi: CredentialReference;
  harvestOAuth2Api: CredentialReference;
}

/** Get data of many clients */
export type HarvestV1ClientGetAllParams = {
  resource: 'client';
  operation: 'getAll';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    accountId?: string | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Whether to only return active clients and false to return inactive clients
     * @default true
     */
    is_active?: boolean | Expression<boolean>;
    /** Only return clients that have been updated since the given date and time
     */
    updated_since?: string | Expression<string>;
  };
};

export type HarvestV1ClientGetAllOutput = {
  created_at?: string;
  currency?: string;
  id?: number;
  is_active?: boolean;
  name?: string;
  statement_key?: string;
  updated_at?: string;
};

export type HarvestV1ClientGetAllNode = {
  type: 'n8n-nodes-base.harvest';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<HarvestV1ClientGetAllParams>;
  output?: Items<HarvestV1ClientGetAllOutput>;
};
