// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mongoDb/v11/resource_search_indexes/operation_create_search_index.schema.js
/**
 * MongoDB Node - Version 1.1 - Zod Schema
 * Discriminator: resource=searchIndexes, operation=createSearchIndex
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('searchIndexes'),
      operation: z.literal('createSearchIndex'),
      collection: stringOrExpression.optional(),
      indexNameRequired: stringOrExpression.optional(),
      indexDefinition: z.union([iDataObjectSchema, z.string()]).optional(),
      indexType: z.union([z.literal('vectorSearch'), z.literal('search'), expressionSchema]).optional(),
    }).optional(),
  });
};
