// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/taiga/v1/resource_task/operation_create.schema.js
/**
 * Taiga Node - Version 1 - Zod Schema
 * Discriminator: resource=task, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('task'),
      operation: z.literal('create').default('create'),
      projectId: stringOrExpression.optional(),
      subject: stringOrExpression.optional(),
      additionalFields: z.object({ assigned_to: stringOrExpression.optional(), blocked_note: stringOrExpression.optional(), description: stringOrExpression.optional(), is_blocked: booleanOrExpression.optional(), milestone: stringOrExpression.optional(), status: stringOrExpression.optional(), tags: z.array(z.string()).optional(), taskboard_order: numberOrExpression.optional(), user_story: stringOrExpression.optional(), us_order: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
