// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/taiga/v1/resource_epic/operation_update.schema.js
/**
 * Taiga Node - Version 1 - Zod Schema
 * Discriminator: resource=epic, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('epic'),
      operation: z.literal('update'),
      projectId: stringOrExpression.optional(),
      epicId: stringOrExpression.optional(),
      updateFields: z.object({ assigned_to: stringOrExpression.optional(), blocked_note: stringOrExpression.optional(), color: stringOrExpression.optional(), description: stringOrExpression.optional(), is_blocked: booleanOrExpression.optional(), subject: stringOrExpression.optional(), tags: z.array(z.string()).optional() }).optional(),
    }).optional(),
  });
};
