// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/taiga/v1/resource_epic/operation_get_all.ts
/**
 * Taiga Node - Version 1
 * Discriminator: resource=epic, operation=getAll
 */


interface Credentials {
  taigaApi: CredentialReference;
}

/** Get many epics */
export type TaigaV1EpicGetAllParams = {
  resource: 'epic';
  operation: 'getAll';
/**
 * ID of the project to which the epic belongs. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    projectId?: string | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** ID of the user whom the epic is assigned to. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    assigned_to?: string | Expression<string>;
    /** Whether the epic is closed
     * @default false
     */
    statusIsClosed?: boolean | Expression<boolean>;
  };
};

export type TaigaV1EpicGetAllNode = {
  type: 'n8n-nodes-base.taiga';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<TaigaV1EpicGetAllParams>;
};
