// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/slack/v1/resource_channel/operation_replies.ts
/**
 * Slack Node - Version 1
 * Discriminator: resource=channel, operation=replies
 */


interface Credentials {
  slackApi: CredentialReference;
  slackOAuth2Api: CredentialReference;
}

/** Get a thread of messages posted to a channel */
export type SlackV1ChannelRepliesParams = {
  resource: 'channel';
  operation: 'replies';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * The name of the channel to create. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    channelId?: string | Expression<string>;
/**
 * Unique identifier of a thread's parent message
 */
    ts?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Whether to include messages with latest or oldest timestamp in results only when either timestamp is specified
     * @default false
     */
    inclusive?: boolean | Expression<boolean>;
    /** End of time range of messages to include in results
     */
    latest?: string | Expression<string> | PlaceholderValue;
    /** Start of time range of messages to include in results
     */
    oldest?: string | Expression<string> | PlaceholderValue;
  };
};

export type SlackV1ChannelRepliesNode = {
  type: 'n8n-nodes-base.slack';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SlackV1ChannelRepliesParams>;
};
