// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/slack/v1/resource_channel/operation_get_all.ts
/**
 * Slack Node - Version 1
 * Discriminator: resource=channel, operation=getAll
 */


interface Credentials {
  slackApi: CredentialReference;
  slackOAuth2Api: CredentialReference;
}

/** Get many channels in a Slack team */
export type SlackV1ChannelGetAllParams = {
  resource: 'channel';
  operation: 'getAll';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Whether to exclude archived channels from the list
     * @default false
     */
    excludeArchived?: boolean | Expression<boolean>;
    /** Mix and match channel types
     * @default ["public_channel"]
     */
    types?: Array<'public_channel' | 'private_channel' | 'mpim' | 'im'>;
  };
};

export type SlackV1ChannelGetAllNode = {
  type: 'n8n-nodes-base.slack';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SlackV1ChannelGetAllParams>;
};
