// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/slack/v1/resource_channel/operation_member.ts
/**
 * Slack Node - Version 1
 * Discriminator: resource=channel, operation=member
 */


interface Credentials {
  slackApi: CredentialReference;
  slackOAuth2Api: CredentialReference;
}

/** List members of a conversation */
export type SlackV1ChannelMemberParams = {
  resource: 'channel';
  operation: 'member';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    channelId?: string | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Whether to resolve the data automatically. By default the response only contain the ID to resource.
 * @default false
 */
    resolveData?: boolean | Expression<boolean>;
};

export type SlackV1ChannelMemberNode = {
  type: 'n8n-nodes-base.slack';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SlackV1ChannelMemberParams>;
};
