// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/wise/v1/resource_exchange_rate/operation_get.ts
/**
 * Wise Node - Version 1
 * Discriminator: resource=exchangeRate, operation=get
 */


interface Credentials {
  wiseApi: CredentialReference;
}

export type WiseV1ExchangeRateGetParams = {
  resource: 'exchangeRate';
  operation: 'get';
/**
 * Code of the source currency to retrieve the exchange rate for
 */
    source?: string | Expression<string> | PlaceholderValue;
/**
 * Code of the target currency to retrieve the exchange rate for
 */
    target?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Interval
     * @default day
     */
    interval?: 'day' | 'hour' | 'minute' | Expression<string>;
    /** Range of time to retrieve the exchange rate for
     * @default {}
     */
    range?: {
        /** Range Properties
     */
    rangeProperties?: {
      /** Range Start
       */
      from?: string | Expression<string>;
      /** Range End
       */
      to?: string | Expression<string>;
    };
  };
    /** Point in time to retrieve the exchange rate for
     */
    time?: string | Expression<string>;
  };
};

export type WiseV1ExchangeRateGetOutput = {
  rate?: number;
  source?: string;
  target?: string;
  time?: string;
};

export type WiseV1ExchangeRateGetNode = {
  type: 'n8n-nodes-base.wise';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<WiseV1ExchangeRateGetParams>;
  output?: Items<WiseV1ExchangeRateGetOutput>;
};
