// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/wise/v1/resource_exchange_rate/operation_get.schema.js
/**
 * Wise Node - Version 1 - Zod Schema
 * Discriminator: resource=exchangeRate, operation=get
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('exchangeRate'),
      operation: z.literal('get'),
      source: stringOrExpression.optional(),
      target: stringOrExpression.optional(),
      additionalFields: z.object({ interval: z.union([z.literal('day'), z.literal('hour'), z.literal('minute'), expressionSchema]).optional(), range: z.unknown().optional(), time: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
