// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/sendy/v1/resource_campaign/operation_create.schema.js
/**
 * Sendy Node - Version 1 - Zod Schema
 * Discriminator: resource=campaign, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('campaign'),
      operation: z.literal('create').default('create'),
      fromName: stringOrExpression.optional(),
      fromEmail: stringOrExpression.optional(),
      replyTo: stringOrExpression.optional(),
      title: stringOrExpression.optional(),
      subject: stringOrExpression.optional(),
      htmlText: stringOrExpression.optional(),
      sendCampaign: booleanOrExpression.optional(),
      brandId: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"sendCampaign":[false]}}, defaults: {"sendCampaign":false} }),
      additionalFields: z.object({ excludeListIds: stringOrExpression.optional(), excludeSegmentIds: stringOrExpression.optional(), listIds: stringOrExpression.optional(), plainText: stringOrExpression.optional(), queryString: stringOrExpression.optional(), segmentIds: stringOrExpression.optional(), trackClicks: booleanOrExpression.optional(), trackOpens: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
