// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftOutlook/v1/resource_message_attachment/operation_add.ts
/**
 * Microsoft Outlook Node - Version 1
 * Discriminator: resource=messageAttachment, operation=add
 */


interface Credentials {
  microsoftOutlookOAuth2Api: CredentialReference;
}

/** Add an attachment to a message */
export type MicrosoftOutlookV1MessageAttachmentAddParams = {
  resource: 'messageAttachment';
  operation: 'add';
/**
 * Message ID
 */
    messageId?: string | Expression<string> | PlaceholderValue;
/**
 * Put Output File in Field
 * @hint The name of the output binary field to put the file in
 * @default data
 */
    binaryPropertyName?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Filename of the attachment. If not set will the file-name of the binary property be used, if it exists.
     */
    fileName?: string | Expression<string> | PlaceholderValue;
  };
};

export type MicrosoftOutlookV1MessageAttachmentAddNode = {
  type: 'n8n-nodes-base.microsoftOutlook';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MicrosoftOutlookV1MessageAttachmentAddParams>;
};
