// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/salesmate/v1/resource_activity/operation_update.schema.js
/**
 * Salesmate Node - Version 1 - Zod Schema
 * Discriminator: resource=activity, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('activity').default('activity'),
      operation: z.literal('update'),
      id: stringOrExpression.optional(),
      rawData: booleanOrExpression.optional(),
      updateFields: z.object({ title: stringOrExpression.optional(), type: stringOrExpression.optional(), owner: stringOrExpression.optional(), description: stringOrExpression.optional(), tags: stringOrExpression.optional(), dueDate: stringOrExpression.optional(), duration: numberOrExpression.optional(), isCalendarInvite: booleanOrExpression.optional(), isCompleted: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
