// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/salesmate/v1/resource_activity/operation_update.ts
/**
 * Salesmate Node - Version 1
 * Discriminator: resource=activity, operation=update
 */


interface Credentials {
  salesmateApi: CredentialReference;
}

/** Update a company */
export type SalesmateV1ActivityUpdateParams = {
  resource: 'activity';
  operation: 'update';
/**
 * Activity ID
 */
    id?: string | Expression<string> | PlaceholderValue;
/**
 * Whether the data should include the fields details
 * @default false
 */
    rawData?: boolean | Expression<boolean>;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** Title
     */
    title?: string | Expression<string> | PlaceholderValue;
    /** Type
     */
    type?: string | Expression<string> | PlaceholderValue;
    /** Owner
     */
    owner?: string | Expression<string> | PlaceholderValue;
    /** This field contains details related to the activity
     */
    description?: string | Expression<string> | PlaceholderValue;
    /** This field contains tags associated with an activity
     */
    tags?: string | Expression<string> | PlaceholderValue;
    /** Expiry date of an activity
     */
    dueDate?: string | Expression<string>;
    /** Time duration of an activity
     */
    duration?: number | Expression<number>;
    /** Whether to send calendar invite
     * @default false
     */
    isCalendarInvite?: boolean | Expression<boolean>;
    /** Whether the activity is completed or not
     * @default false
     */
    isCompleted?: boolean | Expression<boolean>;
  };
};

export type SalesmateV1ActivityUpdateNode = {
  type: 'n8n-nodes-base.salesmate';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SalesmateV1ActivityUpdateParams>;
};
