// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/salesmate/v1/resource_activity/operation_get_all.ts
/**
 * Salesmate Node - Version 1
 * Discriminator: resource=activity, operation=getAll
 */


interface Credentials {
  salesmateApi: CredentialReference;
}

/** Get many companies */
export type SalesmateV1ActivityGetAllParams = {
  resource: 'activity';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 10
 */
    limit?: number | Expression<number>;
/**
 * JSON Parameters
 * @default false
 */
    jsonParameters?: boolean | Expression<boolean>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Comma-separated list of fields to return
     */
    fields?: string | Expression<string> | PlaceholderValue;
    /** The field to sort by
     */
    sortBy?: string | Expression<string> | PlaceholderValue;
    /** Sort Order
     * @default desc
     */
    sortOrder?: 'asc' | 'desc' | Expression<string>;
  };
/**
 * Filters
 * @displayOptions.show { jsonParameters: [true] }
 */
    filtersJson?: IDataObject | string | Expression<string>;
/**
 * Filters
 * @displayOptions.show { jsonParameters: [false] }
 * @default {}
 */
    filters?: {
        /** Filters
     */
    filtersUi?: {
      /** Operator
       * @default AND
       */
      operator?: 'AND' | 'OR' | Expression<string>;
      /** Conditions
       * @default {}
       */
      conditions?: {
        /** Conditions
     */
    conditionsUi?: Array<{
      /** Field
       * @default title
       */
      field?: 'title' | 'tags' | Expression<string>;
      /** Value of the property to set
       * @default EQUALS
       */
      condition?: 'EQUALS' | 'NOT_EQUALS' | 'Contains' | 'DOES_NOT_CONTAINS' | 'EMPTY' | 'NOT_EMPTY' | 'STARTS_WITH' | 'ENDS_WITH' | Expression<string>;
      /** Value
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
    };
  };
};

export type SalesmateV1ActivityGetAllNode = {
  type: 'n8n-nodes-base.salesmate';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SalesmateV1ActivityGetAllParams>;
};
