// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/salesmate/v1/resource_activity/operation_create.ts
/**
 * Salesmate Node - Version 1
 * Discriminator: resource=activity, operation=create
 */


interface Credentials {
  salesmateApi: CredentialReference;
}

/** Create a company */
export type SalesmateV1ActivityCreateParams = {
  resource: 'activity';
  operation: 'create';
/**
 * Title
 */
    title?: string | Expression<string> | PlaceholderValue;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    owner?: string | Expression<string>;
/**
 * This field displays activity type such as call, meeting etc
 */
    type?: string | Expression<string> | PlaceholderValue;
/**
 * Whether the data should include the fields details
 * @default false
 */
    rawData?: boolean | Expression<boolean>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** This field contains details related to the activity
     */
    description?: string | Expression<string> | PlaceholderValue;
    /** This field contains tags associated with an activity
     */
    tags?: string | Expression<string> | PlaceholderValue;
    /** Expiry date of an activity
     */
    dueDate?: string | Expression<string>;
    /** Time duration of an activity
     */
    duration?: number | Expression<number>;
    /** Whether to send calendar invite
     * @default false
     */
    isCalendarInvite?: boolean | Expression<boolean>;
    /** Whether the activity is completed or not
     * @default false
     */
    isCompleted?: boolean | Expression<boolean>;
  };
};

export type SalesmateV1ActivityCreateNode = {
  type: 'n8n-nodes-base.salesmate';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SalesmateV1ActivityCreateParams>;
};
