// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/whatsAppTrigger/v1.ts
/**
 * WhatsApp Trigger Node - Version 1
 * Handle WhatsApp events via webhooks
 */


export interface WhatsAppTriggerV1Params {
  updates?: Array<'account_review_update' | 'account_update' | 'business_capability_update' | 'message_template_quality_update' | 'message_template_status_update' | 'messages' | 'phone_number_name_update' | 'phone_number_quality_update' | 'security' | 'template_category_update'>;
  options?: {
    /** WhatsApp sends notifications to the Trigger when the status of a message changes (for example from Sent to Delivered and from Delivered to Read). To avoid multiple executions for one WhatsApp message, you can set the Trigger to execute only on selected message status updates.
     * @default ["all"]
     */
    messageStatusUpdates?: Array<'all' | 'deleted' | 'delivered' | 'failed' | 'read' | 'sent'>;
  };
}

export interface WhatsAppTriggerV1Credentials {
  whatsAppTriggerApi: CredentialReference;
}

interface WhatsAppTriggerV1NodeBase {
  type: 'n8n-nodes-base.whatsAppTrigger';
  version: 1;
  credentials?: WhatsAppTriggerV1Credentials;
  isTrigger: true;
}

export type WhatsAppTriggerV1ParamsNode = WhatsAppTriggerV1NodeBase & {
  config: NodeConfig<WhatsAppTriggerV1Params>;
};

export type WhatsAppTriggerV1Node = WhatsAppTriggerV1ParamsNode;
