// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/loneScale/v1/resource_item/operation_add.schema.js
/**
 * LoneScale Node - Version 1 - Zod Schema
 * Discriminator: resource=item, operation=add
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('item'),
      operation: z.literal('add'),
      type: z.union([z.literal('COMPANY'), z.literal('PEOPLE')]).optional(),
      list: stringOrExpression.optional(),
      first_name: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"type":["PEOPLE"]}}, defaults: {"type":"PEOPLE"} }),
      last_name: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"type":["PEOPLE"]}}, defaults: {"type":"PEOPLE"} }),
      company_name: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"type":["COMPANY"]}}, defaults: {"type":"PEOPLE"} }),
      peopleAdditionalFields: resolveSchema({ parameters, schema: z.object({ full_name: stringOrExpression.optional(), email: stringOrExpression.optional(), company_name: stringOrExpression.optional(), current_position: stringOrExpression.optional(), domain: stringOrExpression.optional(), linkedin_url: stringOrExpression.optional(), location: stringOrExpression.optional(), contact_id: stringOrExpression.optional() }), required: false, displayOptions: {"show":{"type":["PEOPLE"]}}, defaults: {"type":"PEOPLE"} }),
      companyAdditionalFields: resolveSchema({ parameters, schema: z.object({ linkedin_url: stringOrExpression.optional(), domain: stringOrExpression.optional(), location: stringOrExpression.optional(), contact_id: stringOrExpression.optional() }), required: false, displayOptions: {"show":{"type":["COMPANY"]}}, defaults: {"type":"PEOPLE"} }),
    }).optional(),
  });
};
