// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/venafiTlsProtectCloud/v1/resource_certificate/operation_download.schema.js
/**
 * Venafi TLS Protect Cloud Node - Version 1 - Zod Schema
 * Discriminator: resource=certificate, operation=download
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('certificate'),
      operation: z.literal('download'),
      certificateId: stringOrExpression.optional(),
      downloadItem: z.union([z.literal('certificate'), z.literal('keystore'), expressionSchema]).optional(),
      keystoreType: resolveSchema({ parameters, schema: z.union([z.literal('JKS'), z.literal('PKCS12'), z.literal('PEM'), expressionSchema]), required: false, displayOptions: {"show":{"downloadItem":["keystore"]}}, defaults: {"downloadItem":"certificate"} }),
      certificateLabel: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"downloadItem":["keystore"]}}, defaults: {"downloadItem":"certificate"} }),
      privateKeyPassphrase: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"downloadItem":["keystore"]}}, defaults: {"downloadItem":"certificate"} }),
      keystorePassphrase: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"downloadItem":["keystore"],"keystoreType":["JKS"]}}, defaults: {"downloadItem":"certificate","keystoreType":"PEM"} }),
      binaryProperty: stringOrExpression.optional(),
      options: z.object({ chainOrder: z.union([z.literal('EE_FIRST'), z.literal('EE_ONLY'), z.literal('ROOT_FIRST'), expressionSchema]).optional(), format: z.union([z.literal('PEM'), z.literal('DER'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
