// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/splunk/v1/resource_search_configuration/operation_get_all.ts
/**
 * Splunk Node - Version 1
 * Discriminator: resource=searchConfiguration, operation=getAll
 */


interface Credentials {
  splunkApi: CredentialReference;
}

/** Retrieve many search configurations */
export type SplunkV1SearchConfigurationGetAllParams = {
  resource: 'searchConfiguration';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Whether to include a boolean value for each saved search to show whether the search is orphaned, meaning that it has no valid owner
     * @default false
     */
    add_orphan_field?: boolean | Expression<boolean>;
    /** List Default Actions
     * @default false
     */
    listDefaultActionArgs?: boolean | Expression<boolean>;
  };
};

export type SplunkV1SearchConfigurationGetAllNode = {
  type: 'n8n-nodes-base.splunk';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SplunkV1SearchConfigurationGetAllParams>;
};
