// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/serviceNow/v1/resource_attachment/operation_upload.schema.js
/**
 * ServiceNow Node - Version 1 - Zod Schema
 * Discriminator: resource=attachment, operation=upload
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('attachment'),
      operation: z.literal('upload').default('upload'),
      authentication: z.union([z.literal('basicAuth'), z.literal('oAuth2'), expressionSchema]).optional(),
      tableName: stringOrExpression.optional(),
      id: stringOrExpression.optional(),
      inputDataFieldName: stringOrExpression.optional(),
      options: z.object({ file_name: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
