// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/serviceNow/v1/resource_attachment/operation_get.schema.js
/**
 * ServiceNow Node - Version 1 - Zod Schema
 * Discriminator: resource=attachment, operation=get
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('attachment'),
      operation: z.literal('get'),
      authentication: z.union([z.literal('basicAuth'), z.literal('oAuth2'), expressionSchema]).optional(),
      attachmentId: stringOrExpression.optional(),
      download: booleanOrExpression.optional(),
      outputField: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"download":[true]}}, defaults: {"download":false} }),
      options: z.object({ queryFilter: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
