// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/serviceNow/v1/resource_attachment/operation_get_all.ts
/**
 * ServiceNow Node - Version 1
 * Discriminator: resource=attachment, operation=getAll
 */


interface Credentials {
  serviceNowOAuth2Api: CredentialReference;
  serviceNowBasicApi: CredentialReference;
}

/** Get many attachments on a table */
export type ServiceNowV1AttachmentGetAllParams = {
  resource: 'attachment';
  operation: 'getAll';
/**
 * Authentication method to use
 * @default oAuth2
 */
    authentication?: 'basicAuth' | 'oAuth2' | Expression<string>;
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    tableName?: string | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Download Attachments
 * @default false
 */
    download?: boolean | Expression<boolean>;
/**
 * Field name where downloaded data will be placed
 * @displayOptions.show { download: [true] }
 * @default data
 */
    outputField?: string | Expression<string> | PlaceholderValue;
/**
 * Options
 * @default {}
 */
    options?: {
    /** An encoded query string used to filter the results
     * @hint All parameters are case-sensitive. Queries can contain more than one entry. &lt;a href="https://developer.servicenow.com/dev.do#!/learn/learning-plans/quebec/servicenow_application_developer/app_store_learnv2_rest_quebec_more_about_query_parameters"&gt;More information&lt;/a&gt;.
     */
    queryFilter?: string | Expression<string> | PlaceholderValue;
  };
};

export type ServiceNowV1AttachmentGetAllNode = {
  type: 'n8n-nodes-base.serviceNow';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ServiceNowV1AttachmentGetAllParams>;
};
