// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/serviceNow/v1/resource_attachment/operation_get.ts
/**
 * ServiceNow Node - Version 1
 * Discriminator: resource=attachment, operation=get
 */


interface Credentials {
  serviceNowOAuth2Api: CredentialReference;
  serviceNowBasicApi: CredentialReference;
}

/** Get an attachment */
export type ServiceNowV1AttachmentGetParams = {
  resource: 'attachment';
  operation: 'get';
/**
 * Authentication method to use
 * @default oAuth2
 */
    authentication?: 'basicAuth' | 'oAuth2' | Expression<string>;
/**
 * Sys_id value of the attachment to delete
 */
    attachmentId?: string | Expression<string> | PlaceholderValue;
/**
 * Download Attachments
 * @default false
 */
    download?: boolean | Expression<boolean>;
/**
 * Field name where downloaded data will be placed
 * @displayOptions.show { download: [true] }
 * @default data
 */
    outputField?: string | Expression<string> | PlaceholderValue;
/**
 * Options
 * @default {}
 */
    options?: {
    /** An encoded query string used to filter the results
     * @hint All parameters are case-sensitive. Queries can contain more than one entry. &lt;a href="https://developer.servicenow.com/dev.do#!/learn/learning-plans/quebec/servicenow_application_developer/app_store_learnv2_rest_quebec_more_about_query_parameters"&gt;More information&lt;/a&gt;.
     */
    queryFilter?: string | Expression<string> | PlaceholderValue;
  };
};

export type ServiceNowV1AttachmentGetNode = {
  type: 'n8n-nodes-base.serviceNow';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ServiceNowV1AttachmentGetParams>;
};
