// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/odoo/v1/resource_custom/operation_get_all.schema.js
/**
 * Odoo Node - Version 1 - Zod Schema
 * Discriminator: resource=custom, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('custom'),
      operation: z.literal('getAll'),
      customResource: stringOrExpression.optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      options: z.object({ fieldsList: z.array(z.string()).optional() }).optional(),
      filterRequest: z.object({ filter: z.array(z.object({ fieldName: stringOrExpression.optional(), operator: z.union([z.literal('notEqual'), z.literal('lesserThen'), z.literal('lesserOrEqual'), z.literal('equal'), z.literal('greaterThen'), z.literal('greaterOrEqual'), z.literal('childOf'), z.literal('in'), z.literal('like'), z.literal('notIn'), expressionSchema]).optional(), value: stringOrExpression.optional() })).optional() }).optional(),
    }).optional(),
  });
};
