// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/slackTrigger/v1.schema.js
/**
 * Slack Trigger Node - Version 1 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    authentication: z.unknown().optional(),
    trigger: z.array(z.union([z.literal('any_event'), z.literal('app_mention'), z.literal('file_public'), z.literal('file_share'), z.literal('message'), z.literal('channel_created'), z.literal('team_join'), z.literal('reaction_added')])).optional(),
    watchWorkspace: resolveSchema({ parameters, schema: booleanOrExpression, required: false, displayOptions: {"show":{"trigger":["any_event","message","reaction_added","file_share","app_mention"]}}, defaults: {"trigger":[]} }),
    channelId: resolveSchema({ parameters, schema: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('id'), z.literal('url')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]), required: false, displayOptions: {"show":{"watchWorkspace":[false]}}, defaults: {"watchWorkspace":false} }),
    downloadFiles: resolveSchema({ parameters, schema: booleanOrExpression, required: false, displayOptions: {"show":{"trigger":["any_event","file_share"]}}, defaults: {"trigger":[]} }),
    options: z.object({ resolveIds: booleanOrExpression.optional(), userIds: z.array(z.string()).optional() }).optional(),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
