// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/discord/v2/resource_member/operation_get_all.ts
/**
 * Discord Node - Version 2
 * Discriminator: resource=member, operation=getAll
 */


interface Credentials {
  discordBotApi: CredentialReference;
  discordOAuth2Api: CredentialReference;
  discordWebhookApi?: CredentialReference;
}

/** Retrieve the latest messages in a channel */
export type DiscordV2MemberGetAllParams = {
  resource: 'member';
  operation: 'getAll';
  authentication?: 'botToken' | 'oAuth2' | 'webhook' | Expression<string>;
/**
 * Select the server (guild) that your bot is connected to
 * @displayOptions.show { authentication: ["botToken", "oAuth2"] }
 * @default {"mode":"list","value":""}
 */
    guildId?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
/**
 * Whether to return all results or only up to a given limit
 * @displayOptions.hide { authentication: ["webhook"] }
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @displayOptions.hide { authentication: ["webhook"] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * The ID of the user after which to return the members
 * @displayOptions.hide { authentication: ["webhook"] }
 */
    after?: string | Expression<string> | PlaceholderValue;
/**
 * Options
 * @displayOptions.hide { authentication: ["webhook"] }
 * @default {}
 */
    options?: {
    /** Whether to return a simplified version of the response instead of the raw data
     * @default true
     */
    simplify?: boolean | Expression<boolean>;
  };
};

export type DiscordV2MemberGetAllOutput = {
  communication_disabled_until?: null;
  deaf?: boolean;
  flags?: number;
  joined_at?: string;
  mute?: boolean;
  pending?: boolean;
  roles?: Array<string>;
  user?: {
    accent_color?: null;
    banner?: null;
    banner_color?: null;
    clan?: null;
    discriminator?: string;
    flags?: number;
    id?: string;
    public_flags?: number;
    username?: string;
  };
};

export type DiscordV2MemberGetAllNode = {
  type: 'n8n-nodes-base.discord';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<DiscordV2MemberGetAllParams>;
  output?: Items<DiscordV2MemberGetAllOutput>;
};
