// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/discord/v2/resource_member/operation_role_add.ts
/**
 * Discord Node - Version 2
 * Discriminator: resource=member, operation=roleAdd
 */


interface Credentials {
  discordBotApi: CredentialReference;
  discordOAuth2Api: CredentialReference;
  discordWebhookApi?: CredentialReference;
}

/** Add a role to a member */
export type DiscordV2MemberRoleAddParams = {
  resource: 'member';
  operation: 'roleAdd';
  authentication?: 'botToken' | 'oAuth2' | 'webhook' | Expression<string>;
/**
 * Select the server (guild) that your bot is connected to
 * @displayOptions.show { authentication: ["botToken", "oAuth2"] }
 * @default {"mode":"list","value":""}
 */
    guildId?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
/**
 * Select the user you want to assign a role to
 * @displayOptions.hide { authentication: ["webhook"] }
 * @default {"mode":"list","value":""}
 */
    userId?: { __rl: true; mode: 'list' | 'id'; value: string; cachedResultName?: string };
/**
 * Select the roles you want to add to the user
 * @displayOptions.hide { authentication: ["webhook"] }
 * @default []
 */
    role?: string[];
};

export type DiscordV2MemberRoleAddOutput = {
  success?: boolean;
};

export type DiscordV2MemberRoleAddNode = {
  type: 'n8n-nodes-base.discord';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<DiscordV2MemberRoleAddParams>;
  output?: Items<DiscordV2MemberRoleAddOutput>;
};
