// n8n-nodes-base/dist/credentials/BitbucketAccessTokenApi.credentials.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BitbucketAccessTokenApi = void 0;
class BitbucketAccessTokenApi {
    name = 'bitbucketAccessTokenApi';
    displayName = 'Bitbucket Access Token API';
    documentationUrl = 'bitbuckettokenapi';
    properties = [
        {
            displayName: 'Email',
            name: 'email',
            type: 'string',
            default: '',
        },
        {
            displayName: 'Access Token',
            name: 'accessToken',
            type: 'string',
            typeOptions: { password: true },
            default: '',
        },
    ];
    async authenticate(credentials, requestOptions) {
        const encodedApiKey = Buffer.from(`${credentials.email}:${credentials.accessToken}`).toString('base64');
        if (!requestOptions.headers) {
            requestOptions.headers = {};
        }
        requestOptions.headers.Authorization = `Basic ${encodedApiKey}`;
        return requestOptions;
    }
    test = {
        request: {
            baseURL: 'https://api.bitbucket.org/2.0',
            url: '/user',
        },
    };
}
exports.BitbucketAccessTokenApi = BitbucketAccessTokenApi;
