// n8n-nodes-base/dist/credentials/StripeApi.credentials.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.StripeApi = void 0;
class StripeApi {
    name = 'stripeApi';
    displayName = 'Stripe API';
    documentationUrl = 'stripe';
    properties = [
        {
            displayName: 'Secret Key',
            name: 'secretKey',
            type: 'string',
            typeOptions: { password: true },
            default: '',
        },
        {
            displayName: 'Signature Secret',
            name: 'signatureSecret',
            type: 'string',
            typeOptions: { password: true },
            default: '',
            description: 'The signature secret is used to verify the authenticity of requests sent by Stripe.',
        },
        {
            displayName: 'We strongly recommend setting up a <a href="https://stripe.com/docs/webhooks" target="_blank">signing secret</a> to ensure the authenticity of requests.',
            name: 'notice',
            type: 'notice',
            default: '',
            displayOptions: {
                show: {
                    signatureSecret: [''],
                },
            },
        },
    ];
    authenticate = {
        type: 'generic',
        properties: {
            headers: {
                Authorization: '=Bearer {{$credentials.secretKey}}',
            },
        },
    };
    test = {
        request: {
            baseURL: 'https://api.stripe.com/v1',
            url: '/charges',
            json: true,
        },
    };
}
exports.StripeApi = StripeApi;
