// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/payPal/v1/resource_payout/operation_create.schema.js
/**
 * PayPal Node - Version 1 - Zod Schema
 * Discriminator: resource=payout, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('payout').default('payout'),
      operation: z.literal('create').default('create'),
      senderBatchId: stringOrExpression.optional(),
      jsonParameters: booleanOrExpression.optional(),
      itemsUi: resolveSchema({ parameters, schema: z.object({ itemsValues: z.array(z.object({ recipientType: z.union([z.literal('phone'), z.literal('email'), z.literal('paypalId'), expressionSchema]).optional(), receiverValue: stringOrExpression.optional(), currency: z.union([z.literal('AUD'), z.literal('BRL'), z.literal('CAD'), z.literal('CZK'), z.literal('DKK'), z.literal('EUR'), z.literal('USD'), expressionSchema]).optional(), amount: stringOrExpression.optional(), note: stringOrExpression.optional(), senderItemId: stringOrExpression.optional(), recipientWallet: z.union([z.literal('paypal'), z.literal('venmo'), expressionSchema]).optional() })).optional() }), required: false, displayOptions: {"show":{"jsonParameters":[false]}}, defaults: {"jsonParameters":false} }),
      itemsJson: resolveSchema({ parameters, schema: z.union([iDataObjectSchema, z.string()]), required: false, displayOptions: {"show":{"jsonParameters":[true]}}, defaults: {"jsonParameters":false} }),
      additionalFields: z.object({ emailSubject: stringOrExpression.optional(), emailMessage: stringOrExpression.optional(), note: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
