// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/emelia/v1/resource_contact_list/operation_add.schema.js
/**
 * Emelia Node - Version 1 - Zod Schema
 * Discriminator: resource=contactList, operation=add
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('contactList'),
      operation: z.literal('add'),
      contactListId: stringOrExpression.optional(),
      contactEmail: stringOrExpression.optional(),
      additionalFields: z.object({ customFieldsUi: z.unknown().optional(), firstName: stringOrExpression.optional(), lastContacted: stringOrExpression.optional(), lastName: stringOrExpression.optional(), lastOpen: stringOrExpression.optional(), lastReplied: stringOrExpression.optional(), mailsSent: numberOrExpression.optional(), phoneNumber: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
