// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/uptimeRobot/v1/resource_maintenance_window/operation_create.ts
/**
 * UptimeRobot Node - Version 1
 * Discriminator: resource=maintenanceWindow, operation=create
 */


interface Credentials {
  uptimeRobotApi: CredentialReference;
}

/** Create a monitor */
export type UptimeRobotV1MaintenanceWindowCreateParams = {
  resource: 'maintenanceWindow';
  operation: 'create';
/**
 * The maintenance window activation period (minutes)
 * @default 1
 */
    duration?: number | Expression<number>;
/**
 * The friendly name of the maintenance window
 */
    friendlyName?: string | Expression<string> | PlaceholderValue;
/**
 * The type of the maintenance window
 */
    type?: 1 | 2 | 3 | 4 | Expression<number>;
/**
 * Week Day
 * @displayOptions.show { type: [3] }
 */
    weekDay?: 1 | 2 | 3 | 4 | 5 | 6 | 7 | Expression<number>;
/**
 * Month Day
 * @displayOptions.show { type: [4] }
 * @default 1
 */
    monthDay?: number | Expression<number>;
/**
 * The maintenance window start datetime
 */
    start_time?: string | Expression<string>;
};

export type UptimeRobotV1MaintenanceWindowCreateNode = {
  type: 'n8n-nodes-base.uptimeRobot';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<UptimeRobotV1MaintenanceWindowCreateParams>;
};
