// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/uptimeRobot/v1/resource_maintenance_window/operation_create.schema.js
/**
 * UptimeRobot Node - Version 1 - Zod Schema
 * Discriminator: resource=maintenanceWindow, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('maintenanceWindow'),
      operation: z.literal('create'),
      duration: numberOrExpression.optional(),
      friendlyName: stringOrExpression.optional(),
      type: z.union([z.literal(1), z.literal(2), z.literal(3), z.literal(4), expressionSchema]).optional(),
      weekDay: resolveSchema({ parameters, schema: z.union([z.literal(1), z.literal(2), z.literal(3), z.literal(4), z.literal(5), z.literal(6), z.literal(7), expressionSchema]), required: false, displayOptions: {"show":{"type":[3]}}, defaults: {"type":""} }),
      monthDay: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"type":[4]}}, defaults: {"type":""} }),
      start_time: stringOrExpression.optional(),
    }).optional(),
  });
};
