// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/reddit/v1/resource_post_comment/operation_create.ts
/**
 * Reddit Node - Version 1
 * Discriminator: resource=postComment, operation=create
 */


interface Credentials {
  redditOAuth2Api: CredentialReference;
}

/** Create a top-level comment in a post */
export type RedditV1PostCommentCreateParams = {
  resource: 'postComment';
  operation: 'create';
/**
 * ID of the post to write the comment to. Found in the post URL: &lt;code&gt;/r/[subreddit_name]/comments/[post_id]/[post_title]&lt;/code&gt;
 */
    postId?: string | Expression<string> | PlaceholderValue;
/**
 * Text of the comment. Markdown supported.
 */
    commentText?: string | Expression<string> | PlaceholderValue;
};

export type RedditV1PostCommentCreateOutput = {
  approved_at_utc?: null;
  approved_by?: null;
  archived?: boolean;
  associated_award?: null;
  author?: string;
  author_flair_background_color?: null;
  author_flair_css_class?: null;
  author_flair_template_id?: null;
  author_flair_text?: null;
  author_flair_text_color?: null;
  author_flair_type?: string;
  author_fullname?: string;
  author_is_blocked?: boolean;
  author_patreon_flair?: boolean;
  author_premium?: boolean;
  banned_at_utc?: null;
  banned_by?: null;
  body?: string;
  body_html?: string;
  can_gild?: boolean;
  can_mod_post?: boolean;
  collapsed?: boolean;
  collapsed_because_crowd_control?: null;
  collapsed_reason?: null;
  collapsed_reason_code?: null;
  comment_type?: null;
  controversiality?: number;
  created?: number;
  created_utc?: number;
  distinguished?: null;
  downs?: number;
  edited?: boolean;
  gilded?: number;
  id?: string;
  is_submitter?: boolean;
  likes?: boolean;
  link_id?: string;
  locked?: boolean;
  mod_note?: null;
  mod_reason_by?: null;
  mod_reason_title?: null;
  name?: string;
  no_follow?: boolean;
  parent_id?: string;
  permalink?: string;
  removal_reason?: null;
  replies?: string;
  rte_mode?: string;
  saved?: boolean;
  score?: number;
  score_hidden?: boolean;
  send_replies?: boolean;
  stickied?: boolean;
  subreddit?: string;
  subreddit_id?: string;
  subreddit_name_prefixed?: string;
  subreddit_type?: string;
  top_awarded_type?: null;
  total_awards_received?: number;
  unrepliable_reason?: null;
  ups?: number;
};

export type RedditV1PostCommentCreateNode = {
  type: 'n8n-nodes-base.reddit';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<RedditV1PostCommentCreateParams>;
  output?: Items<RedditV1PostCommentCreateOutput>;
};
