// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/reddit/v1/resource_post_comment/operation_get_all.ts
/**
 * Reddit Node - Version 1
 * Discriminator: resource=postComment, operation=getAll
 */


interface Credentials {
  redditOAuth2Api: CredentialReference;
}

/** Retrieve many comments in a post */
export type RedditV1PostCommentGetAllParams = {
  resource: 'postComment';
  operation: 'getAll';
/**
 * The name of subreddit where the post is
 */
    subreddit?: string | Expression<string> | PlaceholderValue;
/**
 * ID of the post to get all comments from. Found in the post URL: &lt;code&gt;/r/[subreddit_name]/comments/[post_id]/[post_title]&lt;/code&gt;
 */
    postId?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
};

export type RedditV1PostCommentGetAllOutput = {
  approved_at_utc?: null;
  approved_by?: null;
  archived?: boolean;
  associated_award?: null;
  author?: string;
  author_flair_richtext?: Array<{
    a?: string;
    e?: string;
    t?: string;
    u?: string;
  }>;
  author_flair_type?: string;
  author_fullname?: string;
  author_is_blocked?: boolean;
  author_patreon_flair?: boolean;
  author_premium?: boolean;
  banned_at_utc?: null;
  banned_by?: null;
  body?: string;
  body_html?: string;
  can_gild?: boolean;
  can_mod_post?: boolean;
  collapsed?: boolean;
  collapsed_because_crowd_control?: null;
  collapsed_reason?: null;
  comment_type?: null;
  controversiality?: number;
  created?: number;
  created_utc?: number;
  depth?: number;
  downs?: number;
  gilded?: number;
  id?: string;
  is_submitter?: boolean;
  link_id?: string;
  locked?: boolean;
  mod_note?: null;
  mod_reason_by?: null;
  mod_reason_title?: null;
  name?: string;
  no_follow?: boolean;
  num_reports?: null;
  parent_id?: string;
  permalink?: string;
  removal_reason?: null;
  report_reasons?: null;
  saved?: boolean;
  score?: number;
  score_hidden?: boolean;
  send_replies?: boolean;
  stickied?: boolean;
  subreddit?: string;
  subreddit_id?: string;
  subreddit_name_prefixed?: string;
  subreddit_type?: string;
  top_awarded_type?: null;
  total_awards_received?: number;
  unrepliable_reason?: null;
  ups?: number;
};

export type RedditV1PostCommentGetAllNode = {
  type: 'n8n-nodes-base.reddit';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<RedditV1PostCommentGetAllParams>;
  output?: Items<RedditV1PostCommentGetAllOutput>;
};
