// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/harvest/v1/resource_estimate/operation_update.schema.js
/**
 * Harvest Node - Version 1 - Zod Schema
 * Discriminator: resource=estimate, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('estimate'),
      operation: z.literal('update'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      accountId: stringOrExpression.optional(),
      id: stringOrExpression.optional(),
      updateFields: z.object({ client_id: stringOrExpression.optional(), currency: stringOrExpression.optional(), over_budget_notification_percentage: stringOrExpression.optional(), issue_date: stringOrExpression.optional(), number: stringOrExpression.optional(), notes: stringOrExpression.optional(), purchase_order: stringOrExpression.optional(), subject: stringOrExpression.optional(), tax: stringOrExpression.optional(), tax2: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
