// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/raindrop/v1/resource_bookmark/operation_update.ts
/**
 * Raindrop Node - Version 1
 * Discriminator: resource=bookmark, operation=update
 */


interface Credentials {
  raindropOAuth2Api: CredentialReference;
}

export type RaindropV1BookmarkUpdateParams = {
  resource: 'bookmark';
  operation: 'update';
/**
 * The ID of the bookmark to update
 */
    bookmarkId?: string | Expression<string> | PlaceholderValue;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     */
    collectionId?: string | Expression<string>;
    /** Whether this bookmark is marked as favorite
     * @default false
     */
    important?: boolean | Expression<boolean>;
    /** For example if you want to move bookmark to the first place set this field to 0
     * @default 0
     */
    order?: number | Expression<number>;
    /** Whether Raindrop should reload cover, description and HTML for the URL
     * @default false
     */
    pleaseParse?: boolean | Expression<boolean>;
    /** Bookmark tags. Multiple tags can be set separated by comma.
     */
    tags?: string | Expression<string> | PlaceholderValue;
    /** Title of the bookmark to be created
     */
    title?: string | Expression<string> | PlaceholderValue;
  };
};

export type RaindropV1BookmarkUpdateNode = {
  type: 'n8n-nodes-base.raindrop';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<RaindropV1BookmarkUpdateParams>;
};
