// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/raindrop/v1/resource_bookmark/operation_create.ts
/**
 * Raindrop Node - Version 1
 * Discriminator: resource=bookmark, operation=create
 */


interface Credentials {
  raindropOAuth2Api: CredentialReference;
}

export type RaindropV1BookmarkCreateParams = {
  resource: 'bookmark';
  operation: 'create';
/**
 * Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
 */
    collectionId?: string | Expression<string>;
/**
 * Link of the bookmark to be created
 */
    link?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Whether this bookmark is marked as favorite
     * @default false
     */
    important?: boolean | Expression<boolean>;
    /** Sort order for the bookmark. For example, to move it to first place, enter 0.
     * @default 0
     */
    order?: number | Expression<number>;
    /** Whether Raindrop should load cover, description and HTML for the URL
     * @default false
     */
    pleaseParse?: boolean | Expression<boolean>;
    /** Bookmark tags. Multiple tags can be set separated by comma.
     */
    tags?: string | Expression<string> | PlaceholderValue;
    /** Title of the bookmark to create
     */
    title?: string | Expression<string> | PlaceholderValue;
  };
};

export type RaindropV1BookmarkCreateNode = {
  type: 'n8n-nodes-base.raindrop';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<RaindropV1BookmarkCreateParams>;
};
