// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/raindrop/v1/resource_bookmark/operation_update.schema.js
/**
 * Raindrop Node - Version 1 - Zod Schema
 * Discriminator: resource=bookmark, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('bookmark'),
      operation: z.literal('update'),
      bookmarkId: stringOrExpression.optional(),
      updateFields: z.object({ collectionId: stringOrExpression.optional(), important: booleanOrExpression.optional(), order: numberOrExpression.optional(), pleaseParse: booleanOrExpression.optional(), tags: stringOrExpression.optional(), title: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
