// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/rabbitmq/v1.schema.js
/**
 * RabbitMQ Node - Version 1 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    operation: z.unknown().optional(),
    mode: resolveSchema({ parameters, schema: z.union([z.literal('queue'), z.literal('exchange'), expressionSchema]), required: false, displayOptions: {"hide":{"operation":["deleteMessage"]}}, defaults: {"operation":"sendMessage"} }),
    queue: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"mode":["queue"]},"hide":{"operation":["deleteMessage"]}}, defaults: {"mode":"queue","operation":"sendMessage"} }),
    exchange: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"mode":["exchange"]}}, defaults: {"mode":"queue"} }),
    exchangeType: resolveSchema({ parameters, schema: z.union([z.literal('direct'), z.literal('topic'), z.literal('headers'), z.literal('fanout'), expressionSchema]), required: false, displayOptions: {"show":{"mode":["exchange"]}}, defaults: {"mode":"queue"} }),
    routingKey: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"mode":["exchange"]}}, defaults: {"mode":"queue"} }),
    sendInputData: resolveSchema({ parameters, schema: booleanOrExpression, required: false, displayOptions: {"show":{"operation":["sendMessage"]}}, defaults: {"operation":"sendMessage"} }),
    message: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"sendInputData":[false]}}, defaults: {"sendInputData":true} }),
    options: resolveSchema({ parameters, schema: z.object({ alternateExchange: stringOrExpression.optional(), arguments: z.unknown().optional(), autoDelete: booleanOrExpression.optional(), durable: booleanOrExpression.optional(), exclusive: booleanOrExpression.optional(), headers: z.unknown().optional() }), required: false, displayOptions: {"show":{"operation":["sendMessage"]}}, defaults: {"operation":"sendMessage"} }),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
