// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mattermost/v1/resource_user/operation_get_all.schema.js
/**
 * Mattermost Node - Version 1 - Zod Schema
 * Discriminator: resource=user, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('user'),
      operation: z.literal('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":true} }),
      additionalFields: z.object({ inChannel: stringOrExpression.optional(), inTeam: stringOrExpression.optional(), notInTeam: stringOrExpression.optional(), notInChannel: stringOrExpression.optional(), sort: z.union([z.literal('createdAt'), z.literal('lastActivityAt'), z.literal('status'), z.literal('username'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
