// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mattermost/v1/resource_user/operation_get_all.ts
/**
 * Mattermost Node - Version 1
 * Discriminator: resource=user, operation=getAll
 */


interface Credentials {
  mattermostApi: CredentialReference;
}

/** Get many reactions to one or more posts */
export type MattermostV1UserGetAllParams = {
  resource: 'user';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default true
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** The ID of the channel to get users for
     */
    inChannel?: string | Expression<string> | PlaceholderValue;
    /** The ID of the team to get users for
     */
    inTeam?: string | Expression<string> | PlaceholderValue;
    /** The ID of the team to exclude users for
     */
    notInTeam?: string | Expression<string> | PlaceholderValue;
    /** The ID of the channel to exclude users for
     */
    notInChannel?: string | Expression<string> | PlaceholderValue;
    /** The ID of the channel to exclude users for
     * @default username
     */
    sort?: 'createdAt' | 'lastActivityAt' | 'status' | 'username' | Expression<string>;
  };
};

export type MattermostV1UserGetAllNode = {
  type: 'n8n-nodes-base.mattermost';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MattermostV1UserGetAllParams>;
};
