// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/getResponse/v1/resource_contact/operation_delete.ts
/**
 * GetResponse Node - Version 1
 * Discriminator: resource=contact, operation=delete
 */


interface Credentials {
  getResponseApi: CredentialReference;
  getResponseOAuth2Api: CredentialReference;
}

/** Delete a contact */
export type GetResponseV1ContactDeleteParams = {
  resource: 'contact';
  operation: 'delete';
  authentication?: 'apiKey' | 'oAuth2' | Expression<string>;
/**
 * ID of contact to delete
 */
    contactId?: string | Expression<string> | PlaceholderValue;
/**
 * Options
 * @default {}
 */
    options?: {
    /** This makes it possible to pass the IP from which the contact unsubscribed. Used only if the messageId was send.
     */
    ipAddress?: string | Expression<string> | PlaceholderValue;
    /** The ID of a message (such as a newsletter, an autoresponder, or an RSS-newsletter). When passed, this method will simulate the unsubscribe process, as if the contact clicked the unsubscribe link in a given message.
     */
    messageId?: string | Expression<string> | PlaceholderValue;
  };
};

export type GetResponseV1ContactDeleteNode = {
  type: 'n8n-nodes-base.getResponse';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GetResponseV1ContactDeleteParams>;
};
