// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/getResponse/v1/resource_contact/operation_update.ts
/**
 * GetResponse Node - Version 1
 * Discriminator: resource=contact, operation=update
 */


interface Credentials {
  getResponseApi: CredentialReference;
  getResponseOAuth2Api: CredentialReference;
}

/** Update contact properties */
export type GetResponseV1ContactUpdateParams = {
  resource: 'contact';
  operation: 'update';
  authentication?: 'apiKey' | 'oAuth2' | Expression<string>;
/**
 * Unique identifier for a particular contact
 */
    contactId?: string | Expression<string> | PlaceholderValue;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     */
    campaignId?: string | Expression<string>;
    /** Custom Fields
     * @default {}
     */
    customFieldsUi?: {
        /** Custom Field
     */
    customFieldValues?: Array<{
      /** The end user specified key of the user defined data. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
       */
      customFieldId?: string | Expression<string>;
      /** The end user specified value of the user defined data
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
    /** The day on which the contact is in the Autoresponder cycle. null indicates the contacts is not in the cycle.
     */
    dayOfCycle?: string | Expression<string> | PlaceholderValue;
    /** Email
     */
    email?: string | Expression<string> | PlaceholderValue;
    /** The contact's IP address. IPv4 and IPv6 formats are accepted.
     */
    ipAddress?: string | Expression<string> | PlaceholderValue;
    /** Name
     */
    name?: string | Expression<string> | PlaceholderValue;
    /** Note
     */
    note?: string | Expression<string> | PlaceholderValue;
    /** Contact scoring, pass null to remove the score from a contact
     */
    scoring?: number | Expression<number>;
    /** Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
     * @default []
     */
    tags?: string[];
  };
};

export type GetResponseV1ContactUpdateNode = {
  type: 'n8n-nodes-base.getResponse';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GetResponseV1ContactUpdateParams>;
};
