// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/formstackTrigger/v1.ts
/**
 * Formstack Trigger Node - Version 1
 * Starts the workflow on a Formstack form submission.
 */


export interface FormstackTriggerV1Params {
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * The Formstack form to monitor for new submissions. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    formId?: string | Expression<string>;
/**
 * Whether to return a simplified version of the response instead of the raw data
 * @default true
 */
    simple?: boolean | Expression<boolean>;
}

export interface FormstackTriggerV1Credentials {
  formstackApi: CredentialReference;
  formstackOAuth2Api: CredentialReference;
}

interface FormstackTriggerV1NodeBase {
  type: 'n8n-nodes-base.formstackTrigger';
  version: 1;
  credentials?: FormstackTriggerV1Credentials;
  isTrigger: true;
}

export type FormstackTriggerV1ParamsNode = FormstackTriggerV1NodeBase & {
  config: NodeConfig<FormstackTriggerV1Params>;
};

export type FormstackTriggerV1Node = FormstackTriggerV1ParamsNode;
