// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/trello/v1/resource_list/operation_create.schema.js
/**
 * Trello Node - Version 1 - Zod Schema
 * Discriminator: resource=list, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('list'),
      operation: z.literal('create'),
      idBoard: stringOrExpression.optional(),
      name: stringOrExpression.optional(),
      additionalFields: z.object({ idListSource: stringOrExpression.optional(), pos: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
