// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pagerDuty/v1/resource_incident/operation_get_all.schema.js
/**
 * PagerDuty Node - Version 1 - Zod Schema
 * Discriminator: resource=incident, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('incident').default('incident'),
      operation: z.literal('getAll'),
      authentication: z.union([z.literal('apiToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      options: z.object({ dateRange: z.union([z.literal('all'), expressionSchema]).optional(), incidentKey: stringOrExpression.optional(), include: z.array(z.union([z.literal('acknowledgers'), z.literal('assignees'), z.literal('conferenceBridge'), z.literal('escalationPolicies'), z.literal('firstTriggerLogEntries'), z.literal('priorities'), z.literal('services'), z.literal('teams'), z.literal('users')])).optional(), serviceIds: z.array(z.string()).optional(), since: stringOrExpression.optional(), sortBy: stringOrExpression.optional(), statuses: z.array(z.union([z.literal('acknowledged'), z.literal('resolved'), z.literal('triggered')])).optional(), teamIds: stringOrExpression.optional(), timeZone: stringOrExpression.optional(), until: stringOrExpression.optional(), urgencies: z.array(z.union([z.literal('high'), z.literal('low')])).optional(), userIds: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
