// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pagerDuty/v1/resource_incident/operation_update.ts
/**
 * PagerDuty Node - Version 1
 * Discriminator: resource=incident, operation=update
 */


interface Credentials {
  pagerDutyApi: CredentialReference;
  pagerDutyOAuth2Api: CredentialReference;
}

/** Update an incident */
export type PagerDutyV1IncidentUpdateParams = {
  resource: 'incident';
  operation: 'update';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * Unique identifier for the incident
 */
    incidentId?: string | Expression<string> | PlaceholderValue;
/**
 * The email address of a valid user associated with the account making the request
 */
    email?: string | Expression<string> | PlaceholderValue;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** Escalate the incident to this level in the escalation policy
     * @default 0
     */
    escalationLevel?: number | Expression<number>;
    /** Delegate this incident to the specified escalation policy. Cannot be specified if an assignee is given. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    escalationPolicyId?: string | Expression<string>;
    /** The incident will be created on this service. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    priorityId?: string | Expression<string>;
    /** The resolution for this incident if status is set to resolved
     */
    resolution?: string | Expression<string> | PlaceholderValue;
    /** The new status of the incident
     */
    status?: 'acknowledged' | 'resolved' | Expression<string>;
    /** A succinct description of the nature, symptoms, cause, or effect of the incident
     */
    title?: string | Expression<string> | PlaceholderValue;
    /** The urgency of the incident
     */
    urgency?: 'high' | 'low' | Expression<string>;
  };
/**
 * Conference Bridge
 * @default {}
 */
    conferenceBridgeUi?: {
        /** Conference Bridge
     */
    conferenceBridgeValues?: {
      /** Phone numbers should be formatted like +1 415-555-1212,,,,1234#, where a comma (,) represents a one-second wait and pound (#) completes access code input
       */
      conferenceNumber?: string | Expression<string> | PlaceholderValue;
      /** An URL for the conference bridge. This could be a link to a web conference or Slack channel.
       */
      conferenceUrl?: string | Expression<string> | PlaceholderValue;
    };
  };
};

export type PagerDutyV1IncidentUpdateNode = {
  type: 'n8n-nodes-base.pagerDuty';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PagerDutyV1IncidentUpdateParams>;
};
