// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pagerDuty/v1/resource_incident/operation_create.ts
/**
 * PagerDuty Node - Version 1
 * Discriminator: resource=incident, operation=create
 */


interface Credentials {
  pagerDutyApi: CredentialReference;
  pagerDutyOAuth2Api: CredentialReference;
}

/** Create an incident */
export type PagerDutyV1IncidentCreateParams = {
  resource: 'incident';
  operation: 'create';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * A succinct description of the nature, symptoms, cause, or effect of the incident
 */
    title?: string | Expression<string> | PlaceholderValue;
/**
 * The incident will be created on this service. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    serviceId?: string | Expression<string>;
/**
 * The email address of a valid user associated with the account making the request
 */
    email?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Delegate this incident to the specified escalation policy. Cannot be specified if an assignee is given. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    escalationPolicyId?: string | Expression<string>;
    /** Additional details about the incident which will go in the body
     */
    details?: string | Expression<string> | PlaceholderValue;
    /** Sending subsequent requests referencing the same service and with the same incident_key will result in those requests being rejected if an open incident matches that incident_key
     */
    incidentKey?: string | Expression<string> | PlaceholderValue;
    /** The incident will be created on this service. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    priorityId?: string | Expression<string>;
    /** The urgency of the incident
     */
    urgency?: 'high' | 'low' | Expression<string>;
  };
/**
 * Conference Bridge
 * @default {}
 */
    conferenceBridgeUi?: {
        /** Conference Bridge
     */
    conferenceBridgeValues?: {
      /** Phone numbers should be formatted like +1 415-555-1212,,,,1234#, where a comma (,) represents a one-second wait and pound (#) completes access code input
       */
      conferenceNumber?: string | Expression<string> | PlaceholderValue;
      /** An URL for the conference bridge. This could be a link to a web conference or Slack channel.
       */
      conferenceUrl?: string | Expression<string> | PlaceholderValue;
    };
  };
};

export type PagerDutyV1IncidentCreateNode = {
  type: 'n8n-nodes-base.pagerDuty';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PagerDutyV1IncidentCreateParams>;
};
