// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pagerDuty/v1/resource_incident/operation_create.schema.js
/**
 * PagerDuty Node - Version 1 - Zod Schema
 * Discriminator: resource=incident, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('incident').default('incident'),
      operation: z.literal('create').default('create'),
      authentication: z.union([z.literal('apiToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      title: stringOrExpression.optional(),
      serviceId: stringOrExpression.optional(),
      email: stringOrExpression.optional(),
      additionalFields: z.object({ escalationPolicyId: stringOrExpression.optional(), details: stringOrExpression.optional(), incidentKey: stringOrExpression.optional(), priorityId: stringOrExpression.optional(), urgency: z.union([z.literal('high'), z.literal('low'), expressionSchema]).optional() }).optional(),
      conferenceBridgeUi: z.object({ conferenceBridgeValues: z.object({ conferenceNumber: stringOrExpression.optional(), conferenceUrl: stringOrExpression.optional() }).optional() }).optional(),
    }).optional(),
  });
};
