// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pagerDuty/v1/resource_incident/operation_update.schema.js
/**
 * PagerDuty Node - Version 1 - Zod Schema
 * Discriminator: resource=incident, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('incident').default('incident'),
      operation: z.literal('update'),
      authentication: z.union([z.literal('apiToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      incidentId: stringOrExpression.optional(),
      email: stringOrExpression.optional(),
      updateFields: z.object({ escalationLevel: numberOrExpression.optional(), escalationPolicyId: stringOrExpression.optional(), priorityId: stringOrExpression.optional(), resolution: stringOrExpression.optional(), status: z.union([z.literal('acknowledged'), z.literal('resolved'), expressionSchema]).optional(), title: stringOrExpression.optional(), urgency: z.union([z.literal('high'), z.literal('low'), expressionSchema]).optional() }).optional(),
      conferenceBridgeUi: z.object({ conferenceBridgeValues: z.object({ conferenceNumber: stringOrExpression.optional(), conferenceUrl: stringOrExpression.optional() }).optional() }).optional(),
    }).optional(),
  });
};
