// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/wekan/v1/resource_card/operation_get_all.schema.js
/**
 * Wekan Node - Version 1 - Zod Schema
 * Discriminator: resource=card, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('card').default('card'),
      operation: z.literal('getAll'),
      boardId: stringOrExpression.optional(),
      fromObject: z.union([z.literal('list'), z.literal('swimlane'), expressionSchema]).optional(),
      listId: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"fromObject":["list"]}}, defaults: {"fromObject":""} }),
      swimlaneId: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"fromObject":["swimlane"]}}, defaults: {"fromObject":""} }),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
    }).optional(),
  });
};
