// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/wekan/v1/resource_card/operation_create.schema.js
/**
 * Wekan Node - Version 1 - Zod Schema
 * Discriminator: resource=card, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('card').default('card'),
      operation: z.literal('create').default('create'),
      boardId: stringOrExpression.optional(),
      listId: stringOrExpression.optional(),
      title: stringOrExpression.optional(),
      swimlaneId: stringOrExpression.optional(),
      authorId: stringOrExpression.optional(),
      additionalFields: z.object({ assignees: z.array(z.string()).optional(), description: stringOrExpression.optional(), members: z.array(z.string()).optional() }).optional(),
    }).optional(),
  });
};
