// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/quickbooks/v1/resource_customer/operation_update.schema.js
/**
 * QuickBooks Online Node - Version 1 - Zod Schema
 * Discriminator: resource=customer, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('customer').default('customer'),
      operation: z.literal('update'),
      customerId: stringOrExpression.optional(),
      updateFields: z.object({ Active: booleanOrExpression.optional(), Balance: stringOrExpression.optional(), BalanceWithJobs: numberOrExpression.optional(), BillAddr: z.unknown().optional(), BillWithParent: booleanOrExpression.optional(), CompanyName: stringOrExpression.optional(), FamilyName: stringOrExpression.optional(), FullyQualifiedName: stringOrExpression.optional(), GivenName: stringOrExpression.optional(), PreferredDeliveryMethod: z.union([z.literal('Print'), z.literal('Email'), z.literal('None'), expressionSchema]).optional(), PrimaryEmailAddr: stringOrExpression.optional(), PrimaryPhone: stringOrExpression.optional(), PrintOnCheckName: stringOrExpression.optional(), Taxable: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
