// @n8n/n8n-nodes-langchain/dist/nodes/vector_store/VectorStoreChromaDB/VectorStoreChromaDB.node.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VectorStoreChromaDB = void 0;
const chroma_1 = require("@langchain/community/vectorstores/chroma");
const chromadb_1 = require("chromadb");
const n8n_workflow_1 = require("n8n-workflow");
const ai_utilities_1 = require("@n8n/ai-utilities");
const descriptions_1 = require("../shared/descriptions");
function getCredentialType(context) {
    try {
        const authentication = context.getNodeParameter('authentication', 0);
        if (typeof authentication === 'string') {
            return authentication;
        }
    }
    catch (error) {
    }
    const node = context.getNode();
    if (node.credentials?.chromaCloudApi) {
        return 'chromaCloudApi';
    }
    if (node.credentials?.chromaSelfHostedApi) {
        return 'chromaSelfHostedApi';
    }
    return 'chromaSelfHostedApi';
}
async function getChromaClient(context, itemIndex) {
    const credentialType = getCredentialType(context);
    const credentials = await context.getCredentials(credentialType, itemIndex);
    if (credentialType === 'chromaCloudApi') {
        const apiKey = typeof credentials.apiKey === 'string' ? credentials.apiKey : '';
        const config = {
            apiKey,
        };
        if (typeof credentials.tenant === 'string') {
            config.tenant = credentials.tenant;
        }
        if (typeof credentials.database === 'string') {
            config.database = credentials.database;
        }
        return new chromadb_1.CloudClient(config);
    }
    else {
        const baseUrl = typeof credentials.baseUrl === 'string' ? credentials.baseUrl : '';
        const authentication = typeof credentials.authentication === 'string' ? credentials.authentication : '';
        const url = new URL(baseUrl);
        const config = {
            host: url.hostname,
            port: url.port ? parseInt(url.port, 10) : 8000,
            ssl: url.protocol === 'https:',
        };
        if (authentication === 'apiKey' && typeof credentials.apiKey === 'string') {
            config.headers = {
                Authorization: `Bearer ${credentials.apiKey}`,
            };
        }
        else if (authentication === 'token' && typeof credentials.token === 'string') {
            config.headers = {
                'X-Chroma-Token': credentials.token,
            };
        }
        return new chromadb_1.ChromaClient(config);
    }
}
async function getChromaLibConfig(context, collectionName, itemIndex) {
    const credentialType = getCredentialType(context);
    const credentials = await context.getCredentials(credentialType, itemIndex);
    if (credentialType === 'chromaCloudApi') {
        const cloudClientParams = {
            apiKey: typeof credentials.apiKey === 'string' ? credentials.apiKey : '',
        };
        if (typeof credentials.tenant === 'string') {
            cloudClientParams.tenant = credentials.tenant;
        }
        if (typeof credentials.database === 'string') {
            cloudClientParams.database = credentials.database;
        }
        const config = {
            collectionName,
            clientParams: cloudClientParams,
        };
        return config;
    }
    else {
        const baseUrl = typeof credentials.baseUrl === 'string' ? credentials.baseUrl : '';
        const authentication = typeof credentials.authentication === 'string' ? credentials.authentication : '';
        const url = new URL(baseUrl);
        const clientParams = {
            host: url.hostname,
            port: url.port ? parseInt(url.port, 10) : 8000,
            ssl: url.protocol === 'https:',
        };
        if (authentication === 'apiKey' && typeof credentials.apiKey === 'string') {
            clientParams.headers = {
                Authorization: `Bearer ${credentials.apiKey}`,
            };
        }
        else if (authentication === 'token' && typeof credentials.token === 'string') {
            clientParams.headers = {
                'X-Chroma-Token': credentials.token,
            };
        }
        const config = {
            collectionName,
            clientParams,
        };
        return config;
    }
}
class ExtendedChroma extends chroma_1.Chroma {
    async ensureCollection() {
        if (!this.collection) {
            if (!this.index) {
                const clientParams = this.clientParams ?? {};
                if ('apiKey' in clientParams && typeof clientParams.apiKey === 'string') {
                    this.index = new chromadb_1.CloudClient({
                        apiKey: clientParams.apiKey,
                        tenant: typeof clientParams.tenant === 'string' ? clientParams.tenant : undefined,
                        database: typeof clientParams.database === 'string' ? clientParams.database : undefined,
                    });
                }
                else {
                    const { ChromaClient } = await ExtendedChroma.imports();
                    const clientConfig = this.url ? { path: this.url, ...clientParams } : clientParams;
                    this.index = new ChromaClient(clientConfig);
                }
            }
            try {
                this.collection = await this.index.getOrCreateCollection({
                    name: this.collectionName,
                    ...(this.collectionMetadata && { metadata: this.collectionMetadata }),
                    embeddingFunction: null,
                });
            }
            catch (error) {
                const message = error instanceof Error ? error.message : String(error);
                throw new n8n_workflow_1.ApplicationError(`Chroma getOrCreateCollection error: ${message}`);
            }
        }
        if (!this.collection) {
            throw new n8n_workflow_1.ApplicationError('Failed to initialize Chroma collection');
        }
        return this.collection;
    }
    async similaritySearchVectorWithScore(query, k, filter) {
        let flatQuery = [];
        if (query.length > 0 && Array.isArray(query[0])) {
            for (const element of query) {
                if (Array.isArray(element)) {
                    flatQuery.push.apply(flatQuery, element);
                }
                else {
                    flatQuery.push(element);
                }
            }
        }
        else {
            flatQuery = query;
        }
        return await super.similaritySearchVectorWithScore(flatQuery, k, filter);
    }
}
const authenticationProperty = {
    displayName: 'Authentication',
    name: 'authentication',
    type: 'options',
    options: [
        {
            name: 'Self-Hosted',
            value: 'chromaSelfHostedApi',
            description: 'Connect to a self-hosted ChromaDB instance',
        },
        {
            name: 'Cloud',
            value: 'chromaCloudApi',
            description: 'Connect to Chroma Cloud',
        },
    ],
    default: 'chromaSelfHostedApi',
};
const sharedFields = [authenticationProperty, descriptions_1.chromaCollectionRLC];
const retrieveFields = [
    {
        displayName: 'Options',
        name: 'options',
        type: 'collection',
        placeholder: 'Add Option',
        default: {},
        options: [ai_utilities_1.metadataFilterField],
    },
];
const insertFields = [
    {
        displayName: 'Options',
        name: 'options',
        type: 'collection',
        placeholder: 'Add Option',
        default: {},
        options: [
            {
                displayName: 'Clear Collection',
                name: 'clearCollection',
                type: 'boolean',
                default: false,
                description: 'Whether to clear the collection before inserting new data',
            },
        ],
    },
];
class VectorStoreChromaDB extends (0, ai_utilities_1.createVectorStoreNode)({
    meta: {
        displayName: 'Chroma Vector Store',
        name: 'vectorStoreChromaDB',
        description: 'Work with your data in Chroma Vector Store',
        icon: { light: 'file:chroma.svg', dark: 'file:chroma.svg' },
        docsUrl: 'https://docs.n8n.io/integrations/builtin/cluster-nodes/root-nodes/n8n-nodes-langchain.vectorstorechromadb/',
        credentials: [
            {
                name: 'chromaSelfHostedApi',
                required: true,
                displayOptions: {
                    show: {
                        authentication: ['chromaSelfHostedApi'],
                    },
                },
            },
            {
                name: 'chromaCloudApi',
                required: true,
                displayOptions: {
                    show: {
                        authentication: ['chromaCloudApi'],
                    },
                },
            },
        ],
        operationModes: ['load', 'insert', 'retrieve', 'retrieve-as-tool'],
    },
    methods: {
        listSearch: {
            async chromaCollectionsSearch() {
                try {
                    const client = await getChromaClient(this);
                    const collections = await client.listCollections();
                    if (Array.isArray(collections)) {
                        const results = collections.map((collection) => ({
                            name: collection.name,
                            value: collection.name,
                        }));
                        return { results };
                    }
                    return { results: [] };
                }
                catch (error) {
                    const errorMessage = error instanceof Error ? error.message : String(error);
                    if (errorMessage.includes('ECONNREFUSED') || errorMessage.includes('Failed to connect')) {
                        throw new n8n_workflow_1.NodeApiError(this.getNode(), {
                            message: 'Cannot connect to ChromaDB. Please ensure ChromaDB is running and accessible at the configured URL.',
                        });
                    }
                    if (errorMessage.includes('Unauthorized') ||
                        errorMessage.includes('401') ||
                        errorMessage.includes('403')) {
                        throw new n8n_workflow_1.NodeApiError(this.getNode(), {
                            message: 'Authentication failed. Please check your API key or token in the credentials',
                        });
                    }
                    throw new n8n_workflow_1.NodeApiError(this.getNode(), {
                        message: `Failed to list ChromaDB collections: ${errorMessage}`,
                    });
                }
            },
        },
    },
    retrieveFields,
    loadFields: retrieveFields,
    insertFields,
    sharedFields,
    async getVectorStoreClient(context, _filter, embeddings, itemIndex) {
        const collection = context.getNodeParameter('chromaCollection', itemIndex, '', {
            extractValue: true,
        });
        if (typeof collection !== 'string') {
            throw new n8n_workflow_1.NodeOperationError(context.getNode(), 'Collection must be a string');
        }
        try {
            const config = await getChromaLibConfig(context, collection, itemIndex);
            return await ExtendedChroma.fromExistingCollection(embeddings, config);
        }
        catch (error) {
            const message = error instanceof Error ? error.message : 'Unknown error';
            throw new n8n_workflow_1.NodeOperationError(context.getNode(), `Error connecting to ChromaDB: ${message}`, {
                itemIndex,
            });
        }
    },
    async populateVectorStore(context, embeddings, documents, itemIndex) {
        const collection = context.getNodeParameter('chromaCollection', itemIndex, '', {
            extractValue: true,
        });
        if (typeof collection !== 'string') {
            throw new n8n_workflow_1.NodeOperationError(context.getNode(), 'Collection must be a string');
        }
        const options = context.getNodeParameter('options', itemIndex, {});
        const clearCollection = options.clearCollection === true;
        if (clearCollection) {
            try {
                const client = await getChromaClient(context, itemIndex);
                await client.deleteCollection({ name: collection });
                context.logger.info(`Collection ${collection} deleted`);
            }
            catch (error) {
                context.logger.info(`Collection ${collection} does not exist yet or could not be deleted (continuing)`);
            }
        }
        try {
            const config = await getChromaLibConfig(context, collection, itemIndex);
            await ExtendedChroma.fromDocuments(documents, embeddings, config);
        }
        catch (error) {
            const chromaError = error;
            const errorMessage = chromaError.message ?? 'Unknown error';
            const detailMessage = chromaError.response?.data?.detail;
            if (errorMessage.includes('embedding with dimension') ||
                detailMessage?.includes('embedding with dimension')) {
                const displayMessage = detailMessage ?? errorMessage;
                throw new n8n_workflow_1.NodeOperationError(context.getNode(), `ChromaDB embedding dimension mismatch: ${displayMessage}`, {
                    itemIndex,
                    description: 'The collection expects embeddings with different dimensions. Enable "Clear Collection" option to recreate the collection with correct dimensions, or use a different collection name.',
                });
            }
            throw new n8n_workflow_1.NodeOperationError(context.getNode(), `Error inserting documents into ChromaDB: ${errorMessage}`, { itemIndex });
        }
    },
}) {
}
exports.VectorStoreChromaDB = VectorStoreChromaDB;
