// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/slack/v1/resource_message/operation_post.ts
/**
 * Slack Node - Version 1
 * Discriminator: resource=message, operation=post
 */


interface Credentials {
  slackApi: CredentialReference;
  slackOAuth2Api: CredentialReference;
}

/** Post a message into a channel */
export type SlackV1MessagePostParams = {
  resource: 'message';
  operation: 'post';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * The channel to send the message to
 */
    channel?: string | Expression<string> | PlaceholderValue;
/**
 * The text to send
 */
    text?: string | Expression<string> | PlaceholderValue;
/**
 * JSON Parameters
 * @default false
 */
    jsonParameters?: boolean | Expression<boolean>;
/**
 * Other options to set
 * @default {}
 */
    otherOptions?: {
    /** Emoji to use as the icon for this message. Overrides icon_url.
     */
    icon_emoji?: string | Expression<string> | PlaceholderValue;
    /** URL to an image to use as the icon for this message
     */
    icon_url?: string | Expression<string> | PlaceholderValue;
    /** Whether to find and link channel names and usernames
     * @default false
     */
    link_names?: boolean | Expression<boolean>;
    /** Provide another message's ts value to make this message a reply
     */
    thread_ts?: string | Expression<string> | PlaceholderValue;
    /** Whether to use Slack Markdown parsing
     * @default true
     */
    mrkdwn?: boolean | Expression<boolean>;
    /** Whether the reply should be made visible to everyone in the channel or conversation. Use in conjunction with thread_ts.
     * @default false
     */
    reply_broadcast?: boolean | Expression<boolean>;
    /** Whether to enable unfurling of primarily text-based content
     * @default false
     */
    unfurl_links?: boolean | Expression<boolean>;
    /** Whether to disable unfurling of media content
     * @default true
     */
    unfurl_media?: boolean | Expression<boolean>;
    /** The message will be sent from this username (i.e. as if this individual sent the message).
     * @displayOptions.show { /authentication: ["accessToken"] }
     */
    sendAsUser?: string | Expression<string> | PlaceholderValue;
  };
/**
 * The attachment to add
 * @default {}
 */
    attachments?: {
    /** Required plain-text summary of the attachment
     */
    fallback?: string | Expression<string> | PlaceholderValue;
    /** Text to send
     */
    text?: string | Expression<string> | PlaceholderValue;
    /** Title of the message
     */
    title?: string | Expression<string> | PlaceholderValue;
    /** Link of the title
     */
    title_link?: string | Expression<string> | PlaceholderValue;
    /** Color of the line left of text
     * @default #ff0000
     */
    color?: string | Expression<string>;
    /** Text which appears before the message block
     */
    pretext?: string | Expression<string> | PlaceholderValue;
    /** Name that should appear
     */
    author_name?: string | Expression<string> | PlaceholderValue;
    /** Link for the author
     */
    author_link?: string | Expression<string> | PlaceholderValue;
    /** Icon which should appear for the user
     */
    author_icon?: string | Expression<string> | PlaceholderValue;
    /** URL of image
     */
    image_url?: string | Expression<string> | PlaceholderValue;
    /** URL of thumbnail
     */
    thumb_url?: string | Expression<string> | PlaceholderValue;
    /** Text of footer to add
     */
    footer?: string | Expression<string> | PlaceholderValue;
    /** Icon which should appear next to footer
     */
    footer_icon?: string | Expression<string> | PlaceholderValue;
    /** Time message relates to
     */
    ts?: string | Expression<string>;
    /** Fields to add to message
     * @default {}
     */
    fields?: {
        /** Item
     */
    item?: Array<{
      /** Title of the item
       */
      title?: string | Expression<string> | PlaceholderValue;
      /** Value of the item
       */
      value?: string | Expression<string> | PlaceholderValue;
      /** Whether items can be displayed next to each other
       * @default true
       */
      short?: boolean | Expression<boolean>;
    }>;
  };
  };
/**
 * The blocks to add
 * @displayOptions.show { jsonParameters: [false] }
 * @default {}
 */
    blocksUi?: {
        /** Block
     */
    blocksValues?: Array<{
      /** Type
       * @default actions
       */
      type?: 'actions' | 'section' | Expression<string>;
      /** A string acting as a unique identifier for a block. You can use this block_id when you receive an interaction payload to identify the source of the action. If not specified, a block_id will be generated. Maximum length for this field is 255 characters.
       * @displayOptions.show { type: ["actions"] }
       */
      blockId?: string | Expression<string> | PlaceholderValue;
      /** Elements
       * @displayOptions.show { type: ["actions"] }
       * @default {}
       */
      elementsUi?: {
        /** Element
     */
    elementsValues?: Array<{
      /** The type of element
       * @default button
       */
      type?: 'button' | Expression<string>;
      /** The text for the block
       * @displayOptions.show { type: ["button"] }
       */
      text?: string | Expression<string> | PlaceholderValue;
      /** Whether emojis in a text field should be escaped into the colon emoji format
       * @displayOptions.show { type: ["button"] }
       * @default false
       */
      emoji?: boolean | Expression<boolean>;
      /** An identifier for this action. You can use this when you receive an interaction payload to identify the source of the action. Should be unique among all other action_ids used elsewhere by your app.
       * @displayOptions.show { type: ["button"] }
       */
      actionId?: string | Expression<string> | PlaceholderValue;
      /** A URL to load in the user's browser when the button is clicked. Maximum length for this field is 3000 characters. If you're using URL, you'll still receive an interaction payload and will need to send an acknowledgement response.
       * @displayOptions.show { type: ["button"] }
       */
      url?: string | Expression<string> | PlaceholderValue;
      /** The value to send along with the interaction payload
       * @displayOptions.show { type: ["button"] }
       */
      value?: string | Expression<string> | PlaceholderValue;
      /** Decorates buttons with alternative visual color schemes
       * @displayOptions.show { type: ["button"] }
       * @default default
       */
      style?: 'danger' | 'default' | 'primary' | Expression<string>;
      /** Defines an optional confirmation dialog after the button is clicked
       * @default {}
       */
      confirmUi?: {
        /** Confirm
     */
    confirmValue?: {
      /** Defines the dialog's title
       * @default {}
       */
      titleUi?: {
        /** Title
     */
    titleValue?: {
      /** Text of the title
       */
      text?: string | Expression<string> | PlaceholderValue;
      /** Whether emojis in a text field should be escaped into the colon emoji format
       * @default false
       */
      emoji?: boolean | Expression<boolean>;
    };
  };
      /** Defines the explanatory text that appears in the confirm dialog
       * @default {}
       */
      textUi?: {
        /** Text
     */
    textValue?: {
      /** The text for the block
       */
      text?: string | Expression<string> | PlaceholderValue;
      /** Whether emojis in a text field should be escaped into the colon emoji format
       * @default false
       */
      emoji?: boolean | Expression<boolean>;
    };
  };
      /** Defines the text of the button that confirms the action
       * @default {}
       */
      confirmTextUi?: {
        /** Confirm
     */
    confirmValue?: {
      /** Defines the explanatory text that appears in the confirm dialog
       */
      text?: string | Expression<string> | PlaceholderValue;
      /** Whether emojis in a text field should be escaped into the colon emoji format
       * @default false
       */
      emoji?: boolean | Expression<boolean>;
    };
  };
      /** Defines the text of the button that cancels the action
       * @default {}
       */
      denyUi?: {
        /** Deny
     */
    denyValue?: {
      /** Defines the text of the button that cancels the action
       */
      text?: string | Expression<string> | PlaceholderValue;
      /** Whether emojis in a text field should be escaped into the colon emoji format
       * @default false
       */
      emoji?: boolean | Expression<boolean>;
    };
  };
      /** Defines the color scheme applied to the confirm button
       * @default default
       */
      style?: 'danger' | 'default' | 'primary' | Expression<string>;
    };
  };
    }>;
  };
      /** A string acting as a unique identifier for a block. You can use this block_id when you receive an interaction payload to identify the source of the action. If not specified, a block_id will be generated. Maximum length for this field is 255 characters.
       * @displayOptions.show { type: ["section"] }
       */
      blockId?: string | Expression<string> | PlaceholderValue;
      /** Define the text of the button that cancels the action
       * @displayOptions.show { type: ["section"] }
       * @default {}
       */
      textUi?: {
        /** Text
     */
    textValue?: {
      /** The formatting to use for this text object
       * @default mrkwdn
       */
      type?: 'mrkwdn' | 'plainText' | Expression<string>;
      /** The text for the block. This field accepts any of the standard text formatting markup when type is mrkdwn.
       */
      text?: string | Expression<string> | PlaceholderValue;
      /** Whether emojis in a text field should be escaped into the colon emoji format. This field is only usable when type is plain_text.
       * @displayOptions.show { type: ["plainText"] }
       * @default false
       */
      emoji?: boolean | Expression<boolean>;
      /** Whether to set to false (as is default) URLs will be auto-converted into links, conversation names will be link-ified, and certain mentions will be automatically parsed
       * @displayOptions.show { type: ["mrkwdn"] }
       * @default false
       */
      verbatim?: boolean | Expression<boolean>;
    };
  };
      /** An array of text objects. Any text objects included with fields will be rendered in a compact format that allows for 2 columns of side-by-side text. Maximum number of items is 10.
       * @displayOptions.show { type: ["section"] }
       * @default {}
       */
      fieldsUi?: {
        /** Text
     */
    fieldsValues?: Array<{
      /** The formatting to use for this text object
       * @default mrkwdn
       */
      type?: 'mrkwdn' | 'plainText' | Expression<string>;
      /** The text for the block. This field accepts any of the standard text formatting markup when type is mrkdwn.
       */
      text?: string | Expression<string> | PlaceholderValue;
      /** Whether emojis in a text field should be escaped into the colon emoji format. This field is only usable when type is plain_text.
       * @displayOptions.show { type: ["plainText"] }
       * @default false
       */
      emoji?: boolean | Expression<boolean>;
      /** When set to false (as is default) URLs will be auto-converted into links, conversation names will be link-ified, and certain mentions will be automatically parsed
       * @displayOptions.show { type: ["mrkwdn"] }
       * @default false
       */
      verbatim?: boolean | Expression<boolean>;
    }>;
  };
      /** Accessory
       * @displayOptions.show { type: ["section"] }
       * @default {}
       */
      accessoryUi?: {
        /** Accessory
     */
    accessoriesValues?: {
      /** The type of element
       * @default button
       */
      type?: 'button' | Expression<string>;
      /** The text for the block
       * @displayOptions.show { type: ["button"] }
       */
      text?: string | Expression<string> | PlaceholderValue;
      /** Whether emojis in a text field should be escaped into the colon emoji format
       * @displayOptions.show { type: ["button"] }
       * @default false
       */
      emoji?: boolean | Expression<boolean>;
      /** An identifier for this action. You can use this when you receive an interaction payload to identify the source of the action. Should be unique among all other action_ids used elsewhere by your app.
       * @displayOptions.show { type: ["button"] }
       */
      actionId?: string | Expression<string> | PlaceholderValue;
      /** A URL to load in the user's browser when the button is clicked. Maximum length for this field is 3000 characters. If you're using URL, you'll still receive an interaction payload and will need to send an acknowledgement response.
       * @displayOptions.show { type: ["button"] }
       */
      url?: string | Expression<string> | PlaceholderValue;
      /** The value to send along with the interaction payload
       * @displayOptions.show { type: ["button"] }
       */
      value?: string | Expression<string> | PlaceholderValue;
      /** Decorates buttons with alternative visual color schemes
       * @displayOptions.show { type: ["button"] }
       * @default default
       */
      style?: 'danger' | 'default' | 'primary' | Expression<string>;
      /** Defines an optional confirmation dialog after the button is clicked
       * @displayOptions.show { type: ["button"] }
       * @default {}
       */
      confirmUi?: {
        /** Confirm
     */
    confirmValue?: {
      /** Defines an optional confirmation dialog after the button is clicked
       * @default {}
       */
      titleUi?: {
        /** Title
     */
    titleValue?: {
      /** Text of the title
       */
      text?: string | Expression<string> | PlaceholderValue;
      /** Whether emojis in a text field should be escaped into the colon emoji format
       * @default false
       */
      emoji?: boolean | Expression<boolean>;
    };
  };
      /** Defines the explanatory text that appears in the confirm dialog
       * @default {}
       */
      textUi?: {
        /** Text
     */
    textValue?: {
      /** The text for the block
       */
      text?: string | Expression<string> | PlaceholderValue;
      /** Whether emojis in a text field should be escaped into the colon emoji format
       * @default false
       */
      emoji?: boolean | Expression<boolean>;
    };
  };
      /** Defines the explanatory text that appears in the confirm dialog
       * @default {}
       */
      confirmTextUi?: {
        /** Confirm
     */
    confirmValue?: {
      /** Defines the explanatory text that appears in the confirm dialog
       */
      text?: string | Expression<string> | PlaceholderValue;
      /** Whether emojis in a text field should be escaped into the colon emoji format
       * @default false
       */
      emoji?: boolean | Expression<boolean>;
    };
  };
      /** Define the text of the button that cancels the action
       * @default {}
       */
      denyUi?: {
        /** Deny
     */
    denyValue?: {
      /** Define the text of the button that cancels the action
       */
      text?: string | Expression<string> | PlaceholderValue;
      /** Whether emojis in a text field should be escaped into the colon emoji format
       * @default false
       */
      emoji?: boolean | Expression<boolean>;
    };
  };
      /** Defines the color scheme applied to the confirm button
       * @default default
       */
      style?: 'danger' | 'default' | 'primary' | Expression<string>;
    };
  };
    };
  };
    }>;
  };
/**
 * The attachments to add
 * @displayOptions.show { jsonParameters: [true] }
 */
    attachmentsJson?: IDataObject | string | Expression<string>;
/**
 * The blocks to add
 * @displayOptions.show { jsonParameters: [true] }
 */
    blocksJson?: IDataObject | string | Expression<string>;
};

export type SlackV1MessagePostNode = {
  type: 'n8n-nodes-base.slack';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SlackV1MessagePostParams>;
};
