// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/slack/v1/resource_message/operation_post_ephemeral.schema.js
/**
 * Slack Node - Version 1 - Zod Schema
 * Discriminator: resource=message, operation=postEphemeral
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('message').default('message'),
      operation: z.literal('postEphemeral'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      channel: stringOrExpression.optional(),
      user: stringOrExpression.optional(),
      text: stringOrExpression.optional(),
      jsonParameters: booleanOrExpression.optional(),
      otherOptions: z.object({ icon_emoji: stringOrExpression.optional(), icon_url: stringOrExpression.optional(), link_names: booleanOrExpression.optional(), thread_ts: stringOrExpression.optional(), mrkdwn: booleanOrExpression.optional(), reply_broadcast: booleanOrExpression.optional(), unfurl_links: booleanOrExpression.optional(), unfurl_media: booleanOrExpression.optional(), sendAsUser: stringOrExpression.optional() }).optional(),
      attachments: z.object({ fallback: stringOrExpression.optional(), text: stringOrExpression.optional(), title: stringOrExpression.optional(), title_link: stringOrExpression.optional(), color: stringOrExpression.optional(), pretext: stringOrExpression.optional(), author_name: stringOrExpression.optional(), author_link: stringOrExpression.optional(), author_icon: stringOrExpression.optional(), image_url: stringOrExpression.optional(), thumb_url: stringOrExpression.optional(), footer: stringOrExpression.optional(), footer_icon: stringOrExpression.optional(), ts: stringOrExpression.optional(), fields: z.unknown().optional() }).optional(),
    }).optional(),
  });
};
