// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/slack/v1/resource_message/operation_update.schema.js
/**
 * Slack Node - Version 1 - Zod Schema
 * Discriminator: resource=message, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('message').default('message'),
      operation: z.literal('update'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      channelId: stringOrExpression.optional(),
      text: stringOrExpression.optional(),
      ts: stringOrExpression.optional(),
      jsonParameters: booleanOrExpression.optional(),
      updateFields: z.object({ link_names: booleanOrExpression.optional(), parse: z.union([z.literal('client'), z.literal('full'), z.literal('none'), expressionSchema]).optional() }).optional(),
      attachmentsJson: resolveSchema({ parameters, schema: z.union([iDataObjectSchema, z.string()]), required: false, displayOptions: {"show":{"jsonParameters":[true]}}, defaults: {"jsonParameters":false} }),
      blocksJson: resolveSchema({ parameters, schema: z.union([iDataObjectSchema, z.string()]), required: false, displayOptions: {"show":{"jsonParameters":[true]}}, defaults: {"jsonParameters":false} }),
      attachments: resolveSchema({ parameters, schema: z.object({ fallback: stringOrExpression.optional(), text: stringOrExpression.optional(), title: stringOrExpression.optional(), title_link: stringOrExpression.optional(), color: stringOrExpression.optional(), pretext: stringOrExpression.optional(), author_name: stringOrExpression.optional(), author_link: stringOrExpression.optional(), author_icon: stringOrExpression.optional(), image_url: stringOrExpression.optional(), thumb_url: stringOrExpression.optional(), footer: stringOrExpression.optional(), footer_icon: stringOrExpression.optional(), ts: stringOrExpression.optional(), fields: z.unknown().optional() }), required: false, displayOptions: {"show":{"jsonParameters":[false]}}, defaults: {"jsonParameters":false} }),
    }).optional(),
  });
};
