// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleBooks/v1/resource_volume/operation_get_all.ts
/**
 * Google Books Node - Version 1
 * Discriminator: resource=volume, operation=getAll
 */


interface Credentials {
  googleApi: CredentialReference;
  googleBooksOAuth2Api: CredentialReference;
}

/** Get many public bookshelf resource for the specified user */
export type GoogleBooksV1VolumeGetAllParams = {
  resource: 'volume';
  operation: 'getAll';
/**
 * Authentication
 * @default serviceAccount
 */
    authentication?: 'serviceAccount' | 'oAuth2' | Expression<string>;
/**
 * Full-text search query string
 */
    searchQuery?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 40
 */
    limit?: number | Expression<number>;
};

export type GoogleBooksV1VolumeGetAllNode = {
  type: 'n8n-nodes-base.googleBooks';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GoogleBooksV1VolumeGetAllParams>;
};
