// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleBigQuery/v2.ts
/**
 * Google BigQuery Node - Version 2
 * Consume Google BigQuery API
 */


export interface GoogleBigQueryV2Params {
  authentication?: 'oAuth2' | 'serviceAccount';
  resource?: unknown;
/**
 * Operation
 * @displayOptions.show { resource: ["database"] }
 * @default executeQuery
 */
    operation?: 'executeQuery' | 'insert';
/**
 * Projects to which you have been granted any project role
 * @displayOptions.show { resource: ["database"], operation: ["executeQuery", "insert"] }
 * @default {"mode":"list","value":""}
 */
    projectId?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
/**
 * Dataset
 * @displayOptions.show { resource: ["database"], operation: ["insert"] }
 * @default {"mode":"list","value":""}
 */
    datasetId?: { __rl: true; mode: 'list' | 'id'; value: string; cachedResultName?: string };
/**
 * Table
 * @displayOptions.show { resource: ["database"], operation: ["insert"] }
 * @default {"mode":"list","value":""}
 */
    tableId?: { __rl: true; mode: 'list' | 'id'; value: string; cachedResultName?: string };
/**
 * SQL query to execute, you can find more information &lt;a href="https://cloud.google.com/bigquery/docs/reference/standard-sql/query-syntax" target="_blank"&gt;here&lt;/a&gt;. Standard SQL syntax used by default, but you can also use Legacy SQL syntax by using optinon 'Use Legacy SQL'.
 * @displayOptions.show { resource: ["database"], operation: ["executeQuery"] }
 * @displayOptions.hide { /options.useLegacySql: [true] }
 */
    sqlQuery?: string;
/**
 * Options
 * @displayOptions.show { resource: ["database"], operation: ["executeQuery"] }
 * @default {}
 */
    options?: {
    /** If not set, all table names in the query string must be qualified in the format 'datasetId.tableId'. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    defaultDataset?: string | Expression<string>;
    /** Whether set to true BigQuery doesn't run the job. Instead, if the query is valid, BigQuery returns statistics about the job such as how many bytes would be processed. If the query is invalid, an error returns.
     * @default false
     */
    dryRun?: boolean | Expression<boolean>;
    /** Whether to include the schema in the output. If set to true, the output will contain key '_schema' with the schema of the table.
     * @displayOptions.hide { rawOutput: [true] }
     * @default false
     */
    includeSchema?: boolean | Expression<boolean>;
    /** Location or the region where data would be stored and processed. Pricing for storage and analysis is also defined by location of data and reservations, more information &lt;a href="https://cloud.google.com/bigquery/docs/locations" target="_blank"&gt;here&lt;/a&gt;.
     */
    location?: string | Expression<string> | PlaceholderValue;
    /** Limits the bytes billed for this query. Queries with bytes billed above this limit will fail (without incurring a charge). String in &lt;a href="https://developers.google.com/discovery/v1/type-format?utm_source=cloud.google.com&utm_medium=referral" target="_blank"&gt;Int64Value&lt;/a&gt; format
     */
    maximumBytesBilled?: string | Expression<string> | PlaceholderValue;
    /** Maximum number of results to return per page of results. This is particularly useful when dealing with large datasets. It will not affect the total number of results returned, e.g. rows in a table. You can use LIMIT in your SQL query to limit the number of rows returned.
     * @default 1000
     */
    maxResults?: number | Expression<number>;
    /** Specifies the maximum amount of time, in milliseconds, that the client is willing to wait for the query to complete. Be aware that the call is not guaranteed to wait for the specified timeout; it typically returns after around 200 seconds (200,000 milliseconds), even if the query is not complete.
     * @hint How long to wait for the query to complete, in milliseconds
     * @default 10000
     */
    timeoutMs?: number | Expression<number>;
    /** Raw Output
     * @displayOptions.hide { dryRun: [true] }
     * @default false
     */
    rawOutput?: boolean | Expression<boolean>;
    /** Whether to use BigQuery's legacy SQL dialect for this query. If set to false, the query will use BigQuery's standard SQL.
     * @default false
     */
    useLegacySql?: boolean | Expression<boolean>;
    /** Whether all integer values will be returned as numbers. If set to false, all integer values will be returned as strings.
     * @default false
     */
    returnAsNumbers?: boolean | Expression<boolean>;
    /** Use &lt;a href="https://cloud.google.com/bigquery/docs/parameterized-queries#using_structs_in_parameterized_queries" target="_blank"&gt;parameterized queries&lt;/a&gt; to prevent SQL injections. Positional arguments are not supported at the moment. This feature won't be available when using legacy SQL.
     * @displayOptions.hide { /options.useLegacySql: [true] }
     * @default {"namedParameters":[{"name":"","value":""}]}
     */
    queryParameters?: {
        /** Named Parameter
     */
    namedParameters?: Array<{
      /** Name of the parameter
       */
      name?: string | Expression<string> | PlaceholderValue;
      /** The substitute value. It must be a string. Arrays, dates and struct types mentioned in &lt;a href="https://cloud.google.com/bigquery/docs/parameterized-queries#using_structs_in_parameterized_queries" target="_blank"&gt;the official documentation&lt;/a&gt; are not yet supported.
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
    /** Batch Size
     * @default 100
     */
    batchSize?: number | Expression<number>;
    /** Whether to gnore row values that do not match the schema
     * @default false
     */
    ignoreUnknownValues?: boolean | Expression<boolean>;
    /** Whether to skip rows with values that do not match the schema
     * @default false
     */
    skipInvalidRows?: boolean | Expression<boolean>;
    /** Create a new table based on the destination table and insert rows into the new table. The new table will be named &lt;code&gt;{destinationTable}{templateSuffix}&lt;/code&gt;
     */
    templateSuffix?: string | Expression<string> | PlaceholderValue;
    /** Unique ID for the request, for debugging only. It is case-sensitive, limited to up to 36 ASCII characters. A UUID is recommended.
     */
    traceId?: string | Expression<string> | PlaceholderValue;
  };
/**
 * Whether to insert the input data this node receives in the new row
 * @displayOptions.show { resource: ["database"], operation: ["insert"] }
 * @default autoMap
 */
    dataMode?: 'autoMap' | 'define' | Expression<string>;
/**
 * Fields to Send
 * @displayOptions.show { dataMode: ["define"], resource: ["database"], operation: ["insert"] }
 * @default {}
 */
    fieldsUi?: {
        /** Field
     */
    values?: Array<{
      /** Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;
       */
      fieldId?: string | Expression<string>;
      /** Field Value
       */
      fieldValue?: string | Expression<string> | PlaceholderValue;
    }>;
  };
}

export interface GoogleBigQueryV2Credentials {
  googleApi: CredentialReference;
  googleBigQueryOAuth2Api: CredentialReference;
}

interface GoogleBigQueryV2NodeBase {
  type: 'n8n-nodes-base.googleBigQuery';
  version: 2;
  credentials?: GoogleBigQueryV2Credentials;
}

export type GoogleBigQueryV2ParamsNode = GoogleBigQueryV2NodeBase & {
  config: NodeConfig<GoogleBigQueryV2Params>;
};

export type GoogleBigQueryV2Node = GoogleBigQueryV2ParamsNode;
