// @n8n/ai-workflow-builder/dist/ai-workflow-builder-agent.service.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var AiWorkflowBuilderService_1;
Object.defineProperty(exports, "__esModule", { value: true });
exports.AiWorkflowBuilderService = void 0;
const messages_1 = require("@langchain/core/messages");
const tracer_langchain_1 = require("@langchain/core/tracers/tracer_langchain");
const backend_common_1 = require("@n8n/backend-common");
const di_1 = require("@n8n/di");
const ai_assistant_sdk_1 = require("@n8n_io/ai-assistant-sdk");
const assert_1 = __importDefault(require("assert"));
const langsmith_1 = require("langsmith");
const zod_1 = require("zod");
const assistant_1 = require("./assistant");
const errors_1 = require("./errors");
const llm_config_1 = require("./llm-config");
const session_manager_service_1 = require("./session-manager.service");
const workflow_builder_agent_1 = require("./workflow-builder-agent");
let AiWorkflowBuilderService = class AiWorkflowBuilderService {
    static { AiWorkflowBuilderService_1 = this; }
    client;
    logger;
    instanceId;
    instanceUrl;
    n8nVersion;
    onCreditsUpdated;
    onTelemetryEvent;
    nodeDefinitionDirs;
    resourceLocatorCallbackFactory;
    nodeTypes;
    sessionManager;
    constructor(parsedNodeTypes, sessionStorage, client, logger, instanceId, instanceUrl, n8nVersion, onCreditsUpdated, onTelemetryEvent, nodeDefinitionDirs, resourceLocatorCallbackFactory) {
        this.client = client;
        this.logger = logger;
        this.instanceId = instanceId;
        this.instanceUrl = instanceUrl;
        this.n8nVersion = n8nVersion;
        this.onCreditsUpdated = onCreditsUpdated;
        this.onTelemetryEvent = onTelemetryEvent;
        this.nodeDefinitionDirs = nodeDefinitionDirs;
        this.resourceLocatorCallbackFactory = resourceLocatorCallbackFactory;
        this.nodeTypes = this.filterNodeTypes(parsedNodeTypes);
        this.sessionManager = new session_manager_service_1.SessionManagerService(this.nodeTypes, sessionStorage, logger);
    }
    updateNodeTypes(nodeTypes) {
        this.nodeTypes = this.filterNodeTypes(nodeTypes);
        this.sessionManager.updateNodeTypes(this.nodeTypes);
    }
    static async getAnthropicClaudeModel({ baseUrl, authHeaders = {}, apiKey = '-', } = {}) {
        return await (0, llm_config_1.anthropicClaudeSonnet45)({
            baseUrl,
            apiKey,
            headers: {
                ...authHeaders,
                'anthropic-beta': 'prompt-caching-2024-07-31',
            },
        });
    }
    async getApiProxyAuthHeaders(user, userMessageId) {
        (0, assert_1.default)(this.client);
        const authResponse = await this.client.getBuilderApiProxyToken(user, { userMessageId });
        const authHeaders = {
            Authorization: `${authResponse.tokenType} ${authResponse.accessToken}`,
        };
        return authHeaders;
    }
    async setupModels(user, userMessageId) {
        try {
            if (this.client) {
                const authHeaders = await this.getApiProxyAuthHeaders(user, userMessageId);
                const baseUrl = this.client.getApiProxyBaseUrl();
                const modelConfig = {
                    baseUrl: baseUrl + '/anthropic',
                    authHeaders,
                };
                const anthropicClaude = await AiWorkflowBuilderService_1.getAnthropicClaudeModel(modelConfig);
                const tracingClient = new langsmith_1.Client({
                    apiKey: '-',
                    apiUrl: baseUrl + '/langsmith',
                    autoBatchTracing: false,
                    traceBatchConcurrency: 1,
                    fetchOptions: {
                        headers: {
                            ...authHeaders,
                        },
                    },
                });
                return { tracingClient, anthropicClaude, authHeaders };
            }
            const envConfig = { apiKey: process.env.N8N_AI_ANTHROPIC_KEY ?? '' };
            const anthropicClaude = await AiWorkflowBuilderService_1.getAnthropicClaudeModel(envConfig);
            return { anthropicClaude };
        }
        catch (error) {
            const errorMessage = error instanceof Error ? `: ${error.message}` : '';
            const llmError = new errors_1.LLMServiceError(`Failed to connect to LLM Provider${errorMessage}`, {
                cause: error,
                tags: {
                    hasClient: !!this.client,
                    hasUser: !!user,
                },
            });
            throw llmError;
        }
    }
    filterNodeTypes(nodeTypes) {
        const ignoredTypes = new Set([
            '@n8n/n8n-nodes-langchain.toolVectorStore',
            '@n8n/n8n-nodes-langchain.documentGithubLoader',
            '@n8n/n8n-nodes-langchain.code',
        ]);
        const isBuiltInNode = (name) => name.startsWith('n8n-nodes-base.') || name.startsWith('@n8n/');
        const visibleNodeTypes = nodeTypes.filter((nodeType) => isBuiltInNode(nodeType.name) &&
            !ignoredTypes.has(nodeType.name) &&
            (nodeType.hidden !== true || nodeType.name === 'n8n-nodes-base.dataTable'));
        return visibleNodeTypes.map((nodeType) => {
            const isTool = nodeType.name.endsWith('Tool');
            if (!isTool)
                return nodeType;
            const nonToolNode = nodeTypes.find((nt) => nt.name === nodeType.name.replace('Tool', ''));
            if (!nonToolNode)
                return nodeType;
            return {
                ...nonToolNode,
                ...nodeType,
            };
        });
    }
    async getAgent(user, userMessageId, featureFlags) {
        const { anthropicClaude, tracingClient, authHeaders } = await this.setupModels(user, userMessageId);
        const resourceLocatorCallback = this.resourceLocatorCallbackFactory?.(user.id);
        const assistantHandler = this.client
            ? new assistant_1.AssistantHandler(this.client, this.logger)
            : undefined;
        const agent = new workflow_builder_agent_1.WorkflowBuilderAgent({
            parsedNodeTypes: this.nodeTypes,
            stageLLMs: {
                supervisor: anthropicClaude,
                responder: anthropicClaude,
                discovery: anthropicClaude,
                builder: anthropicClaude,
                parameterUpdater: anthropicClaude,
                planner: anthropicClaude,
            },
            logger: this.logger,
            checkpointer: this.sessionManager.getCheckpointer(),
            tracer: tracingClient
                ? new tracer_langchain_1.LangChainTracer({
                    client: tracingClient,
                    projectName: featureFlags?.codeBuilder
                        ? 'code-workflow-builder'
                        : 'n8n-workflow-builder',
                })
                : undefined,
            instanceUrl: this.instanceUrl,
            runMetadata: {
                n8nVersion: this.n8nVersion,
                featureFlags: featureFlags ?? {},
            },
            onGenerationSuccess: async () => await this.onGenerationSuccess(user, authHeaders),
            nodeDefinitionDirs: this.nodeDefinitionDirs,
            resourceLocatorCallback,
            onTelemetryEvent: this.onTelemetryEvent,
            assistantHandler,
        });
        return { agent };
    }
    async onGenerationSuccess(user, authHeaders) {
        try {
            if (this.client) {
                (0, assert_1.default)(authHeaders, 'Auth headers must be set when AI Assistant Service client is used');
                (0, assert_1.default)(user);
                const creditsInfo = await this.client.markBuilderSuccess(user, authHeaders);
                if (this.onCreditsUpdated && user.id && creditsInfo) {
                    this.onCreditsUpdated(user.id, creditsInfo.creditsQuota, creditsInfo.creditsClaimed);
                }
            }
        }
        catch (error) {
            if (error instanceof Error) {
                this.logger?.error(`Unable to mark generation success ${error.message}`, { error });
            }
        }
    }
    async *chat(payload, user, abortSignal) {
        const { agent } = await this.getAgent(user, payload.id, payload.featureFlags);
        const userId = user?.id?.toString();
        const workflowId = payload.workflowContext?.currentWorkflow?.id;
        const isCodeBuilder = payload.featureFlags?.codeBuilder ?? false;
        const threadId = session_manager_service_1.SessionManagerService.generateThreadId(workflowId, userId);
        const historicalMessages = await this.loadSessionMessagesSafe(threadId);
        const pendingHitl = payload.resumeData
            ? this.sessionManager.getAndClearPendingHitl(threadId)
            : undefined;
        if (pendingHitl && payload.resumeData) {
            this.storeHitlInteraction(threadId, pendingHitl, payload.resumeData);
        }
        const resumeInterrupt = pendingHitl?.value;
        const agentPayload = resumeInterrupt ? { ...payload, resumeInterrupt } : payload;
        for await (const output of agent.chat(agentPayload, userId, abortSignal, undefined, historicalMessages)) {
            const streamHitl = this.extractHitlFromStreamOutput(output);
            if (streamHitl) {
                this.sessionManager.setPendingHitl(threadId, streamHitl, payload.id);
            }
            yield output;
        }
        await this.saveSessionSafe(agent, workflowId, userId, threadId);
        if (this.onTelemetryEvent && userId) {
            try {
                await this.trackBuilderReplyTelemetry(agent, workflowId, userId, payload.id, threadId, isCodeBuilder);
            }
            catch (error) {
                this.logger?.error('Failed to track builder reply telemetry', { error });
            }
        }
    }
    storeHitlInteraction(threadId, pendingHitl, resumeData) {
        if (pendingHitl.value.type === 'questions') {
            this.sessionManager.addHitlEntry(threadId, {
                type: 'questions_answered',
                afterMessageId: pendingHitl.triggeringMessageId,
                interrupt: pendingHitl.value,
                answers: resumeData,
            });
        }
        else if (pendingHitl.value.type === 'plan') {
            const decision = resumeData;
            if (decision.action === 'reject' || decision.action === 'modify') {
                this.sessionManager.addHitlEntry(threadId, {
                    type: 'plan_decided',
                    afterMessageId: pendingHitl.triggeringMessageId,
                    plan: pendingHitl.value.plan,
                    decision: decision.action,
                    feedback: decision.feedback,
                });
            }
        }
    }
    extractLastAiMessageContent(messages) {
        const lastAiMessage = messages.findLast((m) => m instanceof messages_1.AIMessage);
        return typeof lastAiMessage?.content === 'string'
            ? lastAiMessage.content
            : JSON.stringify(lastAiMessage?.content ?? '');
    }
    extractUniqueToolNames(messages) {
        const toolMessages = messages.filter((m) => m instanceof messages_1.ToolMessage);
        return [
            ...new Set(toolMessages
                .map((m) => m.name)
                .filter((name) => name !== undefined)),
        ];
    }
    async trackBuilderReplyTelemetry(agent, workflowId, userId, userMessageId, threadId, isCodeBuilder) {
        if (!this.onTelemetryEvent)
            return;
        const state = await agent.getState(workflowId, userId);
        const messages = state?.values?.messages ?? [];
        const properties = {
            user_id: userId,
            instance_id: this.instanceId,
            workflow_id: workflowId,
            sequence_id: threadId,
            message_ai: this.extractLastAiMessageContent(messages),
            tools_called: this.extractUniqueToolNames(messages),
            techniques_categories: state?.values?.techniqueCategories,
            validations: state?.values?.validationHistory,
            ...((state?.values?.templateIds?.length ?? 0) > 0 && {
                templates_selected: state.values.templateIds,
            }),
            user_message_id: userMessageId,
            code_builder: isCodeBuilder,
        };
        this.onTelemetryEvent('Builder replied to user message', properties);
    }
    async getSessions(workflowId, user, codeBuilder) {
        const userId = user?.id?.toString();
        const agentType = codeBuilder ? 'code-builder' : undefined;
        return await this.sessionManager.getSessions(workflowId, userId, agentType);
    }
    async getBuilderInstanceCredits(user) {
        if (this.client) {
            return await this.client.getBuilderInstanceCredits(user);
        }
        return {
            creditsQuota: -1,
            creditsClaimed: 0,
        };
    }
    async truncateMessagesAfter(workflowId, user, messageId, codeBuilder) {
        const agentType = codeBuilder ? 'code-builder' : undefined;
        return await this.sessionManager.truncateMessagesAfter(workflowId, user.id, messageId, agentType);
    }
    async loadSessionMessagesSafe(threadId) {
        try {
            return await this.sessionManager.loadSessionMessages(threadId);
        }
        catch (error) {
            this.logger?.error('Failed to load session messages from storage, continuing without history', { threadId, error });
            return [];
        }
    }
    async saveSessionSafe(agent, workflowId, userId, threadId) {
        try {
            const state = await agent.getState(workflowId, userId);
            const previousSummary = state?.values?.previousSummary;
            await this.sessionManager.saveSessionFromCheckpointer(threadId, previousSummary);
        }
        catch (error) {
            this.logger?.error('Failed to save session to persistent storage', { threadId, error });
        }
    }
    async clearSession(workflowId, user) {
        await this.sessionManager.clearAllSessions(workflowId, user.id);
    }
    static questionsInterruptSchema = zod_1.z.object({
        type: zod_1.z.literal('questions'),
        introMessage: zod_1.z.string().optional(),
        questions: zod_1.z.array(zod_1.z.object({
            id: zod_1.z.string(),
            question: zod_1.z.string(),
            type: zod_1.z.enum(['single', 'multi', 'text']),
            options: zod_1.z.array(zod_1.z.string()).optional(),
        })),
    });
    static planInterruptSchema = zod_1.z.object({
        type: zod_1.z.literal('plan'),
        plan: zod_1.z.object({
            summary: zod_1.z.string(),
            trigger: zod_1.z.string(),
            steps: zod_1.z.array(zod_1.z.object({
                description: zod_1.z.string(),
                subSteps: zod_1.z.array(zod_1.z.string()).optional(),
                suggestedNodes: zod_1.z.array(zod_1.z.string()).optional(),
            })),
            additionalSpecs: zod_1.z.array(zod_1.z.string()).optional(),
        }),
    });
    extractHitlFromStreamOutput(output) {
        if (typeof output !== 'object' || output === null)
            return null;
        if (!('messages' in output))
            return null;
        const messages = output.messages;
        if (!Array.isArray(messages))
            return null;
        for (const message of messages) {
            if (typeof message !== 'object' || message === null)
                continue;
            const m = message;
            if (m.type === 'questions') {
                const parsed = AiWorkflowBuilderService_1.questionsInterruptSchema.safeParse(m);
                if (parsed.success)
                    return parsed.data;
                this.logger?.warn('[HITL] Invalid questions interrupt data', {
                    errors: parsed.error.errors,
                });
                continue;
            }
            if (m.type === 'plan') {
                const parsed = AiWorkflowBuilderService_1.planInterruptSchema.safeParse(m);
                if (parsed.success)
                    return parsed.data;
                this.logger?.warn('[HITL] Invalid plan interrupt data', {
                    errors: parsed.error.errors,
                });
                continue;
            }
        }
        return null;
    }
};
exports.AiWorkflowBuilderService = AiWorkflowBuilderService;
exports.AiWorkflowBuilderService = AiWorkflowBuilderService = AiWorkflowBuilderService_1 = __decorate([
    (0, di_1.Service)(),
    __metadata("design:paramtypes", [Array, Object, ai_assistant_sdk_1.AiAssistantClient,
        backend_common_1.Logger, String, String, String, Function, Function, Array, Function])
], AiWorkflowBuilderService);
