// @n8n/ai-workflow-builder/dist/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ASSISTANT_SDK_TIMEOUT_MS = exports.STREAM_SEPARATOR = exports.MAX_MULTI_AGENT_STREAM_ITERATIONS = exports.MAX_TRIAGE_ITERATIONS = exports.MAX_DISCOVERY_ITERATIONS = exports.MAX_BUILDER_ITERATIONS = exports.MAX_AI_RESPONSE_CHARS = exports.MAX_EXECUTION_DATA_CHARS = exports.MAX_NODE_EXAMPLE_CHARS = exports.AVG_CHARS_PER_TOKEN_ANTHROPIC = exports.MAX_WORKFLOW_LENGTH_TOKENS = exports.DEFAULT_AUTO_COMPACT_THRESHOLD_TOKENS = exports.MAX_PARAMETER_VALUE_LENGTH = exports.MAX_INPUT_TOKENS = exports.MAX_OUTPUT_TOKENS = exports.MAX_TOTAL_TOKENS = exports.MAX_AI_BUILDER_PROMPT_LENGTH = void 0;
exports.MAX_AI_BUILDER_PROMPT_LENGTH = 5000;
exports.MAX_TOTAL_TOKENS = 200_000;
exports.MAX_OUTPUT_TOKENS = 16_000;
exports.MAX_INPUT_TOKENS = exports.MAX_TOTAL_TOKENS - exports.MAX_OUTPUT_TOKENS - 5_000;
exports.MAX_PARAMETER_VALUE_LENGTH = 30_000;
exports.DEFAULT_AUTO_COMPACT_THRESHOLD_TOKENS = exports.MAX_TOTAL_TOKENS - 50_000;
exports.MAX_WORKFLOW_LENGTH_TOKENS = 30_000;
exports.AVG_CHARS_PER_TOKEN_ANTHROPIC = 3.5;
exports.MAX_NODE_EXAMPLE_CHARS = 5000 * exports.AVG_CHARS_PER_TOKEN_ANTHROPIC;
exports.MAX_EXECUTION_DATA_CHARS = 10000;
exports.MAX_AI_RESPONSE_CHARS = 500;
exports.MAX_BUILDER_ITERATIONS = 100;
exports.MAX_DISCOVERY_ITERATIONS = 50;
exports.MAX_TRIAGE_ITERATIONS = 10;
exports.MAX_MULTI_AGENT_STREAM_ITERATIONS = exports.MAX_BUILDER_ITERATIONS + exports.MAX_DISCOVERY_ITERATIONS;
exports.STREAM_SEPARATOR = '⧉⇋⇋➽⌑⧉§§\n';
exports.ASSISTANT_SDK_TIMEOUT_MS = 3 * 60 * 1000;
