// @n8n/ai-workflow-builder/dist/llm-config.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DEFAULT_MODEL = exports.AVAILABLE_MODELS = exports.MODEL_FACTORIES = exports.devstral = exports.gemini3Pro = exports.deepseekV32 = exports.gemini3Flash = exports.glm47 = exports.anthropicClaudeOpus45 = exports.anthropicHaiku45 = exports.anthropicClaudeSonnet45Think = exports.anthropicClaudeSonnet45 = exports.gpt52 = void 0;
exports.getApiKeyEnvVar = getApiKeyEnvVar;
const constants_1 = require("./constants");
const http_proxy_agent_1 = require("./utils/http-proxy-agent");
const gpt52 = async (config) => {
    const { ChatOpenAI } = await Promise.resolve().then(() => __importStar(require('@langchain/openai')));
    return new ChatOpenAI({
        model: 'gpt-5.2-2025-12-11',
        apiKey: config.apiKey,
        temperature: 0,
        maxTokens: -1,
        configuration: {
            baseURL: config.baseUrl,
            defaultHeaders: config.headers,
            fetchOptions: {
                dispatcher: (0, http_proxy_agent_1.getProxyAgent)(config.baseUrl ?? 'https://api.openai.com/v1'),
            },
        },
    });
};
exports.gpt52 = gpt52;
const anthropicClaudeSonnet45 = async (config) => {
    const { ChatAnthropic } = await Promise.resolve().then(() => __importStar(require('@langchain/anthropic')));
    const model = new ChatAnthropic({
        model: 'claude-sonnet-4-5-20250929',
        apiKey: config.apiKey,
        temperature: 0,
        maxTokens: constants_1.MAX_OUTPUT_TOKENS,
        anthropicApiUrl: config.baseUrl,
        clientOptions: {
            defaultHeaders: config.headers,
            fetchOptions: {
                dispatcher: (0, http_proxy_agent_1.getProxyAgent)(config.baseUrl),
            },
        },
    });
    delete model.topP;
    return model;
};
exports.anthropicClaudeSonnet45 = anthropicClaudeSonnet45;
const anthropicClaudeSonnet45Think = async (config) => {
    const { ChatAnthropic } = await Promise.resolve().then(() => __importStar(require('@langchain/anthropic')));
    const model = new ChatAnthropic({
        model: 'claude-sonnet-4-5-20250929',
        apiKey: config.apiKey,
        maxTokens: constants_1.MAX_OUTPUT_TOKENS,
        anthropicApiUrl: config.baseUrl,
        thinking: {
            budget_tokens: 10000,
            type: 'enabled',
        },
        clientOptions: {
            defaultHeaders: config.headers,
            fetchOptions: {
                dispatcher: (0, http_proxy_agent_1.getProxyAgent)(config.baseUrl),
            },
        },
    });
    delete model.topP;
    delete model.temperature;
    return model;
};
exports.anthropicClaudeSonnet45Think = anthropicClaudeSonnet45Think;
const anthropicHaiku45 = async (config) => {
    const { ChatAnthropic } = await Promise.resolve().then(() => __importStar(require('@langchain/anthropic')));
    const model = new ChatAnthropic({
        model: 'claude-haiku-4-5-20251001',
        apiKey: config.apiKey,
        temperature: 0,
        maxTokens: constants_1.MAX_OUTPUT_TOKENS,
        anthropicApiUrl: config.baseUrl,
        clientOptions: {
            defaultHeaders: config.headers,
            fetchOptions: {
                dispatcher: (0, http_proxy_agent_1.getProxyAgent)(config.baseUrl),
            },
        },
    });
    delete model.topP;
    return model;
};
exports.anthropicHaiku45 = anthropicHaiku45;
const anthropicClaudeOpus45 = async (config) => {
    const { ChatAnthropic } = await Promise.resolve().then(() => __importStar(require('@langchain/anthropic')));
    const model = new ChatAnthropic({
        model: 'claude-opus-4-5-20251101',
        apiKey: config.apiKey,
        temperature: 0,
        maxTokens: constants_1.MAX_OUTPUT_TOKENS,
        anthropicApiUrl: config.baseUrl,
        clientOptions: {
            defaultHeaders: config.headers,
            fetchOptions: {
                dispatcher: (0, http_proxy_agent_1.getProxyAgent)(config.baseUrl),
            },
        },
    });
    delete model.topP;
    return model;
};
exports.anthropicClaudeOpus45 = anthropicClaudeOpus45;
const OPENROUTER_BASE_URL = 'https://openrouter.ai/api/v1';
function createOpenRouterModel(modelName) {
    return async (config) => {
        const { ChatOpenAI } = await Promise.resolve().then(() => __importStar(require('@langchain/openai')));
        return new ChatOpenAI({
            model: modelName,
            apiKey: config.apiKey,
            temperature: 0,
            maxTokens: -1,
            configuration: {
                baseURL: OPENROUTER_BASE_URL,
                defaultHeaders: {
                    ...config.headers,
                    'HTTP-Referer': 'https://n8n.io',
                    'X-Title': 'n8n AI Workflow Builder',
                },
                fetchOptions: {
                    dispatcher: (0, http_proxy_agent_1.getProxyAgent)(OPENROUTER_BASE_URL),
                },
            },
        });
    };
}
exports.glm47 = createOpenRouterModel('thudm/glm-4-plus');
exports.gemini3Flash = createOpenRouterModel('google/gemini-3-flash-preview');
exports.deepseekV32 = createOpenRouterModel('deepseek/deepseek-chat-v3-0324');
exports.gemini3Pro = createOpenRouterModel('google/gemini-3-pro-preview');
exports.devstral = createOpenRouterModel('mistralai/devstral-small');
exports.MODEL_FACTORIES = {
    'claude-opus-4.5': exports.anthropicClaudeOpus45,
    'claude-sonnet-4.5': exports.anthropicClaudeSonnet45,
    'claude-sonnet-4.5-think': exports.anthropicClaudeSonnet45Think,
    'claude-haiku-4.5': exports.anthropicHaiku45,
    'gpt-5.2': exports.gpt52,
    'glm-4.7': exports.glm47,
    'gemini-3-flash': exports.gemini3Flash,
    'deepseek-v3.2': exports.deepseekV32,
    'gemini-3-pro': exports.gemini3Pro,
    devstral: exports.devstral,
};
const OPENROUTER_MODELS = [
    'glm-4.7',
    'gemini-3-flash',
    'deepseek-v3.2',
    'gemini-3-pro',
    'devstral',
];
function getApiKeyEnvVar(modelId) {
    if (OPENROUTER_MODELS.includes(modelId)) {
        return 'OPENROUTER_API_KEY';
    }
    if (modelId.startsWith('gpt')) {
        return 'N8N_AI_OPENAI_KEY';
    }
    return 'N8N_AI_ANTHROPIC_KEY';
}
exports.AVAILABLE_MODELS = [
    'claude-opus-4.5',
    'claude-sonnet-4.5',
    'claude-haiku-4.5',
    'gpt-5.2',
    'glm-4.7',
    'gemini-3-flash',
    'deepseek-v3.2',
    'gemini-3-pro',
    'devstral',
];
exports.DEFAULT_MODEL = 'claude-sonnet-4.5';
