// @n8n/ai-workflow-builder/dist/multi-agent-workflow-subgraphs.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createMultiAgentWorkflowWithSubgraphs = createMultiAgentWorkflowWithSubgraphs;
const messages_1 = require("@langchain/core/messages");
const langgraph_1 = require("@langchain/langgraph");
const responder_agent_1 = require("./agents/responder.agent");
const supervisor_agent_1 = require("./agents/supervisor.agent");
const constants_1 = require("./constants");
const parent_graph_state_1 = require("./parent-graph-state");
const builder_subgraph_1 = require("./subgraphs/builder.subgraph");
const discovery_subgraph_1 = require("./subgraphs/discovery.subgraph");
const coordination_1 = require("./types/coordination");
const coordination_log_1 = require("./utils/coordination-log");
const error_sanitizer_1 = require("./utils/error-sanitizer");
const operations_processor_1 = require("./utils/operations-processor");
const state_modifier_1 = require("./utils/state-modifier");
const subgraph_helpers_1 = require("./utils/subgraph-helpers");
function isCoordinationLogEntry(value) {
    if (typeof value !== 'object' || value === null)
        return false;
    const obj = value;
    return (typeof obj.phase === 'string' &&
        typeof obj.status === 'string' &&
        typeof obj.timestamp === 'number' &&
        typeof obj.summary === 'string');
}
function routeToNode(next) {
    const nodeMapping = {
        responder: 'responder',
        discovery: 'discovery_subgraph',
        builder: 'builder_subgraph',
        assistant: 'assistant_subgraph',
    };
    return nodeMapping[next] ?? 'responder';
}
function createSubgraphNodeHandler(phase, execute, logger) {
    return async (state, config) => {
        const startTimestamp = Date.now();
        logger?.debug(`[${phase}_subgraph] Starting ${phase}`);
        try {
            const { output, coordinationLog } = await execute(state, config);
            const validLog = coordinationLog.filter(isCoordinationLogEntry);
            return {
                ...output,
                coordinationLog: [
                    {
                        phase,
                        status: 'in_progress',
                        timestamp: startTimestamp,
                        summary: `Starting ${phase}`,
                        metadata: { phase },
                    },
                    ...validLog,
                ],
            };
        }
        catch (error) {
            if ((0, langgraph_1.isGraphInterrupt)(error)) {
                throw error;
            }
            logger?.error(`[${phase}_subgraph] ERROR:`, { error });
            const rawErrorMessage = error instanceof Error
                ? error.message
                : `An error occurred in ${phase}_subgraph: ${String(error)}`;
            const userFacingMessage = (0, error_sanitizer_1.sanitizeLlmErrorMessage)(error);
            return {
                nextPhase: 'responder',
                messages: [
                    new messages_1.HumanMessage({
                        content: `Error in ${phase}_subgraph: ${userFacingMessage}`,
                        name: 'system_error',
                    }),
                ],
                coordinationLog: [
                    {
                        phase,
                        status: 'in_progress',
                        timestamp: startTimestamp,
                        summary: `Starting ${phase}`,
                        metadata: { phase },
                    },
                    {
                        phase,
                        status: 'error',
                        timestamp: Date.now(),
                        summary: `Error: ${userFacingMessage}`,
                        metadata: (0, coordination_1.createErrorMetadata)({
                            failedSubgraph: phase,
                            errorMessage: rawErrorMessage,
                        }),
                    },
                ],
            };
        }
    };
}
function createCompiledSubgraphExecutor(subgraph, compiledGraph, recursionLimit) {
    return async (state, config) => {
        const input = subgraph.transformInput(state);
        const invokeConfig = { ...config, recursionLimit };
        const result = await compiledGraph.invoke(input, invokeConfig);
        const output = subgraph.transformOutput(result, state);
        const outputLogRaw = Array.isArray(output.coordinationLog) ? output.coordinationLog : [];
        return { output, coordinationLog: outputLogRaw };
    };
}
function createMultiAgentWorkflowWithSubgraphs(config) {
    const { parsedNodeTypes, stageLLMs, logger, instanceUrl, checkpointer, autoCompactThresholdTokens = constants_1.DEFAULT_AUTO_COMPACT_THRESHOLD_TOKENS, featureFlags, onGenerationSuccess, resourceLocatorCallback, assistantHandler, } = config;
    const mergeAskBuild = featureFlags?.mergeAskBuild === true && !!assistantHandler;
    const supervisorAgent = new supervisor_agent_1.SupervisorAgent({
        llm: stageLLMs.supervisor,
        mergeAskBuild,
    });
    const responderAgent = (0, responder_agent_1.createResponderAgent)({
        llm: stageLLMs.responder,
        enableIntrospection: featureFlags?.enableIntrospection,
        logger,
    });
    const discoverySubgraph = new discovery_subgraph_1.DiscoverySubgraph();
    const compiledDiscovery = discoverySubgraph.create({
        parsedNodeTypes,
        llm: stageLLMs.discovery,
        plannerLLM: stageLLMs.planner,
        logger,
        featureFlags,
    });
    const builderSubgraph = new builder_subgraph_1.BuilderSubgraph();
    const compiledBuilder = builderSubgraph.create({
        parsedNodeTypes,
        llm: stageLLMs.builder,
        llmParameterUpdater: stageLLMs.parameterUpdater,
        logger,
        instanceUrl,
        featureFlags,
        resourceLocatorCallback,
    });
    return (new langgraph_1.StateGraph(parent_graph_state_1.ParentGraphState)
        .addNode('supervisor', async (state, config) => {
        const conversationMessages = (0, subgraph_helpers_1.filterOutSubgraphToolMessages)(state.messages);
        const routing = await supervisorAgent.invoke({
            messages: conversationMessages,
            workflowJSON: state.workflowJSON,
            coordinationLog: state.coordinationLog,
            previousSummary: state.previousSummary,
            workflowContext: state.workflowContext,
        }, config);
        logger?.debug(`[supervisor] Routing to: ${routing.next}`, {
            reasoning: routing.reasoning,
        });
        return {
            nextPhase: routing.next,
        };
    })
        .addNode('responder', async (state, config) => {
        const startTimestamp = Date.now();
        const lastCompletedPhase = (0, coordination_log_1.getLastCompletedPhase)(state.coordinationLog);
        const assistantEntry = lastCompletedPhase === 'assistant'
            ? state.coordinationLog.findLast((e) => e.phase === 'assistant' && e.status === 'completed')
            : undefined;
        if (assistantEntry?.output) {
            logger?.debug('[responder] Forwarding assistant response (no credits consumed)', {
                outputLength: assistantEntry.output.length,
                outputPreview: assistantEntry.output.substring(0, 100),
            });
            return {
                coordinationLog: [
                    {
                        phase: 'responder',
                        status: 'completed',
                        timestamp: Date.now(),
                        summary: 'Assistant response forwarded (no credits consumed)',
                        metadata: (0, coordination_1.createResponderMetadata)({ responseLength: 0 }),
                    },
                ],
            };
        }
        const responderMessages = (0, subgraph_helpers_1.filterOutSubgraphToolMessages)(state.messages);
        const { response, introspectionEvents } = await (0, responder_agent_1.invokeResponderAgent)(responderAgent, {
            messages: responderMessages,
            coordinationLog: state.coordinationLog,
            discoveryContext: state.discoveryContext,
            workflowJSON: state.workflowJSON,
            workflowContext: state.workflowContext,
            previousSummary: state.previousSummary,
        }, config, { enableIntrospection: featureFlags?.enableIntrospection });
        if (onGenerationSuccess &&
            !(0, coordination_log_1.hasErrorInLog)(state.coordinationLog) &&
            (0, coordination_log_1.hasBuilderPhaseInLog)(state.coordinationLog)) {
            void Promise.resolve(onGenerationSuccess()).catch((error) => {
                logger?.warn('Failed to execute onGenerationSuccess callback', { error });
            });
        }
        const responseContent = typeof response.content === 'string'
            ? response.content
            : JSON.stringify(response.content);
        return {
            messages: [response],
            coordinationLog: [
                {
                    phase: 'responder',
                    status: 'in_progress',
                    timestamp: startTimestamp,
                    summary: 'Starting responder',
                    metadata: (0, coordination_1.createResponderMetadata)({ responseLength: 0 }),
                },
                {
                    phase: 'responder',
                    status: 'completed',
                    timestamp: Date.now(),
                    summary: `Generated response (${responseContent.length} chars)`,
                    metadata: (0, coordination_1.createResponderMetadata)({ responseLength: responseContent.length }),
                },
            ],
            introspectionEvents,
        };
    })
        .addNode('process_operations', (state) => {
        const result = (0, operations_processor_1.processOperations)(state);
        return {
            ...result,
            workflowOperations: [],
        };
    })
        .addNode('route_next_phase', (state) => {
        const next = (0, coordination_log_1.getNextPhaseFromLog)(state.coordinationLog);
        if (state.planDecision === 'reject') {
            return {
                nextPhase: 'responder',
                planDecision: null,
                planOutput: null,
                planFeedback: null,
                planPrevious: null,
            };
        }
        if (state.planDecision === 'modify') {
            return { nextPhase: 'discovery', planDecision: null, planOutput: null };
        }
        if (next === 'builder' &&
            featureFlags?.planMode === true &&
            state.mode === 'plan' &&
            !state.planOutput) {
            return { nextPhase: 'discovery', planDecision: null };
        }
        return { nextPhase: next, planDecision: null };
    })
        .addNode('check_state', (state) => {
        const action = (0, state_modifier_1.determineStateAction)(state, autoCompactThresholdTokens);
        if (action === 'continue' &&
            featureFlags?.planMode === true &&
            state.mode === 'plan' &&
            !state.planOutput &&
            !mergeAskBuild) {
            return { nextPhase: 'discovery' };
        }
        return { nextPhase: action };
    })
        .addNode('cleanup_dangling', (state) => (0, state_modifier_1.handleCleanupDangling)(state.messages, logger))
        .addNode('compact_messages', async (state, config) => {
        const isAutoCompact = state.messages[state.messages.length - 1]?.content !== '/compact';
        return await (0, state_modifier_1.handleCompactMessages)(state.messages, state.previousSummary ?? '', stageLLMs.responder, isAutoCompact, config);
    })
        .addNode('delete_messages', (state) => (0, state_modifier_1.handleDeleteMessages)(state.messages))
        .addNode('clear_error_state', (state) => (0, state_modifier_1.handleClearErrorState)(state.coordinationLog, logger))
        .addNode('create_workflow_name', async (state, config) => await (0, state_modifier_1.handleCreateWorkflowName)(state.messages, state.workflowJSON, stageLLMs.responder, logger, config))
        .addNode('discovery_subgraph', createSubgraphNodeHandler('discovery', createCompiledSubgraphExecutor(discoverySubgraph, compiledDiscovery, constants_1.MAX_DISCOVERY_ITERATIONS), logger))
        .addNode('builder_subgraph', createSubgraphNodeHandler('builder', createCompiledSubgraphExecutor(builderSubgraph, compiledBuilder, constants_1.MAX_BUILDER_ITERATIONS), logger))
        .addNode('assistant_subgraph', createSubgraphNodeHandler('assistant', async (state, nodeConfig) => {
        if (!assistantHandler) {
            throw new Error('Assistant handler not configured');
        }
        const query = (0, subgraph_helpers_1.extractUserRequest)(state.messages);
        const userId = nodeConfig?.configurable?.userId ?? 'unknown';
        const streamWriter = (0, langgraph_1.getWriter)(nodeConfig);
        let chunksDispatched = 0;
        const writer = (chunk) => {
            if (streamWriter) {
                chunksDispatched++;
                streamWriter(chunk);
            }
        };
        const abortController = new AbortController();
        const timeoutId = setTimeout(() => abortController.abort(), constants_1.ASSISTANT_SDK_TIMEOUT_MS);
        let result;
        try {
            result = await assistantHandler.execute({
                query,
                workflowJSON: state.workflowJSON,
                sdkSessionId: state.sdkSessionId,
            }, userId, writer, abortController.signal);
        }
        finally {
            clearTimeout(timeoutId);
        }
        logger?.debug('[assistant_subgraph] Handler completed', {
            responseTextLength: result.responseText.length,
            sdkSessionId: result.sdkSessionId,
            hasCodeDiff: result.hasCodeDiff,
            suggestionCount: result.suggestionIds.length,
            chunksDispatched,
        });
        return {
            output: {
                messages: [new messages_1.AIMessage({ content: result.responseText })],
                sdkSessionId: result.sdkSessionId,
            },
            coordinationLog: [
                {
                    phase: 'assistant',
                    status: 'completed',
                    timestamp: Date.now(),
                    summary: result.summary,
                    output: result.responseText,
                    metadata: (0, coordination_1.createAssistantMetadata)({
                        hasCodeDiff: result.hasCodeDiff,
                        suggestionCount: result.suggestionIds.length,
                    }),
                },
            ],
        };
    }, logger))
        .addEdge('discovery_subgraph', 'process_operations')
        .addEdge('builder_subgraph', 'process_operations')
        .addEdge('process_operations', 'route_next_phase')
        .addEdge('assistant_subgraph', 'route_next_phase')
        .addEdge(langgraph_1.START, 'check_state')
        .addConditionalEdges('check_state', (state) => {
        const routes = {
            cleanup_dangling: 'cleanup_dangling',
            compact_messages: 'compact_messages',
            delete_messages: 'delete_messages',
            auto_compact_messages: 'compact_messages',
            clear_error_state: 'clear_error_state',
            continue: 'supervisor',
            discovery: 'create_workflow_name',
        };
        return routes[state.nextPhase] ?? 'supervisor';
    })
        .addEdge('cleanup_dangling', 'check_state')
        .addEdge('delete_messages', 'responder')
        .addEdge('clear_error_state', 'check_state')
        .addConditionalEdges('create_workflow_name', (state) => routeToNode(state.nextPhase))
        .addConditionalEdges('compact_messages', (state) => {
        const hasMessages = state.messages.length > 0;
        return hasMessages ? 'check_state' : 'responder';
    })
        .addConditionalEdges('supervisor', (state) => {
        const next = state.nextPhase;
        if (next === 'builder' || next === 'discovery') {
            return 'create_workflow_name';
        }
        return routeToNode(next);
    })
        .addConditionalEdges('route_next_phase', (state) => routeToNode(state.nextPhase))
        .addEdge('responder', langgraph_1.END)
        .compile({ checkpointer }));
}
