// @n8n/ai-workflow-builder/dist/session-manager.service.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var SessionManagerService_1;
Object.defineProperty(exports, "__esModule", { value: true });
exports.SessionManagerService = void 0;
const langgraph_1 = require("@langchain/langgraph");
const backend_common_1 = require("@n8n/backend-common");
const di_1 = require("@n8n/di");
const code_builder_session_1 = require("./code-builder/utils/code-builder-session");
const builder_tools_1 = require("./tools/builder-tools");
const sessions_1 = require("./types/sessions");
const helpers_1 = require("./utils/cache-control/helpers");
const stream_processor_1 = require("./utils/stream-processor");
const thread_id_1 = require("./utils/thread-id");
let SessionManagerService = class SessionManagerService {
    static { SessionManagerService_1 = this; }
    storage;
    logger;
    checkpointer;
    nodeTypes;
    static HITL_TTL_MS = 24 * 60 * 60 * 1000;
    pendingHitlByThreadId = new Map();
    hitlHistoryByThreadId = new Map();
    constructor(parsedNodeTypes, storage, logger) {
        this.storage = storage;
        this.logger = logger;
        this.nodeTypes = parsedNodeTypes;
        this.checkpointer = new langgraph_1.MemorySaver();
        if (storage) {
            this.logger?.debug('Using persistent session storage');
        }
        else {
            this.logger?.debug('Using in-memory session storage (MemorySaver)');
        }
    }
    get usesPersistence() {
        return !!this.storage;
    }
    updateNodeTypes(nodeTypes) {
        this.nodeTypes = nodeTypes;
    }
    static generateThreadId(workflowId, userId, agentType) {
        return (0, thread_id_1.generateThreadId)(workflowId, userId, agentType);
    }
    getCheckpointer() {
        return this.checkpointer;
    }
    setPendingHitl(threadId, value, triggeringMessageId) {
        this.evictExpiredHitl();
        this.pendingHitlByThreadId.set(threadId, {
            value,
            triggeringMessageId,
            expiresAt: Date.now() + SessionManagerService_1.HITL_TTL_MS,
        });
    }
    clearPendingHitl(threadId) {
        this.pendingHitlByThreadId.delete(threadId);
    }
    getAndClearPendingHitl(threadId) {
        this.evictExpiredHitl();
        const entry = this.pendingHitlByThreadId.get(threadId);
        if (!entry)
            return undefined;
        if (Date.now() > entry.expiresAt) {
            this.pendingHitlByThreadId.delete(threadId);
            return undefined;
        }
        this.pendingHitlByThreadId.delete(threadId);
        return { value: entry.value, triggeringMessageId: entry.triggeringMessageId };
    }
    getPendingHitl(threadId) {
        this.evictExpiredHitl();
        const entry = this.pendingHitlByThreadId.get(threadId);
        if (!entry)
            return undefined;
        if (Date.now() > entry.expiresAt) {
            this.pendingHitlByThreadId.delete(threadId);
            return undefined;
        }
        return entry.value;
    }
    addHitlEntry(threadId, entry) {
        const history = this.hitlHistoryByThreadId.get(threadId) ?? [];
        history.push(entry);
        this.hitlHistoryByThreadId.set(threadId, history);
    }
    getHitlHistory(threadId) {
        return this.hitlHistoryByThreadId.get(threadId) ?? [];
    }
    truncateHitlHistory(threadId, survivingMessages) {
        const history = this.hitlHistoryByThreadId.get(threadId);
        if (!history || history.length === 0)
            return;
        const survivingIds = new Set(survivingMessages
            .map((m) => m.additional_kwargs?.messageId)
            .filter(Boolean));
        const filtered = history.filter((entry) => !entry.afterMessageId || survivingIds.has(entry.afterMessageId));
        if (filtered.length === 0) {
            this.hitlHistoryByThreadId.delete(threadId);
        }
        else {
            this.hitlHistoryByThreadId.set(threadId, filtered);
        }
    }
    injectHitlHistory(threadId, formattedMessages) {
        const history = this.getHitlHistory(threadId);
        if (history.length === 0)
            return;
        const groups = new Map();
        for (const entry of history) {
            const key = entry.afterMessageId;
            const msgs = groups.get(key) ?? [];
            msgs.push(...this.hitlEntryToMessages(entry));
            groups.set(key, msgs);
        }
        const keys = [...groups.keys()].reverse();
        for (const afterMessageId of keys) {
            const msgs = groups.get(afterMessageId);
            const insertAt = this.findInsertPosition(formattedMessages, afterMessageId);
            formattedMessages.splice(insertAt, 0, ...msgs);
        }
    }
    hitlEntryToMessages(entry) {
        if (entry.type === 'questions_answered') {
            return [
                {
                    role: 'assistant',
                    type: 'questions',
                    questions: entry.interrupt.questions,
                    ...(entry.interrupt.introMessage ? { introMessage: entry.interrupt.introMessage } : {}),
                },
                {
                    role: 'user',
                    type: 'user_answers',
                    answers: Array.isArray(entry.answers) ? entry.answers : [],
                },
            ];
        }
        const messages = [
            {
                role: 'assistant',
                type: 'plan',
                plan: entry.plan,
            },
        ];
        if (entry.decision === 'reject') {
            messages.push({
                role: 'user',
                type: 'message',
                text: entry.feedback ?? 'Plan rejected',
            });
        }
        else if (entry.decision === 'modify') {
            messages.push({
                role: 'user',
                type: 'message',
                text: entry.feedback ?? 'Please modify the plan',
            });
        }
        return messages;
    }
    findInsertPosition(formattedMessages, afterMessageId) {
        if (afterMessageId) {
            const idx = formattedMessages.findIndex((m) => m.id === afterMessageId);
            if (idx !== -1)
                return idx + 1;
        }
        const firstUserIdx = formattedMessages.findIndex((m) => m.role === 'user');
        return firstUserIdx !== -1 ? firstUserIdx + 1 : 0;
    }
    evictExpiredHitl() {
        const now = Date.now();
        for (const [id, entry] of this.pendingHitlByThreadId) {
            if (now > entry.expiresAt) {
                this.pendingHitlByThreadId.delete(id);
            }
        }
    }
    async loadSessionMessages(threadId) {
        if (!this.storage)
            return [];
        const stored = await this.storage.getSession(threadId);
        if (!stored || stored.messages.length === 0)
            return [];
        (0, helpers_1.stripAllCacheControlMarkers)(stored.messages);
        this.logger?.debug('Loaded session messages from storage', {
            threadId,
            messageCount: stored.messages.length,
        });
        return stored.messages;
    }
    async saveSessionFromCheckpointer(threadId, previousSummary) {
        if (!this.storage)
            return;
        const threadConfig = {
            configurable: { thread_id: threadId },
        };
        const checkpointTuple = await this.checkpointer.getTuple(threadConfig);
        if (!checkpointTuple?.checkpoint)
            return;
        const rawMessages = checkpointTuple.checkpoint.channel_values?.messages;
        const messages = (0, sessions_1.isLangchainMessagesArray)(rawMessages) ? rawMessages : [];
        await this.storage.saveSession(threadId, {
            messages,
            previousSummary,
            updatedAt: new Date(),
        });
        this.logger?.debug('Saved session from checkpointer', {
            threadId,
            messageCount: messages.length,
        });
    }
    async getPreviousSummary(threadId) {
        if (!this.storage)
            return undefined;
        const stored = await this.storage.getSession(threadId);
        return stored?.previousSummary;
    }
    async clearSession(threadId) {
        if (this.storage) {
            await this.storage.deleteSession(threadId);
        }
        const threadConfig = {
            configurable: { thread_id: threadId },
        };
        try {
            const existingTuple = await this.checkpointer.getTuple(threadConfig);
            if (existingTuple?.checkpoint) {
                const emptyCheckpoint = {
                    ...existingTuple.checkpoint,
                    channel_values: {
                        ...existingTuple.checkpoint.channel_values,
                        messages: [],
                    },
                };
                const metadata = existingTuple.metadata ?? {
                    source: 'update',
                    step: -1,
                    parents: {},
                };
                await this.checkpointer.put(threadConfig, emptyCheckpoint, metadata);
            }
        }
        catch (error) {
            this.logger?.debug('Failed to clear in-memory checkpointer state', { threadId, error });
        }
        this.pendingHitlByThreadId.delete(threadId);
        this.hitlHistoryByThreadId.delete(threadId);
        this.logger?.debug('Session cleared', { threadId });
    }
    async clearAllSessions(workflowId, userId) {
        const mainThreadId = SessionManagerService_1.generateThreadId(workflowId, userId);
        await this.clearSession(mainThreadId);
        this.logger?.debug('All sessions cleared for workflow', { workflowId, userId });
    }
    async getSessions(workflowId, userId, agentType) {
        const sessions = [];
        if (!workflowId) {
            return { sessions };
        }
        const threadId = SessionManagerService_1.generateThreadId(workflowId, userId, agentType);
        if (this.storage) {
            const stored = await this.storage.getSession(threadId);
            if (stored && stored.messages.length > 0) {
                const formattedMessages = (0, stream_processor_1.formatMessages)(stored.messages, (0, builder_tools_1.getBuilderToolsForDisplay)({ nodeTypes: this.nodeTypes }));
                this.injectHitlHistory(threadId, formattedMessages);
                const pendingHitl = this.getPendingHitl(threadId);
                if (pendingHitl) {
                    formattedMessages.push({
                        role: 'assistant',
                        type: pendingHitl.type,
                        ...(pendingHitl.type === 'questions'
                            ? {
                                questions: pendingHitl.questions,
                                ...(pendingHitl.introMessage ? { introMessage: pendingHitl.introMessage } : {}),
                            }
                            : {
                                plan: pendingHitl.plan,
                            }),
                    });
                }
                sessions.push({
                    sessionId: threadId,
                    messages: formattedMessages,
                    lastUpdated: stored.updatedAt.toISOString(),
                });
                return { sessions };
            }
        }
        const threadConfig = {
            configurable: { thread_id: threadId },
        };
        try {
            const checkpoint = await this.checkpointer.getTuple(threadConfig);
            if (checkpoint?.checkpoint) {
                const rawMessages = checkpoint.checkpoint.channel_values?.messages;
                const messages = (0, sessions_1.isLangchainMessagesArray)(rawMessages)
                    ? rawMessages
                    : [];
                const formattedMessages = (0, stream_processor_1.formatMessages)(messages, (0, builder_tools_1.getBuilderToolsForDisplay)({ nodeTypes: this.nodeTypes }));
                this.injectHitlHistory(threadId, formattedMessages);
                const pendingHitl = this.getPendingHitl(threadId);
                if (pendingHitl) {
                    formattedMessages.push({
                        role: 'assistant',
                        type: pendingHitl.type,
                        ...(pendingHitl.type === 'questions'
                            ? {
                                questions: pendingHitl.questions,
                                ...(pendingHitl.introMessage ? { introMessage: pendingHitl.introMessage } : {}),
                            }
                            : {
                                plan: pendingHitl.plan,
                            }),
                    });
                }
                sessions.push({
                    sessionId: threadId,
                    messages: formattedMessages,
                    lastUpdated: checkpoint.checkpoint.ts,
                });
            }
        }
        catch (error) {
            this.logger?.debug('No session found for workflow:', { workflowId, error });
        }
        return { sessions };
    }
    async loadMessagesForTruncation(threadId) {
        if (this.storage) {
            const stored = await this.storage.getSession(threadId);
            if (!stored) {
                return null;
            }
            return { messages: stored.messages, previousSummary: stored.previousSummary };
        }
        const threadConfig = {
            configurable: { thread_id: threadId },
        };
        const checkpointTuple = await this.checkpointer.getTuple(threadConfig);
        if (!checkpointTuple?.checkpoint) {
            return null;
        }
        const rawMessages = checkpointTuple.checkpoint.channel_values?.messages;
        if (!(0, sessions_1.isLangchainMessagesArray)(rawMessages)) {
            return null;
        }
        return { messages: rawMessages };
    }
    async truncateMessagesAfter(workflowId, userId, messageId, agentType) {
        const threadId = SessionManagerService_1.generateThreadId(workflowId, userId, agentType);
        try {
            const loaded = await this.loadMessagesForTruncation(threadId);
            if (!loaded) {
                this.logger?.debug('No messages found for truncation', { threadId, messageId });
                return false;
            }
            const { messages, previousSummary } = loaded;
            const msgIndex = messages.findIndex((msg) => msg.additional_kwargs?.messageId === messageId);
            if (msgIndex === -1) {
                this.logger?.debug('Message with messageId not found', { threadId, messageId });
                return false;
            }
            const truncatedMessages = messages.slice(0, msgIndex);
            if (this.storage) {
                await this.storage.saveSession(threadId, {
                    messages: truncatedMessages,
                    previousSummary,
                    updatedAt: new Date(),
                });
            }
            const threadConfig = {
                configurable: { thread_id: threadId },
            };
            const checkpointTuple = await this.checkpointer.getTuple(threadConfig);
            if (checkpointTuple?.checkpoint) {
                const updatedCheckpoint = {
                    ...checkpointTuple.checkpoint,
                    channel_values: {
                        ...checkpointTuple.checkpoint.channel_values,
                        messages: truncatedMessages,
                    },
                };
                const metadata = checkpointTuple.metadata ?? {
                    source: 'update',
                    step: -1,
                    parents: {},
                };
                await this.checkpointer.put(threadConfig, updatedCheckpoint, metadata);
            }
            this.truncateHitlHistory(threadId, truncatedMessages);
            this.clearPendingHitl(threadId);
            this.logger?.debug('Messages truncated successfully', {
                threadId,
                messageId,
                originalCount: messages.length,
                newCount: truncatedMessages.length,
            });
            if (agentType === 'code-builder' && userId) {
                await this.resetCodeBuilderSession(workflowId, userId);
            }
            return true;
        }
        catch (error) {
            this.logger?.error('Failed to truncate messages', {
                threadId,
                messageId,
                error,
            });
            return false;
        }
    }
    async resetCodeBuilderSession(workflowId, userId) {
        const sessionThreadId = (0, code_builder_session_1.generateCodeBuilderThreadId)(workflowId, userId);
        const sessionConfig = {
            configurable: {
                thread_id: sessionThreadId,
            },
        };
        const existingTuple = await this.checkpointer.getTuple(sessionConfig).catch(() => undefined);
        const existingCheckpoint = existingTuple?.checkpoint;
        const checkpoint = existingCheckpoint
            ? {
                ...existingCheckpoint,
                ts: new Date().toISOString(),
                channel_values: {
                    ...existingCheckpoint.channel_values,
                    codeBuilderSession: {
                        conversationEntries: [],
                        previousSummary: undefined,
                    },
                },
            }
            : {
                v: 1,
                id: crypto.randomUUID(),
                ts: new Date().toISOString(),
                channel_values: {
                    codeBuilderSession: {
                        conversationEntries: [],
                        previousSummary: undefined,
                    },
                },
                channel_versions: {},
                versions_seen: {},
            };
        const sessionMetadata = existingTuple?.metadata ?? {
            source: 'update',
            step: -1,
            parents: {},
        };
        await this.checkpointer.put(sessionConfig, checkpoint, sessionMetadata);
        this.logger?.debug('Code-builder session reset after truncation', {
            sessionThreadId,
            workflowId,
        });
    }
};
exports.SessionManagerService = SessionManagerService;
exports.SessionManagerService = SessionManagerService = SessionManagerService_1 = __decorate([
    (0, di_1.Service)(),
    __metadata("design:paramtypes", [Array, Object, backend_common_1.Logger])
], SessionManagerService);
