// @n8n/ai-workflow-builder/dist/workflow-builder-agent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkflowBuilderAgent = void 0;
const messages_1 = require("@langchain/core/messages");
const langgraph_1 = require("@langchain/langgraph");
const n8n_workflow_1 = require("n8n-workflow");
const constants_1 = require("./constants");
const planner_agent_1 = require("./agents/planner.agent");
const code_builder_1 = require("./code-builder");
const triage_agent_1 = require("./code-builder/triage.agent");
const code_builder_session_1 = require("./code-builder/utils/code-builder-session");
const errors_1 = require("./errors");
const multi_agent_workflow_subgraphs_1 = require("./multi-agent-workflow-subgraphs");
const session_manager_service_1 = require("./session-manager.service");
const error_sanitizer_1 = require("./utils/error-sanitizer");
const stream_processor_1 = require("./utils/stream-processor");
const PROMPT_IS_TOO_LARGE_ERROR = 'The current conversation and workflow state is too large to process. Try to simplify your workflow by breaking it into smaller parts.';
const WORKFLOW_TOO_COMPLEX_ERROR = 'Workflow generation stopped: The AI reached the maximum number of steps while building your workflow. This usually means the workflow design became too complex or got stuck in a loop while trying to create the nodes and connections.';
class WorkflowBuilderAgent {
    checkpointer;
    parsedNodeTypes;
    stageLLMs;
    logger;
    tracer;
    instanceUrl;
    runMetadata;
    onGenerationSuccess;
    nodeDefinitionDirs;
    resourceLocatorCallback;
    onTelemetryEvent;
    assistantHandler;
    sessionFeatureFlags;
    constructor(config) {
        this.parsedNodeTypes = config.parsedNodeTypes;
        this.stageLLMs = config.stageLLMs;
        this.logger = config.logger;
        this.checkpointer = config.checkpointer;
        this.tracer = config.tracer;
        this.instanceUrl = config.instanceUrl;
        this.runMetadata = config.runMetadata;
        this.onGenerationSuccess = config.onGenerationSuccess;
        this.nodeDefinitionDirs = config.nodeDefinitionDirs;
        this.resourceLocatorCallback = config.resourceLocatorCallback;
        this.onTelemetryEvent = config.onTelemetryEvent;
        this.assistantHandler = config.assistantHandler;
    }
    createWorkflow(featureFlags) {
        return (0, multi_agent_workflow_subgraphs_1.createMultiAgentWorkflowWithSubgraphs)({
            parsedNodeTypes: this.parsedNodeTypes,
            stageLLMs: this.stageLLMs,
            logger: this.logger,
            instanceUrl: this.instanceUrl,
            checkpointer: this.checkpointer,
            featureFlags,
            onGenerationSuccess: this.onGenerationSuccess,
            resourceLocatorCallback: this.resourceLocatorCallback,
            assistantHandler: this.assistantHandler,
        });
    }
    async getState(workflowId, userId) {
        const workflow = this.createWorkflow();
        const threadId = session_manager_service_1.SessionManagerService.generateThreadId(workflowId, userId);
        return (await workflow.getState({
            configurable: { thread_id: threadId },
        }));
    }
    getDefaultWorkflowJSON(payload) {
        return (payload.workflowContext?.currentWorkflow ?? {
            nodes: [],
            connections: {},
        });
    }
    async *chat(payload, userId, abortSignal, externalCallbacks, historicalMessages) {
        this.validateMessageLength(payload.message);
        const useCodeWorkflowBuilder = payload.featureFlags?.codeBuilder ?? false;
        if (useCodeWorkflowBuilder) {
            const usePlanMode = payload.featureFlags?.planMode === true;
            if (payload.resumeData && payload.resumeInterrupt?.type === 'plan') {
                const decision = (0, planner_agent_1.parsePlanDecision)(payload.resumeData);
                if (decision.action === 'approve') {
                    this.logger?.debug('Plan approved, routing to CodeWorkflowBuilder with plan', {
                        userId,
                    });
                    const codePayload = {
                        ...payload,
                        planOutput: payload.resumeInterrupt.plan,
                        resumeData: undefined,
                        resumeInterrupt: undefined,
                    };
                    yield* this.runCodeWorkflowBuilder(codePayload, userId, abortSignal);
                    return;
                }
                this.logger?.debug('Plan modify/reject, resuming multi-agent system', {
                    userId,
                    action: decision.action,
                });
                yield* this.runMultiAgentSystem(payload, userId, abortSignal, externalCallbacks, historicalMessages);
                return;
            }
            if (usePlanMode && payload.mode === 'plan') {
                this.logger?.debug('Plan mode with code builder, routing to multi-agent for planning', {
                    userId,
                });
                yield* this.runMultiAgentSystem(payload, userId, abortSignal, externalCallbacks, historicalMessages);
                return;
            }
            const isMergeAskBuildEnabled = payload.featureFlags?.mergeAskBuild === true;
            if (isMergeAskBuildEnabled && this.assistantHandler) {
                this.logger?.debug('Routing through triage agent', { userId });
                yield* this.runTriageAgent(payload, userId, abortSignal);
            }
            else {
                this.logger?.debug('Routing to code workflow builder', { userId });
                yield* this.runCodeWorkflowBuilder(payload, userId, abortSignal);
            }
            return;
        }
        this.logger?.debug('Routing to legacy multi-agent system', { userId });
        yield* this.runMultiAgentSystem(payload, userId, abortSignal, externalCallbacks, historicalMessages);
    }
    async *runCodeWorkflowBuilder(payload, userId, abortSignal) {
        const codeWorkflowBuilder = new code_builder_1.CodeWorkflowBuilder({
            llm: this.stageLLMs.builder,
            nodeTypes: this.parsedNodeTypes,
            logger: this.logger,
            nodeDefinitionDirs: this.nodeDefinitionDirs,
            checkpointer: this.checkpointer,
            onGenerationSuccess: this.onGenerationSuccess,
            callbacks: this.tracer ? [this.tracer] : undefined,
            runMetadata: {
                ...this.runMetadata,
                userMessageId: payload.id,
                workflowId: payload.workflowContext?.currentWorkflow?.id,
            },
            onTelemetryEvent: this.onTelemetryEvent,
            generatePinData: payload.featureFlags?.pinData ?? true,
        });
        yield* codeWorkflowBuilder.chat(payload, userId ?? 'unknown', abortSignal);
    }
    async *runTriageAgent(payload, userId, abortSignal) {
        const workflowId = payload.workflowContext?.currentWorkflow?.id;
        const resolvedUserId = userId ?? 'unknown';
        let session;
        let threadId;
        if (workflowId) {
            threadId = (0, code_builder_session_1.generateCodeBuilderThreadId)(workflowId, resolvedUserId);
            session = await (0, code_builder_session_1.loadCodeBuilderSession)(this.checkpointer, threadId);
        }
        const triageAgent = new triage_agent_1.TriageAgent({
            llm: this.stageLLMs.builder,
            assistantHandler: this.assistantHandler,
            buildWorkflow: (p, u, s) => this.runCodeWorkflowBuilder(p, u, s),
            logger: this.logger,
        });
        const lastEntry = session?.conversationEntries?.at(-1);
        const sdkSessionId = lastEntry?.type === 'assistant-exchange' ? session?.sdkSessionId : undefined;
        const gen = triageAgent.run({
            payload,
            userId: resolvedUserId,
            abortSignal,
            sdkSessionId,
            conversationHistory: session?.conversationEntries,
        });
        const { outcome, collectedText } = yield* this.drainTriageGenerator(gen);
        if (session && threadId) {
            await this.saveTriageOutcome(session, threadId, payload.message, outcome, collectedText);
        }
    }
    async *drainTriageGenerator(gen) {
        const collectedText = [];
        let iterResult = await gen.next();
        while (!iterResult.done) {
            yield iterResult.value;
            for (const msg of iterResult.value.messages ?? []) {
                if (msg.type === 'message' && 'text' in msg) {
                    collectedText.push(msg.text);
                }
            }
            iterResult = await gen.next();
        }
        return { outcome: iterResult.value, collectedText };
    }
    async saveTriageOutcome(session, threadId, userMessage, outcome, collectedText) {
        if (outcome.assistantSummary) {
            session.conversationEntries.push({
                type: 'assistant-exchange',
                userQuery: userMessage,
                assistantSummary: outcome.assistantSummary,
            });
            session.sdkSessionId = outcome.sdkSessionId;
        }
        if (outcome.buildExecuted) {
            if (outcome.assistantSummary) {
                await (0, code_builder_session_1.saveCodeBuilderSession)(this.checkpointer, threadId, session);
            }
            return;
        }
        if (!outcome.assistantSummary) {
            session.conversationEntries.push({
                type: 'plan',
                userQuery: userMessage,
                plan: collectedText.join('\n'),
            });
        }
        await (0, code_builder_session_1.saveCodeBuilderSession)(this.checkpointer, threadId, session);
    }
    async *runMultiAgentSystem(payload, userId, abortSignal, externalCallbacks, historicalMessages) {
        const { agent, threadConfig, streamConfig } = this.setupAgentAndConfigs(payload, userId, abortSignal, externalCallbacks);
        try {
            const stream = await this.createAgentStream(payload, streamConfig, agent, historicalMessages);
            yield* this.processAgentStream(stream, agent, threadConfig);
        }
        catch (error) {
            this.handleStreamError(error);
        }
    }
    validateMessageLength(message) {
        if (message.length > constants_1.MAX_AI_BUILDER_PROMPT_LENGTH) {
            this.logger?.warn('Message exceeds maximum length', {
                messageLength: message.length,
                maxLength: constants_1.MAX_AI_BUILDER_PROMPT_LENGTH,
            });
            throw new errors_1.ValidationError(`Message exceeds maximum length of ${constants_1.MAX_AI_BUILDER_PROMPT_LENGTH} characters`);
        }
    }
    setupAgentAndConfigs(payload, userId, abortSignal, externalCallbacks) {
        if (!this.sessionFeatureFlags && payload.featureFlags) {
            this.sessionFeatureFlags = payload.featureFlags;
        }
        const agent = this.createWorkflow(this.sessionFeatureFlags ?? payload.featureFlags);
        const workflowId = payload.workflowContext?.currentWorkflow?.id;
        const threadId = session_manager_service_1.SessionManagerService.generateThreadId(workflowId, userId);
        const threadConfig = {
            configurable: {
                thread_id: threadId,
                userId,
            },
        };
        const callbacks = externalCallbacks ?? (this.tracer ? [this.tracer] : undefined);
        const streamConfig = {
            ...threadConfig,
            streamMode: ['updates', 'custom'],
            recursionLimit: constants_1.MAX_MULTI_AGENT_STREAM_ITERATIONS,
            signal: abortSignal,
            callbacks,
            metadata: this.runMetadata,
            subgraphs: true,
        };
        return { agent, threadConfig, streamConfig };
    }
    async createAgentStream(payload, streamConfig, agent, historicalMessages) {
        const additionalKwargs = {};
        if (payload.versionId)
            additionalKwargs.versionId = payload.versionId;
        if (payload.id)
            additionalKwargs.messageId = payload.id;
        if (payload.resumeData !== undefined)
            additionalKwargs.resumeData = payload.resumeData;
        const humanMessage = new messages_1.HumanMessage({
            id: payload.id,
            content: payload.message,
            additional_kwargs: additionalKwargs,
        });
        const workflowJSON = this.getDefaultWorkflowJSON(payload);
        const workflowContext = payload.workflowContext;
        const mode = payload.mode ?? 'build';
        const messages = [...(historicalMessages ?? []), humanMessage];
        const stream = payload.resumeData
            ? await agent.stream(new langgraph_1.Command({
                resume: payload.resumeData,
                update: {
                    messages: [
                        ...(payload.resumeInterrupt
                            ? [
                                new messages_1.AIMessage({
                                    content: JSON.stringify(payload.resumeInterrupt),
                                    additional_kwargs: { messageType: payload.resumeInterrupt.type },
                                }),
                            ]
                            : []),
                        humanMessage,
                    ],
                    workflowJSON,
                    workflowContext,
                    ...(payload.mode ? { mode: payload.mode } : {}),
                },
            }), streamConfig)
            : await agent.stream({
                messages,
                workflowJSON,
                workflowOperations: [],
                workflowContext,
                mode,
            }, streamConfig);
        return stream;
    }
    handleStreamError(error) {
        const invalidRequestErrorMessage = this.getInvalidRequestError(error);
        if (invalidRequestErrorMessage) {
            throw new errors_1.ValidationError(invalidRequestErrorMessage);
        }
        if (this.isLlmQuotaOrRateLimitError(error)) {
            throw new n8n_workflow_1.OperationalError((0, error_sanitizer_1.sanitizeLlmErrorMessage)(error), {
                cause: error instanceof Error ? error : undefined,
            });
        }
        throw error;
    }
    async *processAgentStream(stream, agent, threadConfig) {
        try {
            const streamProcessor = (0, stream_processor_1.createStreamProcessor)(stream);
            for await (const output of streamProcessor) {
                yield output;
            }
        }
        catch (error) {
            await this.handleAgentStreamError(error, agent, threadConfig);
        }
    }
    async handleAgentStreamError(error, agent, threadConfig) {
        if (error &&
            typeof error === 'object' &&
            'message' in error &&
            typeof error.message === 'string' &&
            ['Abort', 'Aborted'].includes(error.message)) {
            const messages = (await agent.getState(threadConfig)).values.messages;
            const abortedAiMessage = new messages_1.AIMessage({
                content: 'Task aborted',
                id: crypto.randomUUID(),
            });
            await agent.updateState(threadConfig, { messages: [...messages, abortedAiMessage] });
            return;
        }
        if (error instanceof langgraph_1.GraphRecursionError) {
            throw new n8n_workflow_1.ApplicationError(WORKFLOW_TOO_COMPLEX_ERROR);
        }
        if (this.isTokenExpiredError(error)) {
            throw new n8n_workflow_1.ApplicationError(WORKFLOW_TOO_COMPLEX_ERROR);
        }
        throw error;
    }
    isTokenExpiredError(error) {
        const LC_MODEL_AUTHENTICATION_ERROR = 'MODEL_AUTHENTICATION';
        return (!!error &&
            typeof error === 'object' &&
            'lc_error_code' in error &&
            error.lc_error_code === LC_MODEL_AUTHENTICATION_ERROR &&
            'status' in error &&
            error.status === 401);
    }
    isLlmQuotaOrRateLimitError(error) {
        return !!error && typeof error === 'object' && 'status' in error && error.status === 429;
    }
    getInvalidRequestError(error) {
        if (error instanceof Error &&
            'error' in error &&
            typeof error.error === 'object' &&
            error.error) {
            const innerError = error.error;
            if ('error' in innerError && typeof innerError.error === 'object' && innerError.error) {
                const errorDetails = innerError.error;
                if ('type' in errorDetails &&
                    errorDetails.type === 'invalid_request_error' &&
                    'message' in errorDetails &&
                    typeof errorDetails.message === 'string') {
                    if (errorDetails.message.toLocaleLowerCase().includes('prompt is too long')) {
                        return PROMPT_IS_TOO_LARGE_ERROR;
                    }
                    return errorDetails.message;
                }
            }
        }
        return undefined;
    }
}
exports.WorkflowBuilderAgent = WorkflowBuilderAgent;
