// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/todoist/v22/resource_label/operation_update.schema.js
/**
 * Todoist Node - Version 2.2 - Zod Schema
 * Discriminator: resource=label, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('label'),
      operation: z.literal('update'),
      authentication: z.union([z.literal('apiKey'), z.literal('oAuth2'), expressionSchema]).optional(),
      labelId: stringOrExpression.optional(),
      labelUpdateFields: z.object({ name: stringOrExpression.optional(), color: z.union([z.literal('berry_red'), z.literal('red'), z.literal('orange'), z.literal('yellow'), z.literal('olive_green'), z.literal('lime_green'), z.literal('green'), z.literal('mint_green'), z.literal('teal'), z.literal('sky_blue'), z.literal('light_blue'), z.literal('blue'), z.literal('grape'), z.literal('violet'), z.literal('lavender'), z.literal('magenta'), z.literal('salmon'), z.literal('charcoal'), z.literal('grey'), z.literal('taupe'), expressionSchema]).optional(), order: numberOrExpression.optional(), is_favorite: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
