// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mattermost/v1/resource_reaction/operation_create.ts
/**
 * Mattermost Node - Version 1
 * Discriminator: resource=reaction, operation=create
 */


interface Credentials {
  mattermostApi: CredentialReference;
}

/** Create a new channel */
export type MattermostV1ReactionCreateParams = {
  resource: 'reaction';
  operation: 'create';
/**
 * ID of the user sending the reaction. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    userId?: string | Expression<string>;
/**
 * ID of the post to react to. Obtainable from the post link: &lt;code&gt;https://mattermost.internal.n8n.io/[server]/pl/[postId]&lt;/code&gt;
 */
    postId?: string | Expression<string> | PlaceholderValue;
/**
 * Emoji to use for this reaction
 */
    emojiName?: string | Expression<string> | PlaceholderValue;
};

export type MattermostV1ReactionCreateNode = {
  type: 'n8n-nodes-base.mattermost';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MattermostV1ReactionCreateParams>;
};
