// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mattermost/v1/resource_reaction/operation_delete.ts
/**
 * Mattermost Node - Version 1
 * Discriminator: resource=reaction, operation=delete
 */


interface Credentials {
  mattermostApi: CredentialReference;
}

/** Soft delete a channel */
export type MattermostV1ReactionDeleteParams = {
  resource: 'reaction';
  operation: 'delete';
/**
 * ID of the user whose reaction to delete. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    userId?: string | Expression<string>;
/**
 * ID of the post whose reaction to delete. Obtainable from the post link: &lt;code&gt;https://mattermost.internal.n8n.io/[server]/pl/[postId]&lt;/code&gt;
 */
    postId?: string | Expression<string> | PlaceholderValue;
/**
 * Name of the emoji to delete
 */
    emojiName?: string | Expression<string> | PlaceholderValue;
};

export type MattermostV1ReactionDeleteNode = {
  type: 'n8n-nodes-base.mattermost';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MattermostV1ReactionDeleteParams>;
};
