// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/zendesk/v1/resource_user/operation_get.ts
/**
 * Zendesk Node - Version 1
 * Discriminator: resource=user, operation=get
 */


interface Credentials {
  zendeskApi: CredentialReference;
  zendeskOAuth2Api: CredentialReference;
}

/** Manage users */
export type ZendeskV1UserGetParams = {
  resource: 'user';
  operation: 'get';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * User ID
 */
    id?: string | Expression<string> | PlaceholderValue;
};

export type ZendeskV1UserGetOutput = {
  active?: boolean;
  created_at?: string;
  iana_time_zone?: string;
  id?: number;
  locale?: string;
  locale_id?: number;
  moderator?: boolean;
  name?: string;
  only_private_comments?: boolean;
  report_csv?: boolean;
  restricted_agent?: boolean;
  role?: string;
  shared?: boolean;
  shared_agent?: boolean;
  suspended?: boolean;
  tags?: Array<string>;
  time_zone?: string;
  updated_at?: string;
  url?: string;
  verified?: boolean;
};

export type ZendeskV1UserGetNode = {
  type: 'n8n-nodes-base.zendesk';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ZendeskV1UserGetParams>;
  output?: Items<ZendeskV1UserGetOutput>;
};
