// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/zendesk/v1/resource_user/operation_create.schema.js
/**
 * Zendesk Node - Version 1 - Zod Schema
 * Discriminator: resource=user, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('user'),
      operation: z.literal('create').default('create'),
      authentication: z.union([z.literal('apiToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      name: stringOrExpression.optional(),
      additionalFields: z.object({ alias: stringOrExpression.optional(), custom_role_id: numberOrExpression.optional(), details: stringOrExpression.optional(), email: stringOrExpression.optional(), external_id: stringOrExpression.optional(), locale: stringOrExpression.optional(), moderator: booleanOrExpression.optional(), notes: stringOrExpression.optional(), only_private_comments: booleanOrExpression.optional(), organization_id: stringOrExpression.optional(), phone: stringOrExpression.optional(), report_csv: booleanOrExpression.optional(), restricted_agent: booleanOrExpression.optional(), role: z.union([z.literal('end-user'), z.literal('agent'), z.literal('admin'), expressionSchema]).optional(), signature: stringOrExpression.optional(), suspended: booleanOrExpression.optional(), tags: z.array(z.string()).optional(), ticket_restriction: z.union([z.literal('organization'), z.literal('groups'), z.literal('assigned'), z.literal('requested'), expressionSchema]).optional(), time_zone: stringOrExpression.optional(), userFieldsUi: z.unknown().optional(), verified: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
