// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/zendesk/v1/resource_user/operation_create.ts
/**
 * Zendesk Node - Version 1
 * Discriminator: resource=user, operation=create
 */


interface Credentials {
  zendeskApi: CredentialReference;
  zendeskOAuth2Api: CredentialReference;
}

/** Manage users */
export type ZendeskV1UserCreateParams = {
  resource: 'user';
  operation: 'create';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * The user's name
 */
    name?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** An alias displayed to end users
     */
    alias?: string | Expression<string> | PlaceholderValue;
    /** A custom role if the user is an agent on the Enterprise plan
     * @default 0
     */
    custom_role_id?: number | Expression<number>;
    /** Any details you want to store about the user, such as an address
     */
    details?: string | Expression<string> | PlaceholderValue;
    /** The user's primary email address
     */
    email?: string | Expression<string> | PlaceholderValue;
    /** A unique identifier from another system
     */
    external_id?: string | Expression<string> | PlaceholderValue;
    /** The user's locale. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    locale?: string | Expression<string>;
    /** Whether the user has forum moderation capabilities
     * @default false
     */
    moderator?: boolean | Expression<boolean>;
    /** Any notes you want to store about the user
     */
    notes?: string | Expression<string> | PlaceholderValue;
    /** Whether the user can only create private comments
     * @default false
     */
    only_private_comments?: boolean | Expression<boolean>;
    /** The ID of the user's organization. If the user has more than one organization memberships, the ID of the user's default organization. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    organization_id?: string | Expression<string>;
    /** The user's primary phone number
     */
    phone?: string | Expression<string> | PlaceholderValue;
    /** Whether or not the user can access the CSV report on the Search tab of the Reporting page in the Support admin interface
     * @default false
     */
    report_csv?: boolean | Expression<boolean>;
    /** Whether the agent has any restrictions; false for admins and unrestricted agents, true for other agents
     * @default false
     */
    restricted_agent?: boolean | Expression<boolean>;
    /** The user's role
     */
    role?: 'end-user' | 'agent' | 'admin' | Expression<string>;
    /** The user's signature. Only agents and admins can have signatures.
     */
    signature?: string | Expression<string> | PlaceholderValue;
    /** Whether the agent is suspended. Tickets from suspended users are also suspended, and these users cannot sign in to the end user portal.
     * @default false
     */
    suspended?: boolean | Expression<boolean>;
    /** The array of tags applied to this user. Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     * @default []
     */
    tags?: string[];
    /** Specifies which tickets the user has access to
     */
    ticket_restriction?: 'organization' | 'groups' | 'assigned' | 'requested' | Expression<string>;
    /** The user's time zone
     */
    time_zone?: string | Expression<string> | PlaceholderValue;
    /** Values of custom fields in the user's profile
     * @default {}
     */
    userFieldsUi?: {
        /** Field
     */
    userFieldValues?: Array<{
      /** Name of the field to sort on. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
       */
      field?: string | Expression<string>;
      /** Value of the field
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
    /** Whether the user's primary identity is verified or not
     * @default false
     */
    verified?: boolean | Expression<boolean>;
  };
};

export type ZendeskV1UserCreateOutput = {
  active?: boolean;
  alias?: null;
  created_at?: string;
  custom_role_id?: null;
  default_group_id?: null;
  details?: null;
  iana_time_zone?: string;
  id?: number;
  last_active?: string;
  last_login_at?: null;
  locale?: string;
  locale_id?: number;
  moderator?: boolean;
  name?: string;
  only_private_comments?: boolean;
  organization_id?: null;
  photo?: null;
  report_csv?: boolean;
  restricted_agent?: boolean;
  role?: string;
  role_type?: null;
  shared?: boolean;
  shared_agent?: boolean;
  signature?: null;
  suspended?: boolean;
  tags?: Array<string>;
  ticket_restriction?: string;
  time_zone?: string;
  two_factor_auth_enabled?: null;
  updated_at?: string;
  url?: string;
  verified?: boolean;
};

export type ZendeskV1UserCreateNode = {
  type: 'n8n-nodes-base.zendesk';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<ZendeskV1UserCreateParams>;
  output?: Items<ZendeskV1UserCreateOutput>;
};
