// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/securityScorecard/v1/resource_invite/operation_create.ts
/**
 * SecurityScorecard Node - Version 1
 * Discriminator: resource=invite, operation=create
 */


interface Credentials {
  securityScorecardApi: CredentialReference;
}

/** Create an invite for a company/user */
export type SecurityScorecardV1InviteCreateParams = {
  resource: 'invite';
  operation: 'create';
/**
 * Email
 */
    email?: string | Expression<string> | PlaceholderValue;
/**
 * First Name
 */
    firstName?: string | Expression<string> | PlaceholderValue;
/**
 * Last Name
 */
    lastName?: string | Expression<string> | PlaceholderValue;
/**
 * Message for the invitee
 */
    message?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Minimum days to resolve a scorecard issue
     * @default 0
     */
    days_to_resolve_issue?: number | Expression<number>;
    /** Invitee company domain
     */
    domain?: string | Expression<string> | PlaceholderValue;
    /** Request the invitee's organisation to maintain a minimum grade
     */
    grade_to_maintain?: string | Expression<string> | PlaceholderValue;
    /** Is the invitee organisation's point of contact
     * @default false
     */
    is_organization_point_of_contact?: boolean | Expression<boolean>;
    /** Issue Description
     */
    issue_desc?: string | Expression<string> | PlaceholderValue;
    /** Issue Title
     */
    issue_title?: string | Expression<string> | PlaceholderValue;
    /** Issue Type
     */
    issue_type?: string | Expression<string> | PlaceholderValue;
    /** Whether to send a copy of the invite to the requesting user
     * @default false
     */
    sendme_copy?: boolean | Expression<boolean>;
    /** Optional URL to take the invitee to when arriving to the platform
     */
    target_url?: string | Expression<string> | PlaceholderValue;
  };
};

export type SecurityScorecardV1InviteCreateNode = {
  type: 'n8n-nodes-base.securityScorecard';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SecurityScorecardV1InviteCreateParams>;
};
