// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/securityScorecard/v1/resource_invite/operation_create.schema.js
/**
 * SecurityScorecard Node - Version 1 - Zod Schema
 * Discriminator: resource=invite, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('invite'),
      operation: z.literal('create'),
      email: stringOrExpression.optional(),
      firstName: stringOrExpression.optional(),
      lastName: stringOrExpression.optional(),
      message: stringOrExpression.optional(),
      additionalFields: z.object({ days_to_resolve_issue: numberOrExpression.optional(), domain: stringOrExpression.optional(), grade_to_maintain: stringOrExpression.optional(), is_organization_point_of_contact: booleanOrExpression.optional(), issue_desc: stringOrExpression.optional(), issue_title: stringOrExpression.optional(), issue_type: stringOrExpression.optional(), sendme_copy: booleanOrExpression.optional(), target_url: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
