// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/quickbooks/v1/resource_invoice/operation_create.schema.js
/**
 * QuickBooks Online Node - Version 1 - Zod Schema
 * Discriminator: resource=invoice, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('invoice'),
      operation: z.literal('create'),
      CustomerRef: stringOrExpression.optional(),
      Line: z.object({ Amount: numberOrExpression.optional(), Description: stringOrExpression.optional(), DetailType: z.union([z.literal('SalesItemLineDetail'), expressionSchema]).optional(), itemId: stringOrExpression.optional(), LineNum: numberOrExpression.optional(), TaxCodeRef: stringOrExpression.optional(), Qty: numberOrExpression.optional() }).optional(),
      additionalFields: z.object({ Balance: numberOrExpression.optional(), BillAddr: z.unknown().optional(), BillEmail: stringOrExpression.optional(), CustomerMemo: stringOrExpression.optional(), CustomFields: z.unknown().optional(), DocNumber: stringOrExpression.optional(), DueDate: stringOrExpression.optional(), EmailStatus: z.union([z.literal('NotSet'), z.literal('NeedToSend'), z.literal('EmailSent'), expressionSchema]).optional(), PrintStatus: z.union([z.literal('NotSet'), z.literal('NeedToPrint'), z.literal('PrintComplete'), expressionSchema]).optional(), ShipAddr: stringOrExpression.optional(), TotalAmt: numberOrExpression.optional(), TxnDate: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
