// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/securityScorecard/v1/resource_industry/operation_get_factor.ts
/**
 * SecurityScorecard Node - Version 1
 * Discriminator: resource=industry, operation=getFactor
 */


interface Credentials {
  securityScorecardApi: CredentialReference;
}

/** Get company factor scores and issue counts */
export type SecurityScorecardV1IndustryGetFactorParams = {
  resource: 'industry';
  operation: 'getFactor';
/**
 * Industry
 * @default food
 */
    industry?: 'food' | 'healthcare' | 'manofacturing' | 'retail' | 'technology' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Whether to return a simplified version of the response instead of the raw data
 * @default true
 */
    simple?: boolean | Expression<boolean>;
};

export type SecurityScorecardV1IndustryGetFactorNode = {
  type: 'n8n-nodes-base.securityScorecard';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<SecurityScorecardV1IndustryGetFactorParams>;
};
