// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftTeams/v2/resource_channel_message/operation_get_all.ts
/**
 * Microsoft Teams Node - Version 2
 * Discriminator: resource=channelMessage, operation=getAll
 */


interface Credentials {
  microsoftTeamsOAuth2Api: CredentialReference;
}

/** Get many channels */
export type MicrosoftTeamsV2ChannelMessageGetAllParams = {
  resource: 'channelMessage';
  operation: 'getAll';
/**
 * Select the team from the list, by URL, or by ID (the ID is the "groupId" parameter in the URL you get from "Get a link to the team")
 * @default {"mode":"list","value":""}
 */
    teamId?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
/**
 * Select the channel from the list, by URL, or by ID (the ID is the "threadId" in the URL)
 * @default {"mode":"list","value":""}
 */
    channelId?: { __rl: true; mode: 'list' | 'id'; value: string; cachedResultName?: string };
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
};

export type MicrosoftTeamsV2ChannelMessageGetAllOutput = {
  attachments?: Array<{
    contentType?: string;
    id?: string;
    teamsAppId?: null;
    thumbnailUrl?: null;
  }>;
  body?: {
    contentType?: string;
  };
  channelIdentity?: {
    channelId?: string;
    teamId?: string;
  };
  chatId?: null;
  createdDateTime?: string;
  etag?: string;
  from?: {
    device?: null;
    user?: {
      '@odata.type'?: string;
      displayName?: string;
      id?: string;
      tenantId?: string;
      userIdentityType?: string;
    };
  };
  id?: string;
  importance?: string;
  lastModifiedDateTime?: string;
  locale?: string;
  mentions?: Array<{
    id?: number;
    mentioned?: {
      application?: null;
      conversation?: null;
      device?: null;
      tag?: null;
      user?: {
        '@odata.type'?: string;
        displayName?: string;
        id?: string;
        tenantId?: string;
        userIdentityType?: string;
      };
    };
    mentionText?: string;
  }>;
  messageType?: string;
  policyViolation?: null;
  reactions?: Array<{
    createdDateTime?: string;
    displayName?: string;
    reactionContentUrl?: null;
    reactionType?: string;
    user?: {
      application?: null;
      device?: null;
      user?: {
        '@odata.type'?: string;
        displayName?: null;
        id?: string;
        tenantId?: string;
        userIdentityType?: string;
      };
    };
  }>;
  replyToId?: null;
  webUrl?: string;
};

export type MicrosoftTeamsV2ChannelMessageGetAllNode = {
  type: 'n8n-nodes-base.microsoftTeams';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<MicrosoftTeamsV2ChannelMessageGetAllParams>;
  output?: Items<MicrosoftTeamsV2ChannelMessageGetAllOutput>;
};
