// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/executeWorkflowTrigger/v11.ts
/**
 * Execute Workflow Trigger Node - Version 1.1
 * Helpers for calling other n8n workflows. Used for designing modular, microservice-like workflows.
 */


export interface ExecuteWorkflowTriggerV11Params {
  events?: unknown;
/**
 * Input data mode
 * @default workflowInputs
 */
    inputSource?: 'workflowInputs' | 'jsonExample' | 'passthrough';
/**
 * JSON Example
 * @displayOptions.show { inputSource: ["jsonExample"] }
 */
    jsonExample?: IDataObject | string;
/**
 * Define expected input fields. If no inputs are provided, all data from the calling workflow will be passed through.
 * @displayOptions.show { inputSource: ["workflowInputs"] }
 * @default {}
 */
    workflowInputs?: {
        /** Values
     */
    values?: Array<{
      /** A unique name for this workflow input, used to reference it from another workflows
       */
      name?: string;
      /** Expected data type for this input value. Determines how this field's values are stored, validated, and displayed.
       * @default string
       */
      type?: 'any' | 'string' | 'number' | 'boolean' | 'array' | 'object';
    }>;
  };
}

interface ExecuteWorkflowTriggerV11NodeBase {
  type: 'n8n-nodes-base.executeWorkflowTrigger';
  version: 1.1;
  isTrigger: true;
}

export type ExecuteWorkflowTriggerV11ParamsNode = ExecuteWorkflowTriggerV11NodeBase & {
  config: NodeConfig<ExecuteWorkflowTriggerV11Params>;
};

export type ExecuteWorkflowTriggerV11Node = ExecuteWorkflowTriggerV11ParamsNode;
