// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/executeWorkflowTrigger/v11.schema.js
/**
 * Execute Workflow Trigger Node - Version 1.1 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    events: z.unknown().optional(),
    inputSource: z.union([z.literal('workflowInputs'), z.literal('jsonExample'), z.literal('passthrough')]).optional(),
    jsonExample: resolveSchema({ parameters, schema: z.union([iDataObjectSchema, z.string()]), required: false, displayOptions: {"show":{"inputSource":["jsonExample"]}}, defaults: {"inputSource":"workflowInputs"} }),
    workflowInputs: resolveSchema({ parameters, schema: z.object({ values: z.array(z.object({ name: z.string().optional(), type: z.union([z.literal('any'), z.literal('string'), z.literal('number'), z.literal('boolean'), z.literal('array'), z.literal('object')]).optional() })).optional() }), required: false, displayOptions: {"show":{"inputSource":["workflowInputs"]}}, defaults: {"inputSource":"workflowInputs"} }),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
