// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/wordpress/v1/resource_user/operation_get_all.schema.js
/**
 * Wordpress Node - Version 1 - Zod Schema
 * Discriminator: resource=user, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('user'),
      operation: z.literal('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      options: z.object({ context: z.union([z.literal('view'), z.literal('embed'), z.literal('edit'), expressionSchema]).optional(), orderBy: z.union([z.literal('email'), z.literal('id'), z.literal('include'), z.literal('include_slugs'), z.literal('name'), z.literal('registered_date'), z.literal('slug'), z.literal('url'), expressionSchema]).optional(), order: z.union([z.literal('asc'), z.literal('desc'), expressionSchema]).optional(), search: stringOrExpression.optional(), who: z.union([z.literal('authors'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
