// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleDrive/v3/resource_folder/operation_delete_folder.ts
/**
 * Google Drive Node - Version 3
 * Discriminator: resource=folder, operation=deleteFolder
 */


interface Credentials {
  googleApi: CredentialReference;
  googleDriveOAuth2Api: CredentialReference;
}

/** Permanently delete a folder */
export type GoogleDriveV3FolderDeleteFolderParams = {
  resource: 'folder';
  operation: 'deleteFolder';
  authentication?: 'oAuth2' | 'serviceAccount' | Expression<string>;
/**
 * The folder to delete
 * @default {"mode":"list","value":""}
 */
    folderNoRootId?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
/**
 * Options
 * @default {}
 */
    options?: {
    /** Whether to delete the folder immediately. If false, the folder will be moved to the trash.
     * @default false
     */
    deletePermanently?: boolean | Expression<boolean>;
  };
};

export type GoogleDriveV3FolderDeleteFolderOutput = {
  fileId?: string;
  success?: boolean;
};

export type GoogleDriveV3FolderDeleteFolderNode = {
  type: 'n8n-nodes-base.googleDrive';
  version: 3;
  credentials?: Credentials;
  config: NodeConfig<GoogleDriveV3FolderDeleteFolderParams>;
  output?: Items<GoogleDriveV3FolderDeleteFolderOutput>;
};
