// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleDrive/v3/resource_folder/operation_create.ts
/**
 * Google Drive Node - Version 3
 * Discriminator: resource=folder, operation=create
 */


interface Credentials {
  googleApi: CredentialReference;
  googleDriveOAuth2Api: CredentialReference;
}

/** Create a shared drive */
export type GoogleDriveV3FolderCreateParams = {
  resource: 'folder';
  operation: 'create';
  authentication?: 'oAuth2' | 'serviceAccount' | Expression<string>;
/**
 * The name of the new folder. If not set, 'Untitled' will be used.
 */
    name?: string | Expression<string> | PlaceholderValue;
/**
 * The drive where to create the new folder
 * @default {"mode":"list","value":"My Drive"}
 */
    driveId?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
/**
 * The parent folder where to create the new folder
 * @default {"mode":"list","value":"root","cachedResultName":"/ (Root folder)"}
 */
    folderId?: { __rl: true; mode: 'list' | 'url' | 'id'; value: string; cachedResultName?: string };
/**
 * Options
 * @default {}
 */
    options?: {
    /** Whether to return a simplified version of the response instead of all fields
     * @default true
     */
    simplifyOutput?: boolean | Expression<boolean>;
    /** The color of the folder as an RGB hex string. If an unsupported color is specified, the closest color in the palette will be used instead.
     */
    folderColorRgb?: string | Expression<string>;
  };
};

export type GoogleDriveV3FolderCreateOutput = {
  id?: string;
  kind?: string;
  mimeType?: string;
  name?: string;
};

export type GoogleDriveV3FolderCreateNode = {
  type: 'n8n-nodes-base.googleDrive';
  version: 3;
  credentials?: Credentials;
  config: NodeConfig<GoogleDriveV3FolderCreateParams>;
  output?: Items<GoogleDriveV3FolderCreateOutput>;
};
