// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/travisCi/v1/resource_build/operation_get_all.ts
/**
 * TravisCI Node - Version 1
 * Discriminator: resource=build, operation=getAll
 */


interface Credentials {
  travisCiApi: CredentialReference;
}

/** Get many builds */
export type TravisCiV1BuildGetAllParams = {
  resource: 'build';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** List of attributes to eager load
     */
    include?: string | Expression<string> | PlaceholderValue;
    /** You may specify order to sort your response
     * @default asc
     */
    order?: 'asc' | 'desc' | Expression<string>;
    /** Sort By
     * @default number
     */
    sortBy?: 'created_at' | 'finished_at' | 'id' | 'number' | 'started_at' | Expression<string>;
  };
};

export type TravisCiV1BuildGetAllNode = {
  type: 'n8n-nodes-base.travisCi';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<TravisCiV1BuildGetAllParams>;
};
