// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/travisCi/v1/resource_build/operation_get_all.schema.js
/**
 * TravisCI Node - Version 1 - Zod Schema
 * Discriminator: resource=build, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('build').default('build'),
      operation: z.literal('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      additionalFields: z.object({ include: stringOrExpression.optional(), order: z.union([z.literal('asc'), z.literal('desc'), expressionSchema]).optional(), sortBy: z.union([z.literal('created_at'), z.literal('finished_at'), z.literal('id'), z.literal('number'), z.literal('started_at'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
