// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/travisCi/v1/resource_build/operation_trigger.ts
/**
 * TravisCI Node - Version 1
 * Discriminator: resource=build, operation=trigger
 */


interface Credentials {
  travisCiApi: CredentialReference;
}

/** Trigger a build */
export type TravisCiV1BuildTriggerParams = {
  resource: 'build';
  operation: 'trigger';
/**
 * Same as {ownerName}/{repositoryName}
 */
    slug?: string | Expression<string> | PlaceholderValue;
/**
 * Branch requested to be built
 */
    branch?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Travis-ci status message attached to the request
     */
    message?: string | Expression<string> | PlaceholderValue;
    /** Merge Mode
     */
    mergeMode?: 'deep_merge' | 'deep_merge_append' | 'deep_merge_prepend' | 'merge' | 'replace' | Expression<string>;
  };
};

export type TravisCiV1BuildTriggerNode = {
  type: 'n8n-nodes-base.travisCi';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<TravisCiV1BuildTriggerParams>;
};
