// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/rundeck/v1.schema.js
/**
 * Rundeck Node - Version 1 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    resource: z.union([z.literal('job')]).optional(),
    operation: z.union([z.literal('execute'), z.literal('getMetadata')]).optional(),
    jobid: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"operation":["execute","getMetadata"],"resource":["job"]}}, defaults: {"operation":"execute","resource":"job"} }),
    arguments: resolveSchema({ parameters, schema: z.object({ arguments: z.array(z.object({ name: stringOrExpression.optional(), value: stringOrExpression.optional() })).optional() }), required: false, displayOptions: {"show":{"operation":["execute"],"resource":["job"]}}, defaults: {"operation":"execute","resource":"job"} }),
    filter: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"operation":["execute"],"resource":["job"]}}, defaults: {"operation":"execute","resource":"job"} }),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
