// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/gSuiteAdmin/v1/resource_group/operation_update.ts
/**
 * Google Workspace Admin Node - Version 1
 * Discriminator: resource=group, operation=update
 */


interface Credentials {
  gSuiteAdminOAuth2Api: CredentialReference;
}

/** Update a ChromeOS device */
export type GSuiteAdminV1GroupUpdateParams = {
  resource: 'group';
  operation: 'update';
/**
 * Select the group to perform the operation on
 * @default {"mode":"list","value":""}
 */
    groupId?: { __rl: true; mode: 'list' | 'GroupId'; value: string; cachedResultName?: string };
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** An extended description to help users determine the purpose of a group. For example, you can include information about who should join the group, the types of messages to send to the group, links to FAQs about the group, or related groups.
     */
    description?: string | Expression<string> | PlaceholderValue;
    /** The group's email address. If your account has multiple domains, select the appropriate domain for the email address. The email must be unique.
     */
    email?: string | Expression<string> | PlaceholderValue;
    /** The group's display name
     */
    name?: string | Expression<string> | PlaceholderValue;
  };
};

export type GSuiteAdminV1GroupUpdateNode = {
  type: 'n8n-nodes-base.gSuiteAdmin';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GSuiteAdminV1GroupUpdateParams>;
};
