// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mattermost/v1/resource_channel/operation_members.ts
/**
 * Mattermost Node - Version 1
 * Discriminator: resource=channel, operation=members
 */


interface Credentials {
  mattermostApi: CredentialReference;
}

/** Get a page of members for a channel */
export type MattermostV1ChannelMembersParams = {
  resource: 'channel';
  operation: 'members';
/**
 * The Mattermost Team. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    teamId?: string | Expression<string>;
/**
 * The Mattermost Team. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    channelId?: string | Expression<string>;
/**
 * By default the response only contain the ID of the user. If this option gets activated, it will resolve the user automatically.
 * @default true
 */
    resolveData?: boolean | Expression<boolean>;
/**
 * Whether to return all results or only up to a given limit
 * @default true
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
};

export type MattermostV1ChannelMembersNode = {
  type: 'n8n-nodes-base.mattermost';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MattermostV1ChannelMembersParams>;
};
