// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mattermost/v1/resource_channel/operation_search.ts
/**
 * Mattermost Node - Version 1
 * Discriminator: resource=channel, operation=search
 */


interface Credentials {
  mattermostApi: CredentialReference;
}

/** Search for a channel */
export type MattermostV1ChannelSearchParams = {
  resource: 'channel';
  operation: 'search';
/**
 * The Mattermost Team. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    teamId?: string | Expression<string>;
/**
 * The search term for Channels in a Team
 */
    term?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
};

export type MattermostV1ChannelSearchNode = {
  type: 'n8n-nodes-base.mattermost';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MattermostV1ChannelSearchParams>;
};
