// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/slack/v22/resource_user_group/operation_create.schema.js
/**
 * Slack Node - Version 2.2 - Zod Schema
 * Discriminator: resource=userGroup, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('userGroup'),
      operation: z.literal('create').default('create'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      name: stringOrExpression.optional(),
      Options: z.object({ channelIds: z.array(z.string()).optional(), description: stringOrExpression.optional(), handle: stringOrExpression.optional(), include_count: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
