// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/odoo/v1/resource_contact/operation_create.schema.js
/**
 * Odoo Node - Version 1 - Zod Schema
 * Discriminator: resource=contact, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('contact').default('contact'),
      operation: z.literal('create').default('create'),
      contactName: stringOrExpression.optional(),
      additionalFields: z.object({ address: z.unknown().optional(), email: stringOrExpression.optional(), comment: stringOrExpression.optional(), 'function': stringOrExpression.optional(), mobile: stringOrExpression.optional(), phone: stringOrExpression.optional(), vat: stringOrExpression.optional(), website: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
