// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/currents/v1/resource_run/operation_get_all.schema.js
/**
 * Currents Node - Version 1 - Zod Schema
 * Discriminator: resource=run, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('run').default('run'),
      operation: z.literal('getAll').default('getAll'),
      projectId: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      limit: numberOrExpression.optional(),
      filters: z.object({ authors: stringOrExpression.optional(), branches: stringOrExpression.optional(), completionState: z.array(z.union([z.literal('CANCELED'), z.literal('COMPLETE'), z.literal('IN_PROGRESS'), z.literal('TIMEOUT')])).optional(), dateEnd: stringOrExpression.optional(), dateStart: stringOrExpression.optional(), search: stringOrExpression.optional(), status: z.array(z.union([z.literal('FAILED'), z.literal('FAILING'), z.literal('PASSED'), z.literal('RUNNING')])).optional(), tags: stringOrExpression.optional(), tagOperator: z.union([z.literal('AND'), z.literal('OR'), expressionSchema]).optional() }).optional(),
      options: z.object({ startingAfter: stringOrExpression.optional(), endingBefore: stringOrExpression.optional() }).optional(),
      requestOptions: z.object({ batching: z.unknown().optional(), allowUnauthorizedCerts: z.boolean().optional(), proxy: stringOrExpression.optional(), timeout: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
