// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/currents/v1/resource_run/operation_get_all.ts
/**
 * Currents Node - Version 1
 * Discriminator: resource=run, operation=getAll
 */


interface Credentials {
  currentsApi: CredentialReference;
}

/** Test run */
export type CurrentsV1RunGetAllParams = {
  resource: 'run';
  operation: 'getAll';
/**
 * The Currents project
 * @default {"mode":"list","value":""}
 */
    projectId?: { __rl: true; mode: 'list' | 'id'; value: string; cachedResultName?: string };
/**
 * Max number of results to return
 * @default 10
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Filter by git authors (comma-separated for multiple)
     */
    authors?: string | Expression<string> | PlaceholderValue;
    /** Filter by branches (comma-separated for multiple)
     */
    branches?: string | Expression<string> | PlaceholderValue;
    /** Filter by completion state
     * @default []
     */
    completionState?: Array<'CANCELED' | 'COMPLETE' | 'IN_PROGRESS' | 'TIMEOUT'>;
    /** Filter runs created before this date
     */
    dateEnd?: string | Expression<string>;
    /** Filter runs created on or after this date
     */
    dateStart?: string | Expression<string>;
    /** Search by ciBuildId or commit message (max 200 characters)
     */
    search?: string | Expression<string> | PlaceholderValue;
    /** Filter by run status
     * @default []
     */
    status?: Array<'FAILED' | 'FAILING' | 'PASSED' | 'RUNNING'>;
    /** Filter by tags (comma-separated for multiple)
     */
    tags?: string | Expression<string> | PlaceholderValue;
    /** Logical operator for tag filtering
     * @default AND
     */
    tagOperator?: 'AND' | 'OR' | Expression<string>;
  };
/**
 * Options
 * @default {}
 */
    options?: {
    /** Cursor for forward pagination (use cursor from previous response)
     */
    startingAfter?: string | Expression<string> | PlaceholderValue;
    /** Cursor for backward pagination (use cursor from previous response)
     */
    endingBefore?: string | Expression<string> | PlaceholderValue;
  };
  requestOptions?: {
    /** Batching
     * @default {"batch":{}}
     */
    batching?: {
        /** Batching
     */
    batch?: {
      /** Input will be split in batches to throttle requests. -1 for disabled. 0 will be treated as 1.
       * @default 50
       */
      batchSize?: number | Expression<number>;
      /** Time (in milliseconds) between each batch of requests. 0 for disabled.
       * @default 1000
       */
      batchInterval?: number | Expression<number>;
    };
  };
    /** Whether to accept the response even if SSL certificate validation is not possible
     * @default false
     */
    allowUnauthorizedCerts?: boolean;
    /** HTTP proxy to use. If authentication is required it can be defined as follow: http://username:password@myproxy:3128
     */
    proxy?: string | Expression<string> | PlaceholderValue;
    /** Time in ms to wait for the server to send response headers (and start the response body) before aborting the request
     * @default 10000
     */
    timeout?: number | Expression<number>;
  };
};

export type CurrentsV1RunGetAllNode = {
  type: 'n8n-nodes-base.currents';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<CurrentsV1RunGetAllParams>;
};
