// @n8n/n8n-nodes-langchain/dist/node-definitions/nodes/n8n-nodes-langchain/openAi/v11/resource_text/operation_classify.ts
/**
 * OpenAI Node - Version 1.1
 * Discriminator: resource=text, operation=classify
 */


interface Credentials {
  openAiApi: CredentialReference;
}

/** Check whether content complies with usage policies */
export type LcOpenAiV11TextClassifyParams = {
  resource: 'text';
  operation: 'classify';
/**
 * The input text to classify if it is violates the moderation policy
 */
    input?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to return a simplified version of the response instead of the raw data
 * @default false
 */
    simplify?: boolean | Expression<boolean>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Whether to use the stable version of the model instead of the latest version, accuracy may be slightly lower
     * @default false
     */
    useStableModel?: boolean | Expression<boolean>;
  };
};

export interface LcOpenAiV11TextClassifySubnodeConfig {
  tools?: ToolInstance[];
  memory?: MemoryInstance;
}

export type LcOpenAiV11TextClassifyNode = {
  type: '@n8n/n8n-nodes-langchain.openAi';
  version: 1.1;
  credentials?: Credentials;
  config: NodeConfig<LcOpenAiV11TextClassifyParams> & { subnodes?: LcOpenAiV11TextClassifySubnodeConfig };
};
