// @n8n/n8n-nodes-langchain/dist/node-definitions/nodes/n8n-nodes-langchain/openAi/v11/resource_text/operation_message.schema.js
/**
 * OpenAI Node - Version 1.1 - Zod Schema
 * Discriminator: resource=text, operation=message
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, toolInstanceSchema, memoryInstanceSchema }) {

  // Static subnode schema
  const subnodesSchema = z.object({
    tools: z.array(toolInstanceSchema).optional(),
    memory: memoryInstanceSchema.optional(),
  }).strict();

  return z.object({
    parameters: z.object({
      resource: z.literal('text').default('text'),
      operation: z.literal('message').default('message'),
      modelId: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      messages: z.object({ values: z.array(z.object({ content: stringOrExpression.optional(), role: z.union([z.literal('user'), z.literal('assistant'), z.literal('system'), expressionSchema]).optional() })).optional() }).optional(),
      simplify: booleanOrExpression.optional(),
      jsonOutput: booleanOrExpression.optional(),
      options: z.object({ frequency_penalty: numberOrExpression.optional(), maxTokens: numberOrExpression.optional(), n: numberOrExpression.optional(), presence_penalty: numberOrExpression.optional(), temperature: numberOrExpression.optional(), topP: numberOrExpression.optional(), reasoning_effort: z.union([z.literal('low'), z.literal('medium'), z.literal('high'), expressionSchema]).optional(), maxToolsIterations: numberOrExpression.optional() }).optional(),
    }).optional(),
    subnodes: subnodesSchema.optional(),
  });
};
