// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/pipedrive/v1/resource_deal_activity/operation_get_all.ts
/**
 * Pipedrive Node - Version 1
 * Discriminator: resource=dealActivity, operation=getAll
 */


interface Credentials {
  pipedriveApi: CredentialReference;
  pipedriveOAuth2Api: CredentialReference;
}

/** Get data of many activities */
export type PipedriveV1DealActivityGetAllParams = {
  resource: 'dealActivity';
  operation: 'getAll';
  authentication?: 'apiToken' | 'oAuth2' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * The ID of the deal whose activity to retrieve. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    dealId?: string | Expression<string>;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Whether the activity is done or not
     * @default false
     */
    done?: boolean | Expression<boolean>;
    /** A comma-separated Activity IDs, to exclude from result. Ex. 4, 9, 11, ...
     */
    exclude?: string | Expression<string> | PlaceholderValue;
  };
};

export type PipedriveV1DealActivityGetAllOutput = {
  active_flag?: boolean;
  add_time?: string;
  assigned_to_user_id?: number;
  attendees?: null;
  busy_flag?: boolean;
  company_id?: number;
  conference_meeting_client?: null;
  conference_meeting_id?: null;
  conference_meeting_url?: null;
  created_by_user_id?: number;
  deal_dropbox_bcc?: string;
  deal_id?: number;
  deal_title?: string;
  done?: boolean;
  due_date?: string;
  due_time?: string;
  duration?: string;
  id?: number;
  is_recurring?: null;
  last_notification_time?: null;
  last_notification_user_id?: null;
  lead?: null;
  lead_id?: null;
  lead_title?: null;
  location_admin_area_level_1?: null;
  location_admin_area_level_2?: null;
  location_country?: null;
  location_formatted_address?: null;
  location_locality?: null;
  location_postal_code?: null;
  location_route?: null;
  location_street_number?: null;
  location_sublocality?: null;
  location_subpremise?: null;
  marked_as_done_time?: string;
  original_start_time?: null;
  owner_name?: string;
  private?: boolean;
  rec_master_activity_id?: null;
  rec_rule?: null;
  rec_rule_extension?: null;
  reference_id?: null;
  series?: null;
  source_timezone?: null;
  subject?: string;
  type?: string;
  type_name?: string;
  update_time?: string;
  user_id?: number;
};

export type PipedriveV1DealActivityGetAllNode = {
  type: 'n8n-nodes-base.pipedrive';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<PipedriveV1DealActivityGetAllParams>;
  output?: Items<PipedriveV1DealActivityGetAllOutput>;
};
