// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/kafkaTrigger/v12.schema.js
/**
 * Kafka Trigger Node - Version 1.2 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    topic: stringOrExpression.optional(),
    groupId: stringOrExpression.optional(),
    useSchemaRegistry: booleanOrExpression.optional(),
    schemaRegistryUrl: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"useSchemaRegistry":[true]}}, defaults: {"useSchemaRegistry":false} }),
    options: z.object({ allowAutoTopicCreation: booleanOrExpression.optional(), autoCommitThreshold: numberOrExpression.optional(), autoCommitInterval: numberOrExpression.optional(), batchSize: numberOrExpression.optional(), eachBatchAutoResolve: booleanOrExpression.optional(), fetchMaxBytes: numberOrExpression.optional(), fetchMinBytes: numberOrExpression.optional(), heartbeatInterval: numberOrExpression.optional(), heartbeatInterval: numberOrExpression.optional(), maxInFlightRequests: numberOrExpression.optional(), fromBeginning: booleanOrExpression.optional(), jsonParseMessage: booleanOrExpression.optional(), keepBinaryData: booleanOrExpression.optional(), parallelProcessing: booleanOrExpression.optional(), partitionsConsumedConcurrently: numberOrExpression.optional(), onlyMessage: booleanOrExpression.optional(), returnHeaders: booleanOrExpression.optional(), rebalanceTimeout: numberOrExpression.optional(), errorRetryDelay: numberOrExpression.optional(), sessionTimeout: numberOrExpression.optional() }).optional(),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
