// @n8n/n8n-nodes-langchain/dist/nodes/tools/ToolWikipedia/ToolWikipedia.node.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ToolWikipedia = void 0;
const wikipedia_query_run_1 = require("@langchain/community/tools/wikipedia_query_run");
const n8n_workflow_1 = require("n8n-workflow");
const ai_utilities_1 = require("@n8n/ai-utilities");
function getTool(ctx) {
    const WikiTool = new wikipedia_query_run_1.WikipediaQueryRun();
    WikiTool.name = (0, n8n_workflow_1.nodeNameToToolName)(ctx.getNode());
    WikiTool.description =
        'A tool for interacting with and fetching data from the Wikipedia API. The input should always be a string query.';
    return WikiTool;
}
class ToolWikipedia {
    constructor() {
        this.description = {
            displayName: 'Wikipedia',
            name: 'toolWikipedia',
            icon: 'file:wikipedia.svg',
            group: ['transform'],
            version: 1,
            description: 'Search in Wikipedia',
            defaults: {
                name: 'Wikipedia',
            },
            codex: {
                categories: ['AI'],
                subcategories: {
                    AI: ['Tools'],
                    Tools: ['Other Tools'],
                },
                resources: {
                    primaryDocumentation: [
                        {
                            url: 'https://docs.n8n.io/integrations/builtin/cluster-nodes/sub-nodes/n8n-nodes-langchain.toolwikipedia/',
                        },
                    ],
                },
            },
            inputs: [],
            outputs: [n8n_workflow_1.NodeConnectionTypes.AiTool],
            outputNames: ['Tool'],
            properties: [(0, ai_utilities_1.getConnectionHintNoticeField)([n8n_workflow_1.NodeConnectionTypes.AiAgent])],
        };
    }
    async supplyData() {
        return {
            response: (0, ai_utilities_1.logWrapper)(getTool(this), this),
        };
    }
    async execute() {
        const WikiTool = getTool(this);
        const items = this.getInputData();
        const response = [];
        for (let itemIndex = 0; itemIndex < this.getInputData().length; itemIndex++) {
            const item = items[itemIndex];
            if (item === undefined) {
                continue;
            }
            const result = await WikiTool.invoke(item.json);
            response.push({
                json: { response: result },
                pairedItem: { item: itemIndex },
            });
        }
        return [response];
    }
}
exports.ToolWikipedia = ToolWikipedia;
