// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/tapfiliate/v1/resource_affiliate/operation_get_all.schema.js
/**
 * Tapfiliate Node - Version 1 - Zod Schema
 * Discriminator: resource=affiliate, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('affiliate').default('affiliate'),
      operation: z.literal('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filters: z.object({ affiliate_group_id: stringOrExpression.optional(), click_id: stringOrExpression.optional(), email: stringOrExpression.optional(), parentId: stringOrExpression.optional(), referral_code: stringOrExpression.optional(), source_id: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
