// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/rocketchat/v1/resource_chat/operation_post_message.schema.js
/**
 * RocketChat Node - Version 1 - Zod Schema
 * Discriminator: resource=chat, operation=postMessage
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('chat').default('chat'),
      operation: z.literal('postMessage').default('postMessage'),
      channel: stringOrExpression.optional(),
      text: stringOrExpression.optional(),
      jsonParameters: booleanOrExpression.optional(),
      options: z.object({ alias: stringOrExpression.optional(), avatar: stringOrExpression.optional(), emoji: stringOrExpression.optional() }).optional(),
      attachments: resolveSchema({ parameters, schema: z.object({ color: stringOrExpression.optional(), text: stringOrExpression.optional(), ts: stringOrExpression.optional(), thumbUrl: stringOrExpression.optional(), messageLink: stringOrExpression.optional(), collapsed: booleanOrExpression.optional(), authorName: stringOrExpression.optional(), authorLink: stringOrExpression.optional(), authorIcon: stringOrExpression.optional(), title: stringOrExpression.optional(), titleLink: stringOrExpression.optional(), titleLinkDownload: booleanOrExpression.optional(), imageUrl: stringOrExpression.optional(), audioUrl: stringOrExpression.optional(), videoUrl: stringOrExpression.optional(), fields: z.unknown().optional() }), required: false, displayOptions: {"show":{"jsonParameters":[false]}}, defaults: {"jsonParameters":false} }),
      attachmentsJson: resolveSchema({ parameters, schema: z.union([iDataObjectSchema, z.string()]), required: false, displayOptions: {"show":{"jsonParameters":[true]}}, defaults: {"jsonParameters":false} }),
    }).optional(),
  });
};
