// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/gmail/v1/resource_draft/operation_get_all.schema.js
/**
 * Gmail Node - Version 1 - Zod Schema
 * Discriminator: resource=draft, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('draft').default('draft'),
      operation: z.literal('getAll'),
      authentication: z.union([z.literal('oAuth2'), z.literal('serviceAccount'), expressionSchema]).optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      additionalFields: z.object({ dataPropertyAttachmentsPrefixName: stringOrExpression.optional(), format: z.union([z.literal('full'), z.literal('ids'), z.literal('metadata'), z.literal('minimal'), z.literal('raw'), z.literal('resolved'), expressionSchema]).optional(), includeSpamTrash: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
