// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/gmail/v1/resource_draft/operation_get.ts
/**
 * Gmail Node - Version 1
 * Discriminator: resource=draft, operation=get
 */


interface Credentials {
  googleApi: CredentialReference;
  gmailOAuth2: CredentialReference;
}

export type GmailV1DraftGetParams = {
  resource: 'draft';
  operation: 'get';
  authentication?: 'oAuth2' | 'serviceAccount' | Expression<string>;
/**
 * Draft ID
 */
    messageId?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** Prefix for name of the binary property to which to write the attachments. An index starting with 0 will be added. So if name is "attachment_" the first attachment is saved to "attachment_0"
     * @displayOptions.hide { format: ["full", "metadata", "minimal", "raw"] }
     * @default attachment_
     */
    dataPropertyAttachmentsPrefixName?: string | Expression<string> | PlaceholderValue;
    /** The format to return the message in
     * @default resolved
     */
    format?: 'full' | 'metadata' | 'minimal' | 'raw' | 'resolved' | Expression<string>;
  };
};

export type GmailV1DraftGetNode = {
  type: 'n8n-nodes-base.gmail';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GmailV1DraftGetParams>;
};
