// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/dropcontact/v1.schema.js
/**
 * Dropcontact Node - Version 1 - Zod Validation Schemas
 *
 * These schemas validate node configuration at runtime.
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  // Parameters schema
  const parametersSchema = z.object({
    resource: z.union([z.literal('contact')]).optional(),
    operation: z.union([z.literal('enrich'), z.literal('fetchRequest')]).optional(),
    requestId: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"resource":["contact"],"operation":["fetchRequest"]}}, defaults: {"resource":"contact","operation":"enrich"} }),
    email: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"resource":["contact"],"operation":["enrich"]}}, defaults: {"resource":"contact","operation":"enrich"} }),
    simplify: resolveSchema({ parameters, schema: booleanOrExpression, required: false, displayOptions: {"show":{"resource":["contact"],"operation":["enrich"]}}, defaults: {"resource":"contact","operation":"enrich"} }),
    additionalFields: resolveSchema({ parameters, schema: z.object({ num_siren: stringOrExpression.optional(), siret: stringOrExpression.optional(), company: stringOrExpression.optional(), country: stringOrExpression.optional(), first_name: stringOrExpression.optional(), full_name: stringOrExpression.optional(), last_name: stringOrExpression.optional(), linkedin: stringOrExpression.optional(), phone: stringOrExpression.optional(), website: stringOrExpression.optional() }), required: false, displayOptions: {"show":{"resource":["contact"],"operation":["enrich"]}}, defaults: {"resource":"contact","operation":"enrich"} }),
    options: resolveSchema({ parameters, schema: z.object({ waitTime: numberOrExpression.optional(), siren: booleanOrExpression.optional(), language: z.union([z.literal('en'), z.literal('fr'), expressionSchema]).optional() }), required: false, displayOptions: {"show":{"resource":["contact"],"operation":["enrich"]}}, defaults: {"resource":"contact","operation":"enrich"} }),
  });

  // Return combined config schema
  return z.object({
    parameters: parametersSchema.optional(),
  });
};
