// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/linkedIn/v1/resource_post/operation_create.schema.js
/**
 * LinkedIn Node - Version 1 - Zod Schema
 * Discriminator: resource=post, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('post').default('post'),
      operation: z.literal('create').default('create'),
      authentication: z.union([z.literal('standard'), z.literal('communityManagement'), expressionSchema]).optional(),
      postAs: z.union([z.literal('person'), z.literal('organization'), expressionSchema]).optional(),
      person: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"postAs":["person"]}}, defaults: {"postAs":"person"} }),
      organization: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"postAs":["organization"]}}, defaults: {"postAs":"person"} }),
      text: stringOrExpression.optional(),
      shareMediaCategory: z.union([z.literal('NONE'), z.literal('ARTICLE'), z.literal('IMAGE'), expressionSchema]).optional(),
      binaryPropertyName: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"shareMediaCategory":["IMAGE"]}}, defaults: {"shareMediaCategory":"NONE"} }),
      additionalFields: z.object({ description: stringOrExpression.optional(), originalUrl: stringOrExpression.optional(), thumbnailBinaryPropertyName: stringOrExpression.optional(), title: stringOrExpression.optional(), visibility: z.union([z.literal('CONNECTIONS'), z.literal('PUBLIC'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
