// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/hubspot/v1/resource_company/operation_get_recently_modified.ts
/**
 * HubSpot Node - Version 1
 * Discriminator: resource=company, operation=getRecentlyModified
 */


interface Credentials {
  hubspotApi: CredentialReference;
  hubspotAppToken: CredentialReference;
  hubspotOAuth2Api: CredentialReference;
}

/** Get recently modified companies */
export type HubspotV1CompanyGetRecentlyModifiedParams = {
  resource: 'company';
  operation: 'getRecentlyModified';
  authentication?: 'apiKey' | 'appToken' | 'oAuth2' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 100
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Only return companys created after timestamp x
     */
    since?: string | Expression<string>;
    /** By default, you will only get data for the most recent version of a property in the "versions" data. If you include this parameter, you will get data for all previous versions.
     * @default false
     */
    includePropertyVersions?: boolean | Expression<boolean>;
  };
};

export type HubspotV1CompanyGetRecentlyModifiedNode = {
  type: 'n8n-nodes-base.hubspot';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<HubspotV1CompanyGetRecentlyModifiedParams>;
};
