// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/gitlab/v1/resource_file/operation_get.ts
/**
 * GitLab Node - Version 1
 * Discriminator: resource=file, operation=get
 */


interface Credentials {
  gitlabApi: CredentialReference;
  gitlabOAuth2Api: CredentialReference;
}

/** Get the data of a single issue */
export type GitlabV1FileGetParams = {
  resource: 'file';
  operation: 'get';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * User, group or namespace of the project
 */
    owner?: string | Expression<string> | PlaceholderValue;
/**
 * The name of the project
 * @displayOptions.hide { resource: ["user"], operation: ["getRepositories"] }
 */
    repository?: string | Expression<string> | PlaceholderValue;
/**
 * The file path of the file. Has to contain the full path or leave it empty for root folder.
 * @displayOptions.hide { operation: ["list"] }
 */
    filePath?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to set the data of the file as binary property instead of returning the raw API response
 * @default true
 */
    asBinaryProperty?: boolean | Expression<boolean>;
/**
 * Put Output File in Field
 * @hint The name of the output binary field to put the file in
 * @displayOptions.show { asBinaryProperty: [true] }
 * @default data
 */
    binaryPropertyName?: string | Expression<string> | PlaceholderValue;
/**
 * Additional fields to add
 * @default {}
 */
    additionalParameters?: {
    /** The name of the commit/branch/tag. Default: the repository’s default branch (usually main).
     */
    reference?: string | Expression<string> | PlaceholderValue;
  };
};

export type GitlabV1FileGetOutput = {
  blob_id?: string;
  commit_id?: string;
  content?: string;
  content_sha256?: string;
  encoding?: string;
  execute_filemode?: boolean;
  file_name?: string;
  file_path?: string;
  id?: string;
  last_commit_id?: string;
  mode?: string;
  name?: string;
  path?: string;
  ref?: string;
  size?: number;
  type?: string;
};

export type GitlabV1FileGetNode = {
  type: 'n8n-nodes-base.gitlab';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GitlabV1FileGetParams>;
  output?: Items<GitlabV1FileGetOutput>;
};
