// @n8n/n8n-nodes-langchain/dist/node-definitions/nodes/n8n-nodes-langchain/anthropic/v1/resource_text/operation_message.schema.js
/**
 * Anthropic Node - Version 1 - Zod Schema
 * Discriminator: resource=text, operation=message
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema, toolInstanceSchema }) {

  // Static subnode schema
  const subnodesSchema = z.object({
    tools: z.array(toolInstanceSchema).optional(),
  }).strict();

  return z.object({
    parameters: z.object({
      resource: z.literal('text').default('text'),
      operation: z.literal('message'),
      modelId: z.union([z.object({ __rl: z.literal(true), mode: z.union([z.literal('list'), z.literal('id')]), value: z.union([z.string(), z.number()]), cachedResultName: z.string().optional(), cachedResultUrl: z.string().optional() }), expressionSchema]).optional(),
      messages: z.object({ values: z.array(z.object({ content: stringOrExpression.optional(), role: z.union([z.literal('user'), z.literal('assistant'), expressionSchema]).optional() })).optional() }).optional(),
      addAttachments: booleanOrExpression.optional(),
      attachmentsInputType: resolveSchema({ parameters, schema: z.union([z.literal('url'), z.literal('binary'), expressionSchema]), required: false, displayOptions: {"show":{"addAttachments":[true]}}, defaults: {"addAttachments":false} }),
      attachmentsUrls: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"addAttachments":[true],"attachmentsInputType":["url"]}}, defaults: {"addAttachments":false,"attachmentsInputType":"url"} }),
      binaryPropertyName: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"addAttachments":[true],"attachmentsInputType":["binary"]}}, defaults: {"addAttachments":false,"attachmentsInputType":"url"} }),
      simplify: booleanOrExpression.optional(),
      options: z.object({ includeMergedResponse: booleanOrExpression.optional(), system: stringOrExpression.optional(), codeExecution: booleanOrExpression.optional(), webSearch: booleanOrExpression.optional(), maxUses: numberOrExpression.optional(), allowedDomains: stringOrExpression.optional(), blockedDomains: stringOrExpression.optional(), maxTokens: numberOrExpression.optional(), temperature: numberOrExpression.optional(), topP: numberOrExpression.optional(), topK: numberOrExpression.optional(), maxToolsIterations: numberOrExpression.optional() }).optional(),
    }).optional(),
    subnodes: subnodesSchema.optional(),
  });
};
