// @n8n/db/dist/migrations/sqlite/1764276827837-AddCreatorIdToProjectTable.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddCreatorIdToProjectTable1764276827837 = void 0;
const table = {
    project: 'project',
    projectRelation: 'project_relation',
};
const FOREIGN_KEY_NAME = 'projects_creatorId_foreign';
class AddCreatorIdToProjectTable1764276827837 {
    constructor() {
        this.withFKsDisabled = true;
    }
    async up({ escape, schemaBuilder: { addColumns, addForeignKey, column }, queryRunner, }) {
        await addColumns(table.project, [
            column('creatorId').uuid.comment('ID of the user who created the project'),
        ]);
        await addForeignKey(table.project, 'creatorId', ['user', 'id'], FOREIGN_KEY_NAME, 'SET NULL');
        await queryRunner.query(`
			UPDATE ${escape.tableName(table.project)} AS project
			SET ${escape.columnName('creatorId')} = (
				SELECT pr.${escape.columnName('userId')}
				FROM ${escape.tableName(table.projectRelation)} AS pr
				WHERE pr.${escape.columnName('projectId')} = project.${escape.columnName('id')}
					AND pr.${escape.columnName('role')} = 'project:personalOwner'
				LIMIT 1
			)
			WHERE project.${escape.columnName('type')} = 'personal'
				AND project.${escape.columnName('creatorId')} IS NULL;`);
    }
    async down({ schemaBuilder: { dropColumns } }) {
        await dropColumns(table.project, ['creatorId']);
    }
}
exports.AddCreatorIdToProjectTable1764276827837 = AddCreatorIdToProjectTable1764276827837;
