// @n8n/n8n-nodes-langchain/dist/nodes/vendors/OpenAi/v2/actions/image/analyze.operation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.description = void 0;
exports.execute = execute;
const n8n_workflow_1 = require("n8n-workflow");
const transport_1 = require("../../../transport");
const descriptions_1 = require("../descriptions");
const binary_data_1 = require("../../../helpers/binary-data");
const properties = [
    {
        ...(0, descriptions_1.modelRLC)('imageModelSearch'),
    },
    {
        displayName: 'Text Input',
        name: 'text',
        type: 'string',
        placeholder: "e.g. What's in this image?",
        default: "What's in this image?",
        typeOptions: {
            rows: 2,
        },
    },
    {
        displayName: 'Input Type',
        name: 'inputType',
        type: 'options',
        default: 'url',
        options: [
            {
                name: 'Image URL(s)',
                value: 'url',
            },
            {
                name: 'Binary File(s)',
                value: 'base64',
            },
        ],
    },
    {
        displayName: 'URL(s)',
        name: 'imageUrls',
        type: 'string',
        placeholder: 'e.g. https://example.com/image.jpeg',
        description: 'URL(s) of the image(s) to analyze, multiple URLs can be added separated by comma',
        default: '',
        displayOptions: {
            show: {
                inputType: ['url'],
            },
        },
    },
    {
        displayName: 'Input Data Field Name',
        name: 'binaryPropertyName',
        type: 'string',
        default: 'data',
        placeholder: 'e.g. data',
        hint: 'The name of the input field containing the binary file data to be processed',
        description: 'Name of the binary property which contains the image(s)',
        displayOptions: {
            show: {
                inputType: ['base64'],
            },
        },
    },
    {
        displayName: 'Simplify Output',
        name: 'simplify',
        type: 'boolean',
        default: true,
        description: 'Whether to simplify the response or not',
    },
    {
        displayName: 'Options',
        name: 'options',
        placeholder: 'Add Option',
        type: 'collection',
        default: {},
        options: [
            {
                displayName: 'Detail',
                name: 'detail',
                type: 'options',
                default: 'auto',
                options: [
                    {
                        name: 'Auto',
                        value: 'auto',
                        description: 'Model will look at the image input size and decide if it should use the low or high setting',
                    },
                    {
                        name: 'Low',
                        value: 'low',
                        description: 'Return faster responses and consume fewer tokens',
                    },
                    {
                        name: 'High',
                        value: 'high',
                        description: 'Return more detailed responses, consumes more tokens',
                    },
                ],
            },
            {
                displayName: 'Length of Description (Max Tokens)',
                description: 'Fewer tokens will result in shorter, less detailed image description',
                name: 'maxTokens',
                type: 'number',
                default: 300,
                typeOptions: {
                    minValue: 1,
                },
            },
        ],
    },
];
const displayOptions = {
    show: {
        operation: ['analyze'],
        resource: ['image'],
    },
};
exports.description = (0, n8n_workflow_1.updateDisplayOptions)(displayOptions, properties);
async function execute(i) {
    const model = this.getNodeParameter('modelId', i, 'gpt-4o', { extractValue: true });
    const text = this.getNodeParameter('text', i, '');
    const inputType = this.getNodeParameter('inputType', i);
    const options = this.getNodeParameter('options', i, {});
    const content = [
        {
            type: 'input_text',
            text,
        },
    ];
    const detail = options.detail || 'auto';
    if (inputType === 'url') {
        const imageUrls = this.getNodeParameter('imageUrls', i)
            .split(',')
            .map((url) => url.trim());
        for (const url of imageUrls) {
            content.push({
                type: 'input_image',
                detail,
                image_url: url,
            });
        }
    }
    else {
        const binaryPropertyName = this.getNodeParameter('binaryPropertyName', i)
            .split(',')
            .map((propertyName) => propertyName.trim());
        for (const propertyName of binaryPropertyName) {
            const { fileContent, contentType } = await (0, binary_data_1.getBinaryDataFile)(this, i, propertyName);
            const buffer = await this.helpers.binaryToBuffer(fileContent);
            const fileBase64 = buffer.toString('base64');
            content.push({
                type: 'input_image',
                detail,
                image_url: `data:${contentType};base64,${fileBase64}`,
            });
        }
    }
    const body = {
        model,
        input: [
            {
                role: 'user',
                content,
            },
        ],
        max_output_tokens: options.maxTokens || 300,
    };
    const response = (await transport_1.apiRequest.call(this, 'POST', '/responses', {
        body,
    }));
    const simplify = this.getNodeParameter('simplify', i);
    if (simplify) {
        return [
            {
                json: response.output,
                pairedItem: { item: i },
            },
        ];
    }
    return [
        {
            json: response,
            pairedItem: { item: i },
        },
    ];
}
