// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/zendesk/v1/resource_organization/operation_create.schema.js
/**
 * Zendesk Node - Version 1 - Zod Schema
 * Discriminator: resource=organization, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('organization'),
      operation: z.literal('create').default('create'),
      authentication: z.union([z.literal('apiToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      name: stringOrExpression.optional(),
      additionalFields: z.object({ details: stringOrExpression.optional(), domain_names: stringOrExpression.optional(), notes: stringOrExpression.optional(), organizationFieldsUi: z.unknown().optional(), tags: z.array(z.string()).optional() }).optional(),
    }).optional(),
  });
};
