// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mailerLite/v1/resource_subscriber/operation_update.ts
/**
 * MailerLite Node - Version 1
 * Discriminator: resource=subscriber, operation=update
 */


interface Credentials {
  mailerLiteApi: CredentialReference;
}

/** Update an subscriber */
export type MailerLiteV1SubscriberUpdateParams = {
  resource: 'subscriber';
  operation: 'update';
/**
 * Email of subscriber
 */
    subscriberId?: string | Expression<string> | PlaceholderValue;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** Filter by custom fields
     * @default {}
     */
    customFieldsUi?: {
        /** Custom Field
     */
    customFieldsValues?: Array<{
      /** The ID of the field to add custom field to. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
       */
      fieldId?: string | Expression<string>;
      /** The value to set on custom field
       */
      value?: string | Expression<string> | PlaceholderValue;
    }>;
  };
    /** Name
     */
    name?: string | Expression<string> | PlaceholderValue;
    /** Whether it is needed to resend autoresponders
     * @default false
     */
    resend_autoresponders?: boolean | Expression<boolean>;
    /** Type
     */
    type?: 'active' | 'unsubscribed' | 'unconfirmed' | Expression<string>;
  };
};

export type MailerLiteV1SubscriberUpdateNode = {
  type: 'n8n-nodes-base.mailerLite';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MailerLiteV1SubscriberUpdateParams>;
};
