// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mailerLite/v1/resource_subscriber/operation_create.schema.js
/**
 * MailerLite Node - Version 1 - Zod Schema
 * Discriminator: resource=subscriber, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('subscriber').default('subscriber'),
      operation: z.literal('create').default('create'),
      email: stringOrExpression.optional(),
      additionalFields: z.object({ confirmation_timestamp: stringOrExpression.optional(), confirmation_ip: stringOrExpression.optional(), customFieldsUi: z.unknown().optional(), name: stringOrExpression.optional(), resubscribe: booleanOrExpression.optional(), signup_ip: stringOrExpression.optional(), signup_timestamp: stringOrExpression.optional(), type: z.union([z.literal('active'), z.literal('unsubscribed'), z.literal('unconfirmed'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
