// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mailerLite/v1/resource_subscriber/operation_get_all.ts
/**
 * MailerLite Node - Version 1
 * Discriminator: resource=subscriber, operation=getAll
 */


interface Credentials {
  mailerLiteApi: CredentialReference;
}

/** Get many subscribers */
export type MailerLiteV1SubscriberGetAllParams = {
  resource: 'subscriber';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Type
     */
    type?: 'active' | 'unsubscribed' | 'unconfirmed' | Expression<string>;
  };
};

export type MailerLiteV1SubscriberGetAllOutput = {
  id: number;
  name: string;
  email: string;
  sent: number;
  opened: number;
  clicked: number;
  type: string;
  country_id: string;
  signup_ip: null;
  signup_timestamp: null;
  confirmation_ip: null;
  confirmation_timestamp: null;
  fields: Array<{
    key: string;
    value: string;
    type: string;
  }>;
  date_subscribe: string;
  date_created: string;
  date_updated: string;
};

export type MailerLiteV1SubscriberGetAllNode = {
  type: 'n8n-nodes-base.mailerLite';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MailerLiteV1SubscriberGetAllParams>;
  output?: Items<MailerLiteV1SubscriberGetAllOutput>;
};
