// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/slack/v2/resource_user_group/operation_update_users.ts
/**
 * Slack Node - Version 2
 * Discriminator: resource=userGroup, operation=updateUsers
 */


interface Credentials {
  slackApi: CredentialReference;
  slackOAuth2Api: CredentialReference;
}

export type SlackV2UserGroupUpdateUsersParams = {
  resource: 'userGroup';
  operation: 'updateUsers';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * The encoded ID of the User Group to update
 */
    userGroupId?: string | Expression<string> | PlaceholderValue;
/**
 * Users to add to the User Group. Existing users will be preserved. Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 * @default []
 */
    users?: string[];
/**
 * Options
 * @default {}
 */
    options?: {
    /** Whether to include the number of users in the User Group
     * @default true
     */
    include_count?: boolean | Expression<boolean>;
  };
};

export type SlackV2UserGroupUpdateUsersNode = {
  type: 'n8n-nodes-base.slack';
  version: 2;
  credentials?: Credentials;
  config: NodeConfig<SlackV2UserGroupUpdateUsersParams>;
};
