// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/github/v1/resource_user/operation_get_repositories.ts
/**
 * GitHub Node - Version 1
 * Discriminator: resource=user, operation=getRepositories
 */


interface Credentials {
  githubApi: CredentialReference;
  githubOAuth2Api: CredentialReference;
}

/** Returns all repositories of an organization */
export type GithubV1UserGetRepositoriesParams = {
  resource: 'user';
  operation: 'getRepositories';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Repository Owner
 * @displayOptions.hide { operation: ["invite", "getUserIssues"] }
 * @default {"mode":"list","value":""}
 */
    owner?: { __rl: true; mode: 'list' | 'url' | 'name'; value: string; cachedResultName?: string };
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
};

export type GithubV1UserGetRepositoriesOutput = {
  allow_forking?: boolean;
  archive_url?: string;
  archived?: boolean;
  assignees_url?: string;
  blobs_url?: string;
  branches_url?: string;
  clone_url?: string;
  collaborators_url?: string;
  comments_url?: string;
  commits_url?: string;
  compare_url?: string;
  contents_url?: string;
  contributors_url?: string;
  created_at?: string;
  default_branch?: string;
  deployments_url?: string;
  disabled?: boolean;
  downloads_url?: string;
  events_url?: string;
  fork?: boolean;
  forks?: number;
  forks_count?: number;
  forks_url?: string;
  full_name?: string;
  git_commits_url?: string;
  git_refs_url?: string;
  git_tags_url?: string;
  git_url?: string;
  has_discussions?: boolean;
  has_downloads?: boolean;
  has_issues?: boolean;
  has_pages?: boolean;
  has_projects?: boolean;
  has_wiki?: boolean;
  hooks_url?: string;
  html_url?: string;
  id?: number;
  is_template?: boolean;
  issue_comment_url?: string;
  issue_events_url?: string;
  issues_url?: string;
  keys_url?: string;
  labels_url?: string;
  languages_url?: string;
  merges_url?: string;
  milestones_url?: string;
  mirror_url?: null;
  name?: string;
  node_id?: string;
  notifications_url?: string;
  open_issues?: number;
  open_issues_count?: number;
  owner?: {
    avatar_url?: string;
    events_url?: string;
    followers_url?: string;
    following_url?: string;
    gists_url?: string;
    gravatar_id?: string;
    html_url?: string;
    id?: number;
    login?: string;
    node_id?: string;
    organizations_url?: string;
    received_events_url?: string;
    repos_url?: string;
    site_admin?: boolean;
    starred_url?: string;
    subscriptions_url?: string;
    type?: string;
    url?: string;
    user_view_type?: string;
  };
  permissions?: {
    admin?: boolean;
    maintain?: boolean;
    pull?: boolean;
    push?: boolean;
    triage?: boolean;
  };
  private?: boolean;
  pulls_url?: string;
  pushed_at?: string;
  releases_url?: string;
  size?: number;
  ssh_url?: string;
  stargazers_count?: number;
  stargazers_url?: string;
  statuses_url?: string;
  subscribers_url?: string;
  subscription_url?: string;
  svn_url?: string;
  tags_url?: string;
  teams_url?: string;
  topics?: Array<string>;
  trees_url?: string;
  updated_at?: string;
  url?: string;
  visibility?: string;
  watchers?: number;
  watchers_count?: number;
  web_commit_signoff_required?: boolean;
};

export type GithubV1UserGetRepositoriesNode = {
  type: 'n8n-nodes-base.github';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GithubV1UserGetRepositoriesParams>;
  output?: Items<GithubV1UserGetRepositoriesOutput>;
};
