// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/freshservice/v1/resource_change/operation_get_all.ts
/**
 * Freshservice Node - Version 1
 * Discriminator: resource=change, operation=getAll
 */


interface Credentials {
  freshserviceApi: CredentialReference;
}

/** Retrieve many agents */
export type FreshserviceV1ChangeGetAllParams = {
  resource: 'change';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Predefined Filters
     * @default my_open
     */
    filter?: 'closed' | 'my_open' | 'release_requested' | 'requester_id' | 'unassigned' | Expression<string>;
    /** Sort Order
     * @default asc
     */
    sort_by?: 'asc' | 'desc' | Expression<string>;
    /** Updated Since
     */
    updated_since?: string | Expression<string>;
  };
};

export type FreshserviceV1ChangeGetAllNode = {
  type: 'n8n-nodes-base.freshservice';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<FreshserviceV1ChangeGetAllParams>;
};
