// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/freshservice/v1/resource_change/operation_get_all.schema.js
/**
 * Freshservice Node - Version 1 - Zod Schema
 * Discriminator: resource=change, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('change'),
      operation: z.literal('getAll'),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filters: z.object({ filter: z.union([z.literal('closed'), z.literal('my_open'), z.literal('release_requested'), z.literal('requester_id'), z.literal('unassigned'), expressionSchema]).optional(), sort_by: z.union([z.literal('asc'), z.literal('desc'), expressionSchema]).optional(), updated_since: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
