// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/freshservice/v1/resource_change/operation_update.schema.js
/**
 * Freshservice Node - Version 1 - Zod Schema
 * Discriminator: resource=change, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('change'),
      operation: z.literal('update'),
      changeId: stringOrExpression.optional(),
      updateFields: z.object({ agent_id: stringOrExpression.optional(), change_type: z.union([z.literal(1), z.literal(2), z.literal(3), z.literal(4), expressionSchema]).optional(), department_id: stringOrExpression.optional(), description: stringOrExpression.optional(), group_id: stringOrExpression.optional(), impact: z.union([z.literal(1), z.literal(2), z.literal(3), expressionSchema]).optional(), priority: z.union([z.literal(1), z.literal(2), z.literal(3), z.literal(4), expressionSchema]).optional(), requester_id: stringOrExpression.optional(), risk: z.union([z.literal(1), z.literal(2), z.literal(3), z.literal(4), expressionSchema]).optional(), status: z.union([z.literal(1), z.literal(2), z.literal(3), z.literal(4), z.literal(5), z.literal(6), expressionSchema]).optional(), subject: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
