// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/freshservice/v1/resource_change/operation_update.ts
/**
 * Freshservice Node - Version 1
 * Discriminator: resource=change, operation=update
 */


interface Credentials {
  freshserviceApi: CredentialReference;
}

/** Update an agent */
export type FreshserviceV1ChangeUpdateParams = {
  resource: 'change';
  operation: 'update';
/**
 * ID of the change to update
 */
    changeId?: string | Expression<string> | PlaceholderValue;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
    /** ID of the agent to whom the change is assigned. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    agent_id?: string | Expression<string>;
    /** Change Type
     * @default 1
     */
    change_type?: 1 | 2 | 3 | 4 | Expression<number>;
    /** ID of the department requesting the change. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    department_id?: string | Expression<string>;
    /** HTML supported
     */
    description?: string | Expression<string> | PlaceholderValue;
    /** ID of the agent group to which the change is assigned. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    group_id?: string | Expression<string>;
    /** Impact of the change
     * @default 1
     */
    impact?: 1 | 2 | 3 | Expression<number>;
    /** Priority
     * @default 1
     */
    priority?: 1 | 2 | 3 | 4 | Expression<number>;
    /** ID of the requester of the change. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     */
    requester_id?: string | Expression<string>;
    /** Risk
     * @default 1
     */
    risk?: 1 | 2 | 3 | 4 | Expression<number>;
    /** Status
     * @default 1
     */
    status?: 1 | 2 | 3 | 4 | 5 | 6 | Expression<number>;
    /** Subject
     */
    subject?: string | Expression<string> | PlaceholderValue;
  };
};

export type FreshserviceV1ChangeUpdateNode = {
  type: 'n8n-nodes-base.freshservice';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<FreshserviceV1ChangeUpdateParams>;
};
