// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/microsoftTeams/v11/resource_task/operation_get_all.schema.js
/**
 * Microsoft Teams Node - Version 1.1 - Zod Schema
 * Discriminator: resource=task, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('task'),
      operation: z.literal('getAll'),
      groupSource: z.union([z.literal('all'), z.literal('mine'), expressionSchema]).optional(),
      tasksFor: z.union([z.literal('member'), z.literal('plan'), expressionSchema]).optional(),
      groupId: stringOrExpression.optional(),
      memberId: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"tasksFor":["member"]}}, defaults: {"tasksFor":"member"} }),
      planId: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"tasksFor":["plan"]}}, defaults: {"tasksFor":"member"} }),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
    }).optional(),
  });
};
