// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/gitlab/v1/resource_repository/operation_get_issues.ts
/**
 * GitLab Node - Version 1
 * Discriminator: resource=repository, operation=getIssues
 */


interface Credentials {
  gitlabApi: CredentialReference;
  gitlabOAuth2Api: CredentialReference;
}

/** Returns issues of a repository */
export type GitlabV1RepositoryGetIssuesParams = {
  resource: 'repository';
  operation: 'getIssues';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * User, group or namespace of the project
 */
    owner?: string | Expression<string> | PlaceholderValue;
/**
 * The name of the project
 * @displayOptions.hide { resource: ["user"], operation: ["getRepositories"] }
 */
    repository?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 20
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    getRepositoryIssuesFilters?: {
    /** Return only issues which are assigned to a specific user
     */
    assignee_username?: string | Expression<string> | PlaceholderValue;
    /** Return only issues which were created by a specific user
     */
    author_username?: string | Expression<string> | PlaceholderValue;
    /** Search issues against their title and description
     */
    search?: string | Expression<string> | PlaceholderValue;
    /** Return only issues with the given labels. Multiple lables can be separated by comma.
     */
    labels?: string | Expression<string> | PlaceholderValue;
    /** Return only issues updated at or after this time
     */
    updated_after?: string | Expression<string>;
    /** The state to filter by
     * @default opened
     */
    state?: '' | 'closed' | 'opened' | Expression<string>;
    /** The order the issues should be returned in
     * @default created_at
     */
    order_by?: 'created_at' | 'updated_at' | 'priority' | Expression<string>;
    /** The sort order
     * @default desc
     */
    sort?: 'asc' | 'desc' | Expression<string>;
  };
};

export type GitlabV1RepositoryGetIssuesOutput = {
  _links?: {
    award_emoji?: string;
    closed_as_duplicate_of?: null;
    notes?: string;
    project?: string;
    self?: string;
  };
  assignees?: Array<{
    avatar_url?: string;
    id?: number;
    name?: string;
    state?: string;
    username?: string;
    web_url?: string;
  }>;
  author?: {
    avatar_url?: string;
    id?: number;
    locked?: boolean;
    name?: string;
    state?: string;
    username?: string;
    web_url?: string;
  };
  confidential?: boolean;
  created_at?: string;
  discussion_locked?: null;
  downvotes?: number;
  has_tasks?: boolean;
  id?: number;
  iid?: number;
  imported?: boolean;
  imported_from?: string;
  issue_type?: string;
  labels?: Array<string>;
  merge_requests_count?: number;
  moved_to_id?: null;
  project_id?: number;
  references?: {
    full?: string;
    relative?: string;
    short?: string;
  };
  service_desk_reply_to?: null;
  severity?: string;
  state?: string;
  task_completion_status?: {
    completed_count?: number;
    count?: number;
  };
  task_status?: string;
  time_stats?: {
    human_total_time_spent?: null;
    time_estimate?: number;
    total_time_spent?: number;
  };
  title?: string;
  type?: string;
  updated_at?: string;
  upvotes?: number;
  user_notes_count?: number;
  web_url?: string;
};

export type GitlabV1RepositoryGetIssuesNode = {
  type: 'n8n-nodes-base.gitlab';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GitlabV1RepositoryGetIssuesParams>;
  output?: Items<GitlabV1RepositoryGetIssuesOutput>;
};
