// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/gitlab/v1/resource_repository/operation_get_issues.schema.js
/**
 * GitLab Node - Version 1 - Zod Schema
 * Discriminator: resource=repository, operation=getIssues
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('repository'),
      operation: z.literal('getIssues'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      owner: stringOrExpression.optional(),
      repository: stringOrExpression.optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      getRepositoryIssuesFilters: z.object({ assignee_username: stringOrExpression.optional(), author_username: stringOrExpression.optional(), search: stringOrExpression.optional(), labels: stringOrExpression.optional(), updated_after: stringOrExpression.optional(), state: z.union([z.literal(''), z.literal('closed'), z.literal('opened'), expressionSchema]).optional(), order_by: z.union([z.literal('created_at'), z.literal('updated_at'), z.literal('priority'), expressionSchema]).optional(), sort: z.union([z.literal('asc'), z.literal('desc'), expressionSchema]).optional() }).optional(),
    }).optional(),
  });
};
