// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/elasticSecurity/v1/resource_connector/operation_create.schema.js
/**
 * Elastic Security Node - Version 1 - Zod Schema
 * Discriminator: resource=connector, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('connector'),
      operation: z.literal('create').default('create'),
      name: stringOrExpression.optional(),
      connectorType: z.union([z.literal('.resilient'), z.literal('.jira'), z.literal('.servicenow'), expressionSchema]).optional(),
      apiUrl: stringOrExpression.optional(),
      email: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"connectorType":[".jira"]}}, defaults: {"connectorType":".jira"} }),
      apiToken: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"connectorType":[".jira"]}}, defaults: {"connectorType":".jira"} }),
      projectKey: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"connectorType":[".jira"]}}, defaults: {"connectorType":".jira"} }),
      username: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"connectorType":[".servicenow"]}}, defaults: {"connectorType":".jira"} }),
      password: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"connectorType":[".servicenow"]}}, defaults: {"connectorType":".jira"} }),
      apiKeyId: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"connectorType":[".resilient"]}}, defaults: {"connectorType":".jira"} }),
      apiKeySecret: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"connectorType":[".resilient"]}}, defaults: {"connectorType":".jira"} }),
      orgId: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"connectorType":[".resilient"]}}, defaults: {"connectorType":".jira"} }),
    }).optional(),
  });
};
