// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/twist/v1/resource_channel/operation_create.ts
/**
 * Twist Node - Version 1
 * Discriminator: resource=channel, operation=create
 */


interface Credentials {
  twistOAuth2Api: CredentialReference;
}

/** Initiates a public or private channel-based conversation */
export type TwistV1ChannelCreateParams = {
  resource: 'channel';
  operation: 'create';
/**
 * The ID of the workspace. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    workspaceId?: string | Expression<string>;
/**
 * The name of the channel
 */
    name?: string | Expression<string> | PlaceholderValue;
/**
 * Additional Fields
 * @default {}
 */
    additionalFields?: {
    /** The color of the channel
     * @default 0
     */
    color?: 6 | 1 | 4 | 0 | 7 | 9 | 5 | 11 | 8 | 3 | 2 | 10 | Expression<number>;
    /** The description of the channel
     */
    description?: string | Expression<string> | PlaceholderValue;
    /** Whether the channel will be marked as public
     * @default false
     */
    public?: boolean | Expression<boolean>;
    /** The temporary ID of the channel. It needs to be a negative number.
     * @default -1
     */
    temp_id?: number | Expression<number>;
    /** The users that will participate in the channel. Choose from the list, or specify IDs using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
     * @default []
     */
    user_ids?: string[];
  };
};

export type TwistV1ChannelCreateNode = {
  type: 'n8n-nodes-base.twist';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<TwistV1ChannelCreateParams>;
};
