// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/twist/v1/resource_channel/operation_get_all.ts
/**
 * Twist Node - Version 1
 * Discriminator: resource=channel, operation=getAll
 */


interface Credentials {
  twistOAuth2Api: CredentialReference;
}

/** Get many channels */
export type TwistV1ChannelGetAllParams = {
  resource: 'channel';
  operation: 'getAll';
/**
 * The ID of the workspace. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    workspaceId?: string | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Whether only archived conversations are returned
     * @default false
     */
    archived?: boolean | Expression<boolean>;
  };
};

export type TwistV1ChannelGetAllNode = {
  type: 'n8n-nodes-base.twist';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<TwistV1ChannelGetAllParams>;
};
