// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/twist/v1/resource_channel/operation_create.schema.js
/**
 * Twist Node - Version 1 - Zod Schema
 * Discriminator: resource=channel, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('channel'),
      operation: z.literal('create').default('create'),
      workspaceId: stringOrExpression.optional(),
      name: stringOrExpression.optional(),
      additionalFields: z.object({ color: z.union([z.literal(6), z.literal(1), z.literal(4), z.literal(0), z.literal(7), z.literal(9), z.literal(5), z.literal(11), z.literal(8), z.literal(3), z.literal(2), z.literal(10), expressionSchema]).optional(), description: stringOrExpression.optional(), public: booleanOrExpression.optional(), temp_id: numberOrExpression.optional(), user_ids: z.array(z.string()).optional() }).optional(),
    }).optional(),
  });
};
