// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/nextCloud/v1/resource_user/operation_update.ts
/**
 * Nextcloud Node - Version 1
 * Discriminator: resource=user, operation=update
 */


interface Credentials {
  nextCloudApi: CredentialReference;
  nextCloudOAuth2Api: CredentialReference;
}

/** Edit attributes related to a user */
export type NextCloudV1UserUpdateParams = {
  resource: 'user';
  operation: 'update';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Username the user will have
 */
    userId?: string | Expression<string> | PlaceholderValue;
/**
 * Update Fields
 * @default {}
 */
    updateFields?: {
        /** Fields
     */
    field?: {
      /** Key of the updated attribute
       * @default email
       */
      key?: 'address' | 'displayname' | 'email' | 'password' | 'twitter' | 'website' | Expression<string>;
      /** Value of the updated attribute
       */
      value?: string | Expression<string> | PlaceholderValue;
    };
  };
};

export type NextCloudV1UserUpdateNode = {
  type: 'n8n-nodes-base.nextCloud';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<NextCloudV1UserUpdateParams>;
};
