// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/nextCloud/v1/resource_user/operation_get_all.ts
/**
 * Nextcloud Node - Version 1
 * Discriminator: resource=user, operation=getAll
 */


interface Credentials {
  nextCloudApi: CredentialReference;
  nextCloudOAuth2Api: CredentialReference;
}

/** Retrieve a list of users */
export type NextCloudV1UserGetAllParams = {
  resource: 'user';
  operation: 'getAll';
  authentication?: 'accessToken' | 'oAuth2' | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Optional search string
     */
    search?: string | Expression<string> | PlaceholderValue;
    /** Optional offset value
     */
    offset?: number | Expression<number>;
  };
};

export type NextCloudV1UserGetAllNode = {
  type: 'n8n-nodes-base.nextCloud';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<NextCloudV1UserGetAllParams>;
};
