// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/nextCloud/v1/resource_user/operation_update.schema.js
/**
 * Nextcloud Node - Version 1 - Zod Schema
 * Discriminator: resource=user, operation=update
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('user'),
      operation: z.literal('update'),
      authentication: z.union([z.literal('accessToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      userId: stringOrExpression.optional(),
      updateFields: z.object({ field: z.object({ key: z.union([z.literal('address'), z.literal('displayname'), z.literal('email'), z.literal('password'), z.literal('twitter'), z.literal('website'), expressionSchema]).optional(), value: stringOrExpression.optional() }).optional() }).optional(),
    }).optional(),
  });
};
