// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleBigQuery/v1/resource_record/operation_get_all.ts
/**
 * Google BigQuery Node - Version 1
 * Discriminator: resource=record, operation=getAll
 */


interface Credentials {
  googleApi: CredentialReference;
  googleBigQueryOAuth2Api: CredentialReference;
}

/** Retrieve many records */
export type GoogleBigQueryV1RecordGetAllParams = {
  resource: 'record';
  operation: 'getAll';
  authentication?: 'oAuth2' | 'serviceAccount';
/**
 * ID of the project to retrieve all rows from. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    projectId?: string | Expression<string>;
/**
 * ID of the dataset to retrieve all rows from. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    datasetId?: string | Expression<string>;
/**
 * ID of the table to retrieve all rows from. Choose from the list, or specify an ID using an &lt;a href="https://docs.n8n.io/code/expressions/"&gt;expression&lt;/a&gt;.
 */
    tableId?: string | Expression<string>;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Whether to return a simplified version of the response instead of the raw data
 * @default true
 */
    simple?: boolean | Expression<boolean>;
/**
 * Options
 * @default {}
 */
    options?: {
    /** Subset of fields to return, supports select into sub fields. Example: &lt;code&gt;selectedFields = "a,e.d.f"&lt;/code&gt;
     */
    selectedFields?: string | Expression<string> | PlaceholderValue;
  };
};

export type GoogleBigQueryV1RecordGetAllNode = {
  type: 'n8n-nodes-base.googleBigQuery';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<GoogleBigQueryV1RecordGetAllParams>;
};
