// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/googleBigQuery/v1/resource_record/operation_create.schema.js
/**
 * Google BigQuery Node - Version 1 - Zod Schema
 * Discriminator: resource=record, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('record').default('record'),
      operation: z.literal('create').default('create'),
      authentication: z.union([z.literal('oAuth2'), z.literal('serviceAccount')]).optional(),
      projectId: stringOrExpression.optional(),
      datasetId: stringOrExpression.optional(),
      tableId: stringOrExpression.optional(),
      columns: stringOrExpression.optional(),
      options: z.object({ ignoreUnknownValues: booleanOrExpression.optional(), skipInvalidRows: booleanOrExpression.optional(), templateSuffix: stringOrExpression.optional(), traceId: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
