// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/freshworksCrm/v1/resource_appointment/operation_get_all.ts
/**
 * Freshworks CRM Node - Version 1
 * Discriminator: resource=appointment, operation=getAll
 */


interface Credentials {
  freshworksCrmApi: CredentialReference;
}

/** Retrieve many accounts */
export type FreshworksCrmV1AppointmentGetAllParams = {
  resource: 'appointment';
  operation: 'getAll';
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Include
     * @default creater
     */
    include?: 'appointment_attendees' | 'creater' | 'targetable' | Expression<string>;
    /** Time
     * @default upcoming
     */
    filter?: 'past' | 'upcoming' | Expression<string>;
  };
};

export type FreshworksCrmV1AppointmentGetAllNode = {
  type: 'n8n-nodes-base.freshworksCrm';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<FreshworksCrmV1AppointmentGetAllParams>;
};
