// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/onfleet/v1/resource_hub/operation_create.schema.js
/**
 * Onfleet Node - Version 1 - Zod Schema
 * Discriminator: resource=hub, operation=create
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('hub'),
      operation: z.literal('create'),
      name: stringOrExpression.optional(),
      destination: z.object({ destinationProperties: z.object({ unparsed: booleanOrExpression.optional(), address: stringOrExpression.optional(), addressNumber: stringOrExpression.optional(), addressStreet: stringOrExpression.optional(), addressCity: stringOrExpression.optional(), addressState: stringOrExpression.optional(), addressCountry: stringOrExpression.optional(), addressPostalCode: stringOrExpression.optional(), addressName: stringOrExpression.optional(), addressApartment: stringOrExpression.optional(), addressNotes: stringOrExpression.optional() }).optional() }).optional(),
      additionalFields: z.object({ teams: z.array(z.string()).optional() }).optional(),
    }).optional(),
  });
};
