// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/onfleet/v1/resource_task/operation_clone.schema.js
/**
 * Onfleet Node - Version 1 - Zod Schema
 * Discriminator: resource=task, operation=clone
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('task').default('task'),
      operation: z.literal('clone'),
      id: stringOrExpression.optional(),
      overrideFields: z.object({ completeAfter: stringOrExpression.optional(), completeBefore: stringOrExpression.optional(), includeBarcodes: booleanOrExpression.optional(), includeDependencies: booleanOrExpression.optional(), includeMetadata: booleanOrExpression.optional(), notes: stringOrExpression.optional(), pickupTask: booleanOrExpression.optional(), serviceTime: numberOrExpression.optional() }).optional(),
    }).optional(),
  });
};
