// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/linear/v11/resource_comment/operation_add_comment.schema.js
/**
 * Linear Node - Version 1.1 - Zod Schema
 * Discriminator: resource=comment, operation=addComment
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('comment'),
      operation: z.literal('addComment').default('addComment'),
      authentication: z.union([z.literal('apiToken'), z.literal('oAuth2'), expressionSchema]).optional(),
      issueId: stringOrExpression.optional(),
      comment: stringOrExpression.optional(),
      additionalFields: z.object({ parentId: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
