// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/marketstack/v1/resource_end_of_day_data/operation_get_all.schema.js
/**
 * Marketstack Node - Version 1 - Zod Schema
 * Discriminator: resource=endOfDayData, operation=getAll
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('endOfDayData').default('endOfDayData'),
      operation: z.literal('getAll').default('getAll'),
      symbols: stringOrExpression.optional(),
      returnAll: booleanOrExpression.optional(),
      limit: resolveSchema({ parameters, schema: numberOrExpression, required: false, displayOptions: {"show":{"returnAll":[false]}}, defaults: {"returnAll":false} }),
      filters: z.object({ exchange: stringOrExpression.optional(), latest: booleanOrExpression.optional(), sort: z.union([z.literal('ASC'), z.literal('DESC'), expressionSchema]).optional(), specificDate: stringOrExpression.optional(), dateFrom: stringOrExpression.optional(), dateTo: stringOrExpression.optional() }).optional(),
    }).optional(),
  });
};
