// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/marketstack/v1/resource_end_of_day_data/operation_get_all.ts
/**
 * Marketstack Node - Version 1
 * Discriminator: resource=endOfDayData, operation=getAll
 */


interface Credentials {
  marketstackApi: CredentialReference;
}

/** Stock market closing data */
export type MarketstackV1EndOfDayDataGetAllParams = {
  resource: 'endOfDayData';
  operation: 'getAll';
/**
 * One or multiple comma-separated stock symbols (tickers) to retrieve, e.g. &lt;code&gt;AAPL&lt;/code&gt; or &lt;code&gt;AAPL,MSFT&lt;/code&gt;
 */
    symbols?: string | Expression<string> | PlaceholderValue;
/**
 * Whether to return all results or only up to a given limit
 * @default false
 */
    returnAll?: boolean | Expression<boolean>;
/**
 * Max number of results to return
 * @displayOptions.show { returnAll: [false] }
 * @default 50
 */
    limit?: number | Expression<number>;
/**
 * Filters
 * @default {}
 */
    filters?: {
    /** Stock exchange to filter results by, specified by &lt;a href="https://en.wikipedia.org/wiki/Market_Identifier_Code"&gt;Market Identifier Code&lt;/a&gt;, e.g. &lt;code&gt;XNAS&lt;/code&gt;
     */
    exchange?: string | Expression<string> | PlaceholderValue;
    /** Whether to fetch the most recent stock market data
     * @default false
     */
    latest?: boolean | Expression<boolean>;
    /** Order to sort results in
     * @default DESC
     */
    sort?: 'ASC' | 'DESC' | Expression<string>;
    /** Date in YYYY-MM-DD format, e.g. &lt;code&gt;2020-01-01&lt;/code&gt;, or in ISO-8601 date format, e.g. &lt;code&gt;2020-05-21T00:00:00+0000&lt;/code&gt;
     */
    specificDate?: string | Expression<string>;
    /** Timeframe start date in YYYY-MM-DD format, e.g. &lt;code&gt;2020-01-01&lt;/code&gt;, or in ISO-8601 date format, e.g. &lt;code&gt;2020-05-21T00:00:00+0000&lt;/code&gt;
     */
    dateFrom?: string | Expression<string>;
    /** Timeframe end date in YYYY-MM-DD format, e.g. &lt;code&gt;2020-01-01&lt;/code&gt;, or in ISO-8601 date format, e.g. &lt;code&gt;2020-05-21T00:00:00+0000&lt;/code&gt;
     */
    dateTo?: string | Expression<string>;
  };
};

export type MarketstackV1EndOfDayDataGetAllOutput = {
  date?: string;
  exchange?: string;
  split_factor?: number;
  symbol?: string;
  volume?: number;
};

export type MarketstackV1EndOfDayDataGetAllNode = {
  type: 'n8n-nodes-base.marketstack';
  version: 1;
  credentials?: Credentials;
  config: NodeConfig<MarketstackV1EndOfDayDataGetAllParams>;
  output?: Items<MarketstackV1EndOfDayDataGetAllOutput>;
};
