// n8n-nodes-base/dist/node-definitions/nodes/n8n-nodes-base/mistralAi/v1/resource_document/operation_extract_text.schema.js
/**
 * Mistral AI Node - Version 1 - Zod Schema
 * Discriminator: resource=document, operation=extractText
 *
 * Use .parse() for strict validation or .safeParse() for error handling.
 *
 * Schema helpers (z, expressionSchema, etc.) are passed as parameters
 * by the schema-validator, not imported from external files.
 *
 * @generated - CommonJS JavaScript for runtime loading
 */

module.exports = function getSchema({ parameters, z, expressionSchema, stringOrExpression, numberOrExpression, booleanOrExpression, resourceLocatorValueSchema, resourceMapperValueSchema, filterValueSchema, assignmentCollectionValueSchema, iDataObjectSchema, resolveSchema }) {

  return z.object({
    parameters: z.object({
      resource: z.literal('document').default('document'),
      operation: z.literal('extractText').default('extractText'),
      model: z.union([z.literal('mistral-ocr-latest'), expressionSchema]).optional(),
      documentType: z.union([z.literal('document_url'), z.literal('image_url'), expressionSchema]).optional(),
      inputType: z.union([z.literal('binary'), z.literal('url'), expressionSchema]).optional(),
      binaryProperty: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"inputType":["binary"]}}, defaults: {"inputType":"binary"} }),
      url: resolveSchema({ parameters, schema: stringOrExpression, required: false, displayOptions: {"show":{"inputType":["url"]}}, defaults: {"inputType":"binary"} }),
      options: z.object({ batch: booleanOrExpression.optional(), batchSize: numberOrExpression.optional(), deleteFiles: booleanOrExpression.optional() }).optional(),
    }).optional(),
  });
};
